{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# @name graph_v3.2_v20190616\n",
    "# @description notebook to build the NGLY1 Deficiency review knowledge graph v3.2\n",
    "# @author Núria Queralt Rosinach\n",
    "# @date 16 June 2019"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Description\n",
    "\n",
    "This is the notebook for the creation of the first review network and derived hypotheses. \n",
    "\n",
    "* Using intermediary variables from workflow objects. In this workflow variables are directly used for the next step. \n",
    "\n",
    "\n",
    "* Review network: From Monarch knowledge graph, we built a network seeded by 8 nodes, retrieving their explicit relationships and all the relationships among all these nodes. Seed nodes:\n",
    "\n",
    "    - 'MONDO:0007739' HD\n",
    "    - 'HGNC:4851' Htt\n",
    "    - 'CHEBI:18248' Iron (not working)\n",
    "    - 'HGNC:18229' Rhes (RASD2)\n",
    "    \n",
    "Possible seed nodes:\n",
    "https://monarchinitiative.org/search/Iron\n",
    "* Connecting paths: query templates."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import transcriptomics, regulation, curation, monarch, graph, neo4jlib, hypothesis, summary, utils\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Edges library\n",
    "### Review edges to integrate into the knowledge graph and prepare them as individual networks\n",
    "\n",
    "#### TRANSCRIPTOMICS NETWORK\n",
    "#### import transcriptomics\n",
    "We retrieved edges from RNA-seq transcriptomics profiles using the `transcriptomics` module:\n",
    "\n",
    "    - Experimental data sets: from Chow et al. paper [pmid:29346549] (NGLY1 deficiency model on fruit fly)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "The function \"read_data()\" is running...\n",
      "\n",
      "* This is the size of the raw expression data structure: (28087, 10)\n",
      "* These are the expression attributes: Index(['Unnamed: 0', 'symbol', 'baseMean', 'HD.mean', 'Control.mean',\n",
      "       'log2FoldChange', 'lfcSE', 'stat', 'pvalue', 'padj'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "           Unnamed: 0 symbol  baseMean    HD.mean  Control.mean  \\\n",
      "0  ENSG00000069011.10  PITX1  5.645675  18.684286      0.323793   \n",
      "\n",
      "   log2FoldChange     lfcSE      stat        pvalue          padj  \n",
      "0        4.769658  0.366367  13.01879  9.567529e-39  2.687232e-34  \n",
      "\n",
      "The raw data is saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/transcriptomics/HD/data/GSE64810_mlhd_DESeq2_diffexp_DESeq2_outlier_trimmed_adjust.csv\n",
      "\n",
      "\n",
      "Finished read_data().\n",
      "\n",
      "CPU times: user 46.1 ms, sys: 1.52 ms, total: 47.6 ms\n",
      "Wall time: 46.2 ms\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "# prepare data to graph schema\n",
    "csv_path = './transcriptomics/GSE64810_mlhd_DESeq2_diffexp_DESeq2_outlier_trimmed_adjust.txt'\n",
    "data = transcriptomics.read_data(csv_path, \"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "The function \"read_data()\" is running...\n",
      "\n",
      "* This is the size of the raw expression data structure: (28087, 10)\n",
      "* These are the expression attributes: Index(['Unnamed: 0', 'symbol', 'baseMean', 'HD.mean', 'Control.mean',\n",
      "       'log2FoldChange', 'lfcSE', 'stat', 'pvalue', 'padj'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "           Unnamed: 0 symbol  baseMean    HD.mean  Control.mean  \\\n",
      "0  ENSG00000069011.10  PITX1  5.645675  18.684286      0.323793   \n",
      "\n",
      "   log2FoldChange     lfcSE      stat        pvalue          padj  \n",
      "0        4.769658  0.366367  13.01879  9.567529e-39  2.687232e-34  \n",
      "\n",
      "The raw data is saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/transcriptomics/HD/data/GSE64810_mlhd_DESeq2_diffexp_DESeq2_outlier_trimmed_adjust.csv\n",
      "\n",
      "\n",
      "Finished read_data().\n",
      "\n",
      "\n",
      "The function \"clean_data()\" is running. Keeping only data with FC > 1.5 and FDR < 5% ...\n",
      "\n",
      "* This is the size of the clean expression data structure: (3209, 6)\n",
      "* These are the clean expression attributes: Index(['EnsembleID', 'symbol', 'log2FoldChange', 'pvalue', 'padj',\n",
      "       'Regulation'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "           EnsembleID symbol  log2FoldChange        pvalue          padj  \\\n",
      "0  ENSG00000069011.10  PITX1        4.769658  9.567529e-39  2.687232e-34   \n",
      "\n",
      "    Regulation  \n",
      "0  Upregulated  \n",
      "\n",
      "The clean data is saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/transcriptomics/HD/out/Transcriptome_human_BA9.csv\n",
      "\n",
      "\n",
      "Finished clean_data().\n",
      "\n",
      "\n",
      "The function \"prepare_data_edges()\" is running...\n",
      "\n",
      "* This is the size of the expression data structure: (3209, 13)\n",
      "* These are the expression attributes: Index(['ensemble_id', 'symbol', 'log2FoldChange', 'pvalue', 'padj',\n",
      "       'regulation', 'source', 'subject_id', 'subject_label', 'property_id',\n",
      "       'property_label', 'reference_id', 'object_id'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "          ensemble_id symbol  log2FoldChange        pvalue          padj  \\\n",
      "0  ENSG00000069011.10  PITX1        4.769658  9.567529e-39  2.687232e-34   \n",
      "\n",
      "    regulation source               subject_id subject_label property_id  \\\n",
      "0  Upregulated   Chow  ensembl:ENSG00000197386           HTT  RO:0002434   \n",
      "\n",
      "   property_label   reference_id                object_id  \n",
      "0  interacts with  PMID:26636579  ensembl:ENSG00000069011  \n",
      "\n",
      "The HD BA9 transcriptomics expression edges are saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/transcriptomics/HD/out/Transcriptome_human_BA9_edges.csv\n",
      "\n",
      "\n",
      "Finished prepare_data_edges().\n",
      "\n",
      "\n",
      "The function \"prepare_rna_edges()\" is running...\n",
      "\n",
      "* This is the size of the edges data structure: (3209, 12)\n",
      "* These are the edges attributes: Index(['subject_id', 'subject_label', 'property_id', 'property_label',\n",
      "       'object_id', 'object_label', 'log2FoldChange', 'pvalue', 'fdr',\n",
      "       'regulation', 'source', 'reference_id'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "                subject_id subject_label property_id  property_label  \\\n",
      "0  ensembl:ENSG00000197386           HTT  RO:0002434  interacts with   \n",
      "\n",
      "                 object_id object_label  log2FoldChange        pvalue  \\\n",
      "0  ensembl:ENSG00000069011        PITX1        4.769658  9.567529e-39   \n",
      "\n",
      "            fdr   regulation source   reference_id  \n",
      "0  2.687232e-34  Upregulated   Chow  PMID:26636579  \n",
      "\n",
      "This data object is not saved.\n",
      "\n",
      "\n",
      "Finished prepare_rna_edges().\n",
      "\n",
      "\n",
      "The function \"build_edges()\" is running...\n",
      "\n",
      "The function \"build_nodes()\" is running...\n"
     ]
    },
    {
     "ename": "AttributeError",
     "evalue": "'list' object has no attribute 'iterrows'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n",
      "\u001b[0;32m~/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/transcriptomics.py\u001b[0m in \u001b[0;36mbuild_nodes\u001b[0;34m(edges)\u001b[0m\n\u001b[1;32m    306\u001b[0m     \u001b[0;31m# build concept dict: {id:symbol}\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    307\u001b[0m     \u001b[0mconcept_dct\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdict\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 308\u001b[0;31m     \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrow\u001b[0m \u001b[0;32min\u001b[0m \u001b[0medges\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0miterrows\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    309\u001b[0m         \u001b[0;31m# node for subject\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    310\u001b[0m         \u001b[0mconcept_dct\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mrow\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'subject_id'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;34m'preflabel'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mrow\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'subject_label'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'list' object has no attribute 'iterrows'"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "# prepare data to graph schema\n",
    "csv_path = './transcriptomics/GSE64810_mlhd_DESeq2_diffexp_DESeq2_outlier_trimmed_adjust.txt'\n",
    "data = transcriptomics.read_data(csv_path, \"\\t\")\n",
    "clean_data = transcriptomics.clean_data(data)\n",
    "data_edges = transcriptomics.prepare_data_edges(clean_data)\n",
    "rna_network = transcriptomics.prepare_rna_edges(data_edges)\n",
    "\n",
    "# TODO:\n",
    "# This function still produces ENSEMBLE ID's even though the code should already return HGNC ones\n",
    "\n",
    "# build network with graph schema\n",
    "rna_edges = transcriptomics.build_edges(rna_network)\n",
    "rna_nodes, rna_dict = transcriptomics.build_nodes(rna_edges)\n",
    "rna_edges = transcriptomics.rework_edges(rna_edges, rna_dict)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- Transcriptomics network is returned as both digital object (`rna_edges`, `rna_nodes`) and CSV files at _**graph/**_ (`rna_edges_version.csv`, `rna_nodes_version.csv`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "type edges: <class 'dict'>\n",
      "type nodes: <class 'list'>\n",
      "\n",
      "len edges: 3210\n",
      "len nodes: 3210\n",
      "\n"
     ]
    },
    {
     "ename": "KeyError",
     "evalue": "0",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-4-39744771912b>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     10\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     11\u001b[0m \u001b[0;31m# print object attribute\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 12\u001b[0;31m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'attribute edges:'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrna_edges\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mkeys\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     13\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'attribute nodes:'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrna_nodes\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mkeys\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 0"
     ]
    }
   ],
   "source": [
    "# print type of objects\n",
    "print('type edges:', type(rna_edges))\n",
    "print('type nodes:', type(rna_nodes))\n",
    "print()\n",
    "\n",
    "# print objects sizes\n",
    "print('len edges:', len(rna_edges))\n",
    "print('len nodes:', len(rna_nodes))\n",
    "print()\n",
    "\n",
    "# print object attribute\n",
    "print('attribute edges:', rna_edges[0].keys())\n",
    "print('attribute nodes:', rna_nodes[0].keys())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### REGULATION NETWORK\n",
    "#### import regulation\n",
    "\n",
    "We retrieved human TF gene expression regulation edges from several sources using the `regulation` module."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "The function \"prepare_msigdb_data()\" is running...\n",
      "\n",
      "* Number of Transcription Factor Targets (TFT) gene sets: 615\n",
      "\n",
      "The MSigDB raw network is saved at: /home/karolis/Structured review/HD-SR/bioknowledge-reviewer/bioknowledge_reviewer/regulation/msigdb/out/tf_genelist_entrez_msigdb.json. Other reporting files are also saved at the same directory.\n",
      "\n",
      "\n",
      "Finished prepare_msigdb_data().\n",
      "\n",
      "\n",
      "The function \"load_tf_gene_edges()\" is running...\n",
      "\n",
      "Finished load_tf_gene_edges().\n",
      "\n",
      "\n",
      "The function \"get_gene_id_normalization_dictionaries()\" is running...\n",
      "\n",
      "* Querying BioThings to map gene symbols to HGNC and Entrez IDs...\n",
      "querying 1-1000...done.\n",
      "querying 1001-2000...done.\n",
      "querying 2001-3000...done.\n",
      "querying 3001-3071...done.\n",
      "Finished.\n",
      "48 input query terms found no hit:\n",
      "\t['TAL1BETAE47', 'COMP1', 'PTF1BETA', 'MYCMAX', 'NKX62', 'MYOGNF1', 'MEIS1AHOXA9', 'NFMUE1', 'NFKAPPA\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "Saving not found gene symbols at: /home/karolis/Structured review/HD-SR/bioknowledge-reviewer/bioknowledge_reviewer/regulation/not_found_symbols.list\n",
      "\n",
      "\n",
      "* Querying BioThings to map Entrez to HGNC IDs and gene symbols...\n",
      "querying 1-1000...done.\n",
      "querying 1001-2000...done.\n",
      "querying 2001-3000...done.\n",
      "querying 3001-4000...done.\n",
      "querying 4001-5000...done.\n",
      "querying 5001-6000...done.\n",
      "querying 6001-7000...done.\n",
      "querying 7001-8000...done.\n",
      "querying 8001-9000...done.\n",
      "querying 9001-10000...done.\n",
      "querying 10001-11000...done.\n",
      "querying 11001-12000...done.\n",
      "querying 12001-13000...done.\n",
      "querying 13001-14000...done.\n",
      "querying 14001-15000...done.\n",
      "querying 15001-16000...done.\n",
      "querying 16001-16632...done.\n",
      "Finished.\n",
      "144 input query terms found no hit:\n",
      "\t['284475', '9503', '148872', '1231', '284608', '136015', '399819', '388289', '4276', '257357', '2834\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "Saving not found Entrez gene IDs at: /home/karolis/Structured review/HD-SR/bioknowledge-reviewer/bioknowledge_reviewer/regulation/not_found_entrez.list\n",
      "\n",
      "\n",
      "Finished get_gene_id_normalization_dictionaries().\n",
      "\n",
      "\n",
      "The function \"prepare_data_edges()\" is running...\n",
      "\n",
      "The tftargets edges are saved at: /home/karolis/Structured review/HD-SR/bioknowledge-reviewer/bioknowledge_reviewer/regulation/tftargets/tftargets_edges.csv\n",
      "\n",
      "\n",
      "The MSigDB edges are saved at: /home/karolis/Structured review/HD-SR/bioknowledge-reviewer/bioknowledge_reviewer/regulation/msigdb/msigdb_edges.csv\n",
      "\n",
      "\n",
      "Finished prepare_data_edges().\n",
      "\n",
      "\n",
      "The function \"prepare_regulation_edges()\" is running...\n",
      "\n",
      "Finished prepare_regulation_edges().\n",
      "\n",
      "\n",
      "The function \"build_edges()\" is running...\n",
      "\n",
      "* This is the size of the edges file data structure: (197267, 9)\n",
      "* These are the edges attributes: Index(['object_id', 'property_description', 'property_id', 'property_label',\n",
      "       'property_uri', 'reference_date', 'reference_supporting_text',\n",
      "       'reference_uri', 'subject_id'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "   object_id property_description property_id  property_label  \\\n",
      "0  HGNC:8803                   NA  RO:0002434  interacts with   \n",
      "\n",
      "                                property_uri reference_date  \\\n",
      "0  http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
      "\n",
      "                           reference_supporting_text  \\\n",
      "0  This edge comes from the TRED dataset in \"tfta...   \n",
      "\n",
      "                                  reference_uri subject_id  \n",
      "0  https://www.ncbi.nlm.nih.gov/pubmed/17202159  HGNC:8615  \n",
      "\n",
      "The regulation network edges are built and saved at: /home/karolis/Structured review/HD-SR/bioknowledge-reviewer/bioknowledge_reviewer/graph/regulation_edges_v2020-12-12.csv\n",
      "\n",
      "\n",
      "Finished build_edges().\n",
      "\n",
      "\n",
      "The function \"build_nodes()\" is running...\n",
      "\n",
      "* Total number of nodes: 16968\n",
      "\n",
      "* Trap genes without gene symbol, i.e. genes with discontinued entrez ID...\n",
      "* Number of concepts without gene symbol: 144\n",
      "* Check that all genes without gene symbol are identified by entrez ID...\n",
      "* Number of concepts without gene symbol by namespace:  NCBIGene    144\n",
      "Name: id, dtype: int64\n",
      "\n",
      "* Querying BioThings to map retired Entrez to gene symbols...\n",
      "querying 1-144...done.\n",
      "Finished.\n",
      "90 input query terms found no hit:\n",
      "\t['79907', '79911', '93333', '121301', '143902', '146856', '151720', '197379', '219392', '221943', '2\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "* Querying BioThings to retrieve node attributes...\n",
      "querying 1-1000...done.\n",
      "querying 1001-2000...done.\n",
      "querying 2001-3000...done.\n",
      "querying 3001-4000...done.\n",
      "querying 4001-5000...done.\n",
      "querying 5001-6000...done.\n",
      "querying 6001-7000...done.\n",
      "querying 7001-8000...done.\n",
      "querying 8001-9000...done.\n",
      "querying 9001-10000...done.\n",
      "querying 10001-11000...done.\n",
      "querying 11001-12000...done.\n",
      "querying 12001-13000...done.\n",
      "querying 13001-14000...done.\n",
      "querying 14001-15000...done.\n",
      "querying 15001-16000...done.\n",
      "querying 16001-16843...done.\n",
      "Finished.\n",
      "48 input query terms found no hit:\n",
      "\t['PTF1BETA', 'NFMUE1', 'ALPHACP1', 'CREBP1', 'E2F4DP2', 'AMEF2', 'HP1SITEFACTOR', 'AHRARNT', 'CACCCB\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "* This is the size of the nodes file data structure: (16968, 6)\n",
      "* These are the nodes attributes: Index(['description', 'id', 'name', 'preflabel', 'semantic_groups',\n",
      "       'synonyms'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "                                         description         id          name  \\\n",
      "0  This gene is a member of the paired box (PAX) ...  HGNC:8615  paired box 1   \n",
      "\n",
      "  preflabel semantic_groups    synonyms  \n",
      "0      PAX1            GENE  HUP48|OFC2  \n",
      "\n",
      "The regulation network nodes are built and saved at: /home/karolis/Structured review/HD-SR/bioknowledge-reviewer/bioknowledge_reviewer/graph/regulation_nodes_v2020-12-12.csv\n",
      "\n",
      "\n",
      "Finished build_nodes().\n",
      "\n",
      "CPU times: user 3min 33s, sys: 759 ms, total: 3min 34s\n",
      "Wall time: 5min 41s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "# prepare msigdb data\n",
    "gmt_path = './regulation/msigdb/data/c3.tft.v6.1.entrez.gmt'\n",
    "regulation.prepare_msigdb_data(gmt_path)\n",
    "\n",
    "# prepare individual networks\n",
    "data = regulation.load_tf_gene_edges()\n",
    "dicts = regulation.get_gene_id_normalization_dictionaries(data)\n",
    "data_edges = regulation.prepare_data_edges(data, dicts)\n",
    "\n",
    "# prepare regulation network\n",
    "reg_network = regulation.prepare_regulation_edges(data_edges)\n",
    "\n",
    "# build network with graph schema\n",
    "reg_edges = regulation.build_edges(reg_network)\n",
    "reg_nodes = regulation.build_nodes(reg_network)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- Regulation network is returned as both digital object (`reg_edges`, `reg_nodes`) and CSV files at _**graph/**_ (`regulation_edges_version.csv`, `regulation_nodes_version.csv`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "type edges: <class 'list'>\n",
      "type nodes: <class 'list'>\n",
      "\n",
      "len edges: 197267\n",
      "len nodes: 16968\n",
      "\n",
      "attribute edges: dict_keys(['subject_id', 'object_id', 'property_id', 'property_label', 'property_description', 'property_uri', 'reference_uri', 'reference_supporting_text', 'reference_date'])\n",
      "attribute nodes: dict_keys(['id', 'semantic_groups', 'preflabel', 'name', 'synonyms', 'description'])\n"
     ]
    }
   ],
   "source": [
    "# print type of objects\n",
    "print('type edges:', type(reg_edges))\n",
    "print('type nodes:', type(reg_nodes))\n",
    "print()\n",
    "\n",
    "# print objects sizes\n",
    "print('len edges:', len(reg_edges))\n",
    "print('len nodes:', len(reg_nodes))\n",
    "print()\n",
    "\n",
    "# print object attribute\n",
    "print('attribute edges:', reg_edges[0].keys())\n",
    "print('attribute nodes:', reg_nodes[0].keys())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### CURATED NETWORK\n",
    "#### import curation\n",
    "\n",
    "We retrieved and prepared curated edges using the `curation` module. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "curation_edges = pd.read_csv(\"curation/data/HD/Empty_edges.csv\")\n",
    "curation_nodes = pd.read_csv(\"curation/data/HD/Empty_nodes.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "%%time\n",
    "# graph v3.2\n",
    "# read network from drive and concat all curated statements\n",
    "curation_edges, curation_nodes = curation.read_network(version='v20180118')\n",
    "\n",
    "# prepare data edges and nodes\n",
    "data_edges = curation.prepare_data_edges(curation_edges)\n",
    "data_nodes = curation.prepare_data_nodes(curation_nodes)\n",
    "\n",
    "# prepare curated edges and nodes\n",
    "curated_network = curation.prepare_curated_edges(data_edges)\n",
    "curated_concepts = curation.prepare_curated_nodes(data_nodes)\n",
    "\n",
    "\n",
    "# build edges and nodes files\n",
    "curation_edges = curation.build_edges(curated_network)\n",
    "curation_nodes = curation.build_nodes(curated_concepts)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- Curated network is returned as both digital object (`curation_edges`, `curation_nodes`) and CSV files at _**graph/**_ (`curated_graph_edges_version.csv`, `curated_graph_nodes_version.csv`)\n",
    "- The original curated network, i.e. without graph data model normalization, is saved as CSV files at _**curation/**_ (`curated_edges_version.csv`, `curated_nodes_version.csv`)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# print type of objects\n",
    "print('type edges:', type(curation_edges))\n",
    "print('type nodes:', type(curation_nodes))\n",
    "print()\n",
    "\n",
    "# print objects sizes\n",
    "print('len edges:', len(curation_edges))\n",
    "print('len nodes:', len(curation_nodes))\n",
    "print()\n",
    "\n",
    "# print object attribute\n",
    "print('attribute edges:', curation_edges[0].keys())\n",
    "print('attribute nodes:', curation_nodes[0].keys())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### MONARCH NETWORK\n",
    "#### import monarch\n",
    "We retrieved edges from Monarch using the `monarch` module.\n",
    "\n",
    "Tasks:\n",
    "\n",
    "- From 8 seed nodes we retrieved 1st shell nodes\n",
    "- From all seed and 1st shell nodes we retrieved ortho-phenotypes\n",
    "- We retrieved extra edges among all of them, i.e. extra connectivity between: seed, 1st shell, ortholog-phenotype nodes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "The function \"get_neighbours_list()\" is running. Its runtime may take some minutes. If you interrupt the process, you will lose all the nodes retrieved and you should start over the execution of this function.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 4/4 [00:22<00:00,  4.91s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Finished get_neighbours_list().\n",
      "\n",
      "1024\n",
      "\n",
      "The function \"get_orthopheno_list()\" is running. Its runtime may take some hours. If you interrupt the process, you will lose all the nodes retrieved and you should start over the execution of this function.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 4/4 [00:22<00:00,  4.87s/it]\n",
      "100%|██████████| 16/16 [00:50<00:00,  4.90s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Finished get_orthopheno_list().\n",
      "\n",
      "227\n",
      "\n",
      "The function \"get_orthopheno_list()\" is running. Its runtime may take some hours. If you interrupt the process, you will lose all the nodes retrieved and you should start over the execution of this function.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 1024/1024 [1:19:32<00:00,  3.38s/it]\n",
      "100%|██████████| 7307/7307 [6:06:49<00:00,  2.21s/it]   \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Finished get_orthopheno_list().\n",
      "\n",
      "13015\n",
      "genelist:  14270\n",
      "\n",
      "The function \"extract_edges()\" is running. Its runtime may take some hours. If you interrupt the process, you will lose all the edges retrieved and you should start over the execution of this function.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 14021/14021 [15:36:11<00:00,  3.07s/it]   \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Finished extract_edges(). To save the retrieved Monarch edges use the function \"print_network()\".\n",
      "\n",
      "network:  233182\n",
      "\n",
      "Saving Monarch edges at: '/home/karolis/Structured review/HD-SR/bioknowledge-reviewer/bioknowledge_reviewer/monarch/monarch_connections_v2020-12-12.csv'...\n",
      "\n",
      "\n",
      "The function \"build_edges()\" is running...\n",
      "df (233182, 9)\n",
      "\n",
      "* This is the size of the edges file data structure: (233182, 9)\n",
      "* These are the edges attributes: Index(['object_id', 'property_description', 'property_id', 'property_label',\n",
      "       'property_uri', 'reference_date', 'reference_supporting_text',\n",
      "       'reference_uri', 'subject_id'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "    object_id property_description    property_id  \\\n",
      "0  MGI:892004                   NA  RO:HOM0000017   \n",
      "\n",
      "                   property_label  \\\n",
      "0  in orthology relationship with   \n",
      "\n",
      "                                   property_uri reference_date  \\\n",
      "0  http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
      "\n",
      "                           reference_supporting_text reference_uri  \\\n",
      "0  This edge comes from the Monarch Knowledge Gra...            NA   \n",
      "\n",
      "    subject_id  \n",
      "0  RGD:1595815  \n",
      "\n",
      "The Monarch network edges are built and saved at: /home/karolis/Structured review/HD-SR/bioknowledge-reviewer/bioknowledge_reviewer/monarch/monarch_edges_v2020-12-12.csv\n",
      "\n",
      "\n",
      "Finished build_edges().\n",
      "\n",
      "\n",
      "The function \"build_nodes()\" is running...\n",
      "Number of concepts: 14018\n",
      "Number of nodes CURIEs: 31\n",
      "List of nodes CURIEs: dict_keys(['RGD', 'MGI', 'ENSEMBL', 'FlyBase', 'SGD', 'MONARCH', 'HGNC', 'ZFIN', 'HP', 'MP', 'WormBase', 'dictyBase', 'GO', 'ZP', 'PomBase', 'UBERON', 'WBPhenotype', 'Xenbase', 'APO', 'EnsemblGenome', 'EcoGene', 'FBcv', 'MONDO', 'NCBIGene', 'REACT', 'EFO', 'Coriell', 'dbSNP', 'OMIM', 'dbSNPIndividual', 'ClinVarVariant'])\n",
      "\n",
      "Adding BioThings annotation: gene name, synonyms, description...\n",
      "symbols: 7417\n",
      "querying 1-1000...done.\n",
      "querying 1001-2000...done.\n",
      "querying 2001-3000...done.\n",
      "querying 3001-4000...done.\n",
      "querying 4001-5000...done.\n",
      "querying 5001-6000...done.\n",
      "querying 6001-7000...done.\n",
      "querying 7001-7417...done.\n",
      "Finished.\n",
      "819 input query terms found dup hits:\n",
      "\t[('Rheb', 3), ('Dnajb6', 2), ('SKP1', 4), ('Skp1', 2), ('BCAR1', 4), ('CNOT2', 4), ('DDB1', 4), ('He\n",
      "3831 input query terms found no hit:\n",
      "\t['A0A0G2K8R6', 'ENSEMBL:ENSOANG00000006132', 'ENSEMBL:ENSCAFG00000017647', 'ENSEMBL:ENSECAG000000190\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "* This is the size of the nodes file data structure: (14018, 6)\n",
      "* These are the nodes attributes: Index(['description', 'id', 'name', 'preflabel', 'semantic_groups',\n",
      "       'synonyms'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "  description           id                           name preflabel  \\\n",
      "0          NA  RGD:1595815  RT1 class I, locus M6, gene 2  RT1-M6-2   \n",
      "\n",
      "  semantic_groups synonyms  \n",
      "0            GENE       NA  \n",
      "\n",
      "The Monarch network nodes are built and saved at: /home/karolis/Structured review/HD-SR/bioknowledge-reviewer/bioknowledge_reviewer/monarch/monarch_nodes_v2020-12-12.csv\n",
      "\n",
      "\n",
      "Finished build_nodes().\n",
      "\n",
      "CPU times: user 46min 15s, sys: 2min 44s, total: 49min\n",
      "Wall time: 23h 4min 52s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "# prepare data to graph schema\n",
    "# seed nodes\n",
    "seedList = [ \n",
    "    'MONDO:0007739', # HD\n",
    "    'HGNC:4851', # Htt\n",
    "    'HGNC:182293', # Rhes\n",
    "    'REACT:R-HSA-917937' #Iron uptake pathway\n",
    "] \n",
    "\n",
    "# get first shell of neighbours\n",
    "neighboursList = monarch.get_neighbours_list(seedList)\n",
    "print(len(neighboursList))\n",
    "\n",
    "# introduce animal model ortho-phenotypes for seed and 1st shell neighbors\n",
    "## For seed nodes:\n",
    "seed_orthophenoList = monarch.get_orthopheno_list(seedList)\n",
    "print(len(seed_orthophenoList))\n",
    "## For 1st shell nodes:\n",
    "neighbours_orthophenoList = monarch.get_orthopheno_list(neighboursList)\n",
    "print(len(neighbours_orthophenoList))\n",
    "\n",
    "# network nodes: seed + 1shell + ortholog-phentoype\n",
    "geneList = sum([seedList,\n",
    "                neighboursList,\n",
    "                seed_orthophenoList,\n",
    "                neighbours_orthophenoList\n",
    "               ], \n",
    "               [])\n",
    "print('genelist: ',len(geneList))\n",
    "\n",
    "# get Monarch network\n",
    "monarch_network = monarch.extract_edges(geneList)\n",
    "print('network: ',len(monarch_network))\n",
    "\n",
    "# save edges\n",
    "monarch.print_network(monarch_network, 'monarch_connections')\n",
    "\n",
    "# build network with graph schema \n",
    "monarch_edges = monarch.build_edges(monarch_network)\n",
    "monarch_nodes = monarch.build_nodes(monarch_network)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Saving Monarch edges at: '/home/karolis/Structured review/HD-SR/bioknowledge-reviewer/bioknowledge_reviewer/monarch/monarch_connections_v2020-12-12.csv'...\n",
      "\n",
      "\n",
      "The function \"build_edges()\" is running...\n",
      "df (233182, 9)\n",
      "\n",
      "* This is the size of the edges file data structure: (233182, 9)\n",
      "* These are the edges attributes: Index(['object_id', 'property_description', 'property_id', 'property_label',\n",
      "       'property_uri', 'reference_date', 'reference_supporting_text',\n",
      "       'reference_uri', 'subject_id'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "    object_id property_description    property_id  \\\n",
      "0  MGI:892004                   NA  RO:HOM0000017   \n",
      "\n",
      "                   property_label  \\\n",
      "0  in orthology relationship with   \n",
      "\n",
      "                                   property_uri reference_date  \\\n",
      "0  http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
      "\n",
      "                           reference_supporting_text reference_uri  \\\n",
      "0  This edge comes from the Monarch Knowledge Gra...            NA   \n",
      "\n",
      "    subject_id  \n",
      "0  RGD:1595815  \n",
      "\n",
      "The Monarch network edges are built and saved at: /home/karolis/Structured review/HD-SR/bioknowledge-reviewer/bioknowledge_reviewer/monarch/monarch_edges_v2020-12-12.csv\n",
      "\n",
      "\n",
      "Finished build_edges().\n",
      "\n",
      "\n",
      "The function \"build_nodes()\" is running...\n",
      "Number of concepts: 14018\n",
      "Number of nodes CURIEs: 31\n",
      "List of nodes CURIEs: dict_keys(['RGD', 'MGI', 'ENSEMBL', 'FlyBase', 'SGD', 'MONARCH', 'HGNC', 'ZFIN', 'HP', 'MP', 'WormBase', 'dictyBase', 'GO', 'ZP', 'PomBase', 'UBERON', 'WBPhenotype', 'Xenbase', 'APO', 'EnsemblGenome', 'EcoGene', 'FBcv', 'MONDO', 'NCBIGene', 'REACT', 'EFO', 'Coriell', 'dbSNP', 'OMIM', 'dbSNPIndividual', 'ClinVarVariant'])\n",
      "\n",
      "Adding BioThings annotation: gene name, synonyms, description...\n",
      "symbols: 7417\n",
      "querying 1-1000...done.\n",
      "querying 1001-2000...done.\n",
      "querying 2001-3000...done.\n",
      "querying 3001-4000...done.\n",
      "querying 4001-5000...done.\n",
      "querying 5001-6000...done.\n",
      "querying 6001-7000...done.\n",
      "querying 7001-7417...done.\n",
      "Finished.\n",
      "819 input query terms found dup hits:\n",
      "\t[('Rheb', 3), ('Dnajb6', 2), ('SKP1', 4), ('Skp1', 2), ('BCAR1', 4), ('CNOT2', 4), ('DDB1', 4), ('He\n",
      "3831 input query terms found no hit:\n",
      "\t['A0A0G2K8R6', 'ENSEMBL:ENSOANG00000006132', 'ENSEMBL:ENSCAFG00000017647', 'ENSEMBL:ENSECAG000000190\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "* This is the size of the nodes file data structure: (14018, 6)\n",
      "* These are the nodes attributes: Index(['description', 'id', 'name', 'preflabel', 'semantic_groups',\n",
      "       'synonyms'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "  description           id                           name preflabel  \\\n",
      "0          NA  RGD:1595815  RT1 class I, locus M6, gene 2  RT1-M6-2   \n",
      "\n",
      "  semantic_groups synonyms  \n",
      "0            GENE       NA  \n",
      "\n",
      "The Monarch network nodes are built and saved at: /home/karolis/Structured review/HD-SR/bioknowledge-reviewer/bioknowledge_reviewer/monarch/monarch_nodes_v2020-12-12.csv\n",
      "\n",
      "\n",
      "Finished build_nodes().\n",
      "\n"
     ]
    }
   ],
   "source": [
    "monarch.print_network(monarch_network, 'monarch_connections')\n",
    "\n",
    "# build network with graph schema \n",
    "monarch_edges = monarch.build_edges(monarch_network)\n",
    "monarch_nodes = monarch.build_nodes(monarch_network)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "type edges: <class 'list'>\n",
      "type nodes: <class 'list'>\n",
      "\n",
      "len edges: 233182\n",
      "len nodes: 14018\n",
      "\n",
      "attribute edges: dict_keys(['subject_id', 'object_id', 'property_id', 'property_label', 'property_description', 'property_uri', 'reference_uri', 'reference_supporting_text', 'reference_date'])\n",
      "attribute nodes: dict_keys(['id', 'semantic_groups', 'preflabel', 'name', 'synonyms', 'description'])\n"
     ]
    }
   ],
   "source": [
    "# print type of objects\n",
    "print('type edges:', type(monarch_edges))\n",
    "print('type nodes:', type(monarch_nodes))\n",
    "print()\n",
    "\n",
    "# print objects sizes\n",
    "print('len edges:', len(monarch_edges))\n",
    "print('len nodes:', len(monarch_nodes))\n",
    "print()\n",
    "\n",
    "# print object attribute\n",
    "print('attribute edges:', monarch_edges[0].keys())\n",
    "print('attribute nodes:', monarch_nodes[0].keys())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "%%time\n",
    "# prepare data to graph schema\n",
    "# seed nodes\n",
    "seedList = [ \n",
    "    'MONDO:0007739', # HD\n",
    "    'HGNC:4851', # Htt\n",
    "    'HGNC:182293', # Rhes\n",
    "] \n",
    "\n",
    "# get first shell of neighbours\n",
    "neighboursList = monarch.get_neighbours_list(seedList)\n",
    "print(len(neighboursList))\n",
    "\n",
    "# introduce animal model ortho-phenotypes for seed and 1st shell neighbors\n",
    "## For seed nodes:\n",
    "seed_orthophenoList = monarch.get_orthopheno_list(seedList)\n",
    "print(len(seed_orthophenoList))\n",
    "## For 1st shell nodes:\n",
    "neighbours_orthophenoList = monarch.get_orthopheno_list(neighboursList)\n",
    "print(len(neighbours_orthophenoList))\n",
    "\n",
    "# network nodes: seed + 1shell + ortholog-phentoype\n",
    "geneList = sum([seedList,\n",
    "                neighboursList,\n",
    "                #seed_orthophenoList,\n",
    "                #neighbours_orthophenoList\n",
    "               ], \n",
    "               [])\n",
    "print('genelist: ',len(geneList))\n",
    "\n",
    "# get Monarch network\n",
    "monarch_network = monarch.extract_edges(geneList)\n",
    "print('network: ',len(monarch_network))\n",
    "\n",
    "# save edges\n",
    "monarch.print_network(monarch_network, 'monarch_connections')\n",
    "\n",
    "# build network with graph schema \n",
    "monarch_edges = monarch.build_edges(monarch_network)\n",
    "monarch_nodes = monarch.build_nodes(monarch_network)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "monarch_network = monarch.read_connections(\"monarch_connections_v2020-11-15.csv\")\n",
    "monarch_edges = monarch.build_edges(monarch_network)\n",
    "monarch_nodes = monarch.build_nodes(monarch_network)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "monarch_edges = monarch.build_edges(monarch_network)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "monarch_nodes = monarch.build_nodes(monarch_network)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "type edges: <class 'list'>\n",
      "type nodes: <class 'list'>\n",
      "\n",
      "len edges: 233182\n",
      "len nodes: 14018\n",
      "\n",
      "attribute edges: dict_keys(['subject_id', 'object_id', 'property_id', 'property_label', 'property_description', 'property_uri', 'reference_uri', 'reference_supporting_text', 'reference_date'])\n",
      "attribute nodes: dict_keys(['id', 'semantic_groups', 'preflabel', 'name', 'synonyms', 'description'])\n"
     ]
    }
   ],
   "source": [
    "# print type of objects\n",
    "print('type edges:', type(monarch_edges))\n",
    "print('type nodes:', type(monarch_nodes))\n",
    "print()\n",
    "\n",
    "# print objects sizes\n",
    "print('len edges:', len(monarch_edges))\n",
    "print('len nodes:', len(monarch_nodes))\n",
    "print()\n",
    "\n",
    "# print object attribute\n",
    "print('attribute edges:', monarch_edges[0].keys())\n",
    "print('attribute nodes:', monarch_nodes[0].keys())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- Monarch network is returned as both digital object (`monarch_edges`, `monarch_nodes`) and CSV files at _**monarch/**_ (`monarch_edges_version.csv`, `monarch_nodes_version.csv`)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Graph library\n",
    "### Create the review knowledge graph\n",
    "#### import graph\n",
    "\n",
    "Tasks:\n",
    "\n",
    "* Load Networks and calculate graph nodes\n",
    "* Retrieve extra connectivity for the graph from Monarch\n",
    "* Build the review graph"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "curation_edges = pd.read_csv(\"curation/data/HD/Empty_edges.csv\")\n",
    "curation_nodes = pd.read_csv(\"curation/data/HD/Empty_nodes.csv\")\n",
    "curation_nodes = curation_nodes.astype('object')\n",
    "curation_nodes['name'] = [\"NaN\", \"NaN\"]\n",
    "curation_edges = curation_edges.to_dict(\"records\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "* This is the size of the data structure: (233182, 7)\n",
      "* These are the attributes: Index(['object_id', 'object_label', 'reference_id_list', 'relation_id',\n",
      "       'relation_label', 'subject_id', 'subject_label'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "    object_id object_label  reference_id_list    relation_id  \\\n",
      "0  MGI:892004        H2-Bl                NaN  RO:HOM0000017   \n",
      "\n",
      "                   relation_label   subject_id subject_label  \n",
      "0  in orthology relationship with  RGD:1595815      RT1-M6-2  \n",
      "\n",
      "The function \"build_edges()\" is running...\n",
      "df (233182, 9)\n",
      "\n",
      "* This is the size of the edges file data structure: (233182, 9)\n",
      "* These are the edges attributes: Index(['object_id', 'property_description', 'property_id', 'property_label',\n",
      "       'property_uri', 'reference_date', 'reference_supporting_text',\n",
      "       'reference_uri', 'subject_id'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "    object_id property_description    property_id  \\\n",
      "0  MGI:892004                   NA  RO:HOM0000017   \n",
      "\n",
      "                   property_label  \\\n",
      "0  in orthology relationship with   \n",
      "\n",
      "                                   property_uri reference_date  \\\n",
      "0  http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
      "\n",
      "                           reference_supporting_text reference_uri  \\\n",
      "0  This edge comes from the Monarch Knowledge Gra...            NA   \n",
      "\n",
      "    subject_id  \n",
      "0  RGD:1595815  \n",
      "\n",
      "The Monarch network edges are built and saved at: /home/karolis/Structured review/HD-SR/bioknowledge-reviewer/bioknowledge_reviewer/monarch/monarch_edges_v2020-12-13.csv\n",
      "\n",
      "\n",
      "Finished build_edges().\n",
      "\n",
      "\n",
      "The function \"build_nodes()\" is running...\n",
      "Number of concepts: 14018\n",
      "Number of nodes CURIEs: 31\n",
      "List of nodes CURIEs: dict_keys(['RGD', 'MGI', 'ENSEMBL', 'FlyBase', 'SGD', 'MONARCH', 'HGNC', 'ZFIN', 'HP', 'MP', 'WormBase', 'dictyBase', 'GO', 'ZP', 'PomBase', 'UBERON', 'WBPhenotype', 'Xenbase', 'APO', 'EnsemblGenome', 'EcoGene', 'FBcv', 'MONDO', 'NCBIGene', 'REACT', 'EFO', 'Coriell', 'dbSNP', 'OMIM', 'dbSNPIndividual', 'ClinVarVariant'])\n",
      "\n",
      "Adding BioThings annotation: gene name, synonyms, description...\n",
      "symbols: 7417\n",
      "querying 1-1000...done.\n",
      "querying 1001-2000...done.\n",
      "querying 2001-3000...done.\n",
      "querying 3001-4000...done.\n",
      "querying 4001-5000...done.\n",
      "querying 5001-6000...done.\n",
      "querying 6001-7000...done.\n",
      "querying 7001-7417...done.\n",
      "Finished.\n",
      "819 input query terms found dup hits:\n",
      "\t[('Rheb', 3), ('Dnajb6', 2), ('SKP1', 4), ('Skp1', 2), ('BCAR1', 4), ('CNOT2', 4), ('DDB1', 4), ('He\n",
      "3831 input query terms found no hit:\n",
      "\t['A0A0G2K8R6', 'ENSEMBL:ENSOANG00000006132', 'ENSEMBL:ENSCAFG00000017647', 'ENSEMBL:ENSECAG000000190\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "* This is the size of the nodes file data structure: (14018, 6)\n",
      "* These are the nodes attributes: Index(['description', 'id', 'name', 'preflabel', 'semantic_groups',\n",
      "       'synonyms'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "  description           id                           name preflabel  \\\n",
      "0          NA  RGD:1595815  RT1 class I, locus M6, gene 2  RT1-M6-2   \n",
      "\n",
      "  semantic_groups synonyms  \n",
      "0            GENE       NA  \n",
      "\n",
      "The Monarch network nodes are built and saved at: /home/karolis/Structured review/HD-SR/bioknowledge-reviewer/bioknowledge_reviewer/monarch/monarch_nodes_v2020-12-13.csv\n",
      "\n",
      "\n",
      "Finished build_nodes().\n",
      "\n",
      "\n",
      "The function \"read_data()\" is running...\n",
      "\n",
      "* This is the size of the raw expression data structure: (28087, 10)\n",
      "* These are the expression attributes: Index(['Unnamed: 0', 'symbol', 'baseMean', 'HD.mean', 'Control.mean',\n",
      "       'log2FoldChange', 'lfcSE', 'stat', 'pvalue', 'padj'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "           Unnamed: 0 symbol  baseMean    HD.mean  Control.mean  \\\n",
      "0  ENSG00000069011.10  PITX1  5.645675  18.684286      0.323793   \n",
      "\n",
      "   log2FoldChange     lfcSE      stat        pvalue          padj  \n",
      "0        4.769658  0.366367  13.01879  9.567529e-39  2.687232e-34  \n",
      "\n",
      "The raw data is saved at: /home/karolis/Structured review/HD-SR/bioknowledge-reviewer/bioknowledge_reviewer/transcriptomics/HD/data/GSE64810_mlhd_DESeq2_diffexp_DESeq2_outlier_trimmed_adjust.csv\n",
      "\n",
      "\n",
      "Finished read_data().\n",
      "\n",
      "\n",
      "The function \"clean_data()\" is running. Keeping only data with FC > 1.5 and FDR < 5% ...\n",
      "\n",
      "* This is the size of the clean expression data structure: (3209, 6)\n",
      "* These are the clean expression attributes: Index(['EnsembleID', 'symbol', 'log2FoldChange', 'pvalue', 'padj',\n",
      "       'Regulation'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "           EnsembleID symbol  log2FoldChange        pvalue          padj  \\\n",
      "0  ENSG00000069011.10  PITX1        4.769658  9.567529e-39  2.687232e-34   \n",
      "\n",
      "    Regulation  \n",
      "0  Upregulated  \n",
      "\n",
      "The clean data is saved at: /home/karolis/Structured review/HD-SR/bioknowledge-reviewer/bioknowledge_reviewer/transcriptomics/HD/out/Transcriptome_human_BA9.csv\n",
      "\n",
      "\n",
      "Finished clean_data().\n",
      "\n",
      "\n",
      "The function \"prepare_data_edges()\" is running...\n",
      "\n",
      "* This is the size of the expression data structure: (3209, 13)\n",
      "* These are the expression attributes: Index(['ensemble_id', 'symbol', 'log2FoldChange', 'pvalue', 'padj',\n",
      "       'regulation', 'source', 'subject_id', 'subject_label', 'property_id',\n",
      "       'property_label', 'reference_id', 'object_id'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "          ensemble_id symbol  log2FoldChange        pvalue          padj  \\\n",
      "0  ENSG00000069011.10  PITX1        4.769658  9.567529e-39  2.687232e-34   \n",
      "\n",
      "    regulation source               subject_id subject_label property_id  \\\n",
      "0  Upregulated   Chow  ensembl:ENSG00000197386           HTT  RO:0002434   \n",
      "\n",
      "   property_label   reference_id                object_id  \n",
      "0  interacts with  PMID:26636579  ensembl:ENSG00000069011  \n",
      "\n",
      "The HD BA9 transcriptomics expression edges are saved at: /home/karolis/Structured review/HD-SR/bioknowledge-reviewer/bioknowledge_reviewer/transcriptomics/HD/out/Transcriptome_human_BA9_edges.csv\n",
      "\n",
      "\n",
      "Finished prepare_data_edges().\n",
      "\n",
      "\n",
      "The function \"prepare_rna_edges()\" is running...\n",
      "\n",
      "* This is the size of the edges data structure: (3209, 12)\n",
      "* These are the edges attributes: Index(['subject_id', 'subject_label', 'property_id', 'property_label',\n",
      "       'object_id', 'object_label', 'log2FoldChange', 'pvalue', 'fdr',\n",
      "       'regulation', 'source', 'reference_id'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "                subject_id subject_label property_id  property_label  \\\n",
      "0  ensembl:ENSG00000197386           HTT  RO:0002434  interacts with   \n",
      "\n",
      "                 object_id object_label  log2FoldChange        pvalue  \\\n",
      "0  ensembl:ENSG00000069011        PITX1        4.769658  9.567529e-39   \n",
      "\n",
      "            fdr   regulation source   reference_id  \n",
      "0  2.687232e-34  Upregulated   Chow  PMID:26636579  \n",
      "\n",
      "This data object is not saved.\n",
      "\n",
      "\n",
      "Finished prepare_rna_edges().\n",
      "\n",
      "\n",
      "The function \"build_edges()\" is running...\n",
      "\n",
      "* This is the size of the edges file data structure: (3209, 9)\n",
      "* These are the edges attributes: Index(['object_id', 'property_description', 'property_id', 'property_label',\n",
      "       'property_uri', 'reference_date', 'reference_supporting_text',\n",
      "       'reference_uri', 'subject_id'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "                 object_id property_description property_id  property_label  \\\n",
      "0  ensembl:ENSG00000069011                   NA  RO:0002434  interacts with   \n",
      "\n",
      "                                property_uri reference_date  \\\n",
      "0  http://purl.obolibrary.org/obo/RO_0002434     2015-11-04   \n",
      "\n",
      "                           reference_supporting_text  \\\n",
      "0  Here we present a genome-wide analysis of mRNA...   \n",
      "\n",
      "                                  reference_uri               subject_id  \n",
      "0  https://www.ncbi.nlm.nih.gov/pubmed/26636579  ensembl:ENSG00000197386  \n",
      "\n",
      "The transcriptomics network edges are built and saved at: /home/karolis/Structured review/HD-SR/bioknowledge-reviewer/bioknowledge_reviewer/graph/rna_edges_v2020-12-13.csv\n",
      "\n",
      "\n",
      "Finished build_edges().\n",
      "\n",
      "\n",
      "The function \"build_nodes()\" is running...\n",
      "* Total number of nodes: 3210\n",
      "querying 1-1000...done.\n",
      "querying 1001-2000...done.\n",
      "querying 2001-3000...done.\n",
      "querying 3001-3210...done.\n",
      "Finished.\n",
      "178 input query terms found no hit:\n",
      "\t['ENSG00000200127', 'ENSG00000183748', 'ENSG00000140181', 'ENSG00000261589', 'ENSG00000034063', 'ENS\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "* This is the size of the nodes file data structure: (3210, 6)\n",
      "* These are the nodes attributes: Index(['description', 'id', 'name', 'preflabel', 'semantic_groups',\n",
      "       'synonyms'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "                                         description         id        name  \\\n",
      "0  Huntingtin is a disease gene linked to Hunting...  HGNC:4851  huntingtin   \n",
      "\n",
      "  preflabel semantic_groups        synonyms  \n",
      "0       HTT            GENE  HD|IT15|LOMARS  \n",
      "\n",
      "The transcriptomics network nodes are built and saved at: /home/karolis/Structured review/HD-SR/bioknowledge-reviewer/bioknowledge_reviewer/graph/rna_nodes_v2020-12-13.csv\n",
      "\n",
      "\n",
      "Finished build_nodes().\n",
      "\n",
      "\n",
      "The function \"prepare_msigdb_data()\" is running...\n",
      "\n",
      "* Number of Transcription Factor Targets (TFT) gene sets: 615\n",
      "\n",
      "The MSigDB raw network is saved at: /home/karolis/Structured review/HD-SR/bioknowledge-reviewer/bioknowledge_reviewer/regulation/msigdb/out/tf_genelist_entrez_msigdb.json. Other reporting files are also saved at the same directory.\n",
      "\n",
      "\n",
      "Finished prepare_msigdb_data().\n",
      "\n",
      "\n",
      "The function \"load_tf_gene_edges()\" is running...\n",
      "\n",
      "Finished load_tf_gene_edges().\n",
      "\n",
      "\n",
      "The function \"get_gene_id_normalization_dictionaries()\" is running...\n",
      "\n",
      "* Querying BioThings to map gene symbols to HGNC and Entrez IDs...\n",
      "querying 1-1000...done.\n",
      "querying 1001-2000...done.\n",
      "querying 2001-3000...done.\n",
      "querying 3001-3071...done.\n",
      "Finished.\n",
      "48 input query terms found no hit:\n",
      "\t['COREBINDINGFACTOR', 'NKX22', 'AP1FJ', 'CDPCR3', 'E2F1DP1RB', 'CACCCBINDINGFACTOR', 'PTF1BETA', 'AH\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "Saving not found gene symbols at: /home/karolis/Structured review/HD-SR/bioknowledge-reviewer/bioknowledge_reviewer/regulation/not_found_symbols.list\n",
      "\n",
      "\n",
      "* Querying BioThings to map Entrez to HGNC IDs and gene symbols...\n",
      "querying 1-1000...done.\n",
      "querying 1001-2000...done.\n",
      "querying 2001-3000...done.\n",
      "querying 3001-4000...done.\n",
      "querying 4001-5000...done.\n",
      "querying 5001-6000...done.\n",
      "querying 6001-7000...done.\n",
      "querying 7001-8000...done.\n",
      "querying 8001-9000...done.\n",
      "querying 9001-10000...done.\n",
      "querying 10001-11000...done.\n",
      "querying 11001-12000...done.\n",
      "querying 12001-13000...done.\n",
      "querying 13001-14000...done.\n",
      "querying 14001-15000...done.\n",
      "querying 15001-16000...done.\n",
      "querying 16001-16632...done.\n",
      "Finished.\n",
      "144 input query terms found no hit:\n",
      "\t['9142', '157909', '348143', '200058', '145663', '88523', '121301', '100500805', '283476', '340370',\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "Saving not found Entrez gene IDs at: /home/karolis/Structured review/HD-SR/bioknowledge-reviewer/bioknowledge_reviewer/regulation/not_found_entrez.list\n",
      "\n",
      "\n",
      "Finished get_gene_id_normalization_dictionaries().\n",
      "\n",
      "\n",
      "The function \"prepare_data_edges()\" is running...\n",
      "\n",
      "The tftargets edges are saved at: /home/karolis/Structured review/HD-SR/bioknowledge-reviewer/bioknowledge_reviewer/regulation/tftargets/tftargets_edges.csv\n",
      "\n",
      "\n",
      "The MSigDB edges are saved at: /home/karolis/Structured review/HD-SR/bioknowledge-reviewer/bioknowledge_reviewer/regulation/msigdb/msigdb_edges.csv\n",
      "\n",
      "\n",
      "Finished prepare_data_edges().\n",
      "\n",
      "\n",
      "The function \"prepare_regulation_edges()\" is running...\n",
      "\n",
      "Finished prepare_regulation_edges().\n",
      "\n",
      "\n",
      "The function \"build_edges()\" is running...\n",
      "\n",
      "* This is the size of the edges file data structure: (197267, 9)\n",
      "* These are the edges attributes: Index(['object_id', 'property_description', 'property_id', 'property_label',\n",
      "       'property_uri', 'reference_date', 'reference_supporting_text',\n",
      "       'reference_uri', 'subject_id'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "   object_id property_description property_id  property_label  \\\n",
      "0  HGNC:8803                   NA  RO:0002434  interacts with   \n",
      "\n",
      "                                property_uri reference_date  \\\n",
      "0  http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
      "\n",
      "                           reference_supporting_text  \\\n",
      "0  This edge comes from the TRED dataset in \"tfta...   \n",
      "\n",
      "                                  reference_uri subject_id  \n",
      "0  https://www.ncbi.nlm.nih.gov/pubmed/17202159  HGNC:8615  \n",
      "\n",
      "The regulation network edges are built and saved at: /home/karolis/Structured review/HD-SR/bioknowledge-reviewer/bioknowledge_reviewer/graph/regulation_edges_v2020-12-13.csv\n",
      "\n",
      "\n",
      "Finished build_edges().\n",
      "\n",
      "\n",
      "The function \"build_nodes()\" is running...\n",
      "\n",
      "* Total number of nodes: 16968\n",
      "\n",
      "* Trap genes without gene symbol, i.e. genes with discontinued entrez ID...\n",
      "* Number of concepts without gene symbol: 144\n",
      "* Check that all genes without gene symbol are identified by entrez ID...\n",
      "* Number of concepts without gene symbol by namespace:  NCBIGene    144\n",
      "Name: id, dtype: int64\n",
      "\n",
      "* Querying BioThings to map retired Entrez to gene symbols...\n",
      "querying 1-144...done.\n",
      "Finished.\n",
      "90 input query terms found no hit:\n",
      "\t['79907', '79911', '93333', '121301', '143902', '146856', '151720', '197379', '219392', '221943', '2\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "* Querying BioThings to retrieve node attributes...\n",
      "querying 1-1000...done.\n",
      "querying 1001-2000...done.\n",
      "querying 2001-3000...done.\n",
      "querying 3001-4000...done.\n",
      "querying 4001-5000...done.\n",
      "querying 5001-6000...done.\n",
      "querying 6001-7000...done.\n",
      "querying 7001-8000...done.\n",
      "querying 8001-9000...done.\n",
      "querying 9001-10000...done.\n",
      "querying 10001-11000...done.\n",
      "querying 11001-12000...done.\n",
      "querying 12001-13000...done.\n",
      "querying 13001-14000...done.\n",
      "querying 14001-15000...done.\n",
      "querying 15001-16000...done.\n",
      "querying 16001-16843...done.\n",
      "Finished.\n",
      "48 input query terms found no hit:\n",
      "\t['E2F1DP1RB', 'AHRARNT', 'SIN3AK20', 'E2F1DP1', 'HMEF2', 'CACBINDINGPROTEIN', 'CREBP1', 'TAL1BETAE47\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "* This is the size of the nodes file data structure: (16968, 6)\n",
      "* These are the nodes attributes: Index(['description', 'id', 'name', 'preflabel', 'semantic_groups',\n",
      "       'synonyms'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "                                         description         id          name  \\\n",
      "0  This gene is a member of the paired box (PAX) ...  HGNC:8615  paired box 1   \n",
      "\n",
      "  preflabel semantic_groups    synonyms  \n",
      "0      PAX1            GENE  HUP48|OFC2  \n",
      "\n",
      "The regulation network nodes are built and saved at: /home/karolis/Structured review/HD-SR/bioknowledge-reviewer/bioknowledge_reviewer/graph/regulation_nodes_v2020-12-13.csv\n",
      "\n",
      "\n",
      "Finished build_nodes().\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# obtain variables:\n",
    "#curation_edges = pd.read_csv(\"curation/data/HD/Empty_edges.csv\")\n",
    "#curation_nodes = pd.read_csv(\"curation/data/HD/Empty_nodes.csv\")\n",
    "\n",
    "monarch_network = monarch.read_connections(\"monarch_connections_v2020-12-12.csv\")\n",
    "monarch_edges = monarch.build_edges(monarch_network)\n",
    "monarch_nodes = monarch.build_nodes(monarch_network)\n",
    "\n",
    "#monarch_network_graph = monarch.read_connections(\"monarch_connections_graph_v2020-12-02.csv\")\n",
    "#monarch_graph_edges = monarch.build_edges(monarch_network_graph)\n",
    "#monarch_graph_nodes = monarch.build_nodes(monarch_network_graph)\n",
    "\n",
    "csv_path = './transcriptomics/GSE64810_mlhd_DESeq2_diffexp_DESeq2_outlier_trimmed_adjust.txt'\n",
    "data = transcriptomics.read_data(csv_path)\n",
    "clean_data = transcriptomics.clean_data(data)\n",
    "data_edges = transcriptomics.prepare_data_edges(clean_data)\n",
    "rna_network = transcriptomics.prepare_rna_edges(data_edges)\n",
    "\n",
    "# build network with graph schema\n",
    "rna_edges = transcriptomics.build_edges(rna_network)\n",
    "rna_nodes, rna_edges = transcriptomics.build_nodes(rna_network)\n",
    "\n",
    "gmt_path = './regulation/msigdb/data/c3.tft.v6.1.entrez.gmt'\n",
    "regulation.prepare_msigdb_data(gmt_path)\n",
    "\n",
    "# prepare individual networks\n",
    "data = regulation.load_tf_gene_edges()\n",
    "dicts = regulation.get_gene_id_normalization_dictionaries(data)\n",
    "data_edges = regulation.prepare_data_edges(data, dicts)\n",
    "\n",
    "# prepare regulation network\n",
    "reg_network = regulation.prepare_regulation_edges(data_edges)\n",
    "\n",
    "# build network with graph schema\n",
    "reg_edges = regulation.build_edges(reg_network)\n",
    "reg_nodes = regulation.build_nodes(reg_network)\n",
    "\n",
    "#graph_nodes_list, reg_graph_edges = graph.graph_nodes(\n",
    "#    curation=curation_edges,\n",
    "#    monarch=monarch_edges,\n",
    "#    transcriptomics=rna_edges,\n",
    "#    regulation=reg_edges\n",
    "#)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(3209, 12)"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rna_network.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(3209, 12)"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.DataFrame(rna_edges).shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "The function \"graph_nodes()\" is running...\n",
      "\n",
      "Preparing networks...\n",
      "Curated:\n",
      "(2, 9)\n",
      "Index(['object_id', 'property_description', 'property_id', 'property_label',\n",
      "       'property_uri', 'reference_date', 'reference_supporting_text',\n",
      "       'reference_uri', 'subject_id'],\n",
      "      dtype='object')\n",
      "Monarch:\n",
      "(233182, 9)\n",
      "Index(['object_id', 'property_description', 'property_id', 'property_label',\n",
      "       'property_uri', 'reference_date', 'reference_supporting_text',\n",
      "       'reference_uri', 'subject_id'],\n",
      "      dtype='object')\n",
      "Transcriptomics:\n",
      "(3209, 12)\n",
      "Index(['fdr', 'log2FoldChange', 'object_id', 'object_label', 'property_id',\n",
      "       'property_label', 'pvalue', 'reference_id', 'regulation', 'source',\n",
      "       'subject_id', 'subject_label'],\n",
      "      dtype='object')\n",
      "Regulatory:\n",
      "(197267, 9)\n",
      "Index(['object_id', 'property_description', 'property_id', 'property_label',\n",
      "       'property_uri', 'reference_date', 'reference_supporting_text',\n",
      "       'reference_uri', 'subject_id'],\n",
      "      dtype='object')\n",
      "\n",
      "Concatenating into a graph...\n",
      "(236393, 4)\n",
      "\n",
      "Drop duplicated rows...\n",
      "(235387, 4)\n",
      "Index(['object_id', 'property_id', 'property_label', 'subject_id'], dtype='object')\n",
      "\n",
      "Merging tf-gene network to the graph...\n"
     ]
    },
    {
     "ename": "KeyError",
     "evalue": "\"['reference_uri_tf' 'reference_supporting_text_tf' 'reference_date_tf'\\n 'property_description_tf' 'property_uri_tf'] not in index\"",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32m~/Structured review/HD-SR/bioknowledge-reviewer/bioknowledge_reviewer/graph.py\u001b[0m in \u001b[0;36mgraph_nodes\u001b[0;34m(curation, monarch, transcriptomics, regulation, input_from_file)\u001b[0m\n\u001b[1;32m    151\u001b[0m     [['subject_id', 'property_id_tf', 'object_id_tf', 'reference_uri_tf',\n\u001b[1;32m    152\u001b[0m       \u001b[0;34m'reference_supporting_text_tf'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'reference_date_tf'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'property_label_tf'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 153\u001b[0;31m       'property_description_tf', 'property_uri_tf']]\n\u001b[0m\u001b[1;32m    154\u001b[0m         .rename(columns={\n\u001b[1;32m    155\u001b[0m         \u001b[0;34m'property_id_tf'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;34m'property_id'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/sr/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   2680\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mSeries\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mndarray\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mIndex\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlist\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2681\u001b[0m             \u001b[0;31m# either boolean or fancy integer index\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2682\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getitem_array\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2683\u001b[0m         \u001b[0;32melif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mDataFrame\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2684\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getitem_frame\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/sr/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m_getitem_array\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   2724\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_take\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindexer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2725\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2726\u001b[0;31m             \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_convert_to_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2727\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_take\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindexer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2728\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/sr/lib/python3.6/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m_convert_to_indexer\u001b[0;34m(self, obj, axis, is_setter)\u001b[0m\n\u001b[1;32m   1325\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0mmask\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0many\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1326\u001b[0m                     raise KeyError('{mask} not in index'\n\u001b[0;32m-> 1327\u001b[0;31m                                    .format(mask=objarr[mask]))\n\u001b[0m\u001b[1;32m   1328\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1329\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mcom\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_values_from_object\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindexer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: \"['reference_uri_tf' 'reference_supporting_text_tf' 'reference_date_tf'\\n 'property_description_tf' 'property_uri_tf'] not in index\""
     ]
    }
   ],
   "source": [
    "%%time\n",
    "# load networks and calculate graph nodes\n",
    "graph_nodes_list, reg_graph_edges = graph.graph_nodes(\n",
    "    curation=curation_edges,\n",
    "    monarch=monarch_edges,\n",
    "    transcriptomics=rna_edges,\n",
    "    regulation=reg_edges\n",
    ")\n",
    "\n",
    "# Monarch graph connectivity\n",
    "## get Monarch edges\n",
    "monarch_network_graph = monarch.extract_edges(graph_nodes_list)\n",
    "print('network: ',len(monarch_network_graph))\n",
    "\n",
    "## save Monarch network\n",
    "monarch.print_network(monarch_network_graph, 'monarch_connections_graph')\n",
    "\n",
    "## build Monarch network with graph schema\n",
    "monarch_graph_edges = monarch.build_edges(monarch_network_graph)\n",
    "monarch_graph_nodes = monarch.build_nodes(monarch_network_graph)\n",
    "\n",
    "# build review graph\n",
    "edges = graph.build_edges(\n",
    "    curation=curation_edges,\n",
    "    monarch=monarch_graph_edges,\n",
    "    transcriptomics=rna_edges,\n",
    "    regulation=reg_graph_edges\n",
    ")\n",
    "nodes = graph.build_nodes(\n",
    "    statements=edges,\n",
    "    curation=curation_nodes,\n",
    "    monarch=monarch_graph_nodes,\n",
    "    transcriptomics=rna_nodes,\n",
    "    regulation=reg_nodes\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "curated_df = pd.DataFrame(curation_edges)\n",
    "monarch_df = pd.DataFrame(monarch_edges)\n",
    "rna = pd.DataFrame(rna_edges)\n",
    "tf = pd.DataFrame(reg_edges)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>object_id</th>\n",
       "      <th>property_description</th>\n",
       "      <th>property_id</th>\n",
       "      <th>property_label</th>\n",
       "      <th>property_uri</th>\n",
       "      <th>reference_date</th>\n",
       "      <th>reference_supporting_text</th>\n",
       "      <th>reference_uri</th>\n",
       "      <th>subject_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   object_id  property_description  property_id  property_label  property_uri  \\\n",
       "0        NaN                   NaN          NaN             NaN           NaN   \n",
       "1        NaN                   NaN          NaN             NaN           NaN   \n",
       "\n",
       "   reference_date  reference_supporting_text  reference_uri  subject_id  \n",
       "0             NaN                        NaN            NaN         NaN  \n",
       "1             NaN                        NaN            NaN         NaN  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "curated_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>object_id</th>\n",
       "      <th>property_description</th>\n",
       "      <th>property_id</th>\n",
       "      <th>property_label</th>\n",
       "      <th>property_uri</th>\n",
       "      <th>reference_date</th>\n",
       "      <th>reference_supporting_text</th>\n",
       "      <th>reference_uri</th>\n",
       "      <th>subject_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MGI:892004</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>RGD:1595815</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MGI:3704134</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>ENSEMBL:ENSRNOG00000061639</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MGI:97912</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000020</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>FlyBase:FBgn0041191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSEMBL:ENSOANG00000006132</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000020</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>MGI:1344381</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSEMBL:ENSECAG00000019077</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000020</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>ENSEMBL:ENSCAFG00000017647</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MONARCH:APO_0000222APO_0000004</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002200</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>SGD:S000002780</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>FlyBase:FBgn0013275</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>FlyBase:FBgn0036505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ENSEMBL:ENSGALG00000044425</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>ENSEMBL:ENSCAFG00000016509</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>RGD:1359648</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000020</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>ENSEMBL:ENSSSCG00000038628</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ENSEMBL:ENSGALG00000042895</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000020</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>ENSEMBL:ENSMMUG00000002630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ENSEMBL:ENSCAFG00000010109</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>ENSEMBL:ENSOANG00000007850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>HGNC:2717</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>HGNC:7878</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ENSEMBL:ENSBTAG00000020963</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000020</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>ENSEMBL:ENSSSCG00000011407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>MGI:1276525</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>ZFIN:ZDB-GENE-030916-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ENSEMBL:ENSBTAG00000000919</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000020</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>RGD:621405</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ENSEMBL:ENSOANG00000013551</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000020</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>RGD:1565519</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ENSEMBL:ENSBTAG00000014149</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>ZFIN:ZDB-GENE-030911-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>ZFIN:ZDB-GENE-161101-1</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>ENSEMBL:ENSXETG00000009077</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ENSEMBL:ENSECAG00000022370</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000020</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>ZFIN:ZDB-GENE-050522-205</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>FlyBase:FBgn0035161</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>ENSEMBL:ENSMODG00000025146</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002200</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>HGNC:11551</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>MP:0008498</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002200</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>MGI:95805</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>HP:0000431</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002200</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>HGNC:7095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>ENSEMBL:ENSPTRG00000006169</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>WormBase:WBGene00017751</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>ENSEMBL:ENSOANG00000005852</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000020</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>HGNC:9852</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>ENSEMBL:ENSMODG00000018805</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000020</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>ENSEMBL:ENSSSCG00000038558</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>HGNC:16031</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000020</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>FlyBase:FBgn0031857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>GO:0005634</td>\n",
       "      <td>NA</td>\n",
       "      <td>BFO:0000050</td>\n",
       "      <td>is part of</td>\n",
       "      <td>http://purl.obolibrary.org/obo/BFO_0000050</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>dictyBase:DDB_G0286907</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>ENSEMBL:ENSSSCG00000014343</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000020</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>RGD:3052</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>RGD:3494</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>ZFIN:ZDB-GENE-990415-145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233152</th>\n",
       "      <td>ZFIN:ZDB-GENE-130814-1</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>ENSEMBL:ENSACAG00000003922</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233153</th>\n",
       "      <td>ENSEMBL:ENSRNOG00000030638</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>ENSEMBL:ENSCAFG00000018364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233154</th>\n",
       "      <td>ENSEMBL:ENSCAFG00000023721</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>PomBase:SPBC1734.11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233155</th>\n",
       "      <td>ENSEMBL:ENSCAFG00000003395</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>ENSEMBL:ENSRNOG00000030224</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233156</th>\n",
       "      <td>ENSEMBL:ENSCAFG00000012048</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000020</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>ENSEMBL:ENSGALG00000007653</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233157</th>\n",
       "      <td>RGD:631427</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000020</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>MGI:1915881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233158</th>\n",
       "      <td>SGD:S000003588</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>ENSEMBL:ENSCAFG00000012959</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233159</th>\n",
       "      <td>Xenbase:XB-GENE-5892016</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000020</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>ENSEMBL:ENSPTRG00000000484</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233160</th>\n",
       "      <td>MGI:1927140</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>MGI:1926884</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233161</th>\n",
       "      <td>ENSEMBL:ENSECAG00000000796</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>FlyBase:FBgn0051802</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233162</th>\n",
       "      <td>ENSEMBL:ENSECAG00000024558</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>FlyBase:FBgn0040233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233163</th>\n",
       "      <td>MGI:3647514</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>ENSEMBL:ENSXETG00000025015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233164</th>\n",
       "      <td>WormBase:WBGene00006865</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>SGD:S000003072</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233165</th>\n",
       "      <td>ENSEMBL:ENSACAG00000008113</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000020</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>ENSEMBL:ENSECAG00000010532</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233166</th>\n",
       "      <td>ENSEMBL:ENSBTAG00000036009</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000020</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>MGI:108072</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233167</th>\n",
       "      <td>HGNC:5961</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>HGNC:1507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233168</th>\n",
       "      <td>dictyBase:DDB_G0285265</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>ZFIN:ZDB-GENE-030804-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233169</th>\n",
       "      <td>UBERON:0000465</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002206</td>\n",
       "      <td>expressed in</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002206</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>ZFIN:ZDB-GENE-020423-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233170</th>\n",
       "      <td>ZP:0137430</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002200</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>ZFIN:ZDB-GENE-110621-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233171</th>\n",
       "      <td>MGI:2676637</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>ENSEMBL:ENSXETG00000009077</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233172</th>\n",
       "      <td>ENSEMBL:ENSRNOG00000047185</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>HGNC:3272</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233173</th>\n",
       "      <td>ENSEMBL:ENSECAG00000001400</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>MGI:1913331</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233174</th>\n",
       "      <td>ENSEMBL:ENSBTAG00000030416</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>FlyBase:FBgn0028563</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233175</th>\n",
       "      <td>ENSEMBL:ENSSSCG00000036199</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>SGD:S000001057</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233176</th>\n",
       "      <td>HGNC:17345</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>FlyBase:FBgn0030449</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233177</th>\n",
       "      <td>ZFIN:ZDB-GENE-030131-7310</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000020</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>RGD:1311415</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233178</th>\n",
       "      <td>ZFIN:ZDB-GENE-030131-8680</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>dictyBase:DDB_G0271282</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233179</th>\n",
       "      <td>UBERON:0000465</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002206</td>\n",
       "      <td>expressed in</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002206</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>ZFIN:ZDB-GENE-041008-78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233180</th>\n",
       "      <td>MGI:95915</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>ENSEMBL:ENSECAG00000035314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233181</th>\n",
       "      <td>ENSEMBL:ENSECAG00000035532</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "      <td>ENSEMBL:ENSXETG00000034352</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>233182 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             object_id property_description    property_id  \\\n",
       "0                           MGI:892004                   NA  RO:HOM0000017   \n",
       "1                          MGI:3704134                   NA  RO:HOM0000017   \n",
       "2                            MGI:97912                   NA  RO:HOM0000020   \n",
       "3           ENSEMBL:ENSOANG00000006132                   NA  RO:HOM0000020   \n",
       "4           ENSEMBL:ENSECAG00000019077                   NA  RO:HOM0000020   \n",
       "5       MONARCH:APO_0000222APO_0000004                   NA     RO:0002200   \n",
       "6                  FlyBase:FBgn0013275                   NA     RO:0002434   \n",
       "7           ENSEMBL:ENSGALG00000044425                   NA  RO:HOM0000017   \n",
       "8                          RGD:1359648                   NA  RO:HOM0000020   \n",
       "9           ENSEMBL:ENSGALG00000042895                   NA  RO:HOM0000020   \n",
       "10          ENSEMBL:ENSCAFG00000010109                   NA  RO:HOM0000017   \n",
       "11                           HGNC:2717                   NA     RO:0002434   \n",
       "12          ENSEMBL:ENSBTAG00000020963                   NA  RO:HOM0000020   \n",
       "13                         MGI:1276525                   NA  RO:HOM0000017   \n",
       "14          ENSEMBL:ENSBTAG00000000919                   NA  RO:HOM0000020   \n",
       "15          ENSEMBL:ENSOANG00000013551                   NA  RO:HOM0000020   \n",
       "16          ENSEMBL:ENSBTAG00000014149                   NA  RO:HOM0000017   \n",
       "17              ZFIN:ZDB-GENE-161101-1                   NA  RO:HOM0000017   \n",
       "18          ENSEMBL:ENSECAG00000022370                   NA  RO:HOM0000020   \n",
       "19                 FlyBase:FBgn0035161                   NA  RO:HOM0000017   \n",
       "20                          HP:0000343                   NA     RO:0002200   \n",
       "21                          MP:0008498                   NA     RO:0002200   \n",
       "22                          HP:0000431                   NA     RO:0002200   \n",
       "23          ENSEMBL:ENSPTRG00000006169                   NA  RO:HOM0000017   \n",
       "24          ENSEMBL:ENSOANG00000005852                   NA  RO:HOM0000020   \n",
       "25          ENSEMBL:ENSMODG00000018805                   NA  RO:HOM0000020   \n",
       "26                          HGNC:16031                   NA  RO:HOM0000020   \n",
       "27                          GO:0005634                   NA    BFO:0000050   \n",
       "28          ENSEMBL:ENSSSCG00000014343                   NA  RO:HOM0000020   \n",
       "29                            RGD:3494                   NA  RO:HOM0000017   \n",
       "...                                ...                  ...            ...   \n",
       "233152          ZFIN:ZDB-GENE-130814-1                   NA  RO:HOM0000017   \n",
       "233153      ENSEMBL:ENSRNOG00000030638                   NA  RO:HOM0000017   \n",
       "233154      ENSEMBL:ENSCAFG00000023721                   NA  RO:HOM0000017   \n",
       "233155      ENSEMBL:ENSCAFG00000003395                   NA  RO:HOM0000017   \n",
       "233156      ENSEMBL:ENSCAFG00000012048                   NA  RO:HOM0000020   \n",
       "233157                      RGD:631427                   NA  RO:HOM0000020   \n",
       "233158                  SGD:S000003588                   NA  RO:HOM0000017   \n",
       "233159         Xenbase:XB-GENE-5892016                   NA  RO:HOM0000020   \n",
       "233160                     MGI:1927140                   NA     RO:0002434   \n",
       "233161      ENSEMBL:ENSECAG00000000796                   NA  RO:HOM0000017   \n",
       "233162      ENSEMBL:ENSECAG00000024558                   NA  RO:HOM0000017   \n",
       "233163                     MGI:3647514                   NA  RO:HOM0000017   \n",
       "233164         WormBase:WBGene00006865                   NA  RO:HOM0000017   \n",
       "233165      ENSEMBL:ENSACAG00000008113                   NA  RO:HOM0000020   \n",
       "233166      ENSEMBL:ENSBTAG00000036009                   NA  RO:HOM0000020   \n",
       "233167                       HGNC:5961                   NA     RO:0002434   \n",
       "233168          dictyBase:DDB_G0285265                   NA  RO:HOM0000017   \n",
       "233169                  UBERON:0000465                   NA     RO:0002206   \n",
       "233170                      ZP:0137430                   NA     RO:0002200   \n",
       "233171                     MGI:2676637                   NA  RO:HOM0000017   \n",
       "233172      ENSEMBL:ENSRNOG00000047185                   NA  RO:HOM0000017   \n",
       "233173      ENSEMBL:ENSECAG00000001400                   NA  RO:HOM0000017   \n",
       "233174      ENSEMBL:ENSBTAG00000030416                   NA  RO:HOM0000017   \n",
       "233175      ENSEMBL:ENSSSCG00000036199                   NA  RO:HOM0000017   \n",
       "233176                      HGNC:17345                   NA  RO:HOM0000017   \n",
       "233177       ZFIN:ZDB-GENE-030131-7310                   NA  RO:HOM0000020   \n",
       "233178       ZFIN:ZDB-GENE-030131-8680                   NA  RO:HOM0000017   \n",
       "233179                  UBERON:0000465                   NA     RO:0002206   \n",
       "233180                       MGI:95915                   NA  RO:HOM0000017   \n",
       "233181      ENSEMBL:ENSECAG00000035532                   NA  RO:HOM0000017   \n",
       "\n",
       "                               property_label  \\\n",
       "0              in orthology relationship with   \n",
       "1              in orthology relationship with   \n",
       "2       in 1 to 1 orthology relationship with   \n",
       "3       in 1 to 1 orthology relationship with   \n",
       "4       in 1 to 1 orthology relationship with   \n",
       "5                               has phenotype   \n",
       "6                              interacts with   \n",
       "7              in orthology relationship with   \n",
       "8       in 1 to 1 orthology relationship with   \n",
       "9       in 1 to 1 orthology relationship with   \n",
       "10             in orthology relationship with   \n",
       "11                             interacts with   \n",
       "12      in 1 to 1 orthology relationship with   \n",
       "13             in orthology relationship with   \n",
       "14      in 1 to 1 orthology relationship with   \n",
       "15      in 1 to 1 orthology relationship with   \n",
       "16             in orthology relationship with   \n",
       "17             in orthology relationship with   \n",
       "18      in 1 to 1 orthology relationship with   \n",
       "19             in orthology relationship with   \n",
       "20                              has phenotype   \n",
       "21                              has phenotype   \n",
       "22                              has phenotype   \n",
       "23             in orthology relationship with   \n",
       "24      in 1 to 1 orthology relationship with   \n",
       "25      in 1 to 1 orthology relationship with   \n",
       "26      in 1 to 1 orthology relationship with   \n",
       "27                                 is part of   \n",
       "28      in 1 to 1 orthology relationship with   \n",
       "29             in orthology relationship with   \n",
       "...                                       ...   \n",
       "233152         in orthology relationship with   \n",
       "233153         in orthology relationship with   \n",
       "233154         in orthology relationship with   \n",
       "233155         in orthology relationship with   \n",
       "233156  in 1 to 1 orthology relationship with   \n",
       "233157  in 1 to 1 orthology relationship with   \n",
       "233158         in orthology relationship with   \n",
       "233159  in 1 to 1 orthology relationship with   \n",
       "233160                         interacts with   \n",
       "233161         in orthology relationship with   \n",
       "233162         in orthology relationship with   \n",
       "233163         in orthology relationship with   \n",
       "233164         in orthology relationship with   \n",
       "233165  in 1 to 1 orthology relationship with   \n",
       "233166  in 1 to 1 orthology relationship with   \n",
       "233167                         interacts with   \n",
       "233168         in orthology relationship with   \n",
       "233169                           expressed in   \n",
       "233170                          has phenotype   \n",
       "233171         in orthology relationship with   \n",
       "233172         in orthology relationship with   \n",
       "233173         in orthology relationship with   \n",
       "233174         in orthology relationship with   \n",
       "233175         in orthology relationship with   \n",
       "233176         in orthology relationship with   \n",
       "233177  in 1 to 1 orthology relationship with   \n",
       "233178         in orthology relationship with   \n",
       "233179                           expressed in   \n",
       "233180         in orthology relationship with   \n",
       "233181         in orthology relationship with   \n",
       "\n",
       "                                        property_uri reference_date  \\\n",
       "0       http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
       "1       http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
       "2       http://purl.obolibrary.org/obo/RO_HOM0000020             NA   \n",
       "3       http://purl.obolibrary.org/obo/RO_HOM0000020             NA   \n",
       "4       http://purl.obolibrary.org/obo/RO_HOM0000020             NA   \n",
       "5          http://purl.obolibrary.org/obo/RO_0002200             NA   \n",
       "6          http://purl.obolibrary.org/obo/RO_0002434             NA   \n",
       "7       http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
       "8       http://purl.obolibrary.org/obo/RO_HOM0000020             NA   \n",
       "9       http://purl.obolibrary.org/obo/RO_HOM0000020             NA   \n",
       "10      http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
       "11         http://purl.obolibrary.org/obo/RO_0002434             NA   \n",
       "12      http://purl.obolibrary.org/obo/RO_HOM0000020             NA   \n",
       "13      http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
       "14      http://purl.obolibrary.org/obo/RO_HOM0000020             NA   \n",
       "15      http://purl.obolibrary.org/obo/RO_HOM0000020             NA   \n",
       "16      http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
       "17      http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
       "18      http://purl.obolibrary.org/obo/RO_HOM0000020             NA   \n",
       "19      http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
       "20         http://purl.obolibrary.org/obo/RO_0002200             NA   \n",
       "21         http://purl.obolibrary.org/obo/RO_0002200             NA   \n",
       "22         http://purl.obolibrary.org/obo/RO_0002200             NA   \n",
       "23      http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
       "24      http://purl.obolibrary.org/obo/RO_HOM0000020             NA   \n",
       "25      http://purl.obolibrary.org/obo/RO_HOM0000020             NA   \n",
       "26      http://purl.obolibrary.org/obo/RO_HOM0000020             NA   \n",
       "27        http://purl.obolibrary.org/obo/BFO_0000050             NA   \n",
       "28      http://purl.obolibrary.org/obo/RO_HOM0000020             NA   \n",
       "29      http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
       "...                                              ...            ...   \n",
       "233152  http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
       "233153  http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
       "233154  http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
       "233155  http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
       "233156  http://purl.obolibrary.org/obo/RO_HOM0000020             NA   \n",
       "233157  http://purl.obolibrary.org/obo/RO_HOM0000020             NA   \n",
       "233158  http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
       "233159  http://purl.obolibrary.org/obo/RO_HOM0000020             NA   \n",
       "233160     http://purl.obolibrary.org/obo/RO_0002434             NA   \n",
       "233161  http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
       "233162  http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
       "233163  http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
       "233164  http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
       "233165  http://purl.obolibrary.org/obo/RO_HOM0000020             NA   \n",
       "233166  http://purl.obolibrary.org/obo/RO_HOM0000020             NA   \n",
       "233167     http://purl.obolibrary.org/obo/RO_0002434             NA   \n",
       "233168  http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
       "233169     http://purl.obolibrary.org/obo/RO_0002206             NA   \n",
       "233170     http://purl.obolibrary.org/obo/RO_0002200             NA   \n",
       "233171  http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
       "233172  http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
       "233173  http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
       "233174  http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
       "233175  http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
       "233176  http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
       "233177  http://purl.obolibrary.org/obo/RO_HOM0000020             NA   \n",
       "233178  http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
       "233179     http://purl.obolibrary.org/obo/RO_0002206             NA   \n",
       "233180  http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
       "233181  http://purl.obolibrary.org/obo/RO_HOM0000017             NA   \n",
       "\n",
       "                                reference_supporting_text reference_uri  \\\n",
       "0       This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "1       This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "2       This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "3       This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "4       This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "5       This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "6       This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "7       This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "8       This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "9       This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "10      This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "11      This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "12      This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "13      This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "14      This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "15      This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "16      This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "17      This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "18      This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "19      This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "20      This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "21      This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "22      This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "23      This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "24      This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "25      This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "26      This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "27      This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "28      This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "29      This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "...                                                   ...           ...   \n",
       "233152  This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "233153  This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "233154  This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "233155  This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "233156  This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "233157  This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "233158  This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "233159  This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "233160  This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "233161  This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "233162  This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "233163  This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "233164  This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "233165  This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "233166  This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "233167  This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "233168  This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "233169  This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "233170  This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "233171  This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "233172  This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "233173  This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "233174  This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "233175  This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "233176  This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "233177  This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "233178  This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "233179  This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "233180  This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "233181  This edge comes from the Monarch Knowledge Gra...            NA   \n",
       "\n",
       "                        subject_id  \n",
       "0                      RGD:1595815  \n",
       "1       ENSEMBL:ENSRNOG00000061639  \n",
       "2              FlyBase:FBgn0041191  \n",
       "3                      MGI:1344381  \n",
       "4       ENSEMBL:ENSCAFG00000017647  \n",
       "5                   SGD:S000002780  \n",
       "6              FlyBase:FBgn0036505  \n",
       "7       ENSEMBL:ENSCAFG00000016509  \n",
       "8       ENSEMBL:ENSSSCG00000038628  \n",
       "9       ENSEMBL:ENSMMUG00000002630  \n",
       "10      ENSEMBL:ENSOANG00000007850  \n",
       "11                       HGNC:7878  \n",
       "12      ENSEMBL:ENSSSCG00000011407  \n",
       "13          ZFIN:ZDB-GENE-030916-1  \n",
       "14                      RGD:621405  \n",
       "15                     RGD:1565519  \n",
       "16          ZFIN:ZDB-GENE-030911-3  \n",
       "17      ENSEMBL:ENSXETG00000009077  \n",
       "18        ZFIN:ZDB-GENE-050522-205  \n",
       "19      ENSEMBL:ENSMODG00000025146  \n",
       "20                      HGNC:11551  \n",
       "21                       MGI:95805  \n",
       "22                       HGNC:7095  \n",
       "23         WormBase:WBGene00017751  \n",
       "24                       HGNC:9852  \n",
       "25      ENSEMBL:ENSSSCG00000038558  \n",
       "26             FlyBase:FBgn0031857  \n",
       "27          dictyBase:DDB_G0286907  \n",
       "28                        RGD:3052  \n",
       "29        ZFIN:ZDB-GENE-990415-145  \n",
       "...                            ...  \n",
       "233152  ENSEMBL:ENSACAG00000003922  \n",
       "233153  ENSEMBL:ENSCAFG00000018364  \n",
       "233154         PomBase:SPBC1734.11  \n",
       "233155  ENSEMBL:ENSRNOG00000030224  \n",
       "233156  ENSEMBL:ENSGALG00000007653  \n",
       "233157                 MGI:1915881  \n",
       "233158  ENSEMBL:ENSCAFG00000012959  \n",
       "233159  ENSEMBL:ENSPTRG00000000484  \n",
       "233160                 MGI:1926884  \n",
       "233161         FlyBase:FBgn0051802  \n",
       "233162         FlyBase:FBgn0040233  \n",
       "233163  ENSEMBL:ENSXETG00000025015  \n",
       "233164              SGD:S000003072  \n",
       "233165  ENSEMBL:ENSECAG00000010532  \n",
       "233166                  MGI:108072  \n",
       "233167                   HGNC:1507  \n",
       "233168      ZFIN:ZDB-GENE-030804-3  \n",
       "233169      ZFIN:ZDB-GENE-020423-3  \n",
       "233170      ZFIN:ZDB-GENE-110621-1  \n",
       "233171  ENSEMBL:ENSXETG00000009077  \n",
       "233172                   HGNC:3272  \n",
       "233173                 MGI:1913331  \n",
       "233174         FlyBase:FBgn0028563  \n",
       "233175              SGD:S000001057  \n",
       "233176         FlyBase:FBgn0030449  \n",
       "233177                 RGD:1311415  \n",
       "233178      dictyBase:DDB_G0271282  \n",
       "233179     ZFIN:ZDB-GENE-041008-78  \n",
       "233180  ENSEMBL:ENSECAG00000035314  \n",
       "233181  ENSEMBL:ENSXETG00000034352  \n",
       "\n",
       "[233182 rows x 9 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "monarch_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fdr</th>\n",
       "      <th>log2FoldChange</th>\n",
       "      <th>object_id</th>\n",
       "      <th>object_label</th>\n",
       "      <th>property_id</th>\n",
       "      <th>property_label</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>reference_id</th>\n",
       "      <th>regulation</th>\n",
       "      <th>source</th>\n",
       "      <th>subject_id</th>\n",
       "      <th>subject_label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2.687232e-34</td>\n",
       "      <td>4.769658</td>\n",
       "      <td>HGNC:9004</td>\n",
       "      <td>PITX1</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>9.567529e-39</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2.287927e-21</td>\n",
       "      <td>4.760790</td>\n",
       "      <td>HGNC:5120</td>\n",
       "      <td>HOXB9</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>1.629171e-25</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2.724842e-20</td>\n",
       "      <td>4.573976</td>\n",
       "      <td>HGNC:5122</td>\n",
       "      <td>HOXC10</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>2.910430e-24</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2.752160e-20</td>\n",
       "      <td>4.704005</td>\n",
       "      <td>HGNC:5101</td>\n",
       "      <td>HOXA11</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>3.919478e-24</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4.511536e-20</td>\n",
       "      <td>4.273311</td>\n",
       "      <td>HGNC:5100</td>\n",
       "      <td>HOXA10</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>8.031360e-24</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>6.329454e-20</td>\n",
       "      <td>4.602451</td>\n",
       "      <td>HGNC:5133</td>\n",
       "      <td>HOXD10</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>1.352110e-23</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1.371856e-19</td>\n",
       "      <td>3.962235</td>\n",
       "      <td>HGNC:9219</td>\n",
       "      <td>POU4F2</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>3.419015e-23</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2.176886e-19</td>\n",
       "      <td>4.165899</td>\n",
       "      <td>HGNC:5102</td>\n",
       "      <td>HOXA13</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>6.200410e-23</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>3.799669e-15</td>\n",
       "      <td>3.657288</td>\n",
       "      <td>HGNC:5140</td>\n",
       "      <td>HOXD9</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>1.217539e-18</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>5.876207e-15</td>\n",
       "      <td>3.866840</td>\n",
       "      <td>HGNC:5139</td>\n",
       "      <td>HOXD8</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>2.092145e-18</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>1.108606e-14</td>\n",
       "      <td>3.879624</td>\n",
       "      <td>HGNC:5106</td>\n",
       "      <td>HOXA5</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>4.494762e-18</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>1.108606e-14</td>\n",
       "      <td>3.513877</td>\n",
       "      <td>HGNC:23094</td>\n",
       "      <td>SLC16A12</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>4.736450e-18</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>3.160477e-14</td>\n",
       "      <td>3.703297</td>\n",
       "      <td>HGNC:4807</td>\n",
       "      <td>HAND1</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>1.462819e-17</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>7.882863e-14</td>\n",
       "      <td>2.998538</td>\n",
       "      <td>HGNC:8518</td>\n",
       "      <td>OTP</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>3.929223e-17</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>7.117871e-13</td>\n",
       "      <td>1.392757</td>\n",
       "      <td>HGNC:18015</td>\n",
       "      <td>IL17RB</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>3.801334e-16</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>4.366832e-12</td>\n",
       "      <td>2.355393</td>\n",
       "      <td>HGNC:30927</td>\n",
       "      <td>SLC6A20</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>2.487603e-15</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>7.042879e-12</td>\n",
       "      <td>3.608891</td>\n",
       "      <td>HGNC:5128</td>\n",
       "      <td>HOXC6</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>4.262788e-15</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>9.198690e-12</td>\n",
       "      <td>2.490891</td>\n",
       "      <td>HGNC:12706</td>\n",
       "      <td>VNN2</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>6.032251e-15</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>9.198690e-12</td>\n",
       "      <td>1.419149</td>\n",
       "      <td>HGNC:9441</td>\n",
       "      <td>PRKX</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>6.222633e-15</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>1.136339e-11</td>\n",
       "      <td>2.068673</td>\n",
       "      <td>HGNC:4871</td>\n",
       "      <td>HERC2P3</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>8.091568e-15</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>7.152041e-11</td>\n",
       "      <td>2.151887</td>\n",
       "      <td>HGNC:2338</td>\n",
       "      <td>CRABP1</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>5.347415e-14</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>7.718099e-11</td>\n",
       "      <td>2.023018</td>\n",
       "      <td>HGNC:5986</td>\n",
       "      <td>IL18</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>6.045437e-14</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>9.457051e-11</td>\n",
       "      <td>2.590579</td>\n",
       "      <td>HGNC:10854</td>\n",
       "      <td>SHOX2</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>7.744230e-14</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>9.590667e-11</td>\n",
       "      <td>3.096562</td>\n",
       "      <td>HGNC:8126</td>\n",
       "      <td>OGN</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>8.195109e-14</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>9.910100e-11</td>\n",
       "      <td>2.597537</td>\n",
       "      <td>HGNC:7176</td>\n",
       "      <td>MMP9</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>8.820895e-14</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>1.254850e-10</td>\n",
       "      <td>0.787522</td>\n",
       "      <td>HGNC:17278</td>\n",
       "      <td>PNRC1</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>1.161608e-13</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>1.482659e-10</td>\n",
       "      <td>2.078430</td>\n",
       "      <td>HGNC:8134</td>\n",
       "      <td>OMD</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>1.425279e-13</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>2.351391e-10</td>\n",
       "      <td>1.784825</td>\n",
       "      <td>HGNC:11848</td>\n",
       "      <td>TLR2</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>2.427826e-13</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>2.831322e-10</td>\n",
       "      <td>1.312784</td>\n",
       "      <td>HGNC:13448</td>\n",
       "      <td>SLC38A2</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>3.024163e-13</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>3.596331e-10</td>\n",
       "      <td>3.274237</td>\n",
       "      <td>HGNC:24957</td>\n",
       "      <td>HOXA11-AS</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>3.969319e-13</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Upregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3179</th>\n",
       "      <td>4.738907e-02</td>\n",
       "      <td>-0.891618</td>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>CTD-3025N20.2</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>9.024107e-03</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3180</th>\n",
       "      <td>4.748682e-02</td>\n",
       "      <td>-0.796360</td>\n",
       "      <td>HGNC:8672</td>\n",
       "      <td>PCDHA6</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>9.050342e-03</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3181</th>\n",
       "      <td>4.754284e-02</td>\n",
       "      <td>-0.782935</td>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>CTD-3096M3.2</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>9.064403e-03</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3182</th>\n",
       "      <td>4.759288e-02</td>\n",
       "      <td>-0.770710</td>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>OR13E1P</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>9.077333e-03</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3183</th>\n",
       "      <td>4.774771e-02</td>\n",
       "      <td>-1.058550</td>\n",
       "      <td>HGNC:1570</td>\n",
       "      <td>CCKAR</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>9.120464e-03</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3184</th>\n",
       "      <td>4.791022e-02</td>\n",
       "      <td>-1.168875</td>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>Y_RNA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>9.156623e-03</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3185</th>\n",
       "      <td>4.796663e-02</td>\n",
       "      <td>-0.592199</td>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RP4-713B5.2</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>9.169112e-03</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3186</th>\n",
       "      <td>4.809417e-02</td>\n",
       "      <td>-1.124758</td>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RP11-567L7.5</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>9.207190e-03</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3187</th>\n",
       "      <td>4.817799e-02</td>\n",
       "      <td>-0.756181</td>\n",
       "      <td>HGNC:49384</td>\n",
       "      <td>AC129929.5</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>9.239551e-03</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3188</th>\n",
       "      <td>4.820094e-02</td>\n",
       "      <td>-0.788810</td>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>AC079341.1</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>9.249940e-03</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3189</th>\n",
       "      <td>4.853107e-02</td>\n",
       "      <td>-1.165349</td>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>KRT8P15</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>9.342667e-03</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3190</th>\n",
       "      <td>4.862513e-02</td>\n",
       "      <td>-1.256537</td>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RP11-264E20.2</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>9.365968e-03</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3191</th>\n",
       "      <td>4.881452e-02</td>\n",
       "      <td>-0.862867</td>\n",
       "      <td>HGNC:6814</td>\n",
       "      <td>MAGEL2</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>9.419827e-03</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3192</th>\n",
       "      <td>4.881452e-02</td>\n",
       "      <td>-1.127966</td>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RP11-787D11.1</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>9.419447e-03</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3193</th>\n",
       "      <td>4.881452e-02</td>\n",
       "      <td>-0.809353</td>\n",
       "      <td>HGNC:51822</td>\n",
       "      <td>RP11-214C8.2</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>9.415942e-03</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3194</th>\n",
       "      <td>4.887791e-02</td>\n",
       "      <td>-0.949250</td>\n",
       "      <td>HGNC:4941</td>\n",
       "      <td>HLA-DPB2</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>9.440761e-03</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3195</th>\n",
       "      <td>4.897870e-02</td>\n",
       "      <td>-0.824467</td>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>C12orf61</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>9.463717e-03</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3196</th>\n",
       "      <td>4.904817e-02</td>\n",
       "      <td>-0.701350</td>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RP11-1055B8.3</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>9.484124e-03</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3197</th>\n",
       "      <td>4.907942e-02</td>\n",
       "      <td>-1.197316</td>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RN7SL481P</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>9.491914e-03</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3198</th>\n",
       "      <td>4.924376e-02</td>\n",
       "      <td>-1.197465</td>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RP11-390K5.1</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>9.534218e-03</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3199</th>\n",
       "      <td>4.929435e-02</td>\n",
       "      <td>-0.596920</td>\n",
       "      <td>HGNC:2356</td>\n",
       "      <td>CRHBP</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>9.545768e-03</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3200</th>\n",
       "      <td>4.929821e-02</td>\n",
       "      <td>-1.125786</td>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RP11-435O5.6</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>9.550025e-03</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3201</th>\n",
       "      <td>4.932068e-02</td>\n",
       "      <td>-0.599633</td>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RP11-264M12.2</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>9.564230e-03</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3202</th>\n",
       "      <td>4.975431e-02</td>\n",
       "      <td>-1.244305</td>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RP11-9B6.1</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>9.673809e-03</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3203</th>\n",
       "      <td>4.978978e-02</td>\n",
       "      <td>-1.217940</td>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RP11-143A22.1</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>9.685690e-03</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3204</th>\n",
       "      <td>4.978978e-02</td>\n",
       "      <td>-0.931719</td>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RP11-867G2.8</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>9.686025e-03</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3205</th>\n",
       "      <td>4.979908e-02</td>\n",
       "      <td>-0.800030</td>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RP4-682C21.2</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>9.690303e-03</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3206</th>\n",
       "      <td>4.979908e-02</td>\n",
       "      <td>-0.798980</td>\n",
       "      <td>HGNC:53465</td>\n",
       "      <td>AC005592.2</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>9.694926e-03</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3207</th>\n",
       "      <td>4.988450e-02</td>\n",
       "      <td>-0.888370</td>\n",
       "      <td>HGNC:15722</td>\n",
       "      <td>CPA5</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>9.715108e-03</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3208</th>\n",
       "      <td>4.991671e-02</td>\n",
       "      <td>-1.065938</td>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RP11-69I8.3</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>9.735598e-03</td>\n",
       "      <td>PMID:26636579</td>\n",
       "      <td>Downregulated</td>\n",
       "      <td>Chow</td>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HTT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3209 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               fdr  log2FoldChange   object_id   object_label property_id  \\\n",
       "0     2.687232e-34        4.769658   HGNC:9004          PITX1  RO:0002434   \n",
       "1     2.287927e-21        4.760790   HGNC:5120          HOXB9  RO:0002434   \n",
       "2     2.724842e-20        4.573976   HGNC:5122         HOXC10  RO:0002434   \n",
       "3     2.752160e-20        4.704005   HGNC:5101         HOXA11  RO:0002434   \n",
       "4     4.511536e-20        4.273311   HGNC:5100         HOXA10  RO:0002434   \n",
       "5     6.329454e-20        4.602451   HGNC:5133         HOXD10  RO:0002434   \n",
       "6     1.371856e-19        3.962235   HGNC:9219         POU4F2  RO:0002434   \n",
       "7     2.176886e-19        4.165899   HGNC:5102         HOXA13  RO:0002434   \n",
       "8     3.799669e-15        3.657288   HGNC:5140          HOXD9  RO:0002434   \n",
       "9     5.876207e-15        3.866840   HGNC:5139          HOXD8  RO:0002434   \n",
       "10    1.108606e-14        3.879624   HGNC:5106          HOXA5  RO:0002434   \n",
       "11    1.108606e-14        3.513877  HGNC:23094       SLC16A12  RO:0002434   \n",
       "12    3.160477e-14        3.703297   HGNC:4807          HAND1  RO:0002434   \n",
       "13    7.882863e-14        2.998538   HGNC:8518            OTP  RO:0002434   \n",
       "14    7.117871e-13        1.392757  HGNC:18015         IL17RB  RO:0002434   \n",
       "15    4.366832e-12        2.355393  HGNC:30927        SLC6A20  RO:0002434   \n",
       "16    7.042879e-12        3.608891   HGNC:5128          HOXC6  RO:0002434   \n",
       "17    9.198690e-12        2.490891  HGNC:12706           VNN2  RO:0002434   \n",
       "18    9.198690e-12        1.419149   HGNC:9441           PRKX  RO:0002434   \n",
       "19    1.136339e-11        2.068673   HGNC:4871        HERC2P3  RO:0002434   \n",
       "20    7.152041e-11        2.151887   HGNC:2338         CRABP1  RO:0002434   \n",
       "21    7.718099e-11        2.023018   HGNC:5986           IL18  RO:0002434   \n",
       "22    9.457051e-11        2.590579  HGNC:10854          SHOX2  RO:0002434   \n",
       "23    9.590667e-11        3.096562   HGNC:8126            OGN  RO:0002434   \n",
       "24    9.910100e-11        2.597537   HGNC:7176           MMP9  RO:0002434   \n",
       "25    1.254850e-10        0.787522  HGNC:17278          PNRC1  RO:0002434   \n",
       "26    1.482659e-10        2.078430   HGNC:8134            OMD  RO:0002434   \n",
       "27    2.351391e-10        1.784825  HGNC:11848           TLR2  RO:0002434   \n",
       "28    2.831322e-10        1.312784  HGNC:13448        SLC38A2  RO:0002434   \n",
       "29    3.596331e-10        3.274237  HGNC:24957      HOXA11-AS  RO:0002434   \n",
       "...            ...             ...         ...            ...         ...   \n",
       "3179  4.738907e-02       -0.891618    HGNC:nan  CTD-3025N20.2  RO:0002434   \n",
       "3180  4.748682e-02       -0.796360   HGNC:8672         PCDHA6  RO:0002434   \n",
       "3181  4.754284e-02       -0.782935    HGNC:nan   CTD-3096M3.2  RO:0002434   \n",
       "3182  4.759288e-02       -0.770710    HGNC:nan        OR13E1P  RO:0002434   \n",
       "3183  4.774771e-02       -1.058550   HGNC:1570          CCKAR  RO:0002434   \n",
       "3184  4.791022e-02       -1.168875    HGNC:nan          Y_RNA  RO:0002434   \n",
       "3185  4.796663e-02       -0.592199    HGNC:nan    RP4-713B5.2  RO:0002434   \n",
       "3186  4.809417e-02       -1.124758    HGNC:nan   RP11-567L7.5  RO:0002434   \n",
       "3187  4.817799e-02       -0.756181  HGNC:49384     AC129929.5  RO:0002434   \n",
       "3188  4.820094e-02       -0.788810    HGNC:nan     AC079341.1  RO:0002434   \n",
       "3189  4.853107e-02       -1.165349    HGNC:nan        KRT8P15  RO:0002434   \n",
       "3190  4.862513e-02       -1.256537    HGNC:nan  RP11-264E20.2  RO:0002434   \n",
       "3191  4.881452e-02       -0.862867   HGNC:6814         MAGEL2  RO:0002434   \n",
       "3192  4.881452e-02       -1.127966    HGNC:nan  RP11-787D11.1  RO:0002434   \n",
       "3193  4.881452e-02       -0.809353  HGNC:51822   RP11-214C8.2  RO:0002434   \n",
       "3194  4.887791e-02       -0.949250   HGNC:4941       HLA-DPB2  RO:0002434   \n",
       "3195  4.897870e-02       -0.824467    HGNC:nan       C12orf61  RO:0002434   \n",
       "3196  4.904817e-02       -0.701350    HGNC:nan  RP11-1055B8.3  RO:0002434   \n",
       "3197  4.907942e-02       -1.197316    HGNC:nan      RN7SL481P  RO:0002434   \n",
       "3198  4.924376e-02       -1.197465    HGNC:nan   RP11-390K5.1  RO:0002434   \n",
       "3199  4.929435e-02       -0.596920   HGNC:2356          CRHBP  RO:0002434   \n",
       "3200  4.929821e-02       -1.125786    HGNC:nan   RP11-435O5.6  RO:0002434   \n",
       "3201  4.932068e-02       -0.599633    HGNC:nan  RP11-264M12.2  RO:0002434   \n",
       "3202  4.975431e-02       -1.244305    HGNC:nan     RP11-9B6.1  RO:0002434   \n",
       "3203  4.978978e-02       -1.217940    HGNC:nan  RP11-143A22.1  RO:0002434   \n",
       "3204  4.978978e-02       -0.931719    HGNC:nan   RP11-867G2.8  RO:0002434   \n",
       "3205  4.979908e-02       -0.800030    HGNC:nan   RP4-682C21.2  RO:0002434   \n",
       "3206  4.979908e-02       -0.798980  HGNC:53465     AC005592.2  RO:0002434   \n",
       "3207  4.988450e-02       -0.888370  HGNC:15722           CPA5  RO:0002434   \n",
       "3208  4.991671e-02       -1.065938    HGNC:nan    RP11-69I8.3  RO:0002434   \n",
       "\n",
       "      property_label        pvalue   reference_id     regulation source  \\\n",
       "0     interacts with  9.567529e-39  PMID:26636579    Upregulated   Chow   \n",
       "1     interacts with  1.629171e-25  PMID:26636579    Upregulated   Chow   \n",
       "2     interacts with  2.910430e-24  PMID:26636579    Upregulated   Chow   \n",
       "3     interacts with  3.919478e-24  PMID:26636579    Upregulated   Chow   \n",
       "4     interacts with  8.031360e-24  PMID:26636579    Upregulated   Chow   \n",
       "5     interacts with  1.352110e-23  PMID:26636579    Upregulated   Chow   \n",
       "6     interacts with  3.419015e-23  PMID:26636579    Upregulated   Chow   \n",
       "7     interacts with  6.200410e-23  PMID:26636579    Upregulated   Chow   \n",
       "8     interacts with  1.217539e-18  PMID:26636579    Upregulated   Chow   \n",
       "9     interacts with  2.092145e-18  PMID:26636579    Upregulated   Chow   \n",
       "10    interacts with  4.494762e-18  PMID:26636579    Upregulated   Chow   \n",
       "11    interacts with  4.736450e-18  PMID:26636579    Upregulated   Chow   \n",
       "12    interacts with  1.462819e-17  PMID:26636579    Upregulated   Chow   \n",
       "13    interacts with  3.929223e-17  PMID:26636579    Upregulated   Chow   \n",
       "14    interacts with  3.801334e-16  PMID:26636579    Upregulated   Chow   \n",
       "15    interacts with  2.487603e-15  PMID:26636579    Upregulated   Chow   \n",
       "16    interacts with  4.262788e-15  PMID:26636579    Upregulated   Chow   \n",
       "17    interacts with  6.032251e-15  PMID:26636579    Upregulated   Chow   \n",
       "18    interacts with  6.222633e-15  PMID:26636579    Upregulated   Chow   \n",
       "19    interacts with  8.091568e-15  PMID:26636579    Upregulated   Chow   \n",
       "20    interacts with  5.347415e-14  PMID:26636579    Upregulated   Chow   \n",
       "21    interacts with  6.045437e-14  PMID:26636579    Upregulated   Chow   \n",
       "22    interacts with  7.744230e-14  PMID:26636579    Upregulated   Chow   \n",
       "23    interacts with  8.195109e-14  PMID:26636579    Upregulated   Chow   \n",
       "24    interacts with  8.820895e-14  PMID:26636579    Upregulated   Chow   \n",
       "25    interacts with  1.161608e-13  PMID:26636579    Upregulated   Chow   \n",
       "26    interacts with  1.425279e-13  PMID:26636579    Upregulated   Chow   \n",
       "27    interacts with  2.427826e-13  PMID:26636579    Upregulated   Chow   \n",
       "28    interacts with  3.024163e-13  PMID:26636579    Upregulated   Chow   \n",
       "29    interacts with  3.969319e-13  PMID:26636579    Upregulated   Chow   \n",
       "...              ...           ...            ...            ...    ...   \n",
       "3179  interacts with  9.024107e-03  PMID:26636579  Downregulated   Chow   \n",
       "3180  interacts with  9.050342e-03  PMID:26636579  Downregulated   Chow   \n",
       "3181  interacts with  9.064403e-03  PMID:26636579  Downregulated   Chow   \n",
       "3182  interacts with  9.077333e-03  PMID:26636579  Downregulated   Chow   \n",
       "3183  interacts with  9.120464e-03  PMID:26636579  Downregulated   Chow   \n",
       "3184  interacts with  9.156623e-03  PMID:26636579  Downregulated   Chow   \n",
       "3185  interacts with  9.169112e-03  PMID:26636579  Downregulated   Chow   \n",
       "3186  interacts with  9.207190e-03  PMID:26636579  Downregulated   Chow   \n",
       "3187  interacts with  9.239551e-03  PMID:26636579  Downregulated   Chow   \n",
       "3188  interacts with  9.249940e-03  PMID:26636579  Downregulated   Chow   \n",
       "3189  interacts with  9.342667e-03  PMID:26636579  Downregulated   Chow   \n",
       "3190  interacts with  9.365968e-03  PMID:26636579  Downregulated   Chow   \n",
       "3191  interacts with  9.419827e-03  PMID:26636579  Downregulated   Chow   \n",
       "3192  interacts with  9.419447e-03  PMID:26636579  Downregulated   Chow   \n",
       "3193  interacts with  9.415942e-03  PMID:26636579  Downregulated   Chow   \n",
       "3194  interacts with  9.440761e-03  PMID:26636579  Downregulated   Chow   \n",
       "3195  interacts with  9.463717e-03  PMID:26636579  Downregulated   Chow   \n",
       "3196  interacts with  9.484124e-03  PMID:26636579  Downregulated   Chow   \n",
       "3197  interacts with  9.491914e-03  PMID:26636579  Downregulated   Chow   \n",
       "3198  interacts with  9.534218e-03  PMID:26636579  Downregulated   Chow   \n",
       "3199  interacts with  9.545768e-03  PMID:26636579  Downregulated   Chow   \n",
       "3200  interacts with  9.550025e-03  PMID:26636579  Downregulated   Chow   \n",
       "3201  interacts with  9.564230e-03  PMID:26636579  Downregulated   Chow   \n",
       "3202  interacts with  9.673809e-03  PMID:26636579  Downregulated   Chow   \n",
       "3203  interacts with  9.685690e-03  PMID:26636579  Downregulated   Chow   \n",
       "3204  interacts with  9.686025e-03  PMID:26636579  Downregulated   Chow   \n",
       "3205  interacts with  9.690303e-03  PMID:26636579  Downregulated   Chow   \n",
       "3206  interacts with  9.694926e-03  PMID:26636579  Downregulated   Chow   \n",
       "3207  interacts with  9.715108e-03  PMID:26636579  Downregulated   Chow   \n",
       "3208  interacts with  9.735598e-03  PMID:26636579  Downregulated   Chow   \n",
       "\n",
       "     subject_id subject_label  \n",
       "0     HGNC:4851           HTT  \n",
       "1     HGNC:4851           HTT  \n",
       "2     HGNC:4851           HTT  \n",
       "3     HGNC:4851           HTT  \n",
       "4     HGNC:4851           HTT  \n",
       "5     HGNC:4851           HTT  \n",
       "6     HGNC:4851           HTT  \n",
       "7     HGNC:4851           HTT  \n",
       "8     HGNC:4851           HTT  \n",
       "9     HGNC:4851           HTT  \n",
       "10    HGNC:4851           HTT  \n",
       "11    HGNC:4851           HTT  \n",
       "12    HGNC:4851           HTT  \n",
       "13    HGNC:4851           HTT  \n",
       "14    HGNC:4851           HTT  \n",
       "15    HGNC:4851           HTT  \n",
       "16    HGNC:4851           HTT  \n",
       "17    HGNC:4851           HTT  \n",
       "18    HGNC:4851           HTT  \n",
       "19    HGNC:4851           HTT  \n",
       "20    HGNC:4851           HTT  \n",
       "21    HGNC:4851           HTT  \n",
       "22    HGNC:4851           HTT  \n",
       "23    HGNC:4851           HTT  \n",
       "24    HGNC:4851           HTT  \n",
       "25    HGNC:4851           HTT  \n",
       "26    HGNC:4851           HTT  \n",
       "27    HGNC:4851           HTT  \n",
       "28    HGNC:4851           HTT  \n",
       "29    HGNC:4851           HTT  \n",
       "...         ...           ...  \n",
       "3179  HGNC:4851           HTT  \n",
       "3180  HGNC:4851           HTT  \n",
       "3181  HGNC:4851           HTT  \n",
       "3182  HGNC:4851           HTT  \n",
       "3183  HGNC:4851           HTT  \n",
       "3184  HGNC:4851           HTT  \n",
       "3185  HGNC:4851           HTT  \n",
       "3186  HGNC:4851           HTT  \n",
       "3187  HGNC:4851           HTT  \n",
       "3188  HGNC:4851           HTT  \n",
       "3189  HGNC:4851           HTT  \n",
       "3190  HGNC:4851           HTT  \n",
       "3191  HGNC:4851           HTT  \n",
       "3192  HGNC:4851           HTT  \n",
       "3193  HGNC:4851           HTT  \n",
       "3194  HGNC:4851           HTT  \n",
       "3195  HGNC:4851           HTT  \n",
       "3196  HGNC:4851           HTT  \n",
       "3197  HGNC:4851           HTT  \n",
       "3198  HGNC:4851           HTT  \n",
       "3199  HGNC:4851           HTT  \n",
       "3200  HGNC:4851           HTT  \n",
       "3201  HGNC:4851           HTT  \n",
       "3202  HGNC:4851           HTT  \n",
       "3203  HGNC:4851           HTT  \n",
       "3204  HGNC:4851           HTT  \n",
       "3205  HGNC:4851           HTT  \n",
       "3206  HGNC:4851           HTT  \n",
       "3207  HGNC:4851           HTT  \n",
       "3208  HGNC:4851           HTT  \n",
       "\n",
       "[3209 rows x 12 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rna"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>object_id</th>\n",
       "      <th>property_description</th>\n",
       "      <th>property_id</th>\n",
       "      <th>property_label</th>\n",
       "      <th>property_uri</th>\n",
       "      <th>reference_date</th>\n",
       "      <th>reference_supporting_text</th>\n",
       "      <th>reference_uri</th>\n",
       "      <th>subject_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HGNC:8803</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2007-01-01</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>HGNC:8615</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HGNC:12687</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2007-01-01</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>HGNC:8615</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HGNC:10970</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2007-01-01</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>HGNC:8615</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HGNC:10972</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2007-01-01</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>HGNC:8615</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HGNC:25567</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2007-01-01</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>HGNC:8615</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HGNC:14357</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2007-01-01</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>HGNC:8615</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HGNC:1694</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2007-01-01</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>HGNC:3240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>HGNC:11246</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2007-01-01</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>HGNC:3240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>HGNC:12003</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2007-01-01</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>HGNC:3240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>HGNC:11920</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2007-01-01</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>HGNC:3239</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>HGNC:1116</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2007-01-01</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>HGNC:3239</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>HGNC:3594</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2007-01-01</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>HGNC:3239</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HGNC:5103</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2007-01-01</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>HGNC:3239</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>HGNC:5113</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2007-01-01</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>HGNC:3239</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>HGNC:40</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2007-01-01</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>HGNC:3239</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>HGNC:11246</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2007-01-01</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>HGNC:3239</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>HGNC:12003</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2007-01-01</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>HGNC:3239</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>HGNC:2743</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2007-01-01</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>HGNC:3239</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>HGNC:20</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2007-01-01</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>HGNC:3113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>HGNC:29</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2007-01-01</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>HGNC:3113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>HGNC:186</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2007-01-01</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>HGNC:3113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>HGNC:215</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2007-01-01</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>HGNC:3113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>HGNC:245</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2007-01-01</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>HGNC:3113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>HGNC:253</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2007-01-01</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>HGNC:3113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>HGNC:554</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2007-01-01</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>HGNC:3113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>HGNC:320</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2007-01-01</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>HGNC:3113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>HGNC:363</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2007-01-01</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>HGNC:3113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>HGNC:402</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2007-01-01</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>HGNC:3113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>HGNC:436</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2007-01-01</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>HGNC:3113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>HGNC:470</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2007-01-01</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>HGNC:3113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197237</th>\n",
       "      <td>HGNC:18539</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>HGNC:16764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197238</th>\n",
       "      <td>HGNC:547</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>HGNC:16764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197239</th>\n",
       "      <td>HGNC:14343</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>HGNC:16764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197240</th>\n",
       "      <td>HGNC:25821</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>HGNC:16764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197241</th>\n",
       "      <td>HGNC:1917</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>HGNC:16764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197242</th>\n",
       "      <td>HGNC:18585</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>HGNC:16764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197243</th>\n",
       "      <td>HGNC:8574</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>HGNC:16764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197244</th>\n",
       "      <td>HGNC:8114</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>HGNC:16764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197245</th>\n",
       "      <td>HGNC:4237</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>HGNC:16764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197246</th>\n",
       "      <td>HGNC:1919</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>HGNC:16764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197247</th>\n",
       "      <td>HGNC:11795</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>HGNC:16764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197248</th>\n",
       "      <td>HGNC:29153</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>HGNC:16764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197249</th>\n",
       "      <td>HGNC:3444</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>HGNC:16764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197250</th>\n",
       "      <td>HGNC:25965</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>HGNC:16764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197251</th>\n",
       "      <td>HGNC:9086</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>HGNC:16764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197252</th>\n",
       "      <td>HGNC:8761</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>HGNC:16764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197253</th>\n",
       "      <td>HGNC:27387</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>HGNC:16764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197254</th>\n",
       "      <td>HGNC:23564</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>HGNC:16764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197255</th>\n",
       "      <td>HGNC:2347</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>HGNC:16764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197256</th>\n",
       "      <td>HGNC:3029</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>HGNC:16764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197257</th>\n",
       "      <td>HGNC:1020</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>HGNC:16764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197258</th>\n",
       "      <td>HGNC:2663</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>HGNC:16764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197259</th>\n",
       "      <td>HGNC:6132</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>HGNC:16764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197260</th>\n",
       "      <td>HGNC:1725</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>HGNC:16764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197261</th>\n",
       "      <td>HGNC:20610</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>HGNC:16764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197262</th>\n",
       "      <td>HGNC:10931</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>HGNC:16764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197263</th>\n",
       "      <td>HGNC:24561</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>HGNC:16764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197264</th>\n",
       "      <td>HGNC:11209</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>HGNC:16764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197265</th>\n",
       "      <td>HGNC:26314</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>HGNC:16764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197266</th>\n",
       "      <td>HGNC:26137</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>HGNC:16764</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>197267 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         object_id property_description property_id  property_label  \\\n",
       "0        HGNC:8803                   NA  RO:0002434  interacts with   \n",
       "1       HGNC:12687                   NA  RO:0002434  interacts with   \n",
       "2       HGNC:10970                   NA  RO:0002434  interacts with   \n",
       "3       HGNC:10972                   NA  RO:0002434  interacts with   \n",
       "4       HGNC:25567                   NA  RO:0002434  interacts with   \n",
       "5       HGNC:14357                   NA  RO:0002434  interacts with   \n",
       "6        HGNC:1694                   NA  RO:0002434  interacts with   \n",
       "7       HGNC:11246                   NA  RO:0002434  interacts with   \n",
       "8       HGNC:12003                   NA  RO:0002434  interacts with   \n",
       "9       HGNC:11920                   NA  RO:0002434  interacts with   \n",
       "10       HGNC:1116                   NA  RO:0002434  interacts with   \n",
       "11       HGNC:3594                   NA  RO:0002434  interacts with   \n",
       "12       HGNC:5103                   NA  RO:0002434  interacts with   \n",
       "13       HGNC:5113                   NA  RO:0002434  interacts with   \n",
       "14         HGNC:40                   NA  RO:0002434  interacts with   \n",
       "15      HGNC:11246                   NA  RO:0002434  interacts with   \n",
       "16      HGNC:12003                   NA  RO:0002434  interacts with   \n",
       "17       HGNC:2743                   NA  RO:0002434  interacts with   \n",
       "18         HGNC:20                   NA  RO:0002434  interacts with   \n",
       "19         HGNC:29                   NA  RO:0002434  interacts with   \n",
       "20        HGNC:186                   NA  RO:0002434  interacts with   \n",
       "21        HGNC:215                   NA  RO:0002434  interacts with   \n",
       "22        HGNC:245                   NA  RO:0002434  interacts with   \n",
       "23        HGNC:253                   NA  RO:0002434  interacts with   \n",
       "24        HGNC:554                   NA  RO:0002434  interacts with   \n",
       "25        HGNC:320                   NA  RO:0002434  interacts with   \n",
       "26        HGNC:363                   NA  RO:0002434  interacts with   \n",
       "27        HGNC:402                   NA  RO:0002434  interacts with   \n",
       "28        HGNC:436                   NA  RO:0002434  interacts with   \n",
       "29        HGNC:470                   NA  RO:0002434  interacts with   \n",
       "...            ...                  ...         ...             ...   \n",
       "197237  HGNC:18539                   NA  RO:0002434  interacts with   \n",
       "197238    HGNC:547                   NA  RO:0002434  interacts with   \n",
       "197239  HGNC:14343                   NA  RO:0002434  interacts with   \n",
       "197240  HGNC:25821                   NA  RO:0002434  interacts with   \n",
       "197241   HGNC:1917                   NA  RO:0002434  interacts with   \n",
       "197242  HGNC:18585                   NA  RO:0002434  interacts with   \n",
       "197243   HGNC:8574                   NA  RO:0002434  interacts with   \n",
       "197244   HGNC:8114                   NA  RO:0002434  interacts with   \n",
       "197245   HGNC:4237                   NA  RO:0002434  interacts with   \n",
       "197246   HGNC:1919                   NA  RO:0002434  interacts with   \n",
       "197247  HGNC:11795                   NA  RO:0002434  interacts with   \n",
       "197248  HGNC:29153                   NA  RO:0002434  interacts with   \n",
       "197249   HGNC:3444                   NA  RO:0002434  interacts with   \n",
       "197250  HGNC:25965                   NA  RO:0002434  interacts with   \n",
       "197251   HGNC:9086                   NA  RO:0002434  interacts with   \n",
       "197252   HGNC:8761                   NA  RO:0002434  interacts with   \n",
       "197253  HGNC:27387                   NA  RO:0002434  interacts with   \n",
       "197254  HGNC:23564                   NA  RO:0002434  interacts with   \n",
       "197255   HGNC:2347                   NA  RO:0002434  interacts with   \n",
       "197256   HGNC:3029                   NA  RO:0002434  interacts with   \n",
       "197257   HGNC:1020                   NA  RO:0002434  interacts with   \n",
       "197258   HGNC:2663                   NA  RO:0002434  interacts with   \n",
       "197259   HGNC:6132                   NA  RO:0002434  interacts with   \n",
       "197260   HGNC:1725                   NA  RO:0002434  interacts with   \n",
       "197261  HGNC:20610                   NA  RO:0002434  interacts with   \n",
       "197262  HGNC:10931                   NA  RO:0002434  interacts with   \n",
       "197263  HGNC:24561                   NA  RO:0002434  interacts with   \n",
       "197264  HGNC:11209                   NA  RO:0002434  interacts with   \n",
       "197265  HGNC:26314                   NA  RO:0002434  interacts with   \n",
       "197266  HGNC:26137                   NA  RO:0002434  interacts with   \n",
       "\n",
       "                                     property_uri reference_date  \\\n",
       "0       http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
       "1       http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
       "2       http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
       "3       http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
       "4       http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
       "5       http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
       "6       http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
       "7       http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
       "8       http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
       "9       http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
       "10      http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
       "11      http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
       "12      http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
       "13      http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
       "14      http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
       "15      http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
       "16      http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
       "17      http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
       "18      http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
       "19      http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
       "20      http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
       "21      http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
       "22      http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
       "23      http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
       "24      http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
       "25      http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
       "26      http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
       "27      http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
       "28      http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
       "29      http://purl.obolibrary.org/obo/RO_0002434     2007-01-01   \n",
       "...                                           ...            ...   \n",
       "197237  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "197238  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "197239  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "197240  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "197241  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "197242  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "197243  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "197244  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "197245  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "197246  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "197247  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "197248  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "197249  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "197250  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "197251  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "197252  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "197253  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "197254  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "197255  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "197256  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "197257  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "197258  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "197259  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "197260  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "197261  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "197262  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "197263  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "197264  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "197265  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "197266  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "\n",
       "                                reference_supporting_text  \\\n",
       "0       This edge comes from the TRED dataset in \"tfta...   \n",
       "1       This edge comes from the TRED dataset in \"tfta...   \n",
       "2       This edge comes from the TRED dataset in \"tfta...   \n",
       "3       This edge comes from the TRED dataset in \"tfta...   \n",
       "4       This edge comes from the TRED dataset in \"tfta...   \n",
       "5       This edge comes from the TRED dataset in \"tfta...   \n",
       "6       This edge comes from the TRED dataset in \"tfta...   \n",
       "7       This edge comes from the TRED dataset in \"tfta...   \n",
       "8       This edge comes from the TRED dataset in \"tfta...   \n",
       "9       This edge comes from the TRED dataset in \"tfta...   \n",
       "10      This edge comes from the TRED dataset in \"tfta...   \n",
       "11      This edge comes from the TRED dataset in \"tfta...   \n",
       "12      This edge comes from the TRED dataset in \"tfta...   \n",
       "13      This edge comes from the TRED dataset in \"tfta...   \n",
       "14      This edge comes from the TRED dataset in \"tfta...   \n",
       "15      This edge comes from the TRED dataset in \"tfta...   \n",
       "16      This edge comes from the TRED dataset in \"tfta...   \n",
       "17      This edge comes from the TRED dataset in \"tfta...   \n",
       "18      This edge comes from the TRED dataset in \"tfta...   \n",
       "19      This edge comes from the TRED dataset in \"tfta...   \n",
       "20      This edge comes from the TRED dataset in \"tfta...   \n",
       "21      This edge comes from the TRED dataset in \"tfta...   \n",
       "22      This edge comes from the TRED dataset in \"tfta...   \n",
       "23      This edge comes from the TRED dataset in \"tfta...   \n",
       "24      This edge comes from the TRED dataset in \"tfta...   \n",
       "25      This edge comes from the TRED dataset in \"tfta...   \n",
       "26      This edge comes from the TRED dataset in \"tfta...   \n",
       "27      This edge comes from the TRED dataset in \"tfta...   \n",
       "28      This edge comes from the TRED dataset in \"tfta...   \n",
       "29      This edge comes from the TRED dataset in \"tfta...   \n",
       "...                                                   ...   \n",
       "197237  This edge comes from the C3:TFT dataset in \"ms...   \n",
       "197238  This edge comes from the C3:TFT dataset in \"ms...   \n",
       "197239  This edge comes from the C3:TFT dataset in \"ms...   \n",
       "197240  This edge comes from the C3:TFT dataset in \"ms...   \n",
       "197241  This edge comes from the C3:TFT dataset in \"ms...   \n",
       "197242  This edge comes from the C3:TFT dataset in \"ms...   \n",
       "197243  This edge comes from the C3:TFT dataset in \"ms...   \n",
       "197244  This edge comes from the C3:TFT dataset in \"ms...   \n",
       "197245  This edge comes from the C3:TFT dataset in \"ms...   \n",
       "197246  This edge comes from the C3:TFT dataset in \"ms...   \n",
       "197247  This edge comes from the C3:TFT dataset in \"ms...   \n",
       "197248  This edge comes from the C3:TFT dataset in \"ms...   \n",
       "197249  This edge comes from the C3:TFT dataset in \"ms...   \n",
       "197250  This edge comes from the C3:TFT dataset in \"ms...   \n",
       "197251  This edge comes from the C3:TFT dataset in \"ms...   \n",
       "197252  This edge comes from the C3:TFT dataset in \"ms...   \n",
       "197253  This edge comes from the C3:TFT dataset in \"ms...   \n",
       "197254  This edge comes from the C3:TFT dataset in \"ms...   \n",
       "197255  This edge comes from the C3:TFT dataset in \"ms...   \n",
       "197256  This edge comes from the C3:TFT dataset in \"ms...   \n",
       "197257  This edge comes from the C3:TFT dataset in \"ms...   \n",
       "197258  This edge comes from the C3:TFT dataset in \"ms...   \n",
       "197259  This edge comes from the C3:TFT dataset in \"ms...   \n",
       "197260  This edge comes from the C3:TFT dataset in \"ms...   \n",
       "197261  This edge comes from the C3:TFT dataset in \"ms...   \n",
       "197262  This edge comes from the C3:TFT dataset in \"ms...   \n",
       "197263  This edge comes from the C3:TFT dataset in \"ms...   \n",
       "197264  This edge comes from the C3:TFT dataset in \"ms...   \n",
       "197265  This edge comes from the C3:TFT dataset in \"ms...   \n",
       "197266  This edge comes from the C3:TFT dataset in \"ms...   \n",
       "\n",
       "                                            reference_uri  subject_id  \n",
       "0            https://www.ncbi.nlm.nih.gov/pubmed/17202159   HGNC:8615  \n",
       "1            https://www.ncbi.nlm.nih.gov/pubmed/17202159   HGNC:8615  \n",
       "2            https://www.ncbi.nlm.nih.gov/pubmed/17202159   HGNC:8615  \n",
       "3            https://www.ncbi.nlm.nih.gov/pubmed/17202159   HGNC:8615  \n",
       "4            https://www.ncbi.nlm.nih.gov/pubmed/17202159   HGNC:8615  \n",
       "5            https://www.ncbi.nlm.nih.gov/pubmed/17202159   HGNC:8615  \n",
       "6            https://www.ncbi.nlm.nih.gov/pubmed/17202159   HGNC:3240  \n",
       "7            https://www.ncbi.nlm.nih.gov/pubmed/17202159   HGNC:3240  \n",
       "8            https://www.ncbi.nlm.nih.gov/pubmed/17202159   HGNC:3240  \n",
       "9            https://www.ncbi.nlm.nih.gov/pubmed/17202159   HGNC:3239  \n",
       "10           https://www.ncbi.nlm.nih.gov/pubmed/17202159   HGNC:3239  \n",
       "11           https://www.ncbi.nlm.nih.gov/pubmed/17202159   HGNC:3239  \n",
       "12           https://www.ncbi.nlm.nih.gov/pubmed/17202159   HGNC:3239  \n",
       "13           https://www.ncbi.nlm.nih.gov/pubmed/17202159   HGNC:3239  \n",
       "14           https://www.ncbi.nlm.nih.gov/pubmed/17202159   HGNC:3239  \n",
       "15           https://www.ncbi.nlm.nih.gov/pubmed/17202159   HGNC:3239  \n",
       "16           https://www.ncbi.nlm.nih.gov/pubmed/17202159   HGNC:3239  \n",
       "17           https://www.ncbi.nlm.nih.gov/pubmed/17202159   HGNC:3239  \n",
       "18           https://www.ncbi.nlm.nih.gov/pubmed/17202159   HGNC:3113  \n",
       "19           https://www.ncbi.nlm.nih.gov/pubmed/17202159   HGNC:3113  \n",
       "20           https://www.ncbi.nlm.nih.gov/pubmed/17202159   HGNC:3113  \n",
       "21           https://www.ncbi.nlm.nih.gov/pubmed/17202159   HGNC:3113  \n",
       "22           https://www.ncbi.nlm.nih.gov/pubmed/17202159   HGNC:3113  \n",
       "23           https://www.ncbi.nlm.nih.gov/pubmed/17202159   HGNC:3113  \n",
       "24           https://www.ncbi.nlm.nih.gov/pubmed/17202159   HGNC:3113  \n",
       "25           https://www.ncbi.nlm.nih.gov/pubmed/17202159   HGNC:3113  \n",
       "26           https://www.ncbi.nlm.nih.gov/pubmed/17202159   HGNC:3113  \n",
       "27           https://www.ncbi.nlm.nih.gov/pubmed/17202159   HGNC:3113  \n",
       "28           https://www.ncbi.nlm.nih.gov/pubmed/17202159   HGNC:3113  \n",
       "29           https://www.ncbi.nlm.nih.gov/pubmed/17202159   HGNC:3113  \n",
       "...                                                   ...         ...  \n",
       "197237  http://www.broadinstitute.org/gsea/msigdb/card...  HGNC:16764  \n",
       "197238  http://www.broadinstitute.org/gsea/msigdb/card...  HGNC:16764  \n",
       "197239  http://www.broadinstitute.org/gsea/msigdb/card...  HGNC:16764  \n",
       "197240  http://www.broadinstitute.org/gsea/msigdb/card...  HGNC:16764  \n",
       "197241  http://www.broadinstitute.org/gsea/msigdb/card...  HGNC:16764  \n",
       "197242  http://www.broadinstitute.org/gsea/msigdb/card...  HGNC:16764  \n",
       "197243  http://www.broadinstitute.org/gsea/msigdb/card...  HGNC:16764  \n",
       "197244  http://www.broadinstitute.org/gsea/msigdb/card...  HGNC:16764  \n",
       "197245  http://www.broadinstitute.org/gsea/msigdb/card...  HGNC:16764  \n",
       "197246  http://www.broadinstitute.org/gsea/msigdb/card...  HGNC:16764  \n",
       "197247  http://www.broadinstitute.org/gsea/msigdb/card...  HGNC:16764  \n",
       "197248  http://www.broadinstitute.org/gsea/msigdb/card...  HGNC:16764  \n",
       "197249  http://www.broadinstitute.org/gsea/msigdb/card...  HGNC:16764  \n",
       "197250  http://www.broadinstitute.org/gsea/msigdb/card...  HGNC:16764  \n",
       "197251  http://www.broadinstitute.org/gsea/msigdb/card...  HGNC:16764  \n",
       "197252  http://www.broadinstitute.org/gsea/msigdb/card...  HGNC:16764  \n",
       "197253  http://www.broadinstitute.org/gsea/msigdb/card...  HGNC:16764  \n",
       "197254  http://www.broadinstitute.org/gsea/msigdb/card...  HGNC:16764  \n",
       "197255  http://www.broadinstitute.org/gsea/msigdb/card...  HGNC:16764  \n",
       "197256  http://www.broadinstitute.org/gsea/msigdb/card...  HGNC:16764  \n",
       "197257  http://www.broadinstitute.org/gsea/msigdb/card...  HGNC:16764  \n",
       "197258  http://www.broadinstitute.org/gsea/msigdb/card...  HGNC:16764  \n",
       "197259  http://www.broadinstitute.org/gsea/msigdb/card...  HGNC:16764  \n",
       "197260  http://www.broadinstitute.org/gsea/msigdb/card...  HGNC:16764  \n",
       "197261  http://www.broadinstitute.org/gsea/msigdb/card...  HGNC:16764  \n",
       "197262  http://www.broadinstitute.org/gsea/msigdb/card...  HGNC:16764  \n",
       "197263  http://www.broadinstitute.org/gsea/msigdb/card...  HGNC:16764  \n",
       "197264  http://www.broadinstitute.org/gsea/msigdb/card...  HGNC:16764  \n",
       "197265  http://www.broadinstitute.org/gsea/msigdb/card...  HGNC:16764  \n",
       "197266  http://www.broadinstitute.org/gsea/msigdb/card...  HGNC:16764  \n",
       "\n",
       "[197267 rows x 9 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "statements = pd.concat([curated_df, monarch_df, rna], ignore_index=True, join=\"inner\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>object_id</th>\n",
       "      <th>property_id</th>\n",
       "      <th>property_label</th>\n",
       "      <th>subject_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MGI:892004</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>RGD:1595815</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MGI:3704134</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>ENSEMBL:ENSRNOG00000061639</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MGI:97912</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>FlyBase:FBgn0041191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ENSEMBL:ENSOANG00000006132</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>MGI:1344381</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ENSEMBL:ENSECAG00000019077</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>ENSEMBL:ENSCAFG00000017647</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MONARCH:APO_0000222APO_0000004</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>SGD:S000002780</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>FlyBase:FBgn0013275</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>FlyBase:FBgn0036505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ENSEMBL:ENSGALG00000044425</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>ENSEMBL:ENSCAFG00000016509</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>RGD:1359648</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>ENSEMBL:ENSSSCG00000038628</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ENSEMBL:ENSGALG00000042895</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>ENSEMBL:ENSMMUG00000002630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ENSEMBL:ENSCAFG00000010109</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>ENSEMBL:ENSOANG00000007850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>HGNC:2717</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>HGNC:7878</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ENSEMBL:ENSBTAG00000020963</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>ENSEMBL:ENSSSCG00000011407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>MGI:1276525</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>ZFIN:ZDB-GENE-030916-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ENSEMBL:ENSBTAG00000000919</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>RGD:621405</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>ENSEMBL:ENSOANG00000013551</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>RGD:1565519</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ENSEMBL:ENSBTAG00000014149</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>ZFIN:ZDB-GENE-030911-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>ZFIN:ZDB-GENE-161101-1</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>ENSEMBL:ENSXETG00000009077</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>ENSEMBL:ENSECAG00000022370</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>ZFIN:ZDB-GENE-050522-205</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>FlyBase:FBgn0035161</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>ENSEMBL:ENSMODG00000025146</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:11551</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>MP:0008498</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>MGI:95805</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>HP:0000431</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:7095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>ENSEMBL:ENSPTRG00000006169</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>WormBase:WBGene00017751</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ENSEMBL:ENSOANG00000005852</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>HGNC:9852</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>ENSEMBL:ENSMODG00000018805</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>ENSEMBL:ENSSSCG00000038558</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>HGNC:16031</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>FlyBase:FBgn0031857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>GO:0005634</td>\n",
       "      <td>BFO:0000050</td>\n",
       "      <td>is part of</td>\n",
       "      <td>dictyBase:DDB_G0286907</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236363</th>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>HGNC:4851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236364</th>\n",
       "      <td>HGNC:8672</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>HGNC:4851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236365</th>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>HGNC:4851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236366</th>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>HGNC:4851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236367</th>\n",
       "      <td>HGNC:1570</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>HGNC:4851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236368</th>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>HGNC:4851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236369</th>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>HGNC:4851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236370</th>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>HGNC:4851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236371</th>\n",
       "      <td>HGNC:49384</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>HGNC:4851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236372</th>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>HGNC:4851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236373</th>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>HGNC:4851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236374</th>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>HGNC:4851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236375</th>\n",
       "      <td>HGNC:6814</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>HGNC:4851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236376</th>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>HGNC:4851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236377</th>\n",
       "      <td>HGNC:51822</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>HGNC:4851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236378</th>\n",
       "      <td>HGNC:4941</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>HGNC:4851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236379</th>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>HGNC:4851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236380</th>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>HGNC:4851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236381</th>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>HGNC:4851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236382</th>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>HGNC:4851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236383</th>\n",
       "      <td>HGNC:2356</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>HGNC:4851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236384</th>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>HGNC:4851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236385</th>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>HGNC:4851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236386</th>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>HGNC:4851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236387</th>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>HGNC:4851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236388</th>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>HGNC:4851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236389</th>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>HGNC:4851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236390</th>\n",
       "      <td>HGNC:53465</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>HGNC:4851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236391</th>\n",
       "      <td>HGNC:15722</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>HGNC:4851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236392</th>\n",
       "      <td>HGNC:nan</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>HGNC:4851</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>236393 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             object_id    property_id  \\\n",
       "0                                  NaN            NaN   \n",
       "1                                  NaN            NaN   \n",
       "2                           MGI:892004  RO:HOM0000017   \n",
       "3                          MGI:3704134  RO:HOM0000017   \n",
       "4                            MGI:97912  RO:HOM0000020   \n",
       "5           ENSEMBL:ENSOANG00000006132  RO:HOM0000020   \n",
       "6           ENSEMBL:ENSECAG00000019077  RO:HOM0000020   \n",
       "7       MONARCH:APO_0000222APO_0000004     RO:0002200   \n",
       "8                  FlyBase:FBgn0013275     RO:0002434   \n",
       "9           ENSEMBL:ENSGALG00000044425  RO:HOM0000017   \n",
       "10                         RGD:1359648  RO:HOM0000020   \n",
       "11          ENSEMBL:ENSGALG00000042895  RO:HOM0000020   \n",
       "12          ENSEMBL:ENSCAFG00000010109  RO:HOM0000017   \n",
       "13                           HGNC:2717     RO:0002434   \n",
       "14          ENSEMBL:ENSBTAG00000020963  RO:HOM0000020   \n",
       "15                         MGI:1276525  RO:HOM0000017   \n",
       "16          ENSEMBL:ENSBTAG00000000919  RO:HOM0000020   \n",
       "17          ENSEMBL:ENSOANG00000013551  RO:HOM0000020   \n",
       "18          ENSEMBL:ENSBTAG00000014149  RO:HOM0000017   \n",
       "19              ZFIN:ZDB-GENE-161101-1  RO:HOM0000017   \n",
       "20          ENSEMBL:ENSECAG00000022370  RO:HOM0000020   \n",
       "21                 FlyBase:FBgn0035161  RO:HOM0000017   \n",
       "22                          HP:0000343     RO:0002200   \n",
       "23                          MP:0008498     RO:0002200   \n",
       "24                          HP:0000431     RO:0002200   \n",
       "25          ENSEMBL:ENSPTRG00000006169  RO:HOM0000017   \n",
       "26          ENSEMBL:ENSOANG00000005852  RO:HOM0000020   \n",
       "27          ENSEMBL:ENSMODG00000018805  RO:HOM0000020   \n",
       "28                          HGNC:16031  RO:HOM0000020   \n",
       "29                          GO:0005634    BFO:0000050   \n",
       "...                                ...            ...   \n",
       "236363                        HGNC:nan     RO:0002434   \n",
       "236364                       HGNC:8672     RO:0002434   \n",
       "236365                        HGNC:nan     RO:0002434   \n",
       "236366                        HGNC:nan     RO:0002434   \n",
       "236367                       HGNC:1570     RO:0002434   \n",
       "236368                        HGNC:nan     RO:0002434   \n",
       "236369                        HGNC:nan     RO:0002434   \n",
       "236370                        HGNC:nan     RO:0002434   \n",
       "236371                      HGNC:49384     RO:0002434   \n",
       "236372                        HGNC:nan     RO:0002434   \n",
       "236373                        HGNC:nan     RO:0002434   \n",
       "236374                        HGNC:nan     RO:0002434   \n",
       "236375                       HGNC:6814     RO:0002434   \n",
       "236376                        HGNC:nan     RO:0002434   \n",
       "236377                      HGNC:51822     RO:0002434   \n",
       "236378                       HGNC:4941     RO:0002434   \n",
       "236379                        HGNC:nan     RO:0002434   \n",
       "236380                        HGNC:nan     RO:0002434   \n",
       "236381                        HGNC:nan     RO:0002434   \n",
       "236382                        HGNC:nan     RO:0002434   \n",
       "236383                       HGNC:2356     RO:0002434   \n",
       "236384                        HGNC:nan     RO:0002434   \n",
       "236385                        HGNC:nan     RO:0002434   \n",
       "236386                        HGNC:nan     RO:0002434   \n",
       "236387                        HGNC:nan     RO:0002434   \n",
       "236388                        HGNC:nan     RO:0002434   \n",
       "236389                        HGNC:nan     RO:0002434   \n",
       "236390                      HGNC:53465     RO:0002434   \n",
       "236391                      HGNC:15722     RO:0002434   \n",
       "236392                        HGNC:nan     RO:0002434   \n",
       "\n",
       "                               property_label                  subject_id  \n",
       "0                                         NaN                         NaN  \n",
       "1                                         NaN                         NaN  \n",
       "2              in orthology relationship with                 RGD:1595815  \n",
       "3              in orthology relationship with  ENSEMBL:ENSRNOG00000061639  \n",
       "4       in 1 to 1 orthology relationship with         FlyBase:FBgn0041191  \n",
       "5       in 1 to 1 orthology relationship with                 MGI:1344381  \n",
       "6       in 1 to 1 orthology relationship with  ENSEMBL:ENSCAFG00000017647  \n",
       "7                               has phenotype              SGD:S000002780  \n",
       "8                              interacts with         FlyBase:FBgn0036505  \n",
       "9              in orthology relationship with  ENSEMBL:ENSCAFG00000016509  \n",
       "10      in 1 to 1 orthology relationship with  ENSEMBL:ENSSSCG00000038628  \n",
       "11      in 1 to 1 orthology relationship with  ENSEMBL:ENSMMUG00000002630  \n",
       "12             in orthology relationship with  ENSEMBL:ENSOANG00000007850  \n",
       "13                             interacts with                   HGNC:7878  \n",
       "14      in 1 to 1 orthology relationship with  ENSEMBL:ENSSSCG00000011407  \n",
       "15             in orthology relationship with      ZFIN:ZDB-GENE-030916-1  \n",
       "16      in 1 to 1 orthology relationship with                  RGD:621405  \n",
       "17      in 1 to 1 orthology relationship with                 RGD:1565519  \n",
       "18             in orthology relationship with      ZFIN:ZDB-GENE-030911-3  \n",
       "19             in orthology relationship with  ENSEMBL:ENSXETG00000009077  \n",
       "20      in 1 to 1 orthology relationship with    ZFIN:ZDB-GENE-050522-205  \n",
       "21             in orthology relationship with  ENSEMBL:ENSMODG00000025146  \n",
       "22                              has phenotype                  HGNC:11551  \n",
       "23                              has phenotype                   MGI:95805  \n",
       "24                              has phenotype                   HGNC:7095  \n",
       "25             in orthology relationship with     WormBase:WBGene00017751  \n",
       "26      in 1 to 1 orthology relationship with                   HGNC:9852  \n",
       "27      in 1 to 1 orthology relationship with  ENSEMBL:ENSSSCG00000038558  \n",
       "28      in 1 to 1 orthology relationship with         FlyBase:FBgn0031857  \n",
       "29                                 is part of      dictyBase:DDB_G0286907  \n",
       "...                                       ...                         ...  \n",
       "236363                         interacts with                   HGNC:4851  \n",
       "236364                         interacts with                   HGNC:4851  \n",
       "236365                         interacts with                   HGNC:4851  \n",
       "236366                         interacts with                   HGNC:4851  \n",
       "236367                         interacts with                   HGNC:4851  \n",
       "236368                         interacts with                   HGNC:4851  \n",
       "236369                         interacts with                   HGNC:4851  \n",
       "236370                         interacts with                   HGNC:4851  \n",
       "236371                         interacts with                   HGNC:4851  \n",
       "236372                         interacts with                   HGNC:4851  \n",
       "236373                         interacts with                   HGNC:4851  \n",
       "236374                         interacts with                   HGNC:4851  \n",
       "236375                         interacts with                   HGNC:4851  \n",
       "236376                         interacts with                   HGNC:4851  \n",
       "236377                         interacts with                   HGNC:4851  \n",
       "236378                         interacts with                   HGNC:4851  \n",
       "236379                         interacts with                   HGNC:4851  \n",
       "236380                         interacts with                   HGNC:4851  \n",
       "236381                         interacts with                   HGNC:4851  \n",
       "236382                         interacts with                   HGNC:4851  \n",
       "236383                         interacts with                   HGNC:4851  \n",
       "236384                         interacts with                   HGNC:4851  \n",
       "236385                         interacts with                   HGNC:4851  \n",
       "236386                         interacts with                   HGNC:4851  \n",
       "236387                         interacts with                   HGNC:4851  \n",
       "236388                         interacts with                   HGNC:4851  \n",
       "236389                         interacts with                   HGNC:4851  \n",
       "236390                         interacts with                   HGNC:4851  \n",
       "236391                         interacts with                   HGNC:4851  \n",
       "236392                         interacts with                   HGNC:4851  \n",
       "\n",
       "[236393 rows x 4 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "statements"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "merge1 = pd.merge(statements, tf, how='inner', left_on='subject_id', right_on='subject_id',\n",
    "                      suffixes=('_graph', '_tf'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>object_id_graph</th>\n",
       "      <th>property_id_graph</th>\n",
       "      <th>property_label_graph</th>\n",
       "      <th>subject_id</th>\n",
       "      <th>object_id_tf</th>\n",
       "      <th>property_description</th>\n",
       "      <th>property_id_tf</th>\n",
       "      <th>property_label_tf</th>\n",
       "      <th>property_uri</th>\n",
       "      <th>reference_date</th>\n",
       "      <th>reference_supporting_text</th>\n",
       "      <th>reference_uri</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:11551</td>\n",
       "      <td>HGNC:9701</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:11551</td>\n",
       "      <td>HGNC:7879</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:11551</td>\n",
       "      <td>HGNC:8618</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:11551</td>\n",
       "      <td>HGNC:11290</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:11551</td>\n",
       "      <td>HGNC:11743</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:11551</td>\n",
       "      <td>HGNC:11289</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:11551</td>\n",
       "      <td>HGNC:3238</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:11551</td>\n",
       "      <td>HGNC:6206</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:11551</td>\n",
       "      <td>HGNC:3241</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:11551</td>\n",
       "      <td>HGNC:3488</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:11551</td>\n",
       "      <td>HGNC:1539</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:11551</td>\n",
       "      <td>HGNC:13723</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:11551</td>\n",
       "      <td>HGNC:6121</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:11551</td>\n",
       "      <td>HGNC:11208</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:11551</td>\n",
       "      <td>HGNC:8619</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:11551</td>\n",
       "      <td>HGNC:13011</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:11551</td>\n",
       "      <td>HGNC:6122</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:11551</td>\n",
       "      <td>HGNC:6119</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:11551</td>\n",
       "      <td>HGNC:5358</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:11551</td>\n",
       "      <td>HGNC:28510</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:11551</td>\n",
       "      <td>HGNC:11209</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:11551</td>\n",
       "      <td>HGNC:6642</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:11551</td>\n",
       "      <td>HGNC:11207</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:11551</td>\n",
       "      <td>HGNC:10473</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:11551</td>\n",
       "      <td>HGNC:7553</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:11551</td>\n",
       "      <td>HGNC:13718</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:11551</td>\n",
       "      <td>HGNC:3239</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:11551</td>\n",
       "      <td>HGNC:6204</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:11551</td>\n",
       "      <td>HGNC:6116</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>HP:0000343</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:11551</td>\n",
       "      <td>HGNC:644</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693955</th>\n",
       "      <td>HP:0001513</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:7762</td>\n",
       "      <td>HGNC:11557</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693956</th>\n",
       "      <td>HP:0001513</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:7762</td>\n",
       "      <td>HGNC:15979</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693957</th>\n",
       "      <td>HP:0001513</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:7762</td>\n",
       "      <td>HGNC:18668</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693958</th>\n",
       "      <td>HP:0001513</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:7762</td>\n",
       "      <td>HGNC:1835</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693959</th>\n",
       "      <td>HP:0001513</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:7762</td>\n",
       "      <td>HGNC:11205</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693960</th>\n",
       "      <td>HP:0001513</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:7762</td>\n",
       "      <td>HGNC:7565</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693961</th>\n",
       "      <td>HP:0001513</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:7762</td>\n",
       "      <td>HGNC:12869</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693962</th>\n",
       "      <td>HP:0001513</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:7762</td>\n",
       "      <td>HGNC:12679</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693963</th>\n",
       "      <td>HP:0001513</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:7762</td>\n",
       "      <td>HGNC:16617</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693964</th>\n",
       "      <td>HP:0001513</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:7762</td>\n",
       "      <td>HGNC:3113</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693965</th>\n",
       "      <td>HP:0001513</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:7762</td>\n",
       "      <td>HGNC:30721</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693966</th>\n",
       "      <td>HP:0001513</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:7762</td>\n",
       "      <td>HGNC:348</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693967</th>\n",
       "      <td>HP:0001513</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:7762</td>\n",
       "      <td>HGNC:9955</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693968</th>\n",
       "      <td>HP:0001513</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:7762</td>\n",
       "      <td>HGNC:9966</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693969</th>\n",
       "      <td>HP:0001513</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:7762</td>\n",
       "      <td>HGNC:12796</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693970</th>\n",
       "      <td>HP:0001513</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:7762</td>\n",
       "      <td>HGNC:3373</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693971</th>\n",
       "      <td>HP:0001513</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:7762</td>\n",
       "      <td>HGNC:7794</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693972</th>\n",
       "      <td>HP:0001513</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:7762</td>\n",
       "      <td>HGNC:7559</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693973</th>\n",
       "      <td>HP:0001513</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:7762</td>\n",
       "      <td>HGNC:9210</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693974</th>\n",
       "      <td>HP:0001513</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:7762</td>\n",
       "      <td>HGNC:4910</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693975</th>\n",
       "      <td>HP:0001513</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:7762</td>\n",
       "      <td>HGNC:12594</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693976</th>\n",
       "      <td>HP:0001513</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:7762</td>\n",
       "      <td>HGNC:4807</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693977</th>\n",
       "      <td>HP:0001513</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:7762</td>\n",
       "      <td>HGNC:6913</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693978</th>\n",
       "      <td>HP:0001513</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:7762</td>\n",
       "      <td>HGNC:11634</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693979</th>\n",
       "      <td>HP:0001513</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:7762</td>\n",
       "      <td>HGNC:13906</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693980</th>\n",
       "      <td>HP:0001513</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:7762</td>\n",
       "      <td>HGNC:12593</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693981</th>\n",
       "      <td>HP:0001513</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:7762</td>\n",
       "      <td>HGNC:17362</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>2012-09-14</td>\n",
       "      <td>This edge comes from the NEPH2012 dataset in \"...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22959076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693982</th>\n",
       "      <td>HP:0001513</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:7762</td>\n",
       "      <td>HGNC:6081</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the TRRUST dataset in \"tf...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/10545951,1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693983</th>\n",
       "      <td>HP:0001513</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:7762</td>\n",
       "      <td>HGNC:6090</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the TRRUST dataset in \"tf...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/12890672</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693984</th>\n",
       "      <td>HP:0001513</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>HGNC:7762</td>\n",
       "      <td>HGNC:24140</td>\n",
       "      <td>NA</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This edge comes from the TRRUST dataset in \"tf...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/21771782</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>693985 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       object_id_graph property_id_graph property_label_graph  subject_id  \\\n",
       "0           HP:0000343        RO:0002200        has phenotype  HGNC:11551   \n",
       "1           HP:0000343        RO:0002200        has phenotype  HGNC:11551   \n",
       "2           HP:0000343        RO:0002200        has phenotype  HGNC:11551   \n",
       "3           HP:0000343        RO:0002200        has phenotype  HGNC:11551   \n",
       "4           HP:0000343        RO:0002200        has phenotype  HGNC:11551   \n",
       "5           HP:0000343        RO:0002200        has phenotype  HGNC:11551   \n",
       "6           HP:0000343        RO:0002200        has phenotype  HGNC:11551   \n",
       "7           HP:0000343        RO:0002200        has phenotype  HGNC:11551   \n",
       "8           HP:0000343        RO:0002200        has phenotype  HGNC:11551   \n",
       "9           HP:0000343        RO:0002200        has phenotype  HGNC:11551   \n",
       "10          HP:0000343        RO:0002200        has phenotype  HGNC:11551   \n",
       "11          HP:0000343        RO:0002200        has phenotype  HGNC:11551   \n",
       "12          HP:0000343        RO:0002200        has phenotype  HGNC:11551   \n",
       "13          HP:0000343        RO:0002200        has phenotype  HGNC:11551   \n",
       "14          HP:0000343        RO:0002200        has phenotype  HGNC:11551   \n",
       "15          HP:0000343        RO:0002200        has phenotype  HGNC:11551   \n",
       "16          HP:0000343        RO:0002200        has phenotype  HGNC:11551   \n",
       "17          HP:0000343        RO:0002200        has phenotype  HGNC:11551   \n",
       "18          HP:0000343        RO:0002200        has phenotype  HGNC:11551   \n",
       "19          HP:0000343        RO:0002200        has phenotype  HGNC:11551   \n",
       "20          HP:0000343        RO:0002200        has phenotype  HGNC:11551   \n",
       "21          HP:0000343        RO:0002200        has phenotype  HGNC:11551   \n",
       "22          HP:0000343        RO:0002200        has phenotype  HGNC:11551   \n",
       "23          HP:0000343        RO:0002200        has phenotype  HGNC:11551   \n",
       "24          HP:0000343        RO:0002200        has phenotype  HGNC:11551   \n",
       "25          HP:0000343        RO:0002200        has phenotype  HGNC:11551   \n",
       "26          HP:0000343        RO:0002200        has phenotype  HGNC:11551   \n",
       "27          HP:0000343        RO:0002200        has phenotype  HGNC:11551   \n",
       "28          HP:0000343        RO:0002200        has phenotype  HGNC:11551   \n",
       "29          HP:0000343        RO:0002200        has phenotype  HGNC:11551   \n",
       "...                ...               ...                  ...         ...   \n",
       "693955      HP:0001513        RO:0002200        has phenotype   HGNC:7762   \n",
       "693956      HP:0001513        RO:0002200        has phenotype   HGNC:7762   \n",
       "693957      HP:0001513        RO:0002200        has phenotype   HGNC:7762   \n",
       "693958      HP:0001513        RO:0002200        has phenotype   HGNC:7762   \n",
       "693959      HP:0001513        RO:0002200        has phenotype   HGNC:7762   \n",
       "693960      HP:0001513        RO:0002200        has phenotype   HGNC:7762   \n",
       "693961      HP:0001513        RO:0002200        has phenotype   HGNC:7762   \n",
       "693962      HP:0001513        RO:0002200        has phenotype   HGNC:7762   \n",
       "693963      HP:0001513        RO:0002200        has phenotype   HGNC:7762   \n",
       "693964      HP:0001513        RO:0002200        has phenotype   HGNC:7762   \n",
       "693965      HP:0001513        RO:0002200        has phenotype   HGNC:7762   \n",
       "693966      HP:0001513        RO:0002200        has phenotype   HGNC:7762   \n",
       "693967      HP:0001513        RO:0002200        has phenotype   HGNC:7762   \n",
       "693968      HP:0001513        RO:0002200        has phenotype   HGNC:7762   \n",
       "693969      HP:0001513        RO:0002200        has phenotype   HGNC:7762   \n",
       "693970      HP:0001513        RO:0002200        has phenotype   HGNC:7762   \n",
       "693971      HP:0001513        RO:0002200        has phenotype   HGNC:7762   \n",
       "693972      HP:0001513        RO:0002200        has phenotype   HGNC:7762   \n",
       "693973      HP:0001513        RO:0002200        has phenotype   HGNC:7762   \n",
       "693974      HP:0001513        RO:0002200        has phenotype   HGNC:7762   \n",
       "693975      HP:0001513        RO:0002200        has phenotype   HGNC:7762   \n",
       "693976      HP:0001513        RO:0002200        has phenotype   HGNC:7762   \n",
       "693977      HP:0001513        RO:0002200        has phenotype   HGNC:7762   \n",
       "693978      HP:0001513        RO:0002200        has phenotype   HGNC:7762   \n",
       "693979      HP:0001513        RO:0002200        has phenotype   HGNC:7762   \n",
       "693980      HP:0001513        RO:0002200        has phenotype   HGNC:7762   \n",
       "693981      HP:0001513        RO:0002200        has phenotype   HGNC:7762   \n",
       "693982      HP:0001513        RO:0002200        has phenotype   HGNC:7762   \n",
       "693983      HP:0001513        RO:0002200        has phenotype   HGNC:7762   \n",
       "693984      HP:0001513        RO:0002200        has phenotype   HGNC:7762   \n",
       "\n",
       "       object_id_tf property_description property_id_tf property_label_tf  \\\n",
       "0         HGNC:9701                   NA     RO:0002434    interacts with   \n",
       "1         HGNC:7879                   NA     RO:0002434    interacts with   \n",
       "2         HGNC:8618                   NA     RO:0002434    interacts with   \n",
       "3        HGNC:11290                   NA     RO:0002434    interacts with   \n",
       "4        HGNC:11743                   NA     RO:0002434    interacts with   \n",
       "5        HGNC:11289                   NA     RO:0002434    interacts with   \n",
       "6         HGNC:3238                   NA     RO:0002434    interacts with   \n",
       "7         HGNC:6206                   NA     RO:0002434    interacts with   \n",
       "8         HGNC:3241                   NA     RO:0002434    interacts with   \n",
       "9         HGNC:3488                   NA     RO:0002434    interacts with   \n",
       "10        HGNC:1539                   NA     RO:0002434    interacts with   \n",
       "11       HGNC:13723                   NA     RO:0002434    interacts with   \n",
       "12        HGNC:6121                   NA     RO:0002434    interacts with   \n",
       "13       HGNC:11208                   NA     RO:0002434    interacts with   \n",
       "14        HGNC:8619                   NA     RO:0002434    interacts with   \n",
       "15       HGNC:13011                   NA     RO:0002434    interacts with   \n",
       "16        HGNC:6122                   NA     RO:0002434    interacts with   \n",
       "17        HGNC:6119                   NA     RO:0002434    interacts with   \n",
       "18        HGNC:5358                   NA     RO:0002434    interacts with   \n",
       "19       HGNC:28510                   NA     RO:0002434    interacts with   \n",
       "20       HGNC:11209                   NA     RO:0002434    interacts with   \n",
       "21        HGNC:6642                   NA     RO:0002434    interacts with   \n",
       "22       HGNC:11207                   NA     RO:0002434    interacts with   \n",
       "23       HGNC:10473                   NA     RO:0002434    interacts with   \n",
       "24        HGNC:7553                   NA     RO:0002434    interacts with   \n",
       "25       HGNC:13718                   NA     RO:0002434    interacts with   \n",
       "26        HGNC:3239                   NA     RO:0002434    interacts with   \n",
       "27        HGNC:6204                   NA     RO:0002434    interacts with   \n",
       "28        HGNC:6116                   NA     RO:0002434    interacts with   \n",
       "29         HGNC:644                   NA     RO:0002434    interacts with   \n",
       "...             ...                  ...            ...               ...   \n",
       "693955   HGNC:11557                   NA     RO:0002434    interacts with   \n",
       "693956   HGNC:15979                   NA     RO:0002434    interacts with   \n",
       "693957   HGNC:18668                   NA     RO:0002434    interacts with   \n",
       "693958    HGNC:1835                   NA     RO:0002434    interacts with   \n",
       "693959   HGNC:11205                   NA     RO:0002434    interacts with   \n",
       "693960    HGNC:7565                   NA     RO:0002434    interacts with   \n",
       "693961   HGNC:12869                   NA     RO:0002434    interacts with   \n",
       "693962   HGNC:12679                   NA     RO:0002434    interacts with   \n",
       "693963   HGNC:16617                   NA     RO:0002434    interacts with   \n",
       "693964    HGNC:3113                   NA     RO:0002434    interacts with   \n",
       "693965   HGNC:30721                   NA     RO:0002434    interacts with   \n",
       "693966     HGNC:348                   NA     RO:0002434    interacts with   \n",
       "693967    HGNC:9955                   NA     RO:0002434    interacts with   \n",
       "693968    HGNC:9966                   NA     RO:0002434    interacts with   \n",
       "693969   HGNC:12796                   NA     RO:0002434    interacts with   \n",
       "693970    HGNC:3373                   NA     RO:0002434    interacts with   \n",
       "693971    HGNC:7794                   NA     RO:0002434    interacts with   \n",
       "693972    HGNC:7559                   NA     RO:0002434    interacts with   \n",
       "693973    HGNC:9210                   NA     RO:0002434    interacts with   \n",
       "693974    HGNC:4910                   NA     RO:0002434    interacts with   \n",
       "693975   HGNC:12594                   NA     RO:0002434    interacts with   \n",
       "693976    HGNC:4807                   NA     RO:0002434    interacts with   \n",
       "693977    HGNC:6913                   NA     RO:0002434    interacts with   \n",
       "693978   HGNC:11634                   NA     RO:0002434    interacts with   \n",
       "693979   HGNC:13906                   NA     RO:0002434    interacts with   \n",
       "693980   HGNC:12593                   NA     RO:0002434    interacts with   \n",
       "693981   HGNC:17362                   NA     RO:0002434    interacts with   \n",
       "693982    HGNC:6081                   NA     RO:0002434    interacts with   \n",
       "693983    HGNC:6090                   NA     RO:0002434    interacts with   \n",
       "693984   HGNC:24140                   NA     RO:0002434    interacts with   \n",
       "\n",
       "                                     property_uri reference_date  \\\n",
       "0       http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "1       http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "2       http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "3       http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "4       http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "5       http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "6       http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "7       http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "8       http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "9       http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "10      http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "11      http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "12      http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "13      http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "14      http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "15      http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "16      http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "17      http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "18      http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "19      http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "20      http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "21      http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "22      http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "23      http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "24      http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "25      http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "26      http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "27      http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "28      http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "29      http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "...                                           ...            ...   \n",
       "693955  http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "693956  http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "693957  http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "693958  http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "693959  http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "693960  http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "693961  http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "693962  http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "693963  http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "693964  http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "693965  http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "693966  http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "693967  http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "693968  http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "693969  http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "693970  http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "693971  http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "693972  http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "693973  http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "693974  http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "693975  http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "693976  http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "693977  http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "693978  http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "693979  http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "693980  http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "693981  http://purl.obolibrary.org/obo/RO_0002434     2012-09-14   \n",
       "693982  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "693983  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "693984  http://purl.obolibrary.org/obo/RO_0002434            NaN   \n",
       "\n",
       "                                reference_supporting_text  \\\n",
       "0       This edge comes from the NEPH2012 dataset in \"...   \n",
       "1       This edge comes from the NEPH2012 dataset in \"...   \n",
       "2       This edge comes from the NEPH2012 dataset in \"...   \n",
       "3       This edge comes from the NEPH2012 dataset in \"...   \n",
       "4       This edge comes from the NEPH2012 dataset in \"...   \n",
       "5       This edge comes from the NEPH2012 dataset in \"...   \n",
       "6       This edge comes from the NEPH2012 dataset in \"...   \n",
       "7       This edge comes from the NEPH2012 dataset in \"...   \n",
       "8       This edge comes from the NEPH2012 dataset in \"...   \n",
       "9       This edge comes from the NEPH2012 dataset in \"...   \n",
       "10      This edge comes from the NEPH2012 dataset in \"...   \n",
       "11      This edge comes from the NEPH2012 dataset in \"...   \n",
       "12      This edge comes from the NEPH2012 dataset in \"...   \n",
       "13      This edge comes from the NEPH2012 dataset in \"...   \n",
       "14      This edge comes from the NEPH2012 dataset in \"...   \n",
       "15      This edge comes from the NEPH2012 dataset in \"...   \n",
       "16      This edge comes from the NEPH2012 dataset in \"...   \n",
       "17      This edge comes from the NEPH2012 dataset in \"...   \n",
       "18      This edge comes from the NEPH2012 dataset in \"...   \n",
       "19      This edge comes from the NEPH2012 dataset in \"...   \n",
       "20      This edge comes from the NEPH2012 dataset in \"...   \n",
       "21      This edge comes from the NEPH2012 dataset in \"...   \n",
       "22      This edge comes from the NEPH2012 dataset in \"...   \n",
       "23      This edge comes from the NEPH2012 dataset in \"...   \n",
       "24      This edge comes from the NEPH2012 dataset in \"...   \n",
       "25      This edge comes from the NEPH2012 dataset in \"...   \n",
       "26      This edge comes from the NEPH2012 dataset in \"...   \n",
       "27      This edge comes from the NEPH2012 dataset in \"...   \n",
       "28      This edge comes from the NEPH2012 dataset in \"...   \n",
       "29      This edge comes from the NEPH2012 dataset in \"...   \n",
       "...                                                   ...   \n",
       "693955  This edge comes from the NEPH2012 dataset in \"...   \n",
       "693956  This edge comes from the NEPH2012 dataset in \"...   \n",
       "693957  This edge comes from the NEPH2012 dataset in \"...   \n",
       "693958  This edge comes from the NEPH2012 dataset in \"...   \n",
       "693959  This edge comes from the NEPH2012 dataset in \"...   \n",
       "693960  This edge comes from the NEPH2012 dataset in \"...   \n",
       "693961  This edge comes from the NEPH2012 dataset in \"...   \n",
       "693962  This edge comes from the NEPH2012 dataset in \"...   \n",
       "693963  This edge comes from the NEPH2012 dataset in \"...   \n",
       "693964  This edge comes from the NEPH2012 dataset in \"...   \n",
       "693965  This edge comes from the NEPH2012 dataset in \"...   \n",
       "693966  This edge comes from the NEPH2012 dataset in \"...   \n",
       "693967  This edge comes from the NEPH2012 dataset in \"...   \n",
       "693968  This edge comes from the NEPH2012 dataset in \"...   \n",
       "693969  This edge comes from the NEPH2012 dataset in \"...   \n",
       "693970  This edge comes from the NEPH2012 dataset in \"...   \n",
       "693971  This edge comes from the NEPH2012 dataset in \"...   \n",
       "693972  This edge comes from the NEPH2012 dataset in \"...   \n",
       "693973  This edge comes from the NEPH2012 dataset in \"...   \n",
       "693974  This edge comes from the NEPH2012 dataset in \"...   \n",
       "693975  This edge comes from the NEPH2012 dataset in \"...   \n",
       "693976  This edge comes from the NEPH2012 dataset in \"...   \n",
       "693977  This edge comes from the NEPH2012 dataset in \"...   \n",
       "693978  This edge comes from the NEPH2012 dataset in \"...   \n",
       "693979  This edge comes from the NEPH2012 dataset in \"...   \n",
       "693980  This edge comes from the NEPH2012 dataset in \"...   \n",
       "693981  This edge comes from the NEPH2012 dataset in \"...   \n",
       "693982  This edge comes from the TRRUST dataset in \"tf...   \n",
       "693983  This edge comes from the TRRUST dataset in \"tf...   \n",
       "693984  This edge comes from the TRRUST dataset in \"tf...   \n",
       "\n",
       "                                            reference_uri  \n",
       "0            https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "1            https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "2            https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "3            https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "4            https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "5            https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "6            https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "7            https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "8            https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "9            https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "10           https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "11           https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "12           https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "13           https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "14           https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "15           https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "16           https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "17           https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "18           https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "19           https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "20           https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "21           https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "22           https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "23           https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "24           https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "25           https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "26           https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "27           https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "28           https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "29           https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "...                                                   ...  \n",
       "693955       https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "693956       https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "693957       https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "693958       https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "693959       https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "693960       https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "693961       https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "693962       https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "693963       https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "693964       https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "693965       https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "693966       https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "693967       https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "693968       https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "693969       https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "693970       https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "693971       https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "693972       https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "693973       https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "693974       https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "693975       https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "693976       https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "693977       https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "693978       https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "693979       https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "693980       https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "693981       https://www.ncbi.nlm.nih.gov/pubmed/22959076  \n",
       "693982  https://www.ncbi.nlm.nih.gov/pubmed/10545951,1...  \n",
       "693983       https://www.ncbi.nlm.nih.gov/pubmed/12890672  \n",
       "693984       https://www.ncbi.nlm.nih.gov/pubmed/21771782  \n",
       "\n",
       "[693985 rows x 12 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "\"['reference_uri_tf' 'reference_supporting_text_tf' 'reference_date_tf'\\n 'property_description_tf' 'property_uri_tf'] not in index\"",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-9-7ea19627f491>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      2\u001b[0m     [['subject_id', 'property_id_tf', 'object_id_tf', 'reference_uri_tf',\n\u001b[1;32m      3\u001b[0m       \u001b[0;34m'reference_supporting_text_tf'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'reference_date_tf'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'property_label_tf'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 4\u001b[0;31m       'property_description_tf', 'property_uri_tf']]\n\u001b[0m\u001b[1;32m      5\u001b[0m         .rename(columns={\n\u001b[1;32m      6\u001b[0m         \u001b[0;34m'property_id_tf'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;34m'property_id'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/sr/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   2680\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mSeries\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mndarray\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mIndex\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlist\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2681\u001b[0m             \u001b[0;31m# either boolean or fancy integer index\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2682\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getitem_array\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2683\u001b[0m         \u001b[0;32melif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mDataFrame\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2684\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getitem_frame\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/sr/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m_getitem_array\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   2724\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_take\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindexer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2725\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2726\u001b[0;31m             \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_convert_to_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2727\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_take\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindexer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2728\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/sr/lib/python3.6/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m_convert_to_indexer\u001b[0;34m(self, obj, axis, is_setter)\u001b[0m\n\u001b[1;32m   1325\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0mmask\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0many\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1326\u001b[0m                     raise KeyError('{mask} not in index'\n\u001b[0;32m-> 1327\u001b[0;31m                                    .format(mask=objarr[mask]))\n\u001b[0m\u001b[1;32m   1328\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1329\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mcom\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_values_from_object\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindexer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: \"['reference_uri_tf' 'reference_supporting_text_tf' 'reference_date_tf'\\n 'property_description_tf' 'property_uri_tf'] not in index\""
     ]
    }
   ],
   "source": [
    "merge1_clean = (merge1\n",
    "    [['subject_id', 'property_id_tf', 'object_id_tf', 'reference_uri_tf',\n",
    "      'reference_supporting_text_tf', 'reference_date_tf', 'property_label_tf',\n",
    "      'property_description_tf', 'property_uri_tf']]\n",
    "        .rename(columns={\n",
    "        'property_id_tf': 'property_id',\n",
    "        'object_id_tf': 'object_id',\n",
    "        'reference_uri_tf': 'reference_uri',\n",
    "        'reference_supporting_text_tf': 'reference_supporting_text',\n",
    "        'reference_date_tf': 'reference_date',\n",
    "        'property_label_tf': 'property_label',\n",
    "        'property_description_tf': 'property_description',\n",
    "        'property_uri_tf': 'property_uri'\n",
    "    })\n",
    "    )"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# fix curation data:\n",
    "curation_nodes = curation_nodes.astype('object')\n",
    "#curation_nodes.loc[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "curation_nodes = curation_nodes.astype('object')\n",
    "curation_nodes['name'] = [\"NaN\", \"NaN\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "curation_nodes.loc[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "# construct df's\n",
    "edges = graph.build_edges(\n",
    "    curation=curation_edges,\n",
    "    monarch=monarch_edges,\n",
    "    transcriptomics=rna_edges,\n",
    "    regulation=reg_edges\n",
    ")\n",
    "nodes = graph.build_nodes(\n",
    "    statements=edges,\n",
    "    curation=curation_nodes,\n",
    "    monarch=monarch_nodes,\n",
    "    transcriptomics=rna_nodes,\n",
    "    regulation=reg_nodes\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "nodes = pd.read_csv(\"/home/karolis/Structured review/bioknowledge-reviewer/bioknowledge_reviewer/graph/graph_nodes_v2020-11-20.csv\")\n",
    "edges = pd.read_csv(\"/home/karolis/Structured review/bioknowledge-reviewer/bioknowledge_reviewer/graph/graph_edges_v2020-11-20.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- Regulation edges _merged_ with the graph is returned as both digital object (`reg_graph_edges`) and CSV file at _**graph/**_ (`regulation_graph_edges_version.csv`)\n",
    "- Monarch network is returned as both digital object (`monarch_graph_edges`, `monarch_graph_nodes`) and CSV files at _**monarch/**_ (`monarch_edges_version.csv`, `monarch_nodes_version.csv`) overwritten the previous one.\n",
    "- Review knowledge graph is returned as both digital object (`edges`, `nodes`) and CSV files at _**graph/**_ (`graph_edges_version.csv`, `graph_nodes_version.csv`)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Neo4jlib library\n",
    "### Import the graph into Neo4j graph database\n",
    "#### import neo4jlib\n",
    "\n",
    "Tasks:\n",
    "\n",
    "- Create Neo4j server instance\n",
    "- Import review graph into the Neo4j graph database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/karolis/anaconda3/envs/sr/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2728: DtypeWarning: Columns (3,5) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "edges_df = pd.read_csv(\"~/Structured review/HD-SR/bioknowledge-reviewer/bioknowledge_reviewer/graph/graph_edges_v2020-12-02.csv\")\n",
    "nodes_df = pd.read_csv(\"~/Structured review/HD-SR/bioknowledge-reviewer/bioknowledge_reviewer/graph/graph_nodes_v2020-12-02.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Creating a Neo4j community v3.5.6 server instance...\n",
      "Configuration adjusted!\n",
      "Neo4j v3.5.6 is running.\n",
      "The name of the neo4j directory is neo4j-community-3.5.6\n",
      "statements:  433601\n",
      "concepts:  33817\n",
      "\n",
      "File './neo4j-community-3.5.6/import/ngly1/ngly1_statements.csv' saved.\n",
      "\n",
      "File './neo4j-community-3.5.6/import/ngly1/ngly1_concepts.csv' saved.\n",
      "\n",
      "The function \"do_import()\" is running...\n",
      "\n",
      "The graph is imported into the server. Neo4j is running.You can start exploring and querying for hypothesis. If you change ports or authentication in the Neo4j configuration file, the hypothesis-generation modules performance, hypothesis.py and summary.py, will be affected.\n",
      "\n",
      "CPU times: user 6.09 s, sys: 399 ms, total: 6.49 s\n",
      "Wall time: 32.8 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "# create a Neo4j server instance\n",
    "neo4j_dir = neo4jlib.create_neo4j_instance('3.5.6')\n",
    "print('The name of the neo4j directory is {}'.format(neo4j_dir))\n",
    "\n",
    "# import to graph database\n",
    "## prepare the graph to neo4j format\n",
    "#edges_df = utils.get_dataframe(edges)\n",
    "#nodes_df = utils.get_dataframe(nodes)\n",
    "statements = neo4jlib.get_statements(edges_df)\n",
    "concepts = neo4jlib.get_concepts(nodes_df)\n",
    "print('statements: ', len(statements))\n",
    "print('concepts: ',len(concepts))\n",
    "\n",
    "## save files into neo4j import dir\n",
    "neo4j_path = './{}'.format(neo4j_dir)\n",
    "neo4jlib.save_neo4j_files(statements, neo4j_path, file_type = 'statements')\n",
    "neo4jlib.save_neo4j_files(concepts, neo4j_path, file_type = 'concepts')\n",
    "\n",
    "## import graph into neo4j database\n",
    "neo4jlib.do_import(neo4j_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "del edges_df\n",
    "del nodes_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'statements' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-15-9ce45af71828>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;31m# print type of objects\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'type edges:'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstatements\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'type nodes:'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mconcepts\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'statements' is not defined"
     ]
    }
   ],
   "source": [
    "# print type of objects\n",
    "print('type edges:', type(statements))\n",
    "print('type nodes:', type(concepts))\n",
    "print()\n",
    "\n",
    "# print objects sizes\n",
    "print('len edges:', len(statements))\n",
    "print('len nodes:', len(concepts))\n",
    "print()\n",
    "\n",
    "# print object attribute\n",
    "print('attribute edges:', statements.columns)\n",
    "print('attribute nodes:', concepts.columns)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## hypothesis-generation library\n",
    "### Query the graph for mechanistic explanation, then summarize the extracted paths\n",
    "#### import hypothesis, summary\n",
    "\n",
    "\n",
    "Tasks:\n",
    "\n",
    "* Retrieve orthopheno paths with the `query` method.\n",
    "* Retrieve orthopheno paths using relaxing node degree parameters with the `query` method.\n",
    "* Retrieve orthopheno paths from a more open query topology with the `open_query` method.\n",
    "* Get hypothesis summaries"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Ortopheno query with general nodes/relations removed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from neo4j import GraphDatabase\n",
    "import sys,os\n",
    "import json\n",
    "import yaml\n",
    "import datetime\n",
    "import neo4j.exceptions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def parse_path( path ):\n",
    "    \"\"\"\n",
    "    This function parses neo4j results.\n",
    "    :param path: neo4j path object\n",
    "    :return: parsed path dictionary\n",
    "    \"\"\"\n",
    "\n",
    "    out = {}\n",
    "    out['Nodes'] = []\n",
    "    for node in path['path'].nodes:\n",
    "        n = {}\n",
    "        n['idx'] = node.id\n",
    "        n['label'] = list(node.labels)[0]\n",
    "        n['id'] = node.get('id')\n",
    "        n['preflabel'] = node.get('preflabel')\n",
    "        n['name'] = node.get('name')\n",
    "        n['description'] = node.get('description')\n",
    "        out['Nodes'].append(n)\n",
    "    out['Edges'] = []\n",
    "    for edge in path['path'].relationships:\n",
    "        e = {}\n",
    "        e['idx'] = edge.id\n",
    "        e['start_node'] = edge.start_node.id\n",
    "        e['end_node'] = edge.end_node.id\n",
    "        e['type'] = edge.type\n",
    "        e['preflabel'] = edge.get('property_label')\n",
    "        e['references'] = edge.get('reference_uri')\n",
    "        out['Edges'].append(e)\n",
    "    return out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_node(source, target, port='7687'):\n",
    "    \"\"\"\n",
    "    This function checks if a node exists within the graph\n",
    "    \"\"\"\n",
    "    try:\n",
    "        driver = GraphDatabase.driver(\"bolt://localhost:{}\".format(port), auth=(\"neo4j\", \"ngly1\"))\n",
    "    except neo4j.exceptions.ServiceUnavailable:\n",
    "        raise\n",
    "    outputAll = list()\n",
    "    with driver.session() as session:\n",
    "        query = \"\"\"MATCH (source { id: '\"\"\" + source + \"\"\"' }), (target { id: '\"\"\" + target +  \"\"\"' }), p = allShortestPaths((source)-[*..15]-(target)) RETURN p\"\"\"\n",
    "        result = session.run(query)\n",
    "        print(result)\n",
    "        x = []\n",
    "        for record in result:\n",
    "            #path_dct = parse_path(record)\n",
    "            x.append(record)\n",
    "        print(x)\n",
    "        return x, result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "x, result = get_node('HGNC:4851', 'HGNC:18229')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "x[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "result.consume()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def query_shortest_paths(source, target, max_length=4, port='7687'):\n",
    "    \"\"\"\n",
    "    This function gets the shortest paths between source and\n",
    "    target. max_length determines the maximum path size allowed.\n",
    "    \"\"\"\n",
    "    try:\n",
    "        driver = GraphDatabase.driver(\"bolt://localhost:{}\".format(port), auth=(\"neo4j\", \"ngly1\"))\n",
    "    except neo4j.exceptions.ServiceUnavailable:\n",
    "        raise\n",
    "    outputAll = list()\n",
    "    with driver.session() as session:\n",
    "        print(\"kaas\")\n",
    "        query = \"\"\"MATCH (source { id: '\"\"\" + source + \"\"\"' }), (target { id: '\"\"\" + target +  \"\"\"' }), p = shortestPath((source)-[*..15]-(target)) RETURN p\"\"\"\n",
    "        result = session.run(query)\n",
    "        print(result)\n",
    "        return result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "seed = list([\n",
    "        'HGNC:4851',  # Htt human gene\n",
    "        'HGNC:18229'  # Rhes human gene\n",
    "])\n",
    "result = query_shortest_paths(seed[0], seed[1], max_length=999)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "summary = result.consume()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "summary.query"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "result.data()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%time\n",
    "# get orthopheno paths\n",
    "seed = list([\n",
    "        'HGNC:4851',  # Htt human gene\n",
    "        'HGNC:18229'  # Rhes human gene\n",
    "])\n",
    "hypothesis.query(seed,queryname='Htt_Rhes',port='7687') "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%time\n",
    "# get orthopheno paths relaxing pathway and phenotype node degrees\n",
    "seed = list([\n",
    "        'HGNC:4851',  # NGLY1 human gene\n",
    "        'HGNC:18229'  # AQP1 human gene\n",
    "])\n",
    "hypothesis.query(seed, queryname='Htt_Rhes', pwdegree='1000', phdegree='1000', port='7687')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%time\n",
    "# get orthopheno paths from a more open query topogology\n",
    "seed = list([\n",
    "        'HGNC:4851',  # NGLY1 human gene\n",
    "        'HGNC:18229'  # AQP1 human gene\n",
    "])\n",
    "hypothesis.open_query(seed,queryname='Htt_Rhes',port='7687')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import hypothesis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%time\n",
    "seed = list([\n",
    "        'HGNC:4851',  # NGLY1 human gene\n",
    "        'HGNC:18229'  # AQP1 human gene\n",
    "])\n",
    "r = hypothesis.shortest_paths(seed[0], seed[1], max_length=50, port='7687')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(r)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "%%time\n",
    "# get summary\n",
    "data = summary.path_load('./hypothesis/query_ngly1_aqp1_pwdl1000_phdl1000_paths_v2020-09-14.json')\n",
    "\n",
    "# parse data for summarization\n",
    "data_parsed = list()\n",
    "for query in data:\n",
    "    query_parsed = summary.query_parser(query)\n",
    "    data_parsed.append(query_parsed)\n",
    "summary.metapaths(data_parsed)\n",
    "summary.nodes(data_parsed)\n",
    "summary.node_types(data_parsed)\n",
    "summary.edges(data_parsed)\n",
    "summary.edge_types(data_parsed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
