{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# @name graph_v3.2_v20190616\n",
    "# @description notebook to build the NGLY1 Deficiency review knowledge graph v3.2\n",
    "# @author NÃºria Queralt Rosinach\n",
    "# @date 16 June 2019"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Description\n",
    "\n",
    "This is the notebook for the creation of the first review network and derived hypotheses. \n",
    "\n",
    "* Using intermediary variables from workflow objects. In this workflow variables are directly used for the next step. \n",
    "\n",
    "\n",
    "* Review network: From Monarch knowledge graph, we built a network seeded by 8 nodes, retrieving their explicit relationships and all the relationships among all these nodes. Seed nodes:\n",
    "\n",
    "    - 'MONDO:0007739' HD\n",
    "    - 'HGNC:4851' Htt\n",
    "    - 'CHEBI:18248' Iron (not working)\n",
    "    - 'HGNC:18229' Rhes (RASD2)\n",
    "    \n",
    "Possible seed nodes:\n",
    "https://monarchinitiative.org/search/Iron\n",
    "* Connecting paths: query templates."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import transcriptomics, regulation, curation, monarch, graph, neo4jlib, hypothesis, summary, utils\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#onarch.get_connections()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#nodes, edges = monarch.get_neighbours([\"HGNC:18229\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#len(nodes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#nodes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#nodes2, edges2 = monarch.get_neighbours(nodes)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Edges library\n",
    "### Review edges to integrate into the knowledge graph and prepare them as individual networks\n",
    "\n",
    "#### TRANSCRIPTOMICS NETWORK\n",
    "#### import transcriptomics\n",
    "We retrieved edges from RNA-seq transcriptomics profiles using the `transcriptomics` module:\n",
    "\n",
    "    - Experimental data sets: from Chow et al. paper [pmid:29346549] (NGLY1 deficiency model on fruit fly)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "The function \"read_data()\" is running...\n",
      "\n",
      "* This is the size of the raw expression data structure: (28087, 10)\n",
      "* These are the expression attributes: Index(['Unnamed: 0', 'symbol', 'baseMean', 'HD.mean', 'Control.mean',\n",
      "       'log2FoldChange', 'lfcSE', 'stat', 'pvalue', 'padj'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "           Unnamed: 0 symbol  baseMean    HD.mean  Control.mean  \\\n",
      "0  ENSG00000069011.10  PITX1  5.645675  18.684286      0.323793   \n",
      "\n",
      "   log2FoldChange     lfcSE      stat        pvalue          padj  \n",
      "0        4.769658  0.366367  13.01879  9.567529e-39  2.687232e-34  \n",
      "\n",
      "The raw data is saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/transcriptomics/HD/data/GSE64810_mlhd_DESeq2_diffexp_DESeq2_outlier_trimmed_adjust.csv\n",
      "\n",
      "\n",
      "Finished read_data().\n",
      "\n",
      "CPU times: user 45.9 ms, sys: 2.37 ms, total: 48.3 ms\n",
      "Wall time: 46.9 ms\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "# prepare data to graph schema\n",
    "csv_path = './transcriptomics/GSE64810_mlhd_DESeq2_diffexp_DESeq2_outlier_trimmed_adjust.txt'\n",
    "data = transcriptomics.read_data(csv_path, \"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "The function \"read_data()\" is running...\n",
      "\n",
      "* This is the size of the raw expression data structure: (28087, 10)\n",
      "* These are the expression attributes: Index(['Unnamed: 0', 'symbol', 'baseMean', 'HD.mean', 'Control.mean',\n",
      "       'log2FoldChange', 'lfcSE', 'stat', 'pvalue', 'padj'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "           Unnamed: 0 symbol  baseMean    HD.mean  Control.mean  \\\n",
      "0  ENSG00000069011.10  PITX1  5.645675  18.684286      0.323793   \n",
      "\n",
      "   log2FoldChange     lfcSE      stat        pvalue          padj  \n",
      "0        4.769658  0.366367  13.01879  9.567529e-39  2.687232e-34  \n",
      "\n",
      "The raw data is saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/transcriptomics/HD/data/GSE64810_mlhd_DESeq2_diffexp_DESeq2_outlier_trimmed_adjust.csv\n",
      "\n",
      "\n",
      "Finished read_data().\n",
      "\n",
      "\n",
      "The function \"clean_data()\" is running. Keeping only data with FC > 1.5 and FDR < 5% ...\n",
      "\n",
      "* This is the size of the clean expression data structure: (3209, 6)\n",
      "* These are the clean expression attributes: Index(['EnsembleID', 'symbol', 'log2FoldChange', 'pvalue', 'padj',\n",
      "       'Regulation'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "           EnsembleID symbol  log2FoldChange        pvalue          padj  \\\n",
      "0  ENSG00000069011.10  PITX1        4.769658  9.567529e-39  2.687232e-34   \n",
      "\n",
      "    Regulation  \n",
      "0  Upregulated  \n",
      "\n",
      "The clean data is saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/transcriptomics/HD/out/Transcriptome_human_BA9.csv\n",
      "\n",
      "\n",
      "Finished clean_data().\n",
      "\n",
      "\n",
      "The function \"prepare_data_edges()\" is running...\n",
      "\n",
      "* This is the size of the expression data structure: (3209, 13)\n",
      "* These are the expression attributes: Index(['ensemble_id', 'symbol', 'log2FoldChange', 'pvalue', 'padj',\n",
      "       'regulation', 'source', 'subject_id', 'subject_label', 'property_id',\n",
      "       'property_label', 'reference_id', 'object_id'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "          ensemble_id symbol  log2FoldChange        pvalue          padj  \\\n",
      "0  ENSG00000069011.10  PITX1        4.769658  9.567529e-39  2.687232e-34   \n",
      "\n",
      "    regulation source               subject_id subject_label property_id  \\\n",
      "0  Upregulated   Chow  ensembl:ENSG00000197386           HTT  RO:0002434   \n",
      "\n",
      "   property_label   reference_id                object_id  \n",
      "0  interacts with  PMID:26636579  ensembl:ENSG00000069011  \n",
      "\n",
      "The HD BA9 transcriptomics expression edges are saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/transcriptomics/HD/out/Transcriptome_human_BA9_edges.csv\n",
      "\n",
      "\n",
      "Finished prepare_data_edges().\n",
      "\n",
      "\n",
      "The function \"prepare_rna_edges()\" is running...\n",
      "\n",
      "* This is the size of the edges data structure: (3209, 12)\n",
      "* These are the edges attributes: Index(['subject_id', 'subject_label', 'property_id', 'property_label',\n",
      "       'object_id', 'object_label', 'log2FoldChange', 'pvalue', 'fdr',\n",
      "       'regulation', 'source', 'reference_id'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "                subject_id subject_label property_id  property_label  \\\n",
      "0  ensembl:ENSG00000197386           HTT  RO:0002434  interacts with   \n",
      "\n",
      "                 object_id object_label  log2FoldChange        pvalue  \\\n",
      "0  ensembl:ENSG00000069011        PITX1        4.769658  9.567529e-39   \n",
      "\n",
      "            fdr   regulation source   reference_id  \n",
      "0  2.687232e-34  Upregulated   Chow  PMID:26636579  \n",
      "\n",
      "This data object is not saved.\n",
      "\n",
      "\n",
      "Finished prepare_rna_edges().\n",
      "\n",
      "\n",
      "The function \"build_edges()\" is running...\n",
      "\n",
      "The function \"build_nodes()\" is running...\n",
      "* Total number of nodes: 3210\n",
      "querying 1-1000...done.\n",
      "querying 1001-2000...done.\n",
      "querying 2001-3000...done.\n",
      "querying 3001-3210...done.\n",
      "Finished.\n",
      "178 input query terms found no hit:\n",
      "\t['ENSG00000200127', 'ENSG00000183748', 'ENSG00000140181', 'ENSG00000261589', 'ENSG00000034063', 'ENS\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "178 nodes without available information\n",
      "These nodes have retained their original ID. First three nodes are:\n",
      "['ENSG00000200127', 'ENSG00000183748', 'ENSG00000140181']\n",
      "802 nodes do not have a HGNC id, retained original ID\n",
      "\n",
      "* This is the size of the nodes file data structure: (3210, 6)\n",
      "* These are the nodes attributes: Index(['id', 'semantic_groups', 'preflabel', 'name', 'synonyms',\n",
      "       'description'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "          id semantic_groups preflabel        name        synonyms  \\\n",
      "0  HGNC:4851            GENE       HTT  huntingtin  HD|IT15|LOMARS   \n",
      "\n",
      "                                         description  \n",
      "0  Huntingtin is a disease gene linked to Hunting...  \n",
      "\n",
      "The transcriptomics network nodes are built and saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/graph/rna_nodes_v2021-04-14.csv\n",
      "\n",
      "\n",
      "Finished build_nodes().\n",
      "\n",
      "\n",
      "* This is the size of the edges file data structure: (3209, 9)\n",
      "* These are the edges attributes: Index(['subject_id', 'object_id', 'property_id', 'property_label',\n",
      "       'property_description', 'property_uri', 'reference_uri',\n",
      "       'reference_supporting_text', 'reference_date'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "  subject_id  object_id property_id  property_label property_description  \\\n",
      "0  HGNC:4851  HGNC:9004  RO:0002434  interacts with                   NA   \n",
      "\n",
      "                                property_uri  \\\n",
      "0  http://purl.obolibrary.org/obo/RO_0002434   \n",
      "\n",
      "                                  reference_uri  \\\n",
      "0  https://www.ncbi.nlm.nih.gov/pubmed/26636579   \n",
      "\n",
      "                           reference_supporting_text reference_date  \n",
      "0  Here we present a genome-wide analysis of mRNA...     2015-11-04  \n",
      "\n",
      "The transcriptomics network edges are built and saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/graph/rna_edges_v2021-04-14.csv\n",
      "\n",
      "\n",
      "Finished build_edges().\n",
      "\n",
      "CPU times: user 1.42 s, sys: 21.2 ms, total: 1.44 s\n",
      "Wall time: 12.8 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "# prepare data to graph schema\n",
    "csv_path = './transcriptomics/GSE64810_mlhd_DESeq2_diffexp_DESeq2_outlier_trimmed_adjust.txt'\n",
    "data = transcriptomics.read_data(csv_path, \"\\t\")\n",
    "clean_data = transcriptomics.clean_data(data)\n",
    "data_edges = transcriptomics.prepare_data_edges(clean_data)\n",
    "rna_network = transcriptomics.prepare_rna_edges(data_edges)\n",
    "\n",
    "# build network with graph schema\n",
    "rna_edges = transcriptomics.build_edges(rna_network)\n",
    "rna_nodes, nodes_dict = transcriptomics.build_nodes(rna_network)\n",
    "rna_edges = transcriptomics.rework_edges(pd.DataFrame(rna_edges), nodes_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.DataFrame(rna_edges)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv(\"rna_edges_rework.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "* This is the size of the edges file data structure: (3209, 9)\n",
      "* These are the edges attributes: Index(['subject_id', 'object_id', 'property_id', 'property_label',\n",
      "       'property_description', 'property_uri', 'reference_uri',\n",
      "       'reference_supporting_text', 'reference_date'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "  subject_id  object_id property_id  property_label property_description  \\\n",
      "0  HGNC:4851  HGNC:9004  RO:0002434  interacts with                   NA   \n",
      "\n",
      "                                property_uri  \\\n",
      "0  http://purl.obolibrary.org/obo/RO_0002434   \n",
      "\n",
      "                                  reference_uri  \\\n",
      "0  https://www.ncbi.nlm.nih.gov/pubmed/26636579   \n",
      "\n",
      "                           reference_supporting_text reference_date  \n",
      "0  Here we present a genome-wide analysis of mRNA...     2015-11-04  \n",
      "\n",
      "The transcriptomics network edges are built and saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/graph/rna_edges_v2021-04-07.csv\n",
      "\n",
      "\n",
      "Finished build_edges().\n",
      "\n"
     ]
    }
   ],
   "source": [
    "test = transcriptomics.rework_edges(pd.DataFrame(rna_edges), nodes_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subject_id</th>\n",
       "      <th>object_id</th>\n",
       "      <th>property_id</th>\n",
       "      <th>property_label</th>\n",
       "      <th>property_description</th>\n",
       "      <th>property_uri</th>\n",
       "      <th>reference_uri</th>\n",
       "      <th>reference_supporting_text</th>\n",
       "      <th>reference_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HGNC:9004</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>NA</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/26636579</td>\n",
       "      <td>Here we present a genome-wide analysis of mRNA...</td>\n",
       "      <td>2015-11-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HGNC:5120</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>NA</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/26636579</td>\n",
       "      <td>Here we present a genome-wide analysis of mRNA...</td>\n",
       "      <td>2015-11-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HGNC:5122</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>NA</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/26636579</td>\n",
       "      <td>Here we present a genome-wide analysis of mRNA...</td>\n",
       "      <td>2015-11-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HGNC:5101</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>NA</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/26636579</td>\n",
       "      <td>Here we present a genome-wide analysis of mRNA...</td>\n",
       "      <td>2015-11-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HGNC:5100</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>NA</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/26636579</td>\n",
       "      <td>Here we present a genome-wide analysis of mRNA...</td>\n",
       "      <td>2015-11-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3204</th>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HGNC:55493</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>NA</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/26636579</td>\n",
       "      <td>Here we present a genome-wide analysis of mRNA...</td>\n",
       "      <td>2015-11-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3205</th>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>ensembl:ENSG00000181227</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>NA</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/26636579</td>\n",
       "      <td>Here we present a genome-wide analysis of mRNA...</td>\n",
       "      <td>2015-11-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3206</th>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HGNC:53465</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>NA</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/26636579</td>\n",
       "      <td>Here we present a genome-wide analysis of mRNA...</td>\n",
       "      <td>2015-11-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3207</th>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>HGNC:15722</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>NA</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/26636579</td>\n",
       "      <td>Here we present a genome-wide analysis of mRNA...</td>\n",
       "      <td>2015-11-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3208</th>\n",
       "      <td>HGNC:4851</td>\n",
       "      <td>ensembl:ENSG00000227220</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>NA</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/26636579</td>\n",
       "      <td>Here we present a genome-wide analysis of mRNA...</td>\n",
       "      <td>2015-11-04</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3209 rows Ã 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     subject_id                object_id property_id  property_label  \\\n",
       "0     HGNC:4851                HGNC:9004  RO:0002434  interacts with   \n",
       "1     HGNC:4851                HGNC:5120  RO:0002434  interacts with   \n",
       "2     HGNC:4851                HGNC:5122  RO:0002434  interacts with   \n",
       "3     HGNC:4851                HGNC:5101  RO:0002434  interacts with   \n",
       "4     HGNC:4851                HGNC:5100  RO:0002434  interacts with   \n",
       "...         ...                      ...         ...             ...   \n",
       "3204  HGNC:4851               HGNC:55493  RO:0002434  interacts with   \n",
       "3205  HGNC:4851  ensembl:ENSG00000181227  RO:0002434  interacts with   \n",
       "3206  HGNC:4851               HGNC:53465  RO:0002434  interacts with   \n",
       "3207  HGNC:4851               HGNC:15722  RO:0002434  interacts with   \n",
       "3208  HGNC:4851  ensembl:ENSG00000227220  RO:0002434  interacts with   \n",
       "\n",
       "     property_description                               property_uri  \\\n",
       "0                      NA  http://purl.obolibrary.org/obo/RO_0002434   \n",
       "1                      NA  http://purl.obolibrary.org/obo/RO_0002434   \n",
       "2                      NA  http://purl.obolibrary.org/obo/RO_0002434   \n",
       "3                      NA  http://purl.obolibrary.org/obo/RO_0002434   \n",
       "4                      NA  http://purl.obolibrary.org/obo/RO_0002434   \n",
       "...                   ...                                        ...   \n",
       "3204                   NA  http://purl.obolibrary.org/obo/RO_0002434   \n",
       "3205                   NA  http://purl.obolibrary.org/obo/RO_0002434   \n",
       "3206                   NA  http://purl.obolibrary.org/obo/RO_0002434   \n",
       "3207                   NA  http://purl.obolibrary.org/obo/RO_0002434   \n",
       "3208                   NA  http://purl.obolibrary.org/obo/RO_0002434   \n",
       "\n",
       "                                     reference_uri  \\\n",
       "0     https://www.ncbi.nlm.nih.gov/pubmed/26636579   \n",
       "1     https://www.ncbi.nlm.nih.gov/pubmed/26636579   \n",
       "2     https://www.ncbi.nlm.nih.gov/pubmed/26636579   \n",
       "3     https://www.ncbi.nlm.nih.gov/pubmed/26636579   \n",
       "4     https://www.ncbi.nlm.nih.gov/pubmed/26636579   \n",
       "...                                            ...   \n",
       "3204  https://www.ncbi.nlm.nih.gov/pubmed/26636579   \n",
       "3205  https://www.ncbi.nlm.nih.gov/pubmed/26636579   \n",
       "3206  https://www.ncbi.nlm.nih.gov/pubmed/26636579   \n",
       "3207  https://www.ncbi.nlm.nih.gov/pubmed/26636579   \n",
       "3208  https://www.ncbi.nlm.nih.gov/pubmed/26636579   \n",
       "\n",
       "                              reference_supporting_text reference_date  \n",
       "0     Here we present a genome-wide analysis of mRNA...     2015-11-04  \n",
       "1     Here we present a genome-wide analysis of mRNA...     2015-11-04  \n",
       "2     Here we present a genome-wide analysis of mRNA...     2015-11-04  \n",
       "3     Here we present a genome-wide analysis of mRNA...     2015-11-04  \n",
       "4     Here we present a genome-wide analysis of mRNA...     2015-11-04  \n",
       "...                                                 ...            ...  \n",
       "3204  Here we present a genome-wide analysis of mRNA...     2015-11-04  \n",
       "3205  Here we present a genome-wide analysis of mRNA...     2015-11-04  \n",
       "3206  Here we present a genome-wide analysis of mRNA...     2015-11-04  \n",
       "3207  Here we present a genome-wide analysis of mRNA...     2015-11-04  \n",
       "3208  Here we present a genome-wide analysis of mRNA...     2015-11-04  \n",
       "\n",
       "[3209 rows x 9 columns]"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- Transcriptomics network is returned as both digital object (`rna_edges`, `rna_nodes`) and CSV files at _**graph/**_ (`rna_edges_version.csv`, `rna_nodes_version.csv`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "type edges: <class 'list'>\n",
      "type nodes: <class 'list'>\n",
      "\n",
      "len edges: 3209\n",
      "len nodes: 3210\n",
      "\n",
      "attribute edges: Index(['subject_id', 'object_id', 'property_id', 'property_label',\n",
      "       'property_description', 'property_uri', 'reference_uri',\n",
      "       'reference_supporting_text', 'reference_date'],\n",
      "      dtype='object')\n",
      "attribute nodes: dict_keys(['id', 'semantic_groups', 'preflabel', 'name', 'synonyms', 'description'])\n"
     ]
    }
   ],
   "source": [
    "# print type of objects\n",
    "print('type edges:', type(rna_edges))\n",
    "print('type nodes:', type(rna_nodes))\n",
    "print()\n",
    "\n",
    "# print objects sizes\n",
    "print('len edges:', len(rna_edges))\n",
    "print('len nodes:', len(rna_nodes))\n",
    "print()\n",
    "\n",
    "# print object attribute\n",
    "print('attribute edges:', rna_edges[0].keys())\n",
    "print('attribute nodes:', rna_nodes[0].keys())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### REGULATION NETWORK\n",
    "#### import regulation\n",
    "\n",
    "We retrieved human TF gene expression regulation edges from several sources using the `regulation` module."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "The function \"prepare_msigdb_data()\" is running...\n",
      "\n",
      "* Number of Transcription Factor Targets (TFT) gene sets: 615\n",
      "\n",
      "The MSigDB raw network is saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/regulation/msigdb/out/tf_genelist_entrez_msigdb.json. Other reporting files are also saved at the same directory.\n",
      "\n",
      "\n",
      "Finished prepare_msigdb_data().\n",
      "\n",
      "\n",
      "The function \"load_tf_gene_edges()\" is running...\n",
      "\n",
      "Finished load_tf_gene_edges().\n",
      "\n",
      "\n",
      "The function \"get_gene_id_normalization_dictionaries()\" is running...\n",
      "\n",
      "* Querying BioThings to map gene symbols to HGNC and Entrez IDs...\n",
      "querying 1-1000...done.\n",
      "querying 1001-2000...done.\n",
      "querying 2001-3000...done.\n",
      "querying 3001-3071...done.\n",
      "Finished.\n",
      "48 input query terms found no hit:\n",
      "\t['E2F1DP1', 'ALPHACP1', 'E2F4DP1', 'COREBINDINGFACTOR', 'E2F4DP2', 'CEBPGAMMA', 'CDPCR3', 'AMEF2', '\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "Saving not found gene symbols at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/regulation/not_found_symbols.list\n",
      "\n",
      "\n",
      "* Querying BioThings to map Entrez to HGNC IDs and gene symbols...\n",
      "querying 1-1000...done.\n",
      "querying 1001-2000...done.\n",
      "querying 2001-3000...done.\n",
      "querying 3001-4000...done.\n",
      "querying 4001-5000...done.\n",
      "querying 5001-6000...done.\n",
      "querying 6001-7000...done.\n",
      "querying 7001-8000...done.\n",
      "querying 8001-9000...done.\n",
      "querying 9001-10000...done.\n",
      "querying 10001-11000...done.\n",
      "querying 11001-12000...done.\n",
      "querying 12001-13000...done.\n",
      "querying 13001-14000...done.\n",
      "querying 14001-15000...done.\n",
      "querying 15001-16000...done.\n",
      "querying 16001-16632...done.\n",
      "Finished.\n",
      "144 input query terms found no hit:\n",
      "\t['284608', '219392', '9142', '286166', '285229', '100500805', '326332', '145609', '244', '284072', '\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "Saving not found Entrez gene IDs at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/regulation/not_found_entrez.list\n",
      "\n",
      "\n",
      "Finished get_gene_id_normalization_dictionaries().\n",
      "\n",
      "\n",
      "The function \"prepare_data_edges()\" is running...\n",
      "\n",
      "The tftargets edges are saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/regulation/tftargets/tftargets_edges.csv\n",
      "\n",
      "\n",
      "The MSigDB edges are saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/regulation/msigdb/msigdb_edges.csv\n",
      "\n",
      "\n",
      "Finished prepare_data_edges().\n",
      "\n",
      "\n",
      "The function \"prepare_regulation_edges()\" is running...\n",
      "\n",
      "Finished prepare_regulation_edges().\n",
      "\n",
      "\n",
      "The function \"build_edges()\" is running...\n",
      "\n",
      "* This is the size of the edges file data structure: (197267, 9)\n",
      "* These are the edges attributes: Index(['subject_id', 'object_id', 'property_id', 'property_label',\n",
      "       'property_description', 'property_uri', 'reference_uri',\n",
      "       'reference_supporting_text', 'reference_date'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "  subject_id  object_id property_id  property_label property_description  \\\n",
      "0  HGNC:8615  HGNC:8803  RO:0002434  interacts with                   NA   \n",
      "\n",
      "                                property_uri  \\\n",
      "0  http://purl.obolibrary.org/obo/RO_0002434   \n",
      "\n",
      "                                  reference_uri  \\\n",
      "0  https://www.ncbi.nlm.nih.gov/pubmed/17202159   \n",
      "\n",
      "                           reference_supporting_text reference_date  \n",
      "0  This edge comes from the TRED dataset in \"tfta...     2007-01-01  \n",
      "\n",
      "The regulation network edges are built and saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/graph/regulation_edges_v2021-04-07.csv\n",
      "\n",
      "\n",
      "Finished build_edges().\n",
      "\n",
      "\n",
      "The function \"build_nodes()\" is running...\n",
      "\n",
      "* Total number of nodes: 16968\n",
      "\n",
      "* Trap genes without gene symbol, i.e. genes with discontinued entrez ID...\n",
      "* Number of concepts without gene symbol: 144\n",
      "* Check that all genes without gene symbol are identified by entrez ID...\n",
      "* Number of concepts without gene symbol by namespace:  NCBIGene    144\n",
      "Name: id, dtype: int64\n",
      "\n",
      "* Querying BioThings to map retired Entrez to gene symbols...\n",
      "querying 1-144...done.\n",
      "Finished.\n",
      "90 input query terms found no hit:\n",
      "\t['79907', '79911', '93333', '121301', '143902', '146856', '151720', '197379', '219392', '221943', '2\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "* Querying BioThings to retrieve node attributes...\n",
      "querying 1-1000...done.\n",
      "querying 1001-2000...done.\n",
      "querying 2001-3000...done.\n",
      "querying 3001-4000...done.\n",
      "querying 4001-5000...done.\n",
      "querying 5001-6000...done.\n",
      "querying 6001-7000...done.\n",
      "querying 7001-8000...done.\n",
      "querying 8001-9000...done.\n",
      "querying 9001-10000...done.\n",
      "querying 10001-11000...done.\n",
      "querying 11001-12000...done.\n",
      "querying 12001-13000...done.\n",
      "querying 13001-14000...done.\n",
      "querying 14001-15000...done.\n",
      "querying 15001-16000...done.\n",
      "querying 16001-16844...done.\n",
      "Finished.\n",
      "48 input query terms found no hit:\n",
      "\t['E2F4DP1', 'E2F4DP2', 'CEBPGAMMA', 'CDPCR3', 'AMEF2', 'TCF11MAFG', 'NFMUE1', 'LMO2COM', 'COMP1', 'C\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "* This is the size of the nodes file data structure: (16968, 6)\n",
      "* These are the nodes attributes: Index(['id', 'semantic_groups', 'preflabel', 'name', 'synonyms',\n",
      "       'description'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "          id semantic_groups preflabel          name    synonyms  \\\n",
      "0  HGNC:8615            GENE      PAX1  paired box 1  HUP48|OFC2   \n",
      "\n",
      "                                         description  \n",
      "0  This gene is a member of the paired box (PAX) ...  \n",
      "\n",
      "The regulation network nodes are built and saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/graph/regulation_nodes_v2021-04-07.csv\n",
      "\n",
      "\n",
      "Finished build_nodes().\n",
      "\n",
      "CPU times: user 1min 4s, sys: 255 ms, total: 1min 4s\n",
      "Wall time: 3min 20s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "# prepare msigdb data\n",
    "gmt_path = './regulation/msigdb/data/c3.tft.v6.1.entrez.gmt'\n",
    "regulation.prepare_msigdb_data(gmt_path)\n",
    "\n",
    "# prepare individual networks\n",
    "data = regulation.load_tf_gene_edges()\n",
    "dicts = regulation.get_gene_id_normalization_dictionaries(data)\n",
    "data_edges = regulation.prepare_data_edges(data, dicts)\n",
    "\n",
    "# prepare regulation network\n",
    "reg_network = regulation.prepare_regulation_edges(data_edges)\n",
    "\n",
    "# build network with graph schema\n",
    "reg_edges = regulation.build_edges(reg_network)\n",
    "reg_nodes = regulation.build_nodes(reg_network)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- Regulation network is returned as both digital object (`reg_edges`, `reg_nodes`) and CSV files at _**graph/**_ (`regulation_edges_version.csv`, `regulation_nodes_version.csv`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subject_id</th>\n",
       "      <th>object_id</th>\n",
       "      <th>property_id</th>\n",
       "      <th>property_label</th>\n",
       "      <th>property_description</th>\n",
       "      <th>property_uri</th>\n",
       "      <th>reference_uri</th>\n",
       "      <th>reference_supporting_text</th>\n",
       "      <th>reference_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HGNC:8615</td>\n",
       "      <td>HGNC:8803</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>NA</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>2007-01-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HGNC:8615</td>\n",
       "      <td>HGNC:12687</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>NA</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>2007-01-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HGNC:8615</td>\n",
       "      <td>HGNC:10970</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>NA</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>2007-01-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HGNC:8615</td>\n",
       "      <td>HGNC:10972</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>NA</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>2007-01-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HGNC:8615</td>\n",
       "      <td>HGNC:25567</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>NA</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/17202159</td>\n",
       "      <td>This edge comes from the TRED dataset in \"tfta...</td>\n",
       "      <td>2007-01-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197262</th>\n",
       "      <td>HGNC:16764</td>\n",
       "      <td>HGNC:10931</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>NA</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197263</th>\n",
       "      <td>HGNC:16764</td>\n",
       "      <td>HGNC:24561</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>NA</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197264</th>\n",
       "      <td>HGNC:16764</td>\n",
       "      <td>HGNC:11209</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>NA</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197265</th>\n",
       "      <td>HGNC:16764</td>\n",
       "      <td>HGNC:26314</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>NA</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197266</th>\n",
       "      <td>HGNC:16764</td>\n",
       "      <td>HGNC:26137</td>\n",
       "      <td>RO:0002434</td>\n",
       "      <td>interacts with</td>\n",
       "      <td>NA</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002434</td>\n",
       "      <td>http://www.broadinstitute.org/gsea/msigdb/card...</td>\n",
       "      <td>This edge comes from the C3:TFT dataset in \"ms...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>197267 rows Ã 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        subject_id   object_id property_id  property_label  \\\n",
       "0        HGNC:8615   HGNC:8803  RO:0002434  interacts with   \n",
       "1        HGNC:8615  HGNC:12687  RO:0002434  interacts with   \n",
       "2        HGNC:8615  HGNC:10970  RO:0002434  interacts with   \n",
       "3        HGNC:8615  HGNC:10972  RO:0002434  interacts with   \n",
       "4        HGNC:8615  HGNC:25567  RO:0002434  interacts with   \n",
       "...            ...         ...         ...             ...   \n",
       "197262  HGNC:16764  HGNC:10931  RO:0002434  interacts with   \n",
       "197263  HGNC:16764  HGNC:24561  RO:0002434  interacts with   \n",
       "197264  HGNC:16764  HGNC:11209  RO:0002434  interacts with   \n",
       "197265  HGNC:16764  HGNC:26314  RO:0002434  interacts with   \n",
       "197266  HGNC:16764  HGNC:26137  RO:0002434  interacts with   \n",
       "\n",
       "       property_description                               property_uri  \\\n",
       "0                        NA  http://purl.obolibrary.org/obo/RO_0002434   \n",
       "1                        NA  http://purl.obolibrary.org/obo/RO_0002434   \n",
       "2                        NA  http://purl.obolibrary.org/obo/RO_0002434   \n",
       "3                        NA  http://purl.obolibrary.org/obo/RO_0002434   \n",
       "4                        NA  http://purl.obolibrary.org/obo/RO_0002434   \n",
       "...                     ...                                        ...   \n",
       "197262                   NA  http://purl.obolibrary.org/obo/RO_0002434   \n",
       "197263                   NA  http://purl.obolibrary.org/obo/RO_0002434   \n",
       "197264                   NA  http://purl.obolibrary.org/obo/RO_0002434   \n",
       "197265                   NA  http://purl.obolibrary.org/obo/RO_0002434   \n",
       "197266                   NA  http://purl.obolibrary.org/obo/RO_0002434   \n",
       "\n",
       "                                            reference_uri  \\\n",
       "0            https://www.ncbi.nlm.nih.gov/pubmed/17202159   \n",
       "1            https://www.ncbi.nlm.nih.gov/pubmed/17202159   \n",
       "2            https://www.ncbi.nlm.nih.gov/pubmed/17202159   \n",
       "3            https://www.ncbi.nlm.nih.gov/pubmed/17202159   \n",
       "4            https://www.ncbi.nlm.nih.gov/pubmed/17202159   \n",
       "...                                                   ...   \n",
       "197262  http://www.broadinstitute.org/gsea/msigdb/card...   \n",
       "197263  http://www.broadinstitute.org/gsea/msigdb/card...   \n",
       "197264  http://www.broadinstitute.org/gsea/msigdb/card...   \n",
       "197265  http://www.broadinstitute.org/gsea/msigdb/card...   \n",
       "197266  http://www.broadinstitute.org/gsea/msigdb/card...   \n",
       "\n",
       "                                reference_supporting_text reference_date  \n",
       "0       This edge comes from the TRED dataset in \"tfta...     2007-01-01  \n",
       "1       This edge comes from the TRED dataset in \"tfta...     2007-01-01  \n",
       "2       This edge comes from the TRED dataset in \"tfta...     2007-01-01  \n",
       "3       This edge comes from the TRED dataset in \"tfta...     2007-01-01  \n",
       "4       This edge comes from the TRED dataset in \"tfta...     2007-01-01  \n",
       "...                                                   ...            ...  \n",
       "197262  This edge comes from the C3:TFT dataset in \"ms...            NaN  \n",
       "197263  This edge comes from the C3:TFT dataset in \"ms...            NaN  \n",
       "197264  This edge comes from the C3:TFT dataset in \"ms...            NaN  \n",
       "197265  This edge comes from the C3:TFT dataset in \"ms...            NaN  \n",
       "197266  This edge comes from the C3:TFT dataset in \"ms...            NaN  \n",
       "\n",
       "[197267 rows x 9 columns]"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reg_edges"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "type edges: <class 'list'>\n",
      "type nodes: <class 'list'>\n",
      "\n",
      "len edges: 197267\n",
      "len nodes: 16968\n",
      "\n",
      "attribute edges: dict_keys(['subject_id', 'object_id', 'property_id', 'property_label', 'property_description', 'property_uri', 'reference_uri', 'reference_supporting_text', 'reference_date'])\n",
      "attribute nodes: dict_keys(['id', 'semantic_groups', 'preflabel', 'name', 'synonyms', 'description'])\n"
     ]
    }
   ],
   "source": [
    "# print type of objects\n",
    "print('type edges:', type(reg_edges))\n",
    "print('type nodes:', type(reg_nodes))\n",
    "print()\n",
    "\n",
    "# print objects sizes\n",
    "print('len edges:', len(reg_edges))\n",
    "print('len nodes:', len(reg_nodes))\n",
    "print()\n",
    "\n",
    "# print object attribute\n",
    "print('attribute edges:', reg_edges[0].keys())\n",
    "print('attribute nodes:', reg_nodes[0].keys())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### CURATED NETWORK\n",
    "#### import curation\n",
    "\n",
    "We retrieved and prepared curated edges using the `curation` module. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "curation_edges = pd.read_csv(\"curation/data/HD/Empty_edges.csv\")\n",
    "curation_nodes = pd.read_csv(\"curation/data/HD/Empty_nodes.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "The function \"read_network()\" is running...\n",
      "\n",
      "Reading and concatenating all curated statements in the network...\n",
      "\n",
      "* Curation edge files concatenated shape: (322, 22)\n",
      "\n",
      "Reading and concatenating all curated nodes in the network...\n",
      "\n",
      "* Curation node files concatenated shape: (318, 9)\n",
      "\n",
      "Finished read_network().\n",
      "\n",
      "\n",
      "The function \"prepare_data_edges()\" is running...\n",
      "\n",
      "Preparing curated network...\n",
      "\n",
      "Saving curated network at curation/...\n",
      "\n",
      "*Curated edges data structure shape: (321, 9)\n",
      "*Curated edges data structure fields: Index(['subject_id', 'property_id', 'object_id', 'reference_uri',\n",
      "       'reference_supporting_text', 'reference_date', 'property_label',\n",
      "       'property_description', 'property_uri'],\n",
      "      dtype='object')\n",
      "\n",
      "The curated edges are saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/curation/curated_edges_v2021-04-07.csv\n",
      "\n",
      "\n",
      "Finished prepare_data_edges().\n",
      "\n",
      "\n",
      "The function \"prepare_data_nodes()\" is running...\n",
      "\n",
      "Preparing curated nodes...\n",
      "\n",
      "Saving curated nodes at curation/...\n",
      "\n",
      "*Curated nodes data structure shape: (288, 5)\n",
      "*Curated nodes data structure fields: Index(['id', 'semantic_groups', 'preflabel', 'synonyms', 'description'], dtype='object')\n",
      "\n",
      "The curated nodes are saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/curation/curated_nodes_v2021-04-07.csv\n",
      "\n",
      "\n",
      "Finished prepare_data_nodes().\n",
      "\n",
      "\n",
      "The function \"prepare_curated_edges()\" is running...\n",
      "\n",
      "Preparing curated edges to graph schema...\n",
      "\n",
      "Mapping genes to HGNC ID...\n",
      "\n",
      "Mapping genes to HGNC ID...\n",
      "querying 1-18...done.\n",
      "Finished.\n",
      "\n",
      "Adding diseases to MONDO ID network...\n",
      "\n",
      "Adding gene to protein network...\n",
      "querying 1-43...done.\n",
      "Finished.\n",
      "\n",
      "Drop duplicated gene-protein relations...\n"
     ]
    },
    {
     "ename": "KeyError",
     "evalue": "'property_label'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m~/miniconda3/envs/HDSR/lib/python3.9/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   3079\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3080\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcasted_key\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3081\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'property_label'",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n",
      "\u001b[0;32m~/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/curation.py\u001b[0m in \u001b[0;36mprepare_curated_edges\u001b[0;34m(edges_df)\u001b[0m\n\u001b[1;32m    363\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    364\u001b[0m     \u001b[0;31m## Normalize PROTEINS2GENES: add g2p network\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 365\u001b[0;31m     \u001b[0medges\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnormalize_genes_to_proteins_to_graph\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0medges\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mconcept_dct\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    366\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'\\nFinished prepare_curated_edges().\\n'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    367\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/curation.py\u001b[0m in \u001b[0;36mnormalize_genes_to_proteins_to_graph\u001b[0;34m(curated_df, concept_dct)\u001b[0m\n\u001b[1;32m    335\u001b[0m     \u001b[0;31m# mark `has gene product` to be deleted if duplicated\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    336\u001b[0m     curated_df['g2p_mark'] = (\n\u001b[0;32m--> 337\u001b[0;31m         [curated_df.at[idx, 'property_label'] if 'has gene product' in curated_df.at[idx, 'property_label'] else\n\u001b[0m\u001b[1;32m    338\u001b[0m          idx for idx in curated_df.index]\n\u001b[1;32m    339\u001b[0m     )\n",
      "\u001b[0;32m~/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/curation.py\u001b[0m in \u001b[0;36m<listcomp>\u001b[0;34m(.0)\u001b[0m\n\u001b[1;32m    335\u001b[0m     \u001b[0;31m# mark `has gene product` to be deleted if duplicated\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    336\u001b[0m     curated_df['g2p_mark'] = (\n\u001b[0;32m--> 337\u001b[0;31m         [curated_df.at[idx, 'property_label'] if 'has gene product' in curated_df.at[idx, 'property_label'] else\n\u001b[0m\u001b[1;32m    338\u001b[0m          idx for idx in curated_df.index]\n\u001b[1;32m    339\u001b[0m     )\n",
      "\u001b[0;32m~/miniconda3/envs/HDSR/lib/python3.9/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   2154\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2155\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2156\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0msuper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__getitem__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2157\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2158\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__setitem__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/HDSR/lib/python3.9/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   2101\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2102\u001b[0m         \u001b[0mkey\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_convert_key\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2103\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_value\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtakeable\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_takeable\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2104\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2105\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__setitem__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/HDSR/lib/python3.9/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m_get_value\u001b[0;34m(self, index, col, takeable)\u001b[0m\n\u001b[1;32m   3127\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mseries\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_values\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3128\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3129\u001b[0;31m         \u001b[0mseries\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_item_cache\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcol\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3130\u001b[0m         \u001b[0mengine\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3131\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/HDSR/lib/python3.9/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m_get_item_cache\u001b[0;34m(self, item)\u001b[0m\n\u001b[1;32m   3790\u001b[0m             \u001b[0;31m#  pending resolution of GH#33047\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3791\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3792\u001b[0;31m             \u001b[0mloc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3793\u001b[0m             \u001b[0mvalues\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_mgr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0miget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3794\u001b[0m             \u001b[0mres\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_box_col_values\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mloc\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__finalize__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/HDSR/lib/python3.9/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   3080\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcasted_key\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3081\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3082\u001b[0;31m                 \u001b[0;32mraise\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3083\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3084\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mtolerance\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'property_label'"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "# graph v3.2\n",
    "# read network from drive and concat all curated statements\n",
    "curation_edges, curation_nodes = curation.read_network(version='v20180118')\n",
    "\n",
    "# prepare data edges and nodes\n",
    "data_edges = curation.prepare_data_edges(curation_edges)\n",
    "data_nodes = curation.prepare_data_nodes(curation_nodes)\n",
    "\n",
    "# prepare curated edges and nodes\n",
    "curated_network = curation.prepare_curated_edges(data_edges)\n",
    "curated_concepts = curation.prepare_curated_nodes(data_nodes)\n",
    "\n",
    "\n",
    "# build edges and nodes files\n",
    "curation_edges = curation.build_edges(curated_network)\n",
    "curation_nodes = curation.build_nodes(curated_concepts)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- Curated network is returned as both digital object (`curation_edges`, `curation_nodes`) and CSV files at _**graph/**_ (`curated_graph_edges_version.csv`, `curated_graph_nodes_version.csv`)\n",
    "- The original curated network, i.e. without graph data model normalization, is saved as CSV files at _**curation/**_ (`curated_edges_version.csv`, `curated_nodes_version.csv`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "type edges: <class 'pandas.core.frame.DataFrame'>\n",
      "type nodes: <class 'pandas.core.frame.DataFrame'>\n",
      "\n",
      "len edges: 322\n",
      "len nodes: 318\n",
      "\n"
     ]
    },
    {
     "ename": "KeyError",
     "evalue": "0",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m~/miniconda3/envs/HDSR/lib/python3.9/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   3079\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3080\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcasted_key\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3081\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 0",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-10-d154a0ffb6c7>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     10\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     11\u001b[0m \u001b[0;31m# print object attribute\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 12\u001b[0;31m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'attribute edges:'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcuration_edges\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mkeys\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     13\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'attribute nodes:'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcuration_nodes\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mkeys\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/HDSR/lib/python3.9/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3022\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnlevels\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3023\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getitem_multilevel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3024\u001b[0;31m             \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3025\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mis_integer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindexer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3026\u001b[0m                 \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mindexer\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/HDSR/lib/python3.9/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   3080\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcasted_key\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3081\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3082\u001b[0;31m                 \u001b[0;32mraise\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3083\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3084\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mtolerance\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 0"
     ]
    }
   ],
   "source": [
    "# print type of objects\n",
    "print('type edges:', type(curation_edges))\n",
    "print('type nodes:', type(curation_nodes))\n",
    "print()\n",
    "\n",
    "# print objects sizes\n",
    "print('len edges:', len(curation_edges))\n",
    "print('len nodes:', len(curation_nodes))\n",
    "print()\n",
    "\n",
    "# print object attribute\n",
    "print('attribute edges:', curation_edges[0].keys())\n",
    "print('attribute nodes:', curation_nodes[0].keys())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### MONARCH NETWORK\n",
    "#### import monarch\n",
    "We retrieved edges from Monarch using the `monarch` module.\n",
    "\n",
    "Tasks:\n",
    "\n",
    "- From 8 seed nodes we retrieved 1st shell nodes\n",
    "- From all seed and 1st shell nodes we retrieved ortho-phenotypes\n",
    "- We retrieved extra edges among all of them, i.e. extra connectivity between: seed, 1st shell, ortholog-phenotype nodes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%time\n",
    "# prepare data to graph schema\n",
    "# seed nodes\n",
    "seedList = [ \n",
    "    'MONDO:0007739', # HD\n",
    "    'HGNC:4851', # Htt\n",
    "    'HGNC:182293', # Rhes\n",
    "    'REACT:R-HSA-917937' #Iron uptake pathway\n",
    "] \n",
    "\n",
    "# get first shell of neighbours\n",
    "neighboursList = monarch.get_neighbours_list(seedList)\n",
    "print(len(neighboursList))\n",
    "# get second degree of neighbours\n",
    "monarch_network = monarch.get_connections(neighboursList)\n",
    "print(len(monarch_network))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "monarch.print_network(monarch_network, 'monarch_connections')\n",
    "\n",
    "# build network with graph schema \n",
    "monarch_edges = monarch.build_edges(monarch_network)\n",
    "monarch_nodes = monarch.build_nodes(monarch_network)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# print type of objects\n",
    "print('type edges:', type(monarch_edges))\n",
    "print('type nodes:', type(monarch_nodes))\n",
    "print()\n",
    "\n",
    "# print objects sizes\n",
    "print('len edges:', len(monarch_edges))\n",
    "print('len nodes:', len(monarch_nodes))\n",
    "print()\n",
    "\n",
    "# print object attribute\n",
    "print('attribute edges:', monarch_edges[0].keys())\n",
    "print('attribute nodes:', monarch_nodes[0].keys())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "%%time\n",
    "# prepare data to graph schema\n",
    "# seed nodes\n",
    "seedList = [ \n",
    "    'MONDO:0007739', # HD\n",
    "    'HGNC:4851', # Htt\n",
    "    'HGNC:182293', # Rhes\n",
    "] \n",
    "\n",
    "# get first shell of neighbours\n",
    "neighboursList = monarch.get_neighbours_list(seedList)\n",
    "print(len(neighboursList))\n",
    "\n",
    "# introduce animal model ortho-phenotypes for seed and 1st shell neighbors\n",
    "## For seed nodes:\n",
    "seed_orthophenoList = monarch.get_orthopheno_list(seedList)\n",
    "print(len(seed_orthophenoList))\n",
    "## For 1st shell nodes:\n",
    "neighbours_orthophenoList = monarch.get_orthopheno_list(neighboursList)\n",
    "print(len(neighbours_orthophenoList))\n",
    "\n",
    "# network nodes: seed + 1shell + ortholog-phentoype\n",
    "geneList = sum([seedList,\n",
    "                neighboursList,\n",
    "                #seed_orthophenoList,\n",
    "                #neighbours_orthophenoList\n",
    "               ], \n",
    "               [])\n",
    "print('genelist: ',len(geneList))\n",
    "\n",
    "# get Monarch network\n",
    "monarch_network = monarch.extract_edges(geneList)\n",
    "print('network: ',len(monarch_network))\n",
    "\n",
    "# save edges\n",
    "monarch.print_network(monarch_network, 'monarch_connections')\n",
    "\n",
    "# build network with graph schema \n",
    "monarch_edges = monarch.build_edges(monarch_network)\n",
    "monarch_nodes = monarch.build_nodes(monarch_network)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "monarch_network = monarch.read_connections(\"monarch_connections_v2020-11-15.csv\")\n",
    "monarch_edges = monarch.build_edges(monarch_network)\n",
    "monarch_nodes = monarch.build_nodes(monarch_network)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "monarch_edges = monarch.build_edges(monarch_network)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "monarch_nodes = monarch.build_nodes(monarch_network)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# print type of objects\n",
    "print('type edges:', type(monarch_edges))\n",
    "print('type nodes:', type(monarch_nodes))\n",
    "print()\n",
    "\n",
    "# print objects sizes\n",
    "print('len edges:', len(monarch_edges))\n",
    "print('len nodes:', len(monarch_nodes))\n",
    "print()\n",
    "\n",
    "# print object attribute\n",
    "print('attribute edges:', monarch_edges[0].keys())\n",
    "print('attribute nodes:', monarch_nodes[0].keys())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- Monarch network is returned as both digital object (`monarch_edges`, `monarch_nodes`) and CSV files at _**monarch/**_ (`monarch_edges_version.csv`, `monarch_nodes_version.csv`)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Graph library\n",
    "### Create the review knowledge graph\n",
    "#### import graph\n",
    "\n",
    "Tasks:\n",
    "\n",
    "* Load Networks and calculate graph nodes\n",
    "* Retrieve extra connectivity for the graph from Monarch\n",
    "* Build the review graph"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "%%time\n",
    "# load networks and calculate graph nodes\n",
    "graph_nodes_list, reg_graph_edges = graph.graph_nodes(\n",
    "    curation=curation_edges,\n",
    "    monarch=monarch_edges,\n",
    "    transcriptomics=rna_edges,\n",
    "    regulation=reg_edges\n",
    ")\n",
    "\n",
    "# Monarch graph connectivity\n",
    "## get Monarch edges\n",
    "monarch_network_graph = monarch.extract_edges(graph_nodes_list)\n",
    "print('network: ',len(monarch_network_graph))\n",
    "\n",
    "## save Monarch network\n",
    "monarch.print_network(monarch_network_graph, 'monarch_connections_graph')\n",
    "\n",
    "## build Monarch network with graph schema\n",
    "monarch_graph_edges = monarch.build_edges(monarch_network_graph)\n",
    "monarch_graph_nodes = monarch.build_nodes(monarch_network_graph)\n",
    "\n",
    "# build review graph\n",
    "edges = graph.build_edges(\n",
    "    curation=curation_edges,\n",
    "    monarch=monarch_graph_edges,\n",
    "    transcriptomics=rna_edges,\n",
    "    regulation=reg_graph_edges\n",
    ")\n",
    "nodes = graph.build_nodes(\n",
    "    statements=edges,\n",
    "    curation=curation_nodes,\n",
    "    monarch=monarch_graph_nodes,\n",
    "    transcriptomics=rna_nodes,\n",
    "    regulation=reg_nodes\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# obtain variables:\n",
    "curation_edges = pd.read_csv(\"curation/data/HD/Empty_edges.csv\")\n",
    "curation_nodes = pd.read_csv(\"curation/data/HD/Empty_nodes.csv\")\n",
    "\n",
    "#monarch_network = monarch.read_connections(\"monarch_connections_v2020-11-15.csv\")\n",
    "#monarch_edges = monarch.build_edges(monarch_network)\n",
    "#monarch_nodes = monarch.build_nodes(monarch_network)\n",
    "\n",
    "#monarch_network_graph = monarch.read_connections(\"monarch_connections_graph_v2020-11-16.csv\")\n",
    "#monarch_graph_edges = monarch.build_edges(monarch_network_graph)\n",
    "#monarch_graph_nodes = monarch.build_nodes(monarch_network_graph)\n",
    "\n",
    "csv_path = './transcriptomics/GSE64810_mlhd_DESeq2_diffexp_DESeq2_outlier_trimmed_adjust.txt'\n",
    "data = transcriptomics.read_data(csv_path)\n",
    "clean_data = transcriptomics.clean_data(data)\n",
    "data_edges = transcriptomics.prepare_data_edges(clean_data)\n",
    "rna_network = transcriptomics.prepare_rna_edges(data_edges)\n",
    "\n",
    "# build network with graph schema\n",
    "rna_edges = transcriptomics.build_edges(rna_network)\n",
    "rna_nodes = transcriptomics.build_nodes(rna_network)\n",
    "\n",
    "gmt_path = './regulation/msigdb/data/c3.tft.v6.1.entrez.gmt'\n",
    "regulation.prepare_msigdb_data(gmt_path)\n",
    "\n",
    "# prepare individual networks\n",
    "data = regulation.load_tf_gene_edges()\n",
    "dicts = regulation.get_gene_id_normalization_dictionaries(data)\n",
    "data_edges = regulation.prepare_data_edges(data, dicts)\n",
    "\n",
    "# prepare regulation network\n",
    "reg_network = regulation.prepare_regulation_edges(data_edges)\n",
    "\n",
    "# build network with graph schema\n",
    "reg_edges = regulation.build_edges(reg_network)\n",
    "reg_nodes = regulation.build_nodes(reg_network)\n",
    "\n",
    "#graph_nodes_list, reg_graph_edges = graph.graph_nodes(\n",
    "#    curation=curation_edges,\n",
    "#    monarch=monarch_edges,\n",
    "#    transcriptomics=rna_edges,\n",
    "#    regulation=reg_edges\n",
    "#)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# fix curation data:\n",
    "curation_nodes = curation_nodes.astype('object')\n",
    "#curation_nodes.loc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "curation_nodes['name'] = [\"NaN\", \"NaN\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "curation_nodes.loc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# construct df's\n",
    "edges = graph.build_edges(\n",
    "    curation=curation_edges,\n",
    "    monarch=monarch_graph_edges,\n",
    "    transcriptomics=rna_edges,\n",
    "    regulation=reg_graph_edges\n",
    ")\n",
    "nodes = graph.build_nodes(\n",
    "    statements=edges,\n",
    "    curation=curation_nodes,\n",
    "    monarch=monarch_graph_nodes,\n",
    "    transcriptomics=rna_nodes,\n",
    "    regulation=reg_nodes\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "nodes = pd.read_csv(\"/home/karolis/Structured review/bioknowledge-reviewer/bioknowledge_reviewer/graph/graph_nodes_v2020-11-20.csv\")\n",
    "edges = pd.read_csv(\"/home/karolis/Structured review/bioknowledge-reviewer/bioknowledge_reviewer/graph/graph_edges_v2020-11-20.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- Regulation edges _merged_ with the graph is returned as both digital object (`reg_graph_edges`) and CSV file at _**graph/**_ (`regulation_graph_edges_version.csv`)\n",
    "- Monarch network is returned as both digital object (`monarch_graph_edges`, `monarch_graph_nodes`) and CSV files at _**monarch/**_ (`monarch_edges_version.csv`, `monarch_nodes_version.csv`) overwritten the previous one.\n",
    "- Review knowledge graph is returned as both digital object (`edges`, `nodes`) and CSV files at _**graph/**_ (`graph_edges_version.csv`, `graph_nodes_version.csv`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# print type of objects\n",
    "print('type edges:', type(edges))\n",
    "print('type nodes:', type(nodes))\n",
    "print()\n",
    "\n",
    "# print objects sizes\n",
    "print('len edges:', len(edges))\n",
    "print('len nodes:', len(nodes))\n",
    "print()\n",
    "\n",
    "# print object attribute\n",
    "print('attribute edges:', edges.columns)\n",
    "print('attribute nodes:', nodes.columns)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Neo4jlib library\n",
    "### Import the graph into Neo4j graph database\n",
    "#### import neo4jlib\n",
    "\n",
    "Tasks:\n",
    "\n",
    "- Create Neo4j server instance\n",
    "- Import review graph into the Neo4j graph database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "curation_edges = pd.read_csv(\"curation/data/HD/Empty_edges.csv\")\n",
    "curation_nodes = pd.read_csv(\"curation/data/HD/Empty_nodes.csv\")\n",
    "curation_nodes = curation_nodes.astype('object')\n",
    "curation_nodes['name'] = [\"NaN\", \"NaN\"]\n",
    "curation_edges = curation_edges.to_dict(\"records\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "* This is the size of the data structure: (233182, 7)\n",
      "* These are the attributes: Index(['object_id', 'object_label', 'reference_id_list', 'relation_id',\n",
      "       'relation_label', 'subject_id', 'subject_label'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "    object_id object_label  reference_id_list    relation_id  \\\n",
      "0  MGI:892004        H2-Bl                NaN  RO:HOM0000017   \n",
      "\n",
      "                   relation_label   subject_id subject_label  \n",
      "0  in orthology relationship with  RGD:1595815      RT1-M6-2  \n",
      "\n",
      "The function \"build_edges()\" is running...\n",
      "df (233182, 9)\n",
      "\n",
      "* This is the size of the edges file data structure: (233182, 9)\n",
      "* These are the edges attributes: Index(['subject_id', 'object_id', 'property_id', 'property_label',\n",
      "       'property_description', 'property_uri', 'reference_uri',\n",
      "       'reference_supporting_text', 'reference_date'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "    subject_id   object_id    property_id                  property_label  \\\n",
      "0  RGD:1595815  MGI:892004  RO:HOM0000017  in orthology relationship with   \n",
      "\n",
      "  property_description                                  property_uri  \\\n",
      "0                   NA  http://purl.obolibrary.org/obo/RO_HOM0000017   \n",
      "\n",
      "  reference_uri                          reference_supporting_text  \\\n",
      "0            NA  This edge comes from the Monarch Knowledge Gra...   \n",
      "\n",
      "  reference_date  \n",
      "0             NA  \n",
      "\n",
      "The Monarch network edges are built and saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/monarch/monarch_edges_v2021-04-12.csv\n",
      "\n",
      "\n",
      "Finished build_edges().\n",
      "\n",
      "\n",
      "The function \"build_nodes()\" is running...\n",
      "Number of concepts: 14018\n",
      "Number of nodes CURIEs: 31\n",
      "List of nodes CURIEs: dict_keys(['RGD', 'MGI', 'ENSEMBL', 'FlyBase', 'SGD', 'MONARCH', 'HGNC', 'ZFIN', 'HP', 'MP', 'WormBase', 'dictyBase', 'GO', 'ZP', 'PomBase', 'UBERON', 'WBPhenotype', 'Xenbase', 'APO', 'EnsemblGenome', 'EcoGene', 'FBcv', 'MONDO', 'NCBIGene', 'REACT', 'EFO', 'Coriell', 'dbSNP', 'OMIM', 'dbSNPIndividual', 'ClinVarVariant'])\n",
      "\n",
      "Adding BioThings annotation: gene name, synonyms, description...\n",
      "symbols: 7417\n",
      "querying 1-1000...done.\n",
      "querying 1001-2000...done.\n",
      "querying 2001-3000...done.\n",
      "querying 3001-4000...done.\n",
      "querying 4001-5000...done.\n",
      "querying 5001-6000...done.\n",
      "querying 6001-7000...done.\n",
      "querying 7001-7417...done.\n",
      "Finished.\n",
      "819 input query terms found dup hits:\n",
      "\t[('Rheb', 3), ('Dnajb6', 2), ('SKP1', 4), ('Skp1', 2), ('BCAR1', 4), ('CNOT2', 4), ('DDB1', 4), ('He\n",
      "3836 input query terms found no hit:\n",
      "\t['A0A0G2K8R6', 'ENSEMBL:ENSOANG00000006132', 'ENSEMBL:ENSCAFG00000017647', 'ENSEMBL:ENSECAG000000190\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "* This is the size of the nodes file data structure: (14018, 6)\n",
      "* These are the nodes attributes: Index(['id', 'semantic_groups', 'preflabel', 'name', 'synonyms',\n",
      "       'description'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "            id semantic_groups preflabel                           name  \\\n",
      "0  RGD:1595815            GENE  RT1-M6-2  RT1 class I, locus M6, gene 2   \n",
      "\n",
      "  synonyms description  \n",
      "0       NA          NA  \n",
      "\n",
      "The Monarch network nodes are built and saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/monarch/monarch_nodes_v2021-04-12.csv\n",
      "\n",
      "\n",
      "Finished build_nodes().\n",
      "\n",
      "\n",
      "The function \"read_data()\" is running...\n",
      "\n",
      "* This is the size of the raw expression data structure: (28087, 10)\n",
      "* These are the expression attributes: Index(['Unnamed: 0', 'symbol', 'baseMean', 'HD.mean', 'Control.mean',\n",
      "       'log2FoldChange', 'lfcSE', 'stat', 'pvalue', 'padj'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "           Unnamed: 0 symbol  baseMean    HD.mean  Control.mean  \\\n",
      "0  ENSG00000069011.10  PITX1  5.645675  18.684286      0.323793   \n",
      "\n",
      "   log2FoldChange     lfcSE      stat        pvalue          padj  \n",
      "0        4.769658  0.366367  13.01879  9.567529e-39  2.687232e-34  \n",
      "\n",
      "The raw data is saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/transcriptomics/HD/data/GSE64810_mlhd_DESeq2_diffexp_DESeq2_outlier_trimmed_adjust.csv\n",
      "\n",
      "\n",
      "Finished read_data().\n",
      "\n",
      "\n",
      "The function \"clean_data()\" is running. Keeping only data with FC > 1.5 and FDR < 5% ...\n",
      "\n",
      "* This is the size of the clean expression data structure: (3209, 6)\n",
      "* These are the clean expression attributes: Index(['EnsembleID', 'symbol', 'log2FoldChange', 'pvalue', 'padj',\n",
      "       'Regulation'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "           EnsembleID symbol  log2FoldChange        pvalue          padj  \\\n",
      "0  ENSG00000069011.10  PITX1        4.769658  9.567529e-39  2.687232e-34   \n",
      "\n",
      "    Regulation  \n",
      "0  Upregulated  \n",
      "\n",
      "The clean data is saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/transcriptomics/HD/out/Transcriptome_human_BA9.csv\n",
      "\n",
      "\n",
      "Finished clean_data().\n",
      "\n",
      "\n",
      "The function \"prepare_data_edges()\" is running...\n",
      "\n",
      "* This is the size of the expression data structure: (3209, 13)\n",
      "* These are the expression attributes: Index(['ensemble_id', 'symbol', 'log2FoldChange', 'pvalue', 'padj',\n",
      "       'regulation', 'source', 'subject_id', 'subject_label', 'property_id',\n",
      "       'property_label', 'reference_id', 'object_id'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "          ensemble_id symbol  log2FoldChange        pvalue          padj  \\\n",
      "0  ENSG00000069011.10  PITX1        4.769658  9.567529e-39  2.687232e-34   \n",
      "\n",
      "    regulation source               subject_id subject_label property_id  \\\n",
      "0  Upregulated   Chow  ensembl:ENSG00000197386           HTT  RO:0002434   \n",
      "\n",
      "   property_label   reference_id                object_id  \n",
      "0  interacts with  PMID:26636579  ensembl:ENSG00000069011  \n",
      "\n",
      "The HD BA9 transcriptomics expression edges are saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/transcriptomics/HD/out/Transcriptome_human_BA9_edges.csv\n",
      "\n",
      "\n",
      "Finished prepare_data_edges().\n",
      "\n",
      "\n",
      "The function \"prepare_rna_edges()\" is running...\n",
      "\n",
      "* This is the size of the edges data structure: (3209, 12)\n",
      "* These are the edges attributes: Index(['subject_id', 'subject_label', 'property_id', 'property_label',\n",
      "       'object_id', 'object_label', 'log2FoldChange', 'pvalue', 'fdr',\n",
      "       'regulation', 'source', 'reference_id'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "                subject_id subject_label property_id  property_label  \\\n",
      "0  ensembl:ENSG00000197386           HTT  RO:0002434  interacts with   \n",
      "\n",
      "                 object_id object_label  log2FoldChange        pvalue  \\\n",
      "0  ensembl:ENSG00000069011        PITX1        4.769658  9.567529e-39   \n",
      "\n",
      "            fdr   regulation source   reference_id  \n",
      "0  2.687232e-34  Upregulated   Chow  PMID:26636579  \n",
      "\n",
      "This data object is not saved.\n",
      "\n",
      "\n",
      "Finished prepare_rna_edges().\n",
      "\n",
      "\n",
      "The function \"build_edges()\" is running...\n",
      "\n",
      "The function \"build_nodes()\" is running...\n",
      "* Total number of nodes: 3210\n",
      "querying 1-1000...done.\n",
      "querying 1001-2000...done.\n",
      "querying 2001-3000...done.\n",
      "querying 3001-3210...done.\n",
      "Finished.\n",
      "178 input query terms found no hit:\n",
      "\t['ENSG00000200127', 'ENSG00000183748', 'ENSG00000140181', 'ENSG00000261589', 'ENSG00000034063', 'ENS\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "178 nodes without available information\n",
      "These nodes have retained their original ID. First three nodes are:\n",
      "['ENSG00000200127', 'ENSG00000183748', 'ENSG00000140181']\n",
      "802 nodes do not have a HGNC id, retained original ID\n",
      "\n",
      "* This is the size of the nodes file data structure: (3210, 6)\n",
      "* These are the nodes attributes: Index(['id', 'semantic_groups', 'preflabel', 'name', 'synonyms',\n",
      "       'description'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "          id semantic_groups preflabel        name        synonyms  \\\n",
      "0  HGNC:4851            GENE       HTT  huntingtin  HD|IT15|LOMARS   \n",
      "\n",
      "                                         description  \n",
      "0  Huntingtin is a disease gene linked to Hunting...  \n",
      "\n",
      "The transcriptomics network nodes are built and saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/graph/rna_nodes_v2021-04-12.csv\n",
      "\n",
      "\n",
      "Finished build_nodes().\n",
      "\n",
      "\n",
      "The function \"prepare_msigdb_data()\" is running...\n",
      "\n",
      "* Number of Transcription Factor Targets (TFT) gene sets: 615\n",
      "\n",
      "The MSigDB raw network is saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/regulation/msigdb/out/tf_genelist_entrez_msigdb.json. Other reporting files are also saved at the same directory.\n",
      "\n",
      "\n",
      "Finished prepare_msigdb_data().\n",
      "\n",
      "\n",
      "The function \"load_tf_gene_edges()\" is running...\n",
      "\n",
      "Finished load_tf_gene_edges().\n",
      "\n",
      "\n",
      "The function \"get_gene_id_normalization_dictionaries()\" is running...\n",
      "\n",
      "* Querying BioThings to map gene symbols to HGNC and Entrez IDs...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "querying 1-1000...done.\n",
      "querying 1001-2000...done.\n",
      "querying 2001-3000...done.\n",
      "querying 3001-3071...done.\n",
      "Finished.\n",
      "48 input query terms found no hit:\n",
      "\t['MMEF2', 'SIN3AK20', 'CETS1P54', 'CDPCR3', 'AHRARNT', 'ISRE', 'TAL1BETAE47', 'COMP1', 'CEBPGAMMA', \n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "Saving not found gene symbols at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/regulation/not_found_symbols.list\n",
      "\n",
      "\n",
      "* Querying BioThings to map Entrez to HGNC IDs and gene symbols...\n",
      "querying 1-1000...done.\n",
      "querying 1001-2000...done.\n",
      "querying 2001-3000...done.\n",
      "querying 3001-4000...done.\n",
      "querying 4001-5000...done.\n",
      "querying 5001-6000...done.\n",
      "querying 6001-7000...done.\n",
      "querying 7001-8000...done.\n",
      "querying 8001-9000...done.\n",
      "querying 9001-10000...done.\n",
      "querying 10001-11000...done.\n",
      "querying 11001-12000...done.\n",
      "querying 12001-13000...done.\n",
      "querying 13001-14000...done.\n",
      "querying 14001-15000...done.\n",
      "querying 15001-16000...done.\n",
      "querying 16001-16632...done.\n",
      "Finished.\n",
      "144 input query terms found no hit:\n",
      "\t['340432', '338850', '286121', '197379', '100506627', '143902', '100127889', '101243544', '150225', \n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "Saving not found Entrez gene IDs at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/regulation/not_found_entrez.list\n",
      "\n",
      "\n",
      "Finished get_gene_id_normalization_dictionaries().\n",
      "\n",
      "\n",
      "The function \"prepare_data_edges()\" is running...\n",
      "\n",
      "The tftargets edges are saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/regulation/tftargets/tftargets_edges.csv\n",
      "\n",
      "\n",
      "The MSigDB edges are saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/regulation/msigdb/msigdb_edges.csv\n",
      "\n",
      "\n",
      "Finished prepare_data_edges().\n",
      "\n",
      "\n",
      "The function \"prepare_regulation_edges()\" is running...\n",
      "\n",
      "Finished prepare_regulation_edges().\n",
      "\n",
      "\n",
      "The function \"build_edges()\" is running...\n",
      "\n",
      "* This is the size of the edges file data structure: (197267, 9)\n",
      "* These are the edges attributes: Index(['subject_id', 'object_id', 'property_id', 'property_label',\n",
      "       'property_description', 'property_uri', 'reference_uri',\n",
      "       'reference_supporting_text', 'reference_date'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "  subject_id  object_id property_id  property_label property_description  \\\n",
      "0  HGNC:8615  HGNC:8803  RO:0002434  interacts with                   NA   \n",
      "\n",
      "                                property_uri  \\\n",
      "0  http://purl.obolibrary.org/obo/RO_0002434   \n",
      "\n",
      "                                  reference_uri  \\\n",
      "0  https://www.ncbi.nlm.nih.gov/pubmed/17202159   \n",
      "\n",
      "                           reference_supporting_text reference_date  \n",
      "0  This edge comes from the TRED dataset in \"tfta...     2007-01-01  \n",
      "\n",
      "The regulation network edges are built and saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/graph/regulation_edges_v2021-04-12.csv\n",
      "\n",
      "\n",
      "Finished build_edges().\n",
      "\n",
      "\n",
      "The function \"build_nodes()\" is running...\n",
      "\n",
      "* Total number of nodes: 16968\n",
      "\n",
      "* Trap genes without gene symbol, i.e. genes with discontinued entrez ID...\n",
      "* Number of concepts without gene symbol: 144\n",
      "* Check that all genes without gene symbol are identified by entrez ID...\n",
      "* Number of concepts without gene symbol by namespace:  NCBIGene    144\n",
      "Name: id, dtype: int64\n",
      "\n",
      "* Querying BioThings to map retired Entrez to gene symbols...\n",
      "querying 1-144...done.\n",
      "Finished.\n",
      "90 input query terms found no hit:\n",
      "\t['79907', '79911', '93333', '121301', '143902', '146856', '151720', '197379', '219392', '221943', '2\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "* Querying BioThings to retrieve node attributes...\n",
      "querying 1-1000...done.\n",
      "querying 1001-2000...done.\n",
      "querying 2001-3000...done.\n",
      "querying 3001-4000...done.\n",
      "querying 4001-5000...done.\n",
      "querying 5001-6000...done.\n",
      "querying 6001-7000...done.\n",
      "querying 7001-8000...done.\n",
      "querying 8001-9000...done.\n",
      "querying 9001-10000...done.\n",
      "querying 10001-11000...done.\n",
      "querying 11001-12000...done.\n",
      "querying 12001-13000...done.\n",
      "querying 13001-14000...done.\n",
      "querying 14001-15000...done.\n",
      "querying 15001-16000...done.\n",
      "querying 16001-16844...done.\n",
      "Finished.\n",
      "48 input query terms found no hit:\n",
      "\t['MMEF2', 'CDPCR3', 'AHRARNT', 'CREBP1', 'NFMUE1', 'CREL', 'AP1FJ', 'HMEF2', 'TAL1BETAITF2', 'NKX62'\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "* This is the size of the nodes file data structure: (16968, 6)\n",
      "* These are the nodes attributes: Index(['id', 'semantic_groups', 'preflabel', 'name', 'synonyms',\n",
      "       'description'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "          id semantic_groups preflabel          name    synonyms  \\\n",
      "0  HGNC:8615            GENE      PAX1  paired box 1  HUP48|OFC2   \n",
      "\n",
      "                                         description  \n",
      "0  This gene is a member of the paired box (PAX) ...  \n",
      "\n",
      "The regulation network nodes are built and saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/graph/regulation_nodes_v2021-04-12.csv\n",
      "\n",
      "\n",
      "Finished build_nodes().\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# obtain variables:\n",
    "#curation_edges = pd.read_csv(\"curation/data/HD/Empty_edges.csv\")\n",
    "#curation_nodes = pd.read_csv(\"curation/data/HD/Empty_nodes.csv\")\n",
    "\n",
    "monarch_network = monarch.read_connections(\"monarch_connections_v2020-12-12.csv\")\n",
    "monarch_edges = monarch.build_edges(monarch_network)\n",
    "monarch_nodes = monarch.build_nodes(monarch_network)\n",
    "\n",
    "#monarch_network_graph = monarch.read_connections(\"monarch_connections_graph_v2020-12-02.csv\")\n",
    "#monarch_graph_edges = monarch.build_edges(monarch_network_graph)\n",
    "#monarch_graph_nodes = monarch.build_nodes(monarch_network_graph)\n",
    "\n",
    "#csv_path = './transcriptomics/GSE64810_mlhd_DESeq2_diffexp_DESeq2_outlier_trimmed_adjust.txt'\n",
    "#data = transcriptomics.read_data(csv_path, \"\\t\")\n",
    "#clean_data = transcriptomics.clean_data(data)\n",
    "#data_edges = transcriptomics.prepare_data_edges(clean_data)\n",
    "#rna_network = transcriptomics.prepare_rna_edges(data_edges)\n",
    "\n",
    "# build network with graph schema\n",
    "#rna_edges = transcriptomics.build_edges(rna_network)\n",
    "#rna_nodes, rna_edges = transcriptomics.build_nodes(rna_network)\n",
    "csv_path = './transcriptomics/GSE64810_mlhd_DESeq2_diffexp_DESeq2_outlier_trimmed_adjust.txt'\n",
    "data = transcriptomics.read_data(csv_path, \"\\t\")\n",
    "clean_data = transcriptomics.clean_data(data)\n",
    "data_edges = transcriptomics.prepare_data_edges(clean_data)\n",
    "rna_network = transcriptomics.prepare_rna_edges(data_edges)\n",
    "\n",
    "# build network with graph schema\n",
    "rna_edges = transcriptomics.build_edges(rna_network)\n",
    "rna_nodes, nodes_dict = transcriptomics.build_nodes(rna_network)\n",
    "\n",
    "\n",
    "gmt_path = './regulation/msigdb/data/c3.tft.v6.1.entrez.gmt'\n",
    "regulation.prepare_msigdb_data(gmt_path)\n",
    "\n",
    "# prepare individual networks\n",
    "data = regulation.load_tf_gene_edges()\n",
    "dicts = regulation.get_gene_id_normalization_dictionaries(data)\n",
    "data_edges = regulation.prepare_data_edges(data, dicts)\n",
    "\n",
    "# prepare regulation network\n",
    "reg_network = regulation.prepare_regulation_edges(data_edges)\n",
    "\n",
    "# build network with graph schema\n",
    "reg_edges = regulation.build_edges(reg_network)\n",
    "reg_nodes = regulation.build_nodes(reg_network)\n",
    "\n",
    "#graph_nodes_list, reg_graph_edges = graph.graph_nodes(\n",
    "#    curation=curation_edges,\n",
    "#    monarch=monarch_edges,\n",
    "#    transcriptomics=rna_edges,\n",
    "#    regulation=reg_edges\n",
    "#)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "curation_edges = pd.DataFrame(curation_edges)\n",
    "monarch_edges = pd.DataFrame(monarch_edges)\n",
    "rna_edges = pd.DataFrame(rna_edges)\n",
    "curation_edges = curation_edges.iloc[0]\n",
    "curation_edges = pd.DataFrame(curation_edges).T\n",
    "reg_edges = pd.DataFrame(reg_edges)\n",
    "curation_nodes = curation_nodes.iloc[0]\n",
    "curation_nodes = pd.DataFrame(curation_nodes).T\n",
    "monarch_nodes = pd.DataFrame(monarch_nodes)\n",
    "rna_nodes = pd.DataFrame(rna_nodes)\n",
    "reg_nodes = pd.DataFrame(reg_nodes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'id': 'HGNC:8615',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PAX1',\n",
       "  'name': 'paired box 1',\n",
       "  'synonyms': 'HUP48|OFC2',\n",
       "  'description': 'This gene is a member of the paired box (PAX) family of transcription factors. Members of the PAX family typically contain a paired box domain and a paired-type homeodomain. These genes play critical roles during fetal development. This gene plays a role in pattern formation during embryogenesis and may be essential for development of the vertebral column. This gene is silenced by methylation in ovarian and cervical cancers and may be a tumor suppressor gene. Mutations in this gene are also associated with vertebral malformations. [provided by RefSeq, Mar 2012].'},\n",
       " {'id': 'HGNC:8803',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PDGFRA',\n",
       "  'name': 'platelet derived growth factor receptor alpha',\n",
       "  'synonyms': 'CD140A|PDGFR-2|PDGFR2',\n",
       "  'description': 'This gene encodes a cell surface tyrosine kinase receptor for members of the platelet-derived growth factor family. These growth factors are mitogens for cells of mesenchymal origin. The identity of the growth factor bound to a receptor monomer determines whether the functional receptor is a homodimer or a heterodimer, composed of both platelet-derived growth factor receptor alpha and beta polypeptides. Studies suggest that this gene plays a role in organ development, wound healing, and tumor progression. Mutations in this gene have been associated with idiopathic hypereosinophilic syndrome, somatic and familial gastrointestinal stromal tumors, and a variety of other cancers. [provided by RefSeq, Mar 2012].'},\n",
       " {'id': 'HGNC:12687',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'VHL',\n",
       "  'name': 'von Hippel-Lindau tumor suppressor',\n",
       "  'synonyms': 'HRCA1|RCA1|VHL1|pVHL',\n",
       "  'description': 'Von Hippel-Lindau syndrome (VHL) is a dominantly inherited familial cancer syndrome predisposing to a variety of malignant and benign tumors. A germline mutation of this gene is the basis of familial inheritance of VHL syndrome. The protein encoded by this gene is a component of the protein complex that includes elongin B, elongin C, and cullin-2, and possesses ubiquitin ligase E3 activity. This protein is involved in the ubiquitination and degradation of hypoxia-inducible-factor (HIF), which is a transcription factor that plays a central role in the regulation of gene expression by oxygen. RNA polymerase II subunit POLR2G/RPB7 is also reported to be a target of this protein. Alternatively spliced transcript variants encoding distinct isoforms have been observed. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:10970',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SLC22A6',\n",
       "  'name': 'solute carrier family 22 member 6',\n",
       "  'synonyms': 'HOAT1|OAT1|PAHT|ROAT1',\n",
       "  'description': 'The protein encoded by this gene is involved in the sodium-dependent transport and excretion of organic anions, some of which are potentially toxic. The encoded protein is an integral membrane protein and may be localized to the basolateral membrane. Four transcript variants encoding four different isoforms have been found for this gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:10972',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SLC22A8',\n",
       "  'name': 'solute carrier family 22 member 8',\n",
       "  'synonyms': 'OAT3',\n",
       "  'description': 'This gene encodes a protein involved in the sodium-independent transport and excretion of organic anions, some of which are potentially toxic. The encoded protein is an integral membrane protein and appears to be localized to the basolateral membrane of the kidney. Multiple alternatively spliced transcript variants that encode different protein isoforms have been described for this gene. [provided by RefSeq, May 2010].'},\n",
       " {'id': 'HGNC:25567',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ATAD3A',\n",
       "  'name': 'ATPase family AAA domain containing 3A',\n",
       "  'synonyms': 'HAYOS|PHRINL',\n",
       "  'description': 'This gene encodes a ubiquitously expressed mitochondrial membrane protein that contributes to mitochondrial dynamics, nucleoid organization, protein translation, cell growth, and cholesterol metabolism. This gene is a member of the ATPase family AAA-domain containing 3 gene family which, in humans, includes two other paralogs. Naturally occurring mutations in this gene are associated with distinct neurological syndromes including Harel-Yoon syndrome. High-level expression of this gene is associated with poor survival in breast cancer patients. A homozygous knockout of the orthologous gene in mice results in embryonic lethality at day 7.5 due to growth retardation and defective development of the trophoblast lineage. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Feb 2017].'},\n",
       " {'id': 'HGNC:14357',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MMP21',\n",
       "  'name': 'matrix metallopeptidase 21',\n",
       "  'synonyms': 'HTX7|MMP-21',\n",
       "  'description': 'This gene encodes a member of the matrix metalloproteinase family. Proteins in this family are involved in the breakdown of extracellular matrix for both normal physiological processes, such as embryonic development, reproduction, and tissue remodeling, and disease processes, such as asthma and tumor metastasis. The encoded protein may play an important role in embryogenesis, particularly in neuronal cells, as well as in lymphocyte development and survival. [provided by RefSeq, May 2013].'},\n",
       " {'id': 'HGNC:3240',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'EGR3',\n",
       "  'name': 'early growth response 3',\n",
       "  'synonyms': 'EGR-3|PILOT',\n",
       "  'description': 'This gene encodes a transcriptional regulator that belongs to the EGR family of C2H2-type zinc-finger proteins. It is an immediate-early growth response gene which is induced by mitogenic stimulation. The protein encoded by this gene participates in the transcriptional regulation of genes in controling biological rhythm. It may also play a role in a wide variety of processes including muscle development, lymphocyte development, endothelial cell growth and migration, and neuronal development. Alternative splicing results in multiple transcript variants encoding distinct isoforms.[provided by RefSeq, Dec 2010].'},\n",
       " {'id': 'HGNC:1694',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CD69',\n",
       "  'name': 'CD69 molecule',\n",
       "  'synonyms': 'AIM|BL-AC/P26|CLEC2C|EA1|GP32/28|MLR-3',\n",
       "  'description': 'This gene encodes a member of the calcium dependent lectin superfamily of type II transmembrane receptors. Expression of the encoded protein is induced upon activation of T lymphocytes, and may play a role in proliferation. Furthermore, the protein may act to transmit signals in natural killer cells and platelets. [provided by RefSeq, Aug 2011].'},\n",
       " {'id': 'HGNC:11246',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SPINT1',\n",
       "  'name': 'serine peptidase inhibitor, Kunitz type 1',\n",
       "  'synonyms': 'HAI|HAI1|MANSC2',\n",
       "  'description': 'The protein encoded by this gene is a member of the Kunitz family of serine protease inhibitors. The protein is a potent inhibitor specific for HGF activator and is thought to be involved in the regulation of the proteolytic activation of HGF in injured tissues. Alternative splicing results in multiple variants encoding different isoforms. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:12003',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TP73',\n",
       "  'name': 'tumor protein p73',\n",
       "  'synonyms': 'P73',\n",
       "  'description': 'This gene encodes a member of the p53 family of transcription factors involved in cellular responses to stress and development. It maps to a region on chromosome 1p36 that is frequently deleted in neuroblastoma and other tumors, and thought to contain multiple tumor suppressor genes. The demonstration that this gene is monoallelically expressed (likely from the maternal allele), supports the notion that it is a candidate gene for neuroblastoma. Many transcript variants resulting from alternative splicing and/or use of alternate promoters have been found for this gene, but the biological validity and the full-length nature of some variants have not been determined. [provided by RefSeq, Feb 2011].'},\n",
       " {'id': 'HGNC:3239',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'EGR2',\n",
       "  'name': 'early growth response 2',\n",
       "  'synonyms': 'AT591|CHN1|CMT1D|CMT4E|KROX20',\n",
       "  'description': 'The protein encoded by this gene is a transcription factor with three tandem C2H2-type zinc fingers. Defects in this gene are associated with Charcot-Marie-Tooth disease type 1D (CMT1D), Charcot-Marie-Tooth disease type 4E (CMT4E), and with Dejerine-Sottas syndrome (DSS). Multiple transcript variants encoding two different isoforms have been found for this gene. [provided by RefSeq, Oct 2008].'},\n",
       " {'id': 'HGNC:11920',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'FAS',\n",
       "  'name': 'Fas cell surface death receptor',\n",
       "  'synonyms': 'ALPS1A|APO-1|APT1|CD95|FAS1|FASTM|TNFRSF6',\n",
       "  'description': 'The protein encoded by this gene is a member of the TNF-receptor superfamily. This receptor contains a death domain. It has been shown to play a central role in the physiological regulation of programmed cell death, and has been implicated in the pathogenesis of various malignancies and diseases of the immune system. The interaction of this receptor with its ligand allows the formation of a death-inducing signaling complex that includes Fas-associated death domain protein (FADD), caspase 8, and caspase 10. The autoproteolytic processing of the caspases in the complex triggers a downstream caspase cascade, and leads to apoptosis. This receptor has been also shown to activate NF-kappaB, MAPK3/ERK1, and MAPK8/JNK, and is found to be involved in transducing the proliferating signals in normal diploid fibroblast and T cells. Several alternatively spliced transcript variants have been described, some of which are candidates for nonsense-mediated mRNA decay (NMD). The isoforms lacking the transmembrane domain may negatively regulate the apoptosis mediated by the full length isoform. [provided by RefSeq, Mar 2011].'},\n",
       " {'id': 'HGNC:1116',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'BSG',\n",
       "  'name': 'basigin (Ok blood group)',\n",
       "  'synonyms': '5F7|CD147|EMMPRIN|EMPRIN|HAb18G|OK|SLC7A11|TCSF',\n",
       "  'description': 'The protein encoded by this gene, basigin, is a plasma membrane protein that is important in spermatogenesis, embryo implantation, neural network formation, and tumor progression. Basigin is also a member of the immunoglobulin superfamily, ubiquitously expressed in various tissues. Multiple transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, May 2020].'},\n",
       " {'id': 'HGNC:3594',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'FASN',\n",
       "  'name': 'fatty acid synthase',\n",
       "  'synonyms': 'FAS|OA-519|SDR27X1',\n",
       "  'description': 'The enzyme encoded by this gene is a multifunctional protein. Its main function is to catalyze the synthesis of palmitate from acetyl-CoA and malonyl-CoA, in the presence of NADPH, into long-chain saturated fatty acids. In some cancer cell lines, this protein has been found to be fused with estrogen receptor-alpha (ER-alpha), in which the N-terminus of FAS is fused in-frame with the C-terminus of ER-alpha. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:5103',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HOXA2',\n",
       "  'name': 'homeobox A2',\n",
       "  'synonyms': 'HOX1K|MCOHI',\n",
       "  'description': 'In vertebrates, the genes encoding the class of transcription factors called homeobox genes are found in clusters named A, B, C, and D on four separate chromosomes. Expression of these proteins is spatially and temporally regulated during embryonic development. This gene is part of the A cluster on chromosome 7 and encodes a DNA-binding transcription factor which may regulate gene expression, morphogenesis, and differentiation. The encoded protein may be involved in the placement of hindbrain segments in the proper location along the anterior-posterior axis during development. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:5113',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HOXB2',\n",
       "  'name': 'homeobox B2',\n",
       "  'synonyms': 'HOX2|HOX2H|Hox-2.8|K8',\n",
       "  'description': 'This gene is a member of the Antp homeobox family and encodes a nuclear protein with a homeobox DNA-binding domain. It is included in a cluster of homeobox B genes located on chromosome 17. The encoded protein functions as a sequence-specific transcription factor that is involved in development. Increased expression of this gene is associated with pancreatic cancer. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:40',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ABCB1',\n",
       "  'name': 'ATP binding cassette subfamily B member 1',\n",
       "  'synonyms': 'ABC20|CD243|CLCS|GP170|MDR1|P-GP|PGY1|p-170',\n",
       "  'description': 'The membrane-associated protein encoded by this gene is a member of the superfamily of ATP-binding cassette (ABC) transporters. ABC proteins transport various molecules across extra- and intra-cellular membranes. ABC genes are divided into seven distinct subfamilies (ABC1, MDR/TAP, MRP, ALD, OABP, GCN20, White). This protein is a member of the MDR/TAP subfamily. Members of the MDR/TAP subfamily are involved in multidrug resistance. The protein encoded by this gene is an ATP-dependent drug efflux pump for xenobiotic compounds with broad substrate specificity. It is responsible for decreased drug accumulation in multidrug-resistant cells and often mediates the development of resistance to anticancer drugs. This protein also functions as a transporter in the blood-brain barrier. Mutations in this gene are associated with colchicine resistance and Inflammatory bowel disease 13. Alternative splicing and the use of alternative promoters results in multiple transcript variants. [provided by RefSeq, Feb 2017].'},\n",
       " {'id': 'HGNC:2743',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DDX20',\n",
       "  'name': 'DEAD-box helicase 20',\n",
       "  'synonyms': 'DP103|GEMIN3',\n",
       "  'description': 'DEAD box proteins, characterized by the conserved motif Asp-Glu-Ala-Asp (DEAD), are putative RNA helicases. They are implicated in a number of cellular processes involving alteration of RNA secondary structure such as translation initiation, nuclear and mitochondrial splicing, and ribosome and spliceosome assembly. Based on their distribution patterns, some members of this family are believed to be involved in embryogenesis, spermatogenesis, and cellular growth and division. This gene encodes a DEAD box protein, which has an ATPase activity and is a component of the survival of motor neurons (SMN) complex. This protein interacts directly with SMN, the spinal muscular atrophy gene product, and may play a catalytic role in the function of the SMN complex on RNPs. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:3113',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'E2F1',\n",
       "  'name': 'E2F transcription factor 1',\n",
       "  'synonyms': 'E2F-1|RBAP1|RBBP3|RBP3',\n",
       "  'description': 'The protein encoded by this gene is a member of the E2F family of transcription factors. The E2F family plays a crucial role in the control of cell cycle and action of tumor suppressor proteins and is also a target of the transforming proteins of small DNA tumor viruses. The E2F proteins contain several evolutionally conserved domains found in most members of the family. These domains include a DNA binding domain, a dimerization domain which determines interaction with the differentiation regulated transcription factor proteins (DP), a transactivation domain enriched in acidic amino acids, and a tumor suppressor protein association domain which is embedded within the transactivation domain.  This protein and another 2 members, E2F2 and E2F3, have an additional cyclin binding domain. This protein binds preferentially to retinoblastoma protein pRB in a cell-cycle dependent manner. It can mediate both cell proliferation and p53-dependent/independent apoptosis. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:20',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'AARS1',\n",
       "  'name': 'alanyl-tRNA synthetase 1',\n",
       "  'synonyms': 'AARS|CMT2N|DEE29|EIEE29',\n",
       "  'description': 'The human alanyl-tRNA synthetase (AARS) belongs to a family of tRNA synthases, of the class II enzymes.  Class II tRNA synthases evolved early in evolution and are highly conserved. This is reflected by the fact that 498 of the 968-residue polypeptide human AARS shares 41% identity witht the E.coli protein.  tRNA synthases are the enzymes that interpret the RNA code and attach specific aminoacids to the tRNAs that contain the cognate trinucleotide anticodons.  They consist of a catalytic domain which interacts with the amino acid acceptor-T psi C helix of the tRNA, and a second domain which interacts with the rest of the tRNA structure. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:29',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ABCA1',\n",
       "  'name': 'ATP binding cassette subfamily A member 1',\n",
       "  'synonyms': 'ABC-1|ABC1|CERP|HDLCQTL13|HDLDT1|HPALP1|TGD',\n",
       "  'description': 'The membrane-associated protein encoded by this gene is a member of the superfamily of ATP-binding cassette (ABC) transporters. ABC proteins transport various molecules across extra- and intracellular membranes. ABC genes are divided into seven distinct subfamilies (ABC1, MDR/TAP, MRP, ALD, OABP, GCN20, White). This protein is a member of the ABC1 subfamily. Members of the ABC1 subfamily comprise the only major ABC subfamily found exclusively in multicellular eukaryotes. With cholesterol as its substrate, this protein functions as a cholesteral efflux pump in the cellular lipid removal pathway. Mutations in both alleles of this gene cause Tangier disease and familial high-density lipoprotein (HDL) deficiency. [provided by RefSeq, Sep 2019].'},\n",
       " {'id': 'HGNC:186',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ADA',\n",
       "  'name': 'adenosine deaminase',\n",
       "  'synonyms': 'ADA1',\n",
       "  'description': 'This gene encodes an enzyme that catalyzes the hydrolysis of adenosine to inosine in the purine catabolic pathway. Various mutations have been described for this gene and have been linked to human diseases related to impaired immune function such as severe combined immunodeficiency disease (SCID) which is the result of a deficiency in the ADA enzyme. In ADA-deficient individuals there is a marked depletion of T, B, and NK lymphocytes, and consequently, a lack of both humoral and cellular immunity. Conversely, elevated levels of this enzyme are associated with congenital hemolytic anemia. [provided by RefSeq, Sep 2019].'},\n",
       " {'id': 'HGNC:215',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ADAM8',\n",
       "  'name': 'ADAM metallopeptidase domain 8',\n",
       "  'synonyms': 'CD156|CD156a|MS2',\n",
       "  'description': 'This gene encodes a member of the ADAM (a disintegrin and metalloprotease domain) family. Members of this family are membrane-anchored proteins structurally related to snake venom disintegrins, and have been implicated in a variety of biological processes involving cell-cell and cell-matrix interactions, including fertilization, muscle development, and neurogenesis. The protein encoded by this gene may be involved in cell adhesion during neurodegeneration, and it is thought to be a target for allergic respiratory diseases, including asthma. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Aug 2009].'},\n",
       " {'id': 'HGNC:245',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ADD3',\n",
       "  'name': 'adducin 3',\n",
       "  'synonyms': 'ADDL|CPSQ3',\n",
       "  'description': 'Adducins are heteromeric proteins composed of different subunits referred to as adducin alpha, beta and gamma. The three subunits are encoded by distinct genes and belong to a family of membrane skeletal proteins involved in the assembly of spectrin-actin network in erythrocytes and at sites of cell-cell contact in epithelial tissues. While adducins alpha and gamma are ubiquitously expressed, the expression of adducin beta is restricted to brain and hematopoietic tissues. Adducin, originally purified from human erythrocytes, was found to be a heterodimer of adducins alpha and beta. Polymorphisms resulting in amino acid substitutions in these two subunits have been associated with the regulation of blood pressure in an animal model of hypertension. Heterodimers consisting of alpha and gamma subunits have also been described. Structurally, each subunit is comprised of two distinct domains. The amino-terminal region is protease resistant and globular in shape, while the carboxy-terminal region is protease sensitive. The latter contains multiple phosphorylation sites for protein kinase C, the binding site for calmodulin, and is required for association with spectrin and actin. Alternatively spliced adducin gamma transcripts encoding different isoforms have been described. The functions of the different isoforms are not known. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:253',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ADH5',\n",
       "  'name': 'alcohol dehydrogenase 5 (class III), chi polypeptide',\n",
       "  'synonyms': 'ADH-3|ADHX|AMEDS|BMFS7|FALDH|FDH|GSH-FDH|GSNOR|HEL-S-60p',\n",
       "  'description': 'This gene encodes a member of the alcohol dehydrogenase family. Members of this family metabolize a wide variety of substrates, including ethanol, retinol, other aliphatic alcohols, hydroxysteroids, and lipid peroxidation products. The encoded protein forms a homodimer. It has virtually no activity for ethanol oxidation, but exhibits high activity for oxidation of long-chain primary alcohols and for oxidation of S-hydroxymethyl-glutathione, a spontaneous adduct between formaldehyde and glutathione. This enzyme is an important component of cellular metabolism for the elimination of formaldehyde, a potent irritant and sensitizing agent that causes lacrymation, rhinitis, pharyngitis, and contact dermatitis. The human genome contains several non-transcribed pseudogenes related to this gene. [provided by RefSeq, Oct 2008].'},\n",
       " {'id': 'HGNC:554',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'AP1B1',\n",
       "  'name': 'adaptor related protein complex 1 subunit beta 1',\n",
       "  'synonyms': 'ADTB1|AP105A|BAM22|CLAPB2|KIDAR',\n",
       "  'description': 'Adaptor protein complex 1 is found at the cytoplasmic face of coated vesicles located at the Golgi complex, where it mediates both the recruitment of clathrin to the membrane and the recognition of sorting signals within the cytosolic tails of transmembrane receptors. This complex is a heterotetramer composed of two large, one medium, and one small adaptin subunit. The protein encoded by this gene serves as one of the large subunits of this complex and is a member of the adaptin protein family. This gene is a candidate meningioma gene. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Oct 2009].'},\n",
       " {'id': 'HGNC:320',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'AGER',\n",
       "  'name': 'advanced glycosylation end-product specific receptor',\n",
       "  'synonyms': 'RAGE|SCARJ1|sRAGE',\n",
       "  'description': \"The advanced glycosylation end product (AGE) receptor encoded by this gene is a member of the immunoglobulin superfamily of cell surface receptors. It is a multiligand receptor, and besides AGE, interacts with other molecules implicated in homeostasis, development, and inflammation, and certain diseases, such as diabetes and Alzheimer's disease. Many alternatively spliced transcript variants encoding different isoforms, as well as non-protein-coding variants, have been described for this gene (PMID:18089847). [provided by RefSeq, May 2011].\"},\n",
       " {'id': 'HGNC:363',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'AK4',\n",
       "  'name': 'adenylate kinase 4',\n",
       "  'synonyms': 'AK 4|AK3|AK3L1|AK3L2',\n",
       "  'description': 'This gene encodes a member of the adenylate kinase family of enzymes. The encoded protein is localized to the mitochondrial matrix. Adenylate kinases regulate the adenine and guanine nucleotide compositions within a cell by catalyzing the reversible transfer of phosphate group among these nucleotides. Five isozymes of adenylate kinase have been identified in vertebrates. Expression of these isozymes is tissue-specific and developmentally regulated. A pseudogene for this gene has been located on chromosome 17. Three transcript variants encoding the same protein have been identified for this gene. Sequence alignment suggests that the gene defined by NM_013410, NM_203464, and NM_001005353 is located on chromosome 1. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:402',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ALDH1A1',\n",
       "  'name': 'aldehyde dehydrogenase 1 family member A1',\n",
       "  'synonyms': 'ALDC|ALDH-E1|ALDH1|ALDH11|HEL-9|HEL-S-53e|HEL12|PUMB1|RALDH1',\n",
       "  'description': 'The protein encoded by this gene belongs to the aldehyde dehydrogenase family. Aldehyde dehydrogenase is the next enzyme after alcohol dehydrogenase in the major pathway of alcohol metabolism. There are two major aldehyde dehydrogenase isozymes in the liver, cytosolic and mitochondrial, which are encoded by distinct genes, and can be distinguished by their electrophoretic mobility, kinetic properties, and subcellular localization. This gene encodes the cytosolic isozyme. Studies in mice show that through its role in retinol metabolism, this gene may also be involved in the regulation of the metabolic responses to high-fat diet. [provided by RefSeq, Mar 2011].'},\n",
       " {'id': 'HGNC:436',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ALOX5AP',\n",
       "  'name': 'arachidonate 5-lipoxygenase activating protein',\n",
       "  'synonyms': 'FLAP',\n",
       "  'description': 'This gene encodes a protein which, with 5-lipoxygenase, is required for leukotriene synthesis. Leukotrienes are arachidonic acid metabolites which have been implicated in various types of inflammatory responses, including asthma, arthritis and psoriasis. This protein localizes to the plasma membrane. Inhibitors of its function impede translocation of 5-lipoxygenase from the cytoplasm to the cell membrane and inhibit 5-lipoxygenase activation. Alternatively spliced transcript variants encoding different isoforms have been identified for this gene. [provided by RefSeq, Feb 2011].'},\n",
       " {'id': 'HGNC:470',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'AMPD3',\n",
       "  'name': 'adenosine monophosphate deaminase 3',\n",
       "  'synonyms': nan,\n",
       "  'description': 'This gene encodes a member of the AMP deaminase gene family. The encoded protein is a highly regulated enzyme that catalyzes the hydrolytic deamination of adenosine monophosphate to inosine monophosphate, a branch point in the adenylate catabolic pathway. This gene encodes the erythrocyte (E) isoforms, whereas other family members encode isoforms that predominate in muscle (M) and liver (L) cells. Mutations in this gene lead to the clinically asymptomatic, autosomal recessive condition erythrocyte AMP deaminase deficiency. Alternatively spliced transcript variants encoding different isoforms of this gene have been described. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:536',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ANXA13',\n",
       "  'name': 'annexin A13',\n",
       "  'synonyms': 'ANX13|ISA',\n",
       "  'description': 'This gene encodes a member of the annexin family. Members of this calcium-dependent phospholipid-binding protein family play a role in the regulation of cellular growth and in signal transduction pathways. The specific function of this gene has not yet been determined; however, it is associated with the plasma membrane of undifferentiated, proliferating endothelial cells and differentiated villus enterocytes. Alternatively spliced transcript variants encoding different isoforms have been identified. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:576',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'APAF1',\n",
       "  'name': 'apoptotic peptidase activating factor 1',\n",
       "  'synonyms': 'APAF-1|CED4',\n",
       "  'description': 'This gene encodes a cytoplasmic protein that initiates apoptosis. This protein contains several copies of the WD-40 domain, a caspase recruitment domain (CARD), and an ATPase domain (NB-ARC). Upon binding cytochrome c and dATP, this protein forms an oligomeric apoptosome. The apoptosome binds and cleaves caspase 9 preproprotein, releasing its mature, activated form. Activated caspase 9 stimulates the subsequent caspase cascade that commits the cell to apoptosis. Alternative splicing results in several transcript variants encoding different isoforms. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:597',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'APLP1',\n",
       "  'name': 'amyloid beta precursor like protein 1',\n",
       "  'synonyms': 'APLP',\n",
       "  'description': 'This gene encodes a member of the highly conserved amyloid precursor protein gene family. The encoded protein is a membrane-associated glycoprotein that is cleaved by secretases in a manner similar to amyloid beta A4 precursor protein cleavage. This cleavage liberates an intracellular cytoplasmic fragment that may act as a transcriptional activator. The encoded protein may also play a role in synaptic maturation during cortical development. Alternatively spliced transcript variants encoding different isoforms have been described. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:620',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'APP',\n",
       "  'name': 'amyloid beta precursor protein',\n",
       "  'synonyms': 'AAA|ABETA|ABPP|AD1|APPI|CTFgamma|CVAP|PN-II|PN2|alpha-sAPP|preA4',\n",
       "  'description': 'This gene encodes a cell surface receptor and transmembrane precursor protein that is cleaved by secretases to form a number of peptides. Some of these peptides are secreted and can bind to the acetyltransferase complex APBB1/TIP60 to promote transcriptional activation, while others form the protein basis of the amyloid plaques found in the brains of patients with Alzheimer disease. In addition, two of the peptides are antimicrobial peptides, having been shown to have bacteriocidal and antifungal activities. Mutations in this gene have been implicated in autosomal dominant Alzheimer disease and cerebroarterial amyloidosis (cerebral amyloid angiopathy). Multiple transcript variants encoding several different isoforms have been found for this gene. [provided by RefSeq, Aug 2014].'},\n",
       " {'id': 'HGNC:6364',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'KLK3',\n",
       "  'name': 'kallikrein related peptidase 3',\n",
       "  'synonyms': 'APS|KLK2A1|PSA|hK3',\n",
       "  'description': 'Kallikreins are a subgroup of serine proteases having diverse physiological functions. Growing evidence suggests that many kallikreins are implicated in carcinogenesis and some have potential as novel cancer and other disease biomarkers. The gene is one of the fifteen kallikrein subfamily members located in a cluster on chromosome 19. It encodes a single-chain glycoprotein, a protease which is synthesized in the epithelial cells of the prostate gland, and is present in seminal plasma. It is thought to function normally in the liquefaction of seminal coagulum, presumably by hydrolysis of the high molecular mass seminal vesicle protein. The serum level of this protein, called PSA in the clinical setting, is useful in the diagnosis and monitoring of prostatic carcinoma. Alternate splicing of this gene generates several transcript variants encoding different isoforms. [provided by RefSeq, Dec 2019].'},\n",
       " {'id': 'HGNC:700',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ARNT',\n",
       "  'name': 'aryl hydrocarbon receptor nuclear translocator',\n",
       "  'synonyms': 'HIF-1-beta|HIF-1beta|HIF1-beta|HIF1B|HIF1BETA|TANGO|bHLHe2',\n",
       "  'description': 'This gene encodes a protein containing a basic helix-loop-helix domain and two characteristic PAS domains along with a PAC domain. The encoded protein binds to ligand-bound aryl hydrocarbon receptor and aids in the movement of this complex to the nucleus, where it promotes the expression of genes involved in xenobiotic metabolism. This protein is also a co-factor for transcriptional regulation by hypoxia-inducible factor 1. Chromosomal translocation of this locus with the ETV6 (ets variant 6) gene on chromosome 12 have been described in leukemias. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Oct 2013].'},\n",
       " {'id': 'HGNC:840',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ATP5PB',\n",
       "  'name': 'ATP synthase peripheral stalk-membrane subunit b',\n",
       "  'synonyms': 'ATP5F1|PIG47',\n",
       "  'description': 'This gene encodes a subunit of mitochondrial ATP synthase. Mitochondrial ATP synthase catalyzes ATP synthesis, utilizing an electrochemical gradient of protons across the inner membrane during oxidative phosphorylation. ATP synthase is composed of two linked multi-subunit complexes: the soluble catalytic core, F1, and the membrane-spanning component, Fo, comprising the proton channel. The catalytic portion of mitochondrial ATP synthase consists of 5 different subunits (alpha, beta, gamma, delta, and epsilon) assembled with a stoichiometry of 3 alpha, 3 beta, and a single representative of the other 3. The proton channel seems to have nine subunits (a, b, c, d, e, f, g, F6 and 8). This gene encodes the b subunit of the proton channel. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:857',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ATP6V1E1',\n",
       "  'name': 'ATPase H+ transporting V1 subunit E1',\n",
       "  'synonyms': 'ARCL2C|ATP6E|ATP6E2|ATP6V1E|P31|Vma4',\n",
       "  'description': 'This gene encodes a component of vacuolar ATPase (V-ATPase), a multisubunit enzyme that mediates acidification of eukaryotic intracellular organelles. V-ATPase dependent organelle acidification is necessary for such intracellular processes as protein sorting, zymogen activation, receptor-mediated endocytosis, and synaptic vesicle proton gradient generation. V-ATPase is composed of a cytosolic V1 domain and a transmembrane V0 domain. The V1 domain consists of three A, three B, and two G subunits, as well as a C, D, E, F, and H subunit. The V1 domain contains the ATP catalytic site. This gene encodes alternate transcriptional splice variants, encoding different V1 domain E subunit isoforms. Pseudogenes for this gene have been found in the genome. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:861',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ATP6V0B',\n",
       "  'name': 'ATPase H+ transporting V0 subunit b',\n",
       "  'synonyms': 'ATP6F|HATPL|VMA16',\n",
       "  'description': 'This gene encodes a portion of the V0 domain of vacuolar ATPase (V-ATPase), a multisubunit enzyme that mediates acidification of eukaryotic intracellular organelles. Activity of this enzyme is necessary for such varied processes as protein sorting, zymogen activation, receptor-mediated endocytosis, and synaptic vesicle proton gradient generation. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Jun 2014].'},\n",
       " {'id': 'HGNC:1582',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CCND1',\n",
       "  'name': 'cyclin D1',\n",
       "  'synonyms': 'BCL1|D11S287E|PRAD1|U21B31',\n",
       "  'description': 'The protein encoded by this gene belongs to the highly conserved cyclin family, whose members are characterized by a dramatic periodicity in protein abundance throughout the cell cycle. Cyclins function as regulators of CDK kinases. Different cyclins exhibit distinct expression and degradation patterns which contribute to the temporal coordination of each mitotic event. This cyclin forms a complex with and functions as a regulatory subunit of CDK4 or CDK6, whose activity is required for cell cycle G1/S transition. This protein has been shown to interact with tumor suppressor protein Rb and the expression of this gene is regulated positively by Rb. Mutations, amplification and overexpression of this gene, which alters cell cycle progression, are observed frequently in a variety of human cancers. [provided by RefSeq, Dec 2019].'},\n",
       " {'id': 'HGNC:1027',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'BDH1',\n",
       "  'name': '3-hydroxybutyrate dehydrogenase 1',\n",
       "  'synonyms': 'BDH|SDR9C1',\n",
       "  'description': 'This gene encodes a member of the short-chain dehydrogenase/reductase gene family. The encoded protein forms a homotetrameric lipid-requiring enzyme of the mitochondrial membrane and has a specific requirement for phosphatidylcholine for optimal enzymatic activity. The encoded protein catalyzes the interconversion of acetoacetate and (R)-3-hydroxybutyrate, the two major ketone bodies produced during fatty acid catabolism. Alternatively spliced transcript variants encoding the same protein have been described. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:1051',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'BIK',\n",
       "  'name': 'BCL2 interacting killer',\n",
       "  'synonyms': 'BIP1|BP4|NBK',\n",
       "  'description': 'The protein encoded by this gene shares a critical BH3 domain with other death-promoting proteins, such as BID, BAK, BAD and BAX, that is required for its pro-apoptotic activity, and for interaction with anti-apoptotic members of the BCL2 family, and viral survival-promoting proteins. Since the activity of this protein is suppressed in the presence of survival-promoting proteins, it is suggested as a likely target for anti-apoptotic proteins. [provided by RefSeq, Sep 2011].'},\n",
       " {'id': 'HGNC:9346',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PRDM1',\n",
       "  'name': 'PR/SET domain 1',\n",
       "  'synonyms': 'BLIMP1|PRDI-BF1',\n",
       "  'description': 'This gene encodes a protein that acts as a repressor of beta-interferon gene expression. The protein binds specifically to the PRDI (positive regulatory domain I element) of the beta-IFN gene promoter. Transcription of this gene increases upon virus induction. Two alternatively spliced transcript variants that encode different isoforms have been reported. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:1062',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'BLVRA',\n",
       "  'name': 'biliverdin reductase A',\n",
       "  'synonyms': 'BLVR|BVR|BVRA',\n",
       "  'description': 'The protein encoded by this gene belongs to the biliverdin reductase family, members of which catalyze the conversion of biliverdin to bilirubin in the presence of NADPH or NADH. Mutations in this gene are associated with hyperbiliverdinemia. Alternatively spliced transcript variants have been found for this gene. [provided by RefSeq, Dec 2011].'},\n",
       " {'id': 'HGNC:1075',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'BMP8B',\n",
       "  'name': 'bone morphogenetic protein 8b',\n",
       "  'synonyms': 'BMP8|OP2',\n",
       "  'description': 'This gene encodes a secreted ligand of the TGF-beta (transforming growth factor-beta) superfamily of proteins. Ligands of this family bind various TGF-beta receptors leading to recruitment and activation of SMAD family transcription factors that regulate gene expression. The encoded preproprotein is proteolytically processed to generate each subunit of the disulfide-linked homodimer. The encoded protein stimulates thermogenesis in brown adipose tissue. Expression of this gene may be downregulated in pancreatic cancer. This gene may have arose from a gene duplication event and its gene duplicate is also present on chromosome 1. [provided by RefSeq, Jul 2016].'},\n",
       " {'id': 'HGNC:1097',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'BRAF',\n",
       "  'name': 'B-Raf proto-oncogene, serine/threonine kinase',\n",
       "  'synonyms': 'B-RAF1|B-raf|BRAF1|NS7|RAFB1',\n",
       "  'description': 'This gene encodes a protein belonging to the RAF family of serine/threonine protein kinases. This protein plays a role in regulating the MAP kinase/ERK signaling pathway, which affects cell division, differentiation, and secretion. Mutations in this gene, most commonly the V600E mutation, are the most frequently identified cancer-causing mutations in melanoma, and have been identified in various other cancers as well, including non-Hodgkin lymphoma, colorectal cancer, thyroid carcinoma, non-small cell lung carcinoma, hairy cell leukemia and adenocarcinoma of lung. Mutations in this gene are also associated with cardiofaciocutaneous, Noonan, and Costello syndromes, which exhibit overlapping phenotypes. A pseudogene of this gene has been identified on the X chromosome. [provided by RefSeq, Aug 2017].'},\n",
       " {'id': 'HGNC:1247',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'C1S',\n",
       "  'name': 'complement C1s',\n",
       "  'synonyms': 'EDSPD2',\n",
       "  'description': 'This gene encodes a serine protease, which is a major constituent of the human complement subcomponent C1. C1s associates with two other complement components C1r and C1q in order to yield the first component of the serum complement system. Defects in this gene are the cause of selective C1s deficiency. [provided by RefSeq, Mar 2009].'},\n",
       " {'id': 'HGNC:1442',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CALM1',\n",
       "  'name': 'calmodulin 1',\n",
       "  'synonyms': 'CALML2|CAM2|CAM3|CAMB|CAMC|CAMI|CAMIII|CPVT4|DD132|LQT14|PHKD|caM',\n",
       "  'description': 'This gene encodes one of three calmodulin proteins which are members of the EF-hand calcium-binding protein family. Calcium-induced activation of calmodulin regulates and modulates the function of cardiac ion channels. Two pseudogenes have been identified on chromosome 7 and X. Multiple transcript variants encoding different isoforms have been found for this gene.A missense mutation in the CALM1 gene has been associated with ventricular tachycardia.[provided by RefSeq, May 2020].'},\n",
       " {'id': 'HGNC:1455',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CALR',\n",
       "  'name': 'calreticulin',\n",
       "  'synonyms': 'CRT|HEL-S-99n|RO|SSA|cC1qR',\n",
       "  'description': 'Calreticulin is a highly conserved chaperone protein which resides primarily in the endoplasmic reticulum, and is involved in a variety of cellular processes, among them, cell adhesion. Additionally, it functions in protein folding quality control and calcium homeostasis. Calreticulin is also found in the nucleus, suggesting that it may have a role in transcription regulation. Systemic lupus erythematosus is associated with increased autoantibody titers against calreticulin. Recurrent mutations in calreticulin have been linked to various neoplasms, including the myeloproliferative type.[provided by RefSeq, May 2020].'},\n",
       " {'id': 'HGNC:1500',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CASP10',\n",
       "  'name': 'caspase 10',\n",
       "  'synonyms': 'ALPS2|FLICE-2|FLICE2|MCH4',\n",
       "  'description': 'This gene encodes a protein which is a member of the cysteine-aspartic acid protease (caspase) family. Sequential activation of caspases plays a central role in the execution-phase of cell apoptosis. Caspases exist as inactive proenzymes which undergo proteolytic processing at conserved aspartic residues to produce two subunits, large and small, that dimerize to form the active enzyme. This protein cleaves and activates caspases 3 and 7, and the protein itself is processed by caspase 8. Mutations in this gene are associated with type IIA autoimmune lymphoproliferative syndrome, non-Hodgkin lymphoma and gastric cancer. Alternatively spliced transcript variants encoding different isoforms have been described for this gene. [provided by RefSeq, Apr 2011].'},\n",
       " {'id': 'HGNC:1514',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CASR',\n",
       "  'name': 'calcium sensing receptor',\n",
       "  'synonyms': 'CAR|EIG8|FHH|FIH|GPRC2A|HHC|HHC1|HYPOC1|NSHPT|PCAR1|hCasR',\n",
       "  'description': 'The protein encoded by this gene is a plasma membrane G protein-coupled receptor that senses small changes in circulating calcium concentration. The encoded protein couples this information to intracellular signaling pathways that modify parathyroid hormone secretion or renal cation handling, and thus this protein plays an essential role in maintaining mineral ion homeostasis. Mutations in this gene are a cause of familial hypocalciuric hypercalcemia, neonatal severe hyperparathyroidism, and autosomal dominant hypocalcemia. [provided by RefSeq, Aug 2017].'},\n",
       " {'id': 'HGNC:1543',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CBLN1',\n",
       "  'name': 'cerebellin 1 precursor',\n",
       "  'synonyms': nan,\n",
       "  'description': 'This gene encodes a cerebellum-specific precursor protein, precerebellin, with similarity to the globular (non-collagen-like) domain of complement component C1qB. Precerebellin is processed to give rise to several derivatives, including the hexadecapeptide, cerebellin, which is highly enriched in postsynaptic structures of Purkinje cells. Cerebellin has also been found in human and rat adrenals, where it has been shown to enhance the secretory activity of this gland. [provided by RefSeq, Aug 2008].'},\n",
       " {'id': 'HGNC:1550',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CBS',\n",
       "  'name': 'cystathionine beta-synthase',\n",
       "  'synonyms': 'CBSL|HIP4',\n",
       "  'description': 'The protein encoded by this gene acts as a homotetramer to catalyze the conversion of homocysteine to cystathionine, the first step in the transsulfuration pathway. The encoded protein is allosterically activated by adenosyl-methionine and uses pyridoxal phosphate as a cofactor. Defects in this gene can cause cystathionine beta-synthase deficiency (CBSD), which can lead to homocystinuria. This gene is a major contributor to cellular hydrogen sulfide production. Multiple alternatively spliced transcript variants have been found for this gene. [provided by RefSeq, Feb 2016].'},\n",
       " {'id': 'HGNC:1589',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CCNE1',\n",
       "  'name': 'cyclin E1',\n",
       "  'synonyms': 'CCNE|pCCNE1',\n",
       "  'description': 'The protein encoded by this gene belongs to the highly conserved cyclin family, whose members are characterized by a dramatic periodicity in protein abundance through the cell cycle. Cyclins function as regulators of CDK kinases. Different cyclins exhibit distinct expression and degradation patterns which contribute to the temporal coordination of each mitotic event. This cyclin forms a complex with and functions as a regulatory subunit of CDK2, whose activity is required for cell cycle G1/S transition. This protein accumulates at the G1-S phase boundary and is degraded as cells progress through S phase. Overexpression of this gene has been observed in many tumors, which results in chromosome instability, and thus may contribute to tumorigenesis. This protein was found to associate with, and be involved in, the phosphorylation of NPAT protein (nuclear protein mapped to the ATM locus), which participates in cell-cycle regulated histone gene expression and plays a critical role in promoting cell-cycle progression in the absence of pRB. [provided by RefSeq, Apr 2016].'},\n",
       " {'id': 'HGNC:1691',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CD6',\n",
       "  'name': 'CD6 molecule',\n",
       "  'synonyms': 'TP120',\n",
       "  'description': 'This gene encodes a protein found on the outer membrane of T-lymphocytes as well as some other immune cells. The encoded protein contains three scavenger receptor cysteine-rich (SRCR) domains and a binding site for an activated leukocyte cell adhesion molecule. The gene product is important for continuation of T cell activation. This gene may be associated with susceptibility to multiple sclerosis (PMID: 19525953, 21849685). Multiple transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Dec 2011].'},\n",
       " {'id': 'HGNC:1696',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CD72',\n",
       "  'name': 'CD72 molecule',\n",
       "  'synonyms': 'CD72b|LYB2',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:1697',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CD74',\n",
       "  'name': 'CD74 molecule',\n",
       "  'synonyms': 'DHLAG|HLADG|II|Ia-GAMMA|p33',\n",
       "  'description': 'The protein encoded by this gene associates with class II major histocompatibility complex (MHC) and is an important chaperone that regulates antigen presentation for immune response. It also serves as cell surface receptor for the cytokine macrophage migration inhibitory factor (MIF) which, when bound to the encoded protein, initiates survival pathways and cell proliferation. This protein also interacts with amyloid precursor protein (APP) and suppresses the production of amyloid beta (Abeta). Multiple alternatively spliced transcript variants encoding different isoforms have been identified. [provided by RefSeq, Aug 2011].'},\n",
       " {'id': 'HGNC:1722',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CDK1',\n",
       "  'name': 'cyclin dependent kinase 1',\n",
       "  'synonyms': 'CDC2|CDC28A|P34CDC2',\n",
       "  'description': 'The protein encoded by this gene is a member of the Ser/Thr protein kinase family. This protein is a catalytic subunit of the highly conserved protein kinase complex known as M-phase promoting factor (MPF), which is essential for G1/S and G2/M phase transitions of eukaryotic cell cycle. Mitotic cyclins stably associate with this protein and function as regulatory subunits. The kinase activity of this protein is controlled by cyclin accumulation and destruction through the cell cycle. The phosphorylation and dephosphorylation of this protein also play important regulatory roles in cell cycle control. Alternatively spliced transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Mar 2009].'},\n",
       " {'id': 'HGNC:1725',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CDC25A',\n",
       "  'name': 'cell division cycle 25A',\n",
       "  'synonyms': 'CDC25A2',\n",
       "  'description': 'CDC25A is a member of the CDC25 family of phosphatases. CDC25A is required for progression from G1 to the S phase of the cell cycle. It activates the cyclin-dependent kinase CDC2 by removing two phosphate groups. CDC25A is specifically degraded in response to DNA damage, which prevents cells with chromosomal abnormalities from progressing through cell division. CDC25A is an oncogene, although its exact role in oncogenesis has not been demonstrated. Two transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:1764',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CDH5',\n",
       "  'name': 'cadherin 5',\n",
       "  'synonyms': '7B4|CD144',\n",
       "  'description': 'This gene encodes a classical cadherin of the cadherin superfamily. The encoded preproprotein is proteolytically processed to generate the mature glycoprotein. This calcium-dependent cell-cell adhesion molecule is comprised of five extracellular cadherin repeats, a transmembrane region and a highly conserved cytoplasmic tail. Functioning as a classical cadherin by imparting to cells the ability to adhere in a homophilic manner, this protein plays a role in endothelial adherens junction assembly and maintenance. This gene is located in a gene cluster in a region on the long arm of chromosome 16 that is involved in loss of heterozygosity events in breast and prostate cancer. [provided by RefSeq, Nov 2015].'},\n",
       " {'id': 'HGNC:1766',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CDH7',\n",
       "  'name': 'cadherin 7',\n",
       "  'synonyms': 'CDH7L1',\n",
       "  'description': 'This gene encodes a type II classical cadherin of the cadherin superfamily. Alternative splicing results in multiple transcript variants, at least one of which encodes a preproprotein that is proteolytically processed to generate the mature glycoprotein. This calcium dependent cell-cell adhesion molecule is comprised of five extracellular cadherin repeats, a transmembrane region and a highly conserved cytoplasmic tail. Type II (atypical) cadherins are defined based on their lack of a histidine-alanine-valine (HAV) cell adhesion recognition sequence specific to type I cadherins. Cadherins mediate cell-cell binding in a homophilic manner, contributing to the sorting of heterogeneous cell types. Mutations in this gene may be associated with bipolar disease in human patients. This gene is present in a gene cluster on chromosome 18. [provided by RefSeq, May 2016].'},\n",
       " {'id': 'HGNC:1787',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CDKN2A',\n",
       "  'name': 'cyclin dependent kinase inhibitor 2A',\n",
       "  'synonyms': 'ARF|CDK4I|CDKN2|CMM2|INK4|INK4A|MLM|MTS-1|MTS1|P14|P14ARF|P16|P16-INK4A|P16INK4|P16INK4A|P19|P19ARF|TP16',\n",
       "  'description': 'This gene generates several transcript variants which differ in their first exons. At least three alternatively spliced variants encoding distinct proteins have been reported, two of which encode structurally related isoforms known to function as inhibitors of CDK4 kinase. The remaining transcript includes an alternate first exon located 20 Kb upstream of the remainder of the gene; this transcript contains an alternate open reading frame (ARF) that specifies a protein which is structurally unrelated to the products of the other variants. This ARF product functions as a stabilizer of the tumor suppressor protein p53 as it can interact with, and sequester, the E3 ubiquitin-protein ligase MDM2, a protein responsible for the degradation of p53. In spite of the structural and functional differences, the CDK inhibitor isoforms and the ARF product encoded by this gene, through the regulatory roles of CDK4 and p53 in cell cycle G1 progression, share a common functionality in cell cycle G1 control. This gene is frequently mutated or deleted in a wide variety of tumors, and is known to be an important tumor suppressor gene. [provided by RefSeq, Sep 2012].'},\n",
       " {'id': 'HGNC:1789',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CDKN2C',\n",
       "  'name': 'cyclin dependent kinase inhibitor 2C',\n",
       "  'synonyms': 'INK4C|p18|p18-INK4C',\n",
       "  'description': 'The protein encoded by this gene is a member of the INK4 family of cyclin-dependent kinase inhibitors. This protein has been shown to interact with CDK4 or CDK6, and prevent the activation of the CDK kinases, thus function as a cell growth regulator that controls cell cycle G1 progression. Ectopic expression of this gene was shown to suppress the growth of human cells in a manner that appears to correlate with the presence of a wild-type RB1 function. Studies in the knockout mice suggested the roles of this gene in regulating spermatogenesis, as well as in suppressing tumorigenesis. Two alternatively spliced transcript variants of this gene, which encode an identical protein, have been reported. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:1848',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CEL',\n",
       "  'name': 'carboxyl ester lipase',\n",
       "  'synonyms': 'BAL|BSDL|BSSL|CELL|CEase|FAP|FAPP|LIPA|MODY8',\n",
       "  'description': 'The protein encoded by this gene is a glycoprotein secreted from the pancreas into the digestive tract and from the lactating mammary gland into human milk. The physiological role of this protein is in cholesterol and lipid-soluble vitamin ester hydrolysis and absorption. This encoded protein promotes large chylomicron production in the intestine. Also its presence in plasma suggests its interactions with cholesterol and oxidized lipoproteins to modulate the progression of atherosclerosis. In pancreatic tumoral cells, this encoded protein is thought to be sequestrated within the Golgi compartment and is probably not secreted. This gene contains a variable number of tandem repeat (VNTR) polymorphism in the coding region that may influence the function of the encoded protein. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:1869',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CETP',\n",
       "  'name': 'cholesteryl ester transfer protein',\n",
       "  'synonyms': 'BPIFF|HDLCQ10',\n",
       "  'description': 'The protein encoded by this gene is found in plasma, where it is involved in the transfer of cholesteryl ester from high density lipoprotein (HDL) to other lipoproteins. Defects in this gene are a cause of hyperalphalipoproteinemia 1 (HALP1). Two transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Oct 2013].'},\n",
       " {'id': 'HGNC:1937',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CHKA',\n",
       "  'name': 'choline kinase alpha',\n",
       "  'synonyms': 'CHK|CK|CKI|EK',\n",
       "  'description': 'The major pathway for the biosynthesis of phosphatidylcholine occurs via the CDP-choline pathway. The protein encoded by this gene is the initial enzyme in the sequence and may play a regulatory role. The encoded protein also catalyzes the phosphorylation of ethanolamine. Two transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:1940',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CHM',\n",
       "  'name': 'CHM Rab escort protein',\n",
       "  'synonyms': 'DXS540|GGTA|HSD-32|REP-1|TCD',\n",
       "  'description': 'This gene encodes component A of the RAB geranylgeranyl transferase holoenzyme. In the dimeric holoenzyme, this subunit binds unprenylated Rab GTPases and then presents them to the catalytic Rab GGTase subunit for the geranylgeranyl transfer reaction. Rab GTPases need to be geranylgeranyled on either one or two cysteine residues in their C-terminus to localize to the correct intracellular membrane. Mutations in this gene are a cause of choroideremia; also known as tapetochoroidal dystrophy (TCD). This X-linked disease is characterized by progressive dystrophy of the choroid, retinal pigment epithelium and retina. Alternatively spliced transcript variants have been found for this gene. [provided by RefSeq, Mar 2016].'},\n",
       " {'id': 'HGNC:3439',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ERCC8',\n",
       "  'name': 'ERCC excision repair 8, CSA ubiquitin ligase complex subunit',\n",
       "  'synonyms': 'CKN1|CSA|UVSS2',\n",
       "  'description': 'This gene encodes a WD repeat protein, which interacts with Cockayne syndrome type B (CSB) protein and with p44 protein, a subunit of the RNA polymerase II transcription factor IIH. Mutations in this gene have been identified in patients with hereditary disease Cockayne syndrome (CS). CS cells are abnormally sensitive to ultraviolet radiation and are defective in the repair of transcriptionally active genes. Several transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Mar 2014].'},\n",
       " {'id': 'HGNC:2092',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CLTC',\n",
       "  'name': 'clathrin heavy chain',\n",
       "  'synonyms': 'CHC|CHC17|CLH-17|CLTCL2|Hc|MRD56',\n",
       "  'description': 'Clathrin is a major protein component of the cytoplasmic face of intracellular organelles, called coated vesicles and coated pits. These specialized organelles are involved in the intracellular trafficking of receptors and endocytosis of a variety of macromolecules.  The basic subunit of the clathrin coat is composed of three heavy chains and three light chains. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:2279',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'COX6A2',\n",
       "  'name': 'cytochrome c oxidase subunit 6A2',\n",
       "  'synonyms': 'COX6AH|COXVIAH|MC4DN18',\n",
       "  'description': 'Cytochrome c oxidase (COX), the terminal enzyme of the mitochondrial respiratory chain, catalyzes the electron transfer from reduced cytochrome c to oxygen. It is a heteromeric complex consisting of 3 catalytic subunits encoded by mitochondrial genes and multiple structural subunits encoded by nuclear genes. The mitochondrially-encoded subunits function in electron transfer, and the nuclear-encoded subunits may be involved in the regulation and assembly of the complex. This nuclear gene encodes polypeptide 2 (heart/muscle isoform) of subunit VIa, and polypeptide 2 is present only in striated muscles. Polypeptide 1 (liver isoform) of subunit VIa is encoded by a different gene, and is found in all non-muscle tissues. These two polypeptides share 66% amino acid sequence identity. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:2263',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'COX15',\n",
       "  'name': 'cytochrome c oxidase assembly homolog COX15',\n",
       "  'synonyms': 'CEMCOX2|MC4DN6',\n",
       "  'description': \"Cytochrome c oxidase (COX), the terminal component of the mitochondrial respiratory chain, catalyzes the electron transfer from reduced cytochrome c to oxygen. This component is a heteromeric complex consisting of 3 catalytic subunits encoded by mitochondrial genes and multiple structural subunits encoded by nuclear genes. The mitochondrially-encoded subunits function in electron transfer, and the nuclear-encoded subunits may function in the regulation and assembly of the complex. This nuclear gene encodes a protein which is not a structural subunit, but may be essential for the biogenesis of COX formation and may function in the hydroxylation of heme O, according to the yeast mutant studies. This protein is predicted to contain 5 transmembrane domains localized in the mitochondrial inner membrane. Alternative splicing of this gene generates two transcript variants diverging in the 3' region. [provided by RefSeq, Jul 2008].\"},\n",
       " {'id': 'HGNC:2328',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CPT1A',\n",
       "  'name': 'carnitine palmitoyltransferase 1A',\n",
       "  'synonyms': 'CPT1|CPT1-L|L-CPT1',\n",
       "  'description': 'The mitochondrial oxidation of long-chain fatty acids is initiated by the sequential action of carnitine palmitoyltransferase I (which is located in the outer membrane and is detergent-labile) and carnitine palmitoyltransferase II (which is located in the inner membrane and is detergent-stable), together with a carnitine-acylcarnitine translocase. CPT I is the key enzyme in the carnitine-dependent transport across the mitochondrial inner membrane and its deficiency results in a decreased rate of fatty acid beta-oxidation. Alternatively spliced transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:2496',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CTBS',\n",
       "  'name': 'chitobiase',\n",
       "  'synonyms': 'CTB',\n",
       "  'description': 'Chitobiase is a lysosomal glycosidase involved in degradation of asparagine-linked oligosaccharides on glycoproteins (Aronson and Kuranda, 1989 [PubMed 2531691]).[supplied by OMIM, Nov 2010].'},\n",
       " {'id': 'HGNC:2515',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CTNND1',\n",
       "  'name': 'catenin delta 1',\n",
       "  'synonyms': 'BCDS2|CAS|CTNND|P120CAS|P120CTN|p120|p120(CAS)|p120(CTN)',\n",
       "  'description': 'This gene encodes a member of the Armadillo protein family, which function in adhesion between cells and signal transduction. Multiple translation initiation codons and alternative splicing result in many different isoforms being translated. Not all of the full-length natures of the described transcript variants have been determined. Read-through transcription also exists between this gene and the neighboring upstream thioredoxin-related transmembrane protein 2 (TMX2) gene. [provided by RefSeq, Dec 2010].'},\n",
       " {'id': 'HGNC:2579',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CYC1',\n",
       "  'name': 'cytochrome c1',\n",
       "  'synonyms': 'MC3DN6|UQCR4',\n",
       "  'description': 'This gene encodes a subunit of the cytochrome bc1 complex, which plays an important role in the mitochondrial respiratory chain by transferring electrons from the Rieske iron-sulfur protein to cytochrome c. Mutations in this gene may cause mitochondrial complex III deficiency, nuclear type 6. [provided by RefSeq, Dec 2013].'},\n",
       " {'id': 'HGNC:2761',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DEFA1',\n",
       "  'name': 'defensin alpha 1',\n",
       "  'synonyms': 'DEF1|DEFA2|HNP-1|HP-1|HP1|MRS',\n",
       "  'description': 'Defensins are a family of antimicrobial and cytotoxic peptides thought to be involved in host defense. They are abundant in the granules of neutrophils and also found in the epithelia of mucosal surfaces such as those of the intestine, respiratory tract, urinary tract, and vagina. Members of the defensin family are highly similar in protein sequence and distinguished by a conserved cysteine motif. The protein encoded by this gene, defensin, alpha 1, is found in the microbicidal granules of neutrophils and likely plays a role in phagocyte-mediated host defense. Several alpha defensin genes are clustered on chromosome 8. This gene differs from defensin, alpha 3 by only one amino acid. This gene and the gene encoding defensin, alpha 3 are both subject to copy number variation. [provided by RefSeq, Oct 2014].'},\n",
       " {'id': 'HGNC:2964',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DYNC1I2',\n",
       "  'name': 'dynein cytoplasmic 1 intermediate chain 2',\n",
       "  'synonyms': 'DIC74|DNCI2|IC2|NEDMIBA',\n",
       "  'description': 'This gene encodes a member of the dynein intermediate chain family. The encoded protein is a non-catalytic component of the cytoplasmic dynein 1 complex, which acts as a retrograde microtubule motor to transport organelles and vesicles. A pseudogene of this gene is located on chromosome 10. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Nov 2012].'},\n",
       " {'id': 'HGNC:3004',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DPH2',\n",
       "  'name': 'diphthamide biosynthesis 2',\n",
       "  'synonyms': 'DPH2L2',\n",
       "  'description': 'This gene is one of two human genes similar to the yeast gene dph2. The yeast gene was identified by its ability to complement a diphthamide mutant strain, and thus probably functions in diphthamide biosynthesis. Diphthamide is a post-translationally modified histidine residue present in elongation factor 2 (EF2) that is the target of diphtheria toxin ADP-ribosylation. Multiple transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Jan 2016].'},\n",
       " {'id': 'HGNC:3060',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DTX1',\n",
       "  'name': 'deltex E3 ubiquitin ligase 1',\n",
       "  'synonyms': 'RNF140|hDx-1',\n",
       "  'description': 'Studies in Drosophila have identified this gene as encoding a positive regulator of the Notch-signaling pathway. The human gene encodes a protein of unknown function; however, it may play a role in basic helix-loop-helix transcription factor activity. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:3070',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DUSP4',\n",
       "  'name': 'dual specificity phosphatase 4',\n",
       "  'synonyms': 'HVH2|MKP-2|MKP2|TYP',\n",
       "  'description': 'The protein encoded by this gene is a member of the dual specificity protein phosphatase subfamily. These phosphatases inactivate their target kinases by dephosphorylating both the phosphoserine/threonine and phosphotyrosine residues. They negatively regulate members of the mitogen-activated protein (MAP) kinase superfamily (MAPK/ERK, SAPK/JNK, p38), which are associated with cellular proliferation and differentiation. Different members of the family of dual specificity phosphatases show distinct substrate specificities for various MAP kinases, different tissue distribution and subcellular localization, and different modes of inducibility of their expression by extracellular stimuli. This gene product inactivates ERK1, ERK2 and JNK, is expressed in a variety of tissues, and is localized in the nucleus. Two alternatively spliced transcript variants, encoding distinct isoforms, have been observed for this gene. In addition, multiple polyadenylation sites have been reported. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:3078',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DUT',\n",
       "  'name': 'deoxyuridine triphosphatase',\n",
       "  'synonyms': 'dUTPase',\n",
       "  'description': 'This gene encodes an essential enzyme of nucleotide metabolism. The encoded protein forms a ubiquitous, homotetrameric enzyme that hydrolyzes dUTP to dUMP and pyrophosphate. This reaction serves two cellular purposes: providing a precursor (dUMP) for the synthesis of thymine nucleotides needed for DNA replication, and limiting intracellular pools of dUTP. Elevated levels of dUTP lead to increased incorporation of uracil into DNA, which induces extensive excision repair mediated by uracil glycosylase. This repair process, resulting in the removal and reincorporation of dUTP, is self-defeating and leads to DNA fragmentation and cell death. Alternative splicing of this gene leads to different isoforms that localize to either the mitochondrion or nucleus. A related pseudogene is located on chromosome 19. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:3176',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'EDN1',\n",
       "  'name': 'endothelin 1',\n",
       "  'synonyms': 'ARCND3|ET1|HDLCQ7|PPET1|QME',\n",
       "  'description': 'This gene encodes a preproprotein that is proteolytically processed to generate a secreted peptide that belongs to the endothelin/sarafotoxin family. This peptide is a potent vasoconstrictor and its cognate receptors are therapeutic targets in the treatment of pulmonary arterial hypertension. Aberrant expression of this gene may promote tumorigenesis. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Oct 2015].'},\n",
       " {'id': 'HGNC:3229',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'EGF',\n",
       "  'name': 'epidermal growth factor',\n",
       "  'synonyms': 'HOMG4|URG',\n",
       "  'description': 'This gene encodes a member of the epidermal growth factor superfamily. The encoded preproprotein is proteolytically processed to generate the 53-amino acid epidermal growth factor peptide. This protein acts a potent mitogenic factor that plays an important role in the growth, proliferation and differentiation of numerous cell types. This protein acts by binding with high affinity to the cell surface receptor, epidermal growth factor receptor. Defects in this gene are the cause of hypomagnesemia type 4. Dysregulation of this gene has been associated with the growth and progression of certain cancers. Alternative splicing results in multiple transcript variants, at least one of which encodes a preproprotein that is proteolytically processed. [provided by RefSeq, Jan 2016].'},\n",
       " {'id': 'HGNC:3236',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'EGFR',\n",
       "  'name': 'epidermal growth factor receptor',\n",
       "  'synonyms': 'ERBB|ERBB1|ERRP|HER1|NISBD2|PIG61|mENA',\n",
       "  'description': 'The protein encoded by this gene is a transmembrane glycoprotein that is a member of the protein kinase superfamily. This protein is a receptor for members of the epidermal growth factor family. EGFR is a cell surface protein that binds to epidermal growth factor, thus inducing receptor dimerization and tyrosine autophosphorylation leading to cell proliferation. Mutations in this gene are associated with lung cancer. EGFR is a component of the cytokine storm which contributes to a severe form of Coronavirus Disease 2019 (COVID-19) resulting from infection with severe acute respiratory syndrome coronavirus-2 (SARS-CoV-2). [provided by RefSeq, Jul 2020].'},\n",
       " {'id': 'HGNC:3336',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ADGRE1',\n",
       "  'name': 'adhesion G protein-coupled receptor E1',\n",
       "  'synonyms': 'EMR1|TM7LN3',\n",
       "  'description': 'This gene encodes a protein that has a domain resembling seven transmembrane G protein-coupled hormone receptors (7TM receptors) at its C-terminus. The N-terminus of the encoded protein has six EGF-like modules, separated from the transmembrane segments by a serine/threonine-rich domain, a feature reminiscent of mucin-like, single-span, integral membrane glycoproteins with adhesive properties. Multiple alternatively spliced transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Jan 2012].'},\n",
       " {'id': 'HGNC:3342',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'EN1',\n",
       "  'name': 'engrailed homeobox 1',\n",
       "  'synonyms': 'ENDOVESLB',\n",
       "  'description': \"Homeobox-containing genes are thought to have a role in controlling development. In Drosophila, the 'engrailed' (en) gene plays an important role during development in segmentation, where it is required for the formation of posterior compartments. Different mutations in the mouse homologs, En1 and En2, produced different developmental defects that frequently are lethal. The human engrailed homologs 1 and 2 encode homeodomain-containing proteins and have been implicated in the control of pattern formation during development of the central nervous system. [provided by RefSeq, Jul 2008].\"},\n",
       " {'id': 'HGNC:3355',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ENPEP',\n",
       "  'name': 'glutamyl aminopeptidase',\n",
       "  'synonyms': 'APA|CD249|gp160',\n",
       "  'description': 'The ENPEP gene encodes glutamyl aminopeptidase, a type II integral membrane protein with an extracellular zinc-binding domain. This protein can upregulate blood pressure by cleaving the N-terminal aspartate from angiotensin II, and can regulate blood vessel formation and enhance tumorigenesis in some tissues. Along with ANPEP and DPP4, ENPEP was found to be a candidate co-receptor for the coronavirus SARS-CoV-2, which causes COVID-19. [provided by RefSeq, Apr 2020].'},\n",
       " {'id': 'HGNC:3447',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ERH',\n",
       "  'name': 'ERH mRNA splicing and mitosis factor',\n",
       "  'synonyms': 'DROER',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:3569',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ACSL1',\n",
       "  'name': 'acyl-CoA synthetase long chain family member 1',\n",
       "  'synonyms': 'ACS1|FACL1|FACL2|LACS|LACS1|LACS2',\n",
       "  'description': 'The protein encoded by this gene is an isozyme of the long-chain fatty-acid-coenzyme A ligase family. Although differing in substrate specificity, subcellular localization, and tissue distribution, all isozymes of this family convert free long-chain fatty acids into fatty acyl-CoA esters, and thereby play a key role in lipid biosynthesis and fatty acid degradation. Several transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Nov 2013].'},\n",
       " {'id': 'HGNC:3650',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'FEN1',\n",
       "  'name': 'flap structure-specific endonuclease 1',\n",
       "  'synonyms': 'FEN-1|MF1|RAD2',\n",
       "  'description': \"The protein encoded by this gene removes 5' overhanging flaps in DNA repair and processes the 5' ends of Okazaki fragments in lagging strand DNA synthesis. Direct physical interaction between this protein and AP endonuclease 1 during long-patch base excision repair provides coordinated loading of the proteins onto the substrate, thus passing the substrate from one enzyme to another. The protein is a member of the XPG/RAD2 endonuclease family and is one of ten proteins essential for cell-free DNA replication. DNA secondary structure can inhibit flap processing at certain trinucleotide repeats in a length-dependent manner by concealing the 5' end of the flap that is necessary for both binding and cleavage by the protein encoded by this gene. Therefore, secondary structure can deter the protective function of this protein, leading to site-specific trinucleotide expansions. [provided by RefSeq, Jul 2008].\"},\n",
       " {'id': 'HGNC:3694',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'FGG',\n",
       "  'name': 'fibrinogen gamma chain',\n",
       "  'synonyms': nan,\n",
       "  'description': 'The protein encoded by this gene is the gamma component of fibrinogen, a blood-borne glycoprotein comprised of three pairs of nonidentical polypeptide chains. Following vascular injury, fibrinogen is cleaved by thrombin to form fibrin which is the most abundant component of blood clots. In addition, various cleavage products of fibrinogen and fibrin regulate cell adhesion and spreading, display vasoconstrictor and chemotactic activities, and are mitogens for several cell types. Mutations in this gene lead to several disorders, including dysfibrinogenemia, hypofibrinogenemia and thrombophilia. Alternative splicing results in transcript variants encoding different isoforms. [provided by RefSeq, Aug 2015].'},\n",
       " {'id': 'HGNC:3700',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'FH',\n",
       "  'name': 'fumarate hydratase',\n",
       "  'synonyms': 'FMRD|HLRCC|HsFH|LRCC|MCL|MCUL1',\n",
       "  'description': 'The protein encoded by this gene is an enzymatic component of the tricarboxylic acid (TCA) cycle, or Krebs cycle, and catalyzes the formation of L-malate from fumarate. It exists in both a cytosolic form and an N-terminal extended form, differing only in the translation start site used. The N-terminal extended form is targeted to the mitochondrion, where the removal of the extension generates the same form as in the cytoplasm. It is similar to some thermostable class II fumarases and functions as a homotetramer. Mutations in this gene can cause fumarase deficiency and lead to progressive encephalopathy. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:3701',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'FHIT',\n",
       "  'name': 'fragile histidine triad diadenosine triphosphatase',\n",
       "  'synonyms': 'AP3Aase|FRA3B',\n",
       "  'description': \"The protein encoded by this gene is a P1-P3-bis(5'-adenosyl) triphosphate hydrolase involved in purine metabolism. This gene encompasses the common fragile site FRA3B on chromosome 3, where carcinogen-induced damage can lead to translocations and aberrant transcripts. In fact, aberrant transcripts from this gene have been found in about half of all esophageal, stomach, and colon carcinomas. The encoded protein is also a tumor suppressor, as loss of its activity results in replication stress and DNA damage. [provided by RefSeq, Aug 2017].\"},\n",
       " {'id': 'HGNC:3826',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'FPR1',\n",
       "  'name': 'formyl peptide receptor 1',\n",
       "  'synonyms': 'FMLP|FPR',\n",
       "  'description': 'This gene encodes a G protein-coupled receptor of mammalian phagocytic cells that is a member of the G-protein coupled receptor 1 family. The protein mediates the response of phagocytic cells to invasion of the host by microorganisms and is important in host defense and inflammation.[provided by RefSeq, Jul 2010].'},\n",
       " {'id': 'HGNC:4115',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GALC',\n",
       "  'name': 'galactosylceramidase',\n",
       "  'synonyms': nan,\n",
       "  'description': 'This gene encodes a lysosomal protein which hydrolyzes the galactose ester bonds of galactosylceramide, galactosylsphingosine, lactosylceramide, and monogalactosyldiglyceride. Mutations in this gene have been associated with Krabbe disease, also known as globoid cell leukodystrophy. Alternate transcriptional splice variants, encoding different isoforms, have been characterized. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:4119',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GALK2',\n",
       "  'name': 'galactokinase 2',\n",
       "  'synonyms': 'GK2',\n",
       "  'description': 'This gene encodes a highly efficient N-acetylgalactosamine (GalNAc) kinase, which has galactokinase activity when galactose is present at high concentrations. The encoded protein is a member of the GHMP kinase family. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Jun 2017].'},\n",
       " {'id': 'HGNC:4124',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GALNT2',\n",
       "  'name': 'polypeptide N-acetylgalactosaminyltransferase 2',\n",
       "  'synonyms': 'CDG2T|GalNAc-T2',\n",
       "  'description': 'This gene encodes a member of the glycosyltransferase 2 protein family. Members of this family initiate mucin-type O-glycoslation of peptides in the Golgi apparatus. The encoded protein may be involved in O-linked glycosylation of the immunoglobulin A1 hinge region. This gene may influence triglyceride levels, and may be involved Type 2 diabetes, as well as several types of cancer. Alternative splicing results in multiple transcript variants. [provided by RefSeq, May 2014].'},\n",
       " {'id': 'HGNC:4194',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GCHFR',\n",
       "  'name': 'GTP cyclohydrolase I feedback regulator',\n",
       "  'synonyms': 'GFRP|HsT16933|P35',\n",
       "  'description': 'GTP cyclohydrolase I feedback regulatory protein binds to and mediates tetrahydrobiopterin inhibition of GTP cyclohydrolase I. The regulatory protein, GCHFR, consists of a homodimer. It is postulated that GCHFR may play a role in regulating phenylalanine metabolism in the liver and in the production of biogenic amine neurotransmitters and nitric oxide. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:4227',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GDI2',\n",
       "  'name': 'GDP dissociation inhibitor 2',\n",
       "  'synonyms': 'HEL-S-46e|RABGDIB',\n",
       "  'description': 'GDP dissociation inhibitors are proteins that regulate the GDP-GTP exchange reaction of members of the rab family, small GTP-binding proteins of the ras superfamily, that are involved in vesicular trafficking of molecules between cellular organelles. GDIs slow the rate of dissociation of GDP from rab proteins and release GDP from membrane-bound rabs. GDI2 is ubiquitously expressed. The GDI2 gene contains many repetitive elements indicating that it may be prone to inversion/deletion rearrangements. Alternative splicing results in multiple transcript variants encoding distinct isoforms. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:4260',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GGT5',\n",
       "  'name': 'gamma-glutamyltransferase 5',\n",
       "  'synonyms': 'GGL|GGT 5|GGT-REL|GGTLA1',\n",
       "  'description': 'This gene is a member of the gamma-glutamyl transpeptidase gene family, and some reports indicate that it is capable of cleaving the gamma-glutamyl moiety of glutathione. The protein encoded by this gene is synthesized as a single, catalytically-inactive polypeptide, that is processed post-transcriptionally to form a heavy and light subunit, with the catalytic activity contained within the small subunit. The encoded enzyme is able to convert leukotriene C4 to leukotriene D4, but appears to have distinct substrate specificity compared to gamma-glutamyl transpeptidase. Alternative splicing results in multiple transcript variants encoding different isoforms. [provided by RefSeq, Oct 2014].'},\n",
       " {'id': 'HGNC:4312',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GCLM',\n",
       "  'name': 'glutamate-cysteine ligase modifier subunit',\n",
       "  'synonyms': 'GLCLR',\n",
       "  'description': 'Glutamate-cysteine ligase, also known as gamma-glutamylcysteine synthetase, is the first rate limiting enzyme of glutathione synthesis. The enzyme consists of two subunits, a heavy catalytic subunit and a light regulatory subunit. Gamma glutamylcysteine synthetase deficiency has been implicated in some forms of hemolytic anemia. Alternative splicing results in multiple transcript variants encoding different isoforms. [provided by RefSeq, Apr 2015].'},\n",
       " {'id': 'HGNC:4384',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GNAI1',\n",
       "  'name': 'G protein subunit alpha i1',\n",
       "  'synonyms': 'Gi',\n",
       "  'description': 'Guanine nucleotide binding proteins are heterotrimeric signal-transducing molecules consisting of alpha, beta, and gamma subunits. The alpha subunit binds guanine nucleotide, can hydrolyze GTP, and can interact with other proteins. The protein encoded by this gene represents the alpha subunit of an inhibitory complex. The encoded protein is part of a complex that responds to beta-adrenergic signals by inhibiting adenylate cyclase. Two transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Jan 2012].'},\n",
       " {'id': 'HGNC:4456',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GPD2',\n",
       "  'name': 'glycerol-3-phosphate dehydrogenase 2',\n",
       "  'synonyms': 'GDH2|GPDM|mGDH|mGPDH',\n",
       "  'description': 'The protein encoded by this gene localizes to the inner mitochondrial membrane and catalyzes the conversion of glycerol-3-phosphate to dihydroxyacetone phosphate, using FAD as a cofactor. Along with GDP1, the encoded protein constitutes the glycerol phosphate shuttle, which reoxidizes NADH formed during glycolysis. Two transcript variants encoding the same protein have been found for this gene.[provided by RefSeq, Jan 2010].'},\n",
       " {'id': 'HGNC:4553',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GPX1',\n",
       "  'name': 'glutathione peroxidase 1',\n",
       "  'synonyms': 'GPXD|GSHPX1',\n",
       "  'description': \"The protein encoded by this gene belongs to the glutathione peroxidase family, members of which catalyze the reduction of organic hydroperoxides and hydrogen peroxide (H2O2) by glutathione, and thereby protect cells against oxidative damage. Other studies indicate that H2O2 is also essential for growth-factor mediated signal transduction, mitochondrial function, and maintenance of thiol redox-balance; therefore, by limiting H2O2 accumulation, glutathione peroxidases are also involved in modulating these processes. Several isozymes of this gene family exist in vertebrates, which vary in cellular location and substrate specificity. This isozyme is the most abundant, is ubiquitously expressed and localized in the cytoplasm, and whose preferred substrate is hydrogen peroxide. It is also a selenoprotein, containing the rare amino acid selenocysteine (Sec) at its active site. Sec is encoded by the UGA codon, which normally signals translation termination. The 3' UTRs of selenoprotein mRNAs contain a conserved stem-loop structure, designated the Sec insertion sequence (SECIS) element, that is necessary for the recognition of UGA as a Sec codon, rather than as a stop signal. This gene contains an in-frame GCG trinucleotide repeat in the coding region, and three alleles with 4, 5 or 6 repeats have been found in the human population. The allele with 4 GCG repeats has been significantly associated with breast cancer risk in premenopausal women. Alternatively spliced transcript variants have been found for this gene. Pseudogenes of this locus have been identified on chromosomes X and 21. [provided by RefSeq, Aug 2017].\"},\n",
       " {'id': 'HGNC:4555',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GPX3',\n",
       "  'name': 'glutathione peroxidase 3',\n",
       "  'synonyms': 'GPx-P|GSHPx-3|GSHPx-P',\n",
       "  'description': \"The protein encoded by this gene belongs to the glutathione peroxidase family, members of which catalyze the reduction of organic hydroperoxides and hydrogen peroxide (H2O2) by glutathione, and thereby protect cells against oxidative damage. Several isozymes of this gene family exist in vertebrates, which vary in cellular location and substrate specificity. This isozyme is secreted, and is abundantly found in plasma. Downregulation of expression of this gene by promoter hypermethylation has been observed in a wide spectrum of human malignancies, including thyroid cancer, hepatocellular carcinoma and chronic myeloid leukemia. This isozyme is also a selenoprotein, containing the rare amino acid selenocysteine (Sec) at its active site. Sec is encoded by the UGA codon, which normally signals translation termination. The 3' UTRs of selenoprotein mRNAs contain a conserved stem-loop structure, designated the Sec insertion sequence (SECIS) element, that is necessary for the recognition of UGA as a Sec codon, rather than as a stop signal. Alternatively spliced transcript variants have been found for this gene. [provided by RefSeq, Jul 2016].\"},\n",
       " {'id': 'HGNC:4556',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GPX4',\n",
       "  'name': 'glutathione peroxidase 4',\n",
       "  'synonyms': 'GPx-4|GSHPx-4|MCSP|PHGPx|SMDS|snGPx|snPHGPx',\n",
       "  'description': \"The protein encoded by this gene belongs to the glutathione peroxidase family, members of which catalyze the reduction of hydrogen peroxide, organic hydroperoxides and lipid hydroperoxides, and thereby protect cells against oxidative damage. Several isozymes of this gene family exist in vertebrates, which vary in cellular location and substrate specificity. This isozyme has a high preference for lipid hydroperoxides and protects cells against membrane lipid peroxidation and cell death. It is also required for normal sperm development; thus, it has been identified as a 'moonlighting' protein because of its ability to serve dual functions as a peroxidase, as well as a structural protein in mature spermatozoa. Mutations in this gene are associated with Sedaghatian type of spondylometaphyseal dysplasia (SMDS). This isozyme is also a selenoprotein, containing the rare amino acid selenocysteine (Sec) at its active site. Sec is encoded by the UGA codon, which normally signals translation termination. The 3' UTRs of selenoprotein mRNAs contain a conserved stem-loop structure, designated the Sec insertion sequence (SECIS) element, that is necessary for the recognition of UGA as a Sec codon, rather than as a stop signal. Transcript variants resulting from alternative splicing or use of alternate promoters have been described to encode isoforms with different subcellular localization. [provided by RefSeq, Dec 2018].\"},\n",
       " {'id': 'HGNC:4574',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GRIA4',\n",
       "  'name': 'glutamate ionotropic receptor AMPA type subunit 4',\n",
       "  'synonyms': 'GLUR4|GLUR4C|GLURD|GluA4|NEDSGA',\n",
       "  'description': 'Glutamate receptors are the predominant excitatory neurotransmitter receptors in the mammalian brain and are activated in a variety of normal neurophysiologic processes. These receptors are heteromeric protein complexes composed of multiple subunits, arranged to form ligand-gated ion channels. The classification of glutamate receptors is based on their activation by different pharmacologic agonists. The subunit encoded by this gene belongs to a family of AMPA (alpha-amino-3-hydroxy-5-methyl-4-isoxazole propionate)-sensitive glutamate receptors, and is subject to RNA editing (AGA->GGA; R->G). Alternative splicing of this gene results in transcript variants encoding different isoforms, which may vary in their signal transduction properties. Some haplotypes of this gene show a positive association with schizophrenia. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:4583',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GRIK5',\n",
       "  'name': 'glutamate ionotropic receptor kainate type subunit 5',\n",
       "  'synonyms': 'EAA2|GRIK2|GluK5|KA2',\n",
       "  'description': 'This gene encodes a protein that belongs to the glutamate-gated ionic channel family. Glutamate functions as the major excitatory neurotransmitter in the central nervous system through activation of ligand-gated ion channels and G protein-coupled membrane receptors. The protein encoded by this gene forms functional heteromeric kainate-preferring ionic channels with the subunits encoded by related gene family members. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Jul 2014].'},\n",
       " {'id': 'HGNC:4606',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PDIA3',\n",
       "  'name': 'protein disulfide isomerase family A member 3',\n",
       "  'synonyms': 'ER60|ERp57|ERp60|ERp61|GRP57|GRP58|HEL-S-269|HEL-S-93n|HsT17083|P58|PI-PLC',\n",
       "  'description': 'This gene encodes a protein of the endoplasmic reticulum that interacts with lectin chaperones calreticulin and calnexin to modulate folding of newly synthesized glycoproteins. The protein was once thought to be a phospholipase; however, it has been demonstrated that the protein actually has protein disulfide isomerase activity. It is thought that complexes of lectins and this protein mediate protein folding by promoting formation of disulfide bonds in their glycoprotein substrates. This protein also functions as a molecular chaperone that prevents the formation of protein aggregates. [provided by RefSeq, Dec 2016].'},\n",
       " {'id': 'HGNC:4634',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GSTM2',\n",
       "  'name': 'glutathione S-transferase mu 2',\n",
       "  'synonyms': 'GST4|GSTM|GSTM2-2|GTHMUS',\n",
       "  'description': \"Cytosolic and membrane-bound forms of glutathione S-transferase are encoded by two distinct supergene families. At present, eight distinct classes of the soluble cytoplasmic mammalian glutathione S-transferases have been identified: alpha, kappa, mu, omega, pi, sigma, theta and zeta. This gene encodes a glutathione S-transferase that belongs to the mu class. The mu class of enzymes functions in the detoxification of electrophilic compounds, including carcinogens, therapeutic drugs, environmental toxins and products of oxidative stress, by conjugation with glutathione. The genes encoding the mu class of enzymes are organized in a gene cluster on chromosome 1p13.3 and are known to be highly polymorphic. These genetic variations can change an individual's susceptibility to carcinogens and toxins as well as affect the toxicity and efficacy of certain drugs. [provided by RefSeq, Jul 2008].\"},\n",
       " {'id': 'HGNC:4823',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HBA1',\n",
       "  'name': 'hemoglobin subunit alpha 1',\n",
       "  'synonyms': 'ECYT7|HBA-T3|HBH|METHBA',\n",
       "  'description': \"The human alpha globin gene cluster located on chromosome 16 spans about 30 kb and includes seven loci: 5'- zeta - pseudozeta - mu - pseudoalpha-1 - alpha-2 - alpha-1 - theta - 3'. The alpha-2 (HBA2) and alpha-1 (HBA1) coding sequences are identical. These genes differ slightly over the 5' untranslated regions and the introns, but they differ significantly over the 3' untranslated regions. Two alpha chains plus two beta chains constitute HbA, which in normal adult life comprises about 97% of the total hemoglobin; alpha chains combine with delta chains to constitute HbA-2, which with HbF (fetal hemoglobin) makes up the remaining 3% of adult hemoglobin. Alpha thalassemias result from deletions of each of the alpha genes as well as deletions of both HBA2 and HBA1; some nondeletion alpha thalassemias have also been reported. [provided by RefSeq, Jul 2008].\"},\n",
       " {'id': 'HGNC:4827',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HBB',\n",
       "  'name': 'hemoglobin subunit beta',\n",
       "  'synonyms': 'CD113t-C|ECYT6|beta-globin',\n",
       "  'description': \"The alpha (HBA) and beta (HBB) loci determine the structure of the 2 types of polypeptide chains in adult hemoglobin, Hb A. The normal adult hemoglobin tetramer consists of two alpha chains and two beta chains. Mutant beta globin causes sickle cell anemia. Absence of beta chain causes beta-zero-thalassemia. Reduced amounts of detectable beta globin causes beta-plus-thalassemia. The order of the genes in the beta-globin cluster is 5'-epsilon -- gamma-G -- gamma-A -- delta -- beta--3'. [provided by RefSeq, Jul 2008].\"},\n",
       " {'id': 'HGNC:4835',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HBZ',\n",
       "  'name': 'hemoglobin subunit zeta',\n",
       "  'synonyms': 'HBAZ|HBZ-T1|HBZ1',\n",
       "  'description': \"Zeta-globin is an alpha-like hemoglobin. The zeta-globin polypeptide is synthesized in the yolk sac of the early embryo, while alpha-globin is produced throughout fetal and adult life. The zeta-globin gene is a member of the human alpha-globin gene cluster that includes five functional genes and two pseudogenes. The order of genes is: 5' - zeta - pseudozeta - mu - pseudoalpha-1 - alpha-2 -alpha-1 - theta1 - 3'. [provided by RefSeq, Nov 2009].\"},\n",
       " {'id': 'HGNC:4886',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HFE',\n",
       "  'name': 'homeostatic iron regulator',\n",
       "  'synonyms': 'HFE1|HH|HLA-H|MVCD7|TFQTL2',\n",
       "  'description': 'The protein encoded by this gene is a membrane protein that is similar to MHC class I-type proteins and associates with beta2-microglobulin (beta2M). It is thought that this protein functions to regulate iron absorption by regulating the interaction of the transferrin receptor with transferrin. The iron storage disorder, hereditary haemochromatosis, is a recessive genetic disorder that results from defects in this gene. At least nine alternatively spliced variants have been described for this gene. Additional variants have been found but their full-length nature has not been determined. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:4920',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HIVEP1',\n",
       "  'name': 'HIVEP zinc finger 1',\n",
       "  'synonyms': 'CIRIP|CRYBP1|GAAP|MBP-1|PRDII-BF1|Schnurri-1|ZAS1|ZNF40|ZNF40A',\n",
       "  'description': 'This gene encodes a transcription factor belonging to the ZAS family, members of which are large proteins that contain a ZAS domain - a modular protein structure consisting of a pair of C2H2 zinc fingers with an acidic-rich region and a serine/threonine-rich sequence. These proteins bind specifically to the DNA sequence motif, GGGACTTTCC, found in the enhancer elements of several viral promoters, including human immunodeficiency virus (HIV), and to related sequences found in the enhancer elements of a number of cellular promoters. This protein binds to this sequence motif, suggesting a role in the transcriptional regulation of both viral and cellular genes. [provided by RefSeq, Oct 2011].'},\n",
       " {'id': 'HGNC:4922',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HK1',\n",
       "  'name': 'hexokinase 1',\n",
       "  'synonyms': 'HK|HK1-ta|HK1-tb|HK1-tc|HKD|HKI|HMSNR|HXK1|NEDVIBA|RP79|hexokinase',\n",
       "  'description': 'Hexokinases phosphorylate glucose to produce glucose-6-phosphate, the first step in most glucose metabolism pathways. This gene encodes a ubiquitous form of hexokinase which localizes to the outer membrane of mitochondria. Mutations in this gene have been associated with hemolytic anemia due to hexokinase deficiency. Alternative splicing of this gene results in several transcript variants which encode different isoforms, some of which are tissue-specific. [provided by RefSeq, Apr 2016].'},\n",
       " {'id': 'HGNC:4983',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HMGB1',\n",
       "  'name': 'high mobility group box 1',\n",
       "  'synonyms': 'HMG-1|HMG1|HMG3|SBP-1',\n",
       "  'description': 'This gene encodes a protein that belongs to the High Mobility Group-box superfamily. The encoded non-histone, nuclear DNA-binding protein regulates transcription, and is involved in organization of DNA. This protein plays a role in several cellular processes, including inflammation, cell differentiation and tumor cell migration. Multiple pseudogenes of this gene have been identified. Alternative splicing results in multiple transcript variants that encode the same protein. [provided by RefSeq, Sep 2015].'},\n",
       " {'id': 'HGNC:5022',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'FOXA2',\n",
       "  'name': 'forkhead box A2',\n",
       "  'synonyms': 'HNF-3-beta|HNF3B|TCF3B',\n",
       "  'description': 'This gene encodes a member of the forkhead class of DNA-binding proteins. These hepatocyte nuclear factors are transcriptional activators for liver-specific genes such as albumin and transthyretin, and they also interact with chromatin. Similar family members in mice have roles in the regulation of metabolism and in the differentiation of the pancreas and liver. This gene has been linked to sporadic cases of maturity-onset diabetes of the young. Transcript variants encoding different isoforms have been identified for this gene. [provided by RefSeq, Oct 2008].'},\n",
       " {'id': 'HGNC:5119',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HOXB8',\n",
       "  'name': 'homeobox B8',\n",
       "  'synonyms': 'HOX2|HOX2D|Hox-2.4',\n",
       "  'description': 'This gene is a member of the Antp homeobox family and encodes a nuclear protein with a homeobox DNA-binding domain. It is included in a cluster of homeobox B genes located on chromosome 17. The encoded protein functions as a sequence-specific transcription factor that is involved in development. Increased expression of this gene is associated with colorectal cancer. Mice that have had the murine ortholog of this gene knocked out exhibit an excessive pathologic grooming behavior. This behavior is similar to the behavior of humans suffering from the obsessive-compulsive spectrum disorder trichotillomania. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:5120',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HOXB9',\n",
       "  'name': 'homeobox B9',\n",
       "  'synonyms': 'HOX-2.5|HOX2|HOX2E',\n",
       "  'description': 'This gene is a member of the Abd-B homeobox family and encodes a protein with a homeobox DNA-binding domain. It is included in a cluster of homeobox B genes located on chromosome 17. The encoded nuclear protein functions as a sequence-specific transcription factor that is involved in cell proliferation and differentiation. Increased expression of this gene is associated with some cases of leukemia, prostate cancer and lung cancer. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:5261',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HSPD1',\n",
       "  'name': 'heat shock protein family D (Hsp60) member 1',\n",
       "  'synonyms': 'CPN60|GROEL|HLD4|HSP-60|HSP60|HSP65|HuCHA60|SPG13',\n",
       "  'description': 'This gene encodes a member of the chaperonin family. The encoded mitochondrial protein may function as a signaling molecule in the innate immune system. This protein is essential for the folding and assembly of newly imported proteins in the mitochondria. This gene is adjacent to a related family member and the region between the 2 genes functions as a bidirectional promoter. Several pseudogenes have been associated with this gene. Two transcript variants encoding the same protein have been identified for this gene. Mutations associated with this gene cause autosomal recessive spastic paraplegia 13. [provided by RefSeq, Jun 2010].'},\n",
       " {'id': 'HGNC:5292',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HTR1F',\n",
       "  'name': '5-hydroxytryptamine receptor 1F',\n",
       "  'synonyms': '5-HT-1F|5-HT1F|5HT6|HTR1EL|MR77',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:5329',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'IAPP',\n",
       "  'name': 'islet amyloid polypeptide',\n",
       "  'synonyms': 'DAP|IAP',\n",
       "  'description': 'This gene encodes a member of the calcitonin family of peptide hormones. This hormone is released from pancreatic beta cells following food intake to regulate blood glucose levels and act as a satiation signal. Human patients with type 1 and advanced type 2 diabetes exhibit reduced levels of the encoded hormone in blood and pancreas. This protein also exhibits a bactericidal, antimicrobial activity. [provided by RefSeq, Jul 2016].'},\n",
       " {'id': 'HGNC:5362',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ID3',\n",
       "  'name': 'inhibitor of DNA binding 3, HLH protein',\n",
       "  'synonyms': 'HEIR-1|bHLHb25',\n",
       "  'description': 'The protein encoded by this gene is a helix-loop-helix (HLH) protein that can form heterodimers with other HLH proteins. However, the encoded protein lacks a basic DNA-binding domain and therefore inhibits the DNA binding of any HLH protein with which it interacts. [provided by RefSeq, Aug 2011].'},\n",
       " {'id': 'HGNC:5401',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SP110',\n",
       "  'name': 'SP110 nuclear body protein',\n",
       "  'synonyms': 'IFI41|IFI75|IPR1|VODI',\n",
       "  'description': 'The nuclear body is a multiprotein complex that may have a role in the regulation of gene transcription. This gene is a member of the SP100/SP140 family of nuclear body proteins and encodes a leukocyte-specific nuclear body component. The protein can function as an activator of gene transcription and may serve as a nuclear hormone receptor coactivator. In addition, it has been suggested that the protein may play a role in ribosome biogenesis and in the induction of myeloid cell differentiation. Alternative splicing has been observed for this gene and three transcript variants, encoding distinct isoforms, have been identified. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:2654',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CCN1',\n",
       "  'name': 'cellular communication network factor 1',\n",
       "  'synonyms': 'CYR61|GIG1|IGFBP10',\n",
       "  'description': 'The secreted protein encoded by this gene is growth factor-inducible and promotes the adhesion of endothelial cells. The encoded protein interacts with several integrins and with heparan sulfate proteoglycan. This protein also plays a role in cell proliferation, differentiation, angiogenesis, apoptosis, and extracellular matrix formation. [provided by RefSeq, Sep 2011].'},\n",
       " {'id': 'HGNC:5991',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'IL1A',\n",
       "  'name': 'interleukin 1 alpha',\n",
       "  'synonyms': 'IL-1 alpha|IL-1A|IL1|IL1-ALPHA|IL1F1',\n",
       "  'description': \"The protein encoded by this gene is a member of the interleukin 1 cytokine family. This cytokine is a pleiotropic cytokine involved in various immune responses, inflammatory processes, and hematopoiesis. This cytokine is produced by monocytes and macrophages as a proprotein, which is proteolytically processed and released in response to cell injury, and thus induces apoptosis. This gene and eight other interleukin 1 family genes form a cytokine gene cluster on chromosome 2. It has been suggested that the polymorphism of these genes is associated with rheumatoid arthritis and Alzheimer's disease. [provided by RefSeq, Jul 2008].\"},\n",
       " {'id': 'HGNC:5992',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'IL1B',\n",
       "  'name': 'interleukin 1 beta',\n",
       "  'synonyms': 'IL-1|IL1-BETA|IL1F2|IL1beta',\n",
       "  'description': 'The protein encoded by this gene is a member of the interleukin 1 cytokine family. This cytokine is produced by activated macrophages as a proprotein, which is proteolytically processed to its active form by caspase 1 (CASP1/ICE). This cytokine is an important mediator of the inflammatory response, and is involved in a variety of cellular activities, including cell proliferation, differentiation, and apoptosis. The induction of cyclooxygenase-2 (PTGS2/COX2) by this cytokine in the central nervous system (CNS) is found to contribute to inflammatory pain hypersensitivity. Similarly, IL-1B has been implicated in human osteoarthritis pathogenesis. Patients with severe Coronavirus Disease 2019 (COVID-19) present elevated levels of pro-inflammatory cytokines such as IL-1B in bronchial alveolar lavage fluid samples. The lung damage induced by the Severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2) is to a large extent, a result of the inflammatory response promoted by cytokines such as IL-1B. This gene and eight other interleukin 1 family genes form a cytokine gene cluster on chromosome 2. [provided by RefSeq, Jul 2020].'},\n",
       " {'id': 'HGNC:5962',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'IL10',\n",
       "  'name': 'interleukin 10',\n",
       "  'synonyms': 'CSIF|GVHDS|IL-10|IL10A|TGIF',\n",
       "  'description': 'The protein encoded by this gene is a cytokine produced primarily by monocytes and to a lesser extent by lymphocytes. This cytokine has pleiotropic effects in immunoregulation and inflammation. It down-regulates the expression of Th1 cytokines, MHC class II Ags, and costimulatory molecules on macrophages. It also enhances B cell survival, proliferation, and antibody production. This cytokine can block NF-kappa B activity, and is involved in the regulation of the JAK-STAT signaling pathway. Knockout studies in mice suggested the function of this cytokine as an essential immunoregulator in the intestinal tract. Mutations in this gene are associated with an increased susceptibility to HIV-1 infection and rheumatoid arthritis. [provided by RefSeq, May 2020].'},\n",
       " {'id': 'HGNC:5971',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'IL12RB1',\n",
       "  'name': 'interleukin 12 receptor subunit beta 1',\n",
       "  'synonyms': 'CD212|IL-12R-BETA1|IL12RB|IMD30',\n",
       "  'description': 'The protein encoded by this gene is a type I transmembrane protein that belongs to the hemopoietin receptor superfamily. This protein binds to interleukine 12 (IL12) with a low affinity, and is thought to be a part of IL12 receptor complex. This protein forms a disulfide-linked oligomer, which is required for its IL12 binding activity. The coexpression of this and IL12RB2 proteins was shown to lead to the formation of high-affinity IL12 binding sites and reconstitution of IL12 dependent signaling. Mutations in this gene impair the development of interleukin-17-producing T lymphocytes and result in increased susceptibility to mycobacterial and Salmonella infections. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Feb 2014].'},\n",
       " {'id': 'HGNC:6148',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ITGAL',\n",
       "  'name': 'integrin subunit alpha L',\n",
       "  'synonyms': 'CD11A|LFA-1|LFA1A',\n",
       "  'description': 'ITGAL encodes the integrin alpha L chain. Integrins are heterodimeric integral membrane proteins composed of an alpha chain and a beta chain. This I-domain containing alpha integrin combines with the beta 2 chain (ITGB2) to form the integrin lymphocyte function-associated antigen-1 (LFA-1), which is expressed on all leukocytes. LFA-1 plays a central role in leukocyte intercellular adhesion through interactions with its ligands, ICAMs 1-3 (intercellular adhesion molecules 1 through 3), and also functions in lymphocyte costimulatory signaling. Two transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:6166',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ITIH1',\n",
       "  'name': 'inter-alpha-trypsin inhibitor heavy chain 1',\n",
       "  'synonyms': 'H1P|IATIH|IGHEP1|ITI-HC1|ITIH|SHAP',\n",
       "  'description': 'This gene encodes a member of the inter-alpha-trypsin inhibitor family of proteins. Alternative splicing results in multiple transcript variants, at least one of which encodes a preproprotein that is proteolytically processed to generate the heavy chain of the inter-alpha-trypsin inhibitor complex, which is secreted by hepatocytes into the blood. The heavy chain also interacts with hyaluronan, and this interaction may play a role in ovulation and fertilization, and has been implicated in multiple inflammatory diseases. This gene is present in a gene cluster on chromosome 3. [provided by RefSeq, Nov 2015].'},\n",
       " {'id': 'HGNC:6236',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'KCNC4',\n",
       "  'name': 'potassium voltage-gated channel subfamily C member 4',\n",
       "  'synonyms': 'C1orf30|HKSHIIIC|KSHIIIC|KV3.4',\n",
       "  'description': 'The Shaker gene family of Drosophila encodes components of voltage-gated potassium channels and is comprised of four subfamilies. Based on sequence similarity, this gene is similar to the Shaw subfamily. The protein encoded by this gene belongs to the delayed rectifier class of channel proteins and is an integral membrane protein that mediates the voltage-dependent potassium ion permeability of excitable membranes. It generates atypical voltage-dependent transient current that may be important for neuronal excitability. Multiple transcript variants have been found for this gene. [provided by RefSeq, Jul 2010].'},\n",
       " {'id': 'HGNC:6263',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'KCNJ2',\n",
       "  'name': 'potassium inwardly rectifying channel subfamily J member 2',\n",
       "  'synonyms': 'ATFB9|HHBIRK1|HHIRK1|IRK1|KIR2.1|LQT7|SQT3',\n",
       "  'description': 'Potassium channels are present in most mammalian cells, where they participate in a wide range of physiologic responses. The protein encoded by this gene is an integral membrane protein and inward-rectifier type potassium channel. The encoded protein, which has a greater tendency to allow potassium to flow into a cell rather than out of a cell, probably participates in establishing action potential waveform and excitability of neuronal and muscle tissues. Mutations in this gene have been associated with Andersen syndrome, which is characterized by periodic paralysis, cardiac arrhythmias, and dysmorphic features. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:6262',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'KCNJ16',\n",
       "  'name': 'potassium inwardly rectifying channel subfamily J member 16',\n",
       "  'synonyms': 'BIR9|KIR5.1',\n",
       "  'description': 'Potassium channels are present in most mammalian cells, where they participate in a wide range of physiologic responses. The protein encoded by this gene is an integral membrane protein and inward-rectifier type potassium channel. The encoded protein, which tends to allow potassium to flow into rather than out of a cell, can form heterodimers with two other inward-rectifier type potassium channels. It may function in fluid and pH balance regulation. Alternatively spliced transcript variants have been found for this gene. [provided by RefSeq, Apr 2014].'},\n",
       " {'id': 'HGNC:6341',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'KISS1',\n",
       "  'name': 'KiSS-1 metastasis suppressor',\n",
       "  'synonyms': 'HH13|KiSS-1',\n",
       "  'description': 'This gene is a metastasis suppressor gene that suppresses metastases of melanomas and breast carcinomas without affecting tumorigenicity. The encoded protein may inhibit chemotaxis and invasion and thereby attenuate metastasis in malignant melanomas. Studies suggest a putative role in the regulation of events downstream of cell-matrix adhesion, perhaps involving cytoskeletal reorganization. A protein product of this gene, kisspeptin, stimulates gonadotropin-releasing hormone (GnRH)-induced gonadotropin secretion and regulates the pubertal activation of GnRH nuerons. A polymorphism in the terminal exon of this mRNA results in two protein isoforms. An adenosine present at the polymorphic site represents the third position in a stop codon. When the adenosine is absent, a downstream stop codon is utilized and the encoded protein extends for an additional seven amino acid residues. [provided by RefSeq, Mar 2012].'},\n",
       " {'id': 'HGNC:6459',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'KRT82',\n",
       "  'name': 'keratin 82',\n",
       "  'synonyms': 'HB2|Hb-2|KRTHB2',\n",
       "  'description': 'The protein encoded by this gene is a member of the keratin gene family. As a type II hair keratin, it is a basic protein which heterodimerizes with type I keratins to form hair and nails. The type II hair keratins are clustered in a region of chromosome 12q13 and are grouped into two distinct subfamilies based on structure similarity. One subfamily, consisting of KRTHB1, KRTHB3, and KRTHB6, is highly related. The other less-related subfamily includes KRTHB2, KRTHB4, and KRTHB5. All hair keratins are expressed in the hair follicle; this keratin appears to be a hair cuticle-specific keratin. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:6518',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'LBR',\n",
       "  'name': 'lamin B receptor',\n",
       "  'synonyms': 'C14SR|DHCR14B|LMN2R|PHA|PHASK|TDRD18',\n",
       "  'description': 'The protein encoded by this gene belongs to the ERG4/ERG24 family. It localized in the nuclear envelope inner membrane and anchors the lamina and the heterochromatin to the membrane. It may mediate interaction between chromatin and lamin B. Mutations of this gene has been associated with autosomal recessive HEM/Greenberg skeletal dysplasia. Alternative splicing occurs at this locus and two transcript variants encoding the same protein have been identified. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:6619',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'LIPC',\n",
       "  'name': 'lipase C, hepatic type',\n",
       "  'synonyms': 'HDLCQ12|HL|HTGL|LIPH',\n",
       "  'description': 'LIPC encodes hepatic triglyceride lipase, which is expressed in liver. LIPC has the dual functions of triglyceride hydrolase and ligand/bridging factor for receptor-mediated lipoprotein uptake. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:6641',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'LMO1',\n",
       "  'name': 'LIM domain only 1',\n",
       "  'synonyms': 'RBTN1|RHOM1|TTG1',\n",
       "  'description': 'This locus encodes a transcriptional regulator that contains two cysteine-rich LIM domains but lacks a DNA-binding domain. LIM domains may play a role in protein interactions; thus the encoded protein may regulate transcription by competitively binding to specific DNA-binding transcription factors. Alterations at this locus have been associated with acute lymphoblastic T-cell leukemia. Chromosomal rearrangements have been observed between this locus and at least two loci, the delta subunit of the T-cell antigen receptor gene and the LIM domain binding 1 gene. Alternatively spliced transcript variants have been described. [provided by RefSeq, Jul 2012].'},\n",
       " {'id': 'HGNC:6665',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'LOXL1',\n",
       "  'name': 'lysyl oxidase like 1',\n",
       "  'synonyms': 'LOL|LOXL',\n",
       "  'description': 'This gene encodes a member of the lysyl oxidase family of proteins. The prototypic member of the family is essential to the biogenesis of connective tissue, encoding an extracellular copper-dependent amine oxidase that catalyzes the first step in the formation of crosslinks in collagen and elastin. The encoded preproprotein is proteolytically processed to generate the mature enzyme. A highly conserved amino acid sequence at the C-terminus end appears to be sufficient for amine oxidase activity, suggesting that each family member may retain this function. The N-terminus is poorly conserved and may impart additional roles in developmental regulation, senescence, tumor suppression, cell growth control, and chemotaxis to each member of the family. Mutations in this gene are associated with exfoliation syndrome. [provided by RefSeq, Jan 2016].'},\n",
       " {'id': 'HGNC:6815',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MAGOH',\n",
       "  'name': 'mago homolog, exon junction complex subunit',\n",
       "  'synonyms': 'MAGOH1|MAGOHA',\n",
       "  'description': 'Drosophila that have mutations in their mago nashi (grandchildless) gene produce progeny with defects in germplasm assembly and germline development. This gene encodes the mammalian mago nashi homolog. In mammals, mRNA expression is not limited to the germ plasm, but is expressed ubiquitously in adult tissues and can be induced by serum stimulation of quiescent fibroblasts. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:6835',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MAP1A',\n",
       "  'name': 'microtubule associated protein 1A',\n",
       "  'synonyms': 'MAP1L|MTAP1A',\n",
       "  'description': 'This gene encodes a protein that belongs to the microtubule-associated protein family. The proteins of this family are thought to be involved in microtubule assembly, which is an essential step in neurogenesis. The product of this gene is a precursor polypeptide that presumably undergoes proteolytic processing to generate the final MAP1A heavy chain and LC2 light chain. Expression of this gene is almost exclusively in the brain. Studies of the rat microtubule-associated protein 1A gene suggested a role in early events of spinal cord development. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:6945',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MCM3',\n",
       "  'name': 'minichromosome maintenance complex component 3',\n",
       "  'synonyms': 'HCC5|P1-MCM3|P1.h|RLFB',\n",
       "  'description': 'The protein encoded by this gene is one of the highly conserved mini-chromosome maintenance proteins (MCM) that are involved in the initiation of eukaryotic genome replication. The hexameric protein complex formed by MCM proteins is a key component of the pre-replication complex (pre_RC) and may be involved in the formation of replication forks and in the recruitment of other DNA replication related proteins. This protein is a subunit of the protein complex that consists of MCM2-7. It has been shown to interact directly with MCM5/CDC46. This protein also interacts with and is acetylated by MCM3AP, a chromatin-associated acetyltransferase. The acetylation of this protein inhibits the initiation of DNA replication and cell cycle progression. Several transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Oct 2018].'},\n",
       " {'id': 'HGNC:6948',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MCM5',\n",
       "  'name': 'minichromosome maintenance complex component 5',\n",
       "  'synonyms': 'CDC46|MGORS8|P1-CDC46',\n",
       "  'description': 'The protein encoded by this gene is structurally very similar to the CDC46 protein from S. cerevisiae, a protein involved in the initiation of DNA replication. The encoded protein is a member of the MCM family of chromatin-binding proteins and can interact with at least two other members of this family. The encoded protein is upregulated in the transition from the G0 to G1/S phase of the cell cycle and may actively participate in cell cycle regulation. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:7104',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MIPEP',\n",
       "  'name': 'mitochondrial intermediate peptidase',\n",
       "  'synonyms': 'COXPD31|HMIP|MIP',\n",
       "  'description': 'The product of this gene performs the final step in processing a specific class of nuclear-encoded proteins targeted to the mitochondrial matrix or inner membrane. This protein is primarily involved in the maturation of oxidative phosphorylation (OXPHOS)-related proteins. This gene may contribute to the functional effects of frataxin deficiency and the clinical manifestations of Friedreich ataxia. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:7127',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MLH1',\n",
       "  'name': 'mutL homolog 1',\n",
       "  'synonyms': 'COCA2|FCC2|HNPCC|HNPCC2|MMRCS1|hMLH1',\n",
       "  'description': 'The protein encoded by this gene can heterodimerize with mismatch repair endonuclease PMS2 to form MutL alpha, part of the DNA mismatch repair system. When MutL alpha is bound by MutS beta and some accessory proteins, the PMS2 subunit of MutL alpha introduces a single-strand break near DNA mismatches, providing an entry point for exonuclease degradation. The encoded protein is also involved in DNA damage signaling and can heterodimerize with DNA mismatch repair protein MLH3 to form MutL gamma, which is involved in meiosis. This gene was identified as a locus frequently mutated in hereditary nonpolyposis colon cancer (HNPCC). [provided by RefSeq, Aug 2017].'},\n",
       " {'id': 'HGNC:7160',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MMP14',\n",
       "  'name': 'matrix metallopeptidase 14',\n",
       "  'synonyms': 'MMP-14|MMP-X1|MT-MMP|MT-MMP 1|MT1-MMP|MT1MMP|MTMMP1|WNCHRS',\n",
       "  'description': \"Proteins of the matrix metalloproteinase (MMP) family are involved in the breakdown of extracellular matrix in normal physiological processes, such as embryonic development, reproduction, and tissue remodeling, as well as in disease processes, such as arthritis and metastasis. Most MMP's are secreted as inactive proproteins which are activated when cleaved by extracellular proteinases. However, the protein encoded by this gene is a member of the membrane-type MMP (MT-MMP) subfamily; each member of this subfamily contains a potential transmembrane domain suggesting that these proteins are expressed at the cell surface rather than secreted. This protein activates MMP2 protein, and this activity may be involved in tumor invasion. [provided by RefSeq, Jul 2008].\"},\n",
       " {'id': 'HGNC:7161',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MMP15',\n",
       "  'name': 'matrix metallopeptidase 15',\n",
       "  'synonyms': 'MMP-15|MT2-MMP|MT2MMP|MTMMP2|SMCP-2',\n",
       "  'description': 'This gene encodes a member of the peptidase M10 family and membrane-type subfamily of matrix metalloproteinases (MMPs). Proteins in this family are involved in the breakdown of extracellular matrix in normal physiological processes, such as embryonic development, reproduction, and tissue remodeling, as well as in disease processes, such as arthritis and metastasis. Members of this subfamily contain a transmembrane domain suggesting that these proteins are expressed at the cell surface rather than secreted. The encoded preproprotein is proteolytically processed to generate the mature protease. This protein may play a role in cancer progression. [provided by RefSeq, Jan 2016].'},\n",
       " {'id': 'HGNC:7211',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MPG',\n",
       "  'name': 'N-methylpurine DNA glycosylase',\n",
       "  'synonyms': 'AAG|ADPG|APNG|CRA36.1|MDG|Mid1|PIG11|PIG16|anpg',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:7218',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MPO',\n",
       "  'name': 'myeloperoxidase',\n",
       "  'synonyms': nan,\n",
       "  'description': 'Myeloperoxidase (MPO) is a heme protein synthesized during myeloid differentiation that constitutes the major component of neutrophil azurophilic granules. Produced as a single chain precursor, myeloperoxidase is subsequently cleaved into a light and heavy chain. The mature myeloperoxidase is a tetramer composed of 2 light chains and 2 heavy chains. This enzyme produces hypohalous acids central to the microbicidal activity of neutrophils. [provided by RefSeq, Nov 2014].'},\n",
       " {'id': 'HGNC:7223',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MPST',\n",
       "  'name': 'mercaptopyruvate sulfurtransferase',\n",
       "  'synonyms': 'MST|TST2|TUM1',\n",
       "  'description': 'This protein encoded by this gene catalyzes the transfer of a sulfur ion from 3-mercaptopyruvate to cyanide or other thiol compounds. It may be involved in cysteine degradation and cyanide detoxification. There is confusion in literature between this protein (mercaptopyruvate sulfurtransferase, MPST), which appears to be cytoplasmic, and thiosulfate sulfurtransferase (rhodanese, TST, GeneID:7263), which is a mitochondrial protein. Deficiency in MPST activity has been implicated in a rare inheritable disorder known as mercaptolactate-cysteine disulfiduria (MCDU). Alternatively spliced transcript variants encoding same or different isoforms have been identified for this gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:7325',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MSH2',\n",
       "  'name': 'mutS homolog 2',\n",
       "  'synonyms': 'COCA1|FCC1|HNPCC|HNPCC1|LCFS2|MMRCS2|hMSH2',\n",
       "  'description': 'This locus is frequently mutated in hereditary nonpolyposis colon cancer (HNPCC). When cloned, it was discovered to be a human homolog of the E. coli mismatch repair gene mutS, consistent with the characteristic alterations in microsatellite sequences (RER+ phenotype) found in HNPCC. Two transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Apr 2012].'},\n",
       " {'id': 'HGNC:7393',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MT1A',\n",
       "  'name': 'metallothionein 1A',\n",
       "  'synonyms': 'MT-1A|MT-IA|MT1|MT1S|MTC',\n",
       "  'description': 'This gene is a member of the metallothionein family of genes. Proteins encoded by this gene family are low in molecular weight, are cysteine-rich, lack aromatic residues, and bind divalent heavy metal ions. The conserved cysteine residues co-ordinate metal ions using mercaptide linkages. These proteins act as anti-oxidants, protect against hydroxyl free radicals, are important in homeostatic control of metal in the cell, and play a role in detoxification of heavy metals. Disruption of two metallothionein genes in mouse resulted in defects in protection against heavy metals, oxidative stress, immune reactions, carcinogens, and displayed obesity. [provided by RefSeq, Sep 2017].'},\n",
       " {'id': 'HGNC:7405',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MT1X',\n",
       "  'name': 'metallothionein 1X',\n",
       "  'synonyms': 'MT-1l|MT1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:7432',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MTHFD1',\n",
       "  'name': 'methylenetetrahydrofolate dehydrogenase, cyclohydrolase and formyltetrahydrofolate synthetase 1',\n",
       "  'synonyms': 'CIMAH|MTHFC|MTHFD',\n",
       "  'description': 'This gene encodes a protein that possesses three distinct enzymatic activities, 5,10-methylenetetrahydrofolate dehydrogenase, 5,10-methenyltetrahydrofolate cyclohydrolase and 10-formyltetrahydrofolate synthetase. Each of these activities catalyzes one of three sequential reactions in the interconversion of 1-carbon derivatives of tetrahydrofolate, which are substrates for methionine, thymidylate, and de novo purine syntheses. The trifunctional enzymatic activities are conferred by two major domains, an aminoterminal portion containing the dehydrogenase and cyclohydrolase activities and a larger synthetase domain. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:7526',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MMUT',\n",
       "  'name': 'methylmalonyl-CoA mutase',\n",
       "  'synonyms': 'MCM|MUT',\n",
       "  'description': 'This gene encodes the mitochondrial enzyme methylmalonyl Coenzyme A mutase. In humans, the product of this gene is a vitamin B12-dependent enzyme which catalyzes the isomerization of methylmalonyl-CoA to succinyl-CoA, while in other species this enzyme may have different functions. Mutations in this gene may lead to various types of methylmalonic aciduria. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:7534',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MXI1',\n",
       "  'name': 'MAX interactor 1, dimerization protein',\n",
       "  'synonyms': 'MAD2|MXD2|MXI|bHLHc11',\n",
       "  'description': 'Expression of the c-myc gene, which produces an oncogenic transcription factor, is tightly regulated in normal cells but is frequently deregulated in human cancers. The protein encoded by this gene is a transcriptional repressor thought to negatively regulate MYC function, and is therefore a potential tumor suppressor. This protein inhibits the transcriptional activity of MYC by competing for MAX, another basic helix-loop-helix protein that binds to MYC and is required for its function. Defects in this gene are frequently found in patients with prostate tumors. Three alternatively spliced transcripts encoding different isoforms have been described. Additional alternatively spliced transcripts may exist but the products of these transcripts have not been verified experimentally. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:7545',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MYB',\n",
       "  'name': 'MYB proto-oncogene, transcription factor',\n",
       "  'synonyms': 'Cmyb|c-myb|c-myb_CDS|efg',\n",
       "  'description': 'This gene encodes a protein with three HTH DNA-binding domains that functions as a transcription regulator. This protein plays an essential role in the regulation of hematopoiesis. This gene may be aberrently expressed or rearranged or undergo translocation in leukemias and lymphomas, and is considered to be an oncogene. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Jan 2016].'},\n",
       " {'id': 'HGNC:7576',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MYH6',\n",
       "  'name': 'myosin heavy chain 6',\n",
       "  'synonyms': 'ASD3|CMD1EE|CMH14|MYHC|MYHCA|SSS3|alpha-MHC',\n",
       "  'description': 'Cardiac muscle myosin is a hexamer consisting of two heavy chain subunits, two light chain subunits, and two regulatory subunits. This gene encodes the alpha heavy chain subunit of cardiac myosin. The gene is located approximately 4kb downstream of the gene encoding the beta heavy chain subunit of cardiac myosin. Mutations in this gene cause familial hypertrophic cardiomyopathy and atrial septal defect 3. [provided by RefSeq, Feb 2017].'},\n",
       " {'id': 'HGNC:7634',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NAIP',\n",
       "  'name': 'NLR family apoptosis inhibitory protein',\n",
       "  'synonyms': 'BIRC1|NLRB1|psiNAIP',\n",
       "  'description': 'This gene is part of a 500 kb inverted duplication on chromosome 5q13. This duplicated region contains at least four genes and repetitive elements which make it prone to rearrangements and deletions. The repetitiveness and complexity of the sequence have also caused difficulty in determining the organization of this genomic region. This copy of the gene is full length; additional copies with truncations and internal deletions are also present in this region of chromosome 5q13. It is thought that this gene is a modifier of spinal muscular atrophy caused by mutations in a neighboring gene, SMN1. The protein encoded by this gene contains regions of homology to two baculovirus inhibitor of apoptosis proteins, and it is able to suppress apoptosis induced by various signals. Alternative splicing and the use of alternative promoters results in multiple transcript variants. [provided by RefSeq, Nov 2016].'},\n",
       " {'id': 'HGNC:7667',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NCL',\n",
       "  'name': 'nucleolin',\n",
       "  'synonyms': 'C23|Nsr1',\n",
       "  'description': 'Nucleolin (NCL), a eukaryotic nucleolar phosphoprotein, is involved in the synthesis and maturation of ribosomes. It is located mainly in dense fibrillar regions of the nucleolus. Human NCL gene consists of 14 exons with 13 introns and spans approximately 11kb. The intron 11 of the NCL gene encodes a small nucleolar RNA, termed U20. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:7691',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NDUFA7',\n",
       "  'name': 'NADH:ubiquinone oxidoreductase subunit A7',\n",
       "  'synonyms': 'B14.5a|CI-B14.5a',\n",
       "  'description': 'This gene encodes a subunit of NADH:ubiquinone oxidoreductase (complex I), which is a multiprotein complex located in the inner mitochondrial membrane. Complex I functions in the transfer of electrons from NADH to the respiratory chain. [provided by RefSeq, Mar 2011].'},\n",
       " {'id': 'HGNC:7694',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NDUFAB1',\n",
       "  'name': 'NADH:ubiquinone oxidoreductase subunit AB1',\n",
       "  'synonyms': 'ACP|ACP1|FASN2A|SDAP',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:7705',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NDUFC1',\n",
       "  'name': 'NADH:ubiquinone oxidoreductase subunit C1',\n",
       "  'synonyms': 'KFYI',\n",
       "  'description': 'The encoded protein is a subunit of the NADH:ubiquinone oxidoreductase (complex I), the first enzyme complex in the electron transport chain located in the inner mitochondrial membrane. Alternative splicing results in multiple transcript variants. [provided by RefSeq, May 2010].'},\n",
       " {'id': 'HGNC:7716',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NDUFV1',\n",
       "  'name': 'NADH:ubiquinone oxidoreductase core subunit V1',\n",
       "  'synonyms': 'CI-51K|CI51KD|MC1DN4|UQOR1',\n",
       "  'description': \"The mitochondrial respiratory chain provides energy to cells via oxidative phosphorylation and consists of four membrane-bound electron-transporting protein complexes (I-IV) and an ATP synthase (complex V). This gene encodes a 51 kDa subunit of the NADH:ubiquinone oxidoreductase complex I; a large complex with at least 45 nuclear and mitochondrial encoded subunits that liberates electrons from NADH and channels them to ubiquinone. This subunit carries the NADH-binding site as well as flavin mononucleotide (FMN)- and Fe-S-biding sites. Defects in complex I are a common cause of mitochondrial dysfunction; a syndrome that occurs in approximately 1 in 10,000 live births. Mitochondrial complex I deficiency is linked to myopathies, encephalomyopathies, and neurodegenerative disorders such as Parkinson's disease and Leigh syndrome. Alternative splicing results in multiple transcript variants encoding distinct isoforms.[provided by RefSeq, Oct 2009].\"},\n",
       " {'id': 'HGNC:10299',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RPL10A',\n",
       "  'name': 'ribosomal protein L10a',\n",
       "  'synonyms': 'CSA19|Csa-19|L10A|NEDD6',\n",
       "  'description': 'Ribosomes, the organelles that catalyze protein synthesis, consist of a small 40S subunit and a large 60S subunit. Together these subunits are composed of 4 RNA species and approximately 80 structurally distinct proteins. This gene encodes a ribosomal protein that is a component of the 60S subunit. The protein belongs to the L1P family of ribosomal proteins. It is located in the cytoplasm. The expression of this gene is downregulated in the thymus by cyclosporin-A (CsA), an immunosuppressive drug. Studies in mice have shown that the expression of the ribosomal protein L10a gene is downregulated in neural precursor cells during development. This gene previously was referred to as NEDD6 (neural precursor cell expressed, developmentally downregulated 6), but it has been renamed RPL10A (ribosomal protein 10a). As is typical for genes encoding ribosomal proteins, there are multiple processed pseudogenes of this gene dispersed through the genome. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:7745',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NEK2',\n",
       "  'name': 'NIMA related kinase 2',\n",
       "  'synonyms': 'HsPK21|NEK2A|NLK1|PPP1R111|RP67',\n",
       "  'description': 'This gene encodes a serine/threonine-protein kinase that is involved in mitotic regulation. This protein is localized to the centrosome, and undetectable during G1 phase, but accumulates progressively throughout the S phase, reaching maximal levels in late G2 phase. Alternatively spliced transcript variants encoding different isoforms with distinct C-termini have been noted for this gene. [provided by RefSeq, Feb 2011].'},\n",
       " {'id': 'HGNC:7939',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NPPA',\n",
       "  'name': 'natriuretic peptide A',\n",
       "  'synonyms': 'ANF|ANP|ATFB6|ATRST2|CDD|CDD-ANF|CDP|PND',\n",
       "  'description': 'The protein encoded by this gene belongs to the natriuretic peptide family. Natriuretic peptides are implicated in the control of extracellular fluid volume and electrolyte homeostasis. This protein is synthesized as a large precursor (containing a signal peptide), which is processed to release a peptide from the N-terminus with similarity to vasoactive peptide, cardiodilatin, and another peptide from the C-terminus with natriuretic-diuretic activity. Mutations in this gene have been associated with atrial fibrillation familial type 6. This gene is located adjacent to another member of the natriuretic family of peptides on chromosome 1. [provided by RefSeq, Oct 2015].'},\n",
       " {'id': 'HGNC:7953',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NPTX2',\n",
       "  'name': 'neuronal pentraxin 2',\n",
       "  'synonyms': 'NARP|NP-II|NP2',\n",
       "  'description': 'This gene encodes a member of the family of neuronal petraxins, synaptic proteins that are related to C-reactive protein. This protein is involved in excitatory synapse formation. It also plays a role in clustering of alpha-amino-3-hydroxy-5-methyl-4-isoxazolepropionic acid (AMPA)-type glutamate receptors at established synapses, resulting in non-apoptotic cell death of dopaminergic nerve cells. Up-regulation of this gene in Parkinson disease (PD) tissues suggests that the protein may be involved in the pathology of PD. [provided by RefSeq, Feb 2009].'},\n",
       " {'id': 'HGNC:8087',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'OAS2',\n",
       "  'name': \"2'-5'-oligoadenylate synthetase 2\",\n",
       "  'synonyms': nan,\n",
       "  'description': \"This gene encodes a member of the 2-5A synthetase family, essential proteins involved in the innate immune response to viral infection. The encoded protein is induced by interferons and uses adenosine triphosphate in 2'-specific nucleotidyl transfer reactions to synthesize 2',5'-oligoadenylates (2-5As). These molecules activate latent RNase L, which results in viral RNA degradation and the inhibition of viral replication. The three known members of this gene family are located in a cluster on chromosome 12. Alternatively spliced transcript variants encoding different isoforms have been described. [provided by RefSeq, Jul 2008].\"},\n",
       " {'id': 'HGNC:8101',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'OCA2',\n",
       "  'name': 'OCA2 melanosomal transmembrane protein',\n",
       "  'synonyms': 'BEY|BEY1|BEY2|BOCA|D15S12|EYCL|EYCL2|EYCL3|HCL3|P|PED|SHEP1',\n",
       "  'description': 'This gene encodes the human homolog of the mouse p (pink-eyed dilution) gene. The encoded protein is believed to be an integral membrane protein involved in small molecule transport, specifically tyrosine, which is a precursor to melanin synthesis. It is involved in mammalian pigmentation, where it may control skin color variation and act as a determinant of brown or blue eye color. Mutations in this gene result in type 2 oculocutaneous albinism. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Jul 2014].'},\n",
       " {'id': 'HGNC:8529',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'OXTR',\n",
       "  'name': 'oxytocin receptor',\n",
       "  'synonyms': 'OT-R',\n",
       "  'description': 'The protein encoded by this gene belongs to the G-protein coupled receptor family and acts as a receptor for oxytocin. Its activity is mediated by G proteins which activate a phosphatidylinositol-calcium second messenger system. The oxytocin-oxytocin receptor system plays an important role in the uterus during parturition. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:8542',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'P2RY4',\n",
       "  'name': 'pyrimidinergic receptor P2Y4',\n",
       "  'synonyms': 'NRU|P2P|P2Y4|UNR',\n",
       "  'description': 'The product of this gene belongs to the family of G-protein coupled receptors. This family has several receptor subtypes with different pharmacological selectivity, which overlaps in some cases, for various adenosine and uridine nucleotides. This receptor is responsive to uridine nucleotides, partially responsive to ATP, and not responsive to ADP. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:8583',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SERPINE1',\n",
       "  'name': 'serpin family E member 1',\n",
       "  'synonyms': 'PAI|PAI-1|PAI1|PLANH1',\n",
       "  'description': 'This gene encodes a member of the serine proteinase inhibitor (serpin) superfamily. This member is the principal inhibitor of tissue plasminogen activator (tPA) and urokinase (uPA), and hence is an inhibitor of fibrinolysis. The protein also functions as a component of innate antiviral immunity. Defects in this gene are the cause of plasminogen activator inhibitor-1 deficiency (PAI-1 deficiency), and high concentrations of the gene product are associated with thrombophilia. [provided by RefSeq, Aug 2020].'},\n",
       " {'id': 'HGNC:8616',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PAX2',\n",
       "  'name': 'paired box 2',\n",
       "  'synonyms': 'FSGS7|PAPRS',\n",
       "  'description': 'PAX2 encodes paired box gene 2, one of many human homologues of the Drosophila melanogaster gene prd. The central feature of this transcription factor gene family is the conserved DNA-binding paired box domain. PAX2 is believed to be a target of transcriptional supression by the tumor suppressor gene WT1. Mutations within PAX2 have been shown to result in optic nerve colobomas and renal hypoplasia. Alternative splicing of this gene results in multiple transcript variants. [provided by RefSeq, Dec 2014].'},\n",
       " {'id': 'HGNC:8729',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PCNA',\n",
       "  'name': 'proliferating cell nuclear antigen',\n",
       "  'synonyms': 'ATLD2',\n",
       "  'description': 'The protein encoded by this gene is found in the nucleus and is a cofactor of DNA polymerase delta. The encoded protein acts as a homotrimer and helps increase the processivity of leading strand synthesis during DNA replication. In response to DNA damage, this protein is ubiquitinated and is involved in the RAD6-dependent DNA repair pathway. Two transcript variants encoding the same protein have been found for this gene. Pseudogenes of this gene have been described on chromosome 4 and on the X chromosome. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:8738',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PCOLCE',\n",
       "  'name': 'procollagen C-endopeptidase enhancer',\n",
       "  'synonyms': 'PCPE|PCPE-1|PCPE1',\n",
       "  'description': 'Fibrillar collagen types I-III are synthesized as precursor molecules known as procollagens. These precursors contain amino- and carboxyl-terminal peptide extensions known as N- and C-propeptides, respectively, which are cleaved, upon secretion of procollagen from the cell, to yield the mature triple helical, highly structured fibrils. This gene encodes a glycoprotein which binds and drives the enzymatic cleavage of type I procollagen and heightens C-proteinase activity. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:8793',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PDE8A',\n",
       "  'name': 'phosphodiesterase 8A',\n",
       "  'synonyms': 'HsT19550',\n",
       "  'description': 'The protein encoded by this gene belongs to the cyclic nucleotide phosphodiesterase (PDE) family, and PDE8 subfamily. This PDE hydrolyzes the second messenger, cAMP, which is a regulator and mediator of a number of cellular responses to extracellular signals. Thus, by regulating the cellular concentration of cAMP, this protein plays a key role in many important physiological processes. Alternatively spliced transcript variants encoding different isoforms have been found for this gene.[provided by RefSeq, Jul 2011].'},\n",
       " {'id': 'HGNC:8826',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PEG3',\n",
       "  'name': 'paternally expressed 3',\n",
       "  'synonyms': 'PW1|ZKSCAN22|ZNF904|ZSCAN24',\n",
       "  'description': \"In human, ZIM2 and PEG3 are treated as two distinct genes though they share multiple 5' exons and a common promoter and both genes are paternally expressed (PMID:15203203). Alternative splicing events connect their shared 5' exons either with the remaining 4 exons unique to ZIM2, or with the remaining 2 exons unique to PEG3. In contrast, in other mammals ZIM2 does not undergo imprinting and, in mouse, cow, and likely other mammals as well, the ZIM2 and PEG3 genes do not share exons. Human PEG3 protein belongs to the Kruppel C2H2-type zinc finger protein family. PEG3 may play a role in cell proliferation and p53-mediated apoptosis. PEG3 has also shown tumor suppressor activity and tumorigenesis in glioma and ovarian cells. Alternative splicing of this PEG3 gene results in multiple transcript variants encoding distinct isoforms. [provided by RefSeq, Sep 2009].\"},\n",
       " {'id': 'HGNC:8863',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PFAS',\n",
       "  'name': 'phosphoribosylformylglycinamidine synthase',\n",
       "  'synonyms': 'FGAMS|FGAR-AT|FGARAT|GATD8|PURL',\n",
       "  'description': 'Purines are necessary for many cellular processes, including DNA replication, transcription, and energy metabolism. Ten enzymatic steps are required to synthesize inosine monophosphate (IMP) in the de novo pathway of purine biosynthesis. The enzyme encoded by this gene catalyzes the fourth step of IMP biosynthesis. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:8869',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PFDN5',\n",
       "  'name': 'prefoldin subunit 5',\n",
       "  'synonyms': 'MM-1|MM1|PFD5',\n",
       "  'description': 'This gene encodes a member of the prefoldin alpha subunit family. The encoded protein is one of six subunits of prefoldin, a molecular chaperone complex that binds and stabilizes newly synthesized polypeptides, thereby allowing them to fold correctly. The complex, consisting of two alpha and four beta subunits, forms a double beta barrel assembly with six protruding coiled-coils. The encoded protein may also repress the transcriptional activity of the proto-oncogene c-Myc. Alternatively spliced transcript variants encoding different isoforms have been described. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:8882',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PFN2',\n",
       "  'name': 'profilin 2',\n",
       "  'synonyms': 'D3S1319E|PFL',\n",
       "  'description': 'The protein encoded by this gene is a ubiquitous actin monomer-binding protein belonging to the profilin family. It is thought to regulate actin polymerization in response to extracellular signals. There are two alternatively spliced transcript variants encoding different isoforms described for this gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:9053',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PLAUR',\n",
       "  'name': 'plasminogen activator, urokinase receptor',\n",
       "  'synonyms': 'CD87|U-PAR|UPAR|URKR',\n",
       "  'description': 'This gene encodes the receptor for urokinase plasminogen activator and, given its role in localizing and promoting plasmin formation, likely influences many normal and pathological processes related to cell-surface plasminogen activation and localized degradation of the extracellular matrix. It binds both the proprotein and mature forms of urokinase plasminogen activator and permits the activation of the receptor-bound pro-enzyme by plasmin. The protein lacks transmembrane or cytoplasmic domains and may be anchored to the plasma membrane by a glycosyl-phosphatidylinositol (GPI) moiety following cleavage of the nascent polypeptide near its carboxy-terminus. However, a soluble protein is also produced in some cell types. Alternative splicing results in multiple transcript variants encoding different isoforms. The proprotein experiences several post-translational cleavage reactions that have not yet been fully defined. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:9137',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'EXOSC9',\n",
       "  'name': 'exosome component 9',\n",
       "  'synonyms': 'PCH1D|PM/Scl-75|PMSCL1|RRP45|Rrp45p|p5|p6',\n",
       "  'description': 'This gene encodes a component of the human exosome, a exoribonuclease complex which processes and degrades RNA in the nucleus and cytoplasm. This component may play a role in mRNA degradation and the polymyositis/scleroderma autoantigen complex. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Aug 2011].'},\n",
       " {'id': 'HGNC:9173',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'POLA1',\n",
       "  'name': 'DNA polymerase alpha 1, catalytic subunit',\n",
       "  'synonyms': 'NSX|POLA|VEODS|p180',\n",
       "  'description': 'This gene encodes the catalytic subunit of DNA polymerase, which together with a regulatory and two primase subunits, forms the DNA polymerase alpha complex. The catalytic subunit plays an essential role in the initiation of DNA replication. [provided by RefSeq, Mar 2010].'},\n",
       " {'id': 'HGNC:9178',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'POLE2',\n",
       "  'name': 'DNA polymerase epsilon 2, accessory subunit',\n",
       "  'synonyms': 'DPE2',\n",
       "  'description': 'DNA polymerase epsilon, which is involved in DNA repair and replication, is composed of a large catalytic subunit and a small accessory subunit. The protein encoded by this gene represents the small subunit (B). Defects in this gene have been linked to colorectal cancer and to combined immunodeficiency. [provided by RefSeq, Jan 2017].'},\n",
       " {'id': 'HGNC:9189',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'POLR2C',\n",
       "  'name': 'RNA polymerase II subunit C',\n",
       "  'synonyms': 'RPB3|RPB31|hRPB33|hsRPB3',\n",
       "  'description': 'This gene encodes the third largest subunit of RNA polymerase II, the polymerase responsible for synthesizing messenger RNA in eukaryotes. The product of this gene contains a cysteine rich region and exists as a heterodimer with another polymerase subunit, POLR2J. These two subunits form a core subassembly unit of the polymerase. A pseudogene has been identified on chromosome 21. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:9211',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'POU2AF1',\n",
       "  'name': 'POU class 2 homeobox associating factor 1',\n",
       "  'synonyms': 'BOB1|OBF-1|OBF1|OCAB',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:9215',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'POU3F2',\n",
       "  'name': 'POU class 3 homeobox 2',\n",
       "  'synonyms': 'BRN2|N-Oct3|OCT7|OTF-7|OTF7|POUF3|brn-2|oct-7',\n",
       "  'description': 'This gene encodes a member of the POU-III class of neural transcription factors. The encoded protein is involved in neuronal differentiation and enhances the activation of corticotropin-releasing hormone regulated genes. Overexpression of this protein is associated with an increase in the proliferation of melanoma cells. [provided by RefSeq, Mar 2012].'},\n",
       " {'id': 'HGNC:9238',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PPAT',\n",
       "  'name': 'phosphoribosyl pyrophosphate amidotransferase',\n",
       "  'synonyms': 'ATASE|GPAT|PRAT',\n",
       "  'description': 'The protein encoded by this gene is a member of the purine/pyrimidine phosphoribosyltransferase family. It is a regulatory allosteric enzyme that catalyzes the first step of de novo purine nucleotide biosythetic pathway. This gene and PAICS/AIRC gene, a bifunctional enzyme catalyzing steps six and seven of this pathway, are located in close proximity on chromosome 4, and divergently transcribed from an intergenic region. [provided by RefSeq, Mar 2011].'},\n",
       " {'id': 'HGNC:9370',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PRIM2',\n",
       "  'name': 'DNA primase subunit 2',\n",
       "  'synonyms': 'PRIM2A|p58',\n",
       "  'description': 'This gene encodes the 58 kilodalton subunit of DNA primase, an enzyme that plays a key role in the replication of DNA. The encoded protein forms a heterodimer with a 49 kilodalton subunit. This heterodimer functions as a DNA-directed RNA polymerase to synthesize small RNA primers that are used to create Okazaki fragments on the lagging strand of the DNA. Alternative splicing of this gene results in multiple transcript variants. This gene has a related pseudogene, which is also present on chromosome 6. [provided by RefSeq, Apr 2014].'},\n",
       " {'id': 'HGNC:9380',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PRKACA',\n",
       "  'name': 'protein kinase cAMP-activated catalytic subunit alpha',\n",
       "  'synonyms': 'CAFD1|PKACA|PPNAD4',\n",
       "  'description': \"This gene encodes one of the catalytic subunits of protein kinase A, which exists as a tetrameric holoenzyme with two regulatory subunits and two catalytic subunits, in its inactive form. cAMP causes the dissociation of the inactive holoenzyme into a dimer of regulatory subunits bound to four cAMP and two free monomeric catalytic subunits. Four different regulatory subunits and three catalytic subunits have been identified in humans. cAMP-dependent phosphorylation of proteins by protein kinase A is important to many cellular processes, including differentiation, proliferation, and apoptosis. Constitutive activation of this gene caused either by somatic mutations, or genomic duplications of regions that include this gene, have been associated with hyperplasias and adenomas of the adrenal cortex and are linked to corticotropin-independent Cushing's syndrome. Alternative splicing results in multiple transcript variants encoding different isoforms. Tissue-specific isoforms that differ at the N-terminus have been described, and these isoforms may differ in the post-translational modifications that occur at the N-terminus of some isoforms. [provided by RefSeq, Jan 2015].\"},\n",
       " {'id': 'HGNC:9388',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PRKAR1A',\n",
       "  'name': 'protein kinase cAMP-dependent type I regulatory subunit alpha',\n",
       "  'synonyms': 'ACRDYS1|ADOHR|CAR|CNC|CNC1|PKR1|PPNAD1|PRKAR1|TSE1',\n",
       "  'description': 'cAMP is a signaling molecule important for a variety of cellular functions. cAMP exerts its effects by activating the cAMP-dependent protein kinase, which transduces the signal through phosphorylation of different target proteins. The inactive kinase holoenzyme is a tetramer composed of two regulatory and two catalytic subunits. cAMP causes the dissociation of the inactive holoenzyme into a dimer of regulatory subunits bound to four cAMP and two free monomeric catalytic subunits. Four different regulatory subunits and three catalytic subunits have been identified in humans. This gene encodes one of the regulatory subunits. This protein was found to be a tissue-specific extinguisher that down-regulates the expression of seven liver genes in hepatoma x fibroblast hybrids. Mutations in this gene cause Carney complex (CNC). This gene can fuse to the RET protooncogene by gene rearrangement and form the thyroid tumor-specific chimeric oncogene known as PTC2. A nonconventional nuclear localization sequence (NLS) has been found for this protein which suggests a role in DNA replication via the protein serving as a nuclear transport protein for the second subunit of the Replication Factor C (RFC40). Several alternatively spliced transcript variants encoding two different isoforms have been observed. [provided by RefSeq, Jan 2013].'},\n",
       " {'id': 'HGNC:9413',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PRKDC',\n",
       "  'name': 'protein kinase, DNA-activated, catalytic subunit',\n",
       "  'synonyms': 'DNA-PKC|DNA-PKcs|DNAPK|DNAPKc|DNPK1|HYRC|HYRC1|IMD26|XRCC7|p350',\n",
       "  'description': 'This gene encodes the catalytic subunit of the DNA-dependent protein kinase (DNA-PK). It functions with the Ku70/Ku80 heterodimer protein in DNA double strand break repair and recombination. The protein encoded is a member of the PI3/PI4-kinase family.[provided by RefSeq, Jul 2010].'},\n",
       " {'id': 'HGNC:6879',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MAPK6',\n",
       "  'name': 'mitogen-activated protein kinase 6',\n",
       "  'synonyms': 'ERK3|HsT17250|PRKM6|p97MAPK',\n",
       "  'description': 'The protein encoded by this gene is a member of the Ser/Thr protein kinase family, and is most closely related to mitogen-activated protein kinases (MAP kinases). MAP kinases also known as extracellular signal-regulated kinases (ERKs), are activated through protein phosphorylation cascades and act as integration points for multiple biochemical signals. This kinase is localized in the nucleus, and has been reported to be activated in fibroblasts upon treatment with serum or phorbol esters. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:6846',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MAP2K6',\n",
       "  'name': 'mitogen-activated protein kinase kinase 6',\n",
       "  'synonyms': 'MAPKK6|MEK6|MKK6|PRKMK6|SAPKK-3|SAPKK3',\n",
       "  'description': 'This gene encodes a member of the dual specificity protein kinase family, which functions as a mitogen-activated protein (MAP) kinase kinase. MAP kinases, also known as extracellular signal-regulated kinases (ERKs), act as an integration point for multiple biochemical signals. This protein phosphorylates and activates p38 MAP kinase in response to inflammatory cytokines or environmental stress. As an essential component of p38 MAP kinase mediated signal transduction pathway, this gene is involved in many cellular processes such as stress induced cell cycle arrest, transcription activation and apoptosis. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:9533',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PSMA4',\n",
       "  'name': 'proteasome 20S subunit alpha 4',\n",
       "  'synonyms': 'HC9|HsT17706|PSC9',\n",
       "  'description': 'This gene encodes a core alpha subunit of the 20S proteosome, which is a highly ordered ring-shaped structure composed of four rings of 28 non-identical subunits. Proteasomes cleave peptides in an ATP- and ubiquitin-dependent manner. [provided by RefSeq, Aug 2016].'},\n",
       " {'id': 'HGNC:9553',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PSMC6',\n",
       "  'name': 'proteasome 26S subunit, ATPase 6',\n",
       "  'synonyms': 'RPT5|SUG2|p42',\n",
       "  'description': 'The 26S proteasome is a multicatalytic proteinase complex with a highly ordered structure composed of 2 complexes, a 20S core and a 19S regulator. The 20S core is composed of 4 rings of 28 non-identical subunits; 2 rings are composed of 7 alpha subunits and 2 rings are composed of 7 beta subunits. The 19S regulator is composed of a base, which contains 6 ATPase subunits and 2 non-ATPase subunits, and a lid, which contains up to 10 non-ATPase subunits. Proteasomes are distributed throughout eukaryotic cells at a high concentration and cleave peptides in an ATP/ubiquitin-dependent process in a non-lysosomal pathway.  An essential function of a modified proteasome, the immunoproteasome, is the processing of class I MHC peptides. This gene encodes one of the ATPase subunits, a member of the triple-A family of ATPases which have a chaperone-like activity. Pseudogenes have been identified on chromosomes 8 and 12. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:9594',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PTGER2',\n",
       "  'name': 'prostaglandin E receptor 2',\n",
       "  'synonyms': 'EP2',\n",
       "  'description': 'This gene encodes a receptor for prostaglandin E2, a metabolite of arachidonic acid which has different biologic activities in a wide range of tissues. Mutations in this gene are associated with aspirin-induced susceptibility to asthma. [provided by RefSeq, Oct 2009].'},\n",
       " {'id': 'HGNC:9658',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PTPN6',\n",
       "  'name': 'protein tyrosine phosphatase non-receptor type 6',\n",
       "  'synonyms': 'HCP|HCPH|HPTP1C|PTP-1C|SH-PTP1|SHP-1|SHP-1L|SHP1',\n",
       "  'description': 'The protein encoded by this gene is a member of the protein tyrosine phosphatase (PTP) family. PTPs are known to be signaling molecules that regulate a variety of cellular processes including cell growth, differentiation, mitotic cycle, and oncogenic transformation. N-terminal part of this PTP contains two tandem Src homolog (SH2) domains, which act as protein phospho-tyrosine binding domains, and mediate the interaction of this PTP with its substrates. This PTP is expressed primarily in hematopoietic cells, and functions as an important regulator of multiple signaling pathways in hematopoietic cells. This PTP has been shown to interact with, and dephosphorylate a wide spectrum of phospho-proteins involved in hematopoietic cell signaling. Multiple alternatively spliced variants of this gene, which encode distinct isoforms, have been reported. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:9659',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PTPN7',\n",
       "  'name': 'protein tyrosine phosphatase non-receptor type 7',\n",
       "  'synonyms': 'BPTP-4|HEPTP|LC-PTP|LPTP|PTPNI',\n",
       "  'description': 'The protein encoded by this gene is a member of the protein tyrosine phosphatase (PTP) family. PTPs are known to be signaling molecules that regulate a variety of cellular processes including cell growth, differentiation, mitotic cycle, and oncogenic transformation. This gene is preferentially expressed in a variety of hematopoietic cells, and is an early response gene in lymphokine stimulated cells. The non-catalytic N-terminus of this PTP can interact with MAP kinases and suppress the MAP kinase activities. This PTP was shown to be involved in the regulation of T cell antigen receptor (TCR) signaling, which was thought to function through dephosphorylating the molecules related to MAP kinase pathway. Multiple alternatively spliced transcript variants have been found for this gene. [provided by RefSeq, Dec 2010].'},\n",
       " {'id': 'HGNC:9666',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PTPRC',\n",
       "  'name': 'protein tyrosine phosphatase receptor type C',\n",
       "  'synonyms': 'B220|CD45|CD45R|GP180|L-CA|LCA|LY5|T200',\n",
       "  'description': 'The protein encoded by this gene is a member of the protein tyrosine phosphatase (PTP) family. PTPs are known to be signaling molecules that regulate a variety of cellular processes including cell growth, differentiation, mitosis, and oncogenic transformation. This PTP contains an extracellular domain, a single transmembrane segment and two tandem intracytoplasmic catalytic domains, and thus is classified as a receptor type PTP. This PTP has been shown to be an essential regulator of T- and B-cell antigen receptor signaling. It functions through either direct interaction with components of the antigen receptor complexes, or by activating various Src family kinases required for the antigen receptor signaling. This PTP also suppresses JAK kinases, and thus functions as a regulator of cytokine receptor signaling. Alternatively spliced transcripts variants of this gene, which encode distinct isoforms, have been reported. [provided by RefSeq, Jun 2012].'},\n",
       " {'id': 'HGNC:9711',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PWP2',\n",
       "  'name': 'PWP2 small subunit processome component',\n",
       "  'synonyms': 'EHOC-17|PWP2H|UTP1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:9719',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PEX5',\n",
       "  'name': 'peroxisomal biogenesis factor 5',\n",
       "  'synonyms': 'PBD2A|PBD2B|PTS1-BP|PTS1R|PXR1|RCDP5',\n",
       "  'description': 'The product of this gene binds to the C-terminal PTS1-type tripeptide peroxisomal targeting signal (SKL-type) and plays an essential role in peroxisomal protein import. Peroxins (PEXs) are proteins that are essential for the assembly of functional peroxisomes. The peroxisome biogenesis disorders (PBDs) are a group of genetically heterogeneous autosomal recessive, lethal diseases characterized by multiple defects in peroxisome function. The peroxisomal biogenesis disorders are a heterogeneous group with at least 14 complementation groups and with more than 1 phenotype being observed in cases falling into particular complementation groups. Although the clinical features of PBD patients vary, cells from all PBD patients exhibit a defect in the import of one or more classes of peroxisomal matrix proteins into the organelle. Defects in this gene are a cause of neonatal adrenoleukodystrophy (NALD), a cause of Zellweger syndrome (ZWS) as well as may be a cause of infantile Refsum disease (IRD). Alternatively spliced transcript variants encoding different isoforms have been identified. [provided by RefSeq, Oct 2008].'},\n",
       " {'id': 'HGNC:9817',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RAD51',\n",
       "  'name': 'RAD51 recombinase',\n",
       "  'synonyms': 'BRCC5|FANCR|HRAD51|HsRad51|HsT16930|MRMV2|RAD51A|RECA',\n",
       "  'description': 'The protein encoded by this gene is a member of the RAD51 protein family. RAD51 family members are highly similar to bacterial RecA and Saccharomyces cerevisiae Rad51, and are known to be involved in the homologous recombination and repair of DNA. This protein can interact with the ssDNA-binding protein RPA and RAD52, and it is thought to play roles in homologous pairing and strand transfer of DNA. This protein is also found to interact with BRCA1 and BRCA2, which may be important for the cellular response to DNA damage. BRCA2 is shown to regulate both the intracellular localization and DNA-binding ability of this protein. Loss of these controls following BRCA2 inactivation may be a key event leading to genomic instability and tumorigenesis. Multiple transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Aug 2009].'},\n",
       " {'id': 'HGNC:9824',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RAD52',\n",
       "  'name': 'RAD52 homolog, DNA repair protein',\n",
       "  'synonyms': nan,\n",
       "  'description': 'The protein encoded by this gene shares similarity with Saccharomyces cerevisiae Rad52, a protein important for DNA double-strand break repair and homologous recombination. This gene product was shown to bind single-stranded DNA ends, and mediate the DNA-DNA interaction necessary for the annealing of complementary DNA strands. It was also found to interact with DNA recombination protein RAD51, which suggested its role in RAD51 related DNA recombination and repair. A pseudogene of this gene is present on chromosome 2. Alternative splicing results in multiple transcript variants. Additional alternatively spliced transcript variants of this gene have been described, but their full-length nature is not known. [provided by RefSeq, Jul 2014].'},\n",
       " {'id': 'HGNC:9842',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RALGDS',\n",
       "  'name': 'ral guanine nucleotide dissociation stimulator',\n",
       "  'synonyms': 'RGDS|RGF|RalGEF',\n",
       "  'description': 'Guanine nucleotide dissociation stimulators (GDSs, or exchange factors), such as RALGDS, are effectors of Ras-related GTPases (see MIM 190020) that participate in signaling for a variety of cellular processes.[supplied by OMIM, Nov 2010].'},\n",
       " {'id': 'HGNC:9884',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RB1',\n",
       "  'name': 'RB transcriptional corepressor 1',\n",
       "  'synonyms': 'OSRC|PPP1R130|RB|p105-Rb|p110-RB1|pRb|pp110',\n",
       "  'description': 'The protein encoded by this gene is a negative regulator of the cell cycle and was the first tumor suppressor gene found. The encoded protein also stabilizes constitutive heterochromatin to maintain the overall chromatin structure. The active, hypophosphorylated form of the protein binds transcription factor E2F1. Defects in this gene are a cause of childhood cancer retinoblastoma (RB), bladder cancer, and osteogenic sarcoma. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:9893',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RBL1',\n",
       "  'name': 'RB transcriptional corepressor like 1',\n",
       "  'synonyms': 'CP107|PRB1|p107',\n",
       "  'description': 'The protein encoded by this gene is similar in sequence and possibly function to the product of the retinoblastoma 1 (RB1) gene. The RB1 gene product is a tumor suppressor protein that appears to be involved in cell cycle regulation, as it is phosphorylated in the S to M phase transition and is dephosphorylated in the G1 phase of the cell cycle. Both the RB1 protein and the product of this gene can form a complex with adenovirus E1A protein and SV40 large T-antigen, with the SV40 large T-antigen binding only to the unphosphorylated form of each protein. In addition, both proteins can inhibit the transcription of cell cycle genes containing E2F binding sites in their promoters. Due to the sequence and biochemical similarities with the RB1 protein, it is thought that the protein encoded by this gene may also be a tumor suppressor. Two transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:9900',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RBM3',\n",
       "  'name': 'RNA binding motif protein 3',\n",
       "  'synonyms': 'IS1-RNPL|RNPL',\n",
       "  'description': 'This gene is a member of the glycine-rich RNA-binding protein family and encodes a protein with one RNA recognition motif (RRM) domain. Expression of this gene is induced by cold shock and low oxygen tension. A pseudogene exists on chromosome 1. Multiple alternatively spliced transcript variants that are predicted to encode different isoforms have been characterized although some of these variants fit nonsense-mediated decay (NMD) criteria. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:9921',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RBP3',\n",
       "  'name': 'retinol binding protein 3',\n",
       "  'synonyms': 'D10S64|D10S65|D10S66|IRBP|RBPI|RP66',\n",
       "  'description': 'Interphotoreceptor retinol-binding protein is a large glycoprotein known to bind retinoids and found primarily in the interphotoreceptor matrix of the retina between the retinal pigment epithelium and the photoreceptor cells. It is thought to transport retinoids between the retinal pigment epithelium and the photoreceptors, a critical role in the visual process.The human IRBP gene is approximately 9.5 kbp in length and consists of four exons separated by three introns. The introns are 1.6-1.9 kbp long. The gene is transcribed by photoreceptor and retinoblastoma cells into an approximately 4.3-kilobase mRNA that is translated and processed into a glycosylated protein of 135,000 Da. The amino acid sequence of human IRBP can be divided into four contiguous homology domains with 33-38% identity, suggesting a series of gene duplication events. In the gene, the boundaries of these domains are not defined by exon-intron junctions, as might have been expected. The first three homology domains and part of the fourth are all encoded by the first large exon, which is 3,180 base pairs long. The remainder of the fourth domain is encoded in the last three exons, which are 191, 143, and approximately 740 base pairs long, respectively. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:9942',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PRPH2',\n",
       "  'name': 'peripherin 2',\n",
       "  'synonyms': 'AOFMD|AVMD|CACD2|DS|MDBS1|PRPH|RDS|RP7|TSPAN22|rd2',\n",
       "  'description': 'The protein encoded by this gene is a member of the transmembrane 4 superfamily, also known as the tetraspanin family. Most of these members are cell-surface proteins that are characterized by the presence of four hydrophobic domains. The proteins mediate signal transduction events that play a role in the regulation of cell development, activation, growth and motility. This encoded protein is a cell surface glycoprotein found in the outer segment of both rod and cone photoreceptor cells. It may function as an adhesion molecule involved in stabilization and compaction of outer segment disks or in the maintenance of the curvature of the rim. This protein is essential for disk morphogenesis. Defects in this gene are associated with both central and peripheral retinal degenerations. Some of the various phenotypically different disorders are autosomal dominant retinitis pigmentosa, progressive macular degeneration, macular dystrophy and retinitis pigmentosa digenic. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:9964',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DPF2',\n",
       "  'name': 'double PHD fingers 2',\n",
       "  'synonyms': 'CSS7|REQ|UBID4|ubi-d4',\n",
       "  'description': 'The protein encoded by this gene is a member of the d4 domain family, characterized by a zinc finger-like structural motif. This protein functions as a transcription factor which is necessary for the apoptotic response following deprivation of survival factors. It likely serves a regulatory role in rapid hematopoietic cell growth and turnover. This gene is considered a candidate gene for multiple endocrine neoplasia type I, an inherited cancer syndrome involving multiple parathyroid, enteropancreatic, and pituitary tumors. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:9970',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RFC2',\n",
       "  'name': 'replication factor C subunit 2',\n",
       "  'synonyms': 'RFC40',\n",
       "  'description': 'This gene encodes a member of the activator 1 small subunits family. The elongation of primed DNA templates by DNA polymerase delta and epsilon requires the action of the accessory proteins, proliferating cell nuclear antigen (PCNA) and replication factor C (RFC). Replication factor C, also called activator 1, is a protein complex consisting of five distinct subunits. This gene encodes the 40 kD subunit, which has been shown to be responsible for binding ATP and may help promote cell survival. Disruption of this gene is associated with Williams syndrome. Alternatively spliced transcript variants encoding distinct isoforms have been described. A pseudogene of this gene has been defined on chromosome 2. [provided by RefSeq, Jul 2013].'},\n",
       " {'id': 'HGNC:9971',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RFC3',\n",
       "  'name': 'replication factor C subunit 3',\n",
       "  'synonyms': 'RFC38',\n",
       "  'description': 'The elongation of primed DNA templates by DNA polymerase delta and DNA polymerase epsilon requires the accessory proteins proliferating cell nuclear antigen (PCNA) and replication factor C (RFC). RFC, also named activator 1, is a protein complex consisting of five distinct subunits of 140, 40, 38, 37, and 36 kDa. This gene encodes the 38 kDa subunit. This subunit is essential for the interaction between the 140 kDa subunit and the core complex that consists of the 36, 37, and 40 kDa subunits. Alternatively spliced transcript variants encoding distinct isoforms have been described. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:9972',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RFC4',\n",
       "  'name': 'replication factor C subunit 4',\n",
       "  'synonyms': 'A1|RFC37',\n",
       "  'description': 'The elongation of primed DNA templates by DNA polymerase delta and DNA polymerase epsilon requires the accessory proteins proliferating cell nuclear antigen (PCNA) and replication factor C (RFC). RFC, also named activator 1, is a protein complex consisting of five distinct subunits of 140, 40, 38, 37, and 36 kD. This gene encodes the 37 kD subunit. This subunit forms a core complex with the 36 and 40 kDa subunits. The core complex possesses DNA-dependent ATPase activity, which was found to be stimulated by PCNA in an in vitro system. Alternatively spliced transcript variants encoding the same protein have been reported. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:1103',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'BRD2',\n",
       "  'name': 'bromodomain containing 2',\n",
       "  'synonyms': 'BRD2-IT1|D6S113E|FSH|FSRG1|NAT|O27.1.1|RING3|RNF3',\n",
       "  'description': 'This gene encodes a transcriptional regulator that belongs to the BET (bromodomains and extra terminal domain) family of proteins. This protein associates with transcription complexes and with acetylated chromatin during mitosis, and it selectively binds to the acetylated lysine-12 residue of histone H4 via its two bromodomains. The gene maps to the major histocompatability complex (MHC) class II region on chromosome 6p21.3, but sequence comparison suggests that the protein is not involved in the immune response. This gene has been implicated in juvenile myoclonic epilepsy, a common form of epilepsy that becomes apparent in adolescence. Multiple alternatively spliced variants have been described for this gene. [provided by RefSeq, Dec 2010].'},\n",
       " {'id': 'HGNC:10291',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RPA3',\n",
       "  'name': 'replication protein A3',\n",
       "  'synonyms': 'REPA3|RP-A p14',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:10334',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RPL31',\n",
       "  'name': 'ribosomal protein L31',\n",
       "  'synonyms': 'L31',\n",
       "  'description': 'Ribosomes, the organelles that catalyze protein synthesis, consist of a small 40S subunit and a large 60S subunit. Together these subunits are composed of 4 RNA species and approximately 80 structurally distinct proteins. This gene encodes a ribosomal protein that is a component of the 60S subunit. The protein belongs to the L31E family of ribosomal proteins. It is located in the cytoplasm. Higher levels of expression of this gene in familial adenomatous polyps compared to matched normal tissues have been observed. As is typical for genes encoding ribosomal proteins, there are multiple processed pseudogenes of this gene dispersed through the genome. Alternatively spliced transcript variants encoding distinct isoforms have been found for this gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:10436',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RPS6KB1',\n",
       "  'name': 'ribosomal protein S6 kinase B1',\n",
       "  'synonyms': 'PS6K|S6K|S6K-beta-1|S6K1|STK14A|p70 S6KA|p70(S6K)-alpha|p70-S6K|p70-alpha',\n",
       "  'description': 'This gene encodes a member of the ribosomal S6 kinase family of serine/threonine kinases. The encoded protein responds to mTOR (mammalian target of rapamycin) signaling to promote protein synthesis, cell growth, and cell proliferation. Activity of this gene has been associated with human cancer. Alternatively spliced transcript variants have been observed. The use of alternative translation start sites results in isoforms with longer or shorter N-termini which may differ in their subcellular localizations. There are two pseudogenes for this gene on chromosome 17. [provided by RefSeq, Jan 2013].'},\n",
       " {'id': 'HGNC:10451',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RRM1',\n",
       "  'name': 'ribonucleotide reductase catalytic subunit M1',\n",
       "  'synonyms': 'R1|RIR1|RR1',\n",
       "  'description': 'This gene encodes the large and catalytic subunit of ribonucleotide reductase, an enzyme essential for the conversion of ribonucleotides into deoxyribonucleotides. A pool of available deoxyribonucleotides is important for DNA replication during S phase of the cell cycle as well as multiple DNA repair processes. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Dec 2015].'},\n",
       " {'id': 'HGNC:10461',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CLIP1',\n",
       "  'name': 'CAP-Gly domain containing linker protein 1',\n",
       "  'synonyms': 'CLIP|CLIP-170|CLIP170|CYLN1|RSN',\n",
       "  'description': 'The protein encoded by this gene links endocytic vesicles to microtubules. This gene is highly expressed in Reed-Sternberg cells of Hodgkin disease. Several transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Oct 2011].'},\n",
       " {'id': 'HGNC:10468',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RTN2',\n",
       "  'name': 'reticulon 2',\n",
       "  'synonyms': 'NSP2|NSPL1|NSPLI|SPG12',\n",
       "  'description': 'This gene belongs to the family of reticulon encoding genes. Reticulons are associated with the endoplasmic reticulum, and are involved in neuroendocrine secretion or in membrane trafficking in neuroendocrine cells. Reticulon proteins also play an important role in the replication of positive-strand RNA (ssRNA) viruses. Mutations at this locus have been associated with autosomal dominant spastic paraplegia-12. [provided by RefSeq, Aug 2020].'},\n",
       " {'id': 'HGNC:10479',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RXRG',\n",
       "  'name': 'retinoid X receptor gamma',\n",
       "  'synonyms': 'NR2B3|RXRC',\n",
       "  'description': 'This gene encodes a member of the retinoid X receptor (RXR) family of nuclear receptors which are involved in mediating the antiproliferative effects of retinoic acid (RA). This receptor forms dimers with the retinoic acid, thyroid hormone, and vitamin D receptors, increasing both DNA binding and transcriptional function on their respective response elements. This gene is expressed at significantly lower levels in non-small cell lung cancer cells. Alternatively spliced transcript variants have been described. [provided by RefSeq, Jun 2010].'},\n",
       " {'id': 'HGNC:10514',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SAA2',\n",
       "  'name': 'serum amyloid A2',\n",
       "  'synonyms': 'SAA|SAA1',\n",
       "  'description': \"This gene encodes a member of the serum amyloid A family of apolipoproteins. The encoded preproprotein is proteolytically processed to generate the mature protein. This protein is a major acute phase protein that is highly expressed in response to inflammation and tissue injury. This protein also plays an important role in HDL metabolism and cholesterol homeostasis. High levels of this protein are associated with chronic inflammatory diseases including atherosclerosis, rheumatoid arthritis, Alzheimer's disease and Crohn's disease. This protein may also be a potential biomarker for certain tumors. Finally, antimicrobial activity against S. aureus and E. coli resides in the N-terminal portion of the mature protein. [provided by RefSeq, Jul 2020].\"},\n",
       " {'id': 'HGNC:10615',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CCL17',\n",
       "  'name': 'C-C motif chemokine ligand 17',\n",
       "  'synonyms': 'A-152E5.3|ABCD-2|SCYA17|TARC',\n",
       "  'description': 'This antimicrobial gene is one of several Cys-Cys (CC) cytokine genes clustered on the q arm of chromosome 16. Cytokines are a family of secreted proteins involved in immunoregulatory and inflammatory processes. The CC cytokines are proteins characterized by two adjacent cysteines. The cytokine encoded by this gene displays chemotactic activity for T lymphocytes, but not monocytes or granulocytes. The product of this gene binds to chemokine receptors CCR4 and CCR8. This chemokine plays important roles in T cell development in thymus as well as in trafficking and activation of mature T cells. [provided by RefSeq, Sep 2014].'},\n",
       " {'id': 'HGNC:10647',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CX3CL1',\n",
       "  'name': 'C-X3-C motif chemokine ligand 1',\n",
       "  'synonyms': 'ABCD-3|C3Xkine|CXC3|CXC3C|NTN|NTT|SCYD1|fractalkine|neurotactin',\n",
       "  'description': 'This gene belongs to the CX3C subgroup of chemokines, characterized by the number of amino acids located between the conserved cysteine residues. This is the only member of the CX3C subgroup, which contains three amino acids between cysteine residues, resulting in a Cys-X-X-X-Cys configuration. The encoded protein contains an extended mucin-like stalk with a chemokine domain on top, and exists in both a membrane-anchored form where it acts as a binding molecule, or, in soluble form, as a chemotactic cytokine. The mature form of this protein can be cleaved at the cell surface, yielding different soluble forms that can interact with the G-protein coupled receptor, C-X3-C motif chemokine receptor 1 gene product. This gene plays a role in a wide range of diseases, including cancer, vasculitis, neuropathies, atherosclerosis, inflammatory diseases, and in human immunodeficiency virus infections. [provided by RefSeq, Sep 2017].'},\n",
       " {'id': 'HGNC:10818',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SGSH',\n",
       "  'name': 'N-sulfoglucosamine sulfohydrolase',\n",
       "  'synonyms': 'HSS|MPS3A|SFMD',\n",
       "  'description': 'This gene encodes the enzyme sulfamidase; one of several enzymes involved in the lysosomal degradation of heparan sulfate. Mutations in this gene are associated with the lysosomal storage disease mucopolysaccaridosis IIIA, also known as Sanfilippo syndrome A, which results from impaired degradation of heparan sulfate. Transcripts of varying sizes have been reported but their biological validity has not been determined. [provided by RefSeq, Jun 2017].'},\n",
       " {'id': 'HGNC:11007',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SLC2A3',\n",
       "  'name': 'solute carrier family 2 member 3',\n",
       "  'synonyms': 'GLUT3',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:11026',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SLC3A2',\n",
       "  'name': 'solute carrier family 3 member 2',\n",
       "  'synonyms': '4F2|4F2HC|4T2HC|CD98|CD98HC|MDU1|NACAE',\n",
       "  'description': 'This gene is a member of the solute carrier family and encodes a cell surface, transmembrane protein. The protein exists as the heavy chain of a heterodimer, covalently bound through di-sulfide bonds to one of several possible light chains. The encoded transporter plays a role in regulation of intracellular calcium levels and transports L-type amino acids. Alternatively spliced transcript variants, encoding different isoforms, have been characterized. [provided by RefSeq, Nov 2010].'},\n",
       " {'id': 'HGNC:10920',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SLC15A1',\n",
       "  'name': 'solute carrier family 15 member 1',\n",
       "  'synonyms': 'HPECT1|HPEPT1|PEPT1',\n",
       "  'description': 'This gene encodes an intestinal hydrogen peptide cotransporter that is a member of the solute carrier family 15. The encoded protein is localized to the brush border membrane of the intestinal epithelium and mediates the uptake of di- and tripeptides from the lumen into the enterocytes. This protein plays an important role in the uptake and digestion of dietary proteins. This protein also facilitates the absorption of numerous peptidomimetic drugs. [provided by RefSeq, Apr 2010].'},\n",
       " {'id': 'HGNC:11099',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HLTF',\n",
       "  'name': 'helicase like transcription factor',\n",
       "  'synonyms': 'HIP116|HIP116A|HLTF1|RNF80|SMARCA3|SNF2L3|ZBU1',\n",
       "  'description': 'This gene encodes a member of the SWI/SNF family. Members of this family have helicase and ATPase activities and are thought to regulate transcription of certain genes by altering the chromatin structure around those genes. The encoded protein contains a RING finger DNA binding motif. Two transcript variants encoding the same protein have been found for this gene. However, use of an alternative translation start site produces an isoform that is truncated at the N-terminus compared to the full-length protein. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:11169',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SNTB2',\n",
       "  'name': 'syntrophin beta 2',\n",
       "  'synonyms': 'D16S2531E|EST25263|SNT2B2|SNT3|SNTL',\n",
       "  'description': 'Dystrophin is a large, rod-like cytoskeletal protein found at the inner surface of muscle fibers. Dystrophin is missing in Duchenne Muscular Dystrophy patients and is present in reduced amounts in Becker Muscular Dystrophy patients. The protein encoded by this gene is a peripheral membrane protein found associated with dystrophin and dystrophin-related proteins. This gene is a member of the syntrophin gene family, which contains at least two other structurally-related genes. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:11251',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SPOCK1',\n",
       "  'name': 'SPARC (osteonectin), cwcv and kazal like domains proteoglycan 1',\n",
       "  'synonyms': 'SPOCK|TESTICAN|TIC1',\n",
       "  'description': \"This gene encodes the protein core of a seminal plasma proteoglycan containing chondroitin- and heparan-sulfate chains. The protein's function is unknown, although similarity to thyropin-type cysteine protease-inhibitors suggests its function may be related to protease inhibition. [provided by RefSeq, Jul 2008].\"},\n",
       " {'id': 'HGNC:11319',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ITPRID2',\n",
       "  'name': 'ITPR interacting domain containing 2',\n",
       "  'synonyms': 'CS-1|CS1|KRAP|SPAG13|SSFA2',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:11326',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SSR4',\n",
       "  'name': 'signal sequence receptor subunit 4',\n",
       "  'synonyms': 'CDG1Y|TRAPD',\n",
       "  'description': 'This gene encodes the delta subunit of the translocon-associated protein complex which is involved in translocating proteins across the endoplasmic reticulum membrane. The encoded protein is located in the Xq28 region and is arranged in a compact head-to-head manner with the isocitrate dehydrogenase 3 (NAD+) gamma gene and both genes are driven by a CpG-embedded bidirectional promoter. Alternate splicing results in multiple transcript variants.[provided by RefSeq, Mar 2011].'},\n",
       " {'id': 'HGNC:11330',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SSTR1',\n",
       "  'name': 'somatostatin receptor 1',\n",
       "  'synonyms': 'SRIF-2|SS-1-R|SS1-R|SS1R',\n",
       "  'description': 'Somatostatins are peptide hormones that regulate diverse cellular functions such as neurotransmission, cell proliferation, and endocrine signaling as well as inhibiting the release of many hormones and other secretory proteins. Somatostatin has two active forms of 14 and 28 amino acids. The biological effects of somatostatins are mediated by a family of G-protein coupled somatostatin receptors that are expressed in a tissue-specific manner. The protein encoded by this gene is a member of the superfamily of somatostatin receptors having seven transmembrane segments. Somatostatin receptors form homodimers and heterodimers with other members of the superfamily as well as with other G-protein coupled receptors and receptor tyrosine kinases. This somatostatin receptor has greater affinity for somatostatin-14 than -28. [provided by RefSeq, Jul 2012].'},\n",
       " {'id': 'HGNC:11343',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ST13',\n",
       "  'name': 'ST13 Hsp70 interacting protein',\n",
       "  'synonyms': 'AAG2|FAM10A1|FAM10A4|HIP|HOP|HSPABP|HSPABP1|P48|PRO0786|SNC6',\n",
       "  'description': 'The protein encoded by this gene is an adaptor protein that mediates the association of the heat shock proteins HSP70 and HSP90. This protein has been shown to be involved in the assembly process of glucocorticoid receptor, which requires the assistance of multiple molecular chaperones. The expression of this gene is reported to be downregulated in colorectal carcinoma tissue suggesting that it is a candidate tumor suppressor gene. Alternative splicing results in multiple transcript variants encoding different isoforms. [provided by RefSeq, Jun 2013].'},\n",
       " {'id': 'HGNC:11399',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NEK4',\n",
       "  'name': 'NIMA related kinase 4',\n",
       "  'synonyms': 'NRK2|STK2|pp12301',\n",
       "  'description': 'The protein encoded by this gene is a serine/threonine protein kinase required for normal entry into replicative senescence. The encoded protein also is involved in cell cycle arrest in response to double-stranded DNA damage. Finally, this protein plays a role in maintaining cilium integrity, and defects in this gene have been associated with ciliopathies. [provided by RefSeq, Jan 2017].'},\n",
       " {'id': 'HGNC:11408',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'STK4',\n",
       "  'name': 'serine/threonine kinase 4',\n",
       "  'synonyms': 'KRS2|MST1|YSK3',\n",
       "  'description': \"The protein encoded by this gene is a cytoplasmic kinase that is structurally similar to the yeast Ste20p kinase, which acts upstream of the stress-induced mitogen-activated protein kinase cascade. The encoded protein can phosphorylate myelin basic protein and undergoes autophosphorylation. A caspase-cleaved fragment of the encoded protein has been shown to be capable of phosphorylating histone H2B. The particular phosphorylation catalyzed by this protein has been correlated with apoptosis, and it's possible that this protein induces the chromatin condensation observed in this process. [provided by RefSeq, Jul 2008].\"},\n",
       " {'id': 'HGNC:11454',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SULT1A2',\n",
       "  'name': 'sulfotransferase family 1A member 2',\n",
       "  'synonyms': 'HAST4|P-PST|P-PST 2|ST1A2|STP2|TSPST2',\n",
       "  'description': 'Sulfotransferase enzymes catalyze the sulfate conjugation of many hormones, neurotransmitters, drugs, and xenobiotic compounds. These cytosolic enzymes are different in their tissue distributions and substrate specificities. The gene structure (number and length of exons) is similar among family members. This gene encodes one of two phenol sulfotransferases with thermostable enzyme activity. Two alternatively spliced variants that encode the same protein have been described. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:11438',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'STX3',\n",
       "  'name': 'syntaxin 3',\n",
       "  'synonyms': 'STX3A',\n",
       "  'description': 'The gene is a member of the syntaxin family. The encoded protein is targeted to the apical membrane of epithelial cells where it forms clusters and is important in establishing and maintaining polarity necessary for protein trafficking involving vesicle fusion and exocytosis. Alternative splicing results in multiple transcript variants. [provided by RefSeq, May 2010].'},\n",
       " {'id': 'HGNC:11540',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TAF6',\n",
       "  'name': 'TATA-box binding protein associated factor 6',\n",
       "  'synonyms': 'ALYUS|MGC:8964|TAF(II)70|TAF(II)80|TAF2E|TAFII-70|TAFII-80|TAFII70|TAFII80|TAFII85',\n",
       "  'description': 'Initiation of transcription by RNA polymerase II requires the activities of more than 70 polypeptides. The protein that coordinates these activities is transcription factor IID (TFIID), which binds to the core promoter to position the polymerase properly, serves as the scaffold for assembly of the remainder of the transcription complex, and acts as a channel for regulatory signals. TFIID is composed of the TATA-binding protein (TBP) and a group of evolutionarily conserved proteins known as TBP-associated factors or TAFs. TAFs may participate in basal transcription, serve as coactivators, function in promoter recognition or modify general transcription factors (GTFs) to facilitate complex assembly and transcription initiation. This gene encodes one of the smaller subunits of TFIID that binds weakly to TBP but strongly to TAF1, the largest subunit of TFIID. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Jun 2010].'},\n",
       " {'id': 'HGNC:11544',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TAF11',\n",
       "  'name': 'TATA-box binding protein associated factor 11',\n",
       "  'synonyms': 'MGC:15243|PRO2134|TAF2I|TAFII28',\n",
       "  'description': 'Initiation of transcription by RNA polymerase II requires the activities of more than 70 polypeptides. The protein that coordinates these activities is transcription factor IID (TFIID), which binds to the core promoter to position the polymerase properly, serves as the scaffold for assembly of the remainder of the transcription complex, and acts as a channel for regulatory signals. TFIID is composed of the TATA-binding protein (TBP) and a group of evolutionarily conserved proteins known as TBP-associated factors or TAFs. TAFs may participate in basal transcription, serve as coactivators, function in promoter recognition or modify general transcription factors (GTFs) to facilitate complex assembly and transcription initiation. This gene encodes a small subunit of TFIID that is present in all TFIID complexes and interacts with TBP. This subunit also interacts with another small subunit, TAF13, to form a heterodimer with a structure similar to the histone core structure. Two transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Jul 2012].'},\n",
       " {'id': 'HGNC:11580',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TBCC',\n",
       "  'name': 'tubulin folding cofactor C',\n",
       "  'synonyms': 'CFC',\n",
       "  'description': 'Cofactor C is one of four proteins (cofactors A, D, E, and C) involved in the pathway leading to correctly folded beta-tubulin from folding intermediates. Cofactors A and D are believed to play a role in capturing and stabilizing beta-tubulin intermediates in a quasi-native confirmation. Cofactor E binds to the cofactor D/beta-tubulin complex; interaction with cofactor C then causes the release of beta-tubulin polypeptides that are committed to the native state. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:11768',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TGFB2',\n",
       "  'name': 'transforming growth factor beta 2',\n",
       "  'synonyms': 'G-TSF|LDS4|TGF-beta2',\n",
       "  'description': \"This gene encodes a secreted ligand of the TGF-beta (transforming growth factor-beta) superfamily of proteins. Ligands of this family bind various TGF-beta receptors leading to recruitment and activation of SMAD family transcription factors that regulate gene expression. The encoded preproprotein is proteolytically processed to generate a latency-associated peptide (LAP) and a mature peptide, and is found in either a latent form composed of a mature peptide homodimer, a LAP homodimer, and a latent TGF-beta binding protein, or in an active form consisting solely of the mature peptide homodimer. The mature peptide may also form heterodimers with other TGF-beta family members. Disruption of the TGF-beta/SMAD pathway has been implicated in a variety of human cancers. A chromosomal translocation that includes this gene is associated with Peters' anomaly, a congenital defect of the anterior chamber of the eye. Mutations in this gene may be associated with Loeys-Dietz syndrome. This gene encodes multiple isoforms that may undergo similar proteolytic processing. [provided by RefSeq, Aug 2016].\"},\n",
       " {'id': 'HGNC:11810',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'KLF10',\n",
       "  'name': 'Kruppel like factor 10',\n",
       "  'synonyms': 'EGR-alpha|EGRA|TIEG|TIEG1',\n",
       "  'description': 'This gene encodes a member of a family of proteins that feature C2H2-type zinc finger domains. The encoded protein is a transcriptional repressor that acts as an effector of transforming growth factor beta signaling. Activity of this protein may inhibit the growth of cancers, particularly pancreatic cancer. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Jun 2013].'},\n",
       " {'id': 'HGNC:11820',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TIMP1',\n",
       "  'name': 'TIMP metallopeptidase inhibitor 1',\n",
       "  'synonyms': 'CLGI|EPA|EPO|HCI|TIMP|TIMP-1',\n",
       "  'description': 'This gene belongs to the TIMP gene family. The proteins encoded by this gene family are natural inhibitors of the matrix metalloproteinases (MMPs), a group of peptidases involved in degradation of the extracellular matrix. In addition to its inhibitory role against most of the known MMPs, the encoded protein is able to promote cell proliferation in a wide range of cell types, and may also have an anti-apoptotic function. Transcription of this gene is highly inducible in response to many cytokines and hormones. In addition, the expression from some but not all inactive X chromosomes suggests that this gene inactivation is polymorphic in human females. This gene is located within intron 6 of the synapsin I gene and is transcribed in the opposite direction. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:11830',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TK1',\n",
       "  'name': 'thymidine kinase 1',\n",
       "  'synonyms': 'TK2',\n",
       "  'description': 'The protein encoded by this gene is a cytosolic enzyme that catalyzes the addition of a gamma-phosphate group to thymidine. This creates dTMP and is the first step in the biosynthesis of dTTP, which is one component required for DNA replication. The encoded protein, whose levels fluctuate depending on the cell cycle stage, can act as a low activity dimer or a high activity tetramer. High levels of this protein have been used as a biomarker for diagnosing and categorizing many types of cancers. [provided by RefSeq, Oct 2016].'},\n",
       " {'id': 'HGNC:11858',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TSPAN6',\n",
       "  'name': 'tetraspanin 6',\n",
       "  'synonyms': 'T245|TM4SF6|TSPAN-6',\n",
       "  'description': 'The protein encoded by this gene is a member of the transmembrane 4 superfamily, also known as the tetraspanin family. Most of these members are cell-surface proteins that are characterized by the presence of four hydrophobic domains. The proteins mediate signal transduction events that play a role in the regulation of cell development, activation, growth and motility. The protein encoded by this gene is a cell surface glycoprotein and is highly similar in sequence to the transmembrane 4 superfamily member 2 protein. It functions as a negative regulator of retinoic acid-inducible gene I-like receptor-mediated immune signaling via its interaction with the mitochondrial antiviral signaling-centered signalosome. This gene uses alternative polyadenylation sites, and multiple transcript variants result from alternative splicing. [provided by RefSeq, Jul 2013].'},\n",
       " {'id': 'HGNC:2047',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CLDN5',\n",
       "  'name': 'claudin 5',\n",
       "  'synonyms': 'AWAL|BEC1|CPETRL1|TMDVCF|TMVCF',\n",
       "  'description': 'This gene encodes a member of the claudin family. Claudins are integral membrane proteins and components of tight junction strands. Tight junction strands serve as a physical barrier to prevent solutes and water from passing freely through the paracellular space between epithelial or endothelial cell sheets. Mutations in this gene have been found in patients with velocardiofacial syndrome. Alternative splicing results in multiple transcript variants encoding distinct isoforms. [provided by RefSeq, May 2018].'},\n",
       " {'id': 'HGNC:11986',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TOP1',\n",
       "  'name': 'DNA topoisomerase I',\n",
       "  'synonyms': 'TOPI',\n",
       "  'description': 'This gene encodes a DNA topoisomerase, an enzyme that controls and alters the topologic states of DNA during transcription. This enzyme catalyzes the transient breaking and rejoining of a single strand of DNA which allows the strands to pass through one another, thus altering the topology of DNA. This gene is localized to chromosome 20 and has pseudogenes which reside on chromosomes 1 and 22. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:11989',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TOP2A',\n",
       "  'name': 'DNA topoisomerase II alpha',\n",
       "  'synonyms': 'TOP2|TP2A',\n",
       "  'description': 'This gene encodes a DNA topoisomerase, an enzyme that controls and alters the topologic states of DNA during transcription. This nuclear enzyme is involved in processes such as chromosome condensation, chromatid separation, and the relief of torsional stress that occurs during DNA transcription and replication. It catalyzes the transient breaking and rejoining of two strands of duplex DNA which allows the strands to pass through one another, thus altering the topology of DNA. Two forms of this enzyme exist as likely products of a gene duplication event. The gene encoding this form, alpha, is localized to chromosome 17 and the beta gene is localized to chromosome 3. The gene encoding this enzyme functions as the target for several anticancer agents and a variety of mutations in this gene have been associated with the development of drug resistance. Reduced activity of this enzyme may also play a role in ataxia-telangiectasia. [provided by RefSeq, Jul 2010].'},\n",
       " {'id': 'HGNC:12339',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TRPM2',\n",
       "  'name': 'transient receptor potential cation channel subfamily M member 2',\n",
       "  'synonyms': 'EREG1|KNP3|LTRPC2|LTrpC-2|NUDT9H|NUDT9L1|TRPC7',\n",
       "  'description': 'The protein encoded by this gene forms a tetrameric cation channel that is permeable to calcium, sodium, and potassium and is regulated by free intracellular ADP-ribose. The encoded protein is activated by oxidative stress and confers susceptibility to cell death. Alternative splicing results in multiple transcript variants encoding distinct protein isoforms. Additional transcript variants of this gene have been described, but their full-length nature is not known. [provided by RefSeq, Feb 2016].'},\n",
       " {'id': 'HGNC:12362',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TSC1',\n",
       "  'name': 'TSC complex subunit 1',\n",
       "  'synonyms': 'LAM|TSC',\n",
       "  'description': 'This gene is a tumor suppressor gene that encodes the growth inhibitory protein hamartin. The encoded protein interacts with and stabilizes the GTPase activating protein tuberin. This hamartin-tuberin complex negatively regulates mammalian target of rapamycin complex 1 (mTORC1) signalling which is a major regulator of anabolic cell growth. This protein also functions as a co-chaperone for Hsp90 that inhibits its ATPase activity. This protein functions as a facilitator of Hsp90-mediated folding of kinase and non-kinase clients, including Tsc2 and thereby preventing their ubiquitination and proteasomal degradation. Mutations in this gene have been associated with tuberous sclerosis. [provided by RefSeq, Apr 2018].'},\n",
       " {'id': 'HGNC:12407',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TUBA4A',\n",
       "  'name': 'tubulin alpha 4a',\n",
       "  'synonyms': 'ALS22|H2-ALPHA|TUBA1',\n",
       "  'description': 'Microtubules of the eukaryotic cytoskeleton perform essential and diverse functions and are composed of a heterodimer of alpha and beta tubulin. The genes encoding these microtubule constituents are part of the tubulin superfamily, which is composed of six distinct families. Genes from the alpha, beta and gamma tubulin families are found in all eukaryotes. The alpha and beta tubulins represent the major components of microtubules, while gamma tubulin plays a critical role in the nucleation of microtubule assembly. There are multiple alpha and beta tubulin genes and they are highly conserved among and between species. This gene encodes an alpha tubulin that is a highly conserved homolog of a rat testis-specific alpha tubulin. Alternatively spliced transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Jun 2013].'},\n",
       " {'id': 'HGNC:12507',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'UBP1',\n",
       "  'name': 'upstream binding protein 1',\n",
       "  'synonyms': 'LBP-1B|LBP-1a|LBP1A|LBP1B',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:12563',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'UMPS',\n",
       "  'name': 'uridine monophosphate synthetase',\n",
       "  'synonyms': 'OPRT',\n",
       "  'description': \"This gene encodes a uridine 5'-monophosphate synthase. The encoded protein is a bifunctional enzyme that catalyzes the final two steps of the de novo pyrimidine biosynthetic pathway. The first reaction is carried out by the N-terminal enzyme orotate phosphoribosyltransferase which converts orotic acid to orotidine-5'-monophosphate. The terminal reaction is carried out by the C-terminal enzyme OMP decarboxylase which converts orotidine-5'-monophosphate to uridine monophosphate. Defects in this gene are the cause of hereditary orotic aciduria. Alternate splicing results in multiple transcript variants. [provided by RefSeq, Mar 2010].\"},\n",
       " {'id': 'HGNC:12572',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'UNG',\n",
       "  'name': 'uracil DNA glycosylase',\n",
       "  'synonyms': 'DGU|HIGM4|HIGM5|UDG|UNG1|UNG15|UNG2',\n",
       "  'description': 'This gene encodes one of several uracil-DNA glycosylases. One important function of uracil-DNA glycosylases is to prevent mutagenesis by eliminating uracil from DNA molecules by cleaving the N-glycosylic bond and initiating the base-excision repair (BER) pathway. Uracil bases occur from cytosine deamination or misincorporation of dUMP residues. Alternative promoter usage and splicing of this gene leads to two different isoforms: the mitochondrial UNG1 and the nuclear UNG2. The UNG2 term was used as a previous symbol for the CCNO gene (GeneID 10309), which has been confused with this gene, in the literature and some databases. [provided by RefSeq, Nov 2010].'},\n",
       " {'id': 'HGNC:12582',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'UQCRB',\n",
       "  'name': 'ubiquinol-cytochrome c reductase binding protein',\n",
       "  'synonyms': 'MC3DN3|QCR7|QP-C|QPC|UQBC|UQBP|UQCR6|UQPC',\n",
       "  'description': 'This gene encodes a subunit of the ubiquinol-cytochrome c oxidoreductase complex, which consists of one mitochondrial-encoded and 10 nuclear-encoded subunits. The protein encoded by this gene binds ubiquinone and participates in the transfer of electrons when ubiquinone is bound. This protein plays an important role in hypoxia-induced angiogenesis through mitochondrial reactive oxygen species-mediated signaling. Mutations in this gene are associated with mitochondrial complex III deficiency. Alternatively spliced transcript variants have been found for this gene. Related pseudogenes have been identified on chromosomes 1, 5 and X. [provided by RefSeq, Dec 2011].'},\n",
       " {'id': 'HGNC:12607',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'USP1',\n",
       "  'name': 'ubiquitin specific peptidase 1',\n",
       "  'synonyms': 'UBP',\n",
       "  'description': 'This gene encodes a member of the ubiquitin-specific processing (UBP) family of proteases that is a deubiquitinating enzyme (DUB) with His and Cys domains. This protein is located in the cytoplasm and cleaves the ubiquitin moiety from ubiquitin-fused precursors and ubiquitinylated proteins. The protein specifically deubiquitinates a protein in the Fanconi anemia (FA) DNA repair pathway. Alternate transcriptional splice variants have been characterized. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:12635',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'UTRN',\n",
       "  'name': 'utrophin',\n",
       "  'synonyms': 'DMDL|DRP|DRP1',\n",
       "  'description': 'This gene shares both structural and functional similarities with the dystrophin gene. It contains an actin-binding N-terminus, a triple coiled-coil repeat central region, and a C-terminus that consists of protein-protein interaction motifs which interact with dystroglycan protein components. The protein encoded by this gene is located at the neuromuscular synapse and myotendinous junctions, where it participates in post-synaptic membrane maintenance and acetylcholine receptor clustering. Mouse studies suggest that this gene may serve as a functional substitute for the dystrophin gene and therefore, may serve as a potential therapeutic alternative to muscular dystrophy which is caused by mutations in the dystrophin gene. Alternative splicing of the utrophin gene has been described; however, the full-length nature of these variants has not yet been determined. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:12666',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'VCP',\n",
       "  'name': 'valosin containing protein',\n",
       "  'synonyms': 'CDC48|FTDALS6|TERA|p97',\n",
       "  'description': 'This gene encodes a member of the AAA ATPase family of proteins. The encoded protein plays a role in protein degradation, intracellular membrane fusion, DNA repair and replication, regulation of the cell cycle, and activation of the NF-kappa B pathway. This protein forms a homohexameric complex that interacts with a variety of cofactors and extracts ubiquitinated proteins from lipid membranes or protein complexes. Mutations in this gene cause IBMPFD (inclusion body myopathy with paget disease of bone and frontotemporal dementia), ALS (amyotrophic lateral sclerosis) and Charcot-Marie-Tooth disease in human patients. [provided by RefSeq, Aug 2017].'},\n",
       " {'id': 'HGNC:12774',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'WNT1',\n",
       "  'name': 'Wnt family member 1',\n",
       "  'synonyms': 'BMND16|INT1|OI15',\n",
       "  'description': 'The WNT gene family consists of structurally related genes which encode secreted signaling proteins. These proteins have been implicated in oncogenesis and in several developmental processes, including regulation of cell fate and patterning during embryogenesis. This gene is a member of the WNT gene family. It is very conserved in evolution, and the protein encoded by this gene is known to be 98% identical to the mouse Wnt1 protein at the amino acid level. The studies in mouse indicate that the Wnt1 protein functions in the induction of the mesencephalon and cerebellum. This gene was originally considered as a candidate gene for Joubert syndrome, an autosomal recessive disorder with cerebellar hypoplasia as a leading feature. However,  further studies suggested that the gene mutations might not have a significant role in Joubert syndrome. This gene is clustered with another family member, WNT10B, in the chromosome 12q13 region. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:12822',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'XPNPEP1',\n",
       "  'name': 'X-prolyl aminopeptidase 1',\n",
       "  'synonyms': 'APP1|SAMP|XPNPEP|XPNPEPL|XPNPEPL1',\n",
       "  'description': 'This gene encodes the cytosolic form of a metalloaminopeptidase that catalyzes the cleavage of the N-terminal amino acid adjacent to a proline residue. The gene product may play a role in degradation and maturation of tachykinins, neuropeptides, and peptide hormones. Alternative splicing results in multiple transcript variants.[provided by RefSeq, Nov 2009].'},\n",
       " {'id': 'HGNC:13053',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ZNF26',\n",
       "  'name': 'zinc finger protein 26',\n",
       "  'synonyms': 'HEL-179|KOX20',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:12953',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ZNF165',\n",
       "  'name': 'zinc finger protein 165',\n",
       "  'synonyms': 'CT53|LD65|ZSCAN7',\n",
       "  'description': 'This gene encodes a member of the Kruppel family of zinc finger proteins. Members of this DNA-binding protein family act as transcriptional regulators. This gene is located within a cluster of zinc finger family members. The encoded protein may play a role in spermatogenesis. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:12974',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RNF113A',\n",
       "  'name': 'ring finger protein 113A',\n",
       "  'synonyms': 'Cwc24|RNF113|TTD5|ZNF183',\n",
       "  'description': 'This intronless gene encodes a protein which contains a C3H1-type zinc finger domain and a C3HC4 Ring-type (Really Interesting New Gene-type) zinc finger domain. The Ring-type zinc finger domain is identified in various tumor suppressors, DNA repair genes and cytokine receptor-associated molecules, and is probably involved in mediating protein-protein interactions. [provided by RefSeq, May 2010].'},\n",
       " {'id': 'HGNC:12989',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ZMYM2',\n",
       "  'name': 'zinc finger MYM-type containing 2',\n",
       "  'synonyms': 'FIM|MYM|RAMP|SCLL|ZNF198',\n",
       "  'description': 'The protein encoded by this gene is a zinc finger protein that may act as a transcription factor. The encoded protein may be part of a BHC histone deacetylase complex. Translocation of this gene with the fibroblast growth factor receptor-1 gene (FGFR1) results in a fusion gene, which may be a cause of stem cell leukemia lymphoma syndrome (SCLL). Several transcript variants encoding the same protein have been found for this gene. [provided by RefSeq, Jul 2010].'},\n",
       " {'id': 'HGNC:9347',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PRDM2',\n",
       "  'name': 'PR/SET domain 2',\n",
       "  'synonyms': 'HUMHOXY1|KMT8|KMT8A|MTB-ZF|RIZ|RIZ1|RIZ2',\n",
       "  'description': 'This tumor suppressor gene is a member of a nuclear histone/protein methyltransferase superfamily. It encodes a zinc finger protein that can bind to retinoblastoma protein, estrogen receptor, and the TPA-responsive element (MTE) of the heme-oxygenase-1 gene. Although the functions of this protein have not been fully characterized, it may (1) play a role in transcriptional regulation during neuronal differentiation and pathogenesis of retinoblastoma, (2) act as a transcriptional activator of the heme-oxygenase-1 gene, and (3) be a specific effector of estrogen action. Multiple transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Oct 2008].'},\n",
       " {'id': 'HGNC:2561',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CXCR4',\n",
       "  'name': 'C-X-C motif chemokine receptor 4',\n",
       "  'synonyms': 'CD184|D2S201E|FB22|HM89|HSY3RR|LAP-3|LAP3|LCR1|LESTR|NPY3R|NPYR|NPYRL|NPYY3R|WHIM|WHIMS',\n",
       "  'description': 'This gene encodes a CXC chemokine receptor specific for stromal cell-derived factor-1. The protein has 7 transmembrane regions and is located on the cell surface. It acts with the CD4 protein to support HIV entry into cells and is also highly expressed in breast cancer cells. Mutations in this gene have been associated with WHIM (warts, hypogammaglobulinemia, infections, and myelokathexis) syndrome. Alternate transcriptional splice variants, encoding different isoforms, have been characterized. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:11357',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'STAM',\n",
       "  'name': 'signal transducing adaptor molecule',\n",
       "  'synonyms': 'STAM-1|STAM1',\n",
       "  'description': 'This gene encodes a member of the signal-transducing adaptor molecule family. These proteins mediate downstream signaling of cytokine receptors and also play a role in ER to Golgi trafficking by interacting with the coat protein II complex. The encoded protein also associates with hepatocyte growth factor-regulated substrate to form the endosomal sorting complex required for transport-0 (ESCRT-0), which sorts ubiquitinated membrane proteins to the ESCRT-1 complex for lysosomal degradation. Alternatively spliced transcript variants have been observed for this gene. [provided by RefSeq, Feb 2011].'},\n",
       " {'id': 'HGNC:7671',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NCOA4',\n",
       "  'name': 'nuclear receptor coactivator 4',\n",
       "  'synonyms': 'ARA70|ELE1|PTC3|RFG',\n",
       "  'description': 'This gene encodes an androgen receptor coactivator. The encoded protein interacts with the androgen receptor in a ligand-dependent manner to enhance its transcriptional activity. Chromosomal translocations between this gene and the ret tyrosine kinase gene, also located on chromosome 10, have been associated with papillary thyroid carcinoma. Alternatively spliced transcript variants have been described. Pseudogenes are present on chromosomes 4, 5, 10, and 14. [provided by RefSeq, Feb 2009].'},\n",
       " {'id': 'HGNC:2556',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CUL5',\n",
       "  'name': 'cullin 5',\n",
       "  'synonyms': 'CUL-5|VACM-1|VACM1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:10919',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SLC14A2',\n",
       "  'name': 'solute carrier family 14 member 2',\n",
       "  'synonyms': 'HUT2|UT-A2|UT2|UTA|UTR|hUT-A6',\n",
       "  'description': 'The protein encoded by this gene belongs to the urea transporter family. In mammalian cells, urea is the chief end product of nitrogen catabolism, and plays an important role in the urinary concentration mechanism. This protein is expressed in the inner medulla of the kidney, and mediates rapid transepithelial urea transport across the inner medullary collecting duct. Alternatively spliced transcript variants have been found for this gene. [provided by RefSeq, Sep 2011].'},\n",
       " {'id': 'HGNC:4778',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'H3-4',\n",
       "  'name': 'H3.4 histone',\n",
       "  'synonyms': 'H3.4|H3/g|H3FT|H3t|HIST3H3',\n",
       "  'description': 'Histones are basic nuclear proteins that are responsible for the nucleosome structure of the chromosomal fiber in eukaryotes. Nucleosomes consist of approximately 146 bp of DNA wrapped around a histone octamer composed of pairs of each of the four core histones (H2A, H2B, H3, and H4). The chromatin fiber is further compacted through the interaction of a linker histone, H1, with the DNA between the nucleosomes to form higher order chromatin structures. This gene is intronless and encodes a replication-dependent histone that is a member of the histone H3 family. Transcripts from this gene lack polyA tails; instead, they contain a palindromic termination element. This gene is located separately from the other H3 genes that are in the histone gene cluster on chromosome 6p22-p21.3. [provided by RefSeq, Aug 2015].'},\n",
       " {'id': 'HGNC:12347',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TRRAP',\n",
       "  'name': 'transformation/transcription domain associated protein',\n",
       "  'synonyms': 'DEDDFA|DFNA75|PAF350/400|PAF400|STAF40|TR-AP|Tra1',\n",
       "  'description': 'This gene encodes a large multidomain protein of the phosphoinositide 3-kinase-related kinases (PIKK) family. The encoded protein is a common component of many histone acetyltransferase (HAT) complexes and plays a role in transcription and DNA repair by recruiting HAT complexes to chromatin. Deregulation of this gene may play a role in several types of cancer including glioblastoma multiforme. Alternatively spliced transcript variants encoding multiple isoforms have been observed for this gene. [provided by RefSeq, Sep 2011].'},\n",
       " {'id': 'HGNC:1739',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CDC45',\n",
       "  'name': 'cell division cycle 45',\n",
       "  'synonyms': 'CDC45L|CDC45L2|MGORS7|PORC-PI-1',\n",
       "  'description': 'The protein encoded by this gene was identified by its strong similarity with Saccharomyces cerevisiae Cdc45, an essential protein required to the initiation of DNA replication. Cdc45 is a member of the highly conserved multiprotein complex including Cdc6/Cdc18, the minichromosome maintenance proteins (MCMs) and DNA polymerase, which is important for early steps of DNA replication in eukaryotes. This protein has been shown to interact with MCM7 and DNA polymerase alpha. Studies of the similar gene in Xenopus suggested that this protein play a pivotal role in the loading of DNA polymerase alpha onto chromatin. Alternate splicing results in multiple transcript variants. [provided by RefSeq, Jul 2013].'},\n",
       " {'id': 'HGNC:4748',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'H2BC13',\n",
       "  'name': 'H2B clustered histone 13',\n",
       "  'synonyms': 'H2B/c|H2BFC|HIST1H2BL|dJ97D16.4',\n",
       "  'description': 'Histones are basic nuclear proteins that are responsible for the nucleosome structure of the chromosomal fiber in eukaryotes. Two molecules of each of the four core histones (H2A, H2B, H3, and H4) form an octamer, around which approximately 146 bp of DNA is wrapped in repeating units, called nucleosomes. The linker histone, H1, interacts with linker DNA between nucleosomes and functions in the compaction of chromatin into higher order structures. This gene is intronless and encodes a replication-dependent histone that is a member of the histone H2B family. Transcripts from this gene lack polyA tails but instead contain a palindromic termination element. This gene is found in the small histone gene cluster on chromosome 6p22-p21.3. [provided by RefSeq, Aug 2015].'},\n",
       " {'id': 'HGNC:4766',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'H3C1',\n",
       "  'name': 'H3 clustered histone 1',\n",
       "  'synonyms': 'H3/A|H3C10|H3C11|H3C12|H3C2|H3C3|H3C4|H3C6|H3C7|H3C8|H3FA|HIST1H3A',\n",
       "  'description': 'Histones are basic nuclear proteins that are responsible for the nucleosome structure of the chromosomal fiber in eukaryotes. This structure consists of approximately 146 bp of DNA wrapped around a nucleosome, an octamer composed of pairs of each of the four core histones (H2A, H2B, H3, and H4). The chromatin fiber is further compacted through the interaction of a linker histone, H1, with the DNA between the nucleosomes to form higher order chromatin structures. This gene is intronless and encodes a replication-dependent histone that is a member of the histone H3 family. Transcripts from this gene lack polyA tails; instead, they contain a palindromic termination element. This gene is found in the large histone gene cluster on chromosome 6p22-p21.3. [provided by RefSeq, Aug 2015].'},\n",
       " {'id': 'HGNC:11345',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RECK',\n",
       "  'name': 'reversion inducing cysteine rich protein with kazal motifs',\n",
       "  'synonyms': 'ST15',\n",
       "  'description': 'The protein encoded by this gene is a cysteine-rich, extracellular protein with protease inhibitor-like domains whose expression is suppressed strongly in many tumors and cells transformed by various kinds of oncogenes. In normal cells, this membrane-anchored glycoprotein may serve as a negative regulator for matrix metalloproteinase-9, a key enzyme involved in tumor invasion and metastasis. Several transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Oct 2015].'},\n",
       " {'id': 'HGNC:9826',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RAD54L',\n",
       "  'name': 'RAD54 like',\n",
       "  'synonyms': 'HR54|RAD54A|hHR54|hRAD54',\n",
       "  'description': 'The protein encoded by this gene belongs to the DEAD-like helicase superfamily, and shares similarity with Saccharomyces cerevisiae Rad54, a protein known to be involved in the homologous recombination and repair of DNA. This protein has been shown to play a role in homologous recombination related repair of DNA double-strand breaks. The binding of this protein to double-strand DNA induces a DNA topological change, which is thought to facilitate homologous DNA paring, and stimulate DNA recombination. Alternative splicing results in multiple transcript variants encoding the same protein.[provided by RefSeq, Dec 2008].'},\n",
       " {'id': 'HGNC:3066',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DUSP11',\n",
       "  'name': 'dual specificity phosphatase 11',\n",
       "  'synonyms': 'PIR1',\n",
       "  'description': 'The protein encoded by this gene is a member of the dual specificity protein phosphatase subfamily. These phosphatases inactivate their target kinases by dephosphorylating both the phosphoserine/threonine and phosphotyrosine residues. They negatively regulate members of the mitogen-activated protein (MAP) kinase superfamily (MAPK/ERK, SAPK/JNK, p38), which is associated with cellular proliferation and differentiation. Different members of the family of dual specificity phosphatases show distinct substrate specificities for various MAP kinases, different tissue distribution and subcellular localization, and different modes of inducibility of their expression by extracellular stimuli. This gene product is localized to the nucleus and binds directly to RNA and splicing factors, and thus it is suggested to participate in nuclear mRNA metabolism. [provided by RefSeq, Sep 2008].'},\n",
       " {'id': 'HGNC:7827',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NIPSNAP1',\n",
       "  'name': 'nipsnap homolog 1',\n",
       "  'synonyms': nan,\n",
       "  'description': 'This gene encodes a member of the NipSnap family of proteins that may be involved in vesicular transport. A similar protein in mice inhibits the calcium channel TRPV6, and is also localized to the inner mitochondrial membrane where it may play a role in mitochondrial DNA maintenance. A pseudogene of this gene is located on the short arm of chromosome 17. Alternatively spliced transcript variants encoding multiple isoforms have been observed for this gene. [provided by RefSeq, Feb 2011].'},\n",
       " {'id': 'HGNC:4821',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HAT1',\n",
       "  'name': 'histone acetyltransferase 1',\n",
       "  'synonyms': 'KAT1',\n",
       "  'description': 'The protein encoded by this gene is a type B histone acetyltransferase (HAT) that is involved in the rapid acetylation of newly synthesized cytoplasmic histones, which are in turn imported into the nucleus for de novo deposition onto nascent DNA chains. Histone acetylation, particularly of histone H4, plays an important role in replication-dependent chromatin assembly. Specifically, this HAT can acetylate soluble but not nucleosomal histone H4 at lysines 5 and 12, and to a lesser degree, histone H2A at lysine 5. Alternatively spliced transcript variants have been identified for this gene. [provided by RefSeq, Jun 2009].'},\n",
       " {'id': 'HGNC:2727',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DDO',\n",
       "  'name': 'D-aspartate oxidase',\n",
       "  'synonyms': 'DASOX|DDO-1|DDO-2',\n",
       "  'description': 'The protein encoded by this gene is a peroxisomal flavoprotein that catalyzes the oxidative deamination of D-aspartate and N-methyl D-aspartate. Flavin adenine dinucleotide or 6-hydroxyflavin adenine dinucleotide can serve as the cofactor in this reaction. Several transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Jan 2019].'},\n",
       " {'id': 'HGNC:19074',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'THOC5',\n",
       "  'name': 'THO complex 5',\n",
       "  'synonyms': 'C22orf19|Fmip|PK1.3|fSAP79',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:16820',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SCARF1',\n",
       "  'name': 'scavenger receptor class F member 1',\n",
       "  'synonyms': 'SREC|SREC-I|SREC1',\n",
       "  'description': 'The protein encoded by this gene is a scavenger receptor that is expressed in endothelial cells. It regulates the uptake of chemically modified low density lipoproteins, including acetylated low density lipoprotein (Ac-LDL), and it may be involved in atherogenesis. This gene is regulated by the transcription factors ZNF444/EZF-2 and SP1. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Apr 2013].'},\n",
       " {'id': 'HGNC:29674',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RDH16',\n",
       "  'name': 'retinol dehydrogenase 16',\n",
       "  'synonyms': 'RODH-4|SDR9C8|hRDH-E',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:3271',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'EIF3A',\n",
       "  'name': 'eukaryotic translation initiation factor 3 subunit A',\n",
       "  'synonyms': 'EIF3|EIF3S10|P167|TIF32|eIF3-p170|eIF3-theta|p180|p185',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:3270',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'EIF3J',\n",
       "  'name': 'eukaryotic translation initiation factor 3 subunit J',\n",
       "  'synonyms': 'EIF3S1|eIF3-alpha|eIF3-p35',\n",
       "  'description': 'This gene encodes a core subunit of the eukaryotic initiation factor 3 complex, which participates in the initiation of translation by aiding in the recruitment of protein and mRNA components to the 40S ribosome. There are pseudogenes for this gene on chromosomes 1, 3, and 9. Alternative splicing results in multiple transcript variants encoding different isoforms. [provided by RefSeq, Sep 2013].'},\n",
       " {'id': 'HGNC:11429',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'STX11',\n",
       "  'name': 'syntaxin 11',\n",
       "  'synonyms': 'FHL4|HLH4|HPLH4',\n",
       "  'description': 'This gene encodes a member of the syntaxin family. Syntaxins have been implicated in the targeting and fusion of intracellular transport vesicles. This family member may regulate protein transport among late endosomes and the trans-Golgi network. Mutations in this gene have been associated with familial hemophagocytic lymphohistiocytosis. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:1034',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'BECN1',\n",
       "  'name': 'beclin 1',\n",
       "  'synonyms': 'ATG6|VPS30|beclin1',\n",
       "  'description': 'This gene encodes a protein that regulates autophagy, a catabolic process of degradation induced by starvation. The encoded protein is a component of the phosphatidylinositol-3-kinase (PI3K) complex which mediates vesicle-trafficking processes. This protein is thought to play a role in multiple cellular processes, including tumorigenesis, neurodegeneration and apoptosis. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Sep 2015].'},\n",
       " {'id': 'HGNC:11449',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SUCLG1',\n",
       "  'name': 'succinate-CoA ligase GDP/ADP-forming subunit alpha',\n",
       "  'synonyms': 'GALPHA|MTDPS9|SUCLA1',\n",
       "  'description': 'This gene encodes the alpha subunit of the heterodimeric enzyme succinate coenzyme A ligase. This enzyme is targeted to the mitochondria and catalyzes the conversion of succinyl CoA and ADP or GDP to succinate and ATP or GTP. Mutations in this gene are the cause of the metabolic disorder fatal infantile lactic acidosis and mitochondrial DNA depletion. [provided by RefSeq, Feb 2010].'},\n",
       " {'id': 'HGNC:19382',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SOCS2',\n",
       "  'name': 'suppressor of cytokine signaling 2',\n",
       "  'synonyms': 'CIS2|Cish2|SOCS-2|SSI-2|SSI2|STATI2',\n",
       "  'description': 'This gene encodes a member of the suppressor of cytokine signaling (SOCS) family. SOCS family members are cytokine-inducible negative regulators of cytokine receptor signaling via the Janus kinase/signal transducer and activation of transcription pathway (the JAK/STAT pathway). SOCS family proteins interact with major molecules of signaling complexes to block further signal transduction, in part, by proteasomal depletion of receptors or signal-transducing proteins via ubiquitination. The expression of this gene can be induced by a subset of cytokines, including erythropoietin, GM-CSF, IL10, interferon (IFN)-gamma and by cytokine receptors such as growth horomone receptor. The protein encoded by this gene interacts with the cytoplasmic domain of insulin-like growth factor-1 receptor (IGF1R) and is thought to be involved in the regulation of IGF1R mediated cell signaling. This gene has pseudogenes on chromosomes 20 and 22. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Jul 2012].'},\n",
       " {'id': 'HGNC:5392',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'IER3',\n",
       "  'name': 'immediate early response 3',\n",
       "  'synonyms': 'DIF-2|DIF2|GLY96|IEX-1|IEX-1L|IEX1|PRG1',\n",
       "  'description': 'This gene functions in the protection of cells from Fas- or tumor necrosis factor type alpha-induced apoptosis. Partially degraded and unspliced transcripts are found after virus infection in vitro, but these transcripts are not found in vivo and do not generate a valid protein. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:989',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'BCL10',\n",
       "  'name': 'BCL10 immune signaling adaptor',\n",
       "  'synonyms': 'CARMEN|CIPER|CLAP|IMD37|c-E10|mE10',\n",
       "  'description': 'This gene was identified by its translocation in a case of mucosa-associated lymphoid tissue (MALT) lymphoma. The protein encoded by this gene contains a caspase recruitment domain (CARD), and has been shown to induce apoptosis and to activate NF-kappaB. This protein is reported to interact with other CARD domain containing proteins including CARD9, 10, 11 and 14, which are thought to function as upstream regulators in NF-kappaB signaling. This protein is found to form a complex with MALT1, a protein encoded by another gene known to be translocated in MALT lymphoma. MALT1 and this protein are thought to synergize in the activation of NF-kappaB, and the deregulation of either of them may contribute to the same pathogenetic process that leads to the malignancy. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Mar 2016].'},\n",
       " {'id': 'HGNC:1144',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'BTRC',\n",
       "  'name': 'beta-transducin repeat containing E3 ubiquitin protein ligase',\n",
       "  'synonyms': 'BETA-TRCP|FBW1A|FBXW1|FBXW1A|FWD1|bTrCP|bTrCP1|betaTrCP',\n",
       "  'description': 'This gene encodes a member of the F-box protein family which is characterized by an approximately 40 amino acid motif, the F-box. The F-box proteins constitute one of the four subunits of ubiquitin protein ligase complex called SCFs (SKP1-cullin-F-box), which function in phosphorylation-dependent ubiquitination. The F-box proteins are divided into 3 classes: Fbws containing WD-40 domains, Fbls containing leucine-rich repeats, and Fbxs containing either different protein-protein interaction modules or no recognizable motifs. The protein encoded by this gene belongs to the Fbws class; in addition to an F-box, this protein contains multiple WD-40 repeats. The encoded protein mediates degradation of CD4 via its interaction with HIV-1 Vpu. It has also been shown to ubiquitinate phosphorylated NFKBIA (nuclear factor of kappa light polypeptide gene enhancer in B-cells inhibitor, alpha), targeting it for degradation and thus activating nuclear factor kappa-B. Alternatively spliced transcript variants have been described. A related pseudogene exists in chromosome 6. [provided by RefSeq, Mar 2012].'},\n",
       " {'id': 'HGNC:7869',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NOL3',\n",
       "  'name': 'nucleolar protein 3',\n",
       "  'synonyms': 'ARC|FCM|MYOCL1|MYP|NOP|NOP30',\n",
       "  'description': 'This gene encodes an anti-apoptotic protein that has been shown to down-regulate the enzyme activities of caspase 2, caspase 8 and tumor protein p53. Multiple transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Jun 2010].'},\n",
       " {'id': 'HGNC:11534',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TAF1C',\n",
       "  'name': 'TATA-box binding protein associated factor, RNA polymerase I subunit C',\n",
       "  'synonyms': 'MGC:39976|SL1|TAFI110|TAFI95',\n",
       "  'description': 'Initiation of transcription by RNA polymerase I requires the formation of a complex composed of the TATA-binding protein (TBP) and three TBP-associated factors (TAFs) specific for RNA polymerase I. This complex, known as SL1, binds to the core promoter of ribosomal RNA genes to position the polymerase properly and acts as a channel for regulatory signals. This gene encodes the largest SL1-specific TAF. Multiple alternatively spliced transcript variants encoding different isoforms have been identified. [provided by RefSeq, Jul 2011].'},\n",
       " {'id': 'HGNC:6853',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MAP3K14',\n",
       "  'name': 'mitogen-activated protein kinase kinase kinase 14',\n",
       "  'synonyms': 'FTDCR1B|HS|HSNIK|NIK',\n",
       "  'description': 'This gene encodes mitogen-activated protein kinase kinase kinase 14, which is a serine/threonine protein-kinase. This kinase binds to TRAF2 and stimulates NF-kappaB activity. It shares sequence similarity with several other MAPKK kinases. It participates in an NF-kappaB-inducing signalling cascade common to receptors of the tumour-necrosis/nerve-growth factor (TNF/NGF) family and to the interleukin-1 type-I receptor. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:12565',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'UNC119',\n",
       "  'name': 'unc-119 lipid binding chaperone',\n",
       "  'synonyms': 'HRG4|IMD13|POC7|POC7A',\n",
       "  'description': 'This gene is specifically expressed in the photoreceptors in the retina. The encoded product shares strong homology with the C. elegans unc119 protein and it can functionally complement the C. elegans unc119 mutation. It has been localized to the photoreceptor synapses in the outer plexiform layer of the retina, and suggested to play a role in the mechanism of photoreceptor neurotransmitter release through the synaptic vesicle cycle. Two transcript variants encoding different isoforms have been described for this gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:24431',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'KRT75',\n",
       "  'name': 'keratin 75',\n",
       "  'synonyms': 'CK-75|K6HF|K75|KB18|PFB|hK6hf',\n",
       "  'description': 'This gene is a member of the type II keratin family clustered on the long arm of chromosome 12. Type I and type II keratins heteropolymerize to form intermediate-sized filaments in the cytoplasm of epithelial cells. This gene is expressed in the companion layer, upper germinative matrix region of the hair follicle, and medulla of the hair shaft. The encoded protein plays an essential role in hair and nail formation. Variations in this gene have been associated with the hair disorders pseudofolliculitis barbae (PFB) and loose anagen hair syndrome (LAHS). [provided by RefSeq, Oct 2008].'},\n",
       " {'id': 'HGNC:11001',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SLC28A1',\n",
       "  'name': 'solute carrier family 28 member 1',\n",
       "  'synonyms': 'CNT1|HCNT1|URCTU',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:3168',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'LPAR2',\n",
       "  'name': 'lysophosphatidic acid receptor 2',\n",
       "  'synonyms': 'EDG-4|EDG4|LPA-2|LPA2',\n",
       "  'description': 'This gene encodes a member of family I of the G protein-coupled receptors, as well as the EDG family of proteins. This protein functions as a lysophosphatidic acid (LPA) receptor and contributes to Ca2+ mobilization, a critical cellular response to LPA in cells, through association with Gi and Gq proteins. An alternative splice variant has been described but its full length sequence has not been determined. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:13029',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ZMYM5',\n",
       "  'name': 'zinc finger MYM-type containing 5',\n",
       "  'synonyms': 'HSPC050|MYM|ZNF198L1|ZNF237',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:12306',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TRIP12',\n",
       "  'name': 'thyroid hormone receptor interactor 12',\n",
       "  'synonyms': 'MRD49|TRIP-12|TRIPC|ULF',\n",
       "  'description': 'The protein encoded by this gene is an E3 ubiquitin-protein ligase involved in the degradation of the p19ARF/ARF isoform of CDKN2A, a tumor suppressor. The encoded protein also plays a role in the DNA damage response by regulating the stability of USP7, which regulates tumor suppressor p53. [provided by RefSeq, Jan 2017].'},\n",
       " {'id': 'HGNC:12436',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TXNL1',\n",
       "  'name': 'thioredoxin like 1',\n",
       "  'synonyms': 'HEL-S-114|TRP32|TXL-1|TXNL|Txl',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:2375',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MED17',\n",
       "  'name': 'mediator complex subunit 17',\n",
       "  'synonyms': 'CRSP6|CRSP77|DRIP80|SRB4|TRAP80',\n",
       "  'description': 'The activation of gene transcription is a multistep process that is triggered by factors that recognize transcriptional enhancer sites in DNA. These factors work with co-activators to direct transcriptional initiation by the RNA polymerase II apparatus. The protein encoded by this gene is a subunit of the CRSP (cofactor required for SP1 activation) complex, which, along with TFIID, is required for efficient activation by SP1. This protein is also a component of other multisubunit complexes e.g. thyroid hormone receptor-(TR-) associated proteins which interact with TR and facilitate TR function on DNA templates in conjunction with initiation factors and cofactors. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:8959',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PIGB',\n",
       "  'name': 'phosphatidylinositol glycan anchor biosynthesis class B',\n",
       "  'synonyms': 'DEE80|EIEE80|GPI-MT-III|PIG-B',\n",
       "  'description': 'This gene encodes a transmembrane protein that is located in the endoplasmic reticulum and is involved in GPI-anchor biosynthesis. The glycosylphosphatidylinositol (GPI) anchor is a glycolipid found on many blood cells and serves to anchor proteins to the cell surface. This gene is thought to encode a member of a family of dolichol-phosphate-mannose (Dol-P-Man) dependent mannosyltransferases. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:4112',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'XAGE2',\n",
       "  'name': 'X antigen family member 2',\n",
       "  'synonyms': 'CT12.2|GAGED3|XAGE-2|XAGE2B',\n",
       "  'description': \"This gene is a member of the XAGE subfamily, which belongs to the GAGE family. The GAGE genes are expressed in a variety of tumors and in some fetal and reproductive tissues. This gene is strongly expressed in normal testis, and in Ewing's sarcoma, rhabdomyosarcoma, a breast cancer and a germ cell tumor. The protein encoded by this gene shares a sequence similarity with other GAGE/PAGE proteins. Because of the expression pattern and the sequence similarity, this protein also belongs to a family of CT (cancer-testis) antigens. [provided by RefSeq, Jul 2008].\"},\n",
       " {'id': 'NCBIGene:9503',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'XAGE1B',\n",
       "  'name': 'X antigen family member 1B',\n",
       "  'synonyms': 'CT12.1|CT12.1C|CT12.1D|CT12.1E|CT12.1b|CTP9|GAGED2|XAGE-1|XAGE1|XAGE1C|XAGE1D|XAGE1E',\n",
       "  'description': \"This gene is a member of the XAGE subfamily, which belongs to the GAGE family. The GAGE genes are expressed in a variety of tumors and in some fetal and reproductive tissues. This gene is strongly expressed in Ewing's sarcoma, alveolar rhabdomyosarcoma and normal testis. The protein encoded by this gene contains a nuclear localization signal and shares a sequence similarity with other GAGE/PAGE proteins. Because of the expression pattern and the sequence similarity, this protein also belongs to a family of CT (cancer-testis) antigens. Alternative splicing of this gene, in addition to alternative transcription start sites, results in multiple transcript variants. [provided by RefSeq, Jan 2010].\"},\n",
       " {'id': 'HGNC:217',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ADAMTS1',\n",
       "  'name': 'ADAM metallopeptidase with thrombospondin type 1 motif 1',\n",
       "  'synonyms': 'C3-C5|METH1',\n",
       "  'description': 'This gene encodes a member of the ADAMTS (a disintegrin and metalloproteinase with thrombospondin motif) protein family. Members of the family share several distinct protein modules, including a propeptide region, a metalloproteinase domain, a disintegrin-like domain, and a thrombospondin type 1 (TS) motif. Individual members of this family differ in the number of C-terminal TS motifs, and some have unique C-terminal domains. The protein encoded by this gene contains two disintegrin loops and three C-terminal TS motifs and has anti-angiogenic activity. The expression of this gene may be associated with various inflammatory processes as well as development of cancer cachexia. This gene is likely to be necessary for normal growth, fertility, and organ morphology and function. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:21605',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SAFB2',\n",
       "  'name': 'scaffold attachment factor B2',\n",
       "  'synonyms': nan,\n",
       "  'description': 'The protein encoded by this gene, along with its paralog (scaffold attachment factor B1), is a repressor of estrogen receptor alpha. The encoded protein binds scaffold/matrix attachment region (S/MAR) DNA and is involved in cell cycle regulation, apoptosis, differentiation, the stress response, and regulation of immune genes. [provided by RefSeq, May 2016].'},\n",
       " {'id': 'HGNC:29850',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'N4BP1',\n",
       "  'name': 'NEDD4 binding protein 1',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:13744',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HERPUD1',\n",
       "  'name': 'homocysteine inducible ER protein with ubiquitin like domain 1',\n",
       "  'synonyms': 'HERP|Mif1|SUP',\n",
       "  'description': 'The accumulation of unfolded proteins in the endoplasmic reticulum (ER) triggers the ER stress response. This response includes the inhibition of translation to prevent further accumulation of unfolded proteins, the increased expression of proteins involved in polypeptide folding, known as the unfolded protein response (UPR), and the destruction of misfolded proteins by the ER-associated protein degradation (ERAD) system. This gene may play a role in both UPR and ERAD. Its expression is induced by UPR and it has an ER stress response element in its promoter region while the encoded protein has an N-terminal ubiquitin-like domain which may interact with the ERAD system. This protein has been shown to interact with presenilin proteins and to increase the level of amyloid-beta protein following its overexpression. Alternative splicing of this gene produces multiple transcript variants encoding different isoforms. The full-length nature of all transcript variants has not been determined. [provided by RefSeq, Jan 2013].'},\n",
       " {'id': 'NCBIGene:9714',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DOCK10',\n",
       "  'name': 'dedicator of cytokinesis 10',\n",
       "  'synonyms': 'DRIP2|Nbla10300|ZIZ3',\n",
       "  'description': 'This gene encodes a member of the dedicator of cytokinesis protein family. Members of this family are guanosine nucleotide exchange factors for Rho GTPases and defined by the presence of conserved DOCK-homology regions. The encoded protein belongs to the D (or Zizimin) subfamily of DOCK proteins, which also contain an N-terminal pleckstrin homology domain. Alternatively spliced transcript variants that encode different isoforms have been described. [provided by RefSeq, Mar 2014].'},\n",
       " {'id': 'HGNC:24342',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CCP110',\n",
       "  'name': 'centriolar coiled-coil protein 110',\n",
       "  'synonyms': 'CP110|Cep110',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:18683',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'EIF4A3',\n",
       "  'name': 'eukaryotic translation initiation factor 4A3',\n",
       "  'synonyms': 'DDX48|Fal1|MUK34|NMP265|NUK34|RCPS|eIF-4A-III|eIF4A-III|eIF4AIII',\n",
       "  'description': 'This gene encodes a member of the DEAD box protein family. DEAD box proteins, characterized by the conserved motif Asp-Glu-Ala-Asp (DEAD), are putative RNA helicases. They are implicated in a number of cellular processes involving alteration of RNA secondary structure, such as translation initiation, nuclear and mitochondrial splicing, and ribosome and spliceosome assembly. Based on their distribution patterns, some members of this family are believed to be involved in embryogenesis, spermatogenesis, and cellular growth and division. The protein encoded by this gene is a nuclear matrix protein. Its amino acid sequence is highly similar to the amino acid sequences of the translation initiation factors eIF4AI and eIF4AII, two other members of the DEAD box protein family. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:21726',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ARHGEF17',\n",
       "  'name': 'Rho guanine nucleotide exchange factor 17',\n",
       "  'synonyms': 'P164RHOGEF|RHOGEF17|TEM4|p164-RhoGEF',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:28985',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'LPGAT1',\n",
       "  'name': 'lysophosphatidylglycerol acyltransferase 1',\n",
       "  'synonyms': 'FAM34A|FAM34A1|NET8',\n",
       "  'description': 'This gene encodes a member of the lysophospholipid acyltransferase family. The encoded protein catalyzes the reacylation of lysophosphatidylglycerol to phosphatidylglycerol, a membrane phospholipid that is an important precursor for the synthesis of cardiolipin. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Mar 2016].'},\n",
       " {'id': 'HGNC:14013',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SMC4',\n",
       "  'name': 'structural maintenance of chromosomes 4',\n",
       "  'synonyms': 'CAP-C|CAPC|SMC-4|SMC4L1',\n",
       "  'description': \"This gene belongs to the 'structural maintenance of chromosomes' (SMC) gene family. Members of this gene family play a role in two changes in chromosome structure during mitotic segregation of chromosomes- chromosome condensation and sister chromatid cohesion. The protein encoded by this gene is likely a subunit of the 13S condensin complex, which is involved in chromosome condensation. A pseudogene related to this gene is located on chromosome 2. [provided by RefSeq, Jun 2016].\"},\n",
       " {'id': 'HGNC:4280',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GJC1',\n",
       "  'name': 'gap junction protein gamma 1',\n",
       "  'synonyms': 'CX45|GJA7',\n",
       "  'description': 'This gene is a member of the connexin gene family. The encoded protein is a component of gap junctions, which are composed of arrays of intercellular channels that provide a route for the diffusion of low molecular weight materials from cell to cell. Alternatively spliced transcript variants encoding the same isoform have been described. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:3173',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'EDIL3',\n",
       "  'name': 'EGF like repeats and discoidin domains 3',\n",
       "  'synonyms': 'DEL1',\n",
       "  'description': 'The protein encoded by this gene is an integrin ligand. It plays an important role in mediating angiogenesis and may be important in vessel wall remodeling and development. It also influences endothelial cell behavior. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:169',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ACTR2',\n",
       "  'name': 'actin related protein 2',\n",
       "  'synonyms': 'ARP2',\n",
       "  'description': 'The specific function of this gene has not yet been determined; however, the protein it encodes is known to be a major constituent of the ARP2/3 complex. This complex is located at the cell surface and is essential to cell shape and motility through lamellipodial actin assembly and protrusion. Two transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:18641',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SUGP2',\n",
       "  'name': 'SURP and G-patch domain containing 2',\n",
       "  'synonyms': 'SFRS14|SRFS14',\n",
       "  'description': 'This gene encodes a member of the arginine/serine-rich family of splicing factors. The encoded protein functions in mRNA processing. Alternatively spliced transcript variants have been described. [provided by RefSeq, Feb 2009].'},\n",
       " {'id': 'HGNC:16887',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PDZK1IP1',\n",
       "  'name': 'PDZK1 interacting protein 1',\n",
       "  'synonyms': 'DD96|MAP17|SPAP',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:10757',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SERF2',\n",
       "  'name': 'small EDRK-rich factor 2',\n",
       "  'synonyms': '4F5REL|FAM2C|H4F5REL|HsT17089',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:7213',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MPHOSPH10',\n",
       "  'name': 'M-phase phosphoprotein 10',\n",
       "  'synonyms': 'CT90|MPP10|MPP10P|PPP1R106',\n",
       "  'description': 'This gene encodes a protein that is phosphorylated during mitosis. The protein localizes to the nucleolus during interphase and to the chromosomes during M phase. The protein associates with the U3 small nucleolar ribonucleoprotein 60-80S complexes and may be involved in pre-rRNA processing. [provided by RefSeq, Dec 2010].'},\n",
       " {'id': 'HGNC:18349',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DHRS2',\n",
       "  'name': 'dehydrogenase/reductase 2',\n",
       "  'synonyms': 'HEP27|SDR25C1',\n",
       "  'description': 'This gene encodes a member of the short-chain dehydrogenases/reductases (SDR) family, which has over 46,000 members. Members of this family are enzymes that metabolize many different compounds, such as steroid hormones, prostaglandins, retinoids, lipids and xenobiotics. Alternative promoter use and alternative splicing results in multiple transcript variants. [provided by RefSeq, Feb 2017].'},\n",
       " {'id': 'HGNC:20294',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'USPL1',\n",
       "  'name': 'ubiquitin specific peptidase like 1',\n",
       "  'synonyms': 'C13orf22|D13S106E|bA121O19.1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:13176',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'IKZF1',\n",
       "  'name': 'IKAROS family zinc finger 1',\n",
       "  'synonyms': 'CVID13|Hs.54452|IK1|IKAROS|LYF1|LyF-1|PPP1R92|PRO0758|ZNFN1A1',\n",
       "  'description': 'This gene encodes a transcription factor that belongs to the family of zinc-finger DNA-binding proteins associated with chromatin remodeling. The expression of this protein is restricted to the fetal and adult hemo-lymphopoietic system, and it functions as a regulator of lymphocyte differentiation. Several alternatively spliced transcript variants encoding different isoforms have been described for this gene. Most isoforms share a common C-terminal domain, which contains two zinc finger motifs that are required for hetero- or homo-dimerization, and for interactions with other proteins. The isoforms, however, differ in the number of N-terminal zinc finger motifs that bind DNA and in nuclear localization signal presence, resulting in members with and without DNA-binding properties. Only a few isoforms contain the requisite three or more N-terminal zinc motifs that confer high affinity binding to a specific core DNA sequence element in the promoters of target genes. The non-DNA-binding isoforms are largely found in the cytoplasm, and are thought to function as dominant-negative factors. Overexpression of some dominant-negative isoforms have been associated with B-cell malignancies, such as acute lymphoblastic leukemia (ALL). [provided by RefSeq, May 2014].'},\n",
       " {'id': 'HGNC:1140',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'BTN3A3',\n",
       "  'name': 'butyrophilin subfamily 3 member A3',\n",
       "  'synonyms': 'BTF3|BTN3.3',\n",
       "  'description': 'The butyrophilin (BTN) genes are a group of major histocompatibility complex (MHC)-associated genes that encode type I membrane proteins with 2 extracellular immunoglobulin (Ig) domains and an intracellular B30.2 (PRYSPRY) domain. Three subfamilies of human BTN genes are located in the MHC class I region: the single-copy BTN1A1 gene (MIM 601610) and the BTN2 (e.g., BTN2A1; MIM 613590) and BTN3 (e.g., BNT3A3) genes, which have undergone tandem duplication, resulting in 3 copies of each (summary by Smith et al., 2010 [PubMed 20208008]).[supplied by OMIM, Nov 2010].'},\n",
       " {'id': 'HGNC:16909',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NDC80',\n",
       "  'name': 'NDC80 kinetochore complex component',\n",
       "  'synonyms': 'HEC|HEC1|HsHec1|KNTC2|TID3|hsNDC80',\n",
       "  'description': 'This gene encodes a component of the NDC80 kinetochore complex. The encoded protein consists of an N-terminal microtubule binding domain and a C-terminal coiled-coiled domain that interacts with other components of the complex. This protein functions to organize and stabilize microtubule-kinetochore interactions and is required for proper chromosome segregation. [provided by RefSeq, Oct 2011].'},\n",
       " {'id': 'HGNC:16914',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'LRRN2',\n",
       "  'name': 'leucine rich repeat neuronal 2',\n",
       "  'synonyms': 'FIGLER7|GAC1|LRANK1|LRRN5',\n",
       "  'description': 'The protein encoded by this gene belongs to the leucine-rich repeat superfamily. This gene was found to be amplified and overexpressed in malignant gliomas. The encoded protein has homology with other proteins that function as cell-adhesion molecules or as signal transduction receptors and is a candidate for the target gene in the 1q32.1 amplicon in malignant gliomas. Two alternatively spliced transcript variants encoding the same protein have been described for this gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:4989',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HMGN4',\n",
       "  'name': 'high mobility group nucleosomal binding domain 4',\n",
       "  'synonyms': 'HMG17L3|NHC',\n",
       "  'description': 'The protein encoded by this gene, a member of the HMGN protein family, is thought to reduce the compactness of the chromatin fiber in nucleosomes, thereby enhancing transcription from chromatin templates. [provided by RefSeq, Mar 2013].'},\n",
       " {'id': 'HGNC:10867',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ST6GALNAC2',\n",
       "  'name': 'ST6 N-acetylgalactosaminide alpha-2,6-sialyltransferase 2',\n",
       "  'synonyms': 'SAITL1|SIAT7|SIAT7B|SIATL1|ST6GalNAII|STHM',\n",
       "  'description': 'ST6GALNAC2 belongs to a family of sialyltransferases that add sialic acids to the nonreducing ends of glycoconjugates. At the cell surface, these modifications have roles in cell-cell and cell-substrate interactions, bacterial adhesion, and protein targeting (Samyn-Petit et al., 2000 [PubMed 10742600]).[supplied by OMIM, Mar 2008].'},\n",
       " {'id': 'HGNC:15763',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'POLR3F',\n",
       "  'name': 'RNA polymerase III subunit F',\n",
       "  'synonyms': 'C34|RPC39|RPC6',\n",
       "  'description': 'The protein encoded by this gene is one of more than a dozen subunits forming eukaryotic RNA polymerase III (RNA Pol III), which transcribes 5S ribosomal RNA and tRNA genes. This protein has been shown to bind both TFIIIB90 and TBP, two subunits of RNA polymerase III transcription initiation factor IIIB (TFIIIB). Unlike most of the other RNA Pol III subunits, the encoded protein is unique to this polymerase. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Sep 2013].'},\n",
       " {'id': 'HGNC:29602',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PDPN',\n",
       "  'name': 'podoplanin',\n",
       "  'synonyms': 'AGGRUS|GP36|GP40|Gp38|HT1A-1|OTS8|PA2.26|T1A|T1A-2|T1A2|TI1A',\n",
       "  'description': 'This gene encodes a type-I integral membrane glycoprotein with diverse distribution in human tissues. The physiological function of this protein may be related to its mucin-type character. The homologous protein in other species has been described as a differentiation antigen and influenza-virus receptor. The specific function of this protein has not been determined but it has been proposed as a marker of lung injury. Alternatively spliced transcript variants encoding different isoforms have been identified. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:2549',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CELF1',\n",
       "  'name': 'CUGBP Elav-like family member 1',\n",
       "  'synonyms': 'BRUNOL2|CUG-BP|CUGBP|CUGBP1|EDEN-BP|NAB50|NAPOR|hNab50',\n",
       "  'description': 'Members of the CELF/BRUNOL protein family contain two N-terminal RNA recognition motif (RRM) domains, one C-terminal RRM domain, and a divergent segment of 160-230 aa between the second and third RRM domains. Members of this protein family regulate pre-mRNA alternative splicing and may also be involved in mRNA editing, and translation. This gene may play a role in myotonic dystrophy type 1 (DM1) via interactions with the dystrophia myotonica-protein kinase (DMPK) gene. Alternative splicing results in multiple transcript variants encoding different isoforms. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:1623',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CCT8',\n",
       "  'name': 'chaperonin containing TCP1 subunit 8',\n",
       "  'synonyms': 'C21orf112|Cctq|D21S246|PRED71',\n",
       "  'description': 'This gene encodes the theta subunit of the CCT chaperonin, which is abundant in the eukaryotic cytosol and may be involved in the transport and assembly of newly synthesized proteins. Alternative splicing results in multiple transcript variants of this gene. A pseudogene related to this gene is located on chromosome 1. [provided by RefSeq, Sep 2013].'},\n",
       " {'id': 'HGNC:12843',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'YME1L1',\n",
       "  'name': 'YME1 like 1 ATPase',\n",
       "  'synonyms': 'FTSH|MEG4|OPA11|PAMP|YME1L',\n",
       "  'description': 'The protein encoded by this gene is the human ortholog of yeast mitochondrial AAA metalloprotease, Yme1p. It is localized in the mitochondria and can functionally complement a yme1 disruptant yeast strain. It is proposed that this gene plays a role in mitochondrial protein metabolism and could be involved in mitochondrial pathologies. Three transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Dec 2011].'},\n",
       " {'id': 'HGNC:13061',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ZNF268',\n",
       "  'name': 'zinc finger protein 268',\n",
       "  'synonyms': 'HZF3',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:10475',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RUVBL2',\n",
       "  'name': 'RuvB like AAA ATPase 2',\n",
       "  'synonyms': 'CGI-46|ECP-51|ECP51|INO80J|REPTIN|RVB2|TAP54-beta|TIH2|TIP48|TIP49B',\n",
       "  'description': 'This gene encodes the second human homologue of the bacterial RuvB gene. Bacterial RuvB protein is a DNA helicase essential for homologous recombination and DNA double-strand break repair. Functional analysis showed that this gene product has both ATPase and DNA helicase activities. This gene is physically linked to the CGB/LHB gene cluster on chromosome 19q13.3, and is very close (55 nt) to the LHB gene, in the opposite orientation. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:12755',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'WDR3',\n",
       "  'name': 'WD repeat domain 3',\n",
       "  'synonyms': 'DIP2|UTP12',\n",
       "  'description': 'This gene encodes a nuclear protein containing 10 WD repeats. WD repeats are approximately 30- to 40-amino acid domains containing several conserved residues, which usually include a trp-asp at the C-terminal end. Proteins belonging to the WD repeat family are involved in a variety of cellular processes, including cell cycle progression, signal transduction, apoptosis, and gene regulation. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:4524',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PROKR1',\n",
       "  'name': 'prokineticin receptor 1',\n",
       "  'synonyms': 'GPR73|GPR73a|PK-R1|PKR1|ZAQ',\n",
       "  'description': 'This gene encodes a member of the G-protein-coupled receptor family. The encoded protein binds to prokineticins (1 and 2), leading to the activation of MAPK and STAT signaling pathways. Prokineticins are protein ligands involved in angiogenesis and inflammation. The encoded protein is expressed in peripheral tissues such as those comprising the circulatory system, lungs, reproductive system, endocrine system and the gastrointestinal system. The protein may be involved in signaling in human fetal ovary during initiation of primordial follicle formation. Sequence variants in this gene may be associated with recurrent miscarriage. [provided by RefSeq, Aug 2016].'},\n",
       " {'id': 'HGNC:17364',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DBF4',\n",
       "  'name': 'DBF4 zinc finger',\n",
       "  'synonyms': 'ASK|CHIF|DBF4A|ZDBF1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:16988',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SRSF8',\n",
       "  'name': 'serine and arginine rich splicing factor 8',\n",
       "  'synonyms': 'DSM-1|SFRS2B|SRP46',\n",
       "  'description': 'This gene encodes a member of a family of proteins containing a ribonucleoprotein (RNP)-type RNA binding motif and a carboxyl-terminal arginine-serine-rich (RS) domain. The encoded protein functions as a pre-mRNA splicing factor. There is a pseudogene for this gene on chromosome 7. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Jun 2013].'},\n",
       " {'id': 'HGNC:30074',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'POLR3A',\n",
       "  'name': 'RNA polymerase III subunit A',\n",
       "  'synonyms': 'ADDH|C160|HLD7|RPC1|RPC155|WDRTS|hRPC155',\n",
       "  'description': 'The protein encoded by this gene is the catalytic component of RNA polymerase III, which synthesizes small RNAs. The encoded protein also acts as a sensor to detect foreign DNA and trigger an innate immune response. [provided by RefSeq, Aug 2011].'},\n",
       " {'id': 'HGNC:11067',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SLC7A9',\n",
       "  'name': 'solute carrier family 7 member 9',\n",
       "  'synonyms': 'BAT1|CSNU3',\n",
       "  'description': 'This gene encodes a protein that belongs to a family of light subunits of amino acid transporters. This protein plays a role in the high-affinity and sodium-independent transport of cystine and neutral and dibasic amino acids, and appears to function in the reabsorption of cystine in the kidney tubule. Mutations in this gene cause non-type I cystinuria, a disease that leads to cystine stones in the urinary system due to impaired transport of cystine and dibasic amino acids. Alternate transcript variants, which encode the same protein, have been found for this gene. [provided by RefSeq, Jul 2011].'},\n",
       " {'id': 'HGNC:28912',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'WDR45',\n",
       "  'name': 'WD repeat domain 45',\n",
       "  'synonyms': 'JM5|NBIA4|NBIA5|WDRX1|WIPI-4|WIPI4',\n",
       "  'description': 'This gene encodes a member of the WD repeat protein family. WD repeats are minimally conserved regions of approximately 40 amino acids typically bracketed by gly-his and trp-asp (GH-WD), which may facilitate formation of heterotrimeric or multiprotein complexes. Members of this family are involved in a variety of cellular processes, including cell cycle progression, signal transduction, apoptosis, and gene regulation. This gene has a pseudogene at chromosome 4q31.3. Multiple alternatively spliced transcript variants encoding distinct isoforms have been found for this gene, but the biological validity and full-length nature of some variants have not been determined. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:8051',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NUDT4',\n",
       "  'name': 'nudix hydrolase 4',\n",
       "  'synonyms': 'DIPP-2B|DIPP2|DIPP2alpha|DIPP2beta|HDCMB47P|NUDT4B',\n",
       "  'description': 'The protein encoded by this gene regulates the turnover of diphosphoinositol polyphosphates. The turnover of these high-energy diphosphoinositol polyphosphates represents a molecular switching activity with important regulatory consequences. Molecular switching by diphosphoinositol polyphosphates may contribute to regulating intracellular trafficking. Several alternatively spliced transcript variants have been described, but the full-length nature of some variants has not been determined. Isoforms DIPP2alpha and DIPP2beta are distinguishable from each other solely by DIPP2beta possessing one additional amino acid due to intron boundary skidding in alternate splicing. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:6867',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MAP4K5',\n",
       "  'name': 'mitogen-activated protein kinase kinase kinase kinase 5',\n",
       "  'synonyms': 'GCKR|KHS|KHS1|MAPKKKK5',\n",
       "  'description': 'This gene encodes a member of the serine/threonine protein kinase family, that is highly similar to yeast SPS1/STE20 kinase. Yeast SPS1/STE20 functions near the beginning of the MAP kinase signal cascades that is essential for yeast pheromone response. This kinase was shown to activate Jun kinase in mammalian cells, which suggested a role in stress response. Two alternatively spliced transcript variants encoding the same protein have been described for this gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:12739',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'WBP4',\n",
       "  'name': 'WW domain binding protein 4',\n",
       "  'synonyms': 'FBP21',\n",
       "  'description': 'This gene encodes WW domain-containing binding protein 4. The WW domain represents a small and compact globular structure that interacts with proline-rich ligands. This encoded protein is a general spliceosomal protein that may play a role in cross-intron bridging of U1 and U2 snRNPs in the spliceosomal complex A. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:10379',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MRPL3',\n",
       "  'name': 'mitochondrial ribosomal protein L3',\n",
       "  'synonyms': 'COXPD9|MRL3|RPML3',\n",
       "  'description': 'Mammalian mitochondrial ribosomal proteins are encoded by nuclear genes and help in protein synthesis within the mitochondrion. Mitochondrial ribosomes (mitoribosomes) consist of a small 28S subunit and a large 39S subunit. They have an estimated 75% protein to rRNA composition compared to prokaryotic ribosomes, where this ratio is reversed. Another difference between mammalian mitoribosomes and prokaryotic ribosomes is that the latter contain a 5S rRNA. Among different species, the proteins comprising the mitoribosome differ greatly in sequence, and sometimes in biochemical properties, which prevents easy recognition by sequence homology. This gene encodes a 39S subunit protein that belongs to the L3P ribosomal protein family. A pseudogene corresponding to this gene is found on chromosome 13q. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:13232',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RASSF8',\n",
       "  'name': 'Ras association domain family member 8',\n",
       "  'synonyms': 'C12orf2|HOJ1',\n",
       "  'description': 'This gene encodes a member of the Ras-assocation domain family (RASSF) of tumor suppressor proteins. This gene is essential for maintaining adherens junction function in epithelial cells and has a role in epithelial cell migration. It is a lung tumor suppressor gene candidate. A chromosomal translocation t(12;22)(p11.2;q13.3) leading to the fusion of this gene and the FBLN1 gene is found in a complex type of synpolydactyly. Multiple alternatively spliced transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, May 2011].'},\n",
       " {'id': 'HGNC:21082',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SEC63',\n",
       "  'name': 'SEC63 homolog, protein translocation regulator',\n",
       "  'synonyms': 'DNAJC23|ERdj2|PCLD2|PRO2507|SEC63L',\n",
       "  'description': 'The Sec61 complex is the central component of the protein translocation apparatus of the endoplasmic reticulum (ER) membrane. The protein encoded by this gene and SEC62 protein are found to be associated with ribosome-free SEC61 complex. It is speculated that Sec61-Sec62-Sec63 may perform post-translational protein translocation into the ER. The Sec61-Sec62-Sec63 complex might also perform the backward transport of ER proteins that are subject to the ubiquitin-proteasome-dependent degradation pathway. The encoded protein is an integral membrane protein located in the rough ER. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:2713',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DCTN3',\n",
       "  'name': 'dynactin subunit 3',\n",
       "  'synonyms': 'DCTN-22|DCTN22',\n",
       "  'description': 'This gene encodes the smallest subunit of dynactin, a macromolecular complex consisting of 10 subunits ranging in size from 22 to 150 kD. Dynactin binds to both microtubules and cytoplasmic dynein. It is involved in a diverse array of cellular functions, including ER-to-Golgi transport, the centripetal movement of lysosomes and endosomes, spindle formation, cytokinesis, chromosome movement, nuclear positioning, and axonogenesis. This subunit, like most other dynactin subunits, exists only as a part of the dynactin complex. It is primarily an alpha-helical protein with very little coiled coil, and binds directly to the largest subunit (p150) of dynactin. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Jul 2013].'},\n",
       " {'id': 'HGNC:7227',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MRAS',\n",
       "  'name': 'muscle RAS oncogene homolog',\n",
       "  'synonyms': 'M-RAs|NS11|R-RAS3|RRAS3',\n",
       "  'description': 'This gene encodes a member of the Ras family of small GTPases. These membrane-associated proteins function as signal transducers in multiple processes including cell growth and differentiation, and dysregulation of Ras signaling has been associated with many types of cancer. The encoded protein may play a role in the tumor necrosis factor-alpha and MAP kinase signaling pathways. Alternatively spliced transcript variants encoding multiple isoforms have been observed for this gene. [provided by RefSeq, Nov 2011].'},\n",
       " {'id': 'HGNC:29186',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ANKRD26',\n",
       "  'name': 'ankyrin repeat domain 26',\n",
       "  'synonyms': 'THC2|bA145E8.1',\n",
       "  'description': 'This gene encodes a protein containing N-terminal ankyrin repeats which function in protein-protein interactions. Mutations in this gene are associated with autosomal dominant thrombocytopenia-2. Pseudogenes of this gene are found on chromosome 7, 10, 13 and 16. Multiple transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Dec 2011].'},\n",
       " {'id': 'HGNC:6890',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MAPRE1',\n",
       "  'name': 'microtubule associated protein RP/EB family member 1',\n",
       "  'synonyms': 'EB1',\n",
       "  'description': 'The protein encoded by this gene was first identified by its binding to the APC protein which is often mutated in familial and sporadic forms of colorectal cancer. This protein localizes to microtubules, especially the growing ends, in interphase cells. During mitosis, the protein is associated with the centrosomes and spindle microtubules. The protein also associates with components of the dynactin complex and the intermediate chain of cytoplasmic dynein. Because of these associations, it is thought that this protein is involved in the regulation of microtubule structures and chromosome stability. This gene is a member of the RP/EB family. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:19685',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SEPHS1',\n",
       "  'name': 'selenophosphate synthetase 1',\n",
       "  'synonyms': 'SELD|SPS|SPS1',\n",
       "  'description': 'This gene encodes an enzyme that synthesizes selenophosphate from selenide and ATP. Selenophosphate is the selenium donor used to synthesize selenocysteine, which is co-translationally incorporated into selenoproteins at in-frame UGA codons. [provided by RefSeq, Sep 2010].'},\n",
       " {'id': 'HGNC:18410',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ZNF292',\n",
       "  'name': 'zinc finger protein 292',\n",
       "  'synonyms': 'MRD63|MRD64|Nbla00365|ZFP292|ZN-16|Zn-15|bA393I2.3',\n",
       "  'description': 'This gene encodes a growth hormone-dependent, zinc finger transcription factor that functions as a tumor suppressor. Naturally occurring mutations in this gene are associated with gastric cancer, colorectal cancer, and chronic lymphocytic leukemia. [provided by RefSeq, May 2017].'},\n",
       " {'id': 'HGNC:23293',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'WAPL',\n",
       "  'name': 'WAPL cohesin release factor',\n",
       "  'synonyms': 'FOE|KIAA0261|WAPAL',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:445',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SETX',\n",
       "  'name': 'senataxin',\n",
       "  'synonyms': 'ALS4|AOA2|SCAN2|SCAR1|Sen1|bA479K20.2',\n",
       "  'description': 'This gene encodes a protein named for its homology to the Sen1p protein of fungi which has RNA helicase activity encoded by a domain at the C-terminal end of the protein. The protein encoded by this gene contains a DNA/RNA helicase domain at its C-terminal end which suggests that it may be involved in both DNA and RNA processing. Mutations in this gene have been associated with ataxia-ocular apraxia-2 (AOA2) and an autosomal dominant form of juvenile amyotrophic lateral sclerosis (ALS4). [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:17071',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'KDM4C',\n",
       "  'name': 'lysine demethylase 4C',\n",
       "  'synonyms': 'GASC1|JHDM3C|JMJD2C|TDRD14C',\n",
       "  'description': 'This gene is a member of the Jumonji domain 2 (JMJD2) family. The encoded protein is a trimethylation-specific demethylase, and converts specific trimethylated histone residues to the dimethylated form. This enzymatic action regulates gene expression and chromosome segregation. Chromosomal aberrations and changes in expression of this gene may be found in tumor cells. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Jan 2015].'},\n",
       " {'id': 'HGNC:19338',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CDK19',\n",
       "  'name': 'cyclin dependent kinase 19',\n",
       "  'synonyms': 'CDC2L6|CDK11|DEE87|EIEE87|bA346C16.3',\n",
       "  'description': 'This gene encodes a protein that is one of the components of the Mediator co-activator complex. The Mediator complex is a multi-protein complex required for transcriptional activation by DNA binding transcription factors of genes transcribed by RNA polymerase II. The protein encoded by this gene is similar to cyclin-dependent kinase 8 which can also be a component of the Mediator complex. Alternative splicing results in multiple transcript variants encoding distinct isoforms. [provided by RefSeq, Jul 2014].'},\n",
       " {'id': 'HGNC:17078',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CLASP2',\n",
       "  'name': 'cytoplasmic linker associated protein 2',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:29113',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GRAMD4',\n",
       "  'name': 'GRAM domain containing 4',\n",
       "  'synonyms': 'DIP',\n",
       "  'description': 'GRAMD4 is a mitochondrial effector of E2F1 (MIM 189971)-induced apoptosis (Stanelle et al., 2005 [PubMed 15565177]).[supplied by OMIM, Jan 2011].'},\n",
       " {'id': 'HGNC:23697',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PHLDB1',\n",
       "  'name': 'pleckstrin homology like domain family B member 1',\n",
       "  'synonyms': 'LL5A',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:17084',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SYNE2',\n",
       "  'name': 'spectrin repeat containing nuclear envelope protein 2',\n",
       "  'synonyms': 'EDMD5|KASH2|NUA|NUANCE|Nesp2|Nesprin-2|SYNE-2|TROPH',\n",
       "  'description': 'The protein encoded by this gene is a nuclear outer membrane protein that binds cytoplasmic F-actin. This binding tethers the nucleus to the cytoskeleton and aids in the maintenance of the structural integrity of the nucleus. Several transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Mar 2009].'},\n",
       " {'id': 'HGNC:648',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ARC',\n",
       "  'name': 'activity regulated cytoskeleton associated protein',\n",
       "  'synonyms': 'Arg3.1|hArc',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:14132',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DOCK9',\n",
       "  'name': 'dedicator of cytokinesis 9',\n",
       "  'synonyms': 'ZIZ1|ZIZIMIN1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:41',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ABCB10',\n",
       "  'name': 'ATP binding cassette subfamily B member 10',\n",
       "  'synonyms': 'EST20237|M-ABC2|MTABC2',\n",
       "  'description': 'The membrane-associated protein encoded by this gene is a member of the superfamily of ATP-binding cassette (ABC) transporters.  ABC proteins transport various molecules across extra- and intra-cellular membranes.  ABC genes are divided into seven distinct subfamilies (ABC1, MDR/TAP, MRP, ALD, OABP, GCN20, White).  This protein is a member of the MDR/TAP subfamily. Members of the MDR/TAP subfamily are involved in multidrug resistance.  The function of this mitochondrial protein is unknown. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:1555',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CBX5',\n",
       "  'name': 'chromobox 5',\n",
       "  'synonyms': 'HEL25|HP1|HP1A',\n",
       "  'description': 'This gene encodes a highly conserved nonhistone protein, which is a member of the heterochromatin protein family. The protein is enriched in the heterochromatin and associated with centromeres. The protein has a single N-terminal chromodomain which can bind to histone proteins via methylated lysine residues, and a C-terminal chromo shadow-domain (CSD) which is responsible for the homodimerization and interaction with a number of chromatin-associated nonhistone proteins. The encoded product is involved in the formation of functional kinetochore through interaction with essential kinetochore proteins. The gene has a pseudogene located on chromosome 3. Multiple alternatively spliced variants, encoding the same protein, have been identified. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:8848',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PES1',\n",
       "  'name': 'pescadillo ribosomal biogenesis factor 1',\n",
       "  'synonyms': 'NOP7|PES',\n",
       "  'description': 'This gene encodes a nuclear protein that contains a breast cancer associated gene 1 (BRCA1) C-terminal interaction domain. The encoded protein interacts with BOP1 and WDR12 to form the PeBoW complex, which plays a critical role in cell proliferation via pre-rRNA processing and 60S ribosomal subunit maturation. Expression of this gene may play an important role in breast cancer proliferation and tumorigenicity. Alternatively spliced transcript variants encoding multiple isoforms have been observed for this gene. Pseudogenes of this gene are located on the long arm of chromosome 4 and the short arm of chromosome 9. [provided by RefSeq, Aug 2011].'},\n",
       " {'id': 'HGNC:17101',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SUZ12',\n",
       "  'name': 'SUZ12 polycomb repressive complex 2 subunit',\n",
       "  'synonyms': 'CHET9|IMMAS|JJAZ1',\n",
       "  'description': 'This zinc finger gene has been identified at the breakpoints of a recurrent chromosomal translocation reported in endometrial stromal sarcoma. Recombination of these breakpoints results in the fusion of this gene and JAZF1. The protein encoded by this gene contains a zinc finger domain in the C terminus of the coding region. [provided by RefSeq, Jul 2009].'},\n",
       " {'id': 'HGNC:17102',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ARHGAP45',\n",
       "  'name': 'Rho GTPase activating protein 45',\n",
       "  'synonyms': 'HA-1|HLA-HA1|HMHA1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:30035',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PIK3R5',\n",
       "  'name': 'phosphoinositide-3-kinase regulatory subunit 5',\n",
       "  'synonyms': 'F730038I15Rik|FOAP-2|P101-PI3K|p101',\n",
       "  'description': 'Phosphatidylinositol 3-kinases (PI3Ks) phosphorylate the inositol ring of phosphatidylinositol at the 3-prime position, and play important roles in cell growth, proliferation, differentiation, motility, survival and intracellular trafficking. The PI3Ks are divided into three classes: I, II and III, and only the class I PI3Ks are involved in oncogenesis. This gene encodes the 101 kD regulatory subunit of the class I PI3K gamma complex, which is a dimeric enzyme, consisting of a 110 kD catalytic subunit gamma and a regulatory subunit of either 55, 87 or 101 kD. This protein recruits the catalytic subunit from the cytosol to the plasma membrane through high-affinity interaction with G-beta-gamma proteins. Multiple alternatively spliced transcript variants encoding two distinct isoforms have been found. [provided by RefSeq, Oct 2011].'},\n",
       " {'id': 'NCBIGene:23578',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PEF1',\n",
       "  'name': 'penta-EF-hand domain containing 1',\n",
       "  'synonyms': 'ABP32|PEF1A',\n",
       "  'description': 'This gene encodes a calcium-binding protein belonging to the penta-EF-hand protein family. The encoded protein has been shown to form a heterodimer with the programmed cell death 6 gene product and may modulate its function in Ca(2+) signaling. Alternative splicing results in multiple transcript variants and a pseudogene has been identified on chromosome 1.[provided by RefSeq, May 2010].'},\n",
       " {'id': 'HGNC:17150',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CARHSP1',\n",
       "  'name': 'calcium regulated heat stable protein 1',\n",
       "  'synonyms': 'CRHSP-24|CRHSP24|CSDC1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:30073',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'POLA2',\n",
       "  'name': 'DNA polymerase alpha 2, accessory subunit',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:1191',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'EID1',\n",
       "  'name': 'EP300 interacting inhibitor of differentiation 1',\n",
       "  'synonyms': 'C15orf3|CRI1|EID-1|IRO45620|PNAS-22|PTD014|RBP21',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:17165',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TFIP11',\n",
       "  'name': 'tuftelin interacting protein 11',\n",
       "  'synonyms': 'NTR1|STIP|STIP-1|Spp382|TIP39|bK445C9.6',\n",
       "  'description': 'This gene encodes a protein component of the spliceosome that promotes the release of the lariat-intron during late-stage splicing through the recruitment of a pre-mRNA splicing factor called DEAH-box helicase 15. The encoded protein contains a G-patch domain, a hallmark of RNA-processing proteins, that binds DEAH-box helicase 15. This protein contains an atypical nuclear localization sequence as well as a nuclear speckle-targeting sequence, enabling it to localize to distinct speckled regions within the cell nucleus. Polymorphisms in this gene are associated with dental caries suggesting a role in amelogenesis. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Nov 2016].'},\n",
       " {'id': 'HGNC:24999',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ARMC8',\n",
       "  'name': 'armadillo repeat containing 8',\n",
       "  'synonyms': 'GID5|HSPC056|S863-2|VID28',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:24530',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TMEM186',\n",
       "  'name': 'transmembrane protein 186',\n",
       "  'synonyms': 'C16orf51',\n",
       "  'description': 'This gene encodes a potential transmembrane protein. [provided by RefSeq, Dec 2012].'},\n",
       " {'id': 'HGNC:24542',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DPCD',\n",
       "  'name': 'deleted in primary ciliary dyskinesia homolog (mouse)',\n",
       "  'synonyms': nan,\n",
       "  'description': 'This gene in mouse encodes a protein that may be involved in the generation and maintenance of ciliated cells. In mouse, expression of this gene increases during ciliated cell differentiation, and disruption of this gene has been linked to primary ciliary dyskinesia. [provided by RefSeq, Jul 2016].'},\n",
       " {'id': 'HGNC:29931',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GNL3',\n",
       "  'name': 'G protein nucleolar 3',\n",
       "  'synonyms': 'C77032|E2IG3|NNP47|NS',\n",
       "  'description': 'The protein encoded by this gene may interact with p53 and may be involved in tumorigenesis. The encoded protein also appears to be important for stem cell proliferation. This protein is found in both the nucleus and nucleolus. Three transcript variants encoding two different isoforms have been found for this gene. [provided by RefSeq, Nov 2010].'},\n",
       " {'id': 'HGNC:30844',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SEZ6L2',\n",
       "  'name': 'seizure related 6 homolog like 2',\n",
       "  'synonyms': 'BSRPA|PSK-1',\n",
       "  'description': 'This gene encodes a seizure-related protein that is localized on the cell surface. The gene is located in a region of chromosome 16p11.2 that is thought to contain candidate genes for autism spectrum disorders (ASD), though there is no evidence directly implicating this gene in ASD. Increased expression of this gene has been found in lung cancers, and the protein is therefore considered to be a novel prognostic marker for lung cancer. Alternative splicing of this gene results in multiple transcript variants. [provided by RefSeq, Aug 2011].'},\n",
       " {'id': 'HGNC:29990',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NUPR1',\n",
       "  'name': 'nuclear protein 1, transcriptional regulator',\n",
       "  'synonyms': 'COM1|P8',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:24123',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'B9D1',\n",
       "  'name': 'B9 domain containing 1',\n",
       "  'synonyms': 'B9|EPPB9|JBTS27|MKS9|MKSR-1|MKSR1',\n",
       "  'description': 'This gene encodes a B9 domain-containing protein, one of several that are involved in ciliogenesis. Alterations in expression of this gene have been found in a family with Meckel syndrome. Meckel syndrome has been associated with at least six different genes. This gene is located within the Smith-Magenis syndrome region on chromosome 17. [provided by RefSeq, Mar 2016].'},\n",
       " {'id': 'HGNC:17304',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TAF5L',\n",
       "  'name': 'TATA-box binding protein associated factor 5 like',\n",
       "  'synonyms': 'PAF65B',\n",
       "  'description': 'The product of this gene belongs to the WD-repeat TAF5 family of proteins. This gene encodes a protein that is a component of the PCAF histone acetylase complex. The PCAF histone acetylase complex, which is composed of more than 20 polypeptides some of which are TAFs, is required for myogenic transcription and differentiation. TAFs may participate in basal transcription, serve as coactivators, function in promoter recognition or modify general transcription factors to facilitate complex assembly and transcription initiation. The encoded protein is structurally similar to one of the histone-like TAFs, TAF5. Alternatively spliced transcript variants encoding different isoforms have been identified for this gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:23530',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PALD1',\n",
       "  'name': 'phosphatase domain containing paladin 1',\n",
       "  'synonyms': 'KIAA1274|PALD',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:13488',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'VPS4A',\n",
       "  'name': 'vacuolar protein sorting 4 homolog A',\n",
       "  'synonyms': 'SKD1|SKD1A|SKD2|VPS4|VPS4-1',\n",
       "  'description': 'The protein encoded by this gene is a member of the AAA protein family (ATPases associated with diverse cellular activities), and is the homolog of the yeast Vps4 protein. In humans, two paralogs of the yeast protein have been identified. The former share a high degree of aa sequence similarity with each other, and also with yeast Vps4 and mouse Skd1 proteins. The mouse Skd1 (suppressor of K+ transport defect 1) has been shown to be really an yeast Vps4 ortholog. Functional studies indicate that both human paralogs associate with the endosomal compartments, and are involved in intracellular protein trafficking, similar to Vps4 protein in yeast. The gene encoding this paralog has been mapped to chromosome 16; the gene for the other resides on chromosome 18. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:25223',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'COQ2',\n",
       "  'name': 'coenzyme Q2, polyprenyltransferase',\n",
       "  'synonyms': 'CL640|COQ10D1|MSA1|PHB:PPT',\n",
       "  'description': 'This gene encodes an enzyme that functions in the final steps in the biosynthesis of CoQ (ubiquinone), a redox carrier in the mitochondrial respiratory chain and a lipid-soluble antioxidant. This enzyme, which is part of the coenzyme Q10 pathway, catalyzes the prenylation of parahydroxybenzoate with an all-trans polyprenyl group. Mutations in this gene cause coenzyme Q10 deficiency, a mitochondrial encephalomyopathy, and also COQ2 nephropathy, an inherited form of mitochondriopathy with primary renal involvement. [provided by RefSeq, Oct 2009].'},\n",
       " {'id': 'HGNC:11995',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TOR1B',\n",
       "  'name': 'torsin family 1 member B',\n",
       "  'synonyms': 'DQ1',\n",
       "  'description': 'The protein encoded by this gene is an ATPase found primarily in the endoplasmic reticulum and nuclear envelope. This gene has a highly-similar neighboring gene, TOR1A, that encodes a protein that is likely to interact in a complex with this protein. Finally, this protein may act as a chaperone and play a role in maintaining the integrity of the nuclear envelope and endoplasmic reticulum. Several transcript variants, some protein-coding and others non-protein coding, have been found for this gene. [provided by RefSeq, Dec 2015].'},\n",
       " {'id': 'HGNC:30204',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'C11orf54',\n",
       "  'name': 'chromosome 11 open reading frame 54',\n",
       "  'synonyms': 'PTD012|PTOD012',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:23401',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SPCS1',\n",
       "  'name': 'signal peptidase complex subunit 1',\n",
       "  'synonyms': 'HSPC033|SPC1|SPC12|YJR010C-A',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:14493',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MRPL42',\n",
       "  'name': 'mitochondrial ribosomal protein L42',\n",
       "  'synonyms': 'HSPC204|L31MT|L42MT|MRP-L31|MRP-L42|MRP-S32|MRPL31|MRPS32|PTD007|RPML31|S32MT',\n",
       "  'description': 'Mammalian mitochondrial ribosomal proteins are encoded by nuclear genes and help in protein synthesis within the mitochondrion. Mitochondrial ribosomes (mitoribosomes) consist of a small 28S subunit and a large 39S subunit. They have an estimated 75% protein to rRNA composition compared to prokaryotic ribosomes, where this ratio is reversed. Another difference between mammalian mitoribosomes and prokaryotic ribosomes is that the latter contain a 5S rRNA. Among different species, the proteins comprising the mitoribosome differ greatly in sequence, and sometimes in biochemical properties, which prevents easy recognition by sequence homology. This gene encodes a protein identified as belonging to both the 28S and the 39S subunits. Alternative splicing results in multiple transcript variants. Pseudogenes corresponding to this gene are found on chromosomes 4q, 6p, 6q, 7p, and 15q. [provided by RefSeq, May 2011].'},\n",
       " {'id': 'HGNC:30959',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ZC3H7A',\n",
       "  'name': 'zinc finger CCCH-type containing 7A',\n",
       "  'synonyms': 'HSPC055|ZC3H7|ZC3HDC7',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:25002',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CCDC113',\n",
       "  'name': 'coiled-coil domain containing 113',\n",
       "  'synonyms': 'HSPC065',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:25013',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CNIH4',\n",
       "  'name': 'cornichon family AMPA receptor auxiliary protein 4',\n",
       "  'synonyms': 'CNIH-4|CNIH2|HSPC163',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:17679',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RANGRF',\n",
       "  'name': 'RAN guanine nucleotide release factor',\n",
       "  'synonyms': 'HSPC165|HSPC236|MOG1|RANGNRF',\n",
       "  'description': 'This gene encodes a protein that has been shown to function as a guanine nucleotide release factor in mouse and to regulate the expression and function of the Nav1.5 cardiac sodium channel in human. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Apr 2010].'},\n",
       " {'id': 'HGNC:13707',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SCG3',\n",
       "  'name': 'secretogranin III',\n",
       "  'synonyms': 'SGIII',\n",
       "  'description': 'The protein encoded by this gene is a member of the chromogranin/secretogranin family of neuroendocrine secretory proteins. Granins may serve as precursors for biologically active peptides. Some granins have been shown to function as helper proteins in sorting and proteolytic processing of prohormones; however, the function of this protein is unknown. Two transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Sep 2009].'},\n",
       " {'id': 'HGNC:11873',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TMOD3',\n",
       "  'name': 'tropomodulin 3',\n",
       "  'synonyms': 'UTMOD',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:12506',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'UBN1',\n",
       "  'name': 'ubinuclein 1',\n",
       "  'synonyms': 'VT|VT4',\n",
       "  'description': 'Cellular senescence is a hallmark of tumor suppression and tissue aging. Senescent cells contain domains of heterochromatin, called senescence-associated heterochromatin foci (SAHF), that repress proliferation-promoting genes. The protein encoded by this gene binds to proliferation-promoting genes and is required for SAHF formation, enhancing methylation of histone H3. [provided by RefSeq, Oct 2016].'},\n",
       " {'id': 'HGNC:11218',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SPANXA1',\n",
       "  'name': 'sperm protein associated with the nucleus, X-linked, family member A1',\n",
       "  'synonyms': 'CT11.1|CT11.3|NAP-X|SPAN-X|SPAN-Xa|SPAN-Xb|SPANX|SPANX-A',\n",
       "  'description': 'Temporally regulated transcription and translation of several testis-specific genes is required to initiate the series of molecular and morphological changes in the male germ cell lineage necessary for the formation of mature spermatozoa. This gene is a member of the SPANX family of cancer/testis-associated genes, which are located in a cluster on chromosome X. The SPANX genes encode differentially expressed testis-specific proteins that localize to various subcellular compartments. This particular gene maps to chromosome X in a head-to-head orientation with SPANX family member A2, which appears to be a duplication of the A1 locus. The protein encoded by this gene targets to the nucleus where it associates with nuclear vacuoles and the redundant nuclear envelope. Based on its association with these poorly characterized regions of the sperm nucleus, this protein provides a biochemical marker to study unique structures in spermatazoa while attempting to further define its role in spermatogenesis. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:13326',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HUNK',\n",
       "  'name': 'hormonally up-regulated Neu-associated kinase',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:12759',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'WRAP73',\n",
       "  'name': 'WD repeat containing, antisense to TP73',\n",
       "  'synonyms': 'WDR8',\n",
       "  'description': 'This gene encodes a member of the WD repeat protein family. WD repeats are minimally conserved regions of approximately 40 amino acids typically bracketed by gly-his and trp-asp (GH-WD), which may facilitate formation of heterotrimeric or multiprotein complexes. Members of this family are involved in a variety of cellular processes, including cell cycle progression, signal transduction, apoptosis, and gene regulation. Studies of the related mouse protein suggest that the encoded protein may play a role in the process of ossification. [provided by RefSeq, Mar 2009].'},\n",
       " {'id': 'HGNC:18130',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SYCP3',\n",
       "  'name': 'synaptonemal complex protein 3',\n",
       "  'synonyms': 'COR1|RPRGL4|SCP3|SPGF4',\n",
       "  'description': 'This gene encodes an essential structural component of the synaptonemal complex. This complex is involved in synapsis, recombination and segregation of meiotic chromosomes. Mutations in this gene are associated with azoospermia in males and susceptibility to pregnancy loss in females. Alternate splicing results in multiple transcript variants that encode the same protein. [provided by RefSeq, May 2010].'},\n",
       " {'id': 'HGNC:23658',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DELEC1',\n",
       "  'name': 'deleted in esophageal cancer 1',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:17937',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CUZD1',\n",
       "  'name': 'CUB and zona pellucida like domains 1',\n",
       "  'synonyms': 'ERG-1|ERG1|ITMAP1|UO-44',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:24259',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TPRKB',\n",
       "  'name': 'TP53RK binding protein',\n",
       "  'synonyms': 'CGI-121|CGI121|GAMOS5',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:24809',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SSUH2',\n",
       "  'name': 'ssu-2 homolog',\n",
       "  'synonyms': 'C3orf32|SSU-2|fls485',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:14014',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MEMO1',\n",
       "  'name': 'Methylation modifier for class I HLA',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:10833',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SH3GLB1',\n",
       "  'name': 'SH3 domain containing GRB2 like, endophilin B1',\n",
       "  'synonyms': 'Bif-1|CGI-61|PPP1R70|dJ612B15.2',\n",
       "  'description': 'This gene encodes a SRC homology 3 domain-containing protein. The encoded protein interacts with the proapoptotic member of the Bcl-2 family, Bcl-2-associated X protein (Bax) and may be involved in regulating apoptotic signaling pathways. This protein may also be involved in maintaining mitochondrial morphology. Alternate splicing results in multiple transcript variants. [provided by RefSeq, Sep 2011].'},\n",
       " {'id': 'NCBIGene:51145',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'LOC51145',\n",
       "  'name': 'uncharacterized LOC51145',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:29526',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HSPA14',\n",
       "  'name': 'heat shock protein family A (Hsp70) member 14',\n",
       "  'synonyms': 'HSP70-4|HSP70L1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:28035',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TMEM69',\n",
       "  'name': 'transmembrane protein 69',\n",
       "  'synonyms': 'C1orf154',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:16692',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CD320',\n",
       "  'name': 'CD320 molecule',\n",
       "  'synonyms': '8D6|8D6A|TCBLR|TCN2R',\n",
       "  'description': 'This gene encodes the transcobalamin receptor that is expressed at the cell surface. It mediates the cellular uptake of transcobalamin bound cobalamin (vitamin B12), and is involved in B-cell proliferation and immunoglobulin secretion. Mutations in this gene are associated with methylmalonic aciduria. Alternatively spliced transcript variants encoding different isoforms have been found for this gene.[provided by RefSeq, Jan 2011].'},\n",
       " {'id': 'HGNC:20373',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SPG21',\n",
       "  'name': 'SPG21 abhydrolase domain containing, maspardin',\n",
       "  'synonyms': 'ABHD21|ACP33|BM-019|GL010|MAST',\n",
       "  'description': 'The protein encoded by this gene binds to the hydrophobic C-terminal amino acids of CD4 which are involved in repression of T cell activation. The interaction with CD4 is mediated by the noncatalytic alpha/beta hydrolase fold domain of this protein. It is thus proposed that this gene product modulates the stimulatory activity of CD4. Mutations in this gene are associated with autosomal recessive spastic paraplegia 21 (SPG21), also known as mast syndrome. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Aug 2014].'},\n",
       " {'id': 'HGNC:29653',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ZNF771',\n",
       "  'name': 'zinc finger protein 771',\n",
       "  'synonyms': 'DSC43',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:18137',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CHST15',\n",
       "  'name': 'carbohydrate sulfotransferase 15',\n",
       "  'synonyms': 'BRAG|GALNAC4S-6ST',\n",
       "  'description': 'Chondroitin sulfate (CS) is a glycosaminoglycan which is an important structural component of the extracellular matrix and which links to proteins to form proteoglycans. Chondroitin sulfate E (CS-E) is an isomer of chondroitin sulfate in which the C-4 and C-6 hydroxyl groups are sulfated. This gene encodes a type II transmembrane glycoprotein that acts as a sulfotransferase to transfer sulfate to the C-6 hydroxal group of chondroitin sulfate. This gene has also been identified as being co-expressed with RAG1 in B-cells and as potentially acting as a B-cell surface signaling receptor. Alternative splicing results in multiple transcript variants encoding distinct isoforms. [provided by RefSeq, Jul 2012].'},\n",
       " {'id': 'HGNC:24328',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NIP7',\n",
       "  'name': 'nucleolar pre-rRNA processing protein NIP7',\n",
       "  'synonyms': 'CGI-37|HSPC031|KD93',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:30235',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NAT8B',\n",
       "  'name': 'N-acetyltransferase 8B',\n",
       "  'synonyms': 'Cml1|Cml6|Nat8|RGD621605',\n",
       "  'description': 'may act as an N-acetyltransferase; Xenopus laevis homolog camello inhibits gastrulation and may regulate cell surface adhesion [RGD, Feb 2006].'},\n",
       " {'id': 'HGNC:26936',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CTDSPL2',\n",
       "  'name': 'CTD small phosphatase like 2',\n",
       "  'synonyms': 'HSPC058|HSPC129|SCP4',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:30826',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NT5DC3',\n",
       "  'name': \"5'-nucleotidase domain containing 3\",\n",
       "  'synonyms': 'TU12B1-TY|TU12B1TY',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:24270',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DPH5',\n",
       "  'name': 'diphthamide biosynthesis 5',\n",
       "  'synonyms': 'AD-018|CGI-30|HSPC143|NPD015',\n",
       "  'description': 'This gene encodes a component of the diphthamide synthesis pathway. Diphthamide is a post-translationally modified histidine residue found only on translation elongation factor 2. It is conserved from archaebacteria to humans, and is targeted by diphtheria toxin and Pseudomonas exotoxin A to halt cellular protein synthesis. The yeast and Chinese hamster homologs of this protein catalyze the trimethylation of the histidine residue on elongation factor 2, resulting in a diphthine moiety that is subsequently amidated to yield diphthamide. Multiple transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:14509',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MRPS23',\n",
       "  'name': 'mitochondrial ribosomal protein S23',\n",
       "  'synonyms': 'CGI-138|COXPD46|HSPC329|MRP-S23',\n",
       "  'description': 'Mammalian mitochondrial ribosomal proteins are encoded by nuclear genes and help in protein synthesis within the mitochondrion. Mitochondrial ribosomes (mitoribosomes) consist of a small 28S subunit and a large 39S subunit. They have an estimated 75% protein to rRNA composition compared to prokaryotic ribosomes, where this ratio is reversed. Another difference between mammalian mitoribosomes and prokaryotic ribosomes is that the latter contain a 5S rRNA. Among different species, the proteins comprising the mitoribosome differ greatly in sequence, and sometimes in biochemical properties, which prevents easy recognition by sequence homology. This gene encodes a 28S subunit protein. A pseudogene corresponding to this gene is found on chromosome 7p. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:1839',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ADA2',\n",
       "  'name': 'adenosine deaminase 2',\n",
       "  'synonyms': 'ADGF|CECR1|IDGFL|PAN|SNEDS|VAIHS',\n",
       "  'description': 'This gene encodes a member of a subfamily of the adenosine deaminase protein family. The encoded protein is one of two adenosine deaminases found in humans, which regulate levels of the signaling molecule, adenosine. The encoded protein is secreted from monocytes undergoing differentiation and may regulate cell proliferation and differentiation. This gene may be responsible for some of the phenotypic features associated with cat eye syndrome. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Aug 2013].'},\n",
       " {'id': 'HGNC:6273',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'KCNK10',\n",
       "  'name': 'potassium two pore domain channel subfamily K member 10',\n",
       "  'synonyms': 'K2p10.1|PPP1R97|TREK-2|TREK2',\n",
       "  'description': 'The protein encoded by this gene belongs to the family of potassium channel proteins containing two pore-forming P domains. This channel is an open rectifier which primarily passes outward current under physiological K+ concentrations, and is stimulated strongly by arachidonic acid and to a lesser degree by membrane stretching, intracellular acidification, and general anaesthetics. Several alternatively spliced transcript variants encoding different isoforms have been identified for this gene. [provided by RefSeq, Sep 2008].'},\n",
       " {'id': 'HGNC:25231',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'YIPF1',\n",
       "  'name': 'Yip1 domain family member 1',\n",
       "  'synonyms': 'DJ167A19.1|FinGER1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:25937',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'C11orf71',\n",
       "  'name': 'chromosome 11 open reading frame 71',\n",
       "  'synonyms': 'URLC7',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:25611',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CCDC93',\n",
       "  'name': 'coiled-coil domain containing 93',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:23471',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ANKRD16',\n",
       "  'name': 'ankyrin repeat domain 16',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:24944',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DDIT4',\n",
       "  'name': 'DNA damage inducible transcript 4',\n",
       "  'synonyms': 'Dig2|REDD-1|REDD1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:24686',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'FBLIM1',\n",
       "  'name': 'filamin binding LIM protein 1',\n",
       "  'synonyms': 'CAL|FBLP-1|FBLP1',\n",
       "  'description': 'This gene encodes a protein with an N-terminal filamin-binding domain, a central proline-rich domain, and, multiple C-terminal LIM domains. This protein localizes at cell junctions and may link cell adhesion structures to the actin cytoskeleton. This protein may be involved in the assembly and stabilization of actin-filaments and likely plays a role in modulating cell adhesion, cell morphology and cell motility. This protein also localizes to the nucleus and may affect cardiomyocyte differentiation after binding with the CSX/NKX2-5 transcription factor. Alternative splicing results in multiple transcript variants encoding different isoforms. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:13862',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'BTG4',\n",
       "  'name': 'BTG anti-proliferation factor 4',\n",
       "  'synonyms': 'APRO3|OOMD8|PC3B',\n",
       "  'description': 'The protein encoded by this gene is a member of the BTG/Tob family. This family has structurally related proteins that appear to have antiproliferative properties. This encoded protein can induce G1 arrest in the cell cycle. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:19323',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DIRAS2',\n",
       "  'name': 'DIRAS family GTPase 2',\n",
       "  'synonyms': 'Di-Ras2',\n",
       "  'description': 'DIRAS2 belongs to a distinct branch of the functionally diverse Ras (see HRAS; MIM 190020) superfamily of monomeric GTPases.[supplied by OMIM, Apr 2004].'},\n",
       " {'id': 'HGNC:21900',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ALKBH4',\n",
       "  'name': 'alkB homolog 4, lysine demethylase',\n",
       "  'synonyms': 'ABH4',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:20286',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TRIT1',\n",
       "  'name': 'tRNA isopentenyltransferase 1',\n",
       "  'synonyms': 'COXPD35|GRO1|IPPT|IPT|IPTase|MOD5|hGRO1',\n",
       "  'description': 'This gene encodes a protein that that is targeted to the mitochondrion and modifies transfer RNAs (tRNAs) by adding a dimethylallyl group onto the adenine at position 37. This modification is important for maintaining the correct reading frame during protein translation. This gene is considered a tumor suppressor and its expression can decrease cell growth. Alternative splicing results in multiple transcripts variants, most of which are likely non-functional. [provided by RefSeq, Aug 2015].'},\n",
       " {'id': 'HGNC:23510',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'WBP1L',\n",
       "  'name': 'WW domain binding protein 1 like',\n",
       "  'synonyms': 'C10orf26|OPA1L|OPAL1',\n",
       "  'description': nan},\n",
       " {'id': 'NCBIGene:54844',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MCM9',\n",
       "  'name': 'minichromosome maintenance 9 homologous recombination repair factor',\n",
       "  'synonyms': 'C6orf61|MCMDC1|ODG4|dJ329L24.1|dJ329L24.3',\n",
       "  'description': 'The protein encoded by this gene is a member of the mini-chromosome maintenance (MCM) protein family that are essential for the initiation of eukaryotic genome replication. Binding of this protein to chromatin has been shown to be a pre-requisite for recruiting the MCM2-7 helicase to DNA replication origins. This protein also binds, and is a positive regulator of, the chromatin licensing and DNA replication factor 1, CDT1. [provided by RefSeq, Nov 2010].'},\n",
       " {'id': 'HGNC:20243',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CYP2W1',\n",
       "  'name': 'cytochrome P450 family 2 subfamily W member 1',\n",
       "  'synonyms': nan,\n",
       "  'description': 'This gene encodes a member of the cytochrome P450 superfamily of enzymes. The cytochrome P450 proteins are monooxygenases which catalyze many reactions involved in drug metabolism and synthesis of cholesterol, steroids and other lipids. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:31092',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TMEM132A',\n",
       "  'name': 'transmembrane protein 132A',\n",
       "  'synonyms': 'GBP|HSPA5BP1',\n",
       "  'description': 'This gene encodes a protein that is highly similar to the rat Grp78-binding protein (GBP). Alternatively spliced transcript variants encoding different isoforms have been described. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:26059',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CZIB',\n",
       "  'name': 'CXXC motif containing zinc binding protein',\n",
       "  'synonyms': 'C1orf123',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:26060',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ACSM5',\n",
       "  'name': 'acyl-CoA synthetase medium chain family member 5',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:26075',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PIH1D1',\n",
       "  'name': 'PIH1 domain containing 1',\n",
       "  'synonyms': 'DNAAF14|MOT48|NOP17|Pih1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:25486',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RADX',\n",
       "  'name': 'RPA1 related single stranded DNA binding protein, X-linked',\n",
       "  'synonyms': 'CXorf57',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:25583',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ODAD2',\n",
       "  'name': 'outer dynein arm docking complex subunit 2',\n",
       "  'synonyms': 'ARMC4|CILD23|gudu',\n",
       "  'description': 'The protein encoded by this gene contains ten Armadillo repeat motifs (ARMs) and one HEAT repeat, and is thought to be involved in ciliary and flagellar movement. This protein has been shown to localize to the ciliary axonemes and at the ciliary base of respiratory cells. Studies indicate that mutations in this gene cause partial outer dynein arm (ODA) defects in respiratory cilia. The cilia of cells with mutations in this gene displayed either reduced ciliary beat frequency and amplitude, or, complete immotility. Some individuals with primary ciliary dyskensia (PCD) have been shown to have mutations in this gene. PCD is characterized by chronic airway disease and left/right body asymmetry defects. Alternative splicing results in multiple transcript variants encoding different isoforms. [provided by RefSeq, Aug 2015].'},\n",
       " {'id': 'HGNC:20188',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DNAAF2',\n",
       "  'name': 'dynein axonemal assembly factor 2',\n",
       "  'synonyms': 'C14orf104|CILD10|KTU|PF13',\n",
       "  'description': 'This gene encodes a highly conserved protein involved in the preassembly of dynein arm complexes which power cilia. These complexes are found in some cilia and are assembled in the cytoplasm prior to transport for cilia formation. Mutations in this gene have been associated with primary ciliary dyskinesia. Multiple transcript variants encoding different isoforms have been found for this gene.[provided by RefSeq, Oct 2009].'},\n",
       " {'id': 'HGNC:18703',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'FAIM',\n",
       "  'name': 'Fas apoptotic inhibitory molecule',\n",
       "  'synonyms': 'FAIM1',\n",
       "  'description': 'The protein encoded by this gene protects against death receptor-triggered apoptosis and regulates B-cell signaling and differentiation. Several transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Jul 2011].'},\n",
       " {'id': 'NCBIGene:55265',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'KIF24',\n",
       "  'name': 'kinesin family member 24',\n",
       "  'synonyms': 'C9orf48|bA571F15.4',\n",
       "  'description': 'This gene encodes a member of the kinesin superfamily of microtubule-based motor proteins which are involved in the intracellular transport of membranous organelles, protein complexes, and mRNAs. They also play critical roles in mitosis, morphogenesis, and signal transduction. The encoded protein contains an N-terminal sterile alpha motif (SAM) domain and an ATP-binding kinesin motor domain. It binds centriolar coiled coil protein 110 and centrosomal protein 97 and localizes to the mother centriole to regulate ciliogenesis by controlling microtubule polymerization. [provided by RefSeq, Mar 2017].'},\n",
       " {'id': 'HGNC:25612',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ZNF654',\n",
       "  'name': 'zinc finger protein 654',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:17174',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NAGK',\n",
       "  'name': 'N-acetylglucosamine kinase',\n",
       "  'synonyms': 'GNK|HSA242910',\n",
       "  'description': 'This gene encodes a member of the N-acetylhexosamine kinase family. The encoded protein catalyzes the conversion of N-acetyl-D-glucosamine to N-acetyl-D-glucosamine 6-phosphate, and is the major mammalian enzyme which recovers amino sugars. [provided by RefSeq, Nov 2011].'},\n",
       " {'id': 'HGNC:25265',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ZCCHC8',\n",
       "  'name': 'zinc finger CCHC-type containing 8',\n",
       "  'synonyms': 'PFBMFT5',\n",
       "  'description': \"This gene encodes a scaffold protein which serves as an assessory factor to the nuclear RNA exosome complex. The encoded protein forms a trimeric human nuclear exosome targeting (NEXT) complex, together with hMTR4 and the RNA-binding factor RBM7 which promotes the exosomal degradation of non-coding promoter-upstream transcripts, enhancer RNAs and 3'-extended products of histone- and small nuclear RNA transcription. This complex is also thought to recruit the exosome to degrade intronic RNAs via its interaction with both the exosome and the spliceosome. It contains both an N-terminal zinc-knuckle domain and a C-terminal proline-rich domain. [provided by RefSeq, Apr 2017].\"},\n",
       " {'id': 'HGNC:25525',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NDC1',\n",
       "  'name': 'NDC1 transmembrane nucleoporin',\n",
       "  'synonyms': 'NET3|TMEM48',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:18269',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'BTBD7',\n",
       "  'name': 'BTB domain containing 7',\n",
       "  'synonyms': 'FUP1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:18271',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ENAH',\n",
       "  'name': 'ENAH actin regulator',\n",
       "  'synonyms': 'ENA|MENA|NDPP1',\n",
       "  'description': 'This gene encodes a member of the enabled/ vasodilator-stimulated phosphoprotein. Members of this gene family are involved in actin-based motility. This protein is involved in regulating the assembly of actin filaments and modulates cell adhesion and motility. Alternate splice variants of this gene have been correlated with tumor invasiveness in certain tissues and these variants may serve as prognostic markers. A pseudogene of this gene is found on chromosome 3. [provided by RefSeq, Sep 2016].'},\n",
       " {'id': 'HGNC:14456',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'H2AJ',\n",
       "  'name': 'H2A.J histone',\n",
       "  'synonyms': 'H2AFJ',\n",
       "  'description': 'Histones are basic nuclear proteins that are responsible for the nucleosome structure of the chromosomal fiber in eukaryotes. Nucleosomes consist of approximately 146 bp of DNA wrapped around a histone octamer composed of pairs of each of the four core histones (H2A, H2B, H3, and H4). The chromatin fiber is further compacted through the interaction of a linker histone, H1, with the DNA between the nucleosomes to form higher order chromatin structures. This gene is located on chromosome 12 and encodes a replication-independent histone that is a variant H2A histone. The protein is divergent at the C-terminus compared to the consensus H2A histone family member. This gene also encodes an antimicrobial peptide with antibacterial and antifungal activity.[provided by RefSeq, Oct 2015].'},\n",
       " {'id': 'HGNC:18273',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TRERF1',\n",
       "  'name': 'transcriptional regulating factor 1',\n",
       "  'synonyms': 'BCAR2|HSA277276|RAPA|TREP132|TReP-132|dJ139D8.5',\n",
       "  'description': 'This gene encodes a zinc-finger transcriptional regulating protein which interacts with CBP/p300 to regulate the human gene CYP11A1. Alternative splicing results in multiple transcript variants encoding different isoforms. [provided by RefSeq, Jul 2014].'},\n",
       " {'id': 'HGNC:23732',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PPP2R2D',\n",
       "  'name': 'protein phosphatase 2 regulatory subunit Bdelta',\n",
       "  'synonyms': 'B55D|B55delta|MDS026',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:28046',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ZMAT5',\n",
       "  'name': 'zinc finger matrin-type 5',\n",
       "  'synonyms': 'SNRNP20|U11/U12-20K|ZC3H19',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:20392',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SULF2',\n",
       "  'name': 'sulfatase 2',\n",
       "  'synonyms': 'HSULF-2',\n",
       "  'description': 'Heparan sulfate proteoglycans (HSPGs) act as coreceptors for numerous heparin-binding growth factors and cytokines and are involved in cell signaling. Heparan sulfate 6-O-endosulfatases, such as SULF2, selectively remove 6-O-sulfate groups from heparan sulfate. This activity modulates the effects of heparan sulfate by altering binding sites for signaling molecules (Dai et al., 2005 [PubMed 16192265]).[supplied by OMIM, Mar 2008].'},\n",
       " {'id': 'HGNC:18278',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RAD18',\n",
       "  'name': 'RAD18 E3 ubiquitin protein ligase',\n",
       "  'synonyms': 'RNF73',\n",
       "  'description': 'The protein encoded by this gene is highly similar to S. cerevisiae DNA damage repair protein Rad18. Yeast Rad18 functions through its interaction with Rad6, which is an ubiquitin-conjugating enzyme required for post-replication repair of damaged DNA. Similar to its yeast counterpart, this protein is able to interact with the human homolog of yeast Rad6 protein through a conserved ring-finger motif.  Mutation of this motif results in defective replication of UV-damaged DNA and hypersensitivity to multiple mutagens. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:20885',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'AGPAT4',\n",
       "  'name': '1-acylglycerol-3-phosphate O-acyltransferase 4',\n",
       "  'synonyms': '1-AGPAT4|LPAAT-delta|dJ473J16.2',\n",
       "  'description': 'This gene encodes a member of the 1-acylglycerol-3-phosphate O-acyltransferase family. This integral membrane protein converts lysophosphatidic acid to phosphatidic acid, the second step in de novo phospholipid biosynthesis. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:24956',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'OLFML3',\n",
       "  'name': 'olfactomedin like 3',\n",
       "  'synonyms': 'HNOEL-iso|OLF44',\n",
       "  'description': 'This gene encodes a member of the olfactomedin-like gene family which also includes genes encoding noelin, tiarin, myocilin, amassin, optimedin, photomedin, and latrophilin. The encoded protein is a secreted extracellular matrix glycoprotein with a C-terminal olfactomedin domain that facilitates protein-protein interactions, cell adhesion, and intercellular interactions. It serves as both a scaffold protein that recruits bone morphogenetic protein 1 to its substrate chordin, and as a vascular tissue remodeler with pro-angiogenic properties. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Feb 2017].'},\n",
       " {'id': 'HGNC:24641',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'VPS35L',\n",
       "  'name': 'VPS35 endosomal protein sorting factor like',\n",
       "  'synonyms': 'C16orf62|EC97|RTSC3',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:20754',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TRPC7',\n",
       "  'name': 'transient receptor potential cation channel subfamily C member 7',\n",
       "  'synonyms': 'TRP7',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:14085',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RTN4',\n",
       "  'name': 'reticulon 4',\n",
       "  'synonyms': 'ASY|NI220/250|NOGO|NSP|NSP-CL|Nbla00271|Nbla10545|RTN-X|RTN4-A|RTN4-B1|RTN4-B2|RTN4-C',\n",
       "  'description': 'This gene belongs to the family of reticulon encoding genes. Reticulons are associated with the endoplasmic reticulum, and are involved in neuroendocrine secretion or in membrane trafficking in neuroendocrine cells. The product of this gene is a potent neurite outgrowth inhibitor which may also help block the regeneration of the central nervous system in higher vertebrates. Alternatively spliced transcript variants derived both from differential splicing and differential promoter usage and encoding different isoforms have been identified. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:15916',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PAK5',\n",
       "  'name': 'p21 (RAC1) activated kinase 5',\n",
       "  'synonyms': 'PAK7',\n",
       "  'description': 'The protein encoded by this gene is a member of the PAK family of Ser/Thr protein kinases. PAK family members are known to be effectors of Rac/Cdc42 GTPases, which have been implicated in the regulation of cytoskeletal dynamics, proliferation, and cell survival signaling. This kinase contains a CDC42/Rac1 interactive binding (CRIB) motif, and has been shown to bind CDC42 in the presence of GTP. This kinase is predominantly expressed in brain. It is capable of promoting neurite outgrowth, and thus may play a role in neurite development. This kinase is associated with microtubule networks and induces microtubule stabilization. The subcellular localization of this kinase is tightly regulated during cell cycle progression. Alternatively spliced transcript variants encoding the same protein have been described. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:25074',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'LYRM1',\n",
       "  'name': 'LYR motif containing 1',\n",
       "  'synonyms': 'A211C6.1',\n",
       "  'description': 'The protein encoded by this gene belongs to the mitochondrial leucine/tyrosine/arginine motif family of proteins. Proteins of this family are short polypeptides that contain a leucine/tyrosine/arginine motif near the N-terminus. This gene is widely expressed with high levels in omental adipose tissue of obese individuals. In adipose tissue, the protein is localized to the nucleus where it promotes preadipocyte proliferation and lowers the rate of apoptosis to regulate adipose tissue homeostasis. Overexpression of this gene in adipocytes causes abnormal mitochondrial morphology and mitochondrial dysfunction. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Nov 2014].'},\n",
       " {'id': 'HGNC:14202',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'JPH2',\n",
       "  'name': 'junctophilin 2',\n",
       "  'synonyms': 'CMH17|JP-2|JP2',\n",
       "  'description': 'Junctional complexes between the plasma membrane and endoplasmic/sarcoplasmic reticulum are a common feature of all excitable cell types and mediate cross talk between cell surface and intracellular ion channels. The protein encoded by this gene is a component of junctional complexes and is composed of a C-terminal hydrophobic segment spanning the endoplasmic/sarcoplasmic reticulum membrane and a remaining cytoplasmic domain that shows specific affinity for the plasma membrane. This gene is a member of the junctophilin gene family. Alternative splicing has been observed at this locus and two variants encoding distinct isoforms are described. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:23689',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GPR158',\n",
       "  'name': 'G protein-coupled receptor 158',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:17634',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NUFIP2',\n",
       "  'name': 'nuclear FMR1 interacting protein 2',\n",
       "  'synonyms': '182-FIP|82-FIP|FIP-82|PIG1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:29262',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'IFT80',\n",
       "  'name': 'intraflagellar transport 80',\n",
       "  'synonyms': 'ATD2|FAP167|SRTD2|WDR56',\n",
       "  'description': 'The protein encoded by this gene is part of the intraflagellar transport complex B and is necessary for the function of motile and sensory cilia. Defects in this gene are a cause of asphyxiating thoracic dystrophy 2 (ATD2). Three transcript variants encoding two different isoforms have been found for this gene.[provided by RefSeq, Jun 2010].'},\n",
       " {'id': 'HGNC:18357',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ARHGAP20',\n",
       "  'name': 'Rho GTPase activating protein 20',\n",
       "  'synonyms': 'RARHOGAP',\n",
       "  'description': 'The protein encoded by this gene is an activator of RHO-type GTPases, transducing a signal from RAP1 to RHO and impacting neurite outgrowth. [provided by RefSeq, Sep 2016].'},\n",
       " {'id': 'HGNC:18285',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RAB40C',\n",
       "  'name': 'RAB40C, member RAS oncogene family',\n",
       "  'synonyms': 'RARL|RASL8C',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:19079',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PLEKHB1',\n",
       "  'name': 'pleckstrin homology domain containing B1',\n",
       "  'synonyms': 'KPL1|PHR1|PHRET1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:17795',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SAV1',\n",
       "  'name': 'salvador family WW domain containing protein 1',\n",
       "  'synonyms': 'SAV|WW45|WWP4',\n",
       "  'description': 'WW domain-containing proteins are found in all eukaryotes and play an important role in the regulation of a wide variety of cellular functions such as protein degradation, transcription, and RNA splicing. This gene encodes a protein with two WW domains, a SARAH domain, and a coiled-coil region and is ubiquitously expressed in adult tissues. This protein binds to MST1 (mammalian sterile 20-like kinase 1) and promotes MST1-induced apoptosis. It has also been shown to bind to HAX1 (hematopoietic cell-specific protein 1 (HS1)-associated protein X-1) and to attenuate the anti-apoptotic effects of HAX1. Studies in human and mouse suggest this gene acts as a tumor suppressor. [provided by RefSeq, Aug 2012].'},\n",
       " {'id': 'HGNC:14235',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'AASDHPPT',\n",
       "  'name': 'aminoadipate-semialdehyde dehydrogenase-phosphopantetheinyl transferase',\n",
       "  'synonyms': 'AASD-PPT|ACPS|CGI-80|LYS2|LYS5',\n",
       "  'description': 'The protein encoded by this gene is similar to Saccharomyces cerevisiae LYS5, which is required for the activation of the alpha-aminoadipate dehydrogenase in the biosynthetic pathway of lysine. Yeast alpha-aminoadipate dehydrogenase converts alpha-biosynthetic-aminoadipate semialdehyde to alpha-aminoadipate. It has been suggested that defects in the human gene result in pipecolic acidemia. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'NCBIGene:60683',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PDXK',\n",
       "  'name': 'pyridoxal kinase',\n",
       "  'synonyms': 'C21orf124|C21orf97|HEL-S-1a|HMSN6C|PKH|PNK|PRED79',\n",
       "  'description': 'The protein encoded by this gene phosphorylates vitamin B6, a step required for the conversion of vitamin B6 to pyridoxal-5-phosphate, an important cofactor in intermediary metabolism. The encoded protein is cytoplasmic and probably acts as a homodimer. Alternatively spliced transcript variants have been described, but their biological validity has not been determined. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:29639',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MAGEF1',\n",
       "  'name': 'MAGE family member F1',\n",
       "  'synonyms': 'MAGE-F1',\n",
       "  'description': 'This intronless gene encodes a member of the MAGE superfamily. It is ubiquitously expressed in normal tissues and in tumor cells. This gene includes a microsatellite repeat in the coding region. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:23410',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TMBIM1',\n",
       "  'name': 'transmembrane BAX inhibitor motif containing 1',\n",
       "  'synonyms': 'LFG3|MST100|MSTP100|PP1201|RECS1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:17642',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DCLRE1C',\n",
       "  'name': 'DNA cross-link repair 1C',\n",
       "  'synonyms': 'A-SCID|DCLREC1C|RS-SCID|SCIDA|SNM1C',\n",
       "  'description': \"This gene encodes a nuclear protein that is involved in V(D)J recombination and DNA repair. The encoded protein has single-strand-specific 5'-3' exonuclease activity; it also exhibits endonuclease activity on 5' and 3' overhangs and hairpins. The protein also functions in the regulation of the cell cycle in response to DNA damage. Mutations in this gene can cause Athabascan-type severe combined immunodeficiency (SCIDA) and Omenn syndrome. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Jan 2014].\"},\n",
       " {'id': 'HGNC:25078',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MOSPD3',\n",
       "  'name': 'motile sperm domain containing 3',\n",
       "  'synonyms': 'CDS3|NET30',\n",
       "  'description': 'This gene encodes a multi-pass membrane protein with a major sperm protein (MSP) domain. The deletion of a similar mouse gene is associated with defective cardiac development and neonatal lethality. Alternate transcriptional splice variants, encoding different isoforms, have been described. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:26136',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GAREM1',\n",
       "  'name': 'GRB2 associated regulator of MAPK1 subtype 1',\n",
       "  'synonyms': 'C18orf11|FAM59A|GAREM|Gm944',\n",
       "  'description': 'This gene encodes an adaptor protein which functions in the epidermal growth factor (EGF) receptor-mediated signaling pathway. Multiple transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Oct 2011].'},\n",
       " {'id': 'HGNC:25851',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GINS3',\n",
       "  'name': 'GINS complex subunit 3',\n",
       "  'synonyms': 'PSF3',\n",
       "  'description': 'This gene encodes a protein subunit of the GINS heterotetrameric complex, which is essential for the initiation of DNA replication and replisome progression in eukaryotes. Alternatively spliced transcript variants encoding distinct isoforms have been described. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:14491',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MRPL40',\n",
       "  'name': 'mitochondrial ribosomal protein L40',\n",
       "  'synonyms': 'L40mt|MRP-L22|MRP-L40|MRPL22|NLVCF|URIM',\n",
       "  'description': 'Mammalian mitochondrial ribosomal proteins are encoded by nuclear genes and help in protein synthesis within the mitochondrion. Mitochondrial ribosomes (mitoribosomes) consist of a small 28S subunit and a large 39S subunit. They have an estimated 75% protein to rRNA composition compared to prokaryotic ribosomes, where this ratio is reversed. Another difference between mammalian mitoribosomes and prokaryotic ribosomes is that the latter contain a 5S rRNA. Among different species, the proteins comprising the mitoribosome differ greatly in sequence, and sometimes in biochemical properties, which prevents easy recognition by sequence homology. This gene encodes a 39S subunit protein. Deletions in this gene may contribute to the etiology of velo-cardio-facial syndrome and DiGeorge syndrome. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:14466',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NDRG4',\n",
       "  'name': 'NDRG family member 4',\n",
       "  'synonyms': 'BDM1|SMAP-8|SMAP8',\n",
       "  'description': 'This gene is a member of the N-myc downregulated gene family which belongs to the alpha/beta hydrolase superfamily. The protein encoded by this gene is a cytoplasmic protein that is required for cell cycle progression and survival in primary astrocytes and may be involved in the regulation of mitogenic signalling in vascular smooth muscles cells. Alternative splicing results in multiple transcripts encoding different isoforms.[provided by RefSeq, Jun 2011].'},\n",
       " {'id': 'HGNC:20439',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'UPF3B',\n",
       "  'name': 'UPF3B regulator of nonsense mediated mRNA decay',\n",
       "  'synonyms': 'HUPF3B|MRX62|MRX82|MRXS14|RENT3B|UPF3BP1|UPF3BP2|UPF3BP3|UPF3X|Upf3p-X',\n",
       "  'description': 'This gene encodes a protein that is part of a post-splicing multiprotein complex involved in both mRNA nuclear export and mRNA surveillance. The encoded protein is one of two functional homologs to yeast Upf3p. mRNA surveillance detects exported mRNAs with truncated open reading frames and initiates nonsense-mediated mRNA decay (NMD). When translation ends upstream from the last exon-exon junction, this triggers NMD to degrade mRNAs containing premature stop codons. This protein binds to the mRNA and remains bound after nuclear export, acting as a nucleocytoplasmic shuttling protein. It forms with Y14 a complex that binds specifically 20 nt upstream of exon-exon junctions. This gene is located on the long arm of chromosome X. Two splice variants encoding different isoforms have been found for this gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:30798',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PRRG3',\n",
       "  'name': 'proline rich and Gla domain 3',\n",
       "  'synonyms': 'PRGP3|TMG3',\n",
       "  'description': 'This gene encodes a protein which contains a vitamin K-dependent carboxylation/gamma-carboxyglutamic domain. The encoded protein is a member of a family of vitamin K-dependent transmembrane proteins which contain a glutamate-rich extracellular domain. [provided by RefSeq, Aug 2011].'},\n",
       " {'id': 'HGNC:19358',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ALG12',\n",
       "  'name': 'ALG12 alpha-1,6-mannosyltransferase',\n",
       "  'synonyms': 'CDG1G|ECM39|PP14673|hALG12',\n",
       "  'description': 'This gene encodes a member of the glycosyltransferase 22 family. The encoded protein catalyzes the addition of the eighth mannose residue in an alpha-1,6 linkage onto the dolichol-PP-oligosaccharide precursor (dolichol-PP-Man(7)GlcNAc(2)) required for protein glycosylation. Mutations in this gene have been associated with congenital disorder of glycosylation type Ig (CDG-Ig)characterized by abnormal N-glycosylation. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:28369',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'THOC6',\n",
       "  'name': 'THO complex 6',\n",
       "  'synonyms': 'WDR58|fSAP35',\n",
       "  'description': 'This gene encodes a subunit of the multi-protein THO complex, which is involved in coordination between transcription and mRNA processing. The THO complex is a component of the TREX (transcription/export) complex, which is involved in transcription and export of mRNAs. A missense mutation in this gene is associated with a neurodevelopmental disorder called Beaulieu-Boycott-Innes syndrome. [provided by RefSeq, Dec 2016].'},\n",
       " {'id': 'HGNC:21064',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'WDR25',\n",
       "  'name': 'WD repeat domain 25',\n",
       "  'synonyms': 'C14orf67',\n",
       "  'description': 'This gene encodes a protein containing 7 WD repeats. WD repeats are approximately 30 to 40-amino acid domains containing several conserved residues, typically having a Tryptophan-Aspartic acid dipeptide (WD) at the C-terminal end. WD domains are involved in protein-protein interactions in a variety of cellular processes, including cell cycle progression, signal transduction, apoptosis, and gene regulation. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Apr 2017].'},\n",
       " {'id': 'HGNC:21153',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RNF128',\n",
       "  'name': 'ring finger protein 128',\n",
       "  'synonyms': 'GRAIL',\n",
       "  'description': 'The protein encoded by this gene is a type I transmembrane protein that localizes to the endocytic pathway. This protein contains a RING zinc-finger motif and has been shown to possess E3 ubiquitin ligase activity. Expression of this gene in retrovirally transduced T cell hybridoma significantly inhibits activation-induced IL2 and IL4 cytokine production. Induced expression of this gene was observed in anergic CD4(+) T cells, which suggested a role in the induction of anergic phenotype. Alternatively spliced transcript variants encoding distinct isoforms have been reported. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:20991',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'FAM184A',\n",
       "  'name': 'family with sequence similarity 184 member A',\n",
       "  'synonyms': 'C6orf60',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:25792',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'USB1',\n",
       "  'name': 'U6 snRNA biogenesis phosphodiesterase 1',\n",
       "  'synonyms': 'C16orf57|HVSL1|Mpn1|PN|hUsb1',\n",
       "  'description': 'This gene encodes a protein with several conserved domains, however, its exact function is not known. Mutations in this gene are associated with poikiloderma with neutropenia (PN), which shows phenotypic overlap with Rothmund-Thomson syndrome (RTS) caused by mutations in the RECQL4 gene. It is believed that this gene product interacts with RECQL4 protein via SMAD4 proteins, explaining the partial clinical overlap between PN and RTS. Alternatively spliced transcript variants encoding different isoforms have been noted for this gene. [provided by RefSeq, Mar 2011].'},\n",
       " {'id': 'HGNC:2962',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DYNC2H1',\n",
       "  'name': 'dynein cytoplasmic 2 heavy chain 1',\n",
       "  'synonyms': 'ATD3|DHC1b|DHC2|DNCH2|DYH1B|SRPS2B|SRTD3|hdhc11',\n",
       "  'description': 'This gene encodes a large cytoplasmic dynein protein that is involved in retrograde transport in the cilium and has a role in intraflagellar transport, a process required for ciliary/flagellar assembly. Mutations in this gene cause a heterogeneous spectrum of conditions related to altered primary cilium function and often involve polydactyly, abnormal skeletogenesis, and polycystic kidneys. Alternative splicing results in multiple transcript variants encoding distinct proteins. [provided by RefSeq, Jan 2010].'},\n",
       " {'id': 'HGNC:20757',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PLEKHF2',\n",
       "  'name': 'pleckstrin homology and FYVE domain containing 2',\n",
       "  'synonyms': 'EAPF|PHAFIN2|ZFYVE18',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:26265',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NOL9',\n",
       "  'name': 'nucleolar protein 9',\n",
       "  'synonyms': 'Grc3|NET6',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:26144',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PALB2',\n",
       "  'name': 'partner and localizer of BRCA2',\n",
       "  'synonyms': 'FANCN|PNCA3',\n",
       "  'description': 'This gene encodes a protein that may function in tumor suppression. This protein binds to and colocalizes with the breast cancer 2 early onset protein (BRCA2) in nuclear foci and likely permits the stable intranuclear localization and accumulation of BRCA2. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:20652',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TXNDC15',\n",
       "  'name': 'thioredoxin domain containing 15',\n",
       "  'synonyms': 'BUG|C5orf14|UNQ335',\n",
       "  'description': 'This gene encodes a member of the thioredoxin superfamily. Members of this family are characterized by a conserved active motif called the thioredoxin fold that catalyzes disulfide bond formation and isomerization. [provided by RefSeq, Apr 2017].'},\n",
       " {'id': 'HGNC:26253',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ZMYM1',\n",
       "  'name': 'zinc finger MYM-type containing 1',\n",
       "  'synonyms': 'MYM',\n",
       "  'description': nan},\n",
       " {'id': 'NCBIGene:79907',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'NCBIGene:79911',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'HGNC:14671',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ACTR5',\n",
       "  'name': 'actin related protein 5',\n",
       "  'synonyms': 'Arp5|INO80M',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:25773',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'WDR76',\n",
       "  'name': 'WD repeat domain 76',\n",
       "  'synonyms': 'CDW14',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:30729',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CSRNP3',\n",
       "  'name': 'cysteine and serine rich nuclear protein 3',\n",
       "  'synonyms': 'FAM130A2|PPP1R73|TAIP-2|TAIP2',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:25810',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HEXA-AS1',\n",
       "  'name': 'HEXA antisense RNA 1',\n",
       "  'synonyms': 'C15orf34',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:24722',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ZNF614',\n",
       "  'name': 'zinc finger protein 614',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:19018',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TRIM45',\n",
       "  'name': 'tripartite motif containing 45',\n",
       "  'synonyms': 'RNF99',\n",
       "  'description': 'This gene encodes a member of the tripartite motif family. The encoded protein may function as a transcriptional repressor of the mitogen-activated protein kinase pathway. Alternatively spliced transcript variants have been described.[provided by RefSeq, Mar 2009].'},\n",
       " {'id': 'HGNC:30779',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MTERF2',\n",
       "  'name': 'mitochondrial transcription termination factor 2',\n",
       "  'synonyms': 'MTERFD3|mTERFL',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:13403',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PBX4',\n",
       "  'name': 'PBX homeobox 4',\n",
       "  'synonyms': nan,\n",
       "  'description': 'This gene encodes a member of the pre-B cell leukemia transcription factor family. These proteins are homeobox proteins that play critical roles in embryonic development and cellular differentiation both as Hox cofactors and through Hox-independent pathways. The encoded protein contains a homeobox DNA-binding domain, but specific functions of the protein have not been determined. Alternatively spliced transcript variants have been observed for this gene. [provided by RefSeq, May 2011].'},\n",
       " {'id': 'HGNC:14905',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TAS1R2',\n",
       "  'name': 'taste 1 receptor member 2',\n",
       "  'synonyms': 'GPR71|T1R2|TR2',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:29559',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'VMP1',\n",
       "  'name': 'vacuole membrane protein 1',\n",
       "  'synonyms': 'EPG3|TANGO5|TMEM49',\n",
       "  'description': 'This gene encodes a transmembrane protein that plays a key regulatory role in the process of autophagy. The ectopic overexpression of the encoded protein in cultured cells triggers autophagy even under nutrient-rich conditions. This gene is overexpressed in pancreatitis affected acinar cells where the encoded protein mediates sequestration and degradation of potentially deleterious activated zymogen granules in a process termed, zymophagy. [provided by RefSeq, Jul 2016].'},\n",
       " {'id': 'HGNC:24661',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'REXO5',\n",
       "  'name': 'RNA exonuclease 5',\n",
       "  'synonyms': 'NEF-sp',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:23366',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ARPC5L',\n",
       "  'name': 'actin related protein 2/3 complex subunit 5 like',\n",
       "  'synonyms': 'ARC16-2',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:14624',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CDCA3',\n",
       "  'name': 'cell division cycle associated 3',\n",
       "  'synonyms': 'GRCC8|TOME-1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:25461',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PUS3',\n",
       "  'name': 'pseudouridine synthase 3',\n",
       "  'synonyms': '2610020J05Rik|FKSG32|MRT55|NEDMIGS',\n",
       "  'description': 'The protein encoded by this gene catalyzes the formation of tRNA pseudouridine from tRNA uridine at position 39 in the anticodon stem and loop of transfer RNAs. Two transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Dec 2012].'},\n",
       " {'id': 'HGNC:18515',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TMEM47',\n",
       "  'name': 'transmembrane protein 47',\n",
       "  'synonyms': 'BCMP1|TM4SF10|VAB-9',\n",
       "  'description': 'This gene encodes a member of the PMP22/EMP/claudin protein family. The encoded protein is localized to the ER and the plasma membrane. In dogs, transcripts of this gene exist at high levels in the brain. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:28658',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'AMMECR1L',\n",
       "  'name': 'AMMECR1 like',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:23726',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'FAM107B',\n",
       "  'name': 'family with sequence similarity 107 member B',\n",
       "  'synonyms': 'C10orf45|HITS',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:24029',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'INHBE',\n",
       "  'name': 'inhibin subunit beta E',\n",
       "  'synonyms': nan,\n",
       "  'description': 'This gene encodes a member of the TGF-beta (transforming growth factor-beta) superfamily of proteins. The encoded preproprotein is proteolytically processed to generate an inhibin beta subunit. Inhibins have been implicated in regulating numerous cellular processes including cell proliferation, apoptosis, immune response and hormone secretion. This gene may be upregulated under conditions of endoplasmic reticulum stress, and this protein may inhibit cellular proliferation and growth in pancreas and liver. [provided by RefSeq, Sep 2016].'},\n",
       " {'id': 'HGNC:24295',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ESYT3',\n",
       "  'name': 'extended synaptotagmin 3',\n",
       "  'synonyms': 'CHR3SYT|E-Syt3|FAM62C',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:25310',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ANKRD27',\n",
       "  'name': 'ankyrin repeat domain 27',\n",
       "  'synonyms': 'PP12899|VARP',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:24474',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PCBD2',\n",
       "  'name': 'pterin-4 alpha-carbinolamine dehydratase 2',\n",
       "  'synonyms': 'DCOH2|DCOHM|PHS2',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:24103',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ASCC2',\n",
       "  'name': 'activating signal cointegrator 1 complex subunit 2',\n",
       "  'synonyms': 'ASC1p100|p100',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:9657',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PTPN5',\n",
       "  'name': 'protein tyrosine phosphatase non-receptor type 5',\n",
       "  'synonyms': 'PTPSTEP|STEP|STEP61',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:21205',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'LINGO1',\n",
       "  'name': 'leucine rich repeat and Ig domain containing 1',\n",
       "  'synonyms': 'LERN1|LRRN6A|MRT64|UNQ201',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:25907',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CEP89',\n",
       "  'name': 'centrosomal protein 89',\n",
       "  'synonyms': 'CCDC123|CEP123',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:29641',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PWWP3A',\n",
       "  'name': 'PWWP domain containing 3A, DNA repair factor',\n",
       "  'synonyms': 'EXPAND1|HSPC211|MUM-1|MUM1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:13764',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RELT',\n",
       "  'name': 'RELT TNF receptor',\n",
       "  'synonyms': 'AI3C|TNFRSF19L|TRLT',\n",
       "  'description': 'The protein encoded by this gene is a member of the TNF-receptor superfamily. This receptor is especially abundant in hematologic tissues. It has been shown to activate the NF-kappaB pathway and selectively bind TNF receptor-associated factor 1 (TRAF1). This receptor is capable of stimulating T-cell proliferation in the presence of CD3 signaling, which suggests its regulatory role in immune response. Two alternatively spliced transcript variants of this gene encoding the same protein have been reported. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:28221',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'UBL7',\n",
       "  'name': 'ubiquitin like 7',\n",
       "  'synonyms': 'BMSC-UbP|TCBA1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:20302',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SLC22A16',\n",
       "  'name': 'solute carrier family 22 member 16',\n",
       "  'synonyms': 'CT2|FLIPT2|HEL-S-18|OAT6|OCT6|OKB1|dJ261K5.1',\n",
       "  'description': 'This gene encodes a member of the organic zwitterion transporter protein family which transports carnitine. The encoded protein has also been shown to transport anticancer drugs like bleomycin (PMID: 20037140) successful treatment has been correlated with the level of activity of this transporter in tumor cells. [provided by RefSeq, Dec 2011].'},\n",
       " {'id': 'NCBIGene:88523',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'N4BP2L2',\n",
       "  'name': 'NEDD4 binding protein 2 like 2',\n",
       "  'synonyms': '92M18.3|CG005|CG016|PFAAP5',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:15844',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HPS4',\n",
       "  'name': 'HPS4 biogenesis of lysosomal organelles complex 3 subunit 2',\n",
       "  'synonyms': 'BLOC3S2|LE',\n",
       "  'description': 'This gene encodes a protein component of biogenesis of lysosome-related organelles complexes (BLOC). BLOC complexes are important for the formation of endosomal-lysosomal organelles such as melanosomes and platelet dense granules. Mutations in this gene result in subtype 4 of Hermansky-Pudlak syndrome, a form of albinism. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Aug 2012].'},\n",
       " {'id': 'HGNC:17958',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SAMD1',\n",
       "  'name': 'sterile alpha motif domain containing 1',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:30767',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'KNSTRN',\n",
       "  'name': 'kinetochore localized astrin (SPAG5) binding protein',\n",
       "  'synonyms': 'C15orf23|HSD11|SKAP|TRAF4AF1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:28072',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'LYRM7',\n",
       "  'name': 'LYR motif containing 7',\n",
       "  'synonyms': 'C5orf31|MC3DN8|MZM1L',\n",
       "  'description': 'Inner mitochondrial membrane complex III (CIII) is the main enzyme complex in the mitochondrial respiratory chain, and Rieske Fe-S protein (UQCRFS1) is the last catalytic subunit added to the complex. The protein encoded by this gene is a nuclear-encoded mitochondrial matrix protein that stabilizes UQCRFS1 and chaperones it to the CIII complex. Defects in this gene are a cause of mitochondrial complex III deficiency, nuclear type 8. Three transcript variants encoding two different isoforms have been found for this gene. [provided by RefSeq, Jun 2014].'},\n",
       " {'id': 'HGNC:19164',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'STARD13',\n",
       "  'name': 'StAR related lipid transfer domain containing 13',\n",
       "  'synonyms': 'ARHGAP37|DLC2|GT650|LINC00464',\n",
       "  'description': 'This gene encodes a protein which contains an N-terminal sterile alpha motif (SAM) for protein-protein interactions, followed by an ATP/GTP-binding motif, a GTPase-activating protein (GAP) domain, and a C-terminal STAR-related lipid transfer (START) domain. It may be involved in regulation of cytoskeletal reorganization, cell proliferation, and cell motility, and acts as a tumor suppressor in hepatoma cells. The gene is located in a region of chromosome 13 that is associated with loss of heterozygosity in hepatocellular carcinomas. Alternatively spliced transcript variants encoding different isoforms have been described for this gene. [provided by RefSeq, Aug 2011].'},\n",
       " {'id': 'HGNC:25037',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'N4BP2L1',\n",
       "  'name': 'NEDD4 binding protein 2 like 1',\n",
       "  'synonyms': 'CG018',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:14943',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PPP1R3E',\n",
       "  'name': 'protein phosphatase 1 regulatory subunit 3E',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:21711',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ZNF804A',\n",
       "  'name': 'zinc finger protein 804A',\n",
       "  'synonyms': 'C2orf10',\n",
       "  'description': 'The protein encoded by this gene is a zinc finger binding protein. Polymorphisms in this gene, especially rs1344706, are thought to confer increased susceptibility to schizophrenia, bipolar disorder, and heroin addiciton. [provided by RefSeq, Nov 2015].'},\n",
       " {'id': 'HGNC:24456',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DNER',\n",
       "  'name': 'delta/notch like EGF repeat containing',\n",
       "  'synonyms': 'UNQ26|bet',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:19697',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'KCNG4',\n",
       "  'name': 'potassium voltage-gated channel modifier subfamily G member 4',\n",
       "  'synonyms': 'KV6.3|KV6.4',\n",
       "  'description': 'Voltage-gated potassium (Kv) channels represent the most complex class of voltage-gated ion channels from both functional and structural standpoints. Their diverse functions include regulating neurotransmitter release, heart rate, insulin secretion, neuronal excitability, epithelial electrolyte transport, smooth muscle contraction, and cell volume. This gene encodes a member of the potassium channel, voltage-gated, subfamily G. This member functions as a modulatory subunit. The gene has strong expression in brain. Multiple alternatively spliced variants have been found in normal and cancerous tissues. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'NCBIGene:93333',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'HGNC:13579',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PAXBP1',\n",
       "  'name': 'PAX3 and PAX7 binding protein 1',\n",
       "  'synonyms': 'BM020|C21orf66|FSAP105|GCFC|GCFC1',\n",
       "  'description': 'This gene encodes a protein that may bind to GC-rich DNA sequences, which suggests its involvement in the regulation of transcription. Alternative splicing of this gene results in multiple transcript variants. [provided by RefSeq, Jun 2009].'},\n",
       " {'id': 'HGNC:20488',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'L3HYPDH',\n",
       "  'name': 'trans-L-3-hydroxyproline dehydratase',\n",
       "  'synonyms': 'C14orf149',\n",
       "  'description': 'The protein encoded by this gene is a dehydratase that converts trans-3-hydroxy-L-proline to delta(1)-pyrroline-2-carboxylate. This enzyme may function to degrade dietary proteins that contain trans-3-hydroxy-L-proline as well as other proteins such as collagen IV. The encoded protein can be converted to an epimerase by changing a threonine to a cysteine at a catalytic site. [provided by RefSeq, Sep 2016].'},\n",
       " {'id': 'HGNC:16416',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'LINC00313',\n",
       "  'name': 'long intergenic non-protein coding RNA 313',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:16505',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CYGB',\n",
       "  'name': 'cytoglobin',\n",
       "  'synonyms': 'HGB|STAP',\n",
       "  'description': 'This gene encodes a globin protein found in vertebrate cells. The encoded protein is described as a hexacoordinate hemoglobin which binds ligand differently from the pentacoordinate hemoglobins involved in oxygen transport, and may be involved in protection during oxidative stress. This gene is located on chromosome 17 in the same region as a retinal gene which is mutated in progressive rod-cone degeneration, but in the opposite orientation. [provided by RefSeq, Jan 2012].'},\n",
       " {'id': 'HGNC:29404',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'KLHL29',\n",
       "  'name': 'kelch like family member 29',\n",
       "  'synonyms': 'KBTBD9',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:14467',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SLC26A7',\n",
       "  'name': 'solute carrier family 26 member 7',\n",
       "  'synonyms': 'SUT2',\n",
       "  'description': 'This gene is one member of a family of sulfate/anion transporter genes. Family members are well conserved in gene structure and protein length yet have markedly different tissue expression patterns. This gene has abundant and specific expression in the kidney. Alternatively spliced transcript variants that encode different isoforms have been described. [provided by RefSeq, Aug 2013].'},\n",
       " {'id': 'HGNC:25173',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ZNF689',\n",
       "  'name': 'zinc finger protein 689',\n",
       "  'synonyms': 'TIPUH1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:16274',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CYYR1',\n",
       "  'name': 'cysteine and tyrosine rich 1',\n",
       "  'synonyms': 'C21orf95',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:26971',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ABHD15',\n",
       "  'name': 'abhydrolase domain containing 15',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:25177',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TLCD1',\n",
       "  'name': 'TLC domain containing 1',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:23304',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PRAP1',\n",
       "  'name': 'proline rich acidic protein 1',\n",
       "  'synonyms': 'PRO1195|UPA',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:26976',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ANAPC16',\n",
       "  'name': 'anaphase promoting complex subunit 16',\n",
       "  'synonyms': 'APC16|C10orf104|CENP-27|MSAG|bA570G20.3',\n",
       "  'description': nan},\n",
       " {'id': 'NCBIGene:121301',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'HGNC:20510',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'H4-16',\n",
       "  'name': 'H4 histone 16',\n",
       "  'synonyms': 'H4/p|H4C1|H4C11|H4C12|H4C13|H4C14|H4C15|H4C2|H4C3|H4C4|H4C5|H4C6|H4C8|H4C9|HIST4H4',\n",
       "  'description': 'Histones are basic nuclear proteins that are responsible for the nucleosome structure of the chromosomal fiber in eukaryotes. Nucleosomes consist of approximately 146 bp of DNA wrapped around a histone octamer composed of pairs of each of the four core histones (H2A, H2B, H3, and H4). The chromatin fiber is further compacted through the interaction of a linker histone, H1, with the DNA between the nucleosomes to form higher order chromatin structures. This gene is intronless and encodes a replication-dependent histone that is a member of the histone H4 family. Transcripts from this gene lack polyA tails; instead, they contain a palindromic termination element. [provided by RefSeq, Aug 2015].'},\n",
       " {'id': 'HGNC:28479',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'C16orf78',\n",
       "  'name': 'chromosome 16 open reading frame 78',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:28556',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'IQCK',\n",
       "  'name': 'IQ motif containing K',\n",
       "  'synonyms': nan,\n",
       "  'description': 'This gene belongs to the IQ motif-containing family of proteins. The IQ motif serves as a binding site for different EF-hand proteins such as calmodulin. This gene was identified as a potential candidate gene for obsessive-compulsive disorder in a genome-wide association study. Alternative splicing of this gene results in multiple transcript variants. [provided by RefSeq, Feb 2015].'},\n",
       " {'id': 'HGNC:23087',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SLC43A2',\n",
       "  'name': 'solute carrier family 43 member 2',\n",
       "  'synonyms': 'LAT4',\n",
       "  'description': 'This gene encodes a member of the L-amino acid transporter-3 or SLC43 family of transporters. The encoded protein mediates sodium-, chloride-, and pH-independent transport of L-isomers of neutral amino acids, including leucine, phenylalanine, valine and methionine. This protein may contribute to the transfer of amino acids across the placental membrane to the fetus. [provided by RefSeq, Mar 2016].'},\n",
       " {'id': 'HGNC:28356',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TBC1D16',\n",
       "  'name': 'TBC1 domain family member 16',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:25281',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ZNF543',\n",
       "  'name': 'zinc finger protein 543',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:33702',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MICOS13',\n",
       "  'name': 'mitochondrial contact site and cristae organizing system subunit 13',\n",
       "  'synonyms': 'C19orf70|MIC12|MIC13|P117|QIL1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:21269',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NLRP5',\n",
       "  'name': 'NLR family pyrin domain containing 5',\n",
       "  'synonyms': 'CLR19.8|MATER|NALP5|PAN11|PYPAF8',\n",
       "  'description': 'The protein encoded by this gene belongs to the NALP protein family. Members of the NALP protein family typically contain a NACHT domain, a NACHT-associated domain (NAD), a C-terminal leucine-rich repeat (LRR) region, and an N-terminal pyrin domain (PYD). Expression of this gene is restricted to the oocyte. A mouse gene that encodes a maternal oocyte protein, similar to this encoded protein, is required for normal early embryogenesis. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:20503',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'H3C14',\n",
       "  'name': 'H3 clustered histone 14',\n",
       "  'synonyms': 'H3|H3.2|H3/M|H3C13|H3C15|H3F2|H3FM|H3FN|HIST2H3C',\n",
       "  'description': 'Histones are basic nuclear proteins that are responsible for the nucleosome structure of the chromosomal fiber in eukaryotes. This structure consists of approximately 146 bp of DNA wrapped around a nucleosome, an octamer composed of pairs of each of the four core histones (H2A, H2B, H3, and H4). The chromatin fiber is further compacted through the interaction of a linker histone, H1, with the DNA between the nucleosomes to form higher order chromatin structures. This gene is intronless and encodes a replication-dependent histone that is a member of the histone H3 family. Transcripts from this gene lack polyA tails; instead, they contain a palindromic termination element. This gene is found in a histone cluster on chromosome 1. This gene is one of four histone genes in the cluster that are duplicated; this record represents the telomeric copy. [provided by RefSeq, Aug 2015].'},\n",
       " {'id': 'HGNC:20440',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'LYPLAL1',\n",
       "  'name': 'lysophospholipase like 1',\n",
       "  'synonyms': 'Q96AV0',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:14341',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'EDARADD',\n",
       "  'name': 'EDAR associated death domain',\n",
       "  'synonyms': 'ECTD11A|ECTD11B|ED3|EDA3',\n",
       "  'description': 'This gene was identified by its association with ectodermal dysplasia, a genetic disorder characterized by defective development of hair, teeth, and eccrine sweat glands. The protein encoded by this gene is a death domain-containing protein, and is found to interact with EDAR, a death domain receptor known to be required for the development of hair, teeth and other ectodermal derivatives. This protein and EDAR are coexpressed in epithelial cells during the formation of hair follicles and teeth. Through its interaction with EDAR, this protein acts as an adaptor, and links the receptor to downstream signaling pathways. Two alternatively spliced transcript variants of this gene encoding distinct isoforms have been reported. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:18830',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'KLF17',\n",
       "  'name': 'Kruppel like factor 17',\n",
       "  'synonyms': 'ZLF393|ZNF393|Zfp393',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:15791',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PIGU',\n",
       "  'name': 'phosphatidylinositol glycan anchor biosynthesis class U',\n",
       "  'synonyms': 'CDC91L1|GAB1|GPIBD21|NEDBSS',\n",
       "  'description': 'The protein encoded by this gene shares similarity with Saccharomyces cerevisiae Cdc91, a predicted integral membrane protein that may function in cell division control. The protein encoded by this gene is the fifth subunit of GPI transamidase that attaches GPI-anchors to proteins. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:29797',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NUP35',\n",
       "  'name': 'nucleoporin 35',\n",
       "  'synonyms': 'MP-44|MP44|NP44|NUP53',\n",
       "  'description': 'This gene encodes a member of the nucleoporin family. The encoded protein contains two membrane binding regions, is localized to the nuclear rim, and is part of the nuclear pore complex. All molecules entering or leaving the nucleus either diffuse through or are actively transported by the nuclear pore complex. Alternative splicing results in multiple transcript variants. Pseudogenes of this gene have been defined on chromosomes 7 and 10. [provided by RefSeq, Dec 2013].'},\n",
       " {'id': 'HGNC:8111',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'OSR1',\n",
       "  'name': 'odd-skipped related transcription factor 1',\n",
       "  'synonyms': 'ODD',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:34427',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CRYBG3',\n",
       "  'name': 'crystallin beta-gamma domain containing 3',\n",
       "  'synonyms': 'DKFZp667G2110|vlAKAP',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:28386',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'C4orf36',\n",
       "  'name': 'chromosome 4 open reading frame 36',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:17591',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CLDN23',\n",
       "  'name': 'claudin 23',\n",
       "  'synonyms': 'CLDNL|hCG1646163',\n",
       "  'description': 'This gene encodes a member of the claudin family. Claudins are integral membrane proteins and components of tight junction strands. Tight junction strands serve as a physical barrier to prevent solutes and water from passing freely through the paracellular space between epithelial or endothelial cell sheets, and also play critical roles in maintaining cell polarity and signal transductions. This gene is expressed in germinal center B-cells, placenta and stomach as well as in colon tumor. This gene is down-regulated in intestinal type gastric cancer. [provided by RefSeq, Aug 2010].'},\n",
       " {'id': 'HGNC:28625',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NDUFAF6',\n",
       "  'name': 'NADH:ubiquinone oxidoreductase complex assembly factor 6',\n",
       "  'synonyms': 'C8orf38|FRTS5|MC1DN17',\n",
       "  'description': 'This gene encodes a protein that localizes to mitochondria and contains a predicted phytoene synthase domain. The encoded protein plays an important role in the assembly of complex I (NADH-ubiquinone oxidoreductase) of the mitochondrial respiratory chain through regulation of subunit ND1 biogenesis. Mutations in this gene are associated with complex I enzymatic deficiency. [provided by RefSeq, Nov 2011].'},\n",
       " {'id': 'HGNC:23798',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MAGEC3',\n",
       "  'name': 'MAGE family member C3',\n",
       "  'synonyms': 'CT7.2|HCA2|MAGE-C3|MAGEC4',\n",
       "  'description': 'This gene is a member of the MAGEC gene family. The members of this family are not expressed in normal tissues, except for testis, and are expressed in tumors of various histological types. The MAGEC genes are clustered on chromosome Xq26-q27. Two transcript variants encoding distinct isoforms have been found for this gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:17857',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TRIM69',\n",
       "  'name': 'tripartite motif containing 69',\n",
       "  'synonyms': 'HSD-34|HSD34|RNF36|Trif',\n",
       "  'description': 'This gene encodes a member of the RING-B-box-coiled-coil (RBCC) family and encodes a protein with an N-terminal RING finger motif, a PRY domain and a C-terminal SPRY domain. The mouse ortholog of this gene is specifically expressed in germ cells at the round spermatid stages during spermatogenesis and, when overexpressed, induces apoptosis. Alternatively spliced transcript variants encoding distinct isoforms have been described. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:14899',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ADAMTS14',\n",
       "  'name': 'ADAM metallopeptidase with thrombospondin type 1 motif 14',\n",
       "  'synonyms': nan,\n",
       "  'description': 'This gene encodes a member of the ADAMTS (a disintegrin and metalloproteinase with thrombospondin motif) protein family. Members of the family share several distinct protein modules, including a propeptide region, a metalloproteinase domain, a disintegrin-like domain, and a thrombospondin type 1 (TS) motif. Individual members of this family differ in the number of C-terminal TS motifs, and some have unique C-terminal domains. The encoded preproprotein is proteolytically processed to generate the mature enzyme. This enzyme cleaves amino-terminal propeptides from type I procollagen, a necessary step in the formation of collagen fibers. Mutations in this gene may be associated with osteoarthritis in human patients. [provided by RefSeq, May 2016].'},\n",
       " {'id': 'HGNC:16132',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SRXN1',\n",
       "  'name': 'sulfiredoxin 1',\n",
       "  'synonyms': 'C20orf139|Npn3|SRX|SRX1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:9662',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SIRPA',\n",
       "  'name': 'signal regulatory protein alpha',\n",
       "  'synonyms': 'BIT|CD172A|MFR|MYD-1|P84|PTPNS1|SHPS1|SIRP',\n",
       "  'description': 'The protein encoded by this gene is a member of the signal-regulatory-protein (SIRP) family, and also belongs to the immunoglobulin superfamily. SIRP family members are receptor-type transmembrane glycoproteins known to be involved in the negative regulation of receptor tyrosine kinase-coupled signaling processes. This protein can be phosphorylated by tyrosine kinases. The phospho-tyrosine residues of this PTP have been shown to recruit SH2 domain containing tyrosine phosphatases (PTP), and serve as substrates of PTPs. This protein was found to participate in signal transduction mediated by various growth factor receptors. CD47 has been demonstrated to be a ligand for this receptor protein. This gene and its product share very high similarity with several other members of the SIRP family. These related genes are located in close proximity to each other on chromosome 20p13. Multiple alternatively spliced transcript variants have been determined for this gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:23727',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CACUL1',\n",
       "  'name': 'CDK2 associated cullin domain 1',\n",
       "  'synonyms': 'C10orf46|CAC1',\n",
       "  'description': nan},\n",
       " {'id': 'NCBIGene:143902',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'HGNC:28577',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'AMDHD1',\n",
       "  'name': 'amidohydrolase domain containing 1',\n",
       "  'synonyms': 'HMFT1272',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:28520',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TERB2',\n",
       "  'name': 'telomere repeat binding bouquet formation protein 2',\n",
       "  'synonyms': 'C15orf43',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:28558',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TMCO5A',\n",
       "  'name': 'transmembrane and coiled-coil domains 5A',\n",
       "  'synonyms': 'TMCO5',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:24160',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'BEAN1',\n",
       "  'name': 'brain expressed associated with NEDD4 1',\n",
       "  'synonyms': 'BEAN|SCA31',\n",
       "  'description': 'The protein encoded by this gene is one of several proteins that interact with NEDD4, a member of a family of ubiquitin-protein ligases. These proteins have PY motifs in common that bind to the WW domains of NEDD4. NEDD4 is developmentally regulated, and is highly expressed in embryonic tissues. Mutations in this gene (i.e., intronic insertions of >100 copies of pentanucleotide repeats including a (TGGAA)n sequence) are associated with spinocerebellar ataxia type 31. Alternatively spliced transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, May 2010].'},\n",
       " {'id': 'HGNC:26496',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ZNF785',\n",
       "  'name': 'zinc finger protein 785',\n",
       "  'synonyms': 'ZNF688',\n",
       "  'description': nan},\n",
       " {'id': 'NCBIGene:146856',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'HGNC:23616',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SLC35F3',\n",
       "  'name': 'solute carrier family 35 member F3',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:26417',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SLC16A14',\n",
       "  'name': 'solute carrier family 16 member 14',\n",
       "  'synonyms': 'MCT14',\n",
       "  'description': nan},\n",
       " {'id': 'NCBIGene:151720',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'HGNC:18007',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GLIPR2',\n",
       "  'name': 'GLI pathogenesis related 2',\n",
       "  'synonyms': 'C9orf19|GAPR-1|GAPR1',\n",
       "  'description': nan},\n",
       " {'id': 'NCBIGene:157273',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'LOC157273',\n",
       "  'name': 'uncharacterized LOC157273',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:25438',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PRAG1',\n",
       "  'name': 'PEAK1 related, kinase-activating pseudokinase 1',\n",
       "  'synonyms': 'NACK|PEAK2|PRAGMIN|SGK223',\n",
       "  'description': 'This gene encodes an enzyme that belongs to the tyrosine protein kinase family. A similar protein in rat binds to Rho family GTPase 2 (Rnd2) and regulates neurite outgrowth via activation of Ras homolog gene family, member A (RhoA). [provided by RefSeq, Mar 2014].'},\n",
       " {'id': 'HGNC:26368',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MCMDC2',\n",
       "  'name': 'minichromosome maintenance domain containing 2',\n",
       "  'synonyms': 'C8orf45',\n",
       "  'description': nan},\n",
       " {'id': 'NCBIGene:157909',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PAPPA',\n",
       "  'name': 'pappalysin 1',\n",
       "  'synonyms': 'ASBABP2|DIPLA1|IGFBP-4ase|PAPA|PAPP-A|PAPPA1',\n",
       "  'description': 'This gene encodes a secreted metalloproteinase which cleaves insulin-like growth factor binding proteins (IGFBPs). Following IGFBP cleavage, insulin growth factors dissociate from IGFBPs and bind to IGF receptors, resulting in activation of the IGF pathway. The encoded protein plays a role in bone formation, inflammation, wound healing and female fertility. Enhanced expression of this protein is associated with diabetic nephropathy in human patients and this protein may promote tumor invasion and growth in various human cancers. [provided by RefSeq, Aug 2017].'},\n",
       " {'id': 'HGNC:24294',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CSAG1',\n",
       "  'name': 'chondrosarcoma associated gene 1',\n",
       "  'synonyms': 'CSAGE|CT24.1',\n",
       "  'description': 'This gene encodes a member of a family of tumor antigens. The protein is expressed in chondrosarcomas, but may also be expressed in normal tissues such as testis. Alternative splicing of this gene results in two transcript variants encoding the same protein. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:29962',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'FAM47A',\n",
       "  'name': 'family with sequence similarity 47 member A',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:26388',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ARHGAP36',\n",
       "  'name': 'Rho GTPase activating protein 36',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:26607',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SLC35G1',\n",
       "  'name': 'solute carrier family 35 member G1',\n",
       "  'synonyms': 'C10orf60|POST|TMEM20',\n",
       "  'description': 'This gene encodes a transmembrane protein which is a member of the drug/metabolite transporter protein superfamily. The encoded protein may play a role in the regulation of calcium levels inside the cell. [provided by RefSeq, Sep 2016].'},\n",
       " {'id': 'HGNC:28536',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TBC1D21',\n",
       "  'name': 'TBC1 domain family member 21',\n",
       "  'synonyms': 'MgcRabGAP',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:26656',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MFSD6L',\n",
       "  'name': 'major facilitator superfamily domain containing 6 like',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:34333',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ZNF836',\n",
       "  'name': 'zinc finger protein 836',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:25331',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ZNF540',\n",
       "  'name': 'zinc finger protein 540',\n",
       "  'synonyms': 'Nbla10512',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:30930',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'XAGE5',\n",
       "  'name': 'X antigen family member 5',\n",
       "  'synonyms': 'CT12.5|GAGED5|XAGE-5',\n",
       "  'description': 'This gene is a member of the XAGE subfamily, which belongs to the GAGE family. The GAGE genes are expressed in a variety of tumors and in some fetal and reproductive tissues. The protein encoded by this gene shares a sequence similarity with other GAGE/PAGE proteins. Because of the expression pattern and the sequence similarity, this protein also belongs to a family of CT (cancer-testis) antigens. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:18339',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DEFT1P',\n",
       "  'name': 'defensin theta 1, pseudogene',\n",
       "  'synonyms': 'DEFQ1|DEFT1',\n",
       "  'description': nan},\n",
       " {'id': 'NCBIGene:170951',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SLX4',\n",
       "  'name': 'SLX4 structure-specific endonuclease subunit',\n",
       "  'synonyms': 'BTBD12|FANCP|MUS312',\n",
       "  'description': 'This gene encodes a protein that functions as an assembly component of multiple structure-specific endonucleases. These endonuclease complexes are required for repair of specific types of DNA lesions and critical for cellular responses to replication fork failure. Mutations in this gene were found in patients with Fanconi anemia. [provided by RefSeq, Sep 2016].'},\n",
       " {'id': 'HGNC:28328',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ZFC3H1',\n",
       "  'name': 'zinc finger C3H1-type containing',\n",
       "  'synonyms': 'CCDC131|CSRC2|PSRC2',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:26479',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ZNF778',\n",
       "  'name': 'zinc finger protein 778',\n",
       "  'synonyms': nan,\n",
       "  'description': 'The protein encoded by this gene is a member of the krueppel C2H2-type zinc-finger protein family, and it contains one KRAB domain and eighteen C2H2-type zinc fingers. This gene is a candidate gene for autism and variable cognitive impairment in the 16q24.3 microdeletion syndrome. Alternatively spliced transcript variants have been found for this gene. [provided by RefSeq, Jan 2011].'},\n",
       " {'id': 'NCBIGene:197379',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'HGNC:26437',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TSNARE1',\n",
       "  'name': 't-SNARE domain containing 1',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:23752',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CERS3',\n",
       "  'name': 'ceramide synthase 3',\n",
       "  'synonyms': 'ARCI9|LASS3',\n",
       "  'description': 'This gene is a member of the ceramide synthase family of genes. The ceramide synthase enzymes regulate sphingolipid synthesis by catalyzing the formation of ceramides from sphingoid base and acyl-coA substrates. This family member is involved in the synthesis of ceramides with ultra-long-chain acyl moieties (ULC-Cers), important to the epidermis in its role in creating a protective barrier from the environment. The protein encoded by this gene has also been implicated in modification of the lipid structures required for spermatogenesis. Mutations in this gene have been associated with male fertility defects, and epidermal defects, including ichthyosis. Alternative splicing results in multiple transcript variants encoding different isoforms. [provided by RefSeq, Aug 2015].'},\n",
       " {'id': 'NCBIGene:219392',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'HGNC:27346',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'OTUD1',\n",
       "  'name': 'OTU deubiquitinase 1',\n",
       "  'synonyms': 'DUBA7|OTDC1',\n",
       "  'description': 'Deubiquitinating enzymes (DUBs; see MIM 603478) are proteases that specifically cleave ubiquitin (MIM 191339) linkages, negating the action of ubiquitin ligases. DUBA7 belongs to a DUB subfamily characterized by an ovarian tumor (OTU) domain.[supplied by OMIM, May 2008].'},\n",
       " {'id': 'HGNC:18334',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TIGD3',\n",
       "  'name': 'tigger transposable element derived 3',\n",
       "  'synonyms': nan,\n",
       "  'description': 'The protein encoded by this gene belongs to the tigger subfamily of the pogo superfamily of DNA-mediated transposons in humans. These proteins are related to DNA transposons found in fungi and nematodes, and more distantly to the Tc1 and mariner transposases. They are also very similar to the major mammalian centromere protein B. The exact function of this gene is not known. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'NCBIGene:221016',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CCDC7',\n",
       "  'name': 'coiled-coil domain containing 7',\n",
       "  'synonyms': 'BIOT2|BioT2-A|BioT2-B|BioT2-C|C10orf68',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:19010',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ADGRG5',\n",
       "  'name': 'adhesion G protein-coupled receptor G5',\n",
       "  'synonyms': 'GPR114|PGR27',\n",
       "  'description': 'This gene encodes a member of the adhesion family of G-protein coupled receptors. Members of this family are characterized by long N-termini and multiple functional domains. They may play a role in the immune system as well as in the central nervous system. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Jan 2016].'},\n",
       " {'id': 'HGNC:21961',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ZSCAN25',\n",
       "  'name': 'zinc finger and SCAN domain containing 25',\n",
       "  'synonyms': 'ZNF498',\n",
       "  'description': 'This gene encodes a protein that bears some similarity to zinc finger proteins, which are involved in DNA binding and protein-protein interactions. Multiple alternatively spliced transcript variants have been identified, but the full-length nature for most of them has not been determined. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'NCBIGene:221943',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'NCBIGene:221946',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'LOC221946',\n",
       "  'name': 'uncharacterized LOC221946',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:23514',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'BEND7',\n",
       "  'name': 'BEN domain containing 7',\n",
       "  'synonyms': 'C10orf30',\n",
       "  'description': nan},\n",
       " {'id': 'NCBIGene:253129',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'HGNC:21484',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MCM9',\n",
       "  'name': 'minichromosome maintenance 9 homologous recombination repair factor',\n",
       "  'synonyms': 'C6orf61|MCMDC1|ODG4|dJ329L24.1|dJ329L24.3',\n",
       "  'description': 'The protein encoded by this gene is a member of the mini-chromosome maintenance (MCM) protein family that are essential for the initiation of eukaryotic genome replication. Binding of this protein to chromatin has been shown to be a pre-requisite for recruiting the MCM2-7 helicase to DNA replication origins. This protein also binds, and is a positive regulator of, the chromatin licensing and DNA replication factor 1, CDT1. [provided by RefSeq, Nov 2010].'},\n",
       " {'id': 'NCBIGene:256159',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SMIM32',\n",
       "  'name': 'small integral membrane protein 32',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:21397',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SCML4',\n",
       "  'name': 'Scm polycomb group protein like 4',\n",
       "  'synonyms': 'dJ47M23.1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:19653',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SSX7',\n",
       "  'name': 'SSX family member 7',\n",
       "  'synonyms': nan,\n",
       "  'description': 'The product of this gene belongs to the family of highly homologous synovial sarcoma X (SSX) breakpoint proteins. These proteins may function as transcriptional repressors. They are also capable of eliciting spontaneously humoral and cellular immune responses in cancer patients, and are potentially useful targets in cancer vaccine-based immunotherapy. SSX1, SSX2 and SSX4 genes have been involved in the t(X;18) translocation characteristically found in all synovial sarcomas. This gene appears not to be involved in this type of chromosome translocation. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:19654',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SSX8P',\n",
       "  'name': 'SSX family member 8, pseudogene',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'NCBIGene:283071',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CDNF',\n",
       "  'name': 'cerebral dopamine neurotrophic factor',\n",
       "  'synonyms': 'ARMETL1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:23729',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MKX',\n",
       "  'name': 'mohawk homeobox',\n",
       "  'synonyms': 'C10orf48|IFRX|IRXL1',\n",
       "  'description': 'The protein encoded by this gene is an IRX family-related homeobox protein that may play a role in cell adhesion. Studies in mice suggest that this protein may be a regulator of tendon development. Two transcript variants encoding the same protein have been found for this gene.[provided by RefSeq, Jun 2011].'},\n",
       " {'id': 'NCBIGene:283196',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'HGNC:24473',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'OLFML1',\n",
       "  'name': 'olfactomedin like 1',\n",
       "  'synonyms': 'MVAL564|UNQ564',\n",
       "  'description': nan},\n",
       " {'id': 'NCBIGene:283328',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'HGNC:27471',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TRHDE-AS1',\n",
       "  'name': 'TRHDE antisense RNA 1',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'NCBIGene:283401',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SCN8A',\n",
       "  'name': 'sodium voltage-gated channel alpha subunit 8',\n",
       "  'synonyms': 'BFIS5|CERIII|CIAT|DEE13|EIEE13|MED|MYOCL2|NaCh6|Nav1.6|PN4',\n",
       "  'description': 'This gene encodes a member of the sodium channel alpha subunit gene family. The encoded protein forms the ion pore region of the voltage-gated sodium channel. This protein is essential for the rapid membrane depolarization that occurs during the formation of the action potential in excitable neurons. Mutations in this gene are associated with cognitive disability, pancerebellar atrophy and ataxia. Alternate splicing results in multiple transcript variants.[provided by RefSeq, May 2010].'},\n",
       " {'id': 'NCBIGene:283476',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HS6ST3',\n",
       "  'name': 'heparan sulfate 6-O-sulfotransferase 3',\n",
       "  'synonyms': 'HS6ST-3',\n",
       "  'description': 'Heparan sulfate (HS) sulfotransferases, such as HS6ST3, modify HS to generate structures required for interactions between HS and a variety of proteins. These interactions are implicated in proliferation and differentiation, adhesion, migration, inflammation, blood coagulation, and other diverse processes (Habuchi et al., 2000 [PubMed 10644753]).[supplied by OMIM, Mar 2008].'},\n",
       " {'id': 'NCBIGene:283528',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'HGNC:25445',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GPR137C',\n",
       "  'name': 'G protein-coupled receptor 137C',\n",
       "  'synonyms': 'TM7SF1L2',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:20117',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'LINC00523',\n",
       "  'name': 'long intergenic non-protein coding RNA 523',\n",
       "  'synonyms': 'C14orf70',\n",
       "  'description': nan},\n",
       " {'id': 'NCBIGene:283732',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'NCBIGene:283745',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'HGNC:8909',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PGP',\n",
       "  'name': 'phosphoglycolate phosphatase',\n",
       "  'synonyms': 'AUM|G3PP|PGPase',\n",
       "  'description': nan},\n",
       " {'id': 'NCBIGene:283880',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'HGNC:26649',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'C16orf54',\n",
       "  'name': 'chromosome 16 open reading frame 54',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:26874',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'KRT18P55',\n",
       "  'name': 'keratin 18 pseudogene 55',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:23089',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SLC13A5',\n",
       "  'name': 'solute carrier family 13 member 5',\n",
       "  'synonyms': 'DEE25|EIEE25|INDY|NACT|mIndy',\n",
       "  'description': 'This gene encodes a protein belonging to the solute carrier family 13 group of proteins. This family member is a sodium-dependent citrate cotransporter that may regulate metabolic processes. Mutations in this gene cause early infantile epileptic encephalopathy 25. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Aug 2014].'},\n",
       " {'id': 'NCBIGene:284130',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'HGNC:26794',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TMEM105',\n",
       "  'name': 'TMEM105 long non-coding RNA',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'NCBIGene:284211',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'NCBIGene:284241',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'LOC284241',\n",
       "  'name': 'uncharacterized LOC284241',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'NCBIGene:284372',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ZNF616',\n",
       "  'name': 'zinc finger protein 616',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'NCBIGene:284444',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'NCBIGene:284475',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'NCBIGene:284505',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'LINC01555',\n",
       "  'name': 'long intergenic non-protein coding RNA 1555',\n",
       "  'synonyms': 'C1orf180',\n",
       "  'description': nan},\n",
       " {'id': 'NCBIGene:284808',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'HGNC:27669',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MIF-AS1',\n",
       "  'name': 'MIF antisense RNA 1',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:30439',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SH2D6',\n",
       "  'name': 'SH2 domain containing 6',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'NCBIGene:285168',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'NCBIGene:285485',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'HGNC:22042',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ARL10',\n",
       "  'name': 'ADP ribosylation factor like GTPase 10',\n",
       "  'synonyms': 'ARL10A',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:44371',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'FAM86B3P',\n",
       "  'name': 'family with sequence similarity 86, member A pseudogene',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'NCBIGene:286121',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'NCBIGene:286166',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'HGNC:20406',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'KRT6C',\n",
       "  'name': 'keratin 6C',\n",
       "  'synonyms': 'K6E|KRT6E|PPKNEFD',\n",
       "  'description': 'Keratins are intermediate filament proteins responsible for the structural integrity of epithelial cells and are subdivided into epithelial keratins and hair keratins. The type II keratins are clustered in a region of chromosome 12q13. [provided by RefSeq, Jul 2009].'},\n",
       " {'id': 'HGNC:19739',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TTC6',\n",
       "  'name': 'tetratricopeptide repeat domain 6',\n",
       "  'synonyms': 'C14orf25|NCRNA00291',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:20785',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PPM1J',\n",
       "  'name': 'protein phosphatase, Mg2+/Mn2+ dependent 1J',\n",
       "  'synonyms': 'PP2CZ|PP2Czeta|PPP2CZ',\n",
       "  'description': 'This gene encodes the serine/threonine protein phosphatase. The mouse homolog of this gene apparently belongs to the protein phosphatase 2C family of genes. The exact function of this gene is not yet known. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'NCBIGene:338575',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'NCBIGene:338850',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'NCBIGene:339071',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'NCBIGene:339685',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'LOC339685',\n",
       "  'name': 'uncharacterized LOC339685',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:27937',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ESPNL',\n",
       "  'name': 'espin like',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:27960',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SLC6A19',\n",
       "  'name': 'solute carrier family 6 member 19',\n",
       "  'synonyms': 'B0AT1|HND',\n",
       "  'description': 'This gene encodes a system B(0) transmembrane protein that actively transports most neutral amino acids across the apical membrane of epithelial cells. Mutations in this gene may result in Hartnup disorder, an inherited disease with symptoms such as pellagra, cerebellar ataxia, and psychosis. The expression and function of B0AT1 (SLC6A19) in intestinal cells depends on the presence of the accessory protein angiotensin-converting enzyme 2 (ACE2) which, among other functions, acts as a chaperone for membrane trafficking of B0AT1. The ACE2 is also the cellular receptor for severe acute respiratory syndrome-coronavirus (SARS-CoV) and for SARS-CoV-2 that is causing the coronavirus 2019 (COVID-19) pandemic [provided by RefSeq, Jul 2020].'},\n",
       " {'id': 'NCBIGene:340238',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'NCBIGene:340370',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'NCBIGene:340868',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'NCBIGene:342024',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'NCBIGene:343961',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'NCBIGene:347582',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'HGNC:19408',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'LRRTM1',\n",
       "  'name': 'leucine rich repeat transmembrane neuronal 1',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'NCBIGene:347875',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'NCBIGene:348143',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'HGNC:30931',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ACSM2B',\n",
       "  'name': 'acyl-CoA synthetase medium chain family member 2B',\n",
       "  'synonyms': 'ACSM2|HXMA|HYST1046',\n",
       "  'description': nan},\n",
       " {'id': 'NCBIGene:350222',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'HGNC:3114',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'E2F2',\n",
       "  'name': 'E2F transcription factor 2',\n",
       "  'synonyms': 'E2F-2',\n",
       "  'description': 'The protein encoded by this gene is a member of the E2F family of transcription factors. The E2F family plays a crucial role in the control of cell cycle and action of tumor suppressor proteins and is also a target of the transforming proteins of small DNA tumor viruses. The E2F proteins contain several evolutionally conserved domains found in most members of the family. These domains include a DNA binding domain, a dimerization domain which determines interaction with the differentiation regulated transcription factor proteins (DP), a transactivation domain enriched in acidic amino acids, and a tumor suppressor protein association domain which is embedded within the transactivation domain. This protein and another 2 members, E2F1 and E2F3, have an additional cyclin binding domain. This protein binds specifically to retinoblastoma protein pRB in a cell-cycle dependent manner, and it exhibits overall 46% amino acid identity to E2F1. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:549',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'AOC2',\n",
       "  'name': 'amine oxidase copper containing 2',\n",
       "  'synonyms': 'DAO2|RAO|SSAO',\n",
       "  'description': 'Copper amine oxidases catalyze the oxidative conversion of amines to aldehydes and ammonia in the presence of copper and quinone cofactor. This gene shows high sequence similarity to copper amine oxidases from various species ranging from bacteria to mammals. The protein contains several conserved motifs including the active site of amine oxidases and the histidine residues that likely bind copper. It may be a critical modulator of signal transmission in retina, possibly by degrading the biogenic amines dopamine, histamine, and putrescine. This gene may be a candidate gene for hereditary ocular diseases. Alternate splicing results in multiple transcript variants. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:3513',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'EXT2',\n",
       "  'name': 'exostosin glycosyltransferase 2',\n",
       "  'synonyms': 'SOTV|SSMS',\n",
       "  'description': 'This gene encodes one of two glycosyltransferases involved in the chain elongation step of heparan sulfate biosynthesis. Mutations in this gene cause the type II form of multiple exostoses. Alternatively spliced transcript variants encoding different isoforms have been noted for this gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:4392',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GNAS',\n",
       "  'name': 'GNAS complex locus',\n",
       "  'synonyms': 'AHO|C20orf45|GNAS1|GPSA|GSA|GSP|NESP|PITA3|POH|SCG6|SgVI',\n",
       "  'description': \"This locus has a highly complex imprinted expression pattern. It gives rise to maternally, paternally, and biallelically expressed transcripts that are derived from four alternative promoters and 5' exons. Some transcripts contain a differentially methylated region (DMR) at their 5' exons, and this DMR is commonly found in imprinted genes and correlates with transcript expression. An antisense transcript is produced from an overlapping locus on the opposite strand. One of the transcripts produced from this locus, and the antisense transcript, are paternally expressed noncoding RNAs, and may regulate imprinting in this region. In addition, one of the transcripts contains a second overlapping ORF, which encodes a structurally unrelated protein - Alex. Alternative splicing of downstream exons is also observed, which results in different forms of the stimulatory G-protein alpha subunit, a key element of the classical signal transduction pathway linking receptor-ligand interactions with the activation of adenylyl cyclase and a variety of cellular reponses. Multiple transcript variants encoding different isoforms have been found for this gene. Mutations in this gene result in pseudohypoparathyroidism type 1a, pseudohypoparathyroidism type 1b, Albright hereditary osteodystrophy, pseudopseudohypoparathyroidism, McCune-Albright syndrome, progressive osseus heteroplasia, polyostotic fibrous dysplasia of bone, and some pituitary tumors. [provided by RefSeq, Aug 2012].\"},\n",
       " {'id': 'HGNC:4650',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GTF2E1',\n",
       "  'name': 'general transcription factor IIE subunit 1',\n",
       "  'synonyms': 'FE|TF2E1|TFIIE-A',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:4741',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'H2AZ1',\n",
       "  'name': 'H2A.Z variant histone 1',\n",
       "  'synonyms': 'H2A.Z-1|H2A.z|H2A/z|H2AFZ|H2AZ',\n",
       "  'description': 'Histones are basic nuclear proteins that are responsible for the nucleosome structure of the chromosomal fiber in eukaryotes. Nucleosomes consist of approximately 146 bp of DNA wrapped around a histone octamer composed of pairs of each of the four core histones (H2A, H2B, H3, and H4). The chromatin fiber is further compacted through the interaction of a linker histone, H1, with the DNA between the nucleosomes to form higher order chromatin structures. This gene encodes a replication-independent member of the histone H2A family that is distinct from other members of the family. Studies in mice have shown that this particular histone is required for embryonic development and indicate that lack of functional histone H2A leads to embryonic lethality. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:5394',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CFI',\n",
       "  'name': 'complement factor I',\n",
       "  'synonyms': 'AHUS3|ARMD13|C3BINA|C3b-INA|FI|IF|KAF',\n",
       "  'description': 'This gene encodes a serine proteinase that is essential for regulating the complement cascade. The encoded preproprotein is cleaved to produce both heavy and light chains, which are linked by disulfide bonds to form a heterodimeric glycoprotein. This heterodimer can cleave and inactivate the complement components C4b and C3b, and it prevents the assembly of the C3 and C5 convertase enzymes. Defects in this gene cause complement factor I deficiency, an autosomal recessive disease associated with a susceptibility to pyogenic infections. Mutations in this gene have been associated with a predisposition to atypical hemolytic uremic syndrome, a disease characterized by acute renal failure, microangiopathic hemolytic anemia and thrombocytopenia. Primary glomerulonephritis with immune deposits and age-related macular degeneration are other conditions associated with mutations of this gene. [provided by RefSeq, Dec 2015].'},\n",
       " {'id': 'HGNC:5472',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'IGFBP3',\n",
       "  'name': 'insulin like growth factor binding protein 3',\n",
       "  'synonyms': 'BP-53|IBP3',\n",
       "  'description': 'This gene is a member of the insulin-like growth factor binding protein (IGFBP) family and encodes a protein with an IGFBP domain and a thyroglobulin type-I domain. The protein forms a ternary complex with insulin-like growth factor acid-labile subunit (IGFALS) and either insulin-like growth factor (IGF) I or II. In this form, it circulates in the plasma, prolonging the half-life of IGFs and altering their interaction with cell surface receptors. Alternate transcriptional splice variants, encoding different isoforms, have been characterized. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:6695',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'LRP3',\n",
       "  'name': 'LDL receptor related protein 3',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:8091',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'OAT',\n",
       "  'name': 'ornithine aminotransferase',\n",
       "  'synonyms': 'GACR|HOGA|OATASE|OKT',\n",
       "  'description': 'This gene encodes the mitochondrial enzyme ornithine aminotransferase, which is a key enzyme in the pathway that converts arginine and ornithine into the major excitatory and inhibitory neurotransmitters glutamate and GABA. Mutations that result in a deficiency of this enzyme cause the autosomal recessive eye disease Gyrate Atrophy. Alternatively spliced transcript variants encoding different isoforms have been described. Related pseudogenes have been defined on the X chromosome. [provided by RefSeq, Jan 2010].'},\n",
       " {'id': 'HGNC:8541',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'P2RY2',\n",
       "  'name': 'purinergic receptor P2Y2',\n",
       "  'synonyms': 'HP2U|P2RU1|P2U|P2U1|P2UR|P2Y2|P2Y2R',\n",
       "  'description': 'The product of this gene belongs to the family of P2 receptors, which is activated by extracellular nucleotides and subdivided into P2X ligand-gated ion channels and P2Y G-protein coupled receptors. This family has several receptor subtypes with different pharmacological selectivity, which overlaps in some cases, for various adenosine and uridine nucleotides. This receptor, found on many cell types, is activated by ATP and UTP and is reported to be overexpressed on some cancer cell types. It is involved in many cellular functions, such as proliferation, apoptosis and inflammation. Three transcript variants encoding the same protein have been identified for this gene. [provided by RefSeq, Mar 2013].'},\n",
       " {'id': 'HGNC:8759',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PDC',\n",
       "  'name': 'phosducin',\n",
       "  'synonyms': 'MEKA|PHD|PhLOP|PhLP',\n",
       "  'description': 'This gene encodes a phosphoprotein, which is located in the outer and inner segments of the rod cells in the retina. This protein may participate in the regulation of visual phototransduction or in the integration of photoreceptor metabolism. It modulates the phototransduction cascade by interacting with the beta and gamma subunits of the retinal G-protein transducin. This gene is a potential candidate gene for retinitis pigmentosa and Usher syndrome type II. Alternatively spliced transcript variants encoding different isoforms have been identified. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:9005',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PITX2',\n",
       "  'name': 'paired like homeodomain 2',\n",
       "  'synonyms': 'ARP1|ASGD4|Brx1|IDG2|IGDS|IGDS2|IHG2|IRID2|Otlx2|PTX2|RGS|RIEG|RIEG1|RS',\n",
       "  'description': 'This gene encodes a member of the RIEG/PITX homeobox family, which is in the bicoid class of homeodomain proteins. The encoded protein acts as a transcription factor and regulates procollagen lysyl hydroxylase gene expression. This protein plays a role in the terminal differentiation of somatotroph and lactotroph cell phenotypes, is involved in the development of the eye, tooth and abdominal organs, and acts as a transcriptional regulator involved in basal and hormone-regulated activity of prolactin. Mutations in this gene are associated with Axenfeld-Rieger syndrome, iridogoniodysgenesis syndrome, and sporadic cases of Peters anomaly. A similar protein in other vertebrates is involved in the determination of left-right asymmetry during development. Alternatively spliced transcript variants encoding distinct isoforms have been described. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:9605',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PTGS2',\n",
       "  'name': 'prostaglandin-endoperoxide synthase 2',\n",
       "  'synonyms': 'COX-2|COX2|GRIPGHS|PGG/HS|PGHS-2|PHS-2|hCox-2',\n",
       "  'description': 'Prostaglandin-endoperoxide synthase (PTGS), also known as cyclooxygenase, is the key enzyme in prostaglandin biosynthesis, and acts both as a dioxygenase and as a peroxidase. There are two isozymes of PTGS: a constitutive PTGS1 and an inducible PTGS2, which differ in their regulation of expression and tissue distribution. This gene encodes the inducible isozyme. It is regulated by specific stimulatory events, suggesting that it is responsible for the prostanoid biosynthesis involved in inflammation and mitogenesis. [provided by RefSeq, Feb 2009].'},\n",
       " {'id': 'HGNC:9702',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PURB',\n",
       "  'name': 'purine rich element binding protein B',\n",
       "  'synonyms': 'PURBETA',\n",
       "  'description': 'This gene product is a sequence-specific, single-stranded DNA-binding protein. It binds preferentially to the single strand of the purine-rich element termed PUR, which is present at origins of replication and in gene flanking regions in a variety of eukaryotes from yeasts through humans. Thus, it is implicated in the control of both DNA replication and transcription. Deletion of this gene has been associated with myelodysplastic syndrome and acute myelogenous leukemia. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:9858',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RAP1GAP',\n",
       "  'name': 'RAP1 GTPase activating protein',\n",
       "  'synonyms': 'RAP1GA1|RAP1GAP1|RAP1GAPII|RAPGAP',\n",
       "  'description': 'This gene encodes a type of GTPase-activating-protein (GAP) that down-regulates the activity of the ras-related RAP1 protein. RAP1 acts as a molecular switch by cycling between an inactive GDP-bound form and an active GTP-bound form. The product of this gene, RAP1GAP, promotes the hydrolysis of bound GTP and hence returns RAP1 to the inactive state whereas other proteins, guanine nucleotide exchange factors (GEFs), act as RAP1 activators by facilitating the conversion of RAP1 from the GDP- to the GTP-bound form. In general, ras subfamily proteins, such as RAP1, play key roles in receptor-linked signaling pathways that control cell growth and differentiation. RAP1 plays a role in diverse processes such as cell proliferation, adhesion, differentiation, and embryogenesis. Alternative splicing results in multiple transcript variants encoding distinct proteins. [provided by RefSeq, Aug 2011].'},\n",
       " {'id': 'HGNC:10524',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SALL1',\n",
       "  'name': 'spalt like transcription factor 1',\n",
       "  'synonyms': 'HEL-S-89|HSAL1|Sal-1|TBS|ZNF794',\n",
       "  'description': 'The protein encoded by this gene is a zinc finger transcriptional repressor and may be part of the NuRD histone deacetylase complex (HDAC). Defects in this gene are a cause of Townes-Brocks syndrome (TBS) as well as bronchio-oto-renal syndrome (BOR). Two transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:10580',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SCML1',\n",
       "  'name': 'Scm polycomb group protein like 1',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:10860',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ST6GAL1',\n",
       "  'name': 'ST6 beta-galactoside alpha-2,6-sialyltransferase 1',\n",
       "  'synonyms': 'SIAT1|ST6GalI|ST6N',\n",
       "  'description': 'This gene encodes a member of glycosyltransferase family 29. The encoded protein is a type II membrane protein that catalyzes the transfer of sialic acid from CMP-sialic acid to galactose-containing substrates. The protein, which is normally found in the Golgi but can be proteolytically processed to a soluble form, is involved in the generation of the cell-surface carbohydrate determinants and differentiation antigens HB-6, CD75, and CD76. This gene has been incorrectly referred to as CD75. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Jul 2017].'},\n",
       " {'id': 'HGNC:11444',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'STXBP1',\n",
       "  'name': 'syntaxin binding protein 1',\n",
       "  'synonyms': 'DEE4|MUNC18-1|N-Sec1|NSEC1|P67|RBSEC1|UNC18|unc-18A|unc18-1',\n",
       "  'description': 'This gene encodes a syntaxin-binding protein. The encoded protein appears to play a role in release of neurotransmitters via regulation of syntaxin, a transmembrane attachment protein receptor. Mutations in this gene have been associated with infantile epileptic encephalopathy-4. Alternatively spliced transcript variants have been described. [provided by RefSeq, Feb 2010].'},\n",
       " {'id': 'HGNC:11782',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TH',\n",
       "  'name': 'tyrosine hydroxylase',\n",
       "  'synonyms': 'DYT14|DYT5b|TYH',\n",
       "  'description': 'The protein encoded by this gene is involved in the conversion of tyrosine to dopamine. It is the rate-limiting enzyme in the synthesis of catecholamines, hence plays a key role in the physiology of adrenergic neurons. Mutations in this gene have been associated with autosomal recessive Segawa syndrome. Alternatively spliced transcript variants encoding different isoforms have been noted for this gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:3413',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'EPM2A',\n",
       "  'name': 'EPM2A glucan phosphatase, laforin',\n",
       "  'synonyms': 'EPM2|MELF',\n",
       "  'description': 'This gene encodes a dual-specificity phosphatase and may be involved in the regulation of glycogen metabolism. The protein acts on complex carbohydrates to prevent glycogen hyperphosphorylation, thus avoiding the formation of insoluble aggregates. Loss-of-function mutations in this gene have been associated with Lafora disease, a rare, adult-onset recessive neurodegenerative disease, which results in myoclonus epilepsy and usually results in death several years after the onset of symptoms. The disease is characterized by the accumulation of insoluble particles called Lafora bodies, which are derived from glycogen. [provided by RefSeq, Jan 2018].'},\n",
       " {'id': 'HGNC:16821',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NOP14',\n",
       "  'name': 'NOP14 nucleolar protein',\n",
       "  'synonyms': 'C4orf9|NOL14|RES4-25|RES425|UTP2',\n",
       "  'description': 'This gene encodes a protein that plays a role in pre-18s rRNA processing and small ribosomal subunit assembly. The encoded protein may be involved in the regulation of pancreatic cancer cell proliferation and migration. Alternative splicing results in multiple transcript variants. [provided by RefSeq, May 2014].'},\n",
       " {'id': 'HGNC:550',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'AOC3',\n",
       "  'name': 'amine oxidase copper containing 3',\n",
       "  'synonyms': 'HPAO|SSAO|VAP-1|VAP1',\n",
       "  'description': 'This gene encodes a member of the semicarbazide-sensitive amine oxidase family. Copper amine oxidases catalyze the oxidative conversion of amines to aldehydes in the presence of copper and quinone cofactor. The encoded protein is localized to the cell surface, has adhesive properties as well as monoamine oxidase activity, and may be involved in leukocyte trafficking. Alterations in levels of the encoded protein may be associated with many diseases, including diabetes mellitus. A pseudogene of this gene has been described and is located approximately 9-kb downstream on the same chromosome. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Apr 2013].'},\n",
       " {'id': 'HGNC:11907',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TNFRSF10D',\n",
       "  'name': 'TNF receptor superfamily member 10d',\n",
       "  'synonyms': 'CD264|DCR2|TRAIL-R4|TRAILR4|TRUNDD',\n",
       "  'description': 'The protein encoded by this gene is a member of the TNF-receptor superfamily. This receptor contains an extracellular TRAIL-binding domain, a transmembrane domain, and a truncated cytoplamic death domain. This receptor does not induce apoptosis, and has been shown to play an inhibitory role in TRAIL-induced cell apoptosis. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:11904',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TNFRSF10A',\n",
       "  'name': 'TNF receptor superfamily member 10a',\n",
       "  'synonyms': 'APO2|CD261|DR4|TRAILR-1|TRAILR1',\n",
       "  'description': 'The protein encoded by this gene is a member of the TNF-receptor superfamily. This receptor is activated by tumor necrosis factor-related apoptosis inducing ligand (TNFSF10/TRAIL), and thus transduces cell death signal and induces cell apoptosis. Studies with FADD-deficient mice suggested that FADD, a death domain containing adaptor protein, is required for the apoptosis mediated by this protein. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:4877',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HESX1',\n",
       "  'name': 'HESX homeobox 1',\n",
       "  'synonyms': 'ANF|CPHD5|RPX',\n",
       "  'description': 'This gene encodes a conserved homeobox protein that is a transcriptional repressor in the developing forebrain and pituitary gland. Mutations in this gene are associated with septooptic dysplasia, HESX1-related growth hormone deficiency, and combined pituitary hormone deficiency. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:2700',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DCLK1',\n",
       "  'name': 'doublecortin like kinase 1',\n",
       "  'synonyms': 'CL1|CLICK1|DCAMKL1|DCDC3A|DCLK',\n",
       "  'description': 'This gene encodes a member of the protein kinase superfamily and the doublecortin family. The protein encoded by this gene contains two N-terminal doublecortin domains, which bind microtubules and regulate microtubule polymerization, a C-terminal serine/threonine protein kinase domain, which shows substantial homology to Ca2+/calmodulin-dependent protein kinase, and a serine/proline-rich domain in between the doublecortin and the protein kinase domains, which mediates multiple protein-protein interactions. The microtubule-polymerizing activity of the encoded protein is independent of its protein kinase activity. The encoded protein is involved in several different cellular processes, including neuronal migration, retrograde transport, neuronal apoptosis and neurogenesis. This gene is up-regulated by brain-derived neurotrophic factor and associated with memory and general cognitive abilities. Multiple transcript variants generated by two alternative promoter usage and alternative splicing have been reported, but the full-length nature and biological validity of some variants have not been defined. These variants encode different isoforms, which are differentially expressed and have different kinase activities.[provided by RefSeq, Sep 2010].'},\n",
       " {'id': 'HGNC:30757',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'STXBP5L',\n",
       "  'name': 'syntaxin binding protein 5 like',\n",
       "  'synonyms': 'LLGL4',\n",
       "  'description': 'The protein encoded by this gene is similar to syntaxin-binding protein 5 and contains ten N-terminal WD40 repeats, four variable region WD40 repeats, and a C-terminal R-SNARE domain. Studies of the orthologous proteins in mouse and rat have shown that the encoded protein may inhibit exocytosis in neurosecretory cells, and may negatively regulate the secretion of insulin. A missense variant in this gene is likely the cause of an infantile-onset neurodegenerative disorder diagnosed in two siblings of consanguineous parents. [provided by RefSeq, Jan 2017].'},\n",
       " {'id': 'HGNC:19960',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'KIAA0586',\n",
       "  'name': 'KIAA0586',\n",
       "  'synonyms': 'JBTS23|SRTD14|Talpid3',\n",
       "  'description': 'This gene encodes a conserved centrosomal protein that functions in ciliogenesis and responds to hedgehog signaling. Mutations in this gene causes Joubert syndrome 23. Alternative splicing results in multiple transcript variants and protein isoforms. [provided by RefSeq, Aug 2016].'},\n",
       " {'id': 'HGNC:16894',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MFSD10',\n",
       "  'name': 'major facilitator superfamily domain containing 10',\n",
       "  'synonyms': 'TETRAN|TETTRAN',\n",
       "  'description': 'This gene encodes a member of the major facilitator superfamily of transporter proteins. The encoded protein likely functions in efflux of organic anions, including the non-steroidal anti-inflammatory drugs indomethacin and diclofenac. Alternatively spliced transcript variants have been described. [provided by RefSeq, Mar 2009].'},\n",
       " {'id': 'HGNC:30915',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DCAF7',\n",
       "  'name': 'DDB1 and CUL4 associated factor 7',\n",
       "  'synonyms': 'AN11|HAN11|SWAN-1|WDR68',\n",
       "  'description': 'This gene encodes a protein with multiple WD40 repeats which facilitate protein-protein interactions and thereby enable the assembly of multiprotein complexes. This protein has been shown to function as a scaffold protein for protein complexes involved in kinase signaling. This highly conserved gene is present in eukaryotic plants, fungi, and animals. The ortholog of this gene was first identified in plants as a key regulator of anthocyanin biosynthesis and flower pigmentation. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Feb 2014].'},\n",
       " {'id': 'HGNC:16937',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ARL6IP5',\n",
       "  'name': 'ADP ribosylation factor like GTPase 6 interacting protein 5',\n",
       "  'synonyms': 'DERP11|GTRAP3-18|HSPC127|JWA|PRAF3|Yip6b|addicsin|hp22|jmx',\n",
       "  'description': 'Expression of this gene is affected by vitamin A. The encoded protein of this gene may be associated with the cytoskeleton. A similar protein in rats may play a role in the regulation of cell differentiation. The rat protein binds and inhibits the cell membrane glutamate transporter EAAC1. The expression of the rat gene is upregulated by retinoic acid, which results in a specific reduction in EAAC1-mediated glutamate transport. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:14350',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ARID3B',\n",
       "  'name': 'AT-rich interaction domain 3B',\n",
       "  'synonyms': 'BDP|DRIL2',\n",
       "  'description': 'This gene encodes a member of the ARID (AT-rich interaction domain) family of DNA-binding proteins. The encoded protein is homologous with two proteins that bind to the retinoblastoma gene product, and also with the mouse Bright and Drosophila dead ringer proteins. A pseudogene on chromosome 1p31 exists for this gene. Members of the ARID family have roles in embryonic patterning, cell lineage gene regulation, cell cycle control, transcriptional regulation and possibly in chromatin structure modification. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:16701',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MYL12A',\n",
       "  'name': 'myosin light chain 12A',\n",
       "  'synonyms': 'HEL-S-24|MLC-2B|MLCB|MRCL3|MRLC3|MYL2B',\n",
       "  'description': 'This gene encodes a nonsarcomeric myosin regulatory light chain. This protein is activated by phosphorylation and regulates smooth muscle and non-muscle cell contraction. This protein may also be involved in DNA damage repair by sequestering the transcriptional regulator apoptosis-antagonizing transcription factor (AATF)/Che-1 which functions as a repressor of p53-driven apoptosis. Alternate splicing results in multiple transcript variants. A pseudogene of this gene is found on chromosome 8.[provided by RefSeq, Dec 2014].'},\n",
       " {'id': 'HGNC:19943',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TRAPPC4',\n",
       "  'name': 'trafficking protein particle complex subunit 4',\n",
       "  'synonyms': 'CGI-104|HSPC172|NEDESBA|PTD009|SBDN|SYNBINDIN|TRS23',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:1300',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'C21orf58',\n",
       "  'name': 'chromosome 21 open reading frame 58',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:17616',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'IL17RD',\n",
       "  'name': 'interleukin 17 receptor D',\n",
       "  'synonyms': 'HH18|IL-17RD|IL17RLM|SEF',\n",
       "  'description': 'This gene encodes a membrane protein belonging to the interleukin-17 receptor (IL-17R) protein family. The encoded protein is a component of the interleukin-17 receptor signaling complex, and the interaction between this protein and IL-17R does not require the interleukin. The gene product also affects fibroblast growth factor signaling, inhibiting or stimulating growth through MAPK/ERK signaling. Alternate splicing generates multiple transcript variants encoding distinct isoforms. [provided by RefSeq, Jan 2016].'},\n",
       " {'id': 'HGNC:26026',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SOHLH2',\n",
       "  'name': 'spermatogenesis and oogenesis specific basic helix-loop-helix 2',\n",
       "  'synonyms': 'SOSF2|SPATA28|TEB1|bHLHe81',\n",
       "  'description': 'This gene encodes one of testis-specific transcription factors which are essential for spermatogenesis, oogenesis and folliculogenesis. This gene is located on chromosome 13. The proteins encoded by this gene and another testis-specific transcription factor, SOHLH1, can form heterodimers, in addition to homodimers. There is a read-through locus (GeneID: 100526761) that shares sequence identity with this gene and the upstream CCDC169 (GeneID: 728591). Alternatively spliced transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Aug 2013].'},\n",
       " {'id': 'HGNC:26091',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TRMT12',\n",
       "  'name': 'tRNA methyltransferase 12 homolog',\n",
       "  'synonyms': 'TRM12|TYW2',\n",
       "  'description': 'Wybutosine (yW) is a hypermodified guanosine at the 3-prime position adjacent to the anticodon of phenylalanine tRNA that stabilizes codon-anticodon interactions during decoding on the ribosome. TRMT12 is the human homolog of a yeast gene essential for yW synthesis (Noma and Suzuki, 2006).[supplied by OMIM, Mar 2008].'},\n",
       " {'id': 'HGNC:19286',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SCYL2',\n",
       "  'name': 'SCY1 like pseudokinase 2',\n",
       "  'synonyms': 'AMC4|AMCNACC|CVAK104',\n",
       "  'description': 'The protein encoded by this gene associates with clathrin-coated complexes at the plasma membrane and with endocytic coated vesicles. The encoded protein phosphorylates the beta2 subunit of the plasma membrane adapter complex AP2 and interacts with clathrin, showing involvement in clathrin-dependent pathways between the trans-Golgi network and the endosomal system. In addition, this protein has a role in the Wnt signaling pathway by targeting frizzled 5 (Fzd5) for lysosomal degradation. Two transcript variants encoding the same protein have been found for this gene. [provided by RefSeq, Dec 2015].'},\n",
       " {'id': 'HGNC:20444',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MBD5',\n",
       "  'name': 'methyl-CpG binding domain protein 5',\n",
       "  'synonyms': 'MRD1',\n",
       "  'description': 'This gene encodes a member of the methyl-CpG-binding domain (MBD) family. The MBD consists of about 70 residues and is the minimal region required for a methyl-CpG-binding protein binding specifically to methylated DNA. In addition to the MBD domain, this protein contains a PWWP domain (Pro-Trp-Trp-Pro motif), which consists of 100-150 amino acids and is found in numerous proteins that are involved in cell division, growth and differentiation. Mutations in this gene cause an autosomal dominant type of cognitive disability. The encoded protein interacts with the polycomb repressive complex PR-DUB which catalyzes the deubiquitination of a lysine residue of histone 2A. Haploinsufficiency of this gene is associated with a syndrome involving microcephaly, intellectual disabilities, severe speech impairment, and seizures. Alternatively spliced transcript variants have been found, but their full-length nature is not determined. [provided by RefSeq, Jul 2017].'},\n",
       " {'id': 'HGNC:14583',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'VPS11',\n",
       "  'name': 'VPS11 core subunit of CORVET and HOPS complexes',\n",
       "  'synonyms': 'END1|HLD12|PEP5|RNF108|hVPS11',\n",
       "  'description': 'Vesicle mediated protein sorting plays an important role in segregation of intracellular molecules into distinct organelles. Genetic studies in yeast have identified more than 40 vacuolar protein sorting (VPS) genes involved in vesicle transport to vacuoles. This gene encodes the human homolog of yeast class C Vps11 protein. The mammalian class C Vps proteins are predominantly associated with late endosomes/lysosomes, and like their yeast counterparts, may mediate vesicle trafficking steps in the endosome/lysosome pathway. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Feb 2014].'},\n",
       " {'id': 'HGNC:23992',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RTP4',\n",
       "  'name': 'receptor transporter protein 4',\n",
       "  'synonyms': 'IFRG28|Z3CXXC4',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:24025',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ACTR6',\n",
       "  'name': 'actin related protein 6',\n",
       "  'synonyms': 'ARP6|CDA12|HSPC281|MSTP136|hARP6|hARPX',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:21616',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TNS3',\n",
       "  'name': 'tensin 3',\n",
       "  'synonyms': 'TEM6|TENS1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:14051',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MRPS6',\n",
       "  'name': 'mitochondrial ribosomal protein S6',\n",
       "  'synonyms': 'C21orf101|MRP-S6|RPMS6|S6mt',\n",
       "  'description': 'Mammalian mitochondrial ribosomal proteins are encoded by nuclear genes and help in protein synthesis within the mitochondrion. Mitochondrial ribosomes (mitoribosomes) consist of a small 28S subunit and a large 39S subunit. They have an estimated 75% protein to rRNA composition compared to prokaryotic ribosomes, where this ratio is reversed. Another difference between mammalian mitoribosomes and prokaryotic ribosomes is that the latter contain a 5S rRNA. Among different species, the proteins comprising the mitoribosome differ greatly in sequence, and sometimes in biochemical properties, which prevents easy recognition by sequence homology. This gene encodes a 28S subunit protein that belongs to the ribosomal protein S6P family. Pseudogenes corresponding to this gene are found on chromosomes 1p and 12q. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:15829',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ELOVL6',\n",
       "  'name': 'ELOVL fatty acid elongase 6',\n",
       "  'synonyms': 'FACE|FAE|LCE',\n",
       "  'description': 'Fatty acid elongases (EC 6.2.1.3), such as ELOVL6, use malonyl-CoA as a 2-carbon donor in the first and rate-limiting step of fatty acid elongation (Moon et al., 2001 [PubMed 11567032]).[supplied by OMIM, Mar 2008].'},\n",
       " {'id': 'HGNC:28459',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TMUB2',\n",
       "  'name': 'transmembrane and ubiquitin like domain containing 2',\n",
       "  'synonyms': 'FP2653',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:16244',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NPEPL1',\n",
       "  'name': 'aminopeptidase like 1',\n",
       "  'synonyms': 'bA261P9.2',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:23825',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MOB3B',\n",
       "  'name': 'MOB kinase activator 3B',\n",
       "  'synonyms': 'C9orf35|MOB1D|MOBKL2B',\n",
       "  'description': 'The protein encoded by this gene shares similarity with the yeast Mob1 protein. Yeast Mob1 binds Mps1p, a protein kinase essential for spindle pole body duplication and mitotic checkpoint regulation. This gene is located on the opposite strand as the interferon kappa precursor (IFNK) gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:25710',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CEBPA-DT',\n",
       "  'name': 'CEBPA divergent transcript',\n",
       "  'synonyms': 'ADINR|CEBPA-AS1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:26552',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CCDC33',\n",
       "  'name': 'coiled-coil domain containing 33',\n",
       "  'synonyms': 'CT61|HP11097',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:23989',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ACCS',\n",
       "  'name': '1-aminocyclopropane-1-carboxylate synthase homolog (inactive)',\n",
       "  'synonyms': 'ACS|PHACS',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:25917',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'FIZ1',\n",
       "  'name': 'FLT3 interacting zinc finger 1',\n",
       "  'synonyms': 'ZNF798',\n",
       "  'description': 'This gene encodes zinc finger protein, which interacts with a receptor tyrosine kinase involved in the regulation of hematopoietic and lymphoid cells. This gene product also interacts with a transcription factor that regulates the expression of rod-specific genes in retina. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:25464',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ATG16L2',\n",
       "  'name': 'autophagy related 16 like 2',\n",
       "  'synonyms': 'ATG16B|WDR80',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:25135',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'LRSAM1',\n",
       "  'name': 'leucine rich repeat and sterile alpha motif containing 1',\n",
       "  'synonyms': 'CMT2P|RIFLE|TAL',\n",
       "  'description': 'This gene encodes a ring finger protein involved in a variety of functions, including regulation of signaling pathways and cell adhesion, mediation of self-ubiquitylation, and involvement in cargo sorting during receptor endocytosis. Mutations in this gene have been associated with Charcot-Marie-Tooth disease. Multiple transcript variants encoding different isoforms have been identified for this gene. [provided by RefSeq, Jan 2012].'},\n",
       " {'id': 'HGNC:23400',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SEC11C',\n",
       "  'name': 'SEC11 homolog C, signal peptidase complex subunit',\n",
       "  'synonyms': 'SEC11L3|SPC21|SPCS4C',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:15956',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SLC2A13',\n",
       "  'name': 'solute carrier family 2 member 13',\n",
       "  'synonyms': 'HMIT',\n",
       "  'description': nan},\n",
       " {'id': 'NCBIGene:126913',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'HGNC:16216',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'C20orf85',\n",
       "  'name': 'chromosome 20 open reading frame 85',\n",
       "  'synonyms': 'LLC1|bA196N14.1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:25185',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ANKRD54',\n",
       "  'name': 'ankyrin repeat domain 54',\n",
       "  'synonyms': 'LIAR',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:28580',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RTP1',\n",
       "  'name': 'receptor transporter protein 1',\n",
       "  'synonyms': 'Z3CXXC1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:17184',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ASB9',\n",
       "  'name': 'ankyrin repeat and SOCS box containing 9',\n",
       "  'synonyms': nan,\n",
       "  'description': 'This gene encodes a member of the ankyrin repeat and suppressor of cytokine signaling (SOCS) box protein family. Members of this family can interact with the elongin B-C adapter complex via their SOCS box domain and further complex with the cullin and ring box proteins to form E3 ubiquitin ligase complexes. They may function to mediate the substrate-recognition of the E3 ubiquitin ligases. A transcribed pseudogene of this gene has been identified on chromosome 15. Multiple transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Dec 2009].'},\n",
       " {'id': 'HGNC:19766',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ASB14',\n",
       "  'name': 'ankyrin repeat and SOCS box containing 14',\n",
       "  'synonyms': nan,\n",
       "  'description': 'The protein encoded by this gene is a member of the ankyrin repeat and SOCS box-containing (ASB) family of proteins. They contain ankyrin repeat sequence and a SOCS box domain. The SOCS box serves to couple suppressor of cytokine signalling (SOCS) proteins and their binding partners with the elongin B and C complex, possibly targeting them for degradation. Alternative splicing results in multiple transcript variants encoding different isoforms.[provided by RefSeq, Dec 2008].'},\n",
       " {'id': 'HGNC:33111',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ZNF784',\n",
       "  'name': 'zinc finger protein 784',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:21207',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'LINGO2',\n",
       "  'name': 'leucine rich repeat and Ig domain containing 2',\n",
       "  'synonyms': 'LERN3|LRRN6C',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:18053',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PKD1L1',\n",
       "  'name': 'polycystin 1 like 1, transient receptor potential channel interacting',\n",
       "  'synonyms': 'HTX8|PRO19563',\n",
       "  'description': 'This gene encodes a member of the polycystin protein family containing 11 transmembrane domains, a receptor for egg jelly (REJ) domain, and a polycystin-1, lipoxygenase, alpha-toxin (PLAT) domain. The encoded protein may play a role in the male reproductive system. Alternative splice variants have been described but their biological nature has not been determined. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:28337',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'C9orf72',\n",
       "  'name': 'C9orf72-SMCR8 complex subunit',\n",
       "  'synonyms': 'ALSFTD|DENND9|DENNL72|FTDALS|FTDALS1',\n",
       "  'description': \"The protein encoded by this gene plays an important role in the regulation of endosomal trafficking, and has been shown to interact with Rab proteins that are involved in autophagy and endocytic transport. Expansion of a GGGGCC repeat from 2-22 copies to 700-1600 copies in the intronic sequence between alternate 5' exons in transcripts from this gene is associated with 9p-linked ALS (amyotrophic lateral sclerosis) and FTD (frontotemporal dementia) (PMID: 21944778, 21944779). Studies suggest that hexanucleotide expansions could result in the selective stabilization of repeat-containing pre-mRNA, and the accumulation of insoluble dipeptide repeat protein aggregates that could be pathogenic in FTD-ALS patients (PMID: 23393093). Alternative splicing results in multiple transcript variants encoding different isoforms. [provided by RefSeq, Jul 2016].\"},\n",
       " {'id': 'HGNC:22301',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TMED4',\n",
       "  'name': 'transmembrane p24 trafficking protein 4',\n",
       "  'synonyms': 'ERS25|GMP25iso|HNLF|p24a3|p24alpha3',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:23688',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'BCL9L',\n",
       "  'name': 'BCL9 like',\n",
       "  'synonyms': 'B9L|BCL9-2|DLNB11',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:27609',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'EMC10',\n",
       "  'name': 'ER membrane protein complex subunit 10',\n",
       "  'synonyms': 'C19orf63|HSM1|HSS1',\n",
       "  'description': nan},\n",
       " {'id': 'NCBIGene:342098',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'NCBIGene:349298',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'HGNC:3118',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'E2F4',\n",
       "  'name': 'E2F transcription factor 4',\n",
       "  'synonyms': 'E2F-4',\n",
       "  'description': 'The protein encoded by this gene is a member of the E2F family of transcription factors. The E2F family plays a crucial role in the control of cell cycle and action of tumor suppressor proteins and is also a target of the transforming proteins of small DNA tumor viruses. The E2F proteins contain several evolutionally conserved domains found in most members of the family. These domains include a DNA binding domain, a dimerization domain which determines interaction with the differentiation regulated transcription factor proteins (DP), a transactivation domain enriched in acidic amino acids, and a tumor suppressor protein association domain which is embedded within the transactivation domain. This protein binds to all three of the  tumor suppressor proteins pRB, p107 and p130, but with higher affinity to the last two. It plays an important role in the suppression of proliferation-associated genes, and its gene mutation and increased expression may be associated with human cancer. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:129',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ACTA1',\n",
       "  'name': 'actin alpha 1, skeletal muscle',\n",
       "  'synonyms': 'ACTA|ASMA|CFTD|CFTD1|CFTDM|MPFD|NEM1|NEM2|NEM3|SHPM',\n",
       "  'description': 'The product encoded by this gene belongs to the actin family of proteins, which are highly conserved proteins that play a role in cell motility, structure and integrity. Alpha, beta and gamma actin isoforms have been identified, with alpha actins being a major constituent of the contractile apparatus, while beta and gamma actins are involved in the regulation of cell motility. This actin is an alpha actin that is found in skeletal muscle. Mutations in this gene cause a variety of myopathies, including nemaline myopathy, congenital myopathy with excess of thin myofilaments, congenital myopathy with cores, and congenital myopathy with fiber-type disproportion, diseases that lead to muscle fiber defects with manifestations such as hypotonia. [provided by RefSeq, Sep 2019].'},\n",
       " {'id': 'HGNC:132',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ACTB',\n",
       "  'name': 'actin beta',\n",
       "  'synonyms': 'BRWS1|PS1TP5BP1',\n",
       "  'description': 'This gene encodes one of six different actin proteins. Actins are highly conserved proteins that are involved in cell motility, structure, integrity, and intercellular signaling. The encoded protein is a major constituent of the contractile apparatus and one of the two nonmuscle cytoskeletal actins that are ubiquitously expressed. Mutations in this gene cause Baraitser-Winter syndrome 1, which is characterized by intellectual disability with a distinctive facial appearance in human patients. Numerous pseudogenes of this gene have been identified throughout the human genome. [provided by RefSeq, Aug 2017].'},\n",
       " {'id': 'HGNC:226',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ADARB1',\n",
       "  'name': 'adenosine deaminase RNA specific B1',\n",
       "  'synonyms': 'ADAR2|DRABA2|DRADA2|NEDHYMS|RED1',\n",
       "  'description': 'This gene encodes the enzyme responsible for pre-mRNA editing of the glutamate receptor subunit B by site-specific deamination of adenosines. Studies in rat found that this enzyme acted on its own pre-mRNA molecules to convert an AA dinucleotide to an AI dinucleotide which resulted in a new splice site. Alternative splicing of this gene results in several transcript variants, some of which have been characterized by the presence or absence of an ALU cassette insert and a short or long C-terminal region. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:290',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GRK3',\n",
       "  'name': 'G protein-coupled receptor kinase 3',\n",
       "  'synonyms': 'ADRBK2|BARK2',\n",
       "  'description': 'The beta-adrenergic receptor kinase specifically phosphorylates the agonist-occupied form of the beta-adrenergic and related G protein-coupled receptors. Overall, the beta adrenergic receptor kinase 2 has 85% amino acid similarity with beta adrenergic receptor kinase 1, with the protein kinase catalytic domain having 95% similarity. These data suggest the existence of a family of receptor kinases which may serve broadly to regulate receptor function. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:10990',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SLC25A4',\n",
       "  'name': 'solute carrier family 25 member 4',\n",
       "  'synonyms': 'AAC1|ANT|ANT 1|ANT1|MTDPS12|MTDPS12A|PEO2|PEO3|PEOA2|T1',\n",
       "  'description': 'This gene is a member of the mitochondrial carrier subfamily of solute carrier protein genes. The product of this gene functions as a gated pore that translocates ADP from the cytoplasm into the mitochondrial matrix and ATP from the mitochondrial matrix into the cytoplasm. The protein forms a homodimer embedded in the inner mitochondria membrane. Mutations in this gene have been shown to result in autosomal dominant progressive external opthalmoplegia and familial hypertrophic cardiomyopathy. [provided by RefSeq, Jun 2013].'},\n",
       " {'id': 'HGNC:593',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'BIRC5',\n",
       "  'name': 'baculoviral IAP repeat containing 5',\n",
       "  'synonyms': 'API4|EPR-1',\n",
       "  'description': 'This gene is a member of the inhibitor of apoptosis (IAP) gene family, which encode negative regulatory proteins that prevent apoptotic cell death. IAP family members usually contain multiple baculovirus IAP repeat (BIR) domains, but this gene encodes proteins with only a single BIR domain. The encoded proteins also lack a C-terminus RING finger domain. Gene expression is high during fetal development and in most tumors, yet low in adult tissues. Alternatively spliced transcript variants encoding distinct isoforms have been found for this gene. [provided by RefSeq, Jun 2011].'},\n",
       " {'id': 'HGNC:626',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'APRT',\n",
       "  'name': 'adenine phosphoribosyltransferase',\n",
       "  'synonyms': 'AMP|APRTD',\n",
       "  'description': 'Adenine phosphoribosyltransferase belongs to the purine/pyrimidine phosphoribosyltransferase family. A conserved feature of this gene is the distribution of CpG dinucleotides. This enzyme catalyzes the formation of AMP and inorganic pyrophosphate from adenine and 5-phosphoribosyl-1-pyrophosphate (PRPP). It also produces adenine as a by-product of the polyamine biosynthesis pathway. A homozygous deficiency in this enzyme causes 2,8-dihydroxyadenine urolithiasis. Two transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:680',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ARHGDIG',\n",
       "  'name': 'Rho GDP dissociation inhibitor gamma',\n",
       "  'synonyms': 'RHOGDI-3',\n",
       "  'description': 'The GDP-dissociation inhibitors (GDIs) play a primary role in modulating the activation of GTPases by inhibiting the exchange of GDP for GTP. See ARHGDIB (MIM 602843).[supplied by OMIM, Nov 2010].'},\n",
       " {'id': 'HGNC:795',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ATM',\n",
       "  'name': 'ATM serine/threonine kinase',\n",
       "  'synonyms': 'AT1|ATA|ATC|ATD|ATDC|ATE|TEL1|TELO1',\n",
       "  'description': 'The protein encoded by this gene belongs to the PI3/PI4-kinase family. This protein is an important cell cycle checkpoint kinase that phosphorylates; thus, it functions as a regulator of a wide variety of downstream proteins, including tumor suppressor proteins p53 and BRCA1, checkpoint kinase CHK2, checkpoint proteins RAD17 and RAD9, and DNA repair protein NBS1. This protein and the closely related kinase ATR are thought to be master controllers of cell cycle checkpoint signaling pathways that are required for cell response to DNA damage and for genome stability. Mutations in this gene are associated with ataxia telangiectasia, an autosomal recessive disorder. [provided by RefSeq, Aug 2010].'},\n",
       " {'id': 'HGNC:952',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'BARD1',\n",
       "  'name': 'BRCA1 associated RING domain 1',\n",
       "  'synonyms': nan,\n",
       "  'description': 'This gene encodes a protein which interacts with the N-terminal region of BRCA1. In addition to its ability to bind BRCA1 in vivo and in vitro, it shares homology with the 2 most conserved regions of BRCA1: the N-terminal RING motif and the C-terminal BRCT domain. The RING motif is a cysteine-rich sequence found in a variety of proteins that regulate cell growth, including the products of tumor suppressor genes and dominant protooncogenes. This protein also contains 3 tandem ankyrin repeats. The BARD1/BRCA1 interaction is disrupted by tumorigenic amino acid substitutions in BRCA1, implying that the formation of a stable complex between these proteins may be an essential aspect of BRCA1 tumor suppression. This protein may be the target of oncogenic mutations in breast or ovarian cancer. Multiple alternatively spliced transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Sep 2013].'},\n",
       " {'id': 'HGNC:1080',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'POLR3D',\n",
       "  'name': 'RNA polymerase III subunit D',\n",
       "  'synonyms': 'BN51T|C53|RPC4|RPC53|TSBN51',\n",
       "  'description': 'This gene complements a temperature-sensitive mutant isolated from the BHK-21 Syrian hamster cell line. It leads to a block in progression through the G1 phase of the cell cycle at nonpermissive temperatures. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:1100',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'BRCA1',\n",
       "  'name': 'BRCA1 DNA repair associated',\n",
       "  'synonyms': 'BRCAI|BRCC1|BROVCA1|FANCS|IRIS|PNCA4|PPP1R53|PSCP|RNF53',\n",
       "  'description': 'This gene encodes a 190 kD nuclear phosphoprotein that plays a role in maintaining genomic stability, and it also acts as a tumor suppressor. The BRCA1 gene contains 22 exons spanning about 110 kb of DNA. The encoded protein combines with other tumor suppressors, DNA damage sensors, and signal transducers to form a large multi-subunit protein complex known as the BRCA1-associated genome surveillance complex (BASC). This gene product associates with RNA polymerase II, and through the C-terminal domain, also interacts with histone deacetylase complexes. This protein thus plays a role in transcription, DNA repair of double-stranded breaks, and recombination. Mutations in this gene are responsible for approximately 40% of inherited breast cancers and more than 80% of inherited breast and ovarian cancers. Alternative splicing plays a role in modulating the subcellular localization and physiological function of this gene. Many alternatively spliced transcript variants, some of which are disease-associated mutations, have been described for this gene, but the full-length natures of only some of these variants has been described. A related pseudogene, which is also located on chromosome 17, has been identified. [provided by RefSeq, May 2020].'},\n",
       " {'id': 'HGNC:1107',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ZFP36L1',\n",
       "  'name': 'ZFP36 ring finger protein like 1',\n",
       "  'synonyms': 'BRF1|Berg36|ERF-1|ERF1|RNF162B|TIS11B|cMG1',\n",
       "  'description': 'This gene is a member of the TIS11 family of early response genes, which are induced by various agonists such as the phorbol ester TPA and the polypeptide mitogen EGF. This gene is well conserved across species and has a promoter that contains motifs seen in other early-response genes. The encoded protein contains a distinguishing putative zinc finger domain with a repeating cys-his motif. This putative nuclear transcription factor most likely functions in regulating the response to growth factors. Alternatively spliced transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Sep 2011].'},\n",
       " {'id': 'HGNC:1435',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CALB2',\n",
       "  'name': 'calbindin 2',\n",
       "  'synonyms': 'CAB29|CAL2|CR',\n",
       "  'description': 'This gene encodes an intracellular calcium-binding protein belonging to the troponin C superfamily. Members of this protein family have six EF-hand domains which bind calcium. This protein plays a role in diverse cellular functions, including message targeting and intracellular calcium buffering. It also functions as a modulator of neuronal excitability, and is a diagnostic marker for some human diseases, including Hirschsprung disease and some cancers. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Jun 2010].'},\n",
       " {'id': 'HGNC:1527',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CAV1',\n",
       "  'name': 'caveolin 1',\n",
       "  'synonyms': 'BSCL3|CGL3|LCCNS|MSTP085|PPH3|VIP21',\n",
       "  'description': 'The scaffolding protein encoded by this gene is the main component of the caveolae plasma membranes found in most cell types. The protein links integrin subunits to the tyrosine kinase FYN, an initiating step in coupling integrins to the Ras-ERK pathway and promoting cell cycle progression. The gene is a tumor suppressor gene candidate and a negative regulator of the Ras-p42/44 mitogen-activated kinase cascade. Caveolin 1 and caveolin 2 are located next to each other on chromosome 7 and express colocalizing proteins that form a stable hetero-oligomeric complex. Mutations in this gene have been associated with Berardinelli-Seip congenital lipodystrophy. Alternatively spliced transcripts encode alpha and beta isoforms of caveolin 1.[provided by RefSeq, Mar 2010].'},\n",
       " {'id': 'HGNC:1578',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CCNA2',\n",
       "  'name': 'cyclin A2',\n",
       "  'synonyms': 'CCN1|CCNA',\n",
       "  'description': 'The protein encoded by this gene belongs to the highly conserved cyclin family, whose members function as regulators of the cell cycle. This protein binds and activates cyclin-dependent kinase 2 and thus promotes transition through G1/S and G2/M. [provided by RefSeq, Aug 2016].'},\n",
       " {'id': 'HGNC:1744',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CDC6',\n",
       "  'name': 'cell division cycle 6',\n",
       "  'synonyms': 'CDC18L|HsCDC18|HsCDC6|MGORS5',\n",
       "  'description': 'The protein encoded by this gene is highly similar to Saccharomyces cerevisiae Cdc6, a protein essential for the initiation of DNA replication. This protein functions as a regulator at the early steps of DNA replication. It localizes in cell nucleus during cell cyle G1, but translocates to the cytoplasm at the start of S phase. The subcellular translocation of this protein during cell cyle is regulated through its phosphorylation by Cdks. Transcription of this protein was reported to be regulated in response to mitogenic signals through transcriptional control mechanism involving E2F proteins. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:1771',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CDK2',\n",
       "  'name': 'cyclin dependent kinase 2',\n",
       "  'synonyms': 'CDKN2|p33(CDK2)',\n",
       "  'description': 'This gene encodes a member of a family of serine/threonine protein kinases that participate in cell cycle regulation. The encoded protein is the catalytic subunit of the cyclin-dependent protein kinase complex, which regulates progression through the cell cycle. Activity of this protein is especially critical during the G1 to S phase transition. This protein associates with and regulated by other subunits of the complex including cyclin A or E, CDK inhibitor p21Cip1 (CDKN1A), and p27Kip1 (CDKN1B). Alternative splicing results in multiple transcript variants. [provided by RefSeq, Mar 2014].'},\n",
       " {'id': 'HGNC:1786',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CDKN1C',\n",
       "  'name': 'cyclin dependent kinase inhibitor 1C',\n",
       "  'synonyms': 'BWCR|BWS|KIP2|WBS|p57|p57Kip2',\n",
       "  'description': 'This gene is imprinted, with preferential expression of the maternal allele. The encoded protein is a tight-binding, strong inhibitor of several G1 cyclin/Cdk complexes and a negative regulator of cell proliferation. Mutations in this gene are implicated in sporadic cancers and Beckwith-Wiedemann syndorome, suggesting that this gene is a tumor suppressor candidate. Three transcript variants encoding two different isoforms have been found for this gene. [provided by RefSeq, Oct 2010].'},\n",
       " {'id': 'HGNC:1851',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CENPA',\n",
       "  'name': 'centromere protein A',\n",
       "  'synonyms': 'CENP-A|CenH3',\n",
       "  'description': 'Centromeres are the differentiated chromosomal domains that specify the mitotic behavior of chromosomes. This gene encodes a centromere protein which contains a histone H3 related histone fold domain that is required for targeting to the centromere. Centromere protein A is proposed to be a component of a modified nucleosome or nucleosome-like structure in which it replaces 1 or both copies of conventional histone H3 in the (H3-H4)2 tetrameric core of the nucleosome particle. The protein is a replication-independent histone that is a member of the histone H3 family. Alternative splicing results in multiple transcript variants encoding distinct isoforms. [provided by RefSeq, Nov 2015].'},\n",
       " {'id': 'HGNC:1856',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CENPE',\n",
       "  'name': 'centromere protein E',\n",
       "  'synonyms': 'CENP-E|KIF10|MCPH13|PPP1R61',\n",
       "  'description': 'Centrosome-associated protein E (CENPE) is a kinesin-like motor protein that accumulates in the G2 phase of the cell cycle. Unlike other centrosome-associated proteins, it is not present during interphase and first appears at the centromere region of chromosomes during prometaphase. This protein is required for stable spindle microtubule capture at kinetochores which is a necessary step in chromosome alignment during prometaphase. This protein also couples chromosome position to microtubule depolymerizing activity. Alternative splicing results in multiple transcript variants encoding distinct protein isoforms. [provided by RefSeq, Nov 2014].'},\n",
       " {'id': 'HGNC:1925',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CHEK1',\n",
       "  'name': 'checkpoint kinase 1',\n",
       "  'synonyms': 'CHK1',\n",
       "  'description': 'The protein encoded by this gene belongs to the Ser/Thr protein kinase family. It is required for checkpoint mediated cell cycle arrest in response to DNA damage or the presence of unreplicated DNA. This protein acts to integrate signals from ATM and ATR, two cell cycle proteins involved in DNA damage responses, that also associate with chromatin in meiotic prophase I. Phosphorylation of CDC25A protein phosphatase by this protein is required for cells to delay cell cycle progression in response to double-strand DNA breaks. Several alternatively spliced transcript variants have been found for this gene. [provided by RefSeq, Oct 2011].'},\n",
       " {'id': 'HGNC:19083',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CKS1B',\n",
       "  'name': 'CDC28 protein kinase regulatory subunit 1B',\n",
       "  'synonyms': 'CKS1|PNAS-16|PNAS-18|ckshs1',\n",
       "  'description': 'CKS1B protein binds to the catalytic subunit of the cyclin dependent kinases and is essential for their biological function. The CKS1B mRNA is found to be expressed in different patterns through the cell cycle in HeLa cells, which reflects a specialized role for the encoded protein. At least two transcript variants have been identified for this gene, and it appears that only one of them encodes a protein. [provided by RefSeq, Sep 2008].'},\n",
       " {'id': 'HGNC:2000',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CKS2',\n",
       "  'name': 'CDC28 protein kinase regulatory subunit 2',\n",
       "  'synonyms': 'CKSHS2',\n",
       "  'description': 'CKS2 protein binds to the catalytic subunit of the cyclin dependent kinases and is essential for their biological function. The CKS2 mRNA is found to be expressed in different patterns through the cell cycle in HeLa cells, which reflects specialized role for the encoded protein. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:2202',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'COL4A1',\n",
       "  'name': 'collagen type IV alpha 1 chain',\n",
       "  'synonyms': 'BSVD|BSVD1|COL4A1s|PADMAL|RATOR',\n",
       "  'description': 'This gene encodes a type IV collagen alpha protein. Type IV collagen proteins are integral components of basement membranes. This gene shares a bidirectional promoter with a paralogous gene on the opposite strand. The protein consists of an amino-terminal 7S domain, a triple-helix forming collagenous domain, and a carboxy-terminal non-collagenous domain. It functions as part of a heterotrimer and interacts with other extracellular matrix components such as perlecans, proteoglycans, and laminins. In addition, proteolytic cleavage of the non-collagenous carboxy-terminal domain results in a biologically active fragment known as arresten, which has anti-angiogenic and tumor suppressor properties. Mutations in this gene cause porencephaly, cerebrovascular disease, and renal and muscular defects. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Dec 2014].'},\n",
       " {'id': 'HGNC:2203',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'COL4A2',\n",
       "  'name': 'collagen type IV alpha 2 chain',\n",
       "  'synonyms': 'BSVD2|ICH|POREN2',\n",
       "  'description': 'This gene encodes one of the six subunits of type IV collagen, the major structural component of basement membranes. The C-terminal portion of the protein, known as canstatin, is an inhibitor of angiogenesis and tumor growth. Like the other members of the type IV collagen gene family, this gene is organized in a head-to-head conformation with another type IV collagen gene so that each gene pair shares a common promoter. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:2292',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'COX7C',\n",
       "  'name': 'cytochrome c oxidase subunit 7C',\n",
       "  'synonyms': nan,\n",
       "  'description': 'Cytochrome c oxidase (COX), the terminal component of the mitochondrial respiratory chain, catalyzes the electron transfer from reduced cytochrome c to oxygen. This component is a heteromeric complex consisting of 3 catalytic subunits encoded by mitochondrial genes and multiple structural subunits encoded by nuclear genes. The mitochondrially-encoded subunits function in electron transfer, and the nuclear-encoded subunits may function in the regulation and assembly of the complex. This nuclear gene encodes subunit VIIc, which shares 87% and 85% amino acid sequence identity with mouse and bovine COX VIIc, respectively, and is found in all tissues. A pseudogene COX7CP1 has been found on chromosome 13. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:2431',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CSE1L',\n",
       "  'name': 'chromosome segregation 1 like',\n",
       "  'synonyms': 'CAS|CSE1|XPO2',\n",
       "  'description': 'Proteins that carry a nuclear localization signal (NLS) are transported into the nucleus by the importin-alpha/beta heterodimer. Importin-alpha binds the NLS, while importin-beta mediates translocation through the nuclear pore complex. After translocation, RanGTP binds importin-beta and displaces importin-alpha. Importin-alpha must then be returned to the cytoplasm, leaving the NLS protein behind. The protein encoded by this gene binds strongly to NLS-free importin-alpha, and this binding is released in the cytoplasm by the combined action of RANBP1 and RANGAP1. In addition, the encoded protein may play a role both in apoptosis and in cell proliferation. Alternatively spliced transcript variants have been found for this gene. [provided by RefSeq, Jan 2012].'},\n",
       " {'id': 'HGNC:2432',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CSF1',\n",
       "  'name': 'colony stimulating factor 1',\n",
       "  'synonyms': 'CSF-1|MCSF',\n",
       "  'description': 'The protein encoded by this gene is a cytokine that controls the production, differentiation, and function of macrophages. The active form of the protein is found extracellularly as a disulfide-linked homodimer, and is thought to be produced by proteolytic cleavage of membrane-bound precursors. The encoded protein may be involved in development of the placenta. Alternate splicing results in multiple transcript variants. [provided by RefSeq, Sep 2011].'},\n",
       " {'id': 'HGNC:2483',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CSTF1',\n",
       "  'name': 'cleavage stimulation factor subunit 1',\n",
       "  'synonyms': 'CstF-50|CstFp50',\n",
       "  'description': \"This gene encodes one of three subunits which combine to form cleavage stimulation factor (CSTF). CSTF is involved in the polyadenylation and 3'end cleavage of pre-mRNAs. Similar to mammalian G protein beta subunits, this protein contains transducin-like repeats. Several transcript variants with different 5' UTR, but encoding the same protein, have been found for this gene. [provided by RefSeq, Jul 2008].\"},\n",
       " {'id': 'HGNC:2501',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CTH',\n",
       "  'name': 'cystathionine gamma-lyase',\n",
       "  'synonyms': nan,\n",
       "  'description': 'This gene encodes a cytoplasmic enzyme in the trans-sulfuration pathway that converts cystathione derived from methionine into cysteine. Glutathione synthesis in the liver is dependent upon the availability of cysteine. Mutations in this gene cause cystathioninuria. Alternative splicing of this gene results in three transcript variants encoding different isoforms. [provided by RefSeq, Jun 2010].'},\n",
       " {'id': 'HGNC:2597',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CYP1B1',\n",
       "  'name': 'cytochrome P450 family 1 subfamily B member 1',\n",
       "  'synonyms': 'ASGD6|CP1B|CYPIB1|GLC3A|P4501B1',\n",
       "  'description': 'This gene encodes a member of the cytochrome P450 superfamily of enzymes. The cytochrome P450 proteins are monooxygenases which catalyze many reactions involved in drug metabolism and synthesis of cholesterol, steroids and other lipids. The enzyme encoded by this gene localizes to the endoplasmic reticulum and metabolizes procarcinogens such as polycyclic aromatic hydrocarbons and 17beta-estradiol. Mutations in this gene have been associated with primary congenital glaucoma; therefore it is thought that the enzyme also metabolizes a signaling molecule involved in eye development, possibly a steroid. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:4095',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GADD45A',\n",
       "  'name': 'growth arrest and DNA damage inducible alpha',\n",
       "  'synonyms': 'DDIT1|GADD45',\n",
       "  'description': 'This gene is a member of a group of genes whose transcript levels are increased following stressful growth arrest conditions and treatment with DNA-damaging agents. The protein encoded by this gene responds to environmental stresses by mediating activation of the p38/JNK pathway via MTK1/MEKK4 kinase. The DNA damage-induced transcription of this gene is mediated by both p53-dependent and -independent mechanisms. Alternatively spliced transcript variants encoding distinct isoforms have been found for this gene.[provided by RefSeq, Dec 2010].'},\n",
       " {'id': 'HGNC:2861',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DHFR',\n",
       "  'name': 'dihydrofolate reductase',\n",
       "  'synonyms': 'DHFRP1|DYR',\n",
       "  'description': 'Dihydrofolate reductase converts dihydrofolate into tetrahydrofolate, a methyl group shuttle required for the de novo synthesis of purines, thymidylic acid, and certain amino acids. While the functional dihydrofolate reductase gene has been mapped to chromosome 5, multiple intronless processed pseudogenes or dihydrofolate reductase-like genes have been identified on separate chromosomes. Dihydrofolate reductase deficiency has been linked to megaloblastic anemia. Several transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Mar 2014].'},\n",
       " {'id': 'HGNC:2934',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DMRT1',\n",
       "  'name': 'doublesex and mab-3 related transcription factor 1',\n",
       "  'synonyms': 'CT154|DMT1',\n",
       "  'description': 'This gene is found in a cluster with two other members of the gene family, having in common a zinc finger-like DNA-binding motif (DM domain). The DM domain is an ancient, conserved component of the vertebrate sex-determining pathway that is also a key regulator of male development in flies and nematodes. This gene exhibits a gonad-specific and sexually dimorphic expression pattern. Defective testicular development and XY feminization occur when this gene is hemizygous. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:3033',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ATN1',\n",
       "  'name': 'atrophin 1',\n",
       "  'synonyms': 'B37|CHEDDA|D12S755E|DRPLA|HRS|NOD',\n",
       "  'description': 'Dentatorubral pallidoluysian atrophy (DRPLA) is a rare neurodegenerative disorder characterized by cerebellar ataxia, myoclonic epilepsy, choreoathetosis, and dementia. The disorder is related to the expansion from 7-35 copies to 49-93 copies of a trinucleotide repeat (CAG/CAA) within this gene. The encoded protein includes a serine repeat and a region of alternating acidic and basic amino acids, as well as the variable glutamine repeat. Alternative splicing results in two transcripts variants that encode the same protein. [provided by RefSeq, Jul 2016].'},\n",
       " {'id': 'HGNC:3241',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'EGR4',\n",
       "  'name': 'early growth response 4',\n",
       "  'synonyms': 'AT133|NGFI-C|NGFIC|PAT133',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:3541',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'F3',\n",
       "  'name': 'coagulation factor III, tissue factor',\n",
       "  'synonyms': 'CD142|TF|TFA',\n",
       "  'description': 'This gene encodes coagulation factor III which is a cell surface glycoprotein. This factor enables cells to initiate the blood coagulation cascades, and it functions as the high-affinity receptor for the coagulation factor VII. The resulting complex provides a catalytic event that is responsible for initiation of the coagulation protease cascades by specific limited proteolysis. Unlike the other cofactors of these protease cascades, which circulate as nonfunctional precursors, this factor is a potent initiator that is fully functional when expressed on cell surfaces, for example, on monocytes. There are 3 distinct domains of this factor: extracellular, transmembrane, and cytoplasmic. Platelets and monocytes have been shown to express this coagulation factor under procoagulatory and proinflammatory stimuli, and a major role in HIV-associated coagulopathy has been described. Platelet-dependent monocyte expression of coagulation factor III has been described to be associated with Coronavirus Disease 2019 (COVID-19) severity and mortality. This protein is the only one in the coagulation pathway for which a congenital deficiency has not been described. Alternate splicing results in multiple transcript variants.[provided by RefSeq, Aug 2020].'},\n",
       " {'id': 'HGNC:3584',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'FANCC',\n",
       "  'name': 'FA complementation group C',\n",
       "  'synonyms': 'FA3|FAC|FACC',\n",
       "  'description': 'The Fanconi anemia complementation group (FANC) currently includes FANCA, FANCB, FANCC, FANCD1 (also called BRCA2), FANCD2, FANCE, FANCF, FANCG, FANCI, FANCJ (also called BRIP1), FANCL, FANCM and FANCN (also called PALB2). The previously defined group FANCH is the same as FANCA. Fanconi anemia is a genetically heterogeneous recessive disorder characterized by cytogenetic instability, hypersensitivity to DNA crosslinking agents, increased chromosomal breakage, and defective DNA repair. The members of the Fanconi anemia complementation group do not share sequence similarity; they are related by their assembly into a common nuclear protein complex. This gene encodes the protein for complementation group C. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:3629',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'FDFT1',\n",
       "  'name': 'farnesyl-diphosphate farnesyltransferase 1',\n",
       "  'synonyms': 'DGPT|ERG9|SQS|SQSD|SS',\n",
       "  'description': 'This gene encodes a membrane-associated enzyme located at a branch point in the mevalonate pathway. The encoded protein is the first specific enzyme in cholesterol biosynthesis, catalyzing the dimerization of two molecules of farnesyl diphosphate in a two-step reaction to form squalene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:3685',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'FGF7',\n",
       "  'name': 'fibroblast growth factor 7',\n",
       "  'synonyms': 'HBGF-7|KGF',\n",
       "  'description': 'The protein encoded by this gene is a member of the fibroblast growth factor (FGF) family. FGF family members possess broad mitogenic and cell survival activities, and are involved in a variety of biological processes, including embryonic development, cell growth, morphogenesis, tissue repair, tumor growth and invasion. This protein is a potent epithelial cell-specific growth factor, whose mitogenic activity is predominantly exhibited in keratinocytes but not in fibroblasts and endothelial cells. Studies of mouse and rat homologs of this gene implicated roles in morphogenesis of epithelium, reepithelialization of wounds, hair development and early lung organogenesis. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:4057',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'G6PD',\n",
       "  'name': 'glucose-6-phosphate dehydrogenase',\n",
       "  'synonyms': 'G6PD1',\n",
       "  'description': 'This gene encodes glucose-6-phosphate dehydrogenase. This protein is a cytosolic enzyme encoded by a housekeeping X-linked gene whose main function is to produce NADPH, a key electron donor in the defense against oxidizing agents and in reductive biosynthetic reactions. G6PD is remarkable for its genetic diversity. Many variants of G6PD, mostly produced from missense mutations, have been described with wide ranging levels of enzyme activity and associated clinical symptoms. G6PD deficiency may cause neonatal jaundice, acute hemolysis, or severe chronic non-spherocytic hemolytic anemia. Two transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:4071',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GABPA',\n",
       "  'name': 'GA binding protein transcription factor subunit alpha',\n",
       "  'synonyms': 'E4TF1-60|E4TF1A|NFT2|NRF2|NRF2A|RCH04A07',\n",
       "  'description': 'This gene encodes one of three GA-binding protein transcription factor subunits which functions as a DNA-binding subunit. Since this subunit shares identity with a subunit encoding the nuclear respiratory factor 2 gene, it is likely involved in activation of cytochrome oxidase expression and nuclear control of mitochondrial function. This subunit also shares identity with a subunit constituting the transcription factor E4TF1, responsible for expression of the adenovirus E4 gene. Because of its chromosomal localization and ability to form heterodimers with other polypeptides, this gene may play a role in the Down Syndrome phenotype. Two transcript variants encoding the same protein have been found for this gene. [provided by RefSeq, Oct 2010].'},\n",
       " {'id': 'HGNC:4076',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GABRA2',\n",
       "  'name': 'gamma-aminobutyric acid type A receptor subunit alpha2',\n",
       "  'synonyms': 'DEE78|EIEE78',\n",
       "  'description': 'GABA is the major inhibitory neurotransmitter in the mammalian brain where it acts at GABA-A receptors, which are ligand-gated chloride channels. Chloride conductance of these channels can be modulated by agents such as benzodiazepines that bind to the GABA-A receptor. At least 16 distinct subunits of GABA-A receptors have been identified. Alternatively spliced transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Nov 2013].'},\n",
       " {'id': 'HGNC:4284',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GJB2',\n",
       "  'name': 'gap junction protein beta 2',\n",
       "  'synonyms': 'BAPS|CX26|DFNA3|DFNA3A|DFNB1|DFNB1A|HID|KID|NSRD1|PPK',\n",
       "  'description': 'This gene encodes a member of the gap junction protein family. The gap junctions were first characterized by electron microscopy as regionally specialized structures on plasma membranes of contacting adherent cells. These structures were shown to consist of cell-to-cell channels that facilitate the transfer of ions and small molecules between cells. The gap junction proteins, also known as connexins, purified from fractions of enriched gap junctions from different tissues differ. According to sequence similarities at the nucleotide and amino acid levels, the gap junction proteins are divided into two categories, alpha and beta. Mutations in this gene are responsible for as much as 50% of pre-lingual, recessive deafness. [provided by RefSeq, Oct 2008].'},\n",
       " {'id': 'HGNC:4335',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GLUD1',\n",
       "  'name': 'glutamate dehydrogenase 1',\n",
       "  'synonyms': 'GDH|GDH1|GLUD',\n",
       "  'description': 'This gene encodes glutamate dehydrogenase, which is a mitochondrial matrix enzyme that catalyzes the oxidative deamination of glutamate to alpha-ketoglutarate and ammonia. This enzyme has an important role in regulating amino acid-induced insulin secretion. It is allosterically activated by ADP and inhibited by GTP and ATP. Activating mutations in this gene are a common cause of congenital hyperinsulinism. Alternative splicing of this gene results in multiple transcript variants. The related glutamate dehydrogenase 2 gene on the human X-chromosome originated from this gene via retrotransposition and encodes a soluble form of glutamate dehydrogenase. Related pseudogenes have been identified on chromosomes 10, 18 and X. [provided by RefSeq, Jan 2016].'},\n",
       " {'id': 'HGNC:4704',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GYPC',\n",
       "  'name': 'glycophorin C (Gerbich blood group)',\n",
       "  'synonyms': \"CD236|CD236R|GE|GPC|GPD|GYPD|PAS-2|PAS-2'\",\n",
       "  'description': 'Glycophorin C (GYPC) is an integral membrane glycoprotein. It is a minor species carried by human erythrocytes, but plays an important role in regulating the mechanical stability of red cells. A number of glycophorin C mutations have been described. The Gerbich and Yus phenotypes are due to deletion of exon 3 and 2, respectively. The Webb and Duch antigens, also known as glycophorin D, result from single point mutations of the glycophorin C gene. The glycophorin C protein has very little homology with glycophorins A and B. Alternate splicing results in multiple transcript variants. [provided by RefSeq, Feb 2012].'},\n",
       " {'id': 'HGNC:4739',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'H2AX',\n",
       "  'name': 'H2A.X variant histone',\n",
       "  'synonyms': 'H2A.X|H2A/X|H2AFX',\n",
       "  'description': 'Histones are basic nuclear proteins that are responsible for the nucleosome structure of the chromosomal fiber in eukaryotes. Two molecules of each of the four core histones (H2A, H2B, H3, and H4) form an octamer, around which approximately 146 bp of DNA is wrapped in repeating units, called nucleosomes. The linker histone, H1, interacts with linker DNA between nucleosomes and functions in the compaction of chromatin into higher order structures. This gene encodes a replication-independent histone that is a member of the histone H2A family, and generates two transcripts through the use of the conserved stem-loop termination motif, and the polyA addition motif. [provided by RefSeq, Oct 2015].'},\n",
       " {'id': 'HGNC:4747',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'H2BC5',\n",
       "  'name': 'H2B clustered histone 5',\n",
       "  'synonyms': 'H2B.1B|H2B/a|H2B/b|H2B/g|H2B/h|H2B/k|H2B/l|H2BFA|H2BFB|H2BFG|H2BFH|H2BFK|H2BFL|HIRIP2|HIST1H2BC|HIST1H2BD|HIST1H2BE|HIST1H2BF|HIST1H2BG|HIST1H2BI|dJ221C16.6',\n",
       "  'description': 'Histones are basic nuclear proteins that are responsible for the nucleosome structure of the chromosomal fiber in eukaryotes. Nucleosomes consist of approximately 146 bp of DNA wrapped around a histone octamer composed of pairs of each of the four core histones (H2A, H2B, H3, and H4). The chromatin fiber is further compacted through the interaction of a linker histone, H1, with the DNA between the nucleosomes to form higher order chromatin structures. This gene is intronless and encodes a replication-dependent histone that is a member of the histone H2B family. Two transcripts that encode the same protein have been identified for this gene, which is found in the large histone gene cluster on chromosome 6p22-p21.3. [provided by RefSeq, Aug 2015].'},\n",
       " {'id': 'HGNC:5004',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HMGB3',\n",
       "  'name': 'high mobility group box 3',\n",
       "  'synonyms': 'HMG-2a|HMG-4|HMG2A|HMG4',\n",
       "  'description': 'This gene encodes a member of a family of proteins containing one or more high mobility group DNA-binding motifs. The encoded protein plays an important role in maintaining stem cell populations, and may be aberrantly expressed in tumor cells. A mutation in this gene was associated with microphthalmia, syndromic 13. There are numerous pseudogenes of this gene on multiple chromosomes. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Jul 2014].'},\n",
       " {'id': 'HGNC:4986',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HMGN2',\n",
       "  'name': 'high mobility group nucleosomal binding domain 2',\n",
       "  'synonyms': 'HMG17',\n",
       "  'description': 'The protein encoded by this gene binds nucleosomal DNA and is associated with transcriptionally active chromatin. Along with a similar protein, HMGN1, the encoded protein may help maintain an open chromatin configuration around transcribable genes. The protein has also been found to have antimicrobial activity against bacteria, viruses and fungi. [provided by RefSeq, Oct 2014].'},\n",
       " {'id': 'HGNC:5010',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HMGA1',\n",
       "  'name': 'high mobility group AT-hook 1',\n",
       "  'synonyms': 'HMG-R|HMGA1A|HMGIY',\n",
       "  'description': 'This gene encodes a chromatin-associated protein involved in the regulation of gene transcription, integration of retroviruses into chromosomes, and the metastatic progression of cancer cells. The encoded protein preferentially binds to the minor groove of AT-rich regions in double-stranded DNA. Multiple transcript variants encoding different isoforms have been found for this gene. Pseudogenes of this gene have been identified on multiple chromosomes. [provided by RefSeq, Jan 2016].'},\n",
       " {'id': 'HGNC:5012',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HMMR',\n",
       "  'name': 'hyaluronan mediated motility receptor',\n",
       "  'synonyms': 'CD168|IHABP|RHAMM',\n",
       "  'description': 'The protein encoded by this gene is involved in cell motility. It is expressed in breast tissue and together with other proteins, it forms a complex with BRCA1 and BRCA2, thus is potentially associated with higher risk of breast cancer. Alternatively spliced transcript variants encoding different isoforms have been noted for this gene. [provided by RefSeq, Dec 2008].'},\n",
       " {'id': 'HGNC:5024',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HNF4A',\n",
       "  'name': 'hepatocyte nuclear factor 4 alpha',\n",
       "  'synonyms': 'FRTS4|HNF4|HNF4a7|HNF4a8|HNF4a9|HNF4alpha|MODY|MODY1|NR2A1|NR2A21|TCF|TCF14',\n",
       "  'description': 'The protein encoded by this gene is a nuclear transcription factor which binds DNA as a homodimer. The encoded protein controls the expression of several genes, including hepatocyte nuclear factor 1 alpha, a transcription factor which regulates the expression of several hepatic genes. This gene may play a role in development of the liver, kidney, and intestines. Mutations in this gene have been associated with monogenic autosomal dominant non-insulin-dependent diabetes mellitus type I. Alternative splicing of this gene results in multiple transcript variants encoding several different isoforms. [provided by RefSeq, Apr 2012].'},\n",
       " {'id': 'HGNC:5036',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HNRNPD',\n",
       "  'name': 'heterogeneous nuclear ribonucleoprotein D',\n",
       "  'synonyms': 'AUF1|AUF1A|HNRPD|P37|hnRNPD0',\n",
       "  'description': 'This gene belongs to the subfamily of ubiquitously expressed heterogeneous nuclear ribonucleoproteins (hnRNPs). The hnRNPs are nucleic acid binding proteins and they complex with heterogeneous nuclear RNA (hnRNA). These proteins are associated with pre-mRNAs in the nucleus and appear to influence pre-mRNA processing and other aspects of mRNA metabolism and transport. While all of the hnRNPs are present in the nucleus, some seem to shuttle between the nucleus and the cytoplasm. The hnRNP proteins have distinct nucleic acid binding properties. The protein encoded by this gene has two repeats of quasi-RRM domains that bind to RNAs. It localizes to both the nucleus and the cytoplasm. This protein is implicated in the regulation of mRNA stability. Alternative splicing of this gene results in four transcript variants. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:5154',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HPGD',\n",
       "  'name': '15-hydroxyprostaglandin dehydrogenase',\n",
       "  'synonyms': '15-PGDH|PGDH|PGDH1|PHOAR1|SDR36C1',\n",
       "  'description': 'This gene encodes a member of the short-chain nonmetalloenzyme alcohol dehydrogenase protein family. The encoded enzyme is responsible for the metabolism of prostaglandins, which function in a variety of physiologic and cellular processes such as inflammation. Mutations in this gene result in primary autosomal recessive hypertrophic osteoarthropathy and cranioosteoarthropathy. Multiple transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Mar 2009].'},\n",
       " {'id': 'HGNC:5433',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'IFNAR2',\n",
       "  'name': 'interferon alpha and beta receptor subunit 2',\n",
       "  'synonyms': 'IFN-R|IFN-alpha-REC|IFNABR|IFNARB|IMD45',\n",
       "  'description': 'The protein encoded by this gene is a type I membrane protein that forms one of the two chains of a receptor for interferons alpha and beta. Binding and activation of the receptor stimulates Janus protein kinases, which in turn phosphorylate several proteins, including STAT1 and STAT2. The protein belongs to the type II cytokine receptor family. Mutations in this gene are associated with Immunodeficiency 45. [provided by RefSeq, Jul 2020].'},\n",
       " {'id': 'HGNC:6008',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'IL2RA',\n",
       "  'name': 'interleukin 2 receptor subunit alpha',\n",
       "  'synonyms': 'CD25|IDDM10|IL2R|IMD41|TCGFR|p55',\n",
       "  'description': 'The interleukin 2 (IL2) receptor alpha (IL2RA) and beta (IL2RB) chains, together with the common gamma chain (IL2RG), constitute the high-affinity IL2 receptor. Homodimeric alpha chains (IL2RA) result in low-affinity receptor, while homodimeric beta (IL2RB) chains produce a medium-affinity receptor. Normally an integral-membrane protein, soluble IL2RA has been isolated and determined to result from extracellular proteolyisis. Alternately-spliced IL2RA mRNAs have been isolated, but the significance of each is presently unknown. Mutations in this gene are associated with interleukin 2 receptor alpha deficiency. Patients with severe Coronavirus Disease 2019 (COVID-19), the disease caused by the novel severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2), have significantly elevated levels of IL2R in their plasma. Similarly, serum IL-2R levels are found to be elevated in patients with different types of carcinomas. Certain IL2RA and IL2RB gene polymorphisms have been associated with lung cancer risk. [provided by RefSeq, Jul 2020].'},\n",
       " {'id': 'HGNC:6083',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'INSIG1',\n",
       "  'name': 'insulin induced gene 1',\n",
       "  'synonyms': 'CL6',\n",
       "  'description': 'This gene encodes an endoplasmic reticulum membrane protein that regulates cholesterol metabolism, lipogenesis, and glucose homeostasis. The encoded protein has six transmembrane helices which contain an effector protein binding site. It binds the sterol-sensing domains of sterol regulatory element-binding protein (SREBP) cleavage-activating protein (SCAP) and 3-hydroxy-3-methylglutaryl-coenzyme A reductase (HMG-CoA reductase), and is essential for the sterol-mediated trafficking of these two proteins. It promotes the endoplasmic reticulum retention of SCAP and the ubiquitin-mediated degradation of HMG-CoA reductase. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Oct 2016].'},\n",
       " {'id': 'HGNC:6204',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'JUN',\n",
       "  'name': 'Jun proto-oncogene, AP-1 transcription factor subunit',\n",
       "  'synonyms': 'AP-1|AP1|c-Jun|cJUN|p39',\n",
       "  'description': 'This gene is the putative transforming gene of avian sarcoma virus 17. It encodes a protein which is highly similar to the viral protein, and which interacts directly with specific target DNA sequences to regulate gene expression. This gene is intronless and is mapped to 1p32-p31, a chromosomal region involved in both translocations and deletions in human malignancies. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:6391',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'KIF22',\n",
       "  'name': 'kinesin family member 22',\n",
       "  'synonyms': 'A-328A3.2|KID|KNSL4|OBP|OBP-1|OBP-2|SEMDJL2',\n",
       "  'description': 'The protein encoded by this gene is a member of the kinesin-like protein family. The family members are microtubule-dependent molecular motors that transport organelles within cells and move chromosomes during cell division. The C-terminal half of this protein has been shown to bind DNA. Studies with the Xenopus homolog suggests its essential role in metaphase chromosome alignment and maintenance. Alternatively spliced transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Jan 2012].'},\n",
       " {'id': 'HGNC:6436',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'KRT19',\n",
       "  'name': 'keratin 19',\n",
       "  'synonyms': 'CK19|K19|K1CS',\n",
       "  'description': 'The protein encoded by this gene is a member of the keratin family. The keratins are intermediate filament proteins responsible for the structural integrity of epithelial cells and are subdivided into cytokeratins and hair keratins. The type I cytokeratins consist of acidic proteins which are arranged in pairs of heterotypic keratin chains. Unlike its related family members, this smallest known acidic cytokeratin is not paired with a basic cytokeratin in epithelial cells. It is specifically expressed in the periderm, the transiently superficial layer that envelopes the developing epidermis. The type I cytokeratins are clustered in a region of chromosome 17q12-q21. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:6510',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'STMN1',\n",
       "  'name': 'stathmin 1',\n",
       "  'synonyms': 'C1orf215|LAP18|Lag|OP18|PP17|PP19|PR22|SMN',\n",
       "  'description': 'This gene belongs to the stathmin family of genes. It encodes a ubiquitous cytosolic phosphoprotein proposed to function as an intracellular relay integrating regulatory signals of the cellular environment. The encoded protein is involved in the regulation of the microtubule filament system by destabilizing microtubules. It prevents assembly and promotes disassembly of microtubules. Multiple transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Feb 2009].'},\n",
       " {'id': 'HGNC:6763',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MAD2L1',\n",
       "  'name': 'mitotic arrest deficient 2 like 1',\n",
       "  'synonyms': 'HSMAD2|MAD2',\n",
       "  'description': 'MAD2L1 is a component of the mitotic spindle assembly checkpoint that prevents the onset of anaphase until all chromosomes are properly aligned at the metaphase plate.  MAD2L1 is related to the MAD2L2 gene located on chromosome 1.  A MAD2 pseudogene has been mapped to chromosome 14. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:6925',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MBP',\n",
       "  'name': 'myelin basic protein',\n",
       "  'synonyms': nan,\n",
       "  'description': \"The protein encoded by the classic MBP gene is a major constituent of the myelin sheath of oligodendrocytes and Schwann cells in the nervous system. However, MBP-related transcripts are also present in the bone marrow and the immune system. These mRNAs arise from the long MBP gene (otherwise called 'Golli-MBP') that contains 3 additional exons located upstream of the classic MBP exons. Alternative splicing from the Golli and the MBP transcription start sites gives rise to 2 sets of MBP-related transcripts and gene products. The Golli mRNAs contain 3 exons unique to Golli-MBP, spliced in-frame to 1 or more MBP exons. They encode hybrid proteins that have N-terminal Golli aa sequence linked to MBP aa sequence. The second family of transcripts contain only MBP exons and produce the well characterized myelin basic proteins. This complex gene structure is conserved among species suggesting that the MBP transcription unit is an integral part of the Golli transcription unit and that this arrangement is important for the function and/or regulation of these genes. [provided by RefSeq, Jul 2008].\"},\n",
       " {'id': 'HGNC:6949',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MCM6',\n",
       "  'name': 'minichromosome maintenance complex component 6',\n",
       "  'synonyms': 'MCG40308|Mis5|P105MCM',\n",
       "  'description': 'The protein encoded by this gene is one of the highly conserved mini-chromosome maintenance proteins (MCM) that are essential for the initiation of eukaryotic genome replication. The hexameric protein complex formed by the MCM proteins is a key component of the pre-replication complex (pre_RC) and may be involved in the formation of replication forks and in the recruitment of other DNA replication related proteins. The MCM complex consisting of this protein and MCM2, 4 and 7 proteins possesses DNA helicase activity, and may act as a DNA unwinding enzyme. The phosphorylation of the complex by CDC2 kinase reduces the helicase activity, suggesting a role in the regulation of DNA replication. Single nucleotide polymorphisms in the intron regions of this gene are associated with differential transcriptional activation of the promoter of the neighboring lactase gene and, thereby, influence lactose intolerance in early adulthood. [provided by RefSeq, May 2012].'},\n",
       " {'id': 'HGNC:7107',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MKI67',\n",
       "  'name': 'marker of proliferation Ki-67',\n",
       "  'synonyms': 'KIA|MIB-|MIB-1|PPP1R105',\n",
       "  'description': 'This gene encodes a nuclear protein that is associated with and may be necessary for cellular proliferation. Alternatively spliced transcript variants have been described. A related pseudogene exists on chromosome X. [provided by RefSeq, Mar 2009].'},\n",
       " {'id': 'HGNC:7132',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'KMT2A',\n",
       "  'name': 'lysine methyltransferase 2A',\n",
       "  'synonyms': 'ALL-1|CXXC7|HRX|HTRX1|MLL|MLL1|MLL1A|TRX1|WDSTS',\n",
       "  'description': 'This gene encodes a transcriptional coactivator that plays an essential role in regulating gene expression during early development and hematopoiesis. The encoded protein contains multiple conserved functional domains. One of these domains, the SET domain, is responsible for its histone H3 lysine 4 (H3K4) methyltransferase activity which mediates chromatin modifications associated with epigenetic transcriptional activation. This protein is processed by the enzyme Taspase 1 into two fragments, MLL-C and MLL-N. These fragments reassociate and further assemble into different multiprotein complexes that regulate the transcription of specific target genes, including many of the HOX genes. Multiple chromosomal translocations involving this gene are the cause of certain acute lymphoid leukemias and acute myeloid leukemias. Alternate splicing results in multiple transcript variants.[provided by RefSeq, Oct 2010].'},\n",
       " {'id': 'HGNC:7408',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MT3',\n",
       "  'name': 'metallothionein 3',\n",
       "  'synonyms': 'GIF|GIFB|GRIF|ZnMT3',\n",
       "  'description': \"This gene is a member of the metallothionein family of genes. Proteins encoded by this gene family are low in molecular weight, are cysteine-rich, lack aromatic residues, and bind divalent heavy metal ions. This gene family member displays tissue-specific expression, and contains a threonine insert near its N-terminus and a glutamate-rich hexapeptide insert near its C-terminus relative to the proteins encoded by other gene family members. It plays an important role in zinc and copper homeostasis, and is induced under hypoxic conditions. The encoded protein is a growth inhibitory factor, and reduced levels of the protein are observed in the brains of individuals with some metal-linked neurodegenerative disorders such as Alzheimer's disease. [provided by RefSeq, Sep 2017].\"},\n",
       " {'id': 'HGNC:7553',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MYC',\n",
       "  'name': 'MYC proto-oncogene, bHLH transcription factor',\n",
       "  'synonyms': 'MRTL|MYCC|bHLHe39|c-Myc',\n",
       "  'description': 'This gene is a proto-oncogene and encodes a nuclear phosphoprotein that plays a role in cell cycle progression, apoptosis and cellular transformation. The encoded protein forms a heterodimer with the related transcription factor MAX. This complex binds to the E box DNA consensus sequence and regulates the transcription of specific target genes. Amplification of this gene is frequently observed in numerous human cancers. Translocations involving this gene are associated with Burkitt lymphoma and multiple myeloma in human patients. There is evidence to show that translation initiates both from an upstream, in-frame non-AUG (CUG) and a downstream AUG start site, resulting in the production of two isoforms with distinct N-termini. [provided by RefSeq, Aug 2017].'},\n",
       " {'id': 'HGNC:7555',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MYCL',\n",
       "  'name': 'MYCL proto-oncogene, bHLH transcription factor',\n",
       "  'synonyms': 'L-Myc|LMYC|MYCL1|bHLHe38',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:7559',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MYCN',\n",
       "  'name': 'MYCN proto-oncogene, bHLH transcription factor',\n",
       "  'synonyms': 'MODED|N-myc|NMYC|ODED|bHLHe37',\n",
       "  'description': 'This gene is a member of the MYC family and encodes a protein with a basic helix-loop-helix (bHLH) domain. This protein is located in the nucleus and must dimerize with another bHLH protein in order to bind DNA. Amplification of this gene is associated with a variety of tumors, most notably neuroblastomas. Multiple alternatively spliced transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Jun 2014].'},\n",
       " {'id': 'HGNC:7622',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MYT1',\n",
       "  'name': 'myelin transcription factor 1',\n",
       "  'synonyms': 'C20orf36|MTF1|MYTI|NZF2|PLPB1|ZC2H2C1|ZC2HC4A',\n",
       "  'description': 'The protein encoded by this gene is a member of a family of neural specific, zinc finger-containing DNA-binding proteins. The protein binds to the promoter regions of proteolipid proteins of the central nervous system and plays a role in the developing nervous system. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:7640',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NAP1L4',\n",
       "  'name': 'nucleosome assembly protein 1 like 4',\n",
       "  'synonyms': 'NAP1L4b|NAP2|NAP2L|hNAP2',\n",
       "  'description': 'This gene encodes a member of the nucleosome assembly protein (NAP) family which can interact with both core and linker histones. It can shuttle between the cytoplasm and nucleus, suggesting a role as a histone chaperone. This gene is one of several located near the imprinted gene domain of 11p15.5, an important tumor-suppressor gene region. Alterations in this region have been associated with the Beckwith-Wiedemann syndrome, Wilms tumor, rhabdomyosarcoma, adrenocortical carcinoma, and lung, ovarian, and breast cancer. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:3029',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DRG1',\n",
       "  'name': 'developmentally regulated GTP binding protein 1',\n",
       "  'synonyms': 'NEDD3',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:7758',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NEU1',\n",
       "  'name': 'neuraminidase 1',\n",
       "  'synonyms': 'NANH|NEU|SIAL1',\n",
       "  'description': \"The protein encoded by this gene is a lysosomal enzyme that cleaves terminal sialic acid residues from substrates such as glycoproteins and glycolipids. In the lysosome, this enzyme is part of a heterotrimeric complex together with beta-galactosidase and cathepsin A (the latter is also referred to as 'protective protein'). Mutations in this gene can lead to sialidosis, a lysosomal storage disease that can be type 1 (cherry red spot-myoclonus syndrome or normosomatic type), which is late-onset, or type 2 (the dysmorphic type), which occurs at an earlier age with increased severity. [provided by RefSeq, Jul 2008].\"},\n",
       " {'id': 'HGNC:7781',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NFE2L1',\n",
       "  'name': 'nuclear factor, erythroid 2 like 1',\n",
       "  'synonyms': 'LCR-F1|NRF1|TCF11',\n",
       "  'description': \"This gene encodes a protein that is involved in globin gene expression in erythrocytes. Confusion has occurred in bibliographic databases due to the shared symbol of NRF1 for this gene, NFE2L1, and for 'nuclear respiratory factor 1' which has an official symbol of NRF1. [provided by RefSeq, Jul 2008].\"},\n",
       " {'id': 'HGNC:7794',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NFKAPPAB',\n",
       "  'name': 'nuclear factor kappa B subunit 1',\n",
       "  'synonyms': 'CVID12|EBP-1|KBF1|NF-kB|NF-kB1|NF-kappa-B1|NF-kappaB|NF-kappabeta|NFKB-p105|NFKB-p50|NFkappaB',\n",
       "  'description': 'This gene encodes a 105 kD protein which can undergo cotranslational processing by the 26S proteasome to produce a 50 kD protein. The 105 kD protein is a Rel protein-specific transcription inhibitor and the 50 kD protein is a DNA binding subunit of the NF-kappa-B (NFKB) protein complex. NFKB is a transcription regulator that is activated by various intra- and extra-cellular stimuli such as cytokines, oxidant-free radicals, ultraviolet irradiation, and bacterial or viral products. Activated NFKB translocates into the nucleus and stimulates the expression of genes involved in a wide variety of biological functions. Inappropriate activation of NFKB has been associated with a number of inflammatory diseases while persistent inhibition of NFKB leads to inappropriate immune cell development or delayed cell growth. NFKB is a critical regulator of the immediate-early response to viral infection. Alternative splicing results in multiple transcript variants encoding different isoforms, at least one of which is proteolytically processed. [provided by RefSeq, Aug 2020].'},\n",
       " {'id': 'HGNC:7795',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NFKB2',\n",
       "  'name': 'nuclear factor kappa B subunit 2',\n",
       "  'synonyms': 'CVID10|H2TF1|LYT-10|LYT10|NF-kB2|p100|p49/p100|p52',\n",
       "  'description': 'This gene encodes a subunit of the transcription factor complex nuclear factor-kappa-B (NFkB). The NFkB complex is expressed in numerous cell types and functions as a central activator of genes involved in inflammation and immune function. The protein encoded by this gene can function as both a transcriptional activator or repressor depending on its dimerization partner. The p100 full-length protein is co-translationally processed into a p52 active form. Chromosomal rearrangements and translocations of this locus have been observed in B cell lymphomas, some of which may result in the formation of fusion proteins. There is a pseudogene for this gene on chromosome 18. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Dec 2013].'},\n",
       " {'id': 'HGNC:7804',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NFYA',\n",
       "  'name': 'nuclear transcription factor Y subunit alpha',\n",
       "  'synonyms': 'CBF-A|CBF-B|HAP2|NF-YA',\n",
       "  'description': 'The protein encoded by this gene is one subunit of a trimeric complex, forming a highly conserved transcription factor that binds to CCAAT motifs in the promoter regions in a variety of genes. Subunit A associates with a tight dimer composed of the B and C subunits, resulting in a trimer that binds to DNA with high specificity and affinity. The sequence specific interactions of the complex are made by the A subunit, suggesting a role as the regulatory subunit. In addition, there is evidence of post-transcriptional regulation in this gene product, either by protein degradation or control of translation. Further regulation is represented by alternative splicing in the glutamine-rich activation domain, with clear tissue-specific preferences for the two isoforms. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:7833',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NKTR',\n",
       "  'name': 'natural killer cell triggering receptor',\n",
       "  'synonyms': 'p104',\n",
       "  'description': 'This gene encodes a membrane-anchored protein with a hydrophobic amino terminal domain and a cyclophilin-like PPIase domain. It is present on the surface of natural killer cells and facilitates their binding to targets. Its expression is regulated by IL2 activation of the cells. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:8487',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ORC1',\n",
       "  'name': 'origin recognition complex subunit 1',\n",
       "  'synonyms': 'HSORC1|ORC1L|PARC1',\n",
       "  'description': 'The origin recognition complex (ORC) is a highly conserved six subunits protein complex essential for the initiation of the DNA replication in eukaryotic cells. Studies in yeast demonstrated that ORC binds specifically to origins of replication and serves as a platform for the assembly of additional initiation factors such as Cdc6 and Mcm proteins. The protein encoded by this gene is the largest subunit of the ORC complex. While other ORC subunits are stable throughout the cell cycle, the levels of this protein vary during the cell cycle, which has been shown to be controlled by ubiquitin-mediated proteolysis after initiation of DNA replication. This protein is found to be selectively phosphorylated during mitosis. It is also reported to interact with MYST histone acetyltransferase 2 (MyST2/HBO1), a protein involved in control of transcription silencing. Alternatively spliced transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Jun 2010].'},\n",
       " {'id': 'HGNC:16068',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PCNT',\n",
       "  'name': 'pericentrin',\n",
       "  'synonyms': 'KEN|MOPD2|PCN|PCNT2|PCNTB|PCTN2|SCKL4',\n",
       "  'description': 'The protein encoded by this gene binds to calmodulin and is expressed in the centrosome. It is an integral component of the pericentriolar material (PCM). The protein contains a series of coiled-coil domains and a highly conserved PCM targeting motif called the PACT domain near its C-terminus. The protein interacts with the microtubule nucleation component gamma-tubulin and is likely important to normal functioning of the centrosomes, cytoskeleton, and cell-cycle progression. Mutations in this gene cause Seckel syndrome-4 and microcephalic osteodysplastic primordial dwarfism type II. Two transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Oct 2015].'},\n",
       " {'id': 'HGNC:8795',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PDE9A',\n",
       "  'name': 'phosphodiesterase 9A',\n",
       "  'synonyms': 'HSPDE9A2',\n",
       "  'description': 'The protein encoded by this gene catalyzes the hydrolysis of cAMP and cGMP to their corresponding monophosphates. The encoded protein plays a role in signal transduction by regulating the intracellular concentration of these cyclic nucleotides. Multiple transcript variants encoding several different isoforms have been found for this gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:9065',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PLCG1',\n",
       "  'name': 'phospholipase C gamma 1',\n",
       "  'synonyms': 'NCKAP3|PLC-II|PLC1|PLC148|PLCgamma1',\n",
       "  'description': 'The protein encoded by this gene catalyzes the formation of inositol 1,4,5-trisphosphate and diacylglycerol from phosphatidylinositol 4,5-bisphosphate. This reaction uses calcium as a cofactor and plays an important role in the intracellular transduction of receptor-mediated tyrosine kinase activators. For example, when activated by SRC, the encoded protein causes the Ras guanine nucleotide exchange factor RasGRP1 to translocate to the Golgi, where it activates Ras. Also, this protein has been shown to be a major substrate for heparin-binding growth factor 1 (acidic fibroblast growth factor)-activated tyrosine kinase. Two transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:9077',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PLK1',\n",
       "  'name': 'polo like kinase 1',\n",
       "  'synonyms': 'PLK|STPK13',\n",
       "  'description': 'The Ser/Thr protein kinase encoded by this gene belongs to the CDC5/Polo subfamily. It is highly expressed during mitosis and elevated levels are found in many different types of cancer. Depletion of this protein in cancer cells dramatically inhibited cell proliferation and induced apoptosis; hence, it is a target for cancer therapy. [provided by RefSeq, Sep 2015].'},\n",
       " {'id': 'HGNC:9092',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PLSCR1',\n",
       "  'name': 'phospholipid scramblase 1',\n",
       "  'synonyms': 'MMTRA1B',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:9175',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'POLD1',\n",
       "  'name': 'DNA polymerase delta 1, catalytic subunit',\n",
       "  'synonyms': 'CDC2|CRCS10|MDPL|POLD',\n",
       "  'description': \"This gene encodes the 125-kDa catalytic subunit of DNA polymerase delta. DNA polymerase delta possesses both polymerase and 3' to 5' exonuclease activity and plays a critical role in DNA replication and repair. Alternatively spliced transcript variants have been observed for this gene, and a pseudogene of this gene is located on the long arm of chromosome 6. [provided by RefSeq, Mar 2012].\"},\n",
       " {'id': 'HGNC:9176',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'POLD2',\n",
       "  'name': 'DNA polymerase delta 2, accessory subunit',\n",
       "  'synonyms': nan,\n",
       "  'description': \"This gene encodes the 50-kDa catalytic subunit of DNA polymerase delta. DNA polymerase delta possesses both polymerase and 3' to 5' exonuclease activity and plays a critical role in DNA replication and repair. The encoded protein is required for the stimulation of DNA polymerase delta activity by the processivity cofactor proliferating cell nuclear antigen (PCNA). Expression of this gene may be a marker for ovarian carcinomas. Alternatively spliced transcript variants encoding multiple isoforms have been observed for this gene, and a pseudogene of this gene is located on the long arm of chromosome 5. [provided by RefSeq, Mar 2012].\"},\n",
       " {'id': 'HGNC:9187',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'POLR2A',\n",
       "  'name': 'RNA polymerase II subunit A',\n",
       "  'synonyms': 'NEDHIB|POLR2|POLRA|RPB1|RPBh1|RPO2|RPOL2|RpIILS|hRPB220|hsRPB1',\n",
       "  'description': 'This gene encodes the largest subunit of RNA polymerase II, the polymerase responsible for synthesizing messenger RNA in eukaryotes. The product of this gene contains a carboxy terminal domain composed of heptapeptide repeats that are essential for polymerase activity. These repeats contain serine and threonine residues that are phosphorylated in actively transcribing RNA polymerase. In addition, this subunit, in combination with several other polymerase subunits, forms the DNA binding domain of the polymerase, a groove in which the DNA template is transcribed into RNA. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:9191',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'POLR2D',\n",
       "  'name': 'RNA polymerase II subunit D',\n",
       "  'synonyms': 'HSRBP4|HSRPB4|RBP4|RPB16|RPB4',\n",
       "  'description': \"This gene encodes the fourth largest subunit of RNA polymerase II, the polymerase responsible for synthesizing messenger RNA in eukaryotes. In yeast, this polymerase subunit is associated with the polymerase under suboptimal growth conditions and may have a stress protective role. A sequence for a ribosomal pseudogene is contained within the 3' untranslated region of the transcript from this gene. [provided by RefSeq, Jul 2008].\"},\n",
       " {'id': 'HGNC:9253',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PPIA',\n",
       "  'name': 'peptidylprolyl isomerase A',\n",
       "  'synonyms': 'CYPA|CYPH|HEL-S-69p',\n",
       "  'description': 'This gene encodes a member of the peptidyl-prolyl cis-trans isomerase (PPIase) family. PPIases catalyze the cis-trans isomerization of proline imidic peptide bonds in oligopeptides and accelerate the folding of proteins. The encoded protein is a cyclosporin binding-protein and may play a role in cyclosporin A-mediated immunosuppression. The protein can also interact with several HIV proteins, including p55 gag, Vpr, and capsid protein, and has been shown to be necessary for the formation of infectious HIV virions. Multiple pseudogenes that map to different chromosomes have been reported. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:9293',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PPP1R3C',\n",
       "  'name': 'protein phosphatase 1 regulatory subunit 3C',\n",
       "  'synonyms': 'PPP1R5|PTG',\n",
       "  'description': 'This gene encodes a carbohydrate binding protein that is a subunit of the protein phosphatase 1 (PP1) complex. PP1 catalyzes reversible protein phosphorylation, which is important in a wide range of cellular activities. The encoded protein affects glycogen biosynthesis by activating glycogen synthase and limiting glycogen breakdown by reducing glycogen phosphorylase activity. DNA hypermethylation of this gene has been found in colorectal cancer patients. The encoded protein also interacts with the laforin protein, which is a protein tyrosine phosphatase implicated in Lafora disease. [provided by RefSeq, Sep 2016].'},\n",
       " {'id': 'HGNC:9305',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PPP2R2B',\n",
       "  'name': 'protein phosphatase 2 regulatory subunit Bbeta',\n",
       "  'synonyms': 'B55BETA|PP2AB55BETA|PP2ABBETA|PP2APR55B|PP2APR55BETA|PR2AB55BETA|PR2ABBETA|PR2APR55BETA|PR52B|PR55-BETA|PR55BETA|SCA12',\n",
       "  'description': \"The product of this gene belongs to the phosphatase 2 regulatory subunit B family. Protein phosphatase 2 is one of the four major Ser/Thr phosphatases, and it is implicated in the negative control of cell growth and division. It consists of a common heteromeric core enzyme, which is composed of a catalytic subunit and a constant regulatory subunit, that associates with a variety of regulatory subunits. The B regulatory subunit might modulate substrate selectivity and catalytic activity. This gene encodes a beta isoform of the regulatory subunit B55 subfamily. Defects in this gene cause autosomal dominant spinocerebellar ataxia 12 (SCA12), a disease caused by degeneration of the cerebellum, sometimes involving the brainstem and spinal cord, and in resulting in poor coordination of speech and body movements. Multiple alternatively spliced variants, which encode different isoforms, have been identified for this gene. The 5' UTR of some of these variants includes a CAG trinucleotide repeat sequence (7-28 copies) that can be expanded to 55-78 copies in cases of SCA12. [provided by RefSeq, Jul 2016].\"},\n",
       " {'id': 'HGNC:9406',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PKN2',\n",
       "  'name': 'protein kinase N2',\n",
       "  'synonyms': 'PAK2|PRK2|PRKCL2|PRO2042|Pak-2|STK7',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:9411',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PRKCSH',\n",
       "  'name': 'protein kinase C substrate 80K-H',\n",
       "  'synonyms': 'AGE-R2|G19P1|GIIB|PCLD|PCLD1|PKCSH|PLD1|VASAP-60',\n",
       "  'description': 'This gene encodes the beta-subunit of glucosidase II, an N-linked glycan-processing enzyme in the endoplasmic reticulum. The encoded protein is an acidic phosphoprotein known to be a substrate for protein kinase C. Mutations in this gene have been associated with the autosomal dominant polycystic liver disease. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Jan 2014].'},\n",
       " {'id': 'HGNC:9416',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PRKG2',\n",
       "  'name': 'protein kinase cGMP-dependent 2',\n",
       "  'synonyms': 'PKG2|PRKGR2|cGK2|cGKII',\n",
       "  'description': 'This gene encodes a protein that belongs to the serine/threonine protein kinase family of proteins. The encoded protein binds to and inhibits the activation of several receptor tyrosine kinases. The membrane-bound protein is a regulator of intestinal secretion, bone growth and renin secretion. Alternate splicing results in multiple transcript variants encoding distinct isoforms whose regulatory N-termini differ in length but whose C-terminal catalytic domains are identical. [provided by RefSeq, May 2018].'},\n",
       " {'id': 'HGNC:9449',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PRNP',\n",
       "  'name': 'prion protein',\n",
       "  'synonyms': 'ASCR|AltPrP|CD230|CJD|GSS|KURU|PRIP|PrP|PrP27-30|PrP33-35C|PrPc|p27-30',\n",
       "  'description': 'The protein encoded by this gene is a membrane glycosylphosphatidylinositol-anchored glycoprotein that tends to aggregate into rod-like structures. The encoded protein contains a highly unstable region of five tandem octapeptide repeats. This gene is found on chromosome 20, approximately 20 kbp upstream of a gene which encodes a biochemically and structurally similar protein to the one encoded by this gene. Mutations in the repeat region as well as elsewhere in this gene have been associated with Creutzfeldt-Jakob disease, fatal familial insomnia, Gerstmann-Straussler disease, Huntington disease-like 1, and kuru. An overlapping open reading frame has been found for this gene that encodes a smaller, structurally unrelated protein, AltPrp. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Nov 2014].'},\n",
       " {'id': 'HGNC:9466',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PRPSAP1',\n",
       "  'name': 'phosphoribosyl pyrophosphate synthetase associated protein 1',\n",
       "  'synonyms': 'PAP39',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:9476',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HTRA1',\n",
       "  'name': 'HtrA serine peptidase 1',\n",
       "  'synonyms': 'ARMD7|CADASIL2|CARASIL|HtrA|L56|ORF480|PRSS11',\n",
       "  'description': 'This gene encodes a member of the trypsin family of serine proteases. This protein is a secreted enzyme that is proposed to regulate the availability of insulin-like growth factors (IGFs) by cleaving IGF-binding proteins. It has also been suggested to be a regulator of cell growth. Variations in the promoter region of this gene are the cause of susceptibility to age-related macular degeneration type 7. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:9536',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PSMA7',\n",
       "  'name': 'proteasome 20S subunit alpha 7',\n",
       "  'synonyms': 'C6|HEL-S-276|HSPC|RC6-1|XAPC7',\n",
       "  'description': 'The 26S proteasome is a multicatalytic proteinase complex with a highly ordered structure composed of 2 complexes, a 20S core and a 19S regulator. The 20S core is composed of 4 rings of 28 non-identical subunits; 2 rings are composed of 7 alpha subunits and 2 rings are composed of 7 beta subunits. Proteasomes are distributed throughout eukaryotic cells at a high concentration and cleave peptides in an ATP/ubiquitin-dependent process in a non-lysosomal pathway. This gene encodes a member of the peptidase T1A family that functions as a 20S core alpha subunit. The encoded protein interacts with the hepatitis B virus X protein and plays a role in regulating hepatitis C virus internal ribosome entry site (IRES) activity, an activity essential for viral replication. The encoded protein also plays a role in the cellular stress response by regulating hypoxia-inducible factor-1alpha. A pseudogene of this gene is located on the long arm of chromosome 9. [provided by RefSeq, Jul 2012].'},\n",
       " {'id': 'HGNC:9797',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RABIF',\n",
       "  'name': 'RAB interacting factor',\n",
       "  'synonyms': 'MSS4|RASGFR3|RASGRF3',\n",
       "  'description': 'This gene encodes a member of the SCE4/YPT1/RAB family of small GTP-binding proteins that are involved in the regulation of intracellular vesicular transport. This protein stimulates GTP-GDP exchange in SEC4, and to a lesser extent in YPT1 and RAB3A, and may play a general role in vesicular transport. [provided by RefSeq, Oct 2011].'},\n",
       " {'id': 'HGNC:9954',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'REL',\n",
       "  'name': 'REL proto-oncogene, NF-kB subunit',\n",
       "  'synonyms': 'C-Rel|HIVEN86A',\n",
       "  'description': \"This gene encodes a protein that belongs to the Rel homology domain/immunoglobulin-like fold, plexin, transcription factor (RHD/IPT) family. Members of this family regulate genes involved in apoptosis, inflammation, the immune response, and oncogenic processes. This proto-oncogene plays a role in the survival and proliferation of B lymphocytes. Mutation or amplification of this gene is associated with B-cell lymphomas, including Hodgkin's lymphoma. Single nucleotide polymorphisms in this gene are associated with susceptibility to ulcerative colitis and rheumatoid arthritis. Alternative splicing results in multiple transcript variants encoding different isoforms. [provided by RefSeq, Apr 2014].\"},\n",
       " {'id': 'HGNC:10317',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RPL23A',\n",
       "  'name': 'ribosomal protein L23a',\n",
       "  'synonyms': 'L23A|MDA20',\n",
       "  'description': 'Ribosomes, the organelles that catalyze protein synthesis, consist of a small 40S subunit and a large 60S subunit. Together these subunits are composed of 4 RNA species and approximately 80 structurally distinct proteins. This gene encodes a ribosomal protein that is a component of the 60S subunit. The protein belongs to the L23P family of ribosomal proteins. It is located in the cytoplasm. The protein may be one of the target molecules involved in mediating growth inhibition by interferon. In yeast, the corresponding protein binds to a specific site on the 26S rRNA. This gene is co-transcribed with the U42A, U42B, U101A, and U101B small nucleolar RNA genes, which are located in its third, first, second, and fourth introns, respectively. As is typical for genes encoding ribosomal proteins, there are multiple processed pseudogenes of this gene dispersed through the genome. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:10378',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MRPL12',\n",
       "  'name': 'mitochondrial ribosomal protein L12',\n",
       "  'synonyms': '5c5-2|L12mt|MRP-L31/34|MRPL7|MRPL7/L12|RPML12',\n",
       "  'description': 'Mammalian mitochondrial ribosomal proteins are encoded by nuclear genes and help in protein synthesis within the mitochondrion. Mitochondrial ribosomes (mitoribosomes) consist of a small 28S subunit and a large 39S subunit. They have an estimated 75% protein to rRNA composition compared to prokaryotic ribosomes, where this ratio is reversed. Another difference between mammalian mitoribosomes and prokaryotic ribosomes is that the latter contain a 5S rRNA. Among different species, the proteins comprising the mitoribosome differ greatly in sequence, and sometimes in biochemical properties, which prevents easy recognition by sequence homology. This gene encodes a 39S subunit protein which forms homodimers. In prokaryotic ribosomes, two L7/L12 dimers and one L10 protein form the L8 protein complex. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:10421',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RPS3A',\n",
       "  'name': 'ribosomal protein S3A',\n",
       "  'synonyms': 'FTE1|MFTL|S3A',\n",
       "  'description': 'Ribosomes, the organelles that catalyze protein synthesis, consist of a small 40S subunit and a large 60S subunit. Together these subunits are composed of 4 RNA species and approximately 80 structurally distinct proteins. This gene encodes a ribosomal protein that is a component of the 40S subunit. The protein belongs to the S3AE family of ribosomal proteins. It is located in the cytoplasm. Disruption of the gene encoding rat ribosomal protein S3a, also named v-fos transformation effector protein, in v-fos-transformed rat cells results in reversion of the transformed phenotype. This gene is co-transcribed with the U73A and U73B small nucleolar RNA genes, which are located in its fourth and third introns, respectively. As is typical for genes encoding ribosomal proteins, there are multiple processed pseudogenes of this gene dispersed through the genome. Alternatively spliced transcript variants have been found for this gene. [provided by RefSeq, May 2012].'},\n",
       " {'id': 'HGNC:10429',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RPS6',\n",
       "  'name': 'ribosomal protein S6',\n",
       "  'synonyms': 'S6',\n",
       "  'description': 'Ribosomes, the organelles that catalyze protein synthesis, consist of a small 40S subunit and a large 60S subunit. Together these subunits are composed of 4 RNA species and approximately 80 structurally distinct proteins. This gene encodes a cytoplasmic ribosomal protein that is a component of the 40S subunit. The protein belongs to the S6E family of ribosomal proteins. It is the major substrate of protein kinases in the ribosome, with subsets of five C-terminal serine residues phosphorylated by different protein kinases. Phosphorylation is induced by a wide range of stimuli, including growth factors, tumor-promoting agents, and mitogens. Dephosphorylation occurs at growth arrest. The protein may contribute to the control of cell growth and proliferation through the selective translation of particular classes of mRNA. As is typical for genes encoding ribosomal proteins, there are multiple processed pseudogenes of this gene dispersed through the genome. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:10441',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RPS8',\n",
       "  'name': 'ribosomal protein S8',\n",
       "  'synonyms': 'S8',\n",
       "  'description': 'Ribosomes, the organelles that catalyze protein synthesis, consist of a small 40S subunit and a large 60S subunit. Together these subunits are composed of 4 RNA species and approximately 80 structurally distinct proteins. This gene encodes a ribosomal protein that is a component of the 40S subunit. The protein belongs to the S8E family of ribosomal proteins. It is located in the cytoplasm. Increased expression of this gene in colorectal tumors and colon polyps compared to matched normal colonic mucosa has been observed. This gene is co-transcribed with the small nucleolar RNA genes U38A, U38B, U39, and U40, which are located in its fourth, fifth, first, and second introns, respectively. As is typical for genes encoding ribosomal proteins, there are multiple processed pseudogenes of this gene dispersed through the genome. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:10388',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RPS15',\n",
       "  'name': 'ribosomal protein S15',\n",
       "  'synonyms': 'RIG|S15',\n",
       "  'description': 'Ribosomes, the organelles that catalyze protein synthesis, consist of a small 40S subunit and a large 60S subunit. Together these subunits are composed of 4 RNA species and approximately 80 structurally distinct proteins. This gene encodes a ribosomal protein that is a component of the 40S subunit. The protein belongs to the S19P family of ribosomal proteins. It is located in the cytoplasm. This gene has been found to be activated in various tumors, such as insulinomas, esophageal cancers, and colon cancers. As is typical for genes encoding ribosomal proteins, there are multiple processed pseudogenes of this gene dispersed through the genome. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Apr 2015].'},\n",
       " {'id': 'HGNC:10496',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'S100A6',\n",
       "  'name': 'S100 calcium binding protein A6',\n",
       "  'synonyms': '2A9|5B10|CABP|CACY|PRA|S10A6',\n",
       "  'description': 'The protein encoded by this gene is a member of the S100 family of proteins containing 2 EF-hand calcium-binding motifs. S100 proteins are localized in the cytoplasm and/or nucleus of a wide range of cells, and involved in the regulation of a number of cellular processes such as cell cycle progression and differentiation. S100 genes include at least 13 members which are located as a cluster on chromosome 1q21. This protein may function in stimulation of Ca2+-dependent insulin release, stimulation of prolactin secretion, and exocytosis. Chromosomal rearrangements and altered expression of this gene have been implicated in melanoma. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:10774',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SFPQ',\n",
       "  'name': 'splicing factor proline and glutamine rich',\n",
       "  'synonyms': 'POMP100|PPP1R140|PSF',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:10822',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SH3BGR',\n",
       "  'name': 'SH3 domain binding glutamate rich protein',\n",
       "  'synonyms': '21-GARP',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:10899',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SKP1',\n",
       "  'name': 'S-phase kinase associated protein 1',\n",
       "  'synonyms': 'EMC19|OCP-II|OCP2|SKP1A|TCEB1L|p19A',\n",
       "  'description': 'This gene encodes a component of SCF complexes, which are composed of this protein, cullin 1, a ring-box protein, and one member of the F-box family of proteins. This protein binds directly to the F-box motif found in F-box proteins. SCF complexes are involved in the regulated ubiquitination of specific protein substrates, which targets them for degradation by the proteosome. Specific F-box proteins recognize different target protein(s), and many specific SCF substrates have been identified including regulators of cell cycle progression and development. Studies have also characterized the protein as an RNA polymerase II elongation factor. Alternative splicing of this gene results in two transcript variants. A related pseudogene has been identified on chromosome 7. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:11049',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SLC6A3',\n",
       "  'name': 'solute carrier family 6 member 3',\n",
       "  'synonyms': 'DAT|DAT1|PKDYS|PKDYS1',\n",
       "  'description': \"This gene encodes a dopamine transporter which is a member of the sodium- and chloride-dependent neurotransmitter transporter family. The 3' UTR of this gene contains a 40 bp tandem repeat, referred to as a variable number tandem repeat or VNTR, which can be present in 3 to 11 copies. Variation in the number of repeats is associated with idiopathic epilepsy, attention-deficit hyperactivity disorder, dependence on alcohol and cocaine, susceptibility to Parkinson disease and protection against nicotine dependence.[provided by RefSeq, Nov 2009].\"},\n",
       " {'id': 'HGNC:11100',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SMARCA4',\n",
       "  'name': 'SWI/SNF related, matrix associated, actin dependent regulator of chromatin, subfamily a, member 4',\n",
       "  'synonyms': 'BAF190|BAF190A|BRG1|CSS4|MRD16|RTPS2|SNF2|SNF2-beta|SNF2L4|SNF2LB|SWI2|hSNF2b',\n",
       "  'description': 'The protein encoded by this gene is a member of the SWI/SNF family of proteins and is similar to the brahma protein of Drosophila. Members of this family have helicase and ATPase activities and are thought to regulate transcription of certain genes by altering the chromatin structure around those genes. The encoded protein is part of the large ATP-dependent chromatin remodeling complex SNF/SWI, which is required for transcriptional activation of genes normally repressed by chromatin. In addition, this protein can bind BRCA1, as well as regulate the expression of the tumorigenic protein CD44. Mutations in this gene cause rhabdoid tumor predisposition syndrome type 2. Multiple transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, May 2012].'},\n",
       " {'id': 'HGNC:11103',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SMARCB1',\n",
       "  'name': 'SWI/SNF related, matrix associated, actin dependent regulator of chromatin, subfamily b, member 1',\n",
       "  'synonyms': 'BAF47|CSS3|INI1|MRD15|PPP1R144|RDT|RTPS1|SNF5|SNF5L1|SWNTS1|Sfh1p|Snr1|hSNFS',\n",
       "  'description': 'The protein encoded by this gene is part of a complex that relieves repressive chromatin structures, allowing the transcriptional machinery to access its targets more effectively. The encoded nuclear protein may also bind to and enhance the DNA joining activity of HIV-1 integrase. This gene has been found to be a tumor suppressor, and mutations in it have been associated with malignant rhabdoid tumors. Alternatively spliced transcript variants have been found for this gene. [provided by RefSeq, Dec 2015].'},\n",
       " {'id': 'HGNC:11205',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SP1',\n",
       "  'name': 'Sp1 transcription factor',\n",
       "  'synonyms': nan,\n",
       "  'description': 'The protein encoded by this gene is a zinc finger transcription factor that binds to GC-rich motifs of many promoters. The encoded protein is involved in many cellular processes, including cell differentiation, cell growth, apoptosis, immune responses, response to DNA damage, and chromatin remodeling. Post-translational modifications such as phosphorylation, acetylation, glycosylation, and proteolytic processing significantly affect the activity of this protein, which can be an activator or a repressor. Three transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Nov 2014].'},\n",
       " {'id': 'HGNC:11249',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SPN',\n",
       "  'name': 'sialophorin',\n",
       "  'synonyms': 'CD43|GALGP|GPL115|LSN',\n",
       "  'description': 'This gene encodes a highly sialylated glycoprotein that functions in antigen-specific activation of T cells, and is found on the surface of thymocytes, T lymphocytes, monocytes, granulocytes, and some B lymphocytes. It contains a mucin-like extracellular domain, a transmembrane region and a carboxy-terminal intracellular region. The extracellular domain has a high proportion of serine and threonine residues, allowing extensive O-glycosylation, and has one potential N-glycosylation site, while the carboxy-terminal region has potential phosphorylation sites that may mediate transduction of activation signals. Different glycoforms of this protein have been described. In stimulated immune cells, proteolytic cleavage of the extracellular domain occurs in some cell types, releasing a soluble extracellular fragment. Defects in expression of this gene are associated with Wiskott-Aldrich syndrome. [provided by RefSeq, Sep 2017].'},\n",
       " {'id': 'HGNC:11307',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SRPRA',\n",
       "  'name': 'SRP receptor subunit alpha',\n",
       "  'synonyms': 'DP|SRPR|Sralpha',\n",
       "  'description': 'The gene encodes a subunit of the endoplasmic reticulum signal recognition particle receptor that, in conjunction with the signal recognition particle, is involved in the targeting and translocation of signal sequence tagged secretory and membrane proteins across the endoplasmic reticulum. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Apr 2010].'},\n",
       " {'id': 'HGNC:11467',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SUPT4H1',\n",
       "  'name': 'SPT4 homolog, DSIF elongation factor subunit',\n",
       "  'synonyms': 'SPT4|SPT4H|SUPT4H|Supt4a',\n",
       "  'description': 'This gene encodes the small subunit of DRB (5,6-dichloro-1-beta-d-ribofuranosylbenzimidazole) sensitivity-inducing factor (DSIF) complex, which regulates mRNA processing and transcription elongation by RNA polymerase II. The encoded protein is localized to the nucleus and interacts with the large subunit (SUPT5H) to form the DSIF complex. Related pseudogenes have been identified on chromosomes 2 and 12. Alternatively spliced transcript variants have been found for this gene. [provided by RefSeq, Nov 2012].'},\n",
       " {'id': 'HGNC:11479',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SUV39H1',\n",
       "  'name': 'suppressor of variegation 3-9 homolog 1',\n",
       "  'synonyms': 'H3-K9-HMTase 1|KMT1A|MG44|SUV39H',\n",
       "  'description': 'This gene encodes an evolutionarily-conserved protein containing an N-terminal chromodomain and a C-terminal SET domain. The encoded protein is a histone methyltransferase that trimethylates lysine 9 of histone H3, which results in transcriptional gene silencing. Loss of function of this gene disrupts heterochromatin formation and may cause chromosome instability. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Aug 2013].'},\n",
       " {'id': 'HGNC:6859',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MAP3K7',\n",
       "  'name': 'mitogen-activated protein kinase kinase kinase 7',\n",
       "  'synonyms': 'CSCF|FMD2|MEKK7|TAK1|TGF1a',\n",
       "  'description': 'The protein encoded by this gene is a member of the serine/threonine protein kinase family. This kinase mediates the signaling transduction induced by TGF beta and morphogenetic protein (BMP), and controls a variety of cell functions including transcription regulation and apoptosis. In response to IL-1, this protein forms a kinase complex including TRAF6, MAP3K7P1/TAB1 and MAP3K7P2/TAB2; this complex is required for the activation of nuclear factor kappa B. This kinase can also activate MAPK8/JNK, MAP2K4/MKK4, and thus plays a role in the cell response to environmental stresses. Four alternatively spliced transcript variants encoding distinct isoforms have been reported. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:11717',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TEAD4',\n",
       "  'name': 'TEA domain transcription factor 4',\n",
       "  'synonyms': 'EFTR-2|RTEF1|TCF13L1|TEF-3|TEF3|TEFR-1|hRTEF-1B',\n",
       "  'description': 'This gene product is a member of the transcriptional enhancer factor (TEF) family of transcription factors, which contain the TEA/ATTS DNA-binding domain. It is preferentially expressed in the skeletal muscle, and binds to the M-CAT regulatory element found in promoters of muscle-specific genes to direct their gene expression. Alternatively spliced transcripts encoding distinct isoforms, some of which are translated through the use of a non-AUG (UUG) initiation codon, have been described for this gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:11730',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TERT',\n",
       "  'name': 'telomerase reverse transcriptase',\n",
       "  'synonyms': 'CMM9|DKCA2|DKCB4|EST2|PFBMFT1|TCS1|TP2|TRT|hEST2|hTRT',\n",
       "  'description': 'Telomerase is a ribonucleoprotein polymerase that maintains telomere ends by addition of the telomere repeat TTAGGG. The enzyme consists of a protein component with reverse transcriptase activity, encoded by this gene, and an RNA component which serves as a template for the telomere repeat. Telomerase expression plays a role in cellular senescence, as it is normally repressed in postnatal somatic cells resulting in progressive shortening of telomeres. Deregulation of telomerase expression in somatic cells may be involved in oncogenesis. Studies in mouse suggest that telomerase also participates in chromosomal repair, since de novo synthesis of telomere repeats may occur at double-stranded breaks. Alternatively spliced variants encoding different isoforms of telomerase reverse transcriptase have been identified; the full-length sequence of some variants has not been determined. Alternative splicing at this locus is thought to be one mechanism of regulation of telomerase activity. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:11741',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TFAM',\n",
       "  'name': 'transcription factor A, mitochondrial',\n",
       "  'synonyms': 'MTDPS15|MTTF1|MTTFA|TCF6|TCF6L1|TCF6L2|TCF6L3',\n",
       "  'description': \"This gene encodes a key mitochondrial transcription factor containing two high mobility group motifs. The encoded protein also functions in mitochondrial DNA replication and repair. Sequence polymorphisms in this gene are associated with Alzheimer's and Parkinson's diseases. There are pseudogenes for this gene on chromosomes 6, 7, and 11. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Aug 2012].\"},\n",
       " {'id': 'HGNC:11744',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TFAP2C',\n",
       "  'name': 'transcription factor AP-2 gamma',\n",
       "  'synonyms': 'AP2-GAMMA|ERF1|TFAP2G|hAP-2g',\n",
       "  'description': 'The protein encoded by this gene is a sequence-specific DNA-binding transcription factor involved in the activation of several developmental genes. The encoded protein can act as either a homodimer or heterodimer with other family members and is induced during retinoic acid-mediated differentiation. It plays a role in the development of the eyes, face, body wall, limbs, and neural tube. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:11763',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TFRC',\n",
       "  'name': 'transferrin receptor',\n",
       "  'synonyms': 'CD71|IMD46|T9|TFR|TFR1|TR|TRFR|p90',\n",
       "  'description': 'This gene encodes a cell surface receptor necessary for cellular iron uptake by the process of receptor-mediated endocytosis. This receptor is required for erythropoiesis and neurologic development. Multiple alternatively spliced variants have been identified. [provided by RefSeq, Sep 2015].'},\n",
       " {'id': 'HGNC:11793',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'THOP1',\n",
       "  'name': 'thimet oligopeptidase 1',\n",
       "  'synonyms': 'EP24.15|MEPD_HUMAN|MP78|TOP',\n",
       "  'description': 'The protein encoded by this gene is a kininase that uses zinc as a cofactor. The encoded oligopeptidase cleaves cytosolic peptides, making them unavailable for display on antigen-presenting cells. This protein also cleaves neuropeptides under 20 aa in length and can degrade beta-amyloid precursor protein to amyloidogenic peptides. [provided by RefSeq, Nov 2015].'},\n",
       " {'id': 'HGNC:11875',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TMPO',\n",
       "  'name': 'thymopoietin',\n",
       "  'synonyms': 'CMD1T|LAP2|LEMD4|PRO0868|TP',\n",
       "  'description': 'Through alternative splicing, this gene encodes several distinct LEM domain containing protein isoforms. LEM domain proteins include inner nuclear membrane and intranuclear proteins, and are involved in a variety of cellular functions including gene expression, chromatin organization, and replication and cell cycle control. The encoded alpha isoform is broadly diffuse in the nucleus and contains a lamin binding domain, while the beta and gamma isoforms are localized to the nuclear membrane and contain an HDAC3 interaction domain. The distinct isoforms may compete with each other when acting to chaperone other proteins and regulate transcription. [provided by RefSeq, Aug 2019].'},\n",
       " {'id': 'HGNC:11998',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TP53',\n",
       "  'name': 'tumor protein p53',\n",
       "  'synonyms': 'BCC7|BMFS5|LFS1|P53|TRP53',\n",
       "  'description': 'This gene encodes a tumor suppressor protein containing transcriptional activation, DNA binding, and oligomerization domains. The encoded protein responds to diverse cellular stresses to regulate expression of target genes, thereby inducing cell cycle arrest, apoptosis, senescence, DNA repair, or changes in metabolism. Mutations in this gene are associated with a variety of human cancers, including hereditary cancers such as Li-Fraumeni syndrome. Alternative splicing of this gene and the use of alternate promoters result in multiple transcript variants and isoforms. Additional isoforms have also been shown to result from the use of alternate translation initiation codons from identical transcript variants (PMIDs: 12032546, 20937277). [provided by RefSeq, Dec 2016].'},\n",
       " {'id': 'HGNC:12028',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HSP90B1',\n",
       "  'name': 'heat shock protein 90 beta family member 1',\n",
       "  'synonyms': 'ECGP|GP96|GRP94|HEL-S-125m|HEL35|TRA1',\n",
       "  'description': \"This gene encodes a member of a family of adenosine triphosphate(ATP)-metabolizing molecular chaperones with roles in stabilizing and folding other proteins. The encoded protein is localized to melanosomes and the endoplasmic reticulum. Expression of this protein is associated with a variety of pathogenic states, including tumor formation. There is a microRNA gene located within the 5' exon of this gene. There are pseudogenes for this gene on chromosomes 1 and 15. [provided by RefSeq, Aug 2012].\"},\n",
       " {'id': 'HGNC:12032',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TRAF2',\n",
       "  'name': 'TNF receptor associated factor 2',\n",
       "  'synonyms': 'MGC:45012|RNF117|TRAP|TRAP3',\n",
       "  'description': 'The protein encoded by this gene is a member of the TNF receptor associated factor (TRAF) protein family. TRAF proteins associate with, and mediate the signal transduction from members of the TNF receptor superfamily. This protein directly interacts with TNF receptors, and forms a heterodimeric complex with TRAF1. This protein is required for TNF-alpha-mediated activation of MAPK8/JNK and NF-kappaB. The protein complex formed by this protein and TRAF1 interacts with the inhibitor-of-apoptosis proteins (IAPs), and functions as a mediator of the anti-apoptotic signals from TNF receptors. The interaction of this protein with TRADD, a TNF receptor associated apoptotic signal transducer, ensures the recruitment of IAPs for the direct inhibition of caspase activation. BIRC2/c-IAP1, an apoptosis inhibitor possessing ubiquitin ligase activity, can unbiquitinate and induce the degradation of this protein, and thus potentiate TNF-induced apoptosis. Multiple alternatively spliced transcript variants have been found for this gene, but the biological validity of only one transcript has been determined. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:12380',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TSNAX',\n",
       "  'name': 'translin associated factor X',\n",
       "  'synonyms': 'C3PO|TRAX',\n",
       "  'description': 'This gene encodes a protein which specifically interacts with translin, a DNA-binding protein that binds consensus sequences at breakpoint junctions of chromosomal translocations. The encoded protein contains bipartite nuclear targeting sequences that may provide nuclear transport for translin, which lacks any nuclear targeting motifs. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:12401',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TTK',\n",
       "  'name': 'TTK protein kinase',\n",
       "  'synonyms': 'CT96|ESK|MPH1|MPS1|MPS1L1|PYT',\n",
       "  'description': 'This gene encodes a dual specificity protein kinase with the ability to phosphorylate tyrosine, serine and threonine. Associated with cell proliferation, this protein is essential for chromosome alignment at the centromere during mitosis and is required for centrosome duplication. It has been found to be a critical mitotic checkpoint protein for accurate segregation of chromosomes during mitosis. Tumorigenesis may occur when this protein fails to degrade and produces excess centrosomes resulting in aberrant mitotic spindles. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Nov 2009].'},\n",
       " {'id': 'HGNC:12441',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TYMS',\n",
       "  'name': 'thymidylate synthetase',\n",
       "  'synonyms': 'HST422|TMS|TS',\n",
       "  'description': 'Thymidylate synthase catalyzes the methylation of deoxyuridylate to deoxythymidylate using, 10-methylenetetrahydrofolate (methylene-THF) as a cofactor. This function maintains the dTMP (thymidine-5-prime monophosphate) pool critical for DNA replication and repair. The enzyme has been of interest as a target for cancer chemotherapeutic agents. It is considered to be the primary site of action for 5-fluorouracil, 5-fluoro-2-prime-deoxyuridine, and some folate analogs. Expression of this gene and that of a naturally occurring antisense transcript, mitochondrial enolase superfamily member 1 (GeneID:55556), vary inversely when cell-growth progresses from late-log to plateau phase. Polymorphisms in this gene may be associated with etiology of neoplasia, including breast cancer, and response to chemotherapy. [provided by RefSeq, Aug 2017].'},\n",
       " {'id': 'HGNC:12637',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'KDM6A',\n",
       "  'name': 'lysine demethylase 6A',\n",
       "  'synonyms': 'KABUK2|UTX|bA386N14.2',\n",
       "  'description': 'This gene is located on the X chromosome and is the corresponding locus to a Y-linked gene which encodes a tetratricopeptide repeat (TPR) protein. The encoded protein of this gene contains a JmjC-domain and catalyzes the demethylation of tri/dimethylated histone H3. Multiple alternatively spliced transcript variants have been found for this gene. [provided by RefSeq, Apr 2014].'},\n",
       " {'id': 'HGNC:12651',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'VARS1',\n",
       "  'name': 'valyl-tRNA synthetase 1',\n",
       "  'synonyms': 'G7A|NDMSCA|VARS|VARS2',\n",
       "  'description': 'Aminoacyl-tRNA synthetases catalyze the aminoacylation of tRNA by their cognate amino acid. Because of their central role in linking amino acids with nucleotide triplets contained in tRNAs, aminoacyl-tRNA synthetases are thought to be among the first proteins that appeared in evolution. The protein encoded by this gene belongs to class-I aminoacyl-tRNA synthetase family and is located in the class III region of the major histocompatibility complex. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:12741',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'EIF4H',\n",
       "  'name': 'eukaryotic translation initiation factor 4H',\n",
       "  'synonyms': 'WBSCR1|WSCR1|eIF-4H',\n",
       "  'description': 'This gene encodes one of the translation initiation factors, which functions to stimulate the initiation of protein synthesis at the level of mRNA utilization. This gene is deleted in Williams syndrome, a multisystem developmental disorder caused by the deletion of contiguous genes at 7q11.23. Alternative splicing of this gene generates 2 transcript variants. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:12761',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'WEE1',\n",
       "  'name': 'WEE1 G2 checkpoint kinase',\n",
       "  'synonyms': 'WEE1A|WEE1hu',\n",
       "  'description': 'This gene encodes a nuclear protein, which is a tyrosine kinase belonging to the Ser/Thr family of protein kinases. This protein catalyzes the inhibitory tyrosine phosphorylation of CDC2/cyclin B kinase, and appears to coordinate the transition between DNA replication and mitosis by protecting the nucleus from cytoplasmically activated CDC2 kinase. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:12829',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'XRCC2',\n",
       "  'name': 'X-ray repair cross complementing 2',\n",
       "  'synonyms': 'FANCU|POF17|SPGF50',\n",
       "  'description': 'This gene encodes a member of the RecA/Rad51-related protein family that participates in homologous recombination to maintain chromosome stability and repair DNA damage. This gene is involved in the repair of DNA double-strand breaks by homologous recombination and it functionally complements Chinese hamster irs1, a repair-deficient mutant that exhibits hypersensitivity to a number of different DNA-damaging agents. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:12853',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'YWHAH',\n",
       "  'name': 'tyrosine 3-monooxygenase/tryptophan 5-monooxygenase activation protein eta',\n",
       "  'synonyms': 'YWHA1',\n",
       "  'description': \"This gene product belongs to the 14-3-3 family of proteins which mediate signal transduction by binding to phosphoserine-containing proteins. This highly conserved protein family is found in both plants and mammals, and this protein is 99% identical to the mouse, rat and bovine orthologs. This gene contains a 7 bp repeat sequence in its 5' UTR, and changes in the number of this repeat have been associated with early-onset schizophrenia and psychotic bipolar disorder. [provided by RefSeq, Jun 2009].\"},\n",
       " {'id': 'HGNC:13104',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ZSCAN21',\n",
       "  'name': 'zinc finger and SCAN domain containing 21',\n",
       "  'synonyms': 'NY-REN-21|ZNF38|Zipro1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:408',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ALDH5A1',\n",
       "  'name': 'aldehyde dehydrogenase 5 family member A1',\n",
       "  'synonyms': 'SSADH|SSDH',\n",
       "  'description': 'This protein belongs to the aldehyde dehydrogenase family of proteins. This gene encodes a mitochondrial NAD(+)-dependent succinic semialdehyde dehydrogenase. A deficiency of this enzyme, known as 4-hydroxybutyricaciduria, is a rare inborn error in the metabolism of the neurotransmitter 4-aminobutyric acid (GABA). In response to the defect, physiologic fluids from patients accumulate GHB, a compound with numerous neuromodulatory properties. Two transcript variants encoding distinct isoforms have been identified for this gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:13920',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GPANK1',\n",
       "  'name': 'G-patch domain and ankyrin repeats 1',\n",
       "  'synonyms': 'ANKRD59|BAT4|D6S54E|G5|GPATCH10',\n",
       "  'description': 'This gene is located in a cluster of HLA-B-associated transcripts, which is included in the human major histocompatability complex III region. This gene encodes a protein which is thought to play a role in immunity. Multiple alternatively spliced variants, encoding the same protein, have been identified. [provided by RefSeq, Nov 2010].'},\n",
       " {'id': 'HGNC:22979',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SLC10A3',\n",
       "  'name': 'solute carrier family 10 member 3',\n",
       "  'synonyms': 'DXS253E|P3',\n",
       "  'description': 'This gene maps to a GC-rich region of the X chromosome and was identified by its proximity to a CpG island. It is thought to be a housekeeping gene. Alternative splicing results in multiple transcript variants encoding distinct isoforms.[provided by RefSeq, Dec 2008].'},\n",
       " {'id': 'HGNC:389',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'AKR7A2',\n",
       "  'name': 'aldo-keto reductase family 7 member A2',\n",
       "  'synonyms': 'AFAR|AFAR1|AFB1-AR1|AKR7',\n",
       "  'description': 'The protein encoded by this gene belongs to the aldo/keto reductase (AKR) superfamily and AKR7 family, which are involved in the detoxification of aldehydes and ketones. The AKR7 family consists of 3 genes that are present in a cluster on the p arm of chromosome 1. This protein, thought to be localized in the golgi, catalyzes the NADPH-dependent reduction of succinic semialdehyde to the endogenous neuromodulator, gamma-hydroxybutyrate. It may also function as a detoxication enzyme in the reduction of aflatoxin B1 and 2-carboxybenzaldehyde. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Mar 2016].'},\n",
       " {'id': 'HGNC:10020',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RIPK2',\n",
       "  'name': 'receptor interacting serine/threonine kinase 2',\n",
       "  'synonyms': 'CARD3|CARDIAK|CCK|GIG30|RICK|RIP2',\n",
       "  'description': 'This gene encodes a member of the receptor-interacting protein (RIP) family of serine/threonine protein kinases. The encoded protein contains a C-terminal caspase activation and recruitment domain (CARD), and is a component of signaling complexes in both the innate and adaptive immune pathways. It is a potent activator of NF-kappaB and inducer of apoptosis in response to various stimuli. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:3825',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'FPGT',\n",
       "  'name': 'fucose-1-phosphate guanylyltransferase',\n",
       "  'synonyms': 'GFPP',\n",
       "  'description': 'L-fucose is a key sugar in glycoproteins and other complex carbohydrates since it may be involved in many of the functional roles of these macromolecules, such as in cell-cell recognition. The fucosyl donor for these fucosylated oligosaccharides is GDP-beta-L-fucose. There are two alternate pathways for the biosynthesis of GDP-fucose; the major pathway converts GDP-alpha-D-mannose to GDP-beta-L-fucose. The protein encoded by this gene participates in an alternate pathway that is present in certain mammalian tissues, such as liver and kidney, and appears to function as a salvage pathway to reutilize L-fucose arising from the turnover of glycoproteins and glycolipids. This pathway involves the phosphorylation of L-fucose to form beta-L-fucose-1-phosphate, and then condensation of the beta-L-fucose-1-phosphate with GTP by fucose-1-phosphate guanylyltransferase to form GDP-beta-L-fucose. Alternative splicing results in multiple transcript variants. Read-through transcription also exists between this gene and the neighboring downstream TNNI3 interacting kinase (TNNI3K) gene. [provided by RefSeq, Dec 2010].'},\n",
       " {'id': 'HGNC:2351',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CREG1',\n",
       "  'name': 'cellular repressor of E1A stimulated genes 1',\n",
       "  'synonyms': 'CREG',\n",
       "  'description': 'The adenovirus E1A protein both activates and represses gene expression to promote cellular proliferation and inhibit differentiation. The protein encoded by this gene antagonizes transcriptional activation and cellular transformation by E1A. This protein shares limited sequence similarity with E1A and binds both the general transcription factor TBP and the tumor suppressor pRb in vitro. This gene may contribute to the transcriptional control of cell growth and differentiation. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:16665',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'APLN',\n",
       "  'name': 'apelin',\n",
       "  'synonyms': 'APEL|XNPEP2',\n",
       "  'description': 'This gene encodes a peptide that functions as an endogenous ligand for the G-protein coupled apelin receptor. The encoded preproprotein is proteolytically processed into biologically active C-terminal peptide fragments. These peptide fragments activate different tissue specific signaling pathways that regulate diverse biological functions including fluid homeostasis, cardiovascular function and insulin secretion. This protein also functions as a coreceptor for the human immunodeficiency virus 1. [provided by RefSeq, Feb 2016].'},\n",
       " {'id': 'HGNC:3258',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'EIF2B2',\n",
       "  'name': 'eukaryotic translation initiation factor 2B subunit beta',\n",
       "  'synonyms': 'EIF-2Bbeta|EIF2B',\n",
       "  'description': 'This gene encodes the beta subunit of eukaryotic initiation factor-2B (EIF2B). EIF2B is involved in protein synthesis and exchanges GDP and GTP for its activation and deactivation. [provided by RefSeq, Aug 2011].'},\n",
       " {'id': 'HGNC:10984',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SLC25A14',\n",
       "  'name': 'solute carrier family 25 member 14',\n",
       "  'synonyms': 'BMCP1|UCP5',\n",
       "  'description': 'Mitochondrial uncoupling proteins (UCP) are members of the larger family of mitochondrial anion carrier proteins (MACP). Uncoupling proteins separate oxidative phosphorylation from ATP synthesis with energy dissipated as heat, also referred to as the mitochondrial proton leak. Uncoupling proteins facilitate the transfer of anions from the inner to the outer mitochondrial membrane and the return transfer of protons from the outer to the inner mitochondrial membrane. They also reduce the mitochondrial membrane potential in mammalian cells. This gene is widely expressed in many tissues with the greatest abundance in brain and testis. Alternative splicing results in multiple transcript variants. A pseudogene of this gene has been defined on chromosome 4. [provided by RefSeq, Aug 2013].'},\n",
       " {'id': 'HGNC:9341',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PRC1',\n",
       "  'name': 'protein regulator of cytokinesis 1',\n",
       "  'synonyms': 'ASE1',\n",
       "  'description': 'This gene encodes a protein that is involved in cytokinesis. The protein is present at high levels during the S and G2/M phases of mitosis but its levels drop dramatically when the cell exits mitosis and enters the G1 phase. It is located in the nucleus during interphase, becomes associated with mitotic spindles in a highly dynamic manner during mitosis, and localizes to the cell mid-body during cytokinesis. This protein has been shown to be a substrate of several cyclin-dependent kinases (CDKs). It is necessary for polarizing parallel microtubules and concentrating the factors responsible for contractile ring assembly. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Jun 2012].'},\n",
       " {'id': 'HGNC:14135',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PIGQ',\n",
       "  'name': 'phosphatidylinositol glycan anchor biosynthesis class Q',\n",
       "  'synonyms': 'DEE77|EIEE77|GPI1|c407A10.1',\n",
       "  'description': 'This gene is involved in the first step in glycosylphosphatidylinositol (GPI)-anchor biosynthesis. The GPI-anchor is a glycolipid found on many blood cells and serves to anchor proteins to the cell surface. This gene encodes a N-acetylglucosaminyl transferase component that is part of the complex that catalyzes transfer of N-acetylglucosamine (GlcNAc) from UDP-GlcNAc to phosphatidylinositol (PI). Alternatively spliced transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Jun 2012].'},\n",
       " {'id': 'HGNC:11498',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SYNGR1',\n",
       "  'name': 'synaptogyrin 1',\n",
       "  'synonyms': nan,\n",
       "  'description': 'This gene encodes an integral membrane protein associated with presynaptic vesicles in neuronal cells. The exact function of this protein is unclear, but studies of a similar murine protein suggest that it functions in synaptic plasticity without being required for synaptic transmission. The gene product belongs to the synaptogyrin gene family. Three alternatively spliced variants encoding three different isoforms have been identified. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:1151',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'BUB3',\n",
       "  'name': 'BUB3 mitotic checkpoint protein',\n",
       "  'synonyms': 'BUB3L|hBUB3',\n",
       "  'description': 'This gene encodes a protein involved in spindle checkpoint function. The encoded protein contains four WD repeat domains and has sequence similarity with the yeast BUB3 protein. Alternate transcriptional splice variants, encoding different isoforms, have been characterized. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:11390',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'AURKB',\n",
       "  'name': 'aurora kinase B',\n",
       "  'synonyms': 'AIK2|AIM-1|AIM1|ARK-2|ARK2|AurB|IPL1|PPP1R48|STK-1|STK12|STK5|aurkb-sv1|aurkb-sv2',\n",
       "  'description': 'This gene encodes a member of the aurora kinase subfamily of serine/threonine kinases. The genes encoding the other two members of this subfamily are located on chromosomes 19 and 20. These kinases participate in the regulation of alignment and segregation of chromosomes during mitosis and meiosis through association with microtubules. A pseudogene of this gene is located on chromosome 8. Alternatively spliced transcript variants have been found for this gene. [provided by RefSeq, Sep 2015].'},\n",
       " {'id': 'HGNC:9690',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PTTG1',\n",
       "  'name': 'PTTG1 regulator of sister chromatid separation, securin',\n",
       "  'synonyms': 'EAP1|HPTTG|PTTG|TUTR1',\n",
       "  'description': 'The encoded protein is a homolog of yeast securin proteins, which prevent separins from promoting sister chromatid separation. It is an anaphase-promoting complex (APC) substrate that associates with a separin until activation of the APC. The gene product has transforming activity in vitro and tumorigenic activity in vivo, and the gene is highly expressed in various tumors. The gene product contains 2 PXXP motifs, which are required for its transforming and tumorigenic activities, as well as for its stimulation of basic fibroblast growth factor expression. It also contains a destruction box (D box) that is required for its degradation by the APC. The acidic C-terminal region of the encoded protein can act as a transactivation domain. The gene product is mainly a cytosolic protein, although it partially localizes in the nucleus. Three transcript variants encoding the same protein have been found for this gene. [provided by RefSeq, Sep 2013].'},\n",
       " {'id': 'HGNC:3147',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ECEL1',\n",
       "  'name': 'endothelin converting enzyme like 1',\n",
       "  'synonyms': 'DA5D|DINE|ECEX|XCE',\n",
       "  'description': 'This gene encodes a member of the M13 family of endopeptidases. Members of this family are zinc-containing type II integral-membrane proteins that are important regulators of neuropeptide and peptide hormone activity. Mutations in this gene are associated with autosomal recessive distal arthrogryposis, type 5D. This gene has multiple pseudogenes on chromosome 2. Alternative splicing results in multiple transcript variants encoding different isoforms. [provided by RefSeq, Mar 2014].'},\n",
       " {'id': 'HGNC:6732',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NCR2',\n",
       "  'name': 'natural cytotoxicity triggering receptor 2',\n",
       "  'synonyms': 'CD336|LY95|NK-p44|NKP44|dJ149M18.1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:73',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ABCG1',\n",
       "  'name': 'ATP binding cassette subfamily G member 1',\n",
       "  'synonyms': 'ABC8|WHITE1',\n",
       "  'description': 'The protein encoded by this gene is a member of the superfamily of ATP-binding cassette (ABC) transporters. ABC proteins transport various molecules across extra- and intra-cellular membranes. ABC genes are divided into seven distinct subfamilies (ABC1, MDR/TAP, MRP, ALD, OABP, GCN20, White). This protein is a member of the White subfamily. It is involved in macrophage cholesterol and phospholipids transport, and may regulate cellular lipid homeostasis in other cell types. Six alternative splice variants have been identified. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:28980',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GINS1',\n",
       "  'name': 'GINS complex subunit 1',\n",
       "  'synonyms': 'IMD55|PSF1',\n",
       "  'description': 'The yeast heterotetrameric GINS complex is made up of Sld5 (GINS4; MIM 610611), Psf1, Psf2 (GINS2; MIM 610609), and Psf3 (GINS3; MIM 610610). The formation of the GINS complex is essential for the initiation of DNA replication in yeast and Xenopus egg extracts (Ueno et al., 2005 [PubMed 16287864]).[supplied by OMIM, Mar 2008].'},\n",
       " {'id': 'HGNC:29041',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'KLHL21',\n",
       "  'name': 'kelch like family member 21',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:13747',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DLEU1',\n",
       "  'name': 'deleted in lymphocytic leukemia 1',\n",
       "  'synonyms': 'BCMS|BCMS1|DLB1|DLEU2|LEU1|LEU2|LINC00021|NCRNA00021|XTP6',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:13060',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ZNF267',\n",
       "  'name': 'zinc finger protein 267',\n",
       "  'synonyms': 'HZF2',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:14011',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SMC2',\n",
       "  'name': 'structural maintenance of chromosomes 2',\n",
       "  'synonyms': 'CAP-E|CAPE|SMC-2|SMC2L1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:18116',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'KHDRBS1',\n",
       "  'name': 'KH RNA binding domain containing, signal transduction associated 1',\n",
       "  'synonyms': 'Sam68|p62|p68',\n",
       "  'description': \"This gene encodes a member of the K homology domain-containing, RNA-binding, signal transduction-associated protein family. The encoded protein appears to have many functions and may be involved in a variety of cellular processes, including alternative splicing, cell cycle regulation, RNA 3'-end formation, tumorigenesis, and regulation of human immunodeficiency virus gene expression. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Dec 2012].\"},\n",
       " {'id': 'HGNC:8987',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PIM2',\n",
       "  'name': 'Pim-2 proto-oncogene, serine/threonine kinase',\n",
       "  'synonyms': nan,\n",
       "  'description': 'This gene encodes a protooncogene that acts as a serine/threonine protein kinase. Studies determined the encoded protein functions to prevent apoptosis and to promote cell survival.[provided by RefSeq, Nov 2009].'},\n",
       " {'id': 'HGNC:15768',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'OGFR',\n",
       "  'name': 'opioid growth factor receptor',\n",
       "  'synonyms': nan,\n",
       "  'description': 'The protein encoded by this gene is a receptor for opioid growth factor (OGF), also known as [Met(5)]-enkephalin. OGF is a negative regulator of cell proliferation and tissue organization in a variety of processes. The encoded unbound receptor for OGF has been localized to the outer nuclear envelope, where it binds OGF and is translocated into the nucleus. The coding sequence of this gene contains a polymorphic region of 60 nt tandem imperfect repeat units. Several transcripts containing between zero and eight repeat units have been reported. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:5184',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HRH3',\n",
       "  'name': 'histamine receptor H3',\n",
       "  'synonyms': 'GPCR97|HH3R',\n",
       "  'description': 'Histamine is a ubiquitous messenger molecule released from mast cells, enterochromaffin-like cells, and neurons. Its various actions are mediated by histamine receptors H1, H2, H3 and H4. This gene encodes one of the histamine receptors (H3) which belongs to the family 1 of G protein-coupled receptors. It is an integral membrane protein and can regulate neurotransmitter release. This receptor can also increase voltage-dependent calcium current in smooth muscles and innervates the blood vessels and the heart in cardiovascular system. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:16369',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PARK7',\n",
       "  'name': 'Parkinsonism associated deglycase',\n",
       "  'synonyms': 'DJ-1|DJ1|GATD2|HEL-S-67p',\n",
       "  'description': 'The product of this gene belongs to the peptidase C56 family of proteins. It acts as a positive regulator of androgen receptor-dependent transcription. It may also function as a redox-sensitive chaperone, as a sensor for oxidative stress, and it apparently protects neurons against oxidative stress and cell death. Defects in this gene are the cause of autosomal recessive early-onset Parkinson disease 7. Two transcript variants encoding the same protein have been identified for this gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:6157',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ITGB3BP',\n",
       "  'name': 'integrin subunit beta 3 binding protein',\n",
       "  'synonyms': 'CENP-R|CENPR|HSU37139|NRIF3|TAP20',\n",
       "  'description': 'This gene encodes a transcriptional coregulator that binds to and enhances the activity of members of the nuclear receptor families, thyroid hormone receptors and retinoid X receptors. This protein also acts as a corepressor of NF-kappaB-dependent signaling. This protein induces apoptosis in breast cancer cells through a caspase 2-mediated signaling pathway. This protein is also a component of the centromere-specific histone H3 variant nucleosome associated complex (CENP-NAC) and may be involved in mitotic progression by recruiting the histone H3 variant CENP-A to the centromere. Alternate splicing results in multiple transcript variants. [provided by RefSeq, Sep 2011].'},\n",
       " {'id': 'HGNC:16639',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SRRM2',\n",
       "  'name': 'serine/arginine repetitive matrix 2',\n",
       "  'synonyms': '300-KD|CWF21|Cwc21|HSPC075|SRL300|SRm300',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:11250',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SPO11',\n",
       "  'name': 'SPO11 initiator of meiotic double stranded breaks',\n",
       "  'synonyms': 'CT35|SPATA43|TOPOVIA|TOPVIA',\n",
       "  'description': \"Meiotic recombination and chromosome segregation require the formation of double-strand breaks (DSBs) in paired chromosome homologs. During meiosis in yeast, a meiotic recombination protein is covalently-linked to the 5' end of DSBs and is essential for the formation of DSBs. The protein encoded by this gene is similar in sequence and conserved features to the yeast meiotic recombination protein. The encoded protein belongs to the TOP6A protein family. Several transcript variants encoding different isoforms have been found for this gene, but the full-length nature of only two of them have been described. [provided by RefSeq, Jul 2008].\"},\n",
       " {'id': 'HGNC:13339',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'KIF4A',\n",
       "  'name': 'kinesin family member 4A',\n",
       "  'synonyms': 'KIF4|KIF4G1|MRX100',\n",
       "  'description': 'This gene encodes a member of the kinesin 4 subfamily of kinesin related proteins. The encoded protein is an ATP dependent microtubule-based motor protein that is involved in the intracellular transport of membranous organelles. This protein also associates with condensed chromosome arms and may be involved in maintaining chromosome integrity during mitosis. This protein may also be involved in the organization of the central spindle prior to cytokinesis. A pseudogene of this gene is found on chromosome X.[provided by RefSeq, Mar 2010].'},\n",
       " {'id': 'HGNC:15592',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SS18L1',\n",
       "  'name': 'SS18L1 subunit of BAF chromatin remodeling complex',\n",
       "  'synonyms': 'CREST|LP2261',\n",
       "  'description': 'This gene encodes a calcium-responsive transactivator which is an essential subunit of a neuron-specific chromatin-remodeling complex. The structure of this gene is similar to that of the SS18 gene. Mutations in this gene are involved in amyotrophic lateral sclerosis (ALS). Alternatively spliced transcript variants have been found for this gene. [provided by RefSeq, Sep 2014].'},\n",
       " {'id': 'HGNC:2754',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'DECR2',\n",
       "  'name': '2,4-dienoyl-CoA reductase 2',\n",
       "  'synonyms': 'PDCR|SDR17C1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:13241',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RIBC2',\n",
       "  'name': 'RIB43A domain with coiled-coils 2',\n",
       "  'synonyms': 'C22orf11|TRIB',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:20460',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GBGT1',\n",
       "  'name': 'globoside alpha-1,3-N-acetylgalactosaminyltransferase 1 (FORS blood group)',\n",
       "  'synonyms': 'A3GALNT|FS|UNQ2513',\n",
       "  'description': 'This gene encodes a glycosyltransferase that plays a role in the synthesis of Forssman glycolipid (FG), a member of the globoseries glycolipid family. Glycolipids such as FG form attachment sites for the binding of pathogens to cells; expression of this protein may determine host tropism to microorganisms. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Dec 2013].'},\n",
       " {'id': 'HGNC:11586',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TBL2',\n",
       "  'name': 'transducin beta like 2',\n",
       "  'synonyms': 'WBSCR13|WS-betaTRP',\n",
       "  'description': 'This gene encodes a member of the beta-transducin protein family. Most proteins of the beta-transducin family are involved in regulatory functions. This protein is possibly involved in some intracellular signaling pathway. This gene is deleted in Williams-Beuren syndrome, a developmental disorder caused by deletion of multiple genes at 7q11.23. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:10953',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SLCO4A1',\n",
       "  'name': 'solute carrier organic anion transporter family member 4A1',\n",
       "  'synonyms': 'OATP-E|OATP1|OATP4A1|OATPE|OATPRP1|POAT|SLC21A12',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:29523',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CFAP20',\n",
       "  'name': 'cilia and flagella associated protein 20',\n",
       "  'synonyms': 'BUG22|C16orf80|EVORF|GTL3|fSAP23',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:15913',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'NXT1',\n",
       "  'name': 'nuclear transport factor 2 like export factor 1',\n",
       "  'synonyms': 'MTR2|P15',\n",
       "  'description': 'The protein encoded by this gene is located in the nuclear envelope. It has protein similarity to nuclear transport factor 2. This protein functions as a nuclear export factor in both RAN (Ras-related nuclear protein)- and CRM1 (required for chromosome region maintenance)-dependent pathways. It is found to stimulate the export of U1 snRNA in RAN- and CRM1-dependent pathways and the export of tRNA and mRNA in a CRM1-independent pathway. The encoded protein heterodimerizes with Tap protein and may regulate the ability of Tap protein to mediate nuclear mRNA export. The use of alternate polyadenylation sites has been found for this gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:12624',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'USP25',\n",
       "  'name': 'ubiquitin specific peptidase 25',\n",
       "  'synonyms': 'USP21',\n",
       "  'description': 'Ubiquitin (MIM 191339) is a highly conserved 76-amino acid protein involved in regulation of intracellular protein breakdown, cell cycle regulation, and stress response. Ubiquitin is released from degraded proteins by disassembly of the polyubiquitin chains, which is mediated by ubiquitin-specific proteases (USPs), such as USP25 (Valero et al., 1999 [PubMed 10644437]).[supplied by OMIM, Mar 2008].'},\n",
       " {'id': 'HGNC:24271',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CUTC',\n",
       "  'name': 'cutC copper transporter',\n",
       "  'synonyms': 'CGI-32',\n",
       "  'description': 'Members of the CUT family of copper transporters are associated with copper homeostasis and are involved in the uptake, storage, delivery, and efflux of copper (Gupta et al., 1995 [PubMed 7635807]; Li et al., 2005 [PubMed 16182249]).[supplied by OMIM, Mar 2008].'},\n",
       " {'id': 'HGNC:4114',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GAL',\n",
       "  'name': 'galanin and GMAP prepropeptide',\n",
       "  'synonyms': 'ETL8|GAL-GMAP|GALN|GLNN|GMAP',\n",
       "  'description': 'This gene encodes a neuroendocrine peptide that is widely expressed in the central and peripheral nervous systems and also the gastrointestinal tract, pancreas, adrenal gland and urogenital tract. The encoded protein is a precursor that is proteolytically processed to generate two mature peptides: galanin and galanin message-associated peptide (GMAP). Galanin has diverse physiological functions including nociception, feeding and energy homeostasis, osmotic regulation and water balance. GMAP has been demonstrated to possess antifungal activity and hypothesized to be part of the innate immune system. [provided by RefSeq, Jul 2015].'},\n",
       " {'id': 'HGNC:13221',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'BCL11A',\n",
       "  'name': 'BAF chromatin remodeling complex subunit BCL11A',\n",
       "  'synonyms': 'BCL11A-L|BCL11A-S|BCL11A-XL|BCL11a-M|CTIP1|DILOS|EVI9|HBFQTL5|ZNF856',\n",
       "  'description': 'This gene encodes a C2H2 type zinc-finger protein by its similarity to the mouse Bcl11a/Evi9 protein. The corresponding mouse gene is a common site of retroviral integration in myeloid leukemia, and may function as a leukemia disease gene, in part, through its interaction with BCL6. During hematopoietic cell differentiation, this gene is down-regulated. It is possibly involved in lymphoma pathogenesis since translocations associated with B-cell malignancies also deregulates its expression. Multiple transcript variants encoding several different isoforms have been found for this gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'NCBIGene:54073',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'HGNC:14082',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ANLN',\n",
       "  'name': 'anillin actin binding protein',\n",
       "  'synonyms': 'FSGS8|Scraps|scra',\n",
       "  'description': 'This gene encodes an actin-binding protein that plays a role in cell growth and migration, and in cytokinesis. The encoded protein is thought to regulate actin cytoskeletal dynamics in podocytes, components of the glomerulus. Mutations in this gene are associated with focal segmental glomerulosclerosis 8. Alternative splicing results in multiple transcript variants encoding different isoforms. [provided by RefSeq, Oct 2014].'},\n",
       " {'id': 'HGNC:26016',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'LIME1',\n",
       "  'name': 'Lck interacting transmembrane adaptor 1',\n",
       "  'synonyms': 'LIME|dJ583P15.4',\n",
       "  'description': 'This gene encodes a transmembrane adaptor protein that links the T and B-cell receptor stimulation to downstream signaling pathways via its association with the Src family kinases Lck and Lyn, respectively. Alternative splicing of this gene results in multiple transcript variants. [provided by RefSeq, Mar 2015].'},\n",
       " {'id': 'HGNC:17179',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CDC37L1',\n",
       "  'name': 'cell division cycle 37 like 1',\n",
       "  'synonyms': 'CDC37B|HARC',\n",
       "  'description': 'CDC37L1 is a cytoplasmic phosphoprotein that exists in complex with HSP90 (HSPCA; MIM 140571) as well as several other proteins involved in HSP90-mediated protein folding (Scholz et al., 2001 [PubMed 11413142]).[supplied by OMIM, Mar 2008].'},\n",
       " {'id': 'HGNC:21025',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SAYSD1',\n",
       "  'name': 'SAYSVFN motif domain containing 1',\n",
       "  'synonyms': 'C6orf64',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:17563',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'METTL3',\n",
       "  'name': 'methyltransferase like 3',\n",
       "  'synonyms': 'IME4|M6A|MT-A70|Spo8|hMETTL3',\n",
       "  'description': 'This gene encodes the 70 kDa subunit of MT-A which is part of N6-adenosine-methyltransferase. This enzyme is involved in the posttranscriptional methylation of internal adenosine residues in eukaryotic mRNAs, forming N6-methyladenosine. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:30259',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PDXP',\n",
       "  'name': 'pyridoxal phosphatase',\n",
       "  'synonyms': 'CIN|PLP|dJ37E16.5',\n",
       "  'description': 'Pyridoxal 5-prime-phosphate (PLP) is the active form of vitamin B6 that acts as a coenzyme in maintaining biochemical homeostasis. The preferred degradation route from PLP to 4-pyridoxic acid involves the dephosphorylation of PLP by PDXP (Jang et al., 2003 [PubMed 14522954]).[supplied by OMIM, Mar 2008].'},\n",
       " {'id': 'HGNC:16228',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'TRIB3',\n",
       "  'name': 'tribbles pseudokinase 3',\n",
       "  'synonyms': 'C20orf97|NIPK|SINK|SKIP3|TRB3',\n",
       "  'description': 'The protein encoded by this gene is a putative protein kinase that is induced by the transcription factor NF-kappaB. The encoded protein is a negative regulator of NF-kappaB and can also sensitize cells to TNF- and TRAIL-induced apoptosis. In addition, this protein can negatively regulate the cell survival serine-threonine kinase AKT1. Differential promoter usage and alternate splicing result in multiple transcript variants. [provided by RefSeq, Jul 2014].'},\n",
       " {'id': 'HGNC:14152',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ANTKMT',\n",
       "  'name': 'adenine nucleotide translocase lysine methyltransferase',\n",
       "  'synonyms': 'ANT-KMT|C16orf24|FAM173A',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:14177',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'HAGHL',\n",
       "  'name': 'hydroxyacylglutathione hydrolase like',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:16147',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'MCM8',\n",
       "  'name': 'minichromosome maintenance 8 homologous recombination repair factor',\n",
       "  'synonyms': 'C20orf154|POF10|dJ967N21.5',\n",
       "  'description': 'The protein encoded by this gene is one of the highly conserved mini-chromosome maintenance proteins (MCM) that are essential for the initiation of eukaryotic genome replication. The hexameric protein complex formed by the mini-chromosome maintenance proteins is a key component of the pre-replication complex and may be involved in the formation of replication forks and in the recruitment of other DNA replication related proteins. This protein contains the central domain that is conserved among the mini-chromosome maintenance proteins. The encoded protein may interact with other mini-chromosome maintenance proteins and play a role in DNA replication. This gene may be associated with length of reproductive lifespan and menopause. Alternatively spliced transcript variants encoding distinct isoforms have been described. [provided by RefSeq, Jul 2013].'},\n",
       " {'id': 'HGNC:23026',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GNPTG',\n",
       "  'name': 'N-acetylglucosamine-1-phosphate transferase subunit gamma',\n",
       "  'synonyms': 'C16orf27|GNPTAG|LP2537|RJD9',\n",
       "  'description': 'This gene encodes the gamma sunbunit of the N-acetylglucosamine-1-phosphotransferase complex. This hexameric complex, composed of alpha, beta and gamma subunits, catalyzes the first step in synthesis of a mannose 6-phosphate lysosomal recognition marker. This enzyme complex is necessary for targeting of lysosomal hydrolases to the lysosome. Mutations in the gene encoding the gamma subunit have been associated with mucolipidosis IIIC, also known as mucolipidosis III gamma.[provided by RefSeq, Feb 2010].'},\n",
       " {'id': 'NCBIGene:85006',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'PDXK',\n",
       "  'name': 'pyridoxal kinase',\n",
       "  'synonyms': 'C21orf124|C21orf97|HEL-S-1a|HMSN6C|PKH|PNK|PRED79',\n",
       "  'description': 'The protein encoded by this gene phosphorylates vitamin B6, a step required for the conversion of vitamin B6 to pyridoxal-5-phosphate, an important cofactor in intermediary metabolism. The encoded protein is cytoplasmic and probably acts as a homodimer. Alternatively spliced transcript variants have been described, but their biological validity has not been determined. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:16102',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'FAM210B',\n",
       "  'name': 'family with sequence similarity 210 member B',\n",
       "  'synonyms': '5A3|C20orf108|dJ1167H4.1',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:21074',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'GINM1',\n",
       "  'name': 'glycoprotein integral membrane 1',\n",
       "  'synonyms': 'C6orf72',\n",
       "  'description': nan},\n",
       " {'id': 'HGNC:14153',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'CCDC78',\n",
       "  'name': 'coiled-coil domain containing 78',\n",
       "  'synonyms': 'C16orf25|CNM4|JFP10|hsCCDC78',\n",
       "  'description': 'The product of this gene contains two coiled-coil domains. The function of this gene is currently unknown. [provided by RefSeq, Sep 2012].'},\n",
       " {'id': 'HGNC:18501',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'RPS4Y2',\n",
       "  'name': 'ribosomal protein S4 Y-linked 2',\n",
       "  'synonyms': 'RPS4Y2P',\n",
       "  'description': 'The protein encoded by this gene is a ribosomal protein that is highly similar to RPS4Y1. This gene is located in the male-specific region of the Y chromosome. [provided by RefSeq, Aug 2012].'},\n",
       " {'id': 'HGNC:26457',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ZNF681',\n",
       "  'name': 'zinc finger protein 681',\n",
       "  'synonyms': nan,\n",
       "  'description': nan},\n",
       " {'id': 'NCBIGene:348630',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': nan,\n",
       "  'name': None,\n",
       "  'synonyms': None,\n",
       "  'description': None},\n",
       " {'id': 'HGNC:16',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'SERPINA3',\n",
       "  'name': 'serpin family A member 3',\n",
       "  'synonyms': 'AACT|ACT|GIG24|GIG25',\n",
       "  'description': \"The protein encoded by this gene is a member of the serpin family of proteins, a group of proteins that inhibit serine proteases. This gene is one in a cluster of serpin genes located on the q arm of chromosome 14. Polymorphisms in this protein appear to be tissue specific and influence protease targeting. Variations in this protein's sequence have been implicated in Alzheimer's disease, and deficiency of this protein has been associated with liver disease. Mutations have been identified in patients with Parkinson disease and chronic obstructive pulmonary disease. [provided by RefSeq, Jun 2020].\"},\n",
       " {'id': 'HGNC:85',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ACACB',\n",
       "  'name': 'acetyl-CoA carboxylase beta',\n",
       "  'synonyms': 'ACC-beta|ACC2|ACCB|HACC275',\n",
       "  'description': 'Acetyl-CoA carboxylase (ACC) is a complex multifunctional enzyme system. ACC is a biotin-containing enzyme which catalyzes the carboxylation of acetyl-CoA to malonyl-CoA, the rate-limiting step in fatty acid synthesis.  ACC-beta is thought to control fatty acid oxidation by means of the ability of malonyl-CoA to inhibit carnitine-palmitoyl-CoA transferase I, the rate-limiting step in fatty acid uptake and oxidation by mitochondria.  ACC-beta may be involved in the regulation of fatty acid oxidation, rather than fatty acid biosynthesis.  There is evidence for the presence of two ACC-beta isoforms. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:243',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ADD1',\n",
       "  'name': 'adducin 1',\n",
       "  'synonyms': 'ADDA',\n",
       "  'description': 'Adducins are a family of cytoskeletal proteins encoded by three genes (alpha, beta, and gamma). Adducin acts as a heterodimer of the related alpha, beta, or gamma subunits. The protein encoded by this gene represents the alpha subunit. Alpha- and beta-adducin include a protease-resistant N-terminal region and a protease-sensitive, hydrophilic C-terminal region. Adducin binds with high affinity to Ca(2+)/calmodulin and is a substrate for protein kinases A and C. [provided by RefSeq, Aug 2017].'},\n",
       " {'id': 'HGNC:280',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ADRA1D',\n",
       "  'name': 'adrenoceptor alpha 1D',\n",
       "  'synonyms': 'ADRA1|ADRA1A|ADRA1R|ALPHA1|DAR|dJ779E11.2',\n",
       "  'description': 'Alpha-1-adrenergic receptors (alpha-1-ARs) are members of the G protein-coupled receptor superfamily. They activate mitogenic responses and regulate growth and proliferation of many cells. There are 3 alpha-1-AR subtypes: alpha-1A, -1B and -1D, all of which signal through the Gq/11 family of G-proteins and different subtypes show different patterns of activation. This gene encodes alpha-1D-adrenergic receptor. Similar to alpha-1B-adrenergic receptor gene, this gene comprises 2 exons and a single intron that interrupts the coding region. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:277',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ADRA1A',\n",
       "  'name': 'adrenoceptor alpha 1A',\n",
       "  'synonyms': 'ADRA1C|ADRA1L1|ALPHA1AAR',\n",
       "  'description': 'Alpha-1-adrenergic receptors (alpha-1-ARs) are members of the G protein-coupled receptor superfamily. They activate mitogenic responses and regulate growth and proliferation of many cells. There are 3 alpha-1-AR subtypes: alpha-1A, -1B and -1D, all of which signal through the Gq/11 family of G-proteins and different subtypes show different patterns of activation. This gene encodes alpha-1A-adrenergic receptor. Alternative splicing of this gene generates four transcript variants, which encode four different isoforms with distinct C-termini but having similar ligand binding properties. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:282',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ADRA2B',\n",
       "  'name': 'adrenoceptor alpha 2B',\n",
       "  'synonyms': 'ADRA2L1|ADRA2RL1|ADRARL1|ALPHA2BAR|FAME2|alpha-2BAR',\n",
       "  'description': 'This intronless gene encodes a seven-pass transmembrane protein. This protein is a member of a subfamily of G protein-coupled receptors that regulate neurotransmitter release from sympathetic nerves and from adrenergic neurons in the central nervous system. [provided by RefSeq, Apr 2014].'},\n",
       " {'id': 'HGNC:285',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ADRB1',\n",
       "  'name': 'adrenoceptor beta 1',\n",
       "  'synonyms': 'ADRB1R|B1AR|BETA1AR|FNSS2|RHR',\n",
       "  'description': 'The adrenergic receptors (subtypes alpha 1, alpha 2, beta 1, and beta 2) are a prototypic family of guanine nucleotide binding regulatory protein-coupled receptors that mediate the physiological effects of the hormone epinephrine and the neurotransmitter norepinephrine. Beta-1 adrenoceptors are predominately located in the heart. Specific polymorphisms in this gene have been shown to affect the resting heart rate and can be involved in heart failure. [provided by RefSeq, Sep 2019].'},\n",
       " {'id': 'HGNC:319',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ACAN',\n",
       "  'name': 'aggrecan',\n",
       "  'synonyms': 'AGC1|AGCAN|CSPG1|CSPGCP|MSK16|SEDK|SSOAOD',\n",
       "  'description': 'This gene is a member of the aggrecan/versican proteoglycan family. The encoded protein is an integral part of the extracellular matrix in cartilagenous tissue and it withstands compression in cartilage. Mutations in this gene may be involved in skeletal dysplasia and spinal degeneration. Multiple alternatively spliced transcript variants that encode different protein isoforms have been observed in this gene. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:333',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'AGT',\n",
       "  'name': 'angiotensinogen',\n",
       "  'synonyms': 'ANHU|SERPINA8|hFLT1',\n",
       "  'description': 'The protein encoded by this gene, pre-angiotensinogen or angiotensinogen precursor, is expressed in the liver and is cleaved by the enzyme renin in response to lowered blood pressure. The resulting product, angiotensin I, is then cleaved by angiotensin converting enzyme (ACE) to generate the physiologically active enzyme angiotensin II. The protein is involved in maintaining blood pressure, body fluid and electrolyte homeostasis, and in the pathogenesis of essential hypertension and preeclampsia. Mutations in this gene are associated with susceptibility to essential hypertension, and can cause renal tubular dysgenesis, a severe disorder of renal tubular development. Defects in this gene have also been associated with non-familial structural atrial fibrillation, and inflammatory bowel disease. [provided by RefSeq, Nov 2019].'},\n",
       " {'id': 'HGNC:336',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'AGTR1',\n",
       "  'name': 'angiotensin II receptor type 1',\n",
       "  'synonyms': 'AG2S|AGTR1B|AT1|AT1AR|AT1B|AT1BR|AT1R|AT2R1|HAT1R',\n",
       "  'description': 'Angiotensin II is a potent vasopressor hormone and a primary regulator of aldosterone secretion. It is an important effector controlling blood pressure and volume in the cardiovascular system. It acts through at least two types of receptors. This gene encodes the type 1 receptor which is thought to mediate the major cardiovascular effects of angiotensin II. This gene may play a role in the generation of reperfusion arrhythmias following restoration of blood flow to ischemic or infarcted myocardium. It was previously thought that a related gene, denoted as AGTR1B, existed; however, it is now believed that there is only one type 1 receptor gene in humans. Alternative splicing of this gene results in multiple transcript variants. [provided by RefSeq, Aug 2020].'},\n",
       " {'id': 'HGNC:341',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'AGXT',\n",
       "  'name': 'alanine--glyoxylate and serine--pyruvate aminotransferase',\n",
       "  'synonyms': 'AGT|AGT1|AGXT1|PH1|SPAT|SPT|TLH6',\n",
       "  'description': 'This gene is expressed only in the liver and the encoded protein is localized mostly in the peroxisomes, where it is involved in glyoxylate detoxification. Mutations in this gene, some of which alter subcellular targetting, have been associated with type I primary hyperoxaluria. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:399',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ALB',\n",
       "  'name': 'albumin',\n",
       "  'synonyms': 'HSA|PRO0883|PRO0903|PRO1341',\n",
       "  'description': 'This gene encodes the most abundant protein in human blood. This protein functions in the regulation of blood plasma colloid osmotic pressure and acts as a carrier protein for a wide range of endogenous molecules including hormones, fatty acids, and metabolites, as well as exogenous drugs. Additionally, this protein exhibits an esterase-like activity with broad substrate specificity. The encoded preproprotein is proteolytically processed to generate the mature protein. A peptide derived from this protein, EPI-X4, is an endogenous inhibitor of the CXCR4 chemokine receptor. [provided by RefSeq, Jul 2016].'},\n",
       " {'id': 'HGNC:405',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ALDH3A1',\n",
       "  'name': 'aldehyde dehydrogenase 3 family member A1',\n",
       "  'synonyms': 'ALDH3|ALDHIII',\n",
       "  'description': 'Aldehyde dehydrogenases oxidize various aldehydes to the corresponding acids. They are involved in the detoxification of alcohol-derived acetaldehyde and in the metabolism of corticosteroids, biogenic amines, neurotransmitters, and lipid peroxidation. The enzyme encoded by this gene forms a cytoplasmic homodimer that preferentially oxidizes aromatic and medium-chain (6 carbons or more) saturated and unsaturated aldehyde substrates. It is thought to promote resistance to UV and 4-hydroxy-2-nonenal-induced oxidative damage in the cornea. The gene is located within the Smith-Magenis syndrome region on chromosome 17. Multiple alternatively spliced variants, encoding the same protein, have been identified. [provided by RefSeq, Sep 2008].'},\n",
       " {'id': 'HGNC:414',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ALDOA',\n",
       "  'name': 'aldolase, fructose-bisphosphate A',\n",
       "  'synonyms': 'ALDA|GSD12|HEL-S-87p',\n",
       "  'description': 'This gene encodes a member of the class I fructose-bisphosphate aldolase protein family. The encoded protein is a glycolytic enzyme that catalyzes the reversible conversion of fructose-1,6-bisphosphate to glyceraldehyde 3-phosphate and dihydroxyacetone phosphate. Three aldolase isozymes (A, B, and C), encoded by three different genes, are differentially expressed during development. Mutations in this gene have been associated with Glycogen Storage Disease XII, an autosomal recessive disorder associated with hemolytic anemia. Disruption of this gene also plays a role in the progression of multiple types of cancers. Related pseudogenes have been identified on chromosomes 3 and 10. [provided by RefSeq, Sep 2017].'},\n",
       " {'id': 'HGNC:381',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'AKR1B1',\n",
       "  'name': 'aldo-keto reductase family 1 member B',\n",
       "  'synonyms': 'ADR|ALDR1|ALR2|AR',\n",
       "  'description': 'This gene encodes a member of the aldo/keto reductase superfamily, which consists of more than 40 known enzymes and proteins. This member catalyzes the reduction of a number of aldehydes, including the aldehyde form of glucose, and is thereby implicated in the development of diabetic complications by catalyzing the reduction of glucose to sorbitol. Multiple pseudogenes have been identified for this gene. The nomenclature system used by the HUGO Gene Nomenclature Committee to define human aldo-keto reductase family members is known to differ from that used by the Mouse Genome Informatics database. [provided by RefSeq, Feb 2009].'},\n",
       " {'id': 'HGNC:435',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ALOX5',\n",
       "  'name': 'arachidonate 5-lipoxygenase',\n",
       "  'synonyms': '5-LO|5-LOX|5LPG|LOG5',\n",
       "  'description': 'This gene encodes a member of the lipoxygenase gene family and plays a dual role in the synthesis of leukotrienes from arachidonic acid. The encoded protein, which is expressed specifically in bone marrow-derived cells, catalyzes the conversion of arachidonic acid to 5(S)-hydroperoxy-6-trans-8,11,14-cis-eicosatetraenoic acid, and further to the allylic epoxide 5(S)-trans-7,9-trans-11,14-cis-eicosatetrenoic acid (leukotriene A4). Leukotrienes are important mediators of a number of inflammatory and allergic conditions. Mutations in the promoter region of this gene lead to a diminished response to antileukotriene drugs used in the treatment of asthma and may also be associated with atherosclerosis and several cancers. Alternatively spliced transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Jan 2012].'},\n",
       " {'id': 'HGNC:434',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ALOX15B',\n",
       "  'name': 'arachidonate 15-lipoxygenase type B',\n",
       "  'synonyms': '15-LOX-2',\n",
       "  'description': 'This gene encodes a member of the lipoxygenase family of structurally related nonheme iron dioxygenases involved in the production of fatty acid hydroperoxides. The encoded protein converts arachidonic acid exclusively to 15S-hydroperoxyeicosatetraenoic acid, while metabolizing linoleic acid less effectively. This gene is located in a cluster of related genes and a pseudogene that spans approximately 100 kilobases on the short arm of chromosome 17. Alternatively spliced transcript variants encoding different isoforms have been described. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:437',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ALPI',\n",
       "  'name': 'alkaline phosphatase, intestinal',\n",
       "  'synonyms': 'IAP',\n",
       "  'description': 'There are at least four distinct but related alkaline phosphatases: intestinal, placental, placental-like, and liver/bone/kidney (tissue non-specific). The intestinal alkaline phosphatase gene encodes a digestive brush-border enzyme. This enzyme is a component of the gut mucosal defense system and is thought to function in the detoxification of lipopolysaccharide, and in the prevention of bacterial translocation in the gut. [provided by RefSeq, Dec 2014].'},\n",
       " {'id': 'HGNC:464',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'AMH',\n",
       "  'name': 'anti-Mullerian hormone',\n",
       "  'synonyms': 'MIF|MIS',\n",
       "  'description': 'This gene encodes a secreted ligand of the TGF-beta (transforming growth factor-beta) superfamily of proteins. Ligands of this family bind various TGF-beta receptors leading to recruitment and activation of SMAD family transcription factors that regulate gene expression. The encoded preproprotein is proteolytically processed to generate N- and C-terminal cleavage products that homodimerize and associate to form a biologically active noncovalent complex. This complex binds to the anti-Mullerian hormone receptor type 2 and causes the regression of Mullerian ducts in the male embryo that would otherwise differentiate into the uterus and fallopian tubes. This protein also plays a role in Leydig cell differentiation and function and follicular development in adult females. Mutations in this gene result in persistent Mullerian duct syndrome. [provided by RefSeq, Jul 2016].'},\n",
       " {'id': 'HGNC:553',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'AOX1',\n",
       "  'name': 'aldehyde oxidase 1',\n",
       "  'synonyms': 'AO|AOH1',\n",
       "  'description': 'Aldehyde oxidase produces hydrogen peroxide and, under certain conditions, can catalyze the formation of superoxide. Aldehyde oxidase is a candidate gene for amyotrophic lateral sclerosis. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:581',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'APBB1',\n",
       "  'name': 'amyloid beta precursor protein binding family B member 1',\n",
       "  'synonyms': 'FE65|MGC:9072|RIR',\n",
       "  'description': \"The protein encoded by this gene is a member of the Fe65 protein family. It is an adaptor protein localized in the nucleus. It interacts with the Alzheimer's disease amyloid precursor protein (APP), transcription factor CP2/LSF/LBP1 and the low-density lipoprotein receptor-related protein. APP functions as a cytosolic anchoring site that can prevent the gene product's nuclear translocation. This encoded protein could play an important role in the pathogenesis of Alzheimer's disease. It is thought to regulate transcription. Also it is observed to block cell cycle progression by downregulating thymidylate synthase expression. Multiple alternatively spliced transcript variants encoding different isoforms have been described for this gene. [provided by RefSeq, Mar 2012].\"},\n",
       " {'id': 'HGNC:587',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'APEX1',\n",
       "  'name': 'apurinic/apyrimidinic endodeoxyribonuclease 1',\n",
       "  'synonyms': 'APE|APE1|APEN|APEX|APX|HAP1|REF1',\n",
       "  'description': 'The APEX gene encodes the major AP endonuclease in human cells. It encodes the APEX endonuclease, a DNA repair enzyme with apurinic/apyrimidinic (AP) activity. Such AP activity sites occur frequently in DNA molecules by spontaneous hydrolysis, by DNA damaging agents or by DNA glycosylases that remove specific abnormal bases. The AP sites are the most frequent pre-mutagenic lesions that can prevent normal DNA replication. Splice variants have been found for this gene; all encode the same protein. Disruptions in the biological functions related to APEX are associated with many various malignancies and neurodegenerative diseases.[provided by RefSeq, Dec 2019].'},\n",
       " {'id': 'HGNC:644',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'AR',\n",
       "  'name': 'androgen receptor',\n",
       "  'synonyms': 'AIS|AR8|DHTR|HUMARA|HYSP1|KD|NR3C4|SBMA|SMAX1|TFM',\n",
       "  'description': \"The androgen receptor gene is more than 90 kb long and codes for a protein that has 3 major functional domains: the N-terminal domain, DNA-binding domain, and androgen-binding domain. The protein functions as a steroid-hormone activated transcription factor. Upon binding the hormone ligand, the receptor dissociates from accessory proteins, translocates into the nucleus, dimerizes, and then stimulates transcription of androgen responsive genes. This gene contains 2 polymorphic trinucleotide repeat segments that encode polyglutamine and polyglycine tracts in the N-terminal transactivation domain of its protein. Expansion of the polyglutamine tract from the normal 9-34 repeats to the pathogenic 38-62 repeats causes spinal bulbar muscular atrophy (SBMA, also known as Kennedy's disease). Mutations in this gene are also associated with complete androgen insensitivity (CAIS). Alternative splicing results in multiple transcript variants encoding different isoforms. [provided by RefSeq, Jan 2017].\"},\n",
       " {'id': 'HGNC:651',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'AREG',\n",
       "  'name': 'amphiregulin',\n",
       "  'synonyms': 'AR|AREGB|CRDGF|SDGF',\n",
       "  'description': 'The protein encoded by this gene is a member of the epidermal growth factor family. It is an autocrine growth factor as well as a mitogen for astrocytes, Schwann cells and fibroblasts. It is related to epidermal growth factor (EGF) and transforming growth factor alpha (TGF-alpha). The protein interacts with the EGF/TGF-alpha receptor to promote the growth of normal epithelial cells, and it inhibits the growth of certain aggressive carcinoma cell lines. It also functions in mammary gland, oocyte and bone tissue development. This gene is associated with a psoriasis-like skin phenotype, and is also associated with other pathological disorders, including various types of cancers and inflammatory conditions. [provided by RefSeq, Apr 2014].'},\n",
       " {'id': 'HGNC:801',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'ATP1A3',\n",
       "  'name': 'ATPase Na+/K+ transporting subunit alpha 3',\n",
       "  'synonyms': 'AHC2|ATP1A1|CAPOS|DYT12|RDP',\n",
       "  'description': 'The protein encoded by this gene belongs to the family of P-type cation transport ATPases, and to the subfamily of Na+/K+ -ATPases. Na+/K+ -ATPase is an integral membrane protein responsible for establishing and maintaining the electrochemical gradients of Na and K ions across the plasma membrane. These gradients are essential for osmoregulation, for sodium-coupled transport of a variety of organic and inorganic molecules, and for electrical excitability of nerve and muscle. This enzyme is composed of two subunits, a large catalytic subunit (alpha) and a smaller glycoprotein subunit (beta). The catalytic subunit of Na+/K+ -ATPase is encoded by multiple genes. This gene encodes an alpha 3 subunit. Alternatively spliced transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Jan 2012].'},\n",
       " {'id': 'HGNC:959',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'BAX',\n",
       "  'name': 'BCL2 associated X, apoptosis regulator',\n",
       "  'synonyms': 'BCL2L4',\n",
       "  'description': 'The protein encoded by this gene belongs to the BCL2 protein family. BCL2 family members form hetero- or homodimers and act as anti- or pro-apoptotic regulators that are involved in a wide variety of cellular activities. This protein forms a heterodimer with BCL2, and functions as an apoptotic activator. The association and the ratio of BAX to BCL2 also determines survival or death of a cell following an apoptotic stimulus. This protein is reported to interact with, and increase the opening of, the mitochondrial voltage-dependent anion channel (VDAC), which leads to the loss in membrane potential and the release of cytochrome c. The expression of this gene is regulated by the tumor suppressor P53 and has been shown to be involved in P53-mediated apoptosis. Multiple alternatively spliced transcript variants, which encode different isoforms, have been reported for this gene. [provided by RefSeq, Dec 2019].'},\n",
       " {'id': 'HGNC:990',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'BCL2',\n",
       "  'name': 'BCL2 apoptosis regulator',\n",
       "  'synonyms': 'Bcl-2|PPP1R50',\n",
       "  'description': 'This gene encodes an integral outer mitochondrial membrane protein that blocks the apoptotic death of some cells such as lymphocytes. Constitutive expression of BCL2, such as in the case of translocation of BCL2 to Ig heavy chain locus, is thought to be the cause of follicular lymphoma. Alternative splicing results in multiple transcript variants. [provided by RefSeq, Feb 2016].'},\n",
       " {'id': 'HGNC:1030',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'BDKRB2',\n",
       "  'name': 'bradykinin receptor B2',\n",
       "  'synonyms': 'B2R|BK-2|BK2|BKR2|BRB2',\n",
       "  'description': 'This gene encodes a receptor for bradykinin. The 9 aa bradykinin peptide elicits many responses including vasodilation, edema, smooth muscle spasm and pain fiber stimulation. Bradykinin is released upon activation by pathophysiologic conditions such as trauma and inflammation, and binds to its kinin receptors, B1 and B2. The B2 receptor associates with G proteins that stimulate a phosphatidylinositol-calcium second messenger system. [provided by RefSeq, Apr 2020].'},\n",
       " {'id': 'HGNC:1118',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'BST1',\n",
       "  'name': 'bone marrow stromal cell antigen 1',\n",
       "  'synonyms': 'CD157',\n",
       "  'description': 'Bone marrow stromal cell antigen-1 is a stromal cell line-derived glycosylphosphatidylinositol-anchored molecule that facilitates pre-B-cell growth. The deduced amino acid sequence exhibits 33% similarity with CD38. BST1 expression is enhanced in bone marrow stromal cell lines derived from patients with rheumatoid arthritis. The polyclonal B-cell abnormalities in rheumatoid arthritis may be, at least in part, attributed to BST1 overexpression in the stromal cell population. [provided by RefSeq, Jul 2008].'},\n",
       " {'id': 'HGNC:1133',\n",
       "  'semantic_groups': 'GENE',\n",
       "  'preflabel': 'BTK',\n",
       "  'name': 'Bruton tyrosine kinase',\n",
       "  'synonyms': 'AGMX1|AT|ATK|BPK|IGHD3|IMD1|PSCTK1|XLA',\n",
       "  'description': 'The protein encoded by this gene plays a crucial role in B-cell development. Mutations in this gene cause X-linked agammaglobulinemia type 1, which is an immunodeficiency characterized by the failure to produce mature B lymphocytes, and associated with a failure of Ig heavy chain rearrangement. Alternative splicing results in multiple transcript variants encoding different isoforms. [provided by RefSeq, Dec 2013].'},\n",
       " ...]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "monarch_nodes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subject_id</th>\n",
       "      <th>property_id</th>\n",
       "      <th>object_id</th>\n",
       "      <th>reference_uri</th>\n",
       "      <th>reference_supporting_text</th>\n",
       "      <th>reference_date</th>\n",
       "      <th>property_label</th>\n",
       "      <th>property_description</th>\n",
       "      <th>property_uri</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   subject_id  property_id  object_id  reference_uri  \\\n",
       "0         NaN          NaN        NaN            NaN   \n",
       "\n",
       "   reference_supporting_text  reference_date  property_label  \\\n",
       "0                        NaN             NaN             NaN   \n",
       "\n",
       "   property_description  property_uri  \n",
       "0                   NaN           NaN  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "curation_edges"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subject_id</th>\n",
       "      <th>object_id</th>\n",
       "      <th>property_id</th>\n",
       "      <th>property_label</th>\n",
       "      <th>property_description</th>\n",
       "      <th>property_uri</th>\n",
       "      <th>reference_uri</th>\n",
       "      <th>reference_supporting_text</th>\n",
       "      <th>reference_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RGD:1595815</td>\n",
       "      <td>MGI:892004</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>NA</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSEMBL:ENSRNOG00000061639</td>\n",
       "      <td>MGI:3704134</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>NA</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FlyBase:FBgn0041191</td>\n",
       "      <td>MGI:97912</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>NA</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000020</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MGI:1344381</td>\n",
       "      <td>ENSEMBL:ENSOANG00000006132</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>NA</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000020</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSEMBL:ENSCAFG00000017647</td>\n",
       "      <td>ENSEMBL:ENSECAG00000019077</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>NA</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000020</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233177</th>\n",
       "      <td>RGD:1311415</td>\n",
       "      <td>ZFIN:ZDB-GENE-030131-7310</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>NA</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000020</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233178</th>\n",
       "      <td>dictyBase:DDB_G0271282</td>\n",
       "      <td>ZFIN:ZDB-GENE-030131-8680</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>NA</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233179</th>\n",
       "      <td>ZFIN:ZDB-GENE-041008-78</td>\n",
       "      <td>UBERON:0000465</td>\n",
       "      <td>RO:0002206</td>\n",
       "      <td>expressed in</td>\n",
       "      <td>NA</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002206</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233180</th>\n",
       "      <td>ENSEMBL:ENSECAG00000035314</td>\n",
       "      <td>MGI:95915</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>NA</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233181</th>\n",
       "      <td>ENSEMBL:ENSXETG00000034352</td>\n",
       "      <td>ENSEMBL:ENSECAG00000035532</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>NA</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "      <td>NA</td>\n",
       "      <td>This edge comes from the Monarch Knowledge Gra...</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>233182 rows Ã 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        subject_id                   object_id    property_id  \\\n",
       "0                      RGD:1595815                  MGI:892004  RO:HOM0000017   \n",
       "1       ENSEMBL:ENSRNOG00000061639                 MGI:3704134  RO:HOM0000017   \n",
       "2              FlyBase:FBgn0041191                   MGI:97912  RO:HOM0000020   \n",
       "3                      MGI:1344381  ENSEMBL:ENSOANG00000006132  RO:HOM0000020   \n",
       "4       ENSEMBL:ENSCAFG00000017647  ENSEMBL:ENSECAG00000019077  RO:HOM0000020   \n",
       "...                            ...                         ...            ...   \n",
       "233177                 RGD:1311415   ZFIN:ZDB-GENE-030131-7310  RO:HOM0000020   \n",
       "233178      dictyBase:DDB_G0271282   ZFIN:ZDB-GENE-030131-8680  RO:HOM0000017   \n",
       "233179     ZFIN:ZDB-GENE-041008-78              UBERON:0000465     RO:0002206   \n",
       "233180  ENSEMBL:ENSECAG00000035314                   MGI:95915  RO:HOM0000017   \n",
       "233181  ENSEMBL:ENSXETG00000034352  ENSEMBL:ENSECAG00000035532  RO:HOM0000017   \n",
       "\n",
       "                               property_label property_description  \\\n",
       "0              in orthology relationship with                   NA   \n",
       "1              in orthology relationship with                   NA   \n",
       "2       in 1 to 1 orthology relationship with                   NA   \n",
       "3       in 1 to 1 orthology relationship with                   NA   \n",
       "4       in 1 to 1 orthology relationship with                   NA   \n",
       "...                                       ...                  ...   \n",
       "233177  in 1 to 1 orthology relationship with                   NA   \n",
       "233178         in orthology relationship with                   NA   \n",
       "233179                           expressed in                   NA   \n",
       "233180         in orthology relationship with                   NA   \n",
       "233181         in orthology relationship with                   NA   \n",
       "\n",
       "                                        property_uri reference_uri  \\\n",
       "0       http://purl.obolibrary.org/obo/RO_HOM0000017            NA   \n",
       "1       http://purl.obolibrary.org/obo/RO_HOM0000017            NA   \n",
       "2       http://purl.obolibrary.org/obo/RO_HOM0000020            NA   \n",
       "3       http://purl.obolibrary.org/obo/RO_HOM0000020            NA   \n",
       "4       http://purl.obolibrary.org/obo/RO_HOM0000020            NA   \n",
       "...                                              ...           ...   \n",
       "233177  http://purl.obolibrary.org/obo/RO_HOM0000020            NA   \n",
       "233178  http://purl.obolibrary.org/obo/RO_HOM0000017            NA   \n",
       "233179     http://purl.obolibrary.org/obo/RO_0002206            NA   \n",
       "233180  http://purl.obolibrary.org/obo/RO_HOM0000017            NA   \n",
       "233181  http://purl.obolibrary.org/obo/RO_HOM0000017            NA   \n",
       "\n",
       "                                reference_supporting_text reference_date  \n",
       "0       This edge comes from the Monarch Knowledge Gra...             NA  \n",
       "1       This edge comes from the Monarch Knowledge Gra...             NA  \n",
       "2       This edge comes from the Monarch Knowledge Gra...             NA  \n",
       "3       This edge comes from the Monarch Knowledge Gra...             NA  \n",
       "4       This edge comes from the Monarch Knowledge Gra...             NA  \n",
       "...                                                   ...            ...  \n",
       "233177  This edge comes from the Monarch Knowledge Gra...             NA  \n",
       "233178  This edge comes from the Monarch Knowledge Gra...             NA  \n",
       "233179  This edge comes from the Monarch Knowledge Gra...             NA  \n",
       "233180  This edge comes from the Monarch Knowledge Gra...             NA  \n",
       "233181  This edge comes from the Monarch Knowledge Gra...             NA  \n",
       "\n",
       "[233182 rows x 9 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "monarch_edges"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "reg_edges"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "The function \"graph_nodes()\" is running...\n",
      "\n",
      "Preparing networks...\n",
      "Curated:\n",
      "(1, 9)\n",
      "Index(['subject_id', 'property_id', 'object_id', 'reference_uri',\n",
      "       'reference_supporting_text', 'reference_date', 'property_label',\n",
      "       'property_description', 'property_uri'],\n",
      "      dtype='object')\n",
      "Monarch:\n",
      "(233182, 9)\n",
      "Index(['subject_id', 'object_id', 'property_id', 'property_label',\n",
      "       'property_description', 'property_uri', 'reference_uri',\n",
      "       'reference_supporting_text', 'reference_date'],\n",
      "      dtype='object')\n",
      "Transcriptomics:\n",
      "(3209, 9)\n",
      "Index(['subject_id', 'object_id', 'property_id', 'property_label',\n",
      "       'property_description', 'property_uri', 'reference_uri',\n",
      "       'reference_supporting_text', 'reference_date'],\n",
      "      dtype='object')\n",
      "Regulatory:\n",
      "(197267, 9)\n",
      "Index(['subject_id', 'object_id', 'property_id', 'property_label',\n",
      "       'property_description', 'property_uri', 'reference_uri',\n",
      "       'reference_supporting_text', 'reference_date'],\n",
      "      dtype='object')\n",
      "\n",
      "Concatenating into a graph...\n",
      "(236392, 9)\n",
      "\n",
      "Drop duplicated rows...\n",
      "(236392, 9)\n",
      "\n",
      "Merging tf-gene network to the graph...\n",
      "(14164, 9)\n",
      "\n",
      "Saving tf merged edges...\n",
      "\n",
      "The regulation graph merged edges are saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/graph/regulation_graph_edges_v2021-04-12.csv\n",
      "\n",
      "(250556, 9)\n",
      "\n",
      "Drop duplicated rows...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  0%|          | 0/22997 [00:00<?, ?it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(250556, 9)\n",
      "\n",
      "Generating graph nodes...\n",
      "(22997, 1)\n",
      "\n",
      "Finished graph_nodes().\n",
      "\n",
      "\n",
      "The function \"extract_edges()\" is running. Its runtime may take some hours. If you interrupt the process, you will lose all the edges retrieved and you should start over the execution of this function.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|ââââââââââ| 22997/22997 [29:01:03<00:00,  4.54s/it]   \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Finished extract_edges(). To save the retrieved Monarch edges use the function \"print_network()\".\n",
      "\n",
      "network:  575849\n",
      "\n",
      "Saving Monarch edges at: '/home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/monarch/monarch_connections_graph_v2021-04-12.csv'...\n",
      "\n",
      "\n",
      "The function \"build_edges()\" is running...\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=4010, subject_id='HGNC:7675', subject_label='NDN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0002808', object_label='Kyphosis', reference_id_list='NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=6735, subject_id='HGNC:9585', subject_label='PTCH1', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000518', object_label='Cataract', reference_id_list='MONDO:0007187|NCBIBSgene:bcns')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=8381, subject_id='HGNC:7675', subject_label='NDN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0001290', object_label='Generalized hypotonia', reference_id_list='NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=9902, subject_id='HGNC:11164', subject_label='SNRPN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000028', object_label='Cryptorchidism', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=21433, subject_id='HGNC:11164', subject_label='SNRPN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0002650', object_label='Scoliosis', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=29956, subject_id='HGNC:7675', subject_label='NDN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000708', object_label='Behavioral abnormality', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=31898, subject_id='HGNC:9585', subject_label='PTCH1', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0010442', object_label='Polydactyly', reference_id_list='NCBIBSgene:bcns')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=41824, subject_id='HGNC:7675', subject_label='NDN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000939', object_label='Osteoporosis', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=47129, subject_id='HGNC:7675', subject_label='NDN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0002650', object_label='Scoliosis', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=60277, subject_id='HGNC:9586', subject_label='PTCH2', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000486', object_label='Strabismus', reference_id_list='MONDO:0007187|NCBIBSgene:bcns')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=66929, subject_id='HGNC:7675', subject_label='NDN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0002360', object_label='Sleep disturbance', reference_id_list='NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=75093, subject_id='HGNC:6694', subject_label='LRP2', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0001537', object_label='Umbilical hernia', reference_id_list='NCBIBSgene:donnai|MONDO:0009104')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=84803, subject_id='HGNC:6694', subject_label='LRP2', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0005280', object_label='Depressed nasal bridge', reference_id_list='NCBIBSgene:donnai|MONDO:0009104')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=94985, subject_id='HGNC:7675', subject_label='NDN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000028', object_label='Cryptorchidism', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=95706, subject_id='HGNC:11164', subject_label='SNRPN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000486', object_label='Strabismus', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=96753, subject_id='HGNC:9834', subject_label='RAI1', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000750', object_label='Delayed speech and language development', reference_id_list='MONDO:0008434|NCBIBSgene:sms')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=101283, subject_id='HGNC:9585', subject_label='PTCH1', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000568', object_label='Microphthalmia', reference_id_list='NCBIBSgene:bcns')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=109770, subject_id='HGNC:11164', subject_label='SNRPN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0001290', object_label='Generalized hypotonia', reference_id_list='NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=110238, subject_id='HGNC:7675', subject_label='NDN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0002591', object_label='Polyphagia', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=111168, subject_id='HGNC:9834', subject_label='RAI1', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0030680', object_label='Abnormality of cardiovascular system morphology', reference_id_list='MONDO:0008434|NCBIBSgene:sms')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=112207, subject_id='HGNC:9834', subject_label='RAI1', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0002155', object_label='Hypertriglyceridemia', reference_id_list='MONDO:0008434|NCBIBSgene:sms')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=129304, subject_id='HGNC:16466', subject_label='SUFU', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0010442', object_label='Polydactyly', reference_id_list='NCBIBSgene:bcns')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=135192, subject_id='HGNC:11164', subject_label='SNRPN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000939', object_label='Osteoporosis', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=138206, subject_id='HGNC:9834', subject_label='RAI1', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0001156', object_label='Brachydactyly', reference_id_list='MONDO:0008434|NCBIBSgene:sms')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=142231, subject_id='HGNC:9834', subject_label='RAI1', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000272', object_label='Malar flattening', reference_id_list='NCBIBSgene:sms')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=157732, subject_id='HGNC:9834', subject_label='RAI1', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0002715', object_label='Abnormality of the immune system', reference_id_list='NCBIBSgene:sms')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=165715, subject_id='HGNC:9834', subject_label='RAI1', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000248', object_label='Brachycephaly', reference_id_list='MONDO:0008434|NCBIBSgene:sms')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=171719, subject_id='HGNC:11164', subject_label='SNRPN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0002119', object_label='Ventriculomegaly', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=178153, subject_id='HGNC:16466', subject_label='SUFU', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000486', object_label='Strabismus', reference_id_list='MONDO:0007187|PMID:19619433|NCBIBSgene:bcns')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=182075, subject_id='HGNC:9834', subject_label='RAI1', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000490', object_label='Deeply set eye', reference_id_list='MONDO:0008434|NCBIBSgene:sms')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=185870, subject_id='HGNC:11164', subject_label='SNRPN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0001263', object_label='Global developmental delay', reference_id_list='NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=189398, subject_id='HGNC:16466', subject_label='SUFU', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000238', object_label='Hydrocephalus', reference_id_list='MONDO:0016642|MONDO:0007187|MONDO:0007959|NCBIBSgene:bcns')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=191952, subject_id='HGNC:16466', subject_label='SUFU', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0002414', object_label='Spina bifida', reference_id_list='NCBIBSgene:bcns')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=195293, subject_id='HGNC:9586', subject_label='PTCH2', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000175', object_label='Cleft palate', reference_id_list='NCBIBSgene:bcns')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=200415, subject_id='HGNC:11164', subject_label='SNRPN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000545', object_label='Myopia', reference_id_list='NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=219264, subject_id='HGNC:11164', subject_label='SNRPN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000708', object_label='Behavioral abnormality', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=225641, subject_id='HGNC:9834', subject_label='RAI1', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000315', object_label='Abnormality of the orbital region', reference_id_list='NCBIBSgene:sms')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=236643, subject_id='HGNC:7675', subject_label='NDN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0004279', object_label='Short palm', reference_id_list='NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=241515, subject_id='HGNC:7675', subject_label='NDN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0001263', object_label='Global developmental delay', reference_id_list='NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=250563, subject_id='HGNC:11164', subject_label='SNRPN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0004322', object_label='Short stature', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=254917, subject_id='HGNC:9586', subject_label='PTCH2', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0002414', object_label='Spina bifida', reference_id_list='NCBIBSgene:bcns')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=266116, subject_id='HGNC:11164', subject_label='SNRPN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000938', object_label='Osteopenia', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=279316, subject_id='HGNC:9834', subject_label='RAI1', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0004279', object_label='Short palm', reference_id_list='NCBIBSgene:sms')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=294652, subject_id='HGNC:7675', subject_label='NDN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0001773', object_label='Short foot', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=295055, subject_id='HGNC:9585', subject_label='PTCH1', relation_id='RO:0003304', relation_label='contributes to condition', object_id='HP:0000175', object_label='Cleft palate', reference_id_list='MONDO:0019757|MONDO:0019758|MONDO:0019756|PMID:28232668|MONDO:0016355|MONDO:0017219|NCBIBSgene:bcns')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=298007, subject_id='HGNC:6694', subject_label='LRP2', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000520', object_label='Proptosis', reference_id_list='NCBIBSgene:donnai|MONDO:0009104')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=298160, subject_id='HGNC:7675', subject_label='NDN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0001250', object_label='Seizure', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=308196, subject_id='HGNC:9585', subject_label='PTCH1', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000486', object_label='Strabismus', reference_id_list='MONDO:0007187|MONDO:0017219|NCBIBSgene:bcns')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=326742, subject_id='HGNC:7675', subject_label='NDN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000750', object_label='Delayed speech and language development', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=330710, subject_id='HGNC:11164', subject_label='SNRPN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0004279', object_label='Short palm', reference_id_list='NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=333428, subject_id='HGNC:16466', subject_label='SUFU', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000175', object_label='Cleft palate', reference_id_list='PMID:19619433|NCBIBSgene:bcns')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=343505, subject_id='HGNC:7675', subject_label='NDN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000938', object_label='Osteopenia', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=361842, subject_id='HGNC:9586', subject_label='PTCH2', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000238', object_label='Hydrocephalus', reference_id_list='MONDO:0007187|MONDO:0007959|NCBIBSgene:bcns')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=371649, subject_id='HGNC:7675', subject_label='NDN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0001513', object_label='Obesity', reference_id_list='NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=381163, subject_id='HGNC:7675', subject_label='NDN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0008872', object_label='Feeding difficulties in infancy', reference_id_list='NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=391709, subject_id='HGNC:11164', subject_label='SNRPN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000054', object_label='Micropenis', reference_id_list='NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=391970, subject_id='HGNC:11164', subject_label='SNRPN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0001250', object_label='Seizure', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=396571, subject_id='HGNC:11164', subject_label='SNRPN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0001773', object_label='Short foot', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=402986, subject_id='HGNC:9586', subject_label='PTCH2', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0010442', object_label='Polydactyly', reference_id_list='NCBIBSgene:bcns')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=406490, subject_id='HGNC:7675', subject_label='NDN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000717', object_label='Autism', reference_id_list='NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=413651, subject_id='HGNC:11164', subject_label='SNRPN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0002360', object_label='Sleep disturbance', reference_id_list='NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=420602, subject_id='HGNC:11164', subject_label='SNRPN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000750', object_label='Delayed speech and language development', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=423824, subject_id='HGNC:9834', subject_label='RAI1', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0002119', object_label='Ventriculomegaly', reference_id_list='MONDO:0008434|NCBIBSgene:sms')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=427032, subject_id='HGNC:6694', subject_label='LRP2', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0001250', object_label='Seizure', reference_id_list='NCBIBSgene:donnai|MONDO:0009104')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=434830, subject_id='HGNC:7675', subject_label='NDN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000545', object_label='Myopia', reference_id_list='NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=435847, subject_id='HGNC:11164', subject_label='SNRPN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000789', object_label='Infertility', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=440500, subject_id='HGNC:16466', subject_label='SUFU', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000518', object_label='Cataract', reference_id_list='MONDO:0007187|NCBIBSgene:bcns')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=443364, subject_id='HGNC:11164', subject_label='SNRPN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0002591', object_label='Polyphagia', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=454917, subject_id='HGNC:7675', subject_label='NDN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000054', object_label='Micropenis', reference_id_list='NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=457520, subject_id='HGNC:7675', subject_label='NDN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0002205', object_label='Recurrent respiratory infections', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=458393, subject_id='HGNC:6694', subject_label='LRP2', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000541', object_label='Retinal detachment', reference_id_list='NCBIBSgene:donnai|MONDO:0009104')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=458906, subject_id='HGNC:9585', subject_label='PTCH1', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0002414', object_label='Spina bifida', reference_id_list='NCBIBSgene:bcns')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=465537, subject_id='HGNC:7675', subject_label='NDN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0007018', object_label='Attention deficit hyperactivity disorder', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=471065, subject_id='HGNC:7675', subject_label='NDN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0001385', object_label='Hip dysplasia', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=472469, subject_id='HGNC:9586', subject_label='PTCH2', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000568', object_label='Microphthalmia', reference_id_list='NCBIBSgene:bcns')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=482942, subject_id='HGNC:11164', subject_label='SNRPN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0008872', object_label='Feeding difficulties in infancy', reference_id_list='NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=487639, subject_id='HGNC:9585', subject_label='PTCH1', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000238', object_label='Hydrocephalus', reference_id_list='MONDO:0019757|MONDO:0019758|MONDO:0019756|MONDO:0016355|MONDO:0007187|PMID:11941477|NCBIBSgene:bcns')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=493242, subject_id='HGNC:7675', subject_label='NDN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0004322', object_label='Short stature', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=502520, subject_id='HGNC:7675', subject_label='NDN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0002119', object_label='Ventriculomegaly', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=505672, subject_id='HGNC:9586', subject_label='PTCH2', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000518', object_label='Cataract', reference_id_list='MONDO:0007187|NCBIBSgene:bcns')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=509231, subject_id='HGNC:9834', subject_label='RAI1', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0001290', object_label='Generalized hypotonia', reference_id_list='NCBIBSgene:sms')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=509689, subject_id='HGNC:6694', subject_label='LRP2', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0001629', object_label='Ventricular septal defect', reference_id_list='NCBIBSgene:donnai|MONDO:0009104')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=510215, subject_id='HGNC:4319', subject_label='GLI3', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000107', object_label='Renal cyst', reference_id_list='NCBIBSgene:phs')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=513509, subject_id='HGNC:7675', subject_label='NDN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000789', object_label='Infertility', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=521982, subject_id='HGNC:16466', subject_label='SUFU', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000568', object_label='Microphthalmia', reference_id_list='NCBIBSgene:bcns')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=524236, subject_id='HGNC:11164', subject_label='SNRPN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0002808', object_label='Kyphosis', reference_id_list='NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=526554, subject_id='HGNC:9834', subject_label='RAI1', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000541', object_label='Retinal detachment', reference_id_list='MONDO:0008434|NCBIBSgene:sms')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=532195, subject_id='HGNC:9834', subject_label='RAI1', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000733', object_label='Stereotypy', reference_id_list='MONDO:0008434|PMID:17539903|NCBIBSgene:sms')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=532709, subject_id='HGNC:11164', subject_label='SNRPN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000717', object_label='Autism', reference_id_list='NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=543024, subject_id='HGNC:583', subject_label='APC', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000164', object_label='Abnormality of the dentition', reference_id_list='PMID:8929955|PMID:14633595|PMID:24735542|PMID:20223039|PMID:17293347|PMID:8395941|PMID:20685668|PMID:9824584|PMID:9101302|PMID:17411426|PMID:27158207|PMID:12173026|PMID:15108286|PMID:9452101|PMID:23159591|PMID:24664542|PMID:11852337|PMID:8162022|PMID:9890479|PMID:25980754|PMID:1651563|PMID:17963004|PMID:28533537|PMID:12007223|PMID:26681312|PMID:1316610|PMID:25590978|PMID:1338764|PMID:20434453|PMID:15459959|PMID:25317407|PMID:8940264|PMID:23460355|PMID:27081525|PMID:8381579|PMID:14522379|PMID:8956059|PMID:18433509|PMID:21110124|PMID:11247896|PMID:22135120|PMID:10094547|PMID:15311282|PMID:23561487|PMID:8381580|PMID:8381581|PMID:16317745|PMID:7906810|PMID:26511139|PMID:15771908|MONDO:0019336|PMID:10090483|PMID:18794146|PMID:24599579|PMID:20924072|PMID:27087319|PMID:19029688|PMID:7746201|PMID:16461775|PMID:12172928|PMID:12034871|PMID:9487968|PMID:19444466|PMID:11317365|PMID:12486240|PMID:16088911|PMID:8162051|NCBIBSgene:fap|PMID:26613750|PMID:21779980|PMID:8281160')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=549411, subject_id='HGNC:9834', subject_label='RAI1', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000708', object_label='Behavioral abnormality', reference_id_list='NCBIBSgene:sms')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=554500, subject_id='HGNC:11164', subject_label='SNRPN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0007018', object_label='Attention deficit hyperactivity disorder', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=560222, subject_id='HGNC:11164', subject_label='SNRPN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0002205', object_label='Recurrent respiratory infections', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=562001, subject_id='HGNC:11164', subject_label='SNRPN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0001513', object_label='Obesity', reference_id_list='NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=565730, subject_id='HGNC:11164', subject_label='SNRPN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0001385', object_label='Hip dysplasia', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "Warning:\n",
      "Detected a new reference source in Monarch not yet implemented in this module. The new source should be added to the dictionary of sources.Otherwise, the source CURIE cannot be translated to the corresponding URI.\n",
      "In the build_edges() method, update 'uriPrefixes_dct' dictionary with 'NCBIBSgene'\n",
      "The edge that includes this new reference source is Pandas(Index=570014, subject_id='HGNC:7675', subject_label='NDN', relation_id='RO:0002200', relation_label='has phenotype', object_id='HP:0000486', object_label='Strabismus', reference_id_list='MONDO:0008300|NCBIBSgene:pws')\n",
      "The method will continue to run without problem, writing the CURIE instead of the URI,until the dictionary is updated.\n",
      "df (575849, 9)\n",
      "\n",
      "* This is the size of the edges file data structure: (575849, 9)\n",
      "* These are the edges attributes: Index(['subject_id', 'object_id', 'property_id', 'property_label',\n",
      "       'property_description', 'property_uri', 'reference_uri',\n",
      "       'reference_supporting_text', 'reference_date'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "   subject_id            object_id    property_id  \\\n",
      "0  HGNC:23796  FlyBase:FBgn0037481  RO:HOM0000017   \n",
      "\n",
      "                   property_label property_description  \\\n",
      "0  in orthology relationship with                   NA   \n",
      "\n",
      "                                   property_uri reference_uri  \\\n",
      "0  http://purl.obolibrary.org/obo/RO_HOM0000017            NA   \n",
      "\n",
      "                           reference_supporting_text reference_date  \n",
      "0  This edge comes from the Monarch Knowledge Gra...             NA  \n",
      "\n",
      "The Monarch network edges are built and saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/monarch/monarch_edges_v2021-04-12.csv\n",
      "\n",
      "\n",
      "Finished build_edges().\n",
      "\n",
      "\n",
      "The function \"build_nodes()\" is running...\n",
      "Number of concepts: 18492\n",
      "Number of nodes CURIEs: 30\n",
      "List of nodes CURIEs: dict_keys(['HGNC', 'FlyBase', 'GO', 'MONDO', 'HP', 'ENSEMBL', 'WormBase', 'ZFIN', 'MGI', 'UBERON', 'ZP', 'EFO', 'RGD', 'EnsemblGenome', 'SGD', 'MONARCH', 'PomBase', 'dictyBase', 'WBPhenotype', 'MP', 'FBcv', 'Xenbase', 'NCBIGene', 'Coriell', 'dbSNP', 'REACT', 'APO', 'ClinVarVariant', 'dbSNPIndividual', 'OMIM'])\n",
      "\n",
      "Adding BioThings annotation: gene name, synonyms, description...\n",
      "symbols: 12154\n",
      "querying 1-1000...done.\n",
      "querying 1001-2000...done.\n",
      "querying 2001-3000...done.\n",
      "querying 3001-4000...done.\n",
      "querying 4001-5000...done.\n",
      "querying 5001-6000...done.\n",
      "querying 6001-7000...done.\n",
      "querying 7001-8000...done.\n",
      "querying 8001-9000...done.\n",
      "querying 9001-10000...done.\n",
      "querying 10001-11000...done.\n",
      "querying 11001-12000...done.\n",
      "querying 12001-12154...done.\n",
      "Finished.\n",
      "691 input query terms found dup hits:\n",
      "\t[('VIRMA', 3), ('EEF1D', 4), ('KIF9', 3), ('HTT', 3), ('ACTR3', 3), ('MICA', 3), ('HMGA1', 4), ('ZFC\n",
      "3336 input query terms found no hit:\n",
      "\t['ENSEMBL:ENSACAG00000005403', 'ENSEMBL:ENSMODG00000014624', 'ENSEMBL:ENSMODG00000017882', 'CAT (hum\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "* This is the size of the nodes file data structure: (18492, 6)\n",
      "* These are the nodes attributes: Index(['id', 'semantic_groups', 'preflabel', 'name', 'synonyms',\n",
      "       'description'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "           id semantic_groups preflabel                   name  \\\n",
      "0  HGNC:23796            GENE    MAGEB6  MAGE family member B6   \n",
      "\n",
      "                synonyms                                        description  \n",
      "0  CT3.4|MAGE-B6|MAGEB6A  This gene is a member of the MAGEB gene family...  \n",
      "\n",
      "The Monarch network nodes are built and saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/monarch/monarch_nodes_v2021-04-12.csv\n",
      "\n",
      "\n",
      "Finished build_nodes().\n",
      "\n",
      "\n",
      "The function \"build_edges()\" is running...\n",
      "\n",
      "Preparing networks...\n",
      "Curated:\n",
      "(1, 9)\n",
      "Index(['subject_id', 'property_id', 'object_id', 'reference_uri',\n",
      "       'reference_supporting_text', 'reference_date', 'property_label',\n",
      "       'property_description', 'property_uri'],\n",
      "      dtype='object')\n",
      "Monarch:\n",
      "(575849, 9)\n",
      "Index(['subject_id', 'object_id', 'property_id', 'property_label',\n",
      "       'property_description', 'property_uri', 'reference_uri',\n",
      "       'reference_supporting_text', 'reference_date'],\n",
      "      dtype='object')\n",
      "Transcriptomics:\n",
      "(3209, 9)\n",
      "Index(['subject_id', 'object_id', 'property_id', 'property_label',\n",
      "       'property_description', 'property_uri', 'reference_uri',\n",
      "       'reference_supporting_text', 'reference_date'],\n",
      "      dtype='object')\n",
      "Regulatory:\n",
      "(14164, 9)\n",
      "Index(['subject_id', 'property_id', 'object_id', 'reference_uri',\n",
      "       'reference_supporting_text', 'reference_date', 'property_label',\n",
      "       'property_description', 'property_uri'],\n",
      "      dtype='object')\n",
      "\n",
      "Concatenating into a graph...\n",
      "(593223, 9)\n",
      "\n",
      "Drop duplicated rows...\n",
      "(593223, 9)\n",
      "\n",
      "Saving final graph...\n",
      "(593223, 9)\n",
      "Index(['subject_id', 'property_id', 'object_id', 'reference_uri',\n",
      "       'reference_supporting_text', 'reference_date', 'property_label',\n",
      "       'property_description', 'property_uri'],\n",
      "      dtype='object')\n",
      "\n",
      "* This is the size of the edges file data structure: (593223, 9)\n",
      "* These are the edges attributes: Index(['subject_id', 'property_id', 'object_id', 'reference_uri',\n",
      "       'reference_supporting_text', 'reference_date', 'property_label',\n",
      "       'property_description', 'property_uri'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "  subject_id property_id object_id reference_uri reference_supporting_text  \\\n",
      "0        NaN         NaN       NaN           NaN                       NaN   \n",
      "\n",
      "  reference_date property_label property_description property_uri  \n",
      "0            NaN            NaN                  NaN         None  \n",
      "\n",
      "The NGLY1 Deficiency knowledge graph edges are built and saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/graph/graph_edges_v2021-04-12.csv\n",
      "\n",
      "\n",
      "Finished build_edges().\n",
      "\n",
      "\n",
      "The function \"build_nodes()\" is running...\n",
      "\n",
      "Preparing networks...\n",
      "Curated:\n",
      "(1, 6)\n",
      "Index(['id', 'semantic_groups', 'preflabel', 'synonyms', 'description',\n",
      "       'name'],\n",
      "      dtype='object')\n",
      "Monarch:\n",
      "(18492, 6)\n",
      "Index(['id', 'semantic_groups', 'preflabel', 'name', 'synonyms',\n",
      "       'description'],\n",
      "      dtype='object')\n",
      "Transcriptomics:\n",
      "(3210, 6)\n",
      "Index(['id', 'semantic_groups', 'preflabel', 'name', 'synonyms',\n",
      "       'description'],\n",
      "      dtype='object')\n",
      "Regulatory:\n",
      "(16968, 6)\n",
      "Index(['id', 'semantic_groups', 'preflabel', 'name', 'synonyms',\n",
      "       'description'],\n",
      "      dtype='object')\n",
      "\n",
      "Annotating nodes in the graph...\n",
      "graph from e (21951, 1)\n",
      "annotation check\n",
      "curated (1, 6)\n",
      "monarch (18492, 6)\n",
      "rna (1645, 6)\n",
      "regulation (6109, 6)\n",
      "\n",
      "Concatenating all nodes...\n",
      "graph ann (26247, 6)\n",
      "diff {nan, 'ensembl:ENSG00000135736', 'ensembl:ENSG00000146250', 'ensembl:ENSG00000197057', 'ensembl:ENSG00000188313', 'ensembl:ENSG00000203747', 'ensembl:ENSG00000158186', 'ensembl:ENSG00000172554', 'ensembl:ENSG00000122691', 'ensembl:ENSG00000248485', 'ensembl:ENSG00000187634', 'ensembl:ENSG00000166292', 'ensembl:ENSG00000169562', 'ensembl:ENSG00000164850', 'ensembl:ENSG00000125398', 'ensembl:ENSG00000107249', 'ensembl:ENSG00000105383', 'ensembl:ENSG00000186891', 'ensembl:ENSG00000011422', 'ensembl:ENSG00000166743', 'ensembl:ENSG00000116678', 'ensembl:ENSG00000163131', 'ensembl:ENSG00000174885', 'ensembl:ENSG00000141469', 'ensembl:ENSG00000082074', 'ensembl:ENSG00000162998', 'ensembl:ENSG00000138600', 'ensembl:ENSG00000120093', 'ensembl:ENSG00000105483', 'ensembl:ENSG00000136872', 'ensembl:ENSG00000115165', 'ensembl:ENSG00000100234', 'ensembl:ENSG00000270181', 'ensembl:ENSG00000064666', 'ensembl:ENSG00000102760', 'ensembl:ENSG00000243955', 'ensembl:ENSG00000171236', 'ensembl:ENSG00000106333', 'ensembl:ENSG00000257815', 'ensembl:ENSG00000205903', 'ensembl:ENSG00000241697', 'ensembl:ENSG00000002933', 'ensembl:ENSG00000196431', 'ensembl:ENSG00000171345', 'ensembl:ENSG00000120903', 'ensembl:ENSG00000105329', 'ensembl:ENSG00000187193', 'ensembl:ENSG00000102007', 'ensembl:ENSG00000132744', 'ensembl:ENSG00000106560', 'ensembl:ENSG00000174776', 'ensembl:ENSG00000140839', 'ensembl:ENSG00000125780', 'ensembl:ENSG00000238184', 'ensembl:ENSG00000145832', 'ensembl:ENSG00000121742', 'ensembl:ENSG00000137462', 'ensembl:ENSG00000128645', 'ensembl:ENSG00000146278', 'ensembl:ENSG00000166111', 'ensembl:ENSG00000058335', 'ensembl:ENSG00000123975', 'ensembl:ENSG00000115194', 'ensembl:ENSG00000116701', 'ensembl:ENSG00000258673', 'ensembl:ENSG00000088836', 'ensembl:ENSG00000185432', 'ensembl:ENSG00000153246', 'ensembl:ENSG00000100985', 'ensembl:ENSG00000204261', 'ensembl:ENSG00000182533', 'ensembl:ENSG00000145721', 'ensembl:ENSG00000000938', 'ensembl:ENSG00000188747', 'ensembl:ENSG00000111348', 'ensembl:ENSG00000130748', 'ensembl:ENSG00000151715', 'ensembl:ENSG00000225924', 'ensembl:ENSG00000134569', 'ensembl:ENSG00000121552', 'ensembl:ENSG00000162598', 'ensembl:ENSG00000152229', 'ensembl:ENSG00000204516', 'ensembl:ENSG00000177462', 'ensembl:ENSG00000148604', 'ensembl:ENSG00000243364', 'ensembl:ENSG00000089356', 'ensembl:ENSG00000101276', 'ensembl:ENSG00000100368', 'ensembl:ENSG00000081842', 'ensembl:ENSG00000166803', 'ensembl:ENSG00000113088', 'ensembl:ENSG00000128482', 'ensembl:ENSG00000155749', 'ensembl:ENSG00000142319', 'ensembl:ENSG00000157087', 'ensembl:ENSG00000129951', 'ensembl:ENSG00000129680', 'ensembl:ENSG00000105641', 'ensembl:ENSG00000264920', 'ensembl:ENSG00000122367', 'ensembl:ENSG00000145779', 'ensembl:ENSG00000137509', 'ensembl:ENSG00000100344', 'ensembl:ENSG00000196990', 'ensembl:ENSG00000108511', 'ensembl:ENSG00000255398', 'ensembl:ENSG00000145882', 'ensembl:ENSG00000112837', 'ensembl:ENSG00000172824', 'ensembl:ENSG00000230358', 'ensembl:ENSG00000150048', 'ensembl:ENSG00000166535', 'ensembl:ENSG00000150337', 'ensembl:ENSG00000172731', 'ensembl:ENSG00000258099', 'ensembl:ENSG00000135378', 'ensembl:ENSG00000167747', 'ensembl:ENSG00000160307', 'ensembl:ENSG00000259583', 'ensembl:ENSG00000240032', 'ensembl:ENSG00000137507', 'ensembl:ENSG00000224792', 'ensembl:ENSG00000204577', 'ensembl:ENSG00000152661', 'ensembl:ENSG00000138346', 'ensembl:ENSG00000163346', 'ensembl:ENSG00000134532', 'ensembl:ENSG00000259291', 'ensembl:ENSG00000124374', 'ensembl:ENSG00000179111', 'ensembl:ENSG00000198754', 'ensembl:ENSG00000188820', 'ensembl:ENSG00000165168', 'ensembl:ENSG00000225873', 'ensembl:ENSG00000169403', 'ensembl:ENSG00000065675', 'ensembl:ENSG00000230316', 'ensembl:ENSG00000257335', 'ensembl:ENSG00000121039', 'ensembl:ENSG00000149573', 'ensembl:ENSG00000144476', 'ensembl:ENSG00000197430', 'ensembl:ENSG00000148498', 'ensembl:ENSG00000140986', 'ensembl:ENSG00000021645', 'ensembl:ENSG00000157680', 'ensembl:ENSG00000145358', 'ensembl:ENSG00000168679', 'ensembl:ENSG00000147872', 'ensembl:ENSG00000124253', 'ensembl:ENSG00000185885', 'ensembl:ENSG00000163737', 'ensembl:ENSG00000255568', 'ensembl:ENSG00000250295', 'ensembl:ENSG00000198723', 'ensembl:ENSG00000162618', 'ensembl:ENSG00000135838', 'ensembl:ENSG00000231389', 'ensembl:ENSG00000008394', 'ensembl:ENSG00000100276', 'ensembl:ENSG00000142233', 'ensembl:ENSG00000181016', 'ensembl:ENSG00000140832', 'ensembl:ENSG00000131095', 'ensembl:ENSG00000129667', 'ensembl:ENSG00000186951', 'ensembl:ENSG00000081985', 'ensembl:ENSG00000080546', 'ensembl:ENSG00000260630', 'ensembl:ENSG00000151615', 'ensembl:ENSG00000236753', 'ensembl:ENSG00000260231', 'ensembl:ENSG00000163132', 'ensembl:ENSG00000120063', 'ensembl:ENSG00000115041', 'ensembl:ENSG00000171791', 'ensembl:ENSG00000143546', 'ensembl:ENSG00000112812', 'ensembl:ENSG00000113946', 'ensembl:ENSG00000114204', 'ensembl:ENSG00000105246', 'ensembl:ENSG00000115155', 'ensembl:ENSG00000267530', 'ensembl:ENSG00000143552', 'ensembl:ENSG00000155269', 'ensembl:ENSG00000160781', 'ensembl:ENSG00000181449', 'ensembl:ENSG00000235505', 'ensembl:ENSG00000131386', 'ensembl:ENSG00000171940', 'ensembl:ENSG00000119950', 'ensembl:ENSG00000169252', 'ensembl:ENSG00000125810', 'ensembl:ENSG00000180660', 'ensembl:ENSG00000067955', 'ensembl:ENSG00000159588', 'ensembl:ENSG00000176105', 'ensembl:ENSG00000233521', 'ensembl:ENSG00000127993', 'ensembl:ENSG00000164107', 'ensembl:ENSG00000258483', 'ensembl:ENSG00000129675', 'ensembl:ENSG00000163873', 'ensembl:ENSG00000139832', 'ensembl:ENSG00000122870', 'ensembl:ENSG00000173040', 'ensembl:ENSG00000139567', 'ensembl:ENSG00000124469', 'ensembl:ENSG00000162078', 'ensembl:ENSG00000164294', 'ensembl:ENSG00000204287', 'ensembl:ENSG00000119655', 'ensembl:ENSG00000167771', 'ensembl:ENSG00000113594', 'ensembl:ENSG00000106823', 'ensembl:ENSG00000113658', 'ensembl:ENSG00000105967', 'ensembl:ENSG00000185022', 'ensembl:ENSG00000073282', 'ensembl:ENSG00000268001', 'ensembl:ENSG00000172247', 'ensembl:ENSG00000176890', 'ensembl:ENSG00000152192', 'ensembl:ENSG00000129988', 'ensembl:ENSG00000125257', 'ensembl:ENSG00000175130', 'ensembl:ENSG00000135902', 'ensembl:ENSG00000163840', 'ensembl:ENSG00000110203', 'ensembl:ENSG00000169840', 'ensembl:ENSG00000146233', 'ensembl:ENSG00000188211', 'ensembl:ENSG00000011304', 'ensembl:ENSG00000168874', 'ensembl:ENSG00000184608', 'ensembl:ENSG00000248810', 'ensembl:ENSG00000236423', 'ensembl:ENSG00000135077', 'ensembl:ENSG00000101187', 'ensembl:ENSG00000156219', 'ensembl:ENSG00000237686', 'ensembl:ENSG00000125753', 'ensembl:ENSG00000180720', 'ensembl:ENSG00000267909', 'ensembl:ENSG00000150201', 'ensembl:ENSG00000205835', 'ensembl:ENSG00000155850', 'ensembl:ENSG00000198208', 'ensembl:ENSG00000141232', 'ensembl:ENSG00000228544', 'ensembl:ENSG00000081320', 'ensembl:ENSG00000241360', 'ensembl:ENSG00000159403', 'ensembl:ENSG00000081248', 'ensembl:ENSG00000173376', 'ensembl:ENSG00000152518', 'ensembl:ENSG00000164692', 'ensembl:ENSG00000197769', 'ensembl:ENSG00000137267', 'ensembl:ENSG00000157851', 'ensembl:ENSG00000104760', 'ensembl:ENSG00000146648', 'ensembl:ENSG00000204388', 'ensembl:ENSG00000177721', 'ensembl:ENSG00000170962', 'ensembl:ENSG00000168675', 'ensembl:ENSG00000163121', 'ensembl:ENSG00000163659', 'ensembl:ENSG00000090382', 'ensembl:ENSG00000175077', 'ensembl:ENSG00000072840', 'ensembl:ENSG00000187730', 'ensembl:ENSG00000125459', 'ensembl:ENSG00000105467', 'ensembl:ENSG00000136518', 'ensembl:ENSG00000113196', 'ensembl:ENSG00000150907', 'ensembl:ENSG00000188257', 'ensembl:ENSG00000173530', 'ensembl:ENSG00000030419', 'ensembl:ENSG00000120889', 'ensembl:ENSG00000129151', 'ensembl:ENSG00000138039', 'ensembl:ENSG00000087903', 'ensembl:ENSG00000185507', 'ensembl:ENSG00000173548', 'ensembl:ENSG00000172575', 'ensembl:ENSG00000113916', 'ensembl:ENSG00000142698', 'ensembl:ENSG00000228126', 'ensembl:ENSG00000011600', 'ensembl:ENSG00000002745', 'ensembl:ENSG00000170476', 'ensembl:ENSG00000168079', 'ensembl:ENSG00000123374', 'ensembl:ENSG00000248455', 'ensembl:ENSG00000144909', 'ensembl:ENSG00000155629', 'ensembl:ENSG00000158525', 'ensembl:ENSG00000134901', 'ensembl:ENSG00000156966', 'ensembl:ENSG00000186472', 'ensembl:ENSG00000033327', 'ensembl:ENSG00000135439', 'ensembl:ENSG00000181577', 'ensembl:ENSG00000137767', 'ensembl:ENSG00000106538', 'ensembl:ENSG00000126259', 'ensembl:ENSG00000224717', 'ensembl:ENSG00000235919', 'ensembl:ENSG00000100433', 'ensembl:ENSG00000235387', 'ensembl:ENSG00000227456', 'ensembl:ENSG00000126804', 'ensembl:ENSG00000183250', 'ensembl:ENSG00000107738', 'ensembl:ENSG00000261879', 'ensembl:ENSG00000086730', 'ensembl:ENSG00000102575', 'ensembl:ENSG00000234928', 'ensembl:ENSG00000188269', 'ensembl:ENSG00000069702', 'ensembl:ENSG00000185933', 'ensembl:ENSG00000167552', 'ensembl:ENSG00000088826', 'ensembl:ENSG00000108786', 'ensembl:ENSG00000177426', 'ensembl:ENSG00000073331', 'ensembl:ENSG00000177212', 'ensembl:ENSG00000135111', 'ensembl:ENSG00000181773', 'ensembl:ENSG00000104524', 'ensembl:ENSG00000082497', 'ensembl:ENSG00000134762', 'ensembl:ENSG00000152463', 'ensembl:ENSG00000117091', 'ensembl:ENSG00000016391', 'ensembl:ENSG00000178860', 'ensembl:ENSG00000069493', 'ensembl:ENSG00000164849', 'ensembl:ENSG00000253696', 'ensembl:ENSG00000144339', 'ensembl:ENSG00000243477', 'ensembl:ENSG00000138794', 'ensembl:ENSG00000104881', 'ensembl:ENSG00000120645', 'ensembl:ENSG00000187513', 'ensembl:ENSG00000134986', 'ensembl:ENSG00000188404', 'ensembl:ENSG00000113732', 'ensembl:ENSG00000164867', 'ensembl:ENSG00000177551', 'ensembl:ENSG00000074219', 'ensembl:ENSG00000135063', 'ensembl:ENSG00000243811', 'ensembl:ENSG00000170689', 'ensembl:ENSG00000171695', 'ensembl:ENSG00000116584', 'ensembl:ENSG00000111783', 'ensembl:ENSG00000102554', 'ensembl:ENSG00000160255', 'ensembl:ENSG00000053918', 'ensembl:ENSG00000253741', 'ensembl:ENSG00000214866', 'ensembl:ENSG00000121769', 'ensembl:ENSG00000163406', 'ensembl:ENSG00000214226', 'ensembl:ENSG00000139350', 'ensembl:ENSG00000177363', 'ensembl:ENSG00000114812', 'ensembl:ENSG00000204347', 'ensembl:ENSG00000008853', 'ensembl:ENSG00000131097', 'ensembl:ENSG00000132256', 'ensembl:ENSG00000111319', 'ensembl:ENSG00000114315', 'ensembl:ENSG00000134115', 'ensembl:ENSG00000176472', 'ensembl:ENSG00000101292', 'ensembl:ENSG00000167613', 'ensembl:ENSG00000089123', 'ensembl:ENSG00000051523', 'ensembl:ENSG00000084093', 'ensembl:ENSG00000159212', 'ensembl:ENSG00000105695', 'ensembl:ENSG00000260260', 'ensembl:ENSG00000197646', 'ensembl:ENSG00000119681', 'ensembl:ENSG00000185811', 'ensembl:ENSG00000160867', 'ensembl:ENSG00000115604', 'ensembl:ENSG00000177627', 'ensembl:ENSG00000201839', 'ensembl:ENSG00000173905', 'ensembl:ENSG00000182308', 'ensembl:ENSG00000198156', 'ensembl:ENSG00000096060', 'ensembl:ENSG00000198157', 'ensembl:ENSG00000163221', 'ensembl:ENSG00000165125', 'ensembl:ENSG00000116774', 'ensembl:ENSG00000176584', 'ensembl:ENSG00000069011', 'ensembl:ENSG00000142606', 'ensembl:ENSG00000215162', 'ensembl:ENSG00000203711', 'ensembl:ENSG00000179058', 'ensembl:ENSG00000148655', 'ensembl:ENSG00000254585', 'ensembl:ENSG00000235665', 'ensembl:ENSG00000165457', 'ensembl:ENSG00000129990', 'ensembl:ENSG00000149201', 'ensembl:ENSG00000112299', 'ensembl:ENSG00000160223', 'ensembl:ENSG00000179073', 'ensembl:ENSG00000113212', 'ensembl:ENSG00000136274', 'ensembl:ENSG00000267795', 'ensembl:ENSG00000132793', 'ensembl:ENSG00000155380', 'ensembl:ENSG00000115598', 'ensembl:ENSG00000183765', 'ensembl:ENSG00000211574', 'ensembl:ENSG00000090659', 'ensembl:ENSG00000164188', 'ensembl:ENSG00000124664', 'ensembl:ENSG00000164111', 'ensembl:ENSG00000214787', 'ensembl:ENSG00000137757', 'ensembl:ENSG00000089163', 'ensembl:ENSG00000046604', 'ensembl:ENSG00000167641', 'ensembl:ENSG00000224975', 'ensembl:ENSG00000152034', 'ensembl:ENSG00000178462', 'ensembl:ENSG00000198740', 'ensembl:ENSG00000186439', 'ensembl:ENSG00000056487', 'ensembl:ENSG00000175287', 'ensembl:ENSG00000175445', 'ensembl:ENSG00000006327', 'ensembl:ENSG00000237928', 'ensembl:ENSG00000183873', 'ensembl:ENSG00000077942', 'ensembl:ENSG00000102034', 'ensembl:ENSG00000112378', 'ensembl:ENSG00000221039', 'ensembl:ENSG00000076716', 'ensembl:ENSG00000160791', 'ensembl:ENSG00000244482', 'ensembl:ENSG00000241769', 'ensembl:ENSG00000183688', 'ensembl:ENSG00000262686', 'ensembl:ENSG00000188162', 'ensembl:ENSG00000099625', 'ensembl:ENSG00000145423', 'ensembl:ENSG00000260027', 'ensembl:ENSG00000102755', 'ensembl:ENSG00000113739', 'ensembl:ENSG00000154589', 'ensembl:ENSG00000148926', 'ensembl:ENSG00000173369', 'ensembl:ENSG00000111344', 'ensembl:ENSG00000175489', 'ensembl:ENSG00000042062', 'ensembl:ENSG00000138606', 'ensembl:ENSG00000197993', 'ensembl:ENSG00000101460', 'ensembl:ENSG00000134575', 'ensembl:ENSG00000204396', 'ensembl:ENSG00000142513', 'ensembl:ENSG00000126264', 'ensembl:ENSG00000215274', 'ensembl:ENSG00000175895', 'ensembl:ENSG00000126266', 'ensembl:ENSG00000137463', 'ensembl:ENSG00000226761', 'ensembl:ENSG00000122122', 'ensembl:ENSG00000215174', 'ensembl:ENSG00000187554', 'ensembl:ENSG00000111666', 'ensembl:ENSG00000104140', 'ensembl:ENSG00000181885', 'ensembl:ENSG00000188488', 'ensembl:ENSG00000130755', 'ensembl:ENSG00000198576', 'ensembl:ENSG00000255399', 'ensembl:ENSG00000163736', 'ensembl:ENSG00000258947', 'ensembl:ENSG00000100012', 'ensembl:ENSG00000112175', 'ensembl:ENSG00000115596', 'ensembl:ENSG00000143545', 'ensembl:ENSG00000185112', 'ensembl:ENSG00000180155', 'ensembl:ENSG00000123595', 'ensembl:ENSG00000243480', 'ensembl:ENSG00000157005', 'ensembl:ENSG00000170577', 'ensembl:ENSG00000106809', 'ensembl:ENSG00000131401', 'ensembl:ENSG00000151014', 'ensembl:ENSG00000144583', 'ensembl:ENSG00000101347', 'ensembl:ENSG00000003989', 'ensembl:ENSG00000018408', 'ensembl:ENSG00000170425', 'ensembl:ENSG00000163154', 'ensembl:ENSG00000133800', 'ensembl:ENSG00000178695', 'ensembl:ENSG00000197043', 'ensembl:ENSG00000164949', 'ensembl:ENSG00000180573', 'ensembl:ENSG00000145002', 'ensembl:ENSG00000139269', 'ensembl:ENSG00000165970', 'ensembl:ENSG00000118523', 'ensembl:ENSG00000203685', 'ensembl:ENSG00000131042', 'ensembl:ENSG00000168938', 'ensembl:ENSG00000224743', 'ensembl:ENSG00000131797', 'ensembl:ENSG00000015285', 'ensembl:ENSG00000120280', 'ensembl:ENSG00000076248', 'ensembl:ENSG00000178789', 'ensembl:ENSG00000184005', 'ensembl:ENSG00000148175', 'ensembl:ENSG00000231367', 'ensembl:ENSG00000105122', 'ensembl:ENSG00000162909', 'ensembl:ENSG00000101306', 'ensembl:ENSG00000173281', 'ensembl:ENSG00000112312', 'ensembl:ENSG00000175544', 'ensembl:ENSG00000205129', 'ensembl:ENSG00000176853', 'ensembl:ENSG00000230156', 'ensembl:ENSG00000243232', 'ensembl:ENSG00000127377', 'ensembl:ENSG00000198429', 'ensembl:ENSG00000142627', 'ensembl:ENSG00000125772', 'ensembl:ENSG00000167114', 'ensembl:ENSG00000028137', 'ensembl:ENSG00000140092', 'ensembl:ENSG00000183160', 'ensembl:ENSG00000187595', 'ensembl:ENSG00000205436', 'ensembl:ENSG00000196735', 'ensembl:ENSG00000123338', 'ensembl:ENSG00000205913', 'ensembl:ENSG00000160396', 'ensembl:ENSG00000108798', 'ensembl:ENSG00000177181', 'ensembl:ENSG00000063660', 'ensembl:ENSG00000142347', 'ensembl:ENSG00000164181', 'ensembl:ENSG00000086062', 'ensembl:ENSG00000067182', 'ensembl:ENSG00000074047', 'ensembl:ENSG00000094963', 'ensembl:ENSG00000186417', 'ensembl:ENSG00000164039', 'ensembl:ENSG00000168333', 'ensembl:ENSG00000055208', 'ensembl:ENSG00000132002', 'ensembl:ENSG00000198842', 'ensembl:ENSG00000176597', 'ensembl:ENSG00000120594', 'ensembl:ENSG00000171657', 'ensembl:ENSG00000100365', 'ensembl:ENSG00000166682', 'ensembl:ENSG00000185664', 'ensembl:ENSG00000166527', 'ensembl:ENSG00000163734', 'ensembl:ENSG00000213949', 'ensembl:ENSG00000164318', 'ensembl:ENSG00000266405', 'ensembl:ENSG00000173917', 'ensembl:ENSG00000106089', 'ensembl:ENSG00000121807', 'ensembl:ENSG00000145287', 'ensembl:ENSG00000163421', 'ensembl:ENSG00000100156', 'ensembl:ENSG00000142512', 'ensembl:ENSG00000163629', 'ensembl:ENSG00000063015', 'ensembl:ENSG00000175352', 'ensembl:ENSG00000204290', 'ensembl:ENSG00000143839', 'ensembl:ENSG00000170290', 'ensembl:ENSG00000169994', 'ensembl:ENSG00000214300', 'ensembl:ENSG00000118491', 'ensembl:ENSG00000008128', 'ensembl:ENSG00000132965', 'ensembl:ENSG00000221819', 'ensembl:ENSG00000197291', 'ensembl:ENSG00000224557', 'ensembl:ENSG00000254126', 'ensembl:ENSG00000042980', 'ensembl:ENSG00000167895', 'ensembl:ENSG00000205038', 'ensembl:ENSG00000100852', 'ensembl:ENSG00000174125', 'ensembl:ENSG00000163739', 'ensembl:ENSG00000158201', 'ensembl:ENSG00000152952', 'ensembl:ENSG00000153162', 'ensembl:ENSG00000095970', 'ensembl:ENSG00000242258', 'ensembl:ENSG00000090376', 'ensembl:ENSG00000168453', 'ensembl:ENSG00000115474', 'ensembl:ENSG00000061455', 'ensembl:ENSG00000196154', 'ensembl:ENSG00000231789', 'ensembl:ENSG00000164283', 'ensembl:ENSG00000109743', 'ensembl:ENSG00000145777', 'ensembl:ENSG00000205517', 'ensembl:ENSG00000082293', 'ensembl:ENSG00000101017', 'ensembl:ENSG00000126262', 'ensembl:ENSG00000168993', 'ensembl:ENSG00000089558', 'ensembl:ENSG00000181215', 'ensembl:ENSG00000179292', 'ensembl:ENSG00000182791', 'ensembl:ENSG00000165030', 'ensembl:ENSG00000184486', 'ensembl:ENSG00000151224', 'ensembl:ENSG00000239282', 'ensembl:ENSG00000269235', 'ensembl:ENSG00000162975', 'ensembl:ENSG00000149654', 'ensembl:ENSG00000001617', 'ensembl:ENSG00000196139', 'ensembl:ENSG00000230102', 'ensembl:ENSG00000125731', 'ensembl:ENSG00000155719', 'ensembl:ENSG00000232859', 'ensembl:ENSG00000198670', 'ensembl:ENSG00000123636', 'ensembl:ENSG00000184500', 'ensembl:ENSG00000167105', 'ensembl:ENSG00000180176', 'ensembl:ENSG00000197889', 'ensembl:ENSG00000175463', 'ensembl:ENSG00000173156', 'ensembl:ENSG00000185915', 'ensembl:ENSG00000267060', 'ensembl:ENSG00000105499', 'ensembl:ENSG00000131668', 'ensembl:ENSG00000126778', 'ensembl:ENSG00000163285', 'ensembl:ENSG00000124491', 'ensembl:ENSG00000184384', 'ensembl:ENSG00000095585', 'ensembl:ENSG00000162496', 'ensembl:ENSG00000107779', 'ensembl:ENSG00000163513', 'ensembl:ENSG00000119900', 'ensembl:ENSG00000270164', 'ensembl:ENSG00000231742', 'ensembl:ENSG00000250033', 'ensembl:ENSG00000204936', 'ensembl:ENSG00000187116', 'ensembl:ENSG00000118503', 'ensembl:ENSG00000019991', 'ensembl:ENSG00000149257', 'ensembl:ENSG00000137198', 'ensembl:ENSG00000184911', 'ensembl:ENSG00000170989', 'ensembl:ENSG00000254815', 'ensembl:ENSG00000163380', 'ensembl:ENSG00000129654', 'ensembl:ENSG00000196666', 'ensembl:ENSG00000147027', 'ensembl:ENSG00000183963', 'ensembl:ENSG00000081237', 'ensembl:ENSG00000156273', 'ensembl:ENSG00000165023', 'ensembl:ENSG00000103742', 'ensembl:ENSG00000006606', 'ensembl:ENSG00000155087', 'ensembl:ENSG00000099860', 'ensembl:ENSG00000166295', 'ensembl:ENSG00000185499', 'ensembl:ENSG00000169508', 'ensembl:ENSG00000153208', 'ensembl:ENSG00000129514', 'ensembl:ENSG00000155659', 'ensembl:ENSG00000124429', 'ensembl:ENSG00000268996', 'ensembl:ENSG00000136997', 'ensembl:ENSG00000145246', 'ensembl:ENSG00000166523', 'ensembl:ENSG00000183255', 'ensembl:ENSG00000112414', 'ensembl:ENSG00000204257', 'ensembl:ENSG00000153551', 'ensembl:ENSG00000198930', 'ensembl:ENSG00000011465', 'ensembl:ENSG00000151491', 'ensembl:ENSG00000077150', 'ensembl:ENSG00000143199', 'ensembl:ENSG00000125733', 'ensembl:ENSG00000186081', 'ensembl:ENSG00000164620', 'ensembl:ENSG00000115009', 'ensembl:ENSG00000170214', 'ensembl:ENSG00000137976', 'ensembl:ENSG00000142449', 'ensembl:ENSG00000112333', 'ensembl:ENSG00000147257', 'ensembl:ENSG00000105143', 'ensembl:ENSG00000125885', 'ensembl:ENSG00000105851', 'ensembl:ENSG00000099365', 'ensembl:ENSG00000230872', 'ensembl:ENSG00000122585', 'ensembl:ENSG00000108622', 'ensembl:ENSG00000078804', 'ensembl:ENSG00000168843', 'ensembl:ENSG00000107438', 'ensembl:ENSG00000182379', 'ensembl:ENSG00000231768', 'ensembl:ENSG00000071967', 'ensembl:ENSG00000121966', 'ensembl:ENSG00000038945', 'ensembl:ENSG00000137124', 'ensembl:ENSG00000249471', 'ensembl:ENSG00000144644', 'ensembl:ENSG00000134755', 'ensembl:ENSG00000166825', 'ensembl:ENSG00000158008', 'ensembl:ENSG00000133256', 'ensembl:ENSG00000198682', 'ensembl:ENSG00000188157', 'ensembl:ENSG00000180871', 'ensembl:ENSG00000158517', 'ensembl:ENSG00000198604', 'ensembl:ENSG00000099282', 'ensembl:ENSG00000103089', 'ensembl:ENSG00000235618', 'ensembl:ENSG00000136943', 'ensembl:ENSG00000120088', 'ensembl:ENSG00000116815', 'ensembl:ENSG00000131242', 'ensembl:ENSG00000157227', 'ensembl:ENSG00000254440', 'ensembl:ENSG00000148482', 'ensembl:ENSG00000258735', 'ensembl:ENSG00000114638', 'ensembl:ENSG00000159399', 'ensembl:ENSG00000070731', 'ensembl:ENSG00000169213', 'ensembl:ENSG00000204248', 'ensembl:ENSG00000115325', 'ensembl:ENSG00000258227', 'ensembl:ENSG00000171659', 'ensembl:ENSG00000138379', 'ensembl:ENSG00000197471', 'ensembl:ENSG00000196865', 'ensembl:ENSG00000160868', 'ensembl:ENSG00000183814', 'ensembl:ENSG00000172782', 'ensembl:ENSG00000106565', 'ensembl:ENSG00000174137', 'ensembl:ENSG00000151012', 'ensembl:ENSG00000116741', 'ensembl:ENSG00000233930', 'ensembl:ENSG00000198353', 'ensembl:ENSG00000234056', 'ensembl:ENSG00000115290', 'ensembl:ENSG00000242902', 'ensembl:ENSG00000214954', 'ensembl:ENSG00000198835', 'ensembl:ENSG00000158258', 'ensembl:ENSG00000140675', 'ensembl:ENSG00000180767', 'ensembl:ENSG00000224843', 'ensembl:ENSG00000133328', 'ensembl:ENSG00000164707', 'ensembl:ENSG00000137269', 'ensembl:ENSG00000079150', 'ensembl:ENSG00000169413', 'ensembl:ENSG00000156097', 'ensembl:ENSG00000059588', 'ensembl:ENSG00000135114', 'ensembl:ENSG00000213694', 'ensembl:ENSG00000105854', 'ensembl:ENSG00000260001', 'ensembl:ENSG00000179528', 'ensembl:ENSG00000102218', 'ensembl:ENSG00000168060', 'ensembl:ENSG00000164199', 'ensembl:ENSG00000101425', 'ensembl:ENSG00000207611', 'ensembl:ENSG00000096696', 'ensembl:ENSG00000153930', 'ensembl:ENSG00000128585', 'ensembl:ENSG00000118160', 'ensembl:ENSG00000205116', 'ensembl:ENSG00000137693', 'ensembl:ENSG00000182648', 'ensembl:ENSG00000150782', 'ensembl:ENSG00000197766', 'ensembl:ENSG00000170442', 'ensembl:ENSG00000137225', 'ensembl:ENSG00000214688', 'ensembl:ENSG00000103888', 'ensembl:ENSG00000026103', 'ensembl:ENSG00000250337', 'ensembl:ENSG00000249396', 'ensembl:ENSG00000102854', 'ensembl:ENSG00000204420', 'ensembl:ENSG00000138615', 'ensembl:ENSG00000178602', 'ensembl:ENSG00000184292', 'ensembl:ENSG00000104313', 'ensembl:ENSG00000177191', 'ensembl:ENSG00000173714', 'ensembl:ENSG00000169583', 'ensembl:ENSG00000163884', 'ensembl:ENSG00000269028', 'ensembl:ENSG00000187122', 'ensembl:ENSG00000112936', 'ensembl:ENSG00000201388', 'ensembl:ENSG00000203809', 'ensembl:ENSG00000154175', 'ensembl:ENSG00000064201', 'ensembl:ENSG00000257838', 'ensembl:ENSG00000180818', 'ensembl:ENSG00000138075', 'ensembl:ENSG00000178826', 'ensembl:ENSG00000148357', 'ensembl:ENSG00000237943', 'ensembl:ENSG00000143627', 'ensembl:ENSG00000260034', 'ensembl:ENSG00000247774', 'ensembl:ENSG00000135749', 'ensembl:ENSG00000117215', 'ensembl:ENSG00000076706', 'ensembl:ENSG00000138722', 'ensembl:ENSG00000059377', 'ensembl:ENSG00000205632', 'ensembl:ENSG00000178403', 'ensembl:ENSG00000264569', 'ensembl:ENSG00000114841', 'ensembl:ENSG00000153094', 'ensembl:ENSG00000131398', 'ensembl:ENSG00000163110', 'ensembl:ENSG00000139714', 'ensembl:ENSG00000166426', 'ensembl:ENSG00000121316', 'ensembl:ENSG00000138685', 'ensembl:ENSG00000214212', 'ensembl:ENSG00000111713', 'ensembl:ENSG00000137558', 'ensembl:ENSG00000149328', 'ensembl:ENSG00000086717', 'ensembl:ENSG00000101282', 'ensembl:ENSG00000037042', 'ensembl:ENSG00000183019', 'ensembl:ENSG00000183317', 'ensembl:ENSG00000185274', 'ensembl:ENSG00000174032', 'ensembl:ENSG00000153993', 'ensembl:ENSG00000164120', 'ensembl:ENSG00000186212', 'ensembl:ENSG00000152936', 'ensembl:ENSG00000221025', 'ensembl:ENSG00000090534', 'ensembl:ENSG00000134061', 'ensembl:ENSG00000159216', 'ensembl:ENSG00000172322', 'ensembl:ENSG00000204176', 'ensembl:ENSG00000079215', 'ensembl:ENSG00000119729', 'ensembl:ENSG00000101638', 'ensembl:ENSG00000105227', 'ensembl:ENSG00000143858', 'ensembl:ENSG00000111321', 'ensembl:ENSG00000225449', 'ensembl:ENSG00000173698', 'ensembl:ENSG00000255974', 'ensembl:ENSG00000006747', 'ensembl:ENSG00000145703', 'ensembl:ENSG00000117122', 'ensembl:ENSG00000185650', 'ensembl:ENSG00000173535', 'ensembl:ENSG00000179593', 'ensembl:ENSG00000000971', 'ensembl:ENSG00000132141', 'ensembl:ENSG00000160180', 'ensembl:ENSG00000157240', 'ensembl:ENSG00000050438', 'ensembl:ENSG00000186352', 'ensembl:ENSG00000117280', 'ensembl:ENSG00000153404', 'ensembl:ENSG00000049449', 'ensembl:ENSG00000167123', 'ensembl:ENSG00000170385', 'ensembl:ENSG00000138100', 'ensembl:ENSG00000174236', 'ensembl:ENSG00000243836', 'ensembl:ENSG00000138646', 'ensembl:ENSG00000043355', 'ensembl:ENSG00000140835', 'ensembl:ENSG00000158714', 'ensembl:ENSG00000075340', 'ensembl:ENSG00000203952', 'ensembl:ENSG00000243449', 'ensembl:ENSG00000007129', 'ensembl:ENSG00000163605', 'ensembl:ENSG00000197757', 'ensembl:ENSG00000220161', 'ensembl:ENSG00000181649', 'ensembl:ENSG00000105289', 'ensembl:ENSG00000116117', 'ensembl:ENSG00000172296', 'ensembl:ENSG00000168314', 'ensembl:ENSG00000136944', 'ensembl:ENSG00000122180', 'ensembl:ENSG00000170180', 'ensembl:ENSG00000105609', 'ensembl:ENSG00000173585', 'ensembl:ENSG00000172243', 'ensembl:ENSG00000040608', 'ensembl:ENSG00000141750', 'ensembl:ENSG00000224812', 'ensembl:ENSG00000158792', 'ensembl:ENSG00000137752', 'ensembl:ENSG00000136826', 'ensembl:ENSG00000164825', 'ensembl:ENSG00000100504', 'ensembl:ENSG00000176076', 'ensembl:ENSG00000136732', 'ensembl:ENSG00000164117', 'ensembl:ENSG00000054219', 'ensembl:ENSG00000118407', 'ensembl:ENSG00000137094', 'ensembl:ENSG00000005073', 'ensembl:ENSG00000101049', 'ensembl:ENSG00000136250', 'ensembl:ENSG00000172201', 'ensembl:ENSG00000111817', 'ensembl:ENSG00000089250', 'ensembl:ENSG00000199036', 'ensembl:ENSG00000264954', 'ensembl:ENSG00000135480', 'ensembl:ENSG00000026297', 'ensembl:ENSG00000261308', 'ensembl:ENSG00000178573', 'ensembl:ENSG00000168710', 'ensembl:ENSG00000136297', 'ensembl:ENSG00000187736', 'ensembl:ENSG00000205364', 'ensembl:ENSG00000138696', 'ensembl:ENSG00000264462', 'ensembl:ENSG00000182256', 'ensembl:ENSG00000047617', 'ensembl:ENSG00000231752', 'ensembl:ENSG00000089116', 'ensembl:ENSG00000240889', 'ensembl:ENSG00000118707', 'ensembl:ENSG00000259134', 'ensembl:ENSG00000163993', 'ensembl:ENSG00000078401', 'ensembl:ENSG00000264515', 'ensembl:ENSG00000122786', 'ensembl:ENSG00000087510', 'ensembl:ENSG00000144837', 'ensembl:ENSG00000118515', 'ensembl:ENSG00000179583', 'ensembl:ENSG00000122986', 'ensembl:ENSG00000101608', 'ensembl:ENSG00000143473', 'ensembl:ENSG00000114790', 'ensembl:ENSG00000173890', 'ensembl:ENSG00000156127', 'ensembl:ENSG00000124251', 'ensembl:ENSG00000163191', 'ensembl:ENSG00000205403', 'ensembl:ENSG00000018280', 'ensembl:ENSG00000143387', 'ensembl:ENSG00000114698', 'ensembl:ENSG00000164136', 'ensembl:ENSG00000136048', 'ensembl:ENSG00000255929', 'ensembl:ENSG00000236824', 'ensembl:ENSG00000259116', 'ensembl:ENSG00000180914', 'ensembl:ENSG00000167748', 'ensembl:ENSG00000181104', 'ensembl:ENSG00000122884', 'ensembl:ENSG00000008226', 'ensembl:ENSG00000163359', 'ensembl:ENSG00000151025', 'ensembl:ENSG00000118496', 'ensembl:ENSG00000174059', 'ensembl:ENSG00000213901', 'ensembl:ENSG00000163586', 'ensembl:ENSG00000057252', 'ensembl:ENSG00000127083', 'ensembl:ENSG00000161270', 'ensembl:ENSG00000125522', 'ensembl:ENSG00000181481', 'ensembl:ENSG00000126878', 'ensembl:ENSG00000006210', 'ensembl:ENSG00000136859', 'ensembl:ENSG00000171540', 'ensembl:ENSG00000050730', 'ensembl:ENSG00000064012', 'ensembl:ENSG00000204472', 'ensembl:ENSG00000051620', 'ensembl:ENSG00000138594', 'ensembl:ENSG00000144843', 'ensembl:ENSG00000179299', 'ensembl:ENSG00000152467', 'ensembl:ENSG00000157093', 'ensembl:ENSG00000152672', 'ensembl:ENSG00000189058', 'ensembl:ENSG00000104415', 'ensembl:ENSG00000196159', 'ensembl:ENSG00000167508', 'ensembl:ENSG00000205634', 'ensembl:ENSG00000163517', 'ensembl:ENSG00000085719', 'ensembl:ENSG00000232434', 'ensembl:ENSG00000115590', 'ensembl:ENSG00000197019', 'ensembl:ENSG00000149599', 'ensembl:ENSG00000164406', 'ensembl:ENSG00000163106', 'ensembl:ENSG00000259332', 'ensembl:ENSG00000139921', 'ensembl:ENSG00000143514', 'ensembl:ENSG00000139835', 'ensembl:ENSG00000135116', 'ensembl:ENSG00000128340', 'ensembl:ENSG00000147697', 'ensembl:ENSG00000214456', 'ensembl:ENSG00000232803', 'ensembl:ENSG00000136883', 'ensembl:ENSG00000248859', 'ensembl:ENSG00000260549', 'ensembl:ENSG00000089159', 'ensembl:ENSG00000006453', 'ensembl:ENSG00000124507', 'ensembl:ENSG00000105655', 'ensembl:ENSG00000203688', 'ensembl:ENSG00000179673', 'ensembl:ENSG00000174130', 'ensembl:ENSG00000156564', 'ensembl:ENSG00000089847', 'ensembl:ENSG00000134489', 'ensembl:ENSG00000019186', 'ensembl:ENSG00000183615', 'ensembl:ENSG00000229453', 'ensembl:ENSG00000134531', 'ensembl:ENSG00000127920', 'ensembl:ENSG00000152766', 'ensembl:ENSG00000161649', 'ensembl:ENSG00000116729', 'ensembl:ENSG00000138119', 'ensembl:ENSG00000164920', 'ensembl:ENSG00000151655', 'ensembl:ENSG00000138356', 'ensembl:ENSG00000267221', 'ensembl:ENSG00000224870', 'ensembl:ENSG00000197106', 'ensembl:ENSG00000143416', 'ensembl:ENSG00000002746', 'ensembl:ENSG00000112499', 'ensembl:ENSG00000244509', 'ensembl:ENSG00000121764', 'ensembl:ENSG00000170523', 'ensembl:ENSG00000114455', 'ensembl:ENSG00000180340', 'ensembl:ENSG00000110172', 'ensembl:ENSG00000121691', 'ensembl:ENSG00000165164', 'ensembl:ENSG00000124749', 'ensembl:ENSG00000124782', 'ensembl:ENSG00000164080', 'ensembl:ENSG00000140022', 'ensembl:ENSG00000118298', 'ensembl:ENSG00000126500', 'ensembl:ENSG00000114631', 'ensembl:ENSG00000168124', 'ensembl:ENSG00000142583', 'ensembl:ENSG00000111341', 'ensembl:ENSG00000196935', 'ensembl:ENSG00000176658', 'ensembl:ENSG00000187848', 'ensembl:ENSG00000113303', 'ensembl:ENSG00000104894', 'ensembl:ENSG00000167851', 'ensembl:ENSG00000261594', 'ensembl:ENSG00000169908', 'ensembl:ENSG00000226979', 'ensembl:ENSG00000124942', 'ensembl:ENSG00000176714', 'ensembl:ENSG00000187808', 'ensembl:ENSG00000117500', 'ensembl:ENSG00000115594', 'ensembl:ENSG00000199053', 'ensembl:ENSG00000167419', 'ensembl:ENSG00000225930', 'ensembl:ENSG00000144857', 'ensembl:ENSG00000008196', 'ensembl:ENSG00000113721', 'ensembl:ENSG00000067066', 'ensembl:ENSG00000147246', 'ensembl:ENSG00000197565', 'ensembl:ENSG00000197085', 'ensembl:ENSG00000117519', 'ensembl:ENSG00000173728', 'ensembl:ENSG00000122420', 'ensembl:ENSG00000135272', 'ensembl:ENSG00000165730', 'ensembl:ENSG00000188783', 'ensembl:ENSG00000132465', 'ensembl:ENSG00000110852', 'ensembl:ENSG00000197249', 'ensembl:ENSG00000139132', 'ensembl:ENSG00000086506', 'ensembl:ENSG00000155465', 'ensembl:ENSG00000100033', 'ensembl:ENSG00000123999', 'ensembl:ENSG00000164742', 'ensembl:ENSG00000185201', 'ensembl:ENSG00000158169', 'ensembl:ENSG00000244187', 'ensembl:ENSG00000105281', 'ensembl:ENSG00000164089', 'ensembl:ENSG00000133574', 'ensembl:ENSG00000101222', 'ensembl:ENSG00000104081', 'ensembl:ENSG00000102265', 'ensembl:ENSG00000206113', 'ensembl:ENSG00000163239', 'ensembl:ENSG00000117090', 'ensembl:ENSG00000186765', 'ensembl:ENSG00000010319', 'ensembl:ENSG00000197496', 'ensembl:ENSG00000182718', 'ensembl:ENSG00000168754', 'ensembl:ENSG00000183508', 'ensembl:ENSG00000101342', 'ensembl:ENSG00000176909', 'ensembl:ENSG00000170956', 'ensembl:ENSG00000110079', 'ensembl:ENSG00000151929', 'ensembl:ENSG00000168961', 'ensembl:ENSG00000075673', 'ensembl:ENSG00000103241', 'ensembl:ENSG00000155760', 'ensembl:ENSG00000171049', 'ensembl:ENSG00000127530', 'ensembl:ENSG00000254872', 'ensembl:ENSG00000086205', 'ensembl:ENSG00000168229', 'ensembl:ENSG00000204241', 'ensembl:ENSG00000247675', 'ensembl:ENSG00000175985', 'ensembl:ENSG00000206013', 'ensembl:ENSG00000152049', 'ensembl:ENSG00000231738', 'ensembl:ENSG00000178531', 'ensembl:ENSG00000162944', 'ensembl:ENSG00000076770', 'ensembl:ENSG00000151388', 'ensembl:ENSG00000258498', 'ensembl:ENSG00000090104', 'ensembl:ENSG00000052850', 'ensembl:ENSG00000167994', 'ensembl:ENSG00000143119', 'ensembl:ENSG00000178401', 'ensembl:ENSG00000151650', 'ensembl:ENSG00000188766', 'ensembl:ENSG00000169896', 'ensembl:ENSG00000153071', 'ensembl:ENSG00000169385', 'ensembl:ENSG00000163793', 'ensembl:ENSG00000214193', 'ensembl:ENSG00000134463', 'ensembl:ENSG00000187546', 'ensembl:ENSG00000134817', 'ensembl:ENSG00000182107', 'ensembl:ENSG00000140030', 'ensembl:ENSG00000105855', 'ensembl:ENSG00000109861', 'ensembl:ENSG00000256537', 'ensembl:ENSG00000107485', 'ensembl:ENSG00000099194', 'ensembl:ENSG00000242294', 'ensembl:ENSG00000153820', 'ensembl:ENSG00000116044', 'ensembl:ENSG00000169439', 'ensembl:ENSG00000146674', 'ensembl:ENSG00000145390', 'ensembl:ENSG00000157613', 'ensembl:ENSG00000177464', 'ensembl:ENSG00000101336', 'ensembl:ENSG00000172123', 'ensembl:ENSG00000205143', 'ensembl:ENSG00000204889', 'ensembl:ENSG00000132357', 'ensembl:ENSG00000169397', 'ensembl:ENSG00000196169', 'ensembl:ENSG00000122176', 'ensembl:ENSG00000203710', 'ensembl:ENSG00000241563', 'ensembl:ENSG00000130600', 'ensembl:ENSG00000183888', 'ensembl:ENSG00000129474', 'ensembl:ENSG00000204052', 'ensembl:ENSG00000003436', 'ensembl:ENSG00000170374', 'ensembl:ENSG00000174567', 'ensembl:ENSG00000138496', 'ensembl:ENSG00000147443', 'ensembl:ENSG00000120738', 'ensembl:ENSG00000244411', 'ensembl:ENSG00000122862', 'ensembl:ENSG00000007237', 'ensembl:ENSG00000204655', 'ensembl:ENSG00000163083', 'ensembl:ENSG00000133661', 'ensembl:ENSG00000198796', 'ensembl:ENSG00000170381', 'ensembl:ENSG00000214274', 'ensembl:ENSG00000121895', 'ensembl:ENSG00000102174', 'ensembl:ENSG00000125347', 'ensembl:ENSG00000177917', 'ensembl:ENSG00000010704', 'ensembl:ENSG00000196329', 'ensembl:ENSG00000154240', 'ensembl:ENSG00000037965', 'ensembl:ENSG00000212766', 'ensembl:ENSG00000164241', 'ensembl:ENSG00000140798', 'ensembl:ENSG00000226673', 'ensembl:ENSG00000106031', 'ensembl:ENSG00000161653', 'ensembl:ENSG00000139679', 'ensembl:ENSG00000013293', 'ensembl:ENSG00000116132', 'ensembl:ENSG00000163462', 'ensembl:ENSG00000099957', 'ensembl:ENSG00000214353', 'ensembl:ENSG00000180447', 'ensembl:ENSG00000165029', 'ensembl:ENSG00000204175', 'ensembl:ENSG00000189325', 'ensembl:ENSG00000158748', 'ensembl:ENSG00000106211', 'ensembl:ENSG00000106624', 'ensembl:ENSG00000101955', 'ensembl:ENSG00000114127', 'ensembl:ENSG00000161921', 'ensembl:ENSG00000113328', 'ensembl:ENSG00000215045', 'ensembl:ENSG00000134242', 'ensembl:ENSG00000166603', 'ensembl:ENSG00000163394', 'ensembl:ENSG00000070182', 'ensembl:ENSG00000112559', 'ensembl:ENSG00000144642', 'ensembl:ENSG00000139515', 'ensembl:ENSG00000250846', 'ensembl:ENSG00000164621', 'ensembl:ENSG00000128610', 'ensembl:ENSG00000055732', 'ensembl:ENSG00000111602', 'ensembl:ENSG00000128564', 'ensembl:ENSG00000101871', 'ensembl:ENSG00000165474', 'ensembl:ENSG00000132196', 'ensembl:ENSG00000175874', 'ensembl:ENSG00000204644', 'ensembl:ENSG00000116981', 'ensembl:ENSG00000243284', 'ensembl:ENSG00000105877', 'ensembl:ENSG00000154856', 'ensembl:ENSG00000163762', 'ensembl:ENSG00000122859', 'ensembl:ENSG00000121207', 'ensembl:ENSG00000165501', 'ensembl:ENSG00000100979', 'ensembl:ENSG00000115825', 'ensembl:ENSG00000105711', 'ensembl:ENSG00000205730', 'ensembl:ENSG00000140379', 'ensembl:ENSG00000152214', 'ensembl:ENSG00000143556', 'ensembl:ENSG00000179477', 'ensembl:ENSG00000134830', 'ensembl:ENSG00000188039', 'ensembl:ENSG00000159189', 'ensembl:ENSG00000205362', 'ensembl:ENSG00000171885', 'ensembl:ENSG00000206579', 'ensembl:ENSG00000108846', 'ensembl:ENSG00000111087', 'ensembl:ENSG00000188763', 'ensembl:ENSG00000162511', 'ensembl:ENSG00000104043', 'ensembl:ENSG00000122375', 'ensembl:ENSG00000232284', 'ensembl:ENSG00000184709', 'ensembl:ENSG00000207647', 'ensembl:ENSG00000101812', 'ensembl:ENSG00000132874', 'ensembl:ENSG00000158710', 'ensembl:ENSG00000181072', 'ensembl:ENSG00000049089', 'ensembl:ENSG00000171130', 'ensembl:ENSG00000123096', 'ensembl:ENSG00000157514', 'ensembl:ENSG00000163220', 'ensembl:ENSG00000104324', 'ensembl:ENSG00000137731', 'ensembl:ENSG00000163817', 'ensembl:ENSG00000165895', 'ensembl:ENSG00000054598', 'ensembl:ENSG00000128713', 'ensembl:ENSG00000125864', 'ensembl:ENSG00000180537', 'ensembl:ENSG00000188730', 'ensembl:ENSG00000196569', 'ensembl:ENSG00000119283', 'ensembl:ENSG00000256268', 'ensembl:ENSG00000133687', 'ensembl:ENSG00000120075', 'ensembl:ENSG00000151702', 'ensembl:ENSG00000100767', 'ensembl:ENSG00000168566', 'ensembl:ENSG00000182575', 'ensembl:ENSG00000104921', 'ensembl:ENSG00000215190', 'ensembl:ENSG00000238113', 'ensembl:ENSG00000226627', 'ensembl:ENSG00000123080', 'ensembl:ENSG00000105352', 'ensembl:ENSG00000047648', 'ensembl:ENSG00000196954', 'ensembl:ENSG00000123104', 'ensembl:ENSG00000188886', 'ensembl:ENSG00000110931', 'ensembl:ENSG00000142748', 'ensembl:ENSG00000180198', 'ensembl:ENSG00000125531', 'ensembl:ENSG00000139800', 'ensembl:ENSG00000215397', 'ensembl:ENSG00000173786', 'ensembl:ENSG00000248671', 'ensembl:ENSG00000161958', 'ensembl:ENSG00000114013', 'ensembl:ENSG00000183837', 'ensembl:ENSG00000110169', 'ensembl:ENSG00000153165', 'ensembl:ENSG00000187151', 'ensembl:ENSG00000152779', 'ensembl:ENSG00000110324', 'ensembl:ENSG00000130222', 'ensembl:ENSG00000136160', 'ensembl:ENSG00000181350', 'ensembl:ENSG00000100739', 'ensembl:ENSG00000197406', 'ensembl:ENSG00000182327', 'ensembl:ENSG00000184492', 'ensembl:ENSG00000197385', 'ensembl:ENSG00000214922', 'ensembl:ENSG00000196900', 'ensembl:ENSG00000167549', 'ensembl:ENSG00000091879', 'ensembl:ENSG00000197324', 'ensembl:ENSG00000197361', 'ensembl:ENSG00000132199', 'ensembl:ENSG00000205054', 'ensembl:ENSG00000080031', 'ensembl:ENSG00000155926', 'ensembl:ENSG00000142910', 'ensembl:ENSG00000106789', 'ensembl:ENSG00000099937', 'ensembl:ENSG00000111863', 'ensembl:ENSG00000183072', 'ensembl:ENSG00000146410', 'ensembl:ENSG00000150687', 'ensembl:ENSG00000196209', 'ensembl:ENSG00000143603', 'ensembl:ENSG00000185105', 'ensembl:ENSG00000069869', 'ensembl:ENSG00000075275', 'ensembl:ENSG00000115361', 'ensembl:ENSG00000125148', 'ensembl:ENSG00000137142', 'ensembl:ENSG00000220008', 'ensembl:ENSG00000173068', 'ensembl:ENSG00000165092', 'ensembl:ENSG00000130368', 'ensembl:ENSG00000143110', 'ensembl:ENSG00000196972', 'ensembl:ENSG00000248673', 'ensembl:ENSG00000187079', 'ensembl:ENSG00000168685', 'ensembl:ENSG00000165188', 'ensembl:ENSG00000130701', 'ensembl:ENSG00000142973', 'ensembl:ENSG00000106066', 'ensembl:ENSG00000157551', 'ensembl:ENSG00000185792', 'ensembl:ENSG00000136689', 'ensembl:ENSG00000186564', 'ensembl:ENSG00000106258', 'ensembl:ENSG00000139351', 'ensembl:ENSG00000230215', 'ensembl:ENSG00000184557', 'ensembl:ENSG00000104870', 'ensembl:ENSG00000143320', 'ensembl:ENSG00000259142', 'ensembl:ENSG00000142920', 'ensembl:ENSG00000170262', 'ensembl:ENSG00000171812', 'ensembl:ENSG00000087589', 'ensembl:ENSG00000163914', 'ensembl:ENSG00000179178', 'ensembl:ENSG00000245870', 'ensembl:ENSG00000076944', 'ensembl:ENSG00000148908', 'ensembl:ENSG00000178363', 'ensembl:ENSG00000140481', 'ensembl:ENSG00000142494', 'ensembl:ENSG00000172175', 'ensembl:ENSG00000134802', 'ensembl:ENSG00000115919', 'ensembl:ENSG00000228630', 'ensembl:ENSG00000250668', 'ensembl:ENSG00000124549', 'ensembl:ENSG00000174348', 'ensembl:ENSG00000143127', 'ensembl:ENSG00000214381', 'ensembl:ENSG00000219665', 'ensembl:ENSG00000145685', 'ensembl:ENSG00000197520', 'ensembl:ENSG00000188959', 'ensembl:ENSG00000182704', 'ensembl:ENSG00000177875', 'ensembl:ENSG00000183801', 'ensembl:ENSG00000065911', 'ensembl:ENSG00000250742', 'ensembl:ENSG00000173372', 'ensembl:ENSG00000259315', 'ensembl:ENSG00000111961', 'ensembl:ENSG00000137393', 'ensembl:ENSG00000133863', 'ensembl:ENSG00000243468', 'ensembl:ENSG00000075420', 'ensembl:ENSG00000149781', 'ensembl:ENSG00000147606', 'ensembl:ENSG00000185222', 'ensembl:ENSG00000120875', 'ensembl:ENSG00000107611', 'ensembl:ENSG00000109625', 'ensembl:ENSG00000155324', 'ensembl:ENSG00000219435', 'ensembl:ENSG00000154146', 'ensembl:ENSG00000100968', 'ensembl:ENSG00000239998', 'ensembl:ENSG00000131370', 'ensembl:ENSG00000110077', 'ensembl:ENSG00000130052', 'ensembl:ENSG00000173110', 'ensembl:ENSG00000214050', 'ensembl:ENSG00000185669', 'ensembl:ENSG00000162367', 'ensembl:ENSG00000147437', 'ensembl:ENSG00000150656', 'ensembl:ENSG00000258429', 'ensembl:ENSG00000196664', 'ensembl:ENSG00000015520', 'ensembl:ENSG00000148841', 'ensembl:ENSG00000172493', 'ensembl:ENSG00000135269', 'ensembl:ENSG00000167513', 'ensembl:ENSG00000105366', 'ensembl:ENSG00000134160', 'ensembl:ENSG00000180353', 'ensembl:ENSG00000128285', 'ensembl:ENSG00000183760', 'ensembl:ENSG00000232788', 'ensembl:ENSG00000106004', 'ensembl:ENSG00000146192', 'ensembl:ENSG00000175311', 'ensembl:ENSG00000122592', 'ensembl:ENSG00000087253', 'ensembl:ENSG00000196312', 'ensembl:ENSG00000166928', 'ensembl:ENSG00000179023', 'ensembl:ENSG00000238217', 'ensembl:ENSG00000168671', 'ensembl:ENSG00000088882', 'ensembl:ENSG00000176601', 'ensembl:ENSG00000185591', 'ensembl:ENSG00000245848', 'ensembl:ENSG00000102317', 'ensembl:ENSG00000146409', 'ensembl:ENSG00000027697', 'ensembl:ENSG00000161849', 'ensembl:ENSG00000103995', 'ensembl:ENSG00000259070', 'ensembl:ENSG00000175906', 'ensembl:ENSG00000178075', 'ensembl:ENSG00000138448', 'ensembl:ENSG00000160883', 'ensembl:ENSG00000125730', 'ensembl:ENSG00000131435', 'ensembl:ENSG00000234409', 'ensembl:ENSG00000261286', 'ensembl:ENSG00000115648', 'ensembl:ENSG00000119714', 'ensembl:ENSG00000110148', 'ensembl:ENSG00000112309', 'ensembl:ENSG00000149809', 'ensembl:ENSG00000198464', 'ensembl:ENSG00000124564', 'ensembl:ENSG00000149922', 'ensembl:ENSG00000225556', 'ensembl:ENSG00000163735', 'ensembl:ENSG00000149452', 'ensembl:ENSG00000128710', 'ensembl:ENSG00000128709', 'ensembl:ENSG00000198848', 'ensembl:ENSG00000233718', 'ensembl:ENSG00000174326', 'ensembl:ENSG00000161681', 'ensembl:ENSG00000170439', 'ensembl:ENSG00000189238', 'ensembl:ENSG00000102362', 'ensembl:ENSG00000104972', 'ensembl:ENSG00000119535', 'ensembl:ENSG00000135821', 'ensembl:ENSG00000246366', 'ensembl:ENSG00000169429', 'ensembl:ENSG00000111058', 'ensembl:ENSG00000184923', 'ensembl:ENSG00000167676', 'ensembl:ENSG00000123364', 'ensembl:ENSG00000156218', 'ensembl:ENSG00000152558', 'ensembl:ENSG00000183098', 'ensembl:ENSG00000246859', 'ensembl:ENSG00000158373', 'ensembl:ENSG00000162407', 'ensembl:ENSG00000176124', 'ensembl:ENSG00000172216', 'ensembl:ENSG00000112972', 'ensembl:ENSG00000196549', 'ensembl:ENSG00000261105', 'ensembl:ENSG00000199023', 'ensembl:ENSG00000021355', 'ensembl:ENSG00000162753', 'ensembl:ENSG00000172425', 'ensembl:ENSG00000167077', 'ensembl:ENSG00000010610', 'ensembl:ENSG00000005844', 'ensembl:ENSG00000267690', 'ensembl:ENSG00000164946', 'ensembl:ENSG00000160862', 'ensembl:ENSG00000158825', 'ensembl:ENSG00000112303', 'ensembl:ENSG00000139146', 'ensembl:ENSG00000120129', 'ensembl:ENSG00000107099', 'ensembl:ENSG00000243649', 'ensembl:ENSG00000129932', 'ensembl:ENSG00000125144', 'ensembl:ENSG00000135185', 'ensembl:ENSG00000158571', 'ensembl:ENSG00000196616', 'ensembl:ENSG00000226717', 'ensembl:ENSG00000082684', 'ensembl:ENSG00000218358', 'ensembl:ENSG00000117682', 'ensembl:ENSG00000076067', 'ensembl:ENSG00000128604', 'ensembl:ENSG00000215375', 'ensembl:ENSG00000112619', 'ensembl:ENSG00000027644', 'ensembl:ENSG00000154188', 'ensembl:ENSG00000203499', 'ensembl:ENSG00000111181', 'ensembl:ENSG00000079335', 'ensembl:ENSG00000176046', 'ensembl:ENSG00000117602', 'ensembl:ENSG00000102359', 'ensembl:ENSG00000248522', 'ensembl:ENSG00000226124', 'ensembl:ENSG00000138061', 'ensembl:ENSG00000163297', 'ensembl:ENSG00000138944', 'ensembl:ENSG00000255986', 'ensembl:ENSG00000163746', 'ensembl:ENSG00000160593', 'ensembl:ENSG00000026025', 'ensembl:ENSG00000143226', 'ensembl:ENSG00000099622', 'ensembl:ENSG00000186867', 'ensembl:ENSG00000107968', 'ensembl:ENSG00000124107', 'ensembl:ENSG00000145839', 'ensembl:ENSG00000142227', 'ensembl:ENSG00000144152', 'ensembl:ENSG00000165973', 'ensembl:ENSG00000149289', 'ensembl:ENSG00000184497', 'ensembl:ENSG00000073712', 'ensembl:ENSG00000136155', 'ensembl:ENSG00000198807', 'ensembl:ENSG00000165449', 'ensembl:ENSG00000162241', 'ensembl:ENSG00000243742', 'ensembl:ENSG00000228203', 'ensembl:ENSG00000125384', 'ensembl:ENSG00000242574', 'ensembl:ENSG00000142611', 'ensembl:ENSG00000132470', 'ensembl:ENSG00000180229', 'ensembl:ENSG00000179846', 'ensembl:ENSG00000100170', 'ensembl:ENSG00000099814', 'ensembl:ENSG00000237737', 'ensembl:ENSG00000198125', 'ensembl:ENSG00000123427', 'ensembl:ENSG00000267107', 'ensembl:ENSG00000224939', 'ensembl:ENSG00000253293', 'ensembl:ENSG00000147174', 'ensembl:ENSG00000166278', 'ensembl:ENSG00000196460', 'ensembl:ENSG00000077238', 'ensembl:ENSG00000182450', 'ensembl:ENSG00000112773', 'ensembl:ENSG00000143772', 'ensembl:ENSG00000174938', 'ensembl:ENSG00000185551', 'ensembl:ENSG00000167646', 'ensembl:ENSG00000089327', 'ensembl:ENSG00000128274', 'ensembl:ENSG00000185988', 'ensembl:ENSG00000100625', 'ensembl:ENSG00000197256', 'ensembl:ENSG00000165025', 'ensembl:ENSG00000145386', 'ensembl:ENSG00000163430', 'ensembl:ENSG00000269959', 'ensembl:ENSG00000127954', 'ensembl:ENSG00000135046', 'ensembl:ENSG00000001626', 'ensembl:ENSG00000100228', 'ensembl:ENSG00000176293', 'ensembl:ENSG00000031081', 'ensembl:ENSG00000142961', 'ensembl:ENSG00000137819', 'ensembl:ENSG00000114978', 'ensembl:ENSG00000101203', 'ensembl:ENSG00000173451', 'ensembl:ENSG00000267206', 'ensembl:ENSG00000066336', 'ensembl:ENSG00000185585', 'ensembl:ENSG00000164251', 'ensembl:ENSG00000152977', 'ensembl:ENSG00000100577', 'ensembl:ENSG00000118113', 'ensembl:ENSG00000187134', 'ensembl:ENSG00000156381', 'ensembl:ENSG00000197386', 'ensembl:ENSG00000116962', 'ensembl:ENSG00000120690', 'ensembl:ENSG00000234511', 'ensembl:ENSG00000187922', 'ensembl:ENSG00000181458', 'ensembl:ENSG00000072954', 'ensembl:ENSG00000167800', 'ensembl:ENSG00000038427', 'ensembl:ENSG00000205710', 'ensembl:ENSG00000023839', 'ensembl:ENSG00000140968', 'ensembl:ENSG00000188647', 'ensembl:ENSG00000108950', 'ensembl:ENSG00000167207', 'ensembl:ENSG00000127472', 'ensembl:ENSG00000186517', 'ensembl:ENSG00000092758', 'ensembl:ENSG00000123405', 'ensembl:ENSG00000203857', 'ensembl:ENSG00000231185', 'ensembl:ENSG00000111218', 'ensembl:ENSG00000261701', 'ensembl:ENSG00000019169', 'ensembl:ENSG00000143867', 'ensembl:ENSG00000158869', 'ensembl:ENSG00000111057', 'ensembl:ENSG00000069696', 'ensembl:ENSG00000183421', 'ensembl:ENSG00000010379', 'ensembl:ENSG00000177575', 'ensembl:ENSG00000183579', 'ensembl:ENSG00000188176', 'ensembl:ENSG00000111886', 'ensembl:ENSG00000181029', 'ensembl:ENSG00000215808', 'ensembl:ENSG00000267414', 'ensembl:ENSG00000123689', 'ensembl:ENSG00000173926', 'ensembl:ENSG00000174837', 'ensembl:ENSG00000082175', 'ensembl:ENSG00000133561', 'ensembl:ENSG00000078399', 'ensembl:ENSG00000108387', 'ensembl:ENSG00000128016', 'ensembl:ENSG00000157510', 'ensembl:ENSG00000171551', 'ensembl:ENSG00000157833', 'ensembl:ENSG00000102003', 'ensembl:ENSG00000187048', 'ensembl:ENSG00000137868', 'ensembl:ENSG00000263069', 'ensembl:ENSG00000176490', 'ensembl:ENSG00000205358', 'ensembl:ENSG00000251493', 'ensembl:ENSG00000182118', 'ensembl:ENSG00000185155', 'ensembl:ENSG00000105926', 'ensembl:ENSG00000198597', 'ensembl:ENSG00000057294', 'ensembl:ENSG00000145040', 'ensembl:ENSG00000073670', 'ensembl:ENSG00000258667', 'ensembl:ENSG00000215580', 'ensembl:ENSG00000152990', 'ensembl:ENSG00000176014', 'ensembl:ENSG00000147168', 'ensembl:ENSG00000197142', 'ensembl:ENSG00000197614', 'ensembl:ENSG00000100292', 'ensembl:ENSG00000163694', 'ensembl:ENSG00000125850', 'ensembl:ENSG00000223865', 'ensembl:ENSG00000115380', 'ensembl:ENSG00000168994', 'ensembl:ENSG00000168779', 'ensembl:ENSG00000187800', 'ensembl:ENSG00000111674', 'ensembl:ENSG00000241570', 'ensembl:ENSG00000176692', 'ensembl:ENSG00000102934', 'ensembl:ENSG00000057593', 'ensembl:ENSG00000251632', 'ensembl:ENSG00000178395', 'ensembl:ENSG00000186431', 'ensembl:ENSG00000088827', 'ensembl:ENSG00000226435', 'ensembl:ENSG00000168899', 'ensembl:ENSG00000131724', 'ensembl:ENSG00000156535', 'ensembl:ENSG00000223776', 'ensembl:ENSG00000104518', 'ensembl:ENSG00000175063', 'ensembl:ENSG00000117650', 'ensembl:ENSG00000167644', 'ensembl:ENSG00000147571', 'ensembl:ENSG00000117318', 'ensembl:ENSG00000183571', 'ensembl:ENSG00000172382', 'ensembl:ENSG00000204152', 'ensembl:ENSG00000179403', 'ensembl:ENSG00000133055', 'ensembl:ENSG00000189184', 'ensembl:ENSG00000166130', 'ensembl:ENSG00000250423', 'ensembl:ENSG00000128052', 'ensembl:ENSG00000203668', 'ensembl:ENSG00000159674', 'ensembl:ENSG00000122863', 'ensembl:ENSG00000092054', 'ensembl:ENSG00000197540', 'ensembl:ENSG00000163565', 'ensembl:ENSG00000144824', 'ensembl:ENSG00000164292', 'ensembl:ENSG00000163638', 'ensembl:ENSG00000141506', 'ensembl:ENSG00000162913', 'ensembl:ENSG00000182326', 'ensembl:ENSG00000175556', 'ensembl:ENSG00000267586', 'ensembl:ENSG00000241131', 'ensembl:ENSG00000101916', 'ensembl:ENSG00000171357', 'ensembl:ENSG00000214100', 'ensembl:ENSG00000254415', 'ensembl:ENSG00000179044', 'ensembl:ENSG00000159650', 'ensembl:ENSG00000060138', 'ensembl:ENSG00000229619', 'ensembl:ENSG00000177425', 'ensembl:ENSG00000179344', 'ensembl:ENSG00000112343', 'ensembl:ENSG00000196557', 'ensembl:ENSG00000180008', 'ensembl:ENSG00000228135', 'ensembl:ENSG00000162747', 'ensembl:ENSG00000197747', 'ensembl:ENSG00000010671', 'ensembl:ENSG00000197405', 'ensembl:ENSG00000068001', 'ensembl:ENSG00000164938', 'ensembl:ENSG00000235984', 'ensembl:ENSG00000123342', 'ensembl:ENSG00000163623', 'ensembl:ENSG00000233639', 'ensembl:ENSG00000165071', 'ensembl:ENSG00000089169', 'ensembl:ENSG00000177045', 'ensembl:ENSG00000128383', 'ensembl:ENSG00000138411', 'ensembl:ENSG00000112137', 'ensembl:ENSG00000023445', 'ensembl:ENSG00000131873', 'ensembl:ENSG00000114737', 'ensembl:ENSG00000126218', 'ensembl:ENSG00000171450', 'ensembl:ENSG00000137473', 'ensembl:ENSG00000240990', 'ensembl:ENSG00000095596', 'ensembl:ENSG00000092607', 'ensembl:ENSG00000213401', 'ensembl:ENSG00000160097', 'ensembl:ENSG00000157168', 'ensembl:ENSG00000221869', 'ensembl:ENSG00000149131', 'ensembl:ENSG00000186862', 'ensembl:ENSG00000176387', 'ensembl:ENSG00000172548', 'ensembl:ENSG00000138964', 'ensembl:ENSG00000212123', 'ensembl:ENSG00000267121', 'ensembl:ENSG00000135605', 'ensembl:ENSG00000237424', 'ensembl:ENSG00000257596', 'ensembl:ENSG00000121933', 'ensembl:ENSG00000226953', 'ensembl:ENSG00000259721', 'ensembl:ENSG00000257542', 'ensembl:ENSG00000251372', 'ensembl:ENSG00000132514', 'ensembl:ENSG00000132205', 'ensembl:ENSG00000104213', 'ensembl:ENSG00000067048', 'ensembl:ENSG00000176236', 'ensembl:ENSG00000169715', 'ensembl:ENSG00000087116', 'ensembl:ENSG00000115616', 'ensembl:ENSG00000131650', 'ensembl:ENSG00000141736', 'ensembl:ENSG00000135248', 'ensembl:ENSG00000173227', 'ensembl:ENSG00000171777', 'ensembl:ENSG00000012211', 'ensembl:ENSG00000174021', 'ensembl:ENSG00000236296', 'ensembl:ENSG00000173540', 'ensembl:ENSG00000181409', 'ensembl:ENSG00000140623', 'ensembl:ENSG00000265660', 'ensembl:ENSG00000170458', 'ensembl:ENSG00000243715', 'ensembl:ENSG00000100342', 'ensembl:ENSG00000165185', 'ensembl:ENSG00000157734', 'ensembl:ENSG00000231826', 'ensembl:ENSG00000152804', 'ensembl:ENSG00000187764', 'ensembl:ENSG00000166819', 'ensembl:ENSG00000150681', 'ensembl:ENSG00000172578', 'ensembl:ENSG00000121270', 'ensembl:ENSG00000174945', 'ensembl:ENSG00000118640', 'ensembl:ENSG00000239332', 'ensembl:ENSG00000149451', 'ensembl:ENSG00000239713', 'ensembl:ENSG00000160716', 'ensembl:ENSG00000198417', 'ensembl:ENSG00000105419', 'ensembl:ENSG00000115461', 'ensembl:ENSG00000205929', 'ensembl:ENSG00000089225', 'ensembl:ENSG00000148848', 'ensembl:ENSG00000126561', 'ensembl:ENSG00000111679', 'ensembl:ENSG00000124785', 'ensembl:ENSG00000075223', 'ensembl:ENSG00000187837', 'ensembl:ENSG00000187166', 'ensembl:ENSG00000261649', 'ensembl:ENSG00000171860', 'ensembl:ENSG00000170178', 'ensembl:ENSG00000121898', 'ensembl:ENSG00000141574', 'ensembl:ENSG00000160233', 'ensembl:ENSG00000182578', 'ensembl:ENSG00000167798', 'ensembl:ENSG00000135604', 'ensembl:ENSG00000213445', 'ensembl:ENSG00000268592', 'ensembl:ENSG00000007372', 'ensembl:ENSG00000105697', 'ensembl:ENSG00000166927', 'ensembl:ENSG00000138772', 'ensembl:ENSG00000184347', 'ensembl:ENSG00000094796', 'ensembl:ENSG00000184574', 'ensembl:ENSG00000196091', 'ensembl:ENSG00000067167', 'ensembl:ENSG00000166091', 'ensembl:ENSG00000139219', 'ensembl:ENSG00000163687', 'ensembl:ENSG00000047457', 'ensembl:ENSG00000241186', 'ensembl:ENSG00000070190', 'ensembl:ENSG00000100122', 'ensembl:ENSG00000099869', 'ensembl:ENSG00000092531', 'ensembl:ENSG00000123268', 'ensembl:ENSG00000244215', 'ensembl:ENSG00000204389', 'ensembl:ENSG00000064655', 'ensembl:ENSG00000204963', 'ensembl:ENSG00000213214', 'ensembl:ENSG00000125968', 'ensembl:ENSG00000235478', 'ensembl:ENSG00000270604', 'ensembl:ENSG00000099960', 'ensembl:ENSG00000187720', 'ensembl:ENSG00000230667', 'ensembl:ENSG00000124780', 'ensembl:ENSG00000004799', 'ensembl:ENSG00000137834', 'ensembl:ENSG00000172716', 'ensembl:ENSG00000166359', 'ensembl:ENSG00000168404', 'ensembl:ENSG00000166920', 'ensembl:ENSG00000185862', 'ensembl:ENSG00000145708', 'ensembl:ENSG00000163464', 'ensembl:ENSG00000000003', 'ensembl:ENSG00000064787', 'ensembl:ENSG00000125122', 'ensembl:ENSG00000174939', 'ensembl:ENSG00000198346', 'ensembl:ENSG00000105409', 'ensembl:ENSG00000168918', 'ensembl:ENSG00000243566', 'ensembl:ENSG00000173295', 'ensembl:ENSG00000175707', 'ensembl:ENSG00000136167', 'ensembl:ENSG00000188290', 'ensembl:ENSG00000198805', 'ensembl:ENSG00000077984', 'ensembl:ENSG00000130720', 'ensembl:ENSG00000165309', 'ensembl:ENSG00000104611', 'ensembl:ENSG00000055955', 'ensembl:ENSG00000183943', 'ensembl:ENSG00000187783', 'ensembl:ENSG00000165072', 'ensembl:ENSG00000109061', 'ensembl:ENSG00000124440', 'ensembl:ENSG00000145623', 'ensembl:ENSG00000168995', 'ensembl:ENSG00000167693', 'ensembl:ENSG00000166159', 'ensembl:ENSG00000198743', 'ensembl:ENSG00000180263', 'ensembl:ENSG00000214290', 'ensembl:ENSG00000182600', 'ensembl:ENSG00000165806', 'ensembl:ENSG00000130675', 'ensembl:ENSG00000010327', 'ensembl:ENSG00000151632', 'ensembl:ENSG00000172380', 'ensembl:ENSG00000203782', 'ensembl:ENSG00000254726', 'ensembl:ENSG00000019582', 'ensembl:ENSG00000106077', 'ensembl:ENSG00000091317', 'ensembl:ENSG00000182885', 'ensembl:ENSG00000154262', 'ensembl:ENSG00000133639', 'ensembl:ENSG00000121570', 'ensembl:ENSG00000145198', 'ensembl:ENSG00000134516', 'ensembl:ENSG00000172037', 'ensembl:ENSG00000075884', 'ensembl:ENSG00000113140', 'ensembl:ENSG00000251450', 'ensembl:ENSG00000188263', 'ensembl:ENSG00000155307', 'ensembl:ENSG00000167941', 'ensembl:ENSG00000125378', 'ensembl:ENSG00000018625', 'ensembl:ENSG00000188522', 'ensembl:ENSG00000080293', 'ensembl:ENSG00000167588', 'ensembl:ENSG00000150457', 'ensembl:ENSG00000110719', 'ensembl:ENSG00000176153', 'ensembl:ENSG00000204091', 'ensembl:ENSG00000108821', 'ensembl:ENSG00000104447', 'ensembl:ENSG00000255346', 'ensembl:ENSG00000075426', 'ensembl:ENSG00000209482', 'ensembl:ENSG00000078177', 'ensembl:ENSG00000132763', 'ensembl:ENSG00000104833', 'ensembl:ENSG00000101439', 'ensembl:ENSG00000006283', 'ensembl:ENSG00000166265', 'ensembl:ENSG00000196411', 'ensembl:ENSG00000112539', 'ensembl:ENSG00000178342', 'ensembl:ENSG00000241399', 'ensembl:ENSG00000168309', 'ensembl:ENSG00000158467', 'ensembl:ENSG00000182632', 'ensembl:ENSG00000163046', 'ensembl:ENSG00000248383', 'ensembl:ENSG00000166923', 'ensembl:ENSG00000166483', 'ensembl:ENSG00000116353', 'ensembl:ENSG00000224389', 'ensembl:ENSG00000198734', 'ensembl:ENSG00000135245', 'ensembl:ENSG00000118520', 'ensembl:ENSG00000142632', 'ensembl:ENSG00000021488', 'ensembl:ENSG00000164430', 'ensembl:ENSG00000170545', 'ensembl:ENSG00000155962', 'ensembl:ENSG00000224189', 'ensembl:ENSG00000187908', 'ensembl:ENSG00000173039', 'ensembl:ENSG00000122012', 'ensembl:ENSG00000173597', 'ensembl:ENSG00000197594', 'ensembl:ENSG00000135898', 'ensembl:ENSG00000124935', 'ensembl:ENSG00000103490', 'ensembl:ENSG00000134294', 'ensembl:ENSG00000115956', 'ensembl:ENSG00000085265', 'ensembl:ENSG00000102109', 'ensembl:ENSG00000162692', 'ensembl:ENSG00000168209', 'ensembl:ENSG00000255571', 'ensembl:ENSG00000075043', 'ensembl:ENSG00000172508', 'ensembl:ENSG00000160111', 'ensembl:ENSG00000085276', 'ensembl:ENSG00000075234', 'ensembl:ENSG00000130429', 'ensembl:ENSG00000027075', 'ensembl:ENSG00000174946', 'ensembl:ENSG00000147251', 'ensembl:ENSG00000137713', 'ensembl:ENSG00000267421', 'ensembl:ENSG00000237412', 'ensembl:ENSG00000100867', 'ensembl:ENSG00000187942', 'ensembl:ENSG00000110876', 'ensembl:ENSG00000123416', 'ensembl:ENSG00000101096', 'ensembl:ENSG00000119139', 'ensembl:ENSG00000142089', 'ensembl:ENSG00000125966', 'ensembl:ENSG00000177352', 'ensembl:ENSG00000163823', 'ensembl:ENSG00000132274', 'ensembl:ENSG00000169783', 'ensembl:ENSG00000133454', 'ensembl:ENSG00000253250', 'ensembl:ENSG00000158859', 'ensembl:ENSG00000164761', 'ensembl:ENSG00000162069', 'ensembl:ENSG00000143382', 'ensembl:ENSG00000171100', 'ensembl:ENSG00000255850', 'ensembl:ENSG00000255823', 'ensembl:ENSG00000176194', 'ensembl:ENSG00000143819', 'ensembl:ENSG00000183397', 'ensembl:ENSG00000205413', 'ensembl:ENSG00000166689', 'ensembl:ENSG00000197901', 'ensembl:ENSG00000115607', 'ensembl:ENSG00000084628', 'ensembl:ENSG00000105939', 'ensembl:ENSG00000271605', 'ensembl:ENSG00000104490', 'ensembl:ENSG00000197971', 'ensembl:ENSG00000151176', 'ensembl:ENSG00000107821', 'ensembl:ENSG00000243927', 'ensembl:ENSG00000123094', 'ensembl:ENSG00000134250', 'ensembl:ENSG00000144802', 'ensembl:ENSG00000176845', 'ensembl:ENSG00000128714', 'ensembl:ENSG00000154727', 'ensembl:ENSG00000146373', 'ensembl:ENSG00000107807', 'ensembl:ENSG00000171873', 'ensembl:ENSG00000230427', 'ensembl:ENSG00000137936', 'ensembl:ENSG00000184371', 'ensembl:ENSG00000056736', 'ensembl:ENSG00000135973', 'ensembl:ENSG00000167711', 'ensembl:ENSG00000005243', 'ensembl:ENSG00000146477', 'ensembl:ENSG00000258743', 'ensembl:ENSG00000139734', 'ensembl:ENSG00000171051', 'ensembl:ENSG00000004939', 'ensembl:ENSG00000116096', 'ensembl:ENSG00000092820', 'ensembl:ENSG00000156925', 'ensembl:ENSG00000123610', 'ensembl:ENSG00000106511', 'ensembl:ENSG00000236279', 'ensembl:ENSG00000214814', 'ensembl:ENSG00000166118', 'ensembl:ENSG00000170231', 'ensembl:ENSG00000110400', 'ensembl:ENSG00000093134', 'ensembl:ENSG00000149575', 'ensembl:ENSG00000203663', 'ensembl:ENSG00000174944', 'ensembl:ENSG00000129450', 'ensembl:ENSG00000140931', 'ensembl:ENSG00000178202', 'ensembl:ENSG00000092969', 'ensembl:ENSG00000157111', 'ensembl:ENSG00000125089', 'ensembl:ENSG00000175764', 'ensembl:ENSG00000182177', 'ensembl:ENSG00000144908', 'ensembl:ENSG00000128602', 'ensembl:ENSG00000136869', 'ensembl:ENSG00000019549', 'ensembl:ENSG00000183307', 'ensembl:ENSG00000115129', 'ensembl:ENSG00000184302', 'ensembl:ENSG00000256162', 'ensembl:ENSG00000186854', 'ensembl:ENSG00000163874', 'ensembl:ENSG00000167208', 'ensembl:ENSG00000175879', 'ensembl:ENSG00000156414', 'ensembl:ENSG00000161664', 'ensembl:ENSG00000183682', 'ensembl:ENSG00000066294', 'ensembl:ENSG00000127951', 'ensembl:ENSG00000178947', 'ensembl:ENSG00000146221', 'ensembl:ENSG00000170370', 'ensembl:ENSG00000203485', 'ensembl:ENSG00000107317', 'ensembl:ENSG00000101974', 'ensembl:ENSG00000073792', 'ensembl:ENSG00000248144', 'ensembl:ENSG00000259831', 'ensembl:ENSG00000111907', 'ensembl:ENSG00000145794', 'ensembl:ENSG00000103316', 'ensembl:ENSG00000102287', 'ensembl:ENSG00000164236', 'ensembl:ENSG00000112799', 'ensembl:ENSG00000068697', 'ensembl:ENSG00000167244', 'ensembl:ENSG00000235770', 'ensembl:ENSG00000213719', 'ensembl:ENSG00000260019', 'ensembl:ENSG00000117400', 'ensembl:ENSG00000144891', 'ensembl:ENSG00000177119', 'ensembl:ENSG00000141622', 'ensembl:ENSG00000257771', 'ensembl:ENSG00000164093', 'ensembl:ENSG00000072694', 'ensembl:ENSG00000162739', 'ensembl:ENSG00000213654', 'ensembl:ENSG00000129048', 'ensembl:ENSG00000143891', 'ensembl:ENSG00000184988', 'ensembl:ENSG00000168306', 'ensembl:ENSG00000196126', 'ensembl:ENSG00000162066', 'ensembl:ENSG00000135069', 'ensembl:ENSG00000100906', 'ensembl:ENSG00000174718', 'ensembl:ENSG00000130158', 'ensembl:ENSG00000188107', 'ensembl:ENSG00000236404', 'ensembl:ENSG00000171388', 'ensembl:ENSG00000135477', 'ensembl:ENSG00000180638', 'ensembl:ENSG00000139209', 'ensembl:ENSG00000150722', 'ensembl:ENSG00000124875'}\n",
      "\n",
      "Drop duplicated rows...\n",
      "(21504, 6)\n",
      "\n",
      "Drop duplicated nodes...\n",
      "(19721, 6)\n",
      "\n",
      "There is a problem in the annotation of nodes.\n",
      "The number of annotated nodes is different than the number of nodes in the graph.\n",
      "Curated nodes not in the graph: set()\n",
      "Monarch nodes not in the graph: set()\n",
      "RNA-seq nodes not in the graph: {'HGNC:32554', 'HGNC:53304', 'HGNC:190', 'HGNC:22058', 'HGNC:24764', 'HGNC:21923', 'HGNC:6925', 'HGNC:385', 'HGNC:21361', 'HGNC:10925', 'HGNC:9033', 'HGNC:15523', 'HGNC:29218', 'HGNC:10046', 'HGNC:21732', 'HGNC:14351', 'HGNC:2067', 'HGNC:49847', 'HGNC:4871', 'HGNC:26277', 'HGNC:7745', 'HGNC:49621', 'HGNC:19174', 'HGNC:13406', 'HGNC:1951', 'HGNC:10919', 'HGNC:52696', 'HGNC:1506', 'HGNC:548', 'HGNC:20568', 'HGNC:123', 'HGNC:51497', 'HGNC:11828', 'HGNC:28904', 'HGNC:11917', 'HGNC:29800', 'HGNC:20565', 'HGNC:10777', 'HGNC:51124', 'HGNC:10707', 'HGNC:1621', 'HGNC:41186', 'HGNC:53466', 'HGNC:19086', 'HGNC:37156', 'HGNC:27502', 'HGNC:4028', 'HGNC:26689', 'HGNC:33391', 'HGNC:20374', 'HGNC:24473', 'HGNC:33724', 'HGNC:44162', 'HGNC:20601', 'HGNC:29394', 'HGNC:4941', 'HGNC:11027', 'HGNC:48592', 'HGNC:2156', 'HGNC:33100', 'HGNC:16267', 'HGNC:30085', 'HGNC:23712', 'HGNC:30706', 'HGNC:25460', 'HGNC:4403', 'HGNC:34548', 'HGNC:29821', 'HGNC:51126', 'HGNC:21079', 'HGNC:18437', 'HGNC:9480', 'HGNC:28018', 'HGNC:34437', 'HGNC:17936', 'HGNC:11873', 'HGNC:2063', 'HGNC:18606', 'HGNC:3542', 'HGNC:16262', 'HGNC:22196', 'HGNC:43642', 'HGNC:23534', 'HGNC:4862', 'HGNC:15975', 'HGNC:27800', 'HGNC:15475', 'HGNC:10970', 'HGNC:23489', 'HGNC:53976', 'HGNC:23483', 'HGNC:15707', 'HGNC:33767', 'HGNC:48959', 'HGNC:43943', 'HGNC:6027', 'HGNC:5295', 'HGNC:27270', 'HGNC:20564', 'HGNC:6259', 'HGNC:40396', 'HGNC:13287', 'HGNC:6060', 'HGNC:3129', 'HGNC:4479', 'HGNC:4485', 'HGNC:6244', 'HGNC:18167', 'HGNC:9256', 'HGNC:14465', 'HGNC:33766', 'HGNC:1953', 'HGNC:1816', 'HGNC:4235', 'HGNC:28522', 'HGNC:31409', 'HGNC:18791', 'HGNC:1686', 'HGNC:17025', 'HGNC:2436', 'HGNC:15830', 'HGNC:18628', 'HGNC:31704', 'HGNC:19323', 'HGNC:30200', 'HGNC:18958', 'HGNC:48329', 'HGNC:6024', 'HGNC:33781', 'HGNC:8677', 'HGNC:4711', 'HGNC:31339', 'HGNC:10059', 'HGNC:26624', 'HGNC:29612', 'HGNC:48639', 'HGNC:37284', 'HGNC:23067', 'HGNC:4581', 'HGNC:2536', 'HGNC:14965', 'HGNC:24956', 'HGNC:15020', 'HGNC:55347', 'HGNC:26740', 'HGNC:21573', 'HGNC:5414', 'HGNC:6819', 'HGNC:14678', 'HGNC:4712', 'HGNC:1683', 'HGNC:15767', 'HGNC:10048', 'HGNC:2481', 'HGNC:24844', 'HGNC:9087', 'HGNC:29957', 'HGNC:10488', 'HGNC:20304', 'HGNC:12769', 'HGNC:7227', 'HGNC:6528', 'HGNC:9024', 'HGNC:32063', 'HGNC:3042', 'HGNC:25607', 'HGNC:670', 'HGNC:26654', 'HGNC:11774', 'HGNC:3656', 'HGNC:16431', 'HGNC:13843', 'HGNC:612', 'HGNC:27349', 'HGNC:9934', 'HGNC:1318', 'HGNC:14525', 'HGNC:3025', 'HGNC:9070', 'HGNC:29193', 'HGNC:17131', 'HGNC:30358', 'HGNC:232', 'HGNC:1338', 'HGNC:21718', 'HGNC:3383', 'HGNC:1248', 'HGNC:8113', 'HGNC:28664', 'HGNC:4905', 'HGNC:37253', 'HGNC:17265', 'HGNC:30843', 'HGNC:4161', 'HGNC:17167', 'HGNC:656', 'HGNC:33784', 'HGNC:3602', 'HGNC:25579', 'HGNC:30910', 'HGNC:25742', 'HGNC:648', 'HGNC:13886', 'HGNC:16442', 'HGNC:1641', 'HGNC:23396', 'HGNC:42400', 'HGNC:16309', 'HGNC:2433', 'HGNC:26032', 'HGNC:50538', 'HGNC:49278', 'HGNC:14211', 'HGNC:44570', 'HGNC:12006', 'HGNC:34414', 'HGNC:41001', 'HGNC:1441', 'HGNC:54248', 'HGNC:8723', 'HGNC:17383', 'HGNC:7607', 'HGNC:38661', 'HGNC:14670', 'HGNC:27402', 'HGNC:53657', 'HGNC:11046', 'HGNC:10497', 'HGNC:55433', 'HGNC:3770', 'HGNC:30132', 'HGNC:16911', 'HGNC:3620', 'HGNC:21206', 'HGNC:17960', 'HGNC:30035', 'HGNC:9780', 'HGNC:44064', 'HGNC:384', 'HGNC:7189', 'HGNC:29853', 'HGNC:27841', 'HGNC:26292', 'HGNC:14679', 'HGNC:14640', 'HGNC:31041', 'HGNC:16711', 'HGNC:910', 'HGNC:7908', 'HGNC:352', 'HGNC:28580', 'HGNC:2216', 'HGNC:20458', 'HGNC:29667', 'HGNC:5113', 'HGNC:53387', 'HGNC:18718', 'HGNC:13937', 'HGNC:3613', 'HGNC:17357', 'HGNC:14451', 'HGNC:11701', 'HGNC:27570', 'HGNC:26708', 'HGNC:11738', 'HGNC:1167', 'HGNC:26977', 'HGNC:974', 'HGNC:2610', 'HGNC:964', 'HGNC:23497', 'HGNC:18260', 'HGNC:13549', 'HGNC:24063', 'HGNC:21723', 'HGNC:5239', 'HGNC:1688', 'HGNC:25116', 'HGNC:6469', 'HGNC:7892', 'HGNC:41192', 'HGNC:10571', 'HGNC:21918', 'HGNC:21555', 'HGNC:7033', 'HGNC:10668', 'HGNC:28725', 'HGNC:6482', 'HGNC:10625', 'HGNC:40014', 'HGNC:24712', 'HGNC:3520', 'HGNC:6517', 'HGNC:44466', 'HGNC:1479', 'HGNC:12419', 'HGNC:55297', 'HGNC:15878', 'HGNC:6294', 'HGNC:22941', 'HGNC:2766', 'HGNC:11034', 'HGNC:25625', 'HGNC:6261', 'HGNC:7567', 'HGNC:2662', 'HGNC:1304', 'HGNC:3664', 'HGNC:244', 'HGNC:15548', 'HGNC:704', 'HGNC:1073', 'HGNC:19177', 'HGNC:25057', 'HGNC:8515', 'HGNC:8090', 'HGNC:16247', 'HGNC:24745', 'HGNC:9240', 'HGNC:6079', 'HGNC:27244', 'HGNC:19007', 'HGNC:1705', 'HGNC:28468', 'HGNC:28339', 'HGNC:32696', 'HGNC:4702', 'HGNC:37165', 'HGNC:22231', 'HGNC:27718', 'HGNC:11896', 'HGNC:5964', 'HGNC:36', 'HGNC:17353', 'HGNC:30555', 'HGNC:15598', 'HGNC:11906', 'HGNC:1730', 'HGNC:52163', 'HGNC:6729', 'HGNC:5133', 'HGNC:9991', 'HGNC:44672', 'HGNC:23602', 'HGNC:27619', 'HGNC:9600', 'HGNC:30229', 'HGNC:29898', 'HGNC:21387', 'HGNC:430', 'HGNC:675', 'HGNC:11396', 'HGNC:16526', 'HGNC:11309', 'HGNC:26572', 'HGNC:30614', 'HGNC:14555', 'HGNC:14620', 'HGNC:11867', 'HGNC:54284', 'HGNC:10643', 'HGNC:16642', 'HGNC:18539', 'HGNC:30525', 'HGNC:18581', 'HGNC:16497', 'HGNC:9789', 'HGNC:27615', 'HGNC:14554', 'HGNC:16147', 'HGNC:53363', 'HGNC:15722', 'HGNC:21293', 'HGNC:48605', 'HGNC:29859', 'HGNC:7655', 'HGNC:49513', 'HGNC:7898', 'HGNC:44117', 'HGNC:16824', 'HGNC:53869', 'HGNC:28834', 'HGNC:27001', 'HGNC:24368', 'HGNC:18816', 'HGNC:4519', 'HGNC:960', 'HGNC:6246', 'HGNC:9581', 'HGNC:8805', 'HGNC:24294', 'HGNC:21378', 'HGNC:386', 'HGNC:48598', 'HGNC:27877', 'HGNC:14284', 'HGNC:44128', 'HGNC:6899', 'HGNC:3515', 'HGNC:42319', 'HGNC:33277', 'HGNC:5474', 'HGNC:10967', 'HGNC:6250', 'HGNC:29593', 'HGNC:51180', 'HGNC:28275', 'HGNC:22128', 'HGNC:541', 'HGNC:52652', 'HGNC:29807', 'HGNC:27031', 'HGNC:43690', 'HGNC:3611', 'HGNC:7400', 'HGNC:25588', 'HGNC:6915', 'HGNC:9802', 'HGNC:16910', 'HGNC:11010', 'HGNC:2053', 'HGNC:1712', 'HGNC:1142', 'HGNC:51178', 'HGNC:24911', 'HGNC:15478', 'HGNC:16315', 'HGNC:21015', 'HGNC:15525', 'HGNC:29216', 'HGNC:10528', 'HGNC:21650', 'HGNC:6603', 'HGNC:251', 'HGNC:2899', 'HGNC:16378', 'HGNC:14446', 'HGNC:25357', 'HGNC:7172', 'HGNC:24355', 'HGNC:18455', 'HGNC:7577', 'HGNC:2054', 'HGNC:32675', 'HGNC:7448', 'HGNC:37286', 'HGNC:39433', 'HGNC:42874', 'HGNC:6111', 'HGNC:20310', 'HGNC:40537', 'HGNC:24126', 'HGNC:7405', 'HGNC:15836', 'HGNC:28613', 'HGNC:18005', 'HGNC:15855', 'HGNC:4449', 'HGNC:8266', 'HGNC:28036', 'HGNC:14874', 'HGNC:11037', 'HGNC:6814', 'HGNC:6588', 'HGNC:17173', 'HGNC:50702', 'HGNC:6903', 'HGNC:9594', 'HGNC:120', 'HGNC:13300', 'HGNC:6461', 'HGNC:13839', 'HGNC:27371', 'HGNC:51408', 'HGNC:10727', 'HGNC:49234', 'HGNC:5185', 'HGNC:29424', 'HGNC:10504', 'HGNC:30619', 'HGNC:37099', 'HGNC:4713', 'HGNC:22036', 'HGNC:26439', 'HGNC:5134', 'HGNC:9591', 'HGNC:5139', 'HGNC:28875', 'HGNC:9582', 'HGNC:1971', 'HGNC:33463', 'HGNC:28497', 'HGNC:6678', 'HGNC:24262', 'HGNC:9878', 'HGNC:15588', 'HGNC:16253', 'HGNC:50843', 'HGNC:23421', 'HGNC:15862', 'HGNC:27924', 'HGNC:19191', 'HGNC:55493', 'HGNC:53034', 'HGNC:19182', 'HGNC:11510', 'HGNC:1956', 'HGNC:17852', 'HGNC:41777', 'HGNC:14307', 'HGNC:11049', 'HGNC:220', 'HGNC:16168', 'HGNC:10722', 'HGNC:6064', 'HGNC:2196', 'HGNC:1941', 'HGNC:50501', 'HGNC:88', 'HGNC:16414', 'HGNC:2000', 'HGNC:4223', 'HGNC:30133', 'HGNC:3788', 'HGNC:10918', 'HGNC:2538', 'HGNC:19392', 'HGNC:29147', 'HGNC:17032', 'HGNC:32047', 'HGNC:18049', 'HGNC:3165', 'HGNC:40049', 'HGNC:19881', 'HGNC:35462', 'HGNC:23298', 'HGNC:33722', 'HGNC:2545', 'HGNC:33808', 'HGNC:19768', 'HGNC:21391', 'HGNC:22195', 'HGNC:28701', 'HGNC:21872', 'HGNC:4278', 'HGNC:3723', 'HGNC:4503', 'HGNC:6605', 'HGNC:14399', 'HGNC:3356', 'HGNC:16500', 'HGNC:49262', 'HGNC:13711', 'HGNC:17827', 'HGNC:30400', 'HGNC:30964', 'HGNC:7142', 'HGNC:24731', 'HGNC:1319', 'HGNC:4716', 'HGNC:10573', 'HGNC:3395', 'HGNC:34395', 'HGNC:17057', 'HGNC:22944', 'HGNC:7398', 'HGNC:21740', 'HGNC:17693', 'HGNC:21578', 'HGNC:21487', 'HGNC:9909', 'HGNC:24930', 'HGNC:9350', 'HGNC:9082', 'HGNC:1372', 'HGNC:24149', 'HGNC:9243', 'HGNC:2439', 'HGNC:30227', 'HGNC:13554', 'HGNC:17736', 'HGNC:2316', 'HGNC:43938', 'HGNC:3618', 'HGNC:49574', 'HGNC:9042', 'HGNC:6967', 'HGNC:14940', 'HGNC:27537', 'HGNC:10599', 'HGNC:2642', 'HGNC:6667', 'HGNC:55330', 'HGNC:31696', 'HGNC:685', 'HGNC:48995', 'HGNC:9990', 'HGNC:26249', 'HGNC:4513', 'HGNC:28859', 'HGNC:20822', 'HGNC:49497', 'HGNC:25559', 'HGNC:20148', 'HGNC:18794', 'HGNC:5972', 'HGNC:6169', 'HGNC:31255', 'HGNC:11067', 'HGNC:31399', 'HGNC:10921', 'HGNC:33758', 'HGNC:50861', 'HGNC:3760', 'HGNC:33708', 'HGNC:51517', 'HGNC:11502', 'HGNC:49576', 'HGNC:20348', 'HGNC:53437', 'HGNC:33631', 'HGNC:25162', 'HGNC:44584', 'HGNC:9075', 'HGNC:1606', 'HGNC:11914', 'HGNC:6249', 'HGNC:16276', 'HGNC:12749', 'HGNC:27900', 'HGNC:19663', 'HGNC:28250', 'HGNC:19742', 'HGNC:44146', 'HGNC:13396', 'HGNC:29805', 'HGNC:7750', 'HGNC:52447', 'HGNC:29480', 'HGNC:44159', 'HGNC:22644', 'HGNC:1980', 'HGNC:13353', 'HGNC:2991', 'HGNC:12785', 'HGNC:26976', 'HGNC:30668', 'HGNC:18856', 'HGNC:18349', 'HGNC:23015', 'HGNC:28477', 'HGNC:20448', 'HGNC:2475', 'HGNC:5986', 'HGNC:13375', 'HGNC:26574', 'HGNC:48896', 'HGNC:53149', 'HGNC:40209', 'HGNC:34500', 'HGNC:18521', 'HGNC:19176', 'HGNC:4341', 'HGNC:37442', 'HGNC:6029', 'HGNC:24957', 'HGNC:10659', 'HGNC:33510', 'HGNC:5188', 'HGNC:8978', 'HGNC:8194', 'HGNC:54184', 'HGNC:4408', 'HGNC:29326', 'HGNC:30435', 'HGNC:37210', 'HGNC:28387', 'HGNC:33846', 'HGNC:40499', 'HGNC:2940', 'HGNC:1844', 'HGNC:18414', 'HGNC:2309', 'HGNC:16668', 'HGNC:27069', 'HGNC:14346', 'HGNC:55411', 'HGNC:44358', 'HGNC:3311', 'HGNC:51381', 'HGNC:27949', 'HGNC:9020', 'HGNC:20867', 'HGNC:9037', 'HGNC:24820', 'HGNC:24550', 'HGNC:52753', 'HGNC:1602', 'HGNC:8373', 'HGNC:29634', 'HGNC:1346', 'HGNC:1386', 'HGNC:25645', 'HGNC:29645', 'HGNC:51658', 'HGNC:2396', 'HGNC:53322', 'HGNC:16187', 'HGNC:15565', 'HGNC:4533', 'HGNC:48994', 'HGNC:14006', 'HGNC:49593', 'HGNC:55341', 'HGNC:3070', 'HGNC:4234', 'HGNC:26766', 'HGNC:9408', 'HGNC:2926', 'HGNC:22788', 'HGNC:28705', 'HGNC:30620', 'HGNC:29673', 'HGNC:30084', 'HGNC:3612', 'HGNC:6515', 'HGNC:52866', 'HGNC:17466', 'HGNC:6463', 'HGNC:11329', 'HGNC:16837', 'HGNC:15710', 'HGNC:3619', 'HGNC:1529', 'HGNC:1247', 'HGNC:16701', 'HGNC:1095', 'HGNC:16270', 'HGNC:13771', 'HGNC:1603', 'HGNC:206', 'HGNC:19129', 'HGNC:24318', 'HGNC:15495', 'HGNC:11087', 'HGNC:4374', 'HGNC:32351', 'HGNC:30743', 'HGNC:30323', 'HGNC:25181', 'HGNC:16797', 'HGNC:2397', 'HGNC:2334', 'HGNC:51838', 'HGNC:14558', 'HGNC:1708', 'HGNC:21199', 'HGNC:6448', 'HGNC:23693', 'HGNC:19161', 'HGNC:2771', 'HGNC:17494', 'HGNC:31539', 'HGNC:13232', 'HGNC:1610', 'HGNC:2548', 'HGNC:15859', 'HGNC:26918', 'HGNC:19351', 'HGNC:7727', 'HGNC:259', 'HGNC:11853', 'HGNC:13009', 'HGNC:12968', 'HGNC:18130', 'HGNC:12646', 'HGNC:17056', 'HGNC:15632', 'HGNC:27723', 'HGNC:28297', 'HGNC:6932', 'HGNC:31686', 'HGNC:2255', 'HGNC:28496', 'HGNC:30702', 'HGNC:4838', 'HGNC:19356', 'HGNC:27540', 'HGNC:2257', 'HGNC:52452', 'HGNC:1022', 'HGNC:15600', 'HGNC:20637', 'HGNC:23305', 'HGNC:5999', 'HGNC:23427', 'HGNC:6649', 'HGNC:30138', 'HGNC:7151', 'HGNC:264', 'HGNC:12578', 'HGNC:23399', 'HGNC:23730', 'HGNC:2355', 'HGNC:5171', 'HGNC:407', 'HGNC:9775', 'HGNC:25034', 'HGNC:19125', 'HGNC:33584', 'HGNC:8945', 'HGNC:9873', 'HGNC:31776', 'HGNC:4091', 'HGNC:3695', 'HGNC:6179', 'HGNC:7060', 'HGNC:23721', 'HGNC:7146', 'HGNC:6067', 'HGNC:20120', 'HGNC:15631', 'HGNC:1965', 'HGNC:1505', 'HGNC:5362', 'HGNC:13363', 'HGNC:52678', 'HGNC:26129', 'HGNC:41204', 'HGNC:26260', 'HGNC:53508', 'HGNC:12412', 'HGNC:11065', 'HGNC:2699', 'HGNC:16608', 'HGNC:24936', 'HGNC:28354', 'HGNC:10737', 'HGNC:11760', 'HGNC:18515', 'HGNC:14272', 'HGNC:53465', 'HGNC:11850', 'HGNC:10732', 'HGNC:28321', 'HGNC:19350', 'HGNC:18150', 'HGNC:1976', 'HGNC:33849', 'HGNC:16885', 'HGNC:32327', 'HGNC:30238', 'HGNC:11219', 'HGNC:30282', 'HGNC:10725', 'HGNC:31098', 'HGNC:13444', 'HGNC:8672', 'HGNC:16438', 'HGNC:28344', 'HGNC:11012', 'HGNC:11085', 'HGNC:9403', 'HGNC:18126', 'HGNC:23581', 'HGNC:19319', 'HGNC:20122', 'HGNC:18464', 'HGNC:30963', 'HGNC:23336', 'HGNC:6895', 'HGNC:53141', 'HGNC:30198', 'HGNC:26634', 'HGNC:1611', 'HGNC:1224', 'HGNC:4844', 'HGNC:30827', 'HGNC:7399', 'HGNC:11069', 'HGNC:44152', 'HGNC:4086', 'HGNC:21174', 'HGNC:32443', 'HGNC:3793', 'HGNC:23429', 'HGNC:37272', 'HGNC:4626', 'HGNC:2479', 'HGNC:7882', 'HGNC:5713', 'HGNC:30670', 'HGNC:13378', 'HGNC:32071', 'HGNC:31859', 'HGNC:6357', 'HGNC:3496', 'HGNC:20776', 'HGNC:2959', 'HGNC:23438', 'HGNC:9076', 'HGNC:19315', 'HGNC:24029', 'HGNC:19747', 'HGNC:24893', 'HGNC:14458', 'HGNC:44689', 'HGNC:29596', 'HGNC:34451', 'HGNC:25667', 'HGNC:973', 'HGNC:25252', 'HGNC:483', 'HGNC:26741', 'HGNC:43711', 'HGNC:15937', 'HGNC:19254', 'HGNC:10017', 'HGNC:18296', 'HGNC:12705', 'HGNC:42784', 'HGNC:30830', 'HGNC:20990', 'HGNC:49147', 'HGNC:6308', 'HGNC:7600', 'HGNC:10351', 'HGNC:17493', 'HGNC:13229', 'HGNC:24739', 'HGNC:9762', 'HGNC:23953', 'HGNC:20322', 'HGNC:27081', 'HGNC:33107', 'HGNC:29362', 'HGNC:9592', 'HGNC:13251', 'HGNC:51899', 'HGNC:800', 'HGNC:9441', 'HGNC:1820', 'HGNC:25417', 'HGNC:13138', 'HGNC:8918', 'HGNC:23494', 'HGNC:13307', 'HGNC:29335', 'HGNC:39758', 'HGNC:53474', 'HGNC:40702', 'HGNC:5988', 'HGNC:31745', 'HGNC:33143', 'HGNC:27266', 'HGNC:31713', 'HGNC:21158', 'HGNC:32991', 'HGNC:10902', 'HGNC:20892', 'HGNC:9900', 'HGNC:53084', 'HGNC:26281', 'HGNC:17824', 'HGNC:13805', 'HGNC:8673', 'HGNC:28764', 'HGNC:124', 'HGNC:17890', 'HGNC:3420', 'HGNC:29393', 'HGNC:34303', 'HGNC:17087', 'HGNC:5412', 'HGNC:20757', 'HGNC:26257', 'HGNC:18396', 'HGNC:26795', 'HGNC:21209', 'HGNC:14268', 'HGNC:7376', 'HGNC:11177', 'HGNC:50602', 'HGNC:49257', 'HGNC:31886', 'HGNC:20369', 'HGNC:26042', 'HGNC:7661', 'HGNC:27049', 'HGNC:51822', 'HGNC:20081', 'HGNC:5993', 'HGNC:26444', 'HGNC:26735', 'HGNC:436', 'HGNC:28721', 'HGNC:33769', 'HGNC:11979', 'HGNC:30652', 'HGNC:30811', 'HGNC:16112', 'HGNC:16175', 'HGNC:29370', 'HGNC:21375', 'HGNC:16274', 'HGNC:16644', 'HGNC:26579', 'HGNC:13202', 'HGNC:19770', 'HGNC:6015', 'HGNC:29268', 'HGNC:11059', 'HGNC:13618', 'HGNC:1397', 'HGNC:20854', 'HGNC:40374', 'HGNC:54083', 'HGNC:25240', 'HGNC:3697', 'HGNC:32926', 'HGNC:44655', 'HGNC:4528', 'HGNC:6696', 'HGNC:7043', 'HGNC:7217', 'HGNC:33749', 'HGNC:50581', 'HGNC:1040', 'HGNC:4565', 'HGNC:10998', 'HGNC:27584', 'HGNC:44670', 'HGNC:21495', 'HGNC:15520', 'HGNC:3621', 'HGNC:11593', 'HGNC:16394', 'HGNC:16916', 'HGNC:637', 'HGNC:30968', 'HGNC:26587', 'HGNC:30491', 'HGNC:21785', 'HGNC:10586', 'HGNC:24132', 'HGNC:2988', 'HGNC:34499', 'HGNC:40051', 'HGNC:21233', 'HGNC:10944', 'HGNC:16008', 'HGNC:9868', 'HGNC:30365', 'HGNC:29573', 'HGNC:19189', 'HGNC:6715', 'HGNC:26009', 'HGNC:1913', 'HGNC:9725', 'HGNC:3625', 'HGNC:26633', 'HGNC:28327', 'HGNC:11829', 'HGNC:40401', 'HGNC:11851', 'HGNC:34513', 'HGNC:25194', 'HGNC:10966', 'HGNC:10012', 'HGNC:28462', 'HGNC:17819', 'HGNC:18731', 'HGNC:4088', 'HGNC:10867', 'HGNC:24104', 'HGNC:3959', 'HGNC:16494', 'HGNC:3623', 'HGNC:28447', 'HGNC:21115', 'HGNC:6607', 'HGNC:10593', 'HGNC:10274', 'HGNC:27257', 'HGNC:54068', 'HGNC:16051', 'HGNC:32403', 'HGNC:17661', 'HGNC:12261', 'HGNC:14605', 'HGNC:618', 'HGNC:44327', 'HGNC:3801', 'HGNC:6487', 'HGNC:4376', 'HGNC:52851', 'HGNC:7660', 'HGNC:39886', 'HGNC:13670', 'HGNC:20917', 'HGNC:17761', 'HGNC:18877', 'HGNC:33196', 'HGNC:1478', 'HGNC:19127', 'HGNC:8421', 'HGNC:11898', 'HGNC:9875', 'HGNC:18655', 'HGNC:10877', 'HGNC:11847', 'HGNC:25144', 'HGNC:26353', 'HGNC:21444', 'HGNC:55311', 'HGNC:7321', 'HGNC:32958', 'HGNC:20675', 'HGNC:303', 'HGNC:8946', 'HGNC:7402', 'HGNC:23536', 'HGNC:25236', 'HGNC:48910', 'HGNC:44151', 'HGNC:5345', 'HGNC:26131', 'HGNC:24766', 'HGNC:50329', 'HGNC:27254', 'HGNC:3978', 'HGNC:23696', 'HGNC:49754', 'HGNC:24723', 'HGNC:2644', 'HGNC:44256', 'HGNC:4848', 'HGNC:25857', 'HGNC:26430', 'HGNC:3336', 'HGNC:10589', 'HGNC:5394', 'HGNC:5331', 'HGNC:17343', 'HGNC:44378', 'HGNC:24525', 'HGNC:9410', 'HGNC:6799', 'HGNC:54720', 'HGNC:11568', 'HGNC:52824', 'HGNC:21553', 'HGNC:7371', 'HGNC:20096', 'HGNC:1072', 'HGNC:44371', 'HGNC:11693', 'HGNC:8861', 'HGNC:8134', 'HGNC:14449', 'HGNC:32422', 'HGNC:13797', 'HGNC:3497', 'HGNC:30459', 'HGNC:1815', 'HGNC:21214', 'HGNC:1197', 'HGNC:18345', 'HGNC:13358', 'HGNC:13841', 'HGNC:11131', 'HGNC:42937', 'HGNC:12706', 'HGNC:21969', 'HGNC:23672', 'HGNC:3401', 'HGNC:19987', 'HGNC:49266', 'HGNC:18113', 'HGNC:922', 'HGNC:40308', 'HGNC:17354', 'HGNC:3774', 'HGNC:2356', 'HGNC:24118', 'HGNC:23109', 'HGNC:14404', 'HGNC:28240', 'HGNC:13410', 'HGNC:20474', 'HGNC:14932', 'HGNC:16144', 'HGNC:28553', 'HGNC:29514', 'HGNC:4747', 'HGNC:14871', 'HGNC:48971', 'HGNC:6158', 'HGNC:29571', 'HGNC:20240', 'HGNC:500', 'HGNC:10045', 'HGNC:31455', 'HGNC:11127', 'HGNC:13900', 'HGNC:25161', 'HGNC:25802', 'HGNC:11530', 'HGNC:17156', 'HGNC:1245', 'HGNC:3538', 'HGNC:49383', 'HGNC:14467', 'HGNC:7135', 'HGNC:1470', 'HGNC:3827', 'HGNC:3555', 'HGNC:20860', 'HGNC:24490', 'HGNC:4883', 'HGNC:24670', 'HGNC:55416', 'HGNC:11779', 'HGNC:32389', 'HGNC:31597', 'HGNC:26306', 'HGNC:28016', 'HGNC:1704', 'HGNC:6477', 'HGNC:248', 'HGNC:28692', 'HGNC:6134', 'HGNC:8676', 'HGNC:33482', 'HGNC:12449', 'HGNC:1077', 'HGNC:3531', 'HGNC:21392', 'HGNC:15718', 'HGNC:11322', 'HGNC:7404', 'HGNC:16834', 'HGNC:2176', 'HGNC:27291', 'HGNC:26545', 'HGNC:1348', 'HGNC:52924', 'HGNC:52980', 'HGNC:15805', 'HGNC:10878', 'HGNC:4490', 'HGNC:6570', 'HGNC:4707', 'HGNC:13639', 'HGNC:26102', 'HGNC:26586', 'HGNC:9031', 'HGNC:1546', 'HGNC:43944', 'HGNC:9778', 'HGNC:23508', 'HGNC:18757', 'HGNC:21144', 'HGNC:278', 'HGNC:26348', 'HGNC:54402', 'HGNC:27884', 'HGNC:30012', 'HGNC:1662', 'HGNC:32960', 'HGNC:13770', 'HGNC:30259', 'HGNC:13741', 'HGNC:490', 'HGNC:17468', 'HGNC:18411', 'HGNC:1666', 'HGNC:2338', 'HGNC:269', 'HGNC:7434', 'HGNC:26082', 'HGNC:5994', 'HGNC:12694', 'HGNC:2218', 'HGNC:44297', 'HGNC:3128', 'HGNC:29123', 'HGNC:4516', 'HGNC:27867', 'HGNC:11060', 'HGNC:13427', 'HGNC:17198', 'HGNC:3500', 'HGNC:25835', 'HGNC:12647', 'HGNC:30654', 'HGNC:53220', 'HGNC:12919', 'HGNC:21197', 'HGNC:22204', 'HGNC:21649', 'HGNC:10876', 'HGNC:8516', 'HGNC:23689', 'HGNC:30390', 'HGNC:23723', 'HGNC:3691', 'HGNC:21', 'HGNC:52213', 'HGNC:8614', 'HGNC:10903', 'HGNC:4497', 'HGNC:20707', 'HGNC:27172', 'HGNC:2577', 'HGNC:31536', 'HGNC:20222', 'HGNC:16781', 'HGNC:13371', 'HGNC:37192', 'HGNC:26617', 'HGNC:52425', 'HGNC:11605', 'HGNC:4925', 'HGNC:18772', 'HGNC:12841', 'HGNC:26215', 'HGNC:24139', 'HGNC:9357', 'HGNC:29604', 'HGNC:544', 'HGNC:13587', 'HGNC:6135', 'HGNC:30650', 'HGNC:11909', 'HGNC:35240', 'HGNC:6010', 'HGNC:26683', 'HGNC:2157', 'HGNC:23405', 'HGNC:42813', 'HGNC:15620', 'HGNC:725', 'HGNC:34073', 'HGNC:26256', 'HGNC:27388', 'HGNC:17449', 'HGNC:50277', 'HGNC:7662', 'HGNC:28276', 'HGNC:2158', 'HGNC:26402', 'HGNC:16243', 'HGNC:40340', 'HGNC:9038', 'HGNC:21695', 'HGNC:16423', 'HGNC:6026', 'HGNC:7723', 'HGNC:3037', 'HGNC:30451', 'HGNC:3557', 'HGNC:18149', 'HGNC:26791', 'HGNC:44166', 'HGNC:51334', 'HGNC:3616', 'HGNC:35316', 'HGNC:5413', 'HGNC:14908', 'HGNC:6167', 'HGNC:25480', 'HGNC:23791', 'HGNC:23495', 'HGNC:26999', 'HGNC:14439', 'HGNC:4036', 'HGNC:2705', 'HGNC:4530', 'HGNC:51152', 'HGNC:24702', 'HGNC:6783', 'HGNC:34496', 'HGNC:18742', 'HGNC:17020', 'HGNC:18553', 'HGNC:30034', 'HGNC:7598', 'HGNC:26810', 'HGNC:1241', 'HGNC:20603', 'HGNC:13939', 'HGNC:28867', 'HGNC:11916', 'HGNC:35167', 'HGNC:33748', 'HGNC:30456', 'HGNC:9998', 'HGNC:26596', 'HGNC:28065', 'HGNC:3826', 'HGNC:1076', 'HGNC:33257', 'HGNC:963', 'HGNC:51375', 'HGNC:10720', 'HGNC:2578', 'HGNC:8736', 'HGNC:30449', 'HGNC:29025', 'HGNC:37238', 'HGNC:679', 'HGNC:682', 'HGNC:7175', 'HGNC:28514', 'HGNC:2939', 'HGNC:21356', 'HGNC:29915', 'HGNC:29269', 'HGNC:11253', 'HGNC:31658', 'HGNC:27013', 'HGNC:2942', 'HGNC:43436', 'HGNC:3154', 'HGNC:6460', 'HGNC:23677', 'HGNC:43713', 'HGNC:27080', 'HGNC:30455', 'HGNC:9942', 'HGNC:20774', 'HGNC:48661', 'HGNC:1452', 'HGNC:10487', 'HGNC:28288', 'HGNC:14261', 'HGNC:991', 'HGNC:28706', 'HGNC:11257', 'HGNC:1570', 'HGNC:53776', 'HGNC:10728', 'HGNC:33763', 'HGNC:21285', 'HGNC:29399', 'HGNC:29936', 'HGNC:4833', 'HGNC:4511', 'HGNC:42780', 'HGNC:20259', 'HGNC:20362', 'HGNC:26645', 'HGNC:13728', 'HGNC:10931', 'HGNC:21226', 'HGNC:9506', 'HGNC:25743', 'HGNC:32416', 'HGNC:25697', 'HGNC:4047', 'HGNC:25500', 'HGNC:13536', 'HGNC:1037', 'HGNC:18653', 'HGNC:11089', 'HGNC:26888', 'HGNC:29678', 'HGNC:14639', 'HGNC:10953', 'HGNC:11038', 'HGNC:4604', 'HGNC:29298', 'HGNC:537', 'HGNC:7014', 'HGNC:21755', 'HGNC:17845', 'HGNC:477', 'HGNC:10489', 'HGNC:8692', 'HGNC:50677', 'HGNC:26978', 'HGNC:44265', 'HGNC:10941', 'HGNC:21685', 'HGNC:19178', 'HGNC:26957', 'HGNC:1571', 'HGNC:15716', 'HGNC:25316', 'HGNC:6609', 'HGNC:16397', 'HGNC:8368', 'HGNC:29255', 'HGNC:35279', 'HGNC:1183', 'HGNC:25833', 'HGNC:33756', 'HGNC:37261', 'HGNC:29887', 'HGNC:24099', 'HGNC:53167', 'HGNC:21179', 'HGNC:31108', 'HGNC:27631', 'HGNC:14376', 'HGNC:32222', 'HGNC:52624', 'HGNC:26714', 'HGNC:34346', 'HGNC:28133', 'HGNC:8604', 'HGNC:18601', 'HGNC:329', 'HGNC:815', 'HGNC:13524', 'HGNC:7027', 'HGNC:9666', 'HGNC:15874', 'HGNC:21367', 'HGNC:3795', 'HGNC:2001', 'HGNC:6606', 'HGNC:37454', 'HGNC:14254', 'HGNC:19262', 'HGNC:14051', 'HGNC:30072', 'HGNC:30125', 'HGNC:17416', 'HGNC:11858', 'HGNC:3333', 'HGNC:26707', 'HGNC:51388', 'HGNC:54819', 'HGNC:29806', 'HGNC:19084', 'HGNC:14668', 'HGNC:23520', 'HGNC:10719', 'HGNC:53079', 'HGNC:49384', 'HGNC:30927', 'HGNC:21789', 'HGNC:402', 'HGNC:17972', 'HGNC:17337', 'HGNC:33160', 'HGNC:25760', 'HGNC:2855', 'HGNC:4554', 'HGNC:250', 'HGNC:14654', 'HGNC:33718', 'HGNC:45010', 'HGNC:1228', 'HGNC:54065', 'HGNC:7821', 'HGNC:18809', 'HGNC:1305', 'HGNC:30586', 'HGNC:28662', 'HGNC:533', 'HGNC:53947', 'HGNC:8126'}\n",
      "Regulation nodes not in the graph: 10859\n",
      "\n",
      "Saving final graph...\n",
      "(19721, 6)\n",
      "Index(['id', 'semantic_groups', 'preflabel', 'synonyms', 'name',\n",
      "       'description'],\n",
      "      dtype='object')\n",
      "\n",
      "* This is the size of the edges file data structure: (19721, 6)\n",
      "* These are the edges attributes: Index(['id', 'semantic_groups', 'preflabel', 'synonyms', 'name',\n",
      "       'description'],\n",
      "      dtype='object')\n",
      "* This is the first record:\n",
      "    id semantic_groups preflabel synonyms name description\n",
      "0  NaN             NaN       NaN      NaN  NaN         NaN\n",
      "\n",
      "The NGLY1 Deficiency knowledge graph nodes are built and saved at: /home/karolis/LUMC/HDSR/bioknowledge-reviewer/bioknowledge_reviewer/graph/graph_nodes_v2021-04-12.csv\n",
      "\n",
      "\n",
      "Finished build_nodes().\n",
      "\n",
      "CPU times: user 36min 15s, sys: 2min, total: 38min 16s\n",
      "Wall time: 1d 5h 2min 52s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "# load networks and calculate graph nodes\n",
    "graph_nodes_list, reg_graph_edges = graph.graph_nodes(\n",
    "    curation=curation_edges,\n",
    "    monarch=monarch_edges,\n",
    "    transcriptomics=rna_edges,\n",
    "    regulation=reg_edges\n",
    ")\n",
    "\n",
    "# Monarch graph connectivity\n",
    "## get Monarch edges\n",
    "monarch_network_graph = monarch.extract_edges(graph_nodes_list)\n",
    "print('network: ',len(monarch_network_graph))\n",
    "\n",
    "## save Monarch network\n",
    "monarch.print_network(monarch_network_graph, 'monarch_connections_graph')\n",
    "\n",
    "## build Monarch network with graph schema\n",
    "monarch_graph_edges = monarch.build_edges(monarch_network_graph)\n",
    "monarch_graph_nodes = monarch.build_nodes(monarch_network_graph)\n",
    "\n",
    "# build review graph\n",
    "edges = graph.build_edges(\n",
    "    curation=curation_edges,\n",
    "    monarch=monarch_graph_edges,\n",
    "    transcriptomics=rna_edges,\n",
    "    regulation=reg_graph_edges\n",
    ")\n",
    "nodes = graph.build_nodes(\n",
    "    statements=edges,\n",
    "    curation=curation_nodes,\n",
    "    monarch=monarch_graph_nodes,\n",
    "    transcriptomics=rna_nodes,\n",
    "    regulation=reg_nodes\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'reg_graph_edges' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-21-a6121cbfb742>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mreg_graph_edges\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'reg_graph_edges' is not defined"
     ]
    }
   ],
   "source": [
    "reg_graph_edges"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "edges_df = pd.read_csv(\"./graph/graph_edges_v2021-04-12_alt_del.csv\")\n",
    "nodes_df = pd.read_csv(\"./graph/graph_nodes_v2021-04-12.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Creating a Neo4j community v4.2.1 server instance...\n",
      "Preparing the server...\n",
      "Configuration adjusted!\n",
      "Starting the server...\n",
      "Neo4j v4.2.1 is running.\n",
      "The name of the neo4j directory is neo4j-community-4.2.1\n",
      "statements:  593222\n",
      "concepts:  19721\n",
      "\n",
      "File './neo4j-community-4.2.1/import/ngly1/ngly1_statements.csv' saved.\n",
      "\n",
      "File './neo4j-community-4.2.1/import/ngly1/ngly1_concepts.csv' saved.\n",
      "\n",
      "The function \"do_import()\" is running...\n",
      "\n",
      "The graph is imported into the server. Neo4j is running.You can start exploring and querying for hypothesis. If you change ports or authentication in the Neo4j configuration file, the hypothesis-generation modules performance, hypothesis.py and summary.py, will be affected.\n",
      "\n",
      "CPU times: user 4.22 s, sys: 179 ms, total: 4.4 s\n",
      "Wall time: 18.5 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "# create a Neo4j server instance\n",
    "neo4j_dir = neo4jlib.create_neo4j_instance('4.2.1')\n",
    "print('The name of the neo4j directory is {}'.format(neo4j_dir))\n",
    "\n",
    "# import to graph database\n",
    "## prepare the graph to neo4j format\n",
    "#edges_df = utils.get_dataframe(edges)\n",
    "#nodes_df = utils.get_dataframe(nodes)\n",
    "statements = neo4jlib.get_statements(edges_df)\n",
    "concepts = neo4jlib.get_concepts(nodes_df)\n",
    "print('statements: ', len(statements))\n",
    "print('concepts: ',len(concepts))\n",
    "\n",
    "## save files into neo4j import dir\n",
    "neo4j_path = './{}'.format(neo4j_dir)\n",
    "neo4jlib.save_neo4j_files(statements, neo4j_path, file_type = 'statements')\n",
    "neo4jlib.save_neo4j_files(concepts, neo4j_path, file_type = 'concepts')\n",
    "\n",
    "## import graph into neo4j database\n",
    "neo4jlib.do_import(neo4j_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# print type of objects\n",
    "print('type edges:', type(statements))\n",
    "print('type nodes:', type(concepts))\n",
    "print()\n",
    "\n",
    "# print objects sizes\n",
    "print('len edges:', len(statements))\n",
    "print('len nodes:', len(concepts))\n",
    "print()\n",
    "\n",
    "# print object attribute\n",
    "print('attribute edges:', statements.columns)\n",
    "print('attribute nodes:', concepts.columns)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## hypothesis-generation library\n",
    "### Query the graph for mechanistic explanation, then summarize the extracted paths\n",
    "#### import hypothesis, summary\n",
    "\n",
    "\n",
    "Tasks:\n",
    "\n",
    "* Retrieve orthopheno paths with the `query` method.\n",
    "* Retrieve orthopheno paths using relaxing node degree parameters with the `query` method.\n",
    "* Retrieve orthopheno paths from a more open query topology with the `open_query` method.\n",
    "* Get hypothesis summaries"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Ortopheno query with general nodes/relations removed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "ename": "ModuleNotFoundError",
     "evalue": "No module named 'neo4j'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mModuleNotFoundError\u001b[0m                       Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-3-c396c590e696>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0;32mfrom\u001b[0m \u001b[0mneo4j\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mGraphDatabase\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0msys\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mos\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mjson\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0myaml\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mdatetime\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mModuleNotFoundError\u001b[0m: No module named 'neo4j'"
     ]
    }
   ],
   "source": [
    "from neo4j import GraphDatabase\n",
    "import sys,os\n",
    "import json\n",
    "import yaml\n",
    "import datetime\n",
    "import neo4j.exceptions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def parse_path( path ):\n",
    "    \"\"\"\n",
    "    This function parses neo4j results.\n",
    "    :param path: neo4j path object\n",
    "    :return: parsed path dictionary\n",
    "    \"\"\"\n",
    "\n",
    "    out = {}\n",
    "    out['Nodes'] = []\n",
    "    for node in path['path'].nodes:\n",
    "        n = {}\n",
    "        n['idx'] = node.id\n",
    "        n['label'] = list(node.labels)[0]\n",
    "        n['id'] = node.get('id')\n",
    "        n['preflabel'] = node.get('preflabel')\n",
    "        n['name'] = node.get('name')\n",
    "        n['description'] = node.get('description')\n",
    "        out['Nodes'].append(n)\n",
    "    out['Edges'] = []\n",
    "    for edge in path['path'].relationships:\n",
    "        e = {}\n",
    "        e['idx'] = edge.id\n",
    "        e['start_node'] = edge.start_node.id\n",
    "        e['end_node'] = edge.end_node.id\n",
    "        e['type'] = edge.type\n",
    "        e['preflabel'] = edge.get('property_label')\n",
    "        e['references'] = edge.get('reference_uri')\n",
    "        out['Edges'].append(e)\n",
    "    return out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_node(source, target, port='7687'):\n",
    "    \"\"\"\n",
    "    This function checks if a node exists within the graph\n",
    "    \"\"\"\n",
    "    try:\n",
    "        driver = GraphDatabase.driver(\"bolt://localhost:{}\".format(port), auth=(\"neo4j\", \"ngly1\"))\n",
    "    except neo4j.exceptions.ServiceUnavailable:\n",
    "        raise\n",
    "    outputAll = list()\n",
    "    with driver.session() as session:\n",
    "        query = \"\"\"MATCH (source { id: '\"\"\" + source + \"\"\"' }), (target { id: '\"\"\" + target +  \"\"\"' }), p = allShortestPaths((source)-[*..15]-(target)) RETURN p\"\"\"\n",
    "        result = session.run(query)\n",
    "        print(result)\n",
    "        x = []\n",
    "        for record in result:\n",
    "            #path_dct = parse_path(record)\n",
    "            x.append(record)\n",
    "        print(x)\n",
    "        return x, result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "x, result = get_node('HGNC:4851', 'HGNC:18229')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "x[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "result.consume()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def query_shortest_paths(source, target, max_length=4, port='7687'):\n",
    "    \"\"\"\n",
    "    This function gets the shortest paths between source and\n",
    "    target. max_length determines the maximum path size allowed.\n",
    "    \"\"\"\n",
    "    try:\n",
    "        driver = GraphDatabase.driver(\"bolt://localhost:{}\".format(port), auth=(\"neo4j\", \"ngly1\"))\n",
    "    except neo4j.exceptions.ServiceUnavailable:\n",
    "        raise\n",
    "    outputAll = list()\n",
    "    with driver.session() as session:\n",
    "        print(\"kaas\")\n",
    "        query = \"\"\"MATCH (source { id: '\"\"\" + source + \"\"\"' }), (target { id: '\"\"\" + target +  \"\"\"' }), p = shortestPath((source)-[*..15]-(target)) RETURN p\"\"\"\n",
    "        result = session.run(query)\n",
    "        print(result)\n",
    "        return result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "seed = list([\n",
    "        'HGNC:4851',  # Htt human gene\n",
    "        'HGNC:18229'  # Rhes human gene\n",
    "])\n",
    "result = query_shortest_paths(seed[0], seed[1], max_length=999)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "summary = result.consume()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "summary.query"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "result.data()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%time\n",
    "# get orthopheno paths\n",
    "seed = list([\n",
    "        'HGNC:4851',  # Htt human gene\n",
    "        'HGNC:18229'  # Rhes human gene\n",
    "])\n",
    "hypothesis.query(seed,queryname='Htt_Rhes',port='7687') "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%time\n",
    "# get orthopheno paths relaxing pathway and phenotype node degrees\n",
    "seed = list([\n",
    "        'HGNC:4851',  # NGLY1 human gene\n",
    "        'HGNC:18229'  # AQP1 human gene\n",
    "])\n",
    "hypothesis.query(seed, queryname='Htt_Rhes', pwdegree='1000', phdegree='1000', port='7687')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%time\n",
    "# get orthopheno paths from a more open query topogology\n",
    "seed = list([\n",
    "        'HGNC:4851',  # NGLY1 human gene\n",
    "        'HGNC:18229'  # AQP1 human gene\n",
    "])\n",
    "hypothesis.open_query(seed,queryname='Htt_Rhes',port='7687')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import hypothesis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%time\n",
    "seed = list([\n",
    "        'HGNC:4851',  # NGLY1 human gene\n",
    "        'HGNC:18229'  # AQP1 human gene\n",
    "])\n",
    "r = hypothesis.shortest_paths(seed[0], seed[1], max_length=50, port='7687')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(r)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "%%time\n",
    "# get summary\n",
    "data = summary.path_load('./hypothesis/query_ngly1_aqp1_pwdl1000_phdl1000_paths_v2020-09-14.json')\n",
    "\n",
    "# parse data for summarization\n",
    "data_parsed = list()\n",
    "for query in data:\n",
    "    query_parsed = summary.query_parser(query)\n",
    "    data_parsed.append(query_parsed)\n",
    "summary.metapaths(data_parsed)\n",
    "summary.nodes(data_parsed)\n",
    "summary.node_types(data_parsed)\n",
    "summary.edges(data_parsed)\n",
    "summary.edge_types(data_parsed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
